<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-04-28" />

<title>Compare different settings: cs index filter or not</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Compare different settings: cs index filter
or not</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-04-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed9f4daaee8ab0a2d363e29c8cd4cdf639049dbb9targetblankd9f4daaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d9f4daaee8ab0a2d363e29c8cd4cdf639049dbb9" target="_blank">d9f4daa</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed9f4daaee8ab0a2d363e29c8cd4cdf639049dbb9targetblankd9f4daaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d9f4daaee8ab0a2d363e29c8cd4cdf639049dbb9" target="_blank">d9f4daa</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    cv/

Unstaged changes:
    Modified:   analysis/comparing_differnt_settings_csindex.Rmd
    Deleted:    slurm-30624233.out

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/comparing_differnt_settings_csindex.Rmd</code>) and HTML
(<code>docs/comparing_differnt_settings_csindex.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/3b4edbeb0ca45598313a293861a9d653315fb3e2/analysis/comparing_differnt_settings_csindex.Rmd" target="_blank">3b4edbe</a>
</td>
<td>
XSun
</td>
<td>
2025-04-28
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/3b4edbeb0ca45598313a293861a9d653315fb3e2/docs/comparing_differnt_settings_csindex.html" target="_blank">3b4edbe</a>
</td>
<td>
XSun
</td>
<td>
2025-04-28
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(scales)
library(tidyr)
library(ggplot2)
library(eulerr)

traits_silver &lt;- c(&quot;T2D&quot;,&quot;LDL&quot;,&quot;BMI&quot;,&quot;RBC&quot;,&quot;IBD&quot;,&quot;SCZ&quot;,&quot;aFib&quot;)
names(traits_silver) &lt;- c(&quot;T2D-panukb&quot;,&quot;LDL-ukb-d-30780_irnt&quot;,&quot;BMI-panukb&quot;,&quot;RBC-panukb&quot;,&quot;IBD-ebi-a-GCST004131&quot;,&quot;SCZ-ieu-b-5102&quot;,&quot;aFib-ebi-a-GCST006414&quot;)

traits &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;IBD-ebi-a-GCST004131&quot;,&quot;BMI-panukb&quot;,&quot;RBC-panukb&quot;,&quot;SCZ-ieu-b-5102&quot;,&quot;aFib-ebi-a-GCST006414&quot;,&quot;T2D-panukb&quot;)
db &lt;- &quot;GO_Biological_Process_2023&quot;

folder_results &lt;- &quot;/project/xinhe/xsun/multi_group_ctwas/15.susie_weights/snakemake_outputs/&quot;

create_summary_plot_withTP &lt;- function(df, columns_to_plot, x_var = &quot;setting&quot;, x_order = NULL, title = NULL) {

  # Reshape data
  df_long &lt;- df %&gt;%
    pivot_longer(
      cols = all_of(columns_to_plot),
      names_to = &quot;variable&quot;,
      values_to = &quot;value&quot;
    )
  
  # Convert to factor with specified order if x_order is provided
  if (!is.null(x_order)) {
    df_long &lt;- df_long %&gt;%
      mutate(across(all_of(x_var), ~factor(., levels = x_order)))
  }
  
  # Identify the max value for scaling
  max_main &lt;- max(df_long$value[df_long$variable != &quot;TP_rate&quot;], na.rm = TRUE)
  max_tp_rate &lt;- max(df_long$value[df_long$variable == &quot;TP_rate&quot;], na.rm = TRUE)
  
  if(max_tp_rate ==0) {
    max_tp_rate &lt;- 1
  }
  
  # Rescale TP_rate
  df_long &lt;- df_long %&gt;%
    mutate(scaled_value = ifelse(variable == &quot;TP_rate&quot;, 
                                 value * (max_main / max_tp_rate), value))
  
  # Create plot
  ggplot(df_long, aes(x = .data[[x_var]], y = scaled_value, color = variable, shape = variable)) +
    #geom_point(size = 3, position = position_jitter(width = 0.2)) +
    geom_point(size = 3) + 
    scale_y_continuous(
      name = &quot;Count&quot;,
      sec.axis = sec_axis(~ . * (max_tp_rate / max_main), name = &quot;TP Rate&quot;)
    ) +
    labs(x = &quot;Settings&quot;, title = title) +
    theme_minimal() +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
      legend.position = &quot;right&quot;,
      legend.title = element_blank()
    ) +
    scale_color_brewer(palette = &quot;Set1&quot;)
}

DT::datatable(matrix())</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"data":[[null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>thin &lt;- 1
var_struc &lt;- &quot;shared_all&quot;
L &lt;- 5
st &lt;- &quot;with_susieST&quot;

setting_names &lt;- c()
num_gene_pip08_all &lt;- c()
num_silver_pip08_all &lt;- c()
num_bystander_pip08_all &lt;- c()
go_num_fdr005 &lt;- c()
for (trait in traits) {
  for (cs in c(&quot;CS_filtered&quot;,&quot;CS_NOT_filtered&quot;)){
    
    if(cs == &quot;CS_filtered&quot;){
      cs_setting &lt;- NULL
    }else{
      cs_setting &lt;- &quot;_csF&quot;
    }
    
    setting_names &lt;- c(setting_names, paste0(traits_silver[trait],&quot;-&quot;,cs))
    
    # num_gene_pip08 -- cs filtered
    combined_pip_by_group &lt;- readRDS(paste0(folder_results,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.combined_pip_bygroup_final&quot;,cs_setting,&quot;.RDS&quot;))
    combined_pip_sig &lt;- combined_pip_by_group[combined_pip_by_group$combined_pip &gt; 0.8,]
    
    num_gene_pip08_all &lt;- c(num_gene_pip08_all, nrow(combined_pip_sig))
    
    # silver_standard genes
    known &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/data/silverstandard/known_annotations_&quot;,traits_silver[trait],&quot;.RDS&quot;))
    bystander &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/data/silverstandard/bystanders_&quot;,traits_silver[trait],&quot;.RDS&quot;))
    
    num_silver_pip08_all &lt;- c(num_silver_pip08_all,sum(combined_pip_sig$gene_name %in% known))
    num_bystander_pip08_all &lt;- c(num_bystander_pip08_all,sum(combined_pip_sig$gene_name %in% bystander))
    
    file_go &lt;- paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_&quot;,cs,&quot;.RDS&quot;)
    if(!file.exists(file_go)){
      go_num_fdr005 &lt;- c(go_num_fdr005,0)
    }else{
      go &lt;- readRDS(paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_&quot;,cs,&quot;.RDS&quot;))
      go_num_fdr005 &lt;- c(go_num_fdr005,nrow(go))
    }
    
    
  }
}

df &lt;- data.frame(setting = setting_names, 
                 num_gene_pip08 = num_gene_pip08_all,
                 num_silver_pip08 = num_silver_pip08_all,
                 num_bystander_pip08 = num_bystander_pip08_all,
                 TP_rate = num_silver_pip08_all/(num_silver_pip08_all+num_bystander_pip08_all),
                 go_num_fdr005 = go_num_fdr005)

create_summary_plot_withTP(df,x_order = setting_names, 
                    columns_to_plot = c(&quot;num_gene_pip08&quot;,&quot;num_silver_pip08&quot;,&quot;TP_rate&quot;))</code></pre>
<p><img src="figure/comparing_differnt_settings_csindex.Rmd/unnamed-chunk-2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/3b4edbeb0ca45598313a293861a9d653315fb3e2/docs/figure/comparing_differnt_settings_csindex.Rmd/unnamed-chunk-2-1.png" target="_blank">3b4edbe</a>
</td>
<td>
XSun
</td>
<td>
2025-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>create_summary_plot_withTP(df,x_order = setting_names, 
                    columns_to_plot = c(&quot;go_num_fdr005&quot;), title = &quot;Number of GO terms at p.adjust &lt; 0.05&quot;)</code></pre>
<p><img src="figure/comparing_differnt_settings_csindex.Rmd/unnamed-chunk-2-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DT::datatable(df,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\"><\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["LDL-CS_filtered","LDL-CS_NOT_filtered","IBD-CS_filtered","IBD-CS_NOT_filtered","BMI-CS_filtered","BMI-CS_NOT_filtered","RBC-CS_filtered","RBC-CS_NOT_filtered","SCZ-CS_filtered","SCZ-CS_NOT_filtered","aFib-CS_filtered","aFib-CS_NOT_filtered","T2D-CS_filtered","T2D-CS_NOT_filtered"],[60,72,22,28,52,61,152,194,11,16,43,57,8,15],[10,10,5,5,0,0,3,3,0,0,10,10,1,1],[4,5,0,0,0,0,20,24,1,1,5,5,3,3],[0.714285714285714,0.666666666666667,1,1,null,null,0.130434782608696,0.111111111111111,0,0,0.666666666666667,0.666666666666667,0.25,0.25],[50,47,190,139,0,0,13,0,34,0,28,13,63,3]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>setting<\/th>\n      <th>num_gene_pip08<\/th>\n      <th>num_silver_pip08<\/th>\n      <th>num_bystander_pip08<\/th>\n      <th>TP_rate<\/th>\n      <th>go_num_fdr005<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>traits &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;IBD-ebi-a-GCST004131&quot;)

for (trait in traits) {
  
  file_go_filtered &lt;- paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_CS_filtered.RDS&quot;)
  if(!file.exists(file_go_filtered)){
    go_filtered &lt;- NULL
  }else{
    go_filtered &lt;- readRDS(file_go_filtered)
  }
  

  
  file_go_nofiltered &lt;- paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_CS_NOT_filtered.RDS&quot;)
  if(!file.exists(file_go_nofiltered)){
    go_nofiltered &lt;- NULL
  }else{
    go_nofiltered &lt;- readRDS(file_go_nofiltered)
  }
  
  set1 &lt;- unique(go_filtered$Term)
  set2 &lt;- unique(go_nofiltered$Term)
  
  venn_input &lt;- list(
    Filtered = set1,
    NoFiltered = set2
  )
  
  fit &lt;- euler(venn_input)
  
  print(plot(fit,
       fills = c(&quot;skyblue&quot;, &quot;orange&quot;),
       labels = TRUE,
       quantities = TRUE,
       main = trait))
  
    
}</code></pre>
<p><img src="figure/comparing_differnt_settings_csindex.Rmd/unnamed-chunk-3-1.png" width="384" style="display: block; margin: auto;" /><img src="figure/comparing_differnt_settings_csindex.Rmd/unnamed-chunk-3-2.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>traits &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;IBD-ebi-a-GCST004131&quot;)

for (trait in traits) {
  
  file_go_filtered &lt;- paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_CS_filtered.RDS&quot;)
  if(!file.exists(file_go_filtered)){
    go_filtered &lt;- NULL
  }else{
    go_filtered &lt;- readRDS(file_go_filtered)
  }
  

  
  file_go_nofiltered &lt;- paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.L&quot;,L, &quot;.enrichr_&quot;,db,&quot;_CS_NOT_filtered.RDS&quot;)
  if(!file.exists(file_go_nofiltered)){
    go_nofiltered &lt;- NULL
  }else{
    go_nofiltered &lt;- readRDS(file_go_nofiltered)
  }

  
  cat(&quot;\n\n&quot;)
  cat(knitr::knit_print(DT::datatable(go_filtered[!go_filtered$Term %in% go_nofiltered$Term,], caption = htmltools::tags$caption( style = &#39;font-size: 150%; caption-side: topleft; text-align = left; color:black;&#39;,paste0(&quot;Unique GO terms at p.adjust &lt; 0.05 -- &quot;,trait, &quot;--CS_filtered&quot;)),options = list(pageLength = 5))))
  cat(&quot;\n\n&quot;)
  
  cat(&quot;\n\n&quot;)
  cat(knitr::knit_print(DT::datatable(go_nofiltered[!go_nofiltered$Term %in% go_filtered$Term,], caption = htmltools::tags$caption( style = &#39;font-size: 150%; caption-side: topleft; text-align = left; color:black;&#39;,paste0(&quot;Unique GO terms at p.adjust &lt; 0.05 -- &quot;,trait, &quot;--CS_NOT_filtered&quot;)),options = list(pageLength = 5))))
  cat(&quot;\n\n&quot;)
  
 
    
}</code></pre>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"font-size: 150%; caption-side: topleft; text-align = left; color:black;\">Unique GO terms at p.adjust &lt; 0.05 -- LDL-ukb-d-30780_irnt--CS_filtered<\/caption>","data":[["35","36","37","38","39","41","42","43","44","45","46","48","49","50"],["Response To Insulin (GO:0032868)","Positive Regulation Of Endocytosis (GO:0045807)","Regulation Of Protein Export From Nucleus (GO:0046825)","Dendritic Cell Differentiation (GO:0097028)","Positive Regulation Of Cholesterol Efflux (GO:0010875)","Secondary Alcohol Biosynthetic Process (GO:1902653)","Mononuclear Cell Differentiation (GO:1903131)","Negative Regulation Of Macromolecule Metabolic Process (GO:0010605)","Cellular Response To Peptide Hormone Stimulus (GO:0071375)","Organophosphate Ester Transport (GO:0015748)","Cholesterol Biosynthetic Process (GO:0006695)","Negative Regulation Of Receptor-Mediated Endocytosis (GO:0048261)","Positive Regulation Of Microtubule Polymerization Or Depolymerization (GO:0031112)","Regulation Of Actin Filament Bundle Assembly (GO:0032231)"],["3/79","3/79","2/22","2/22","2/22","2/24","2/24","4/186","3/89","2/25","2/25","2/26","2/26","2/26"],[0.00172623576251799,0.00172623576251799,0.00196693859693445,0.00196693859693445,0.00196693859693445,0.00234106985124172,0.00234106985124172,0.00235319177997096,0.00242697845689802,0.00253974446225944,0.00253974446225944,0.00274609634217686,0.00274609634217686,0.00274609634217686],[0.0416694132674482,0.0416694132674482,0.0438274266855394,0.0438274266855394,0.0438274266855394,0.0475563641115062,0.0475563641115062,0.0475563641115062,0.0477271544270338,0.0477271544270338,0.0477271544270338,0.0477271544270338,0.0477271544270338,0.0477271544270338],[0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0],[13.7562326869806,13.7562326869806,34.3448275862069,34.3448275862069,34.3448275862069,31.2194357366771,31.2194357366771,7.75431711145997,12.1505507955936,29.8605697151424,29.8605697151424,28.6149425287356,28.6149425287356,28.6149425287356],[87.5145675709806,87.5145675709806,214.012132718709,214.012132718709,214.012132718709,189.100719403629,189.100719403629,46.9289927615703,73.159781378704,178.437561841603,178.437561841603,168.758766470475,168.758766470475,168.758766470475],["INSIG2;PCSK9;INHBB","RAB21;HFE;GAS6","GSK3B;GAS6","FLT3;GAS6","ABCA8;PLTP","INSIG2;HMGCR","FLT3;GAS6","GSK3B;PCSK9;PARP9;LDLR","INSIG2;PCSK9;INHBB","LDLR;PLTP","INSIG2;HMGCR","PCSK9;RAC1","PSRC1;RAC1","ASAP3;RAC1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Term<\/th>\n      <th>Overlap<\/th>\n      <th>P.value<\/th>\n      <th>Adjusted.P.value<\/th>\n      <th>Old.P.value<\/th>\n      <th>Old.Adjusted.P.value<\/th>\n      <th>Odds.Ratio<\/th>\n      <th>Combined.Score<\/th>\n      <th>Genes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-93928d62269dcef60a51" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"font-size: 150%; caption-side: topleft; text-align = left; color:black;\">Unique GO terms at p.adjust &lt; 0.05 -- LDL-ukb-d-30780_irnt--CS_NOT_filtered<\/caption>","data":[["26","27","31","32","33","34","35","38","41","44","47"],["Regulation Of Cytoplasmic Transport (GO:1903649)","Regulation Of Mitochondrion Organization (GO:0010821)","Phosphorylation (GO:0016310)","Protein Modification Process (GO:0036211)","Cellular Response To Starvation (GO:0009267)","Regulation Of Cholesterol Storage (GO:0010885)","Positive Regulation Of Protein Binding (GO:0032092)","Activin Receptor Signaling Pathway (GO:0032924)","Excitatory Postsynaptic Potential (GO:0060079)","Negative Regulation Of Protein Binding (GO:0032091)","Positive Regulation Of Protein Phosphorylation (GO:0001934)"],["2/11","3/50","7/429","9/711","4/126","2/14","3/59","2/16","2/18","3/71","6/377"],[0.000688306132347803,0.000776571159573549,0.000882624308239369,0.000991903459908926,0.00110982132766907,0.00113090369603985,0.00125855549078998,0.00148437429368802,0.00188378999792587,0.00214552673996789,0.00236502378858464],[0.0258644265886078,0.026836409713089,0.0278169015854795,0.0302840525103444,0.0324968503244392,0.0324968503244392,0.0348919299100449,0.0381640443403473,0.0448893372676481,0.0476404460215596,0.0491623030095148],[0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0],[63.2349206349206,18.3913043478261,4.97783448778709,3.91249491249491,9.549662487946,47.4190476190476,15.4285714285714,40.6408163265306,35.5571428571429,12.6982097186701,4.79220779220779],[460.430964311659,131.693183637102,35.0071731761757,27.0583639938926,64.9716658353351,321.725825441771,103.02877005264,264.683962049075,223.102211034977,78.0225012659559,28.9783233278319],["LRRK2;DNAJC13","PRMT6;GSK3B;LRRK2","GSK3B;PXK;ACVR1C;PKN3;LRRK2;PRKD2;GAS6","PRMT6;GSK3B;CSNK1G3;PXK;ACVR1C;PKN3;LRRK2;PRKD2;GAS6","LRRK2;PCSK9;INHBB;FADS1","APOB;TTC39B","GSK3B;HFE;LRRK2","ACVR1C;INHBB","GSK3B;LRRK2","GSK3B;HFE;LRRK2","LRRK2;HFE;PRKD2;INHBB;RAC1;GAS6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Term<\/th>\n      <th>Overlap<\/th>\n      <th>P.value<\/th>\n      <th>Adjusted.P.value<\/th>\n      <th>Old.P.value<\/th>\n      <th>Old.Adjusted.P.value<\/th>\n      <th>Odds.Ratio<\/th>\n      <th>Combined.Score<\/th>\n      <th>Genes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-16707cd634e93674753d" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"font-size: 150%; caption-side: topleft; text-align = left; color:black;\">Unique GO terms at p.adjust &lt; 0.05 -- IBD-ebi-a-GCST004131--CS_filtered<\/caption>","data":[["113","126","127","132","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190"],["Negative Regulation Of Cell Population Proliferation (GO:0008285)","Negative Regulation Of Transcription By RNA Polymerase II (GO:0000122)","Transmembrane Receptor Protein Serine/Threonine Kinase Signaling Pathway (GO:0007178)","Positive Regulation Of DNA-templated Transcription (GO:0045893)","Negative Regulation Of Response To External Stimulus (GO:0032102)","Negative Regulation Of Platelet-Derived Growth Factor Receptor Signaling Pathway (GO:0010642)","Visual Learning (GO:0008542)","Positive Regulation Of Platelet Aggregation (GO:1901731)","Lysosomal Microautophagy (GO:0016237)","Regulation Of Type II Interferon-Mediated Signaling Pathway (GO:0060334)","Positive Regulation Of Circadian Rhythm (GO:0042753)","C-terminal Protein Amino Acid Modification (GO:0018410)","Eye Photoreceptor Cell Differentiation (GO:0001754)","Negative Regulation Of Locomotion (GO:0040013)","Negative Regulation Of Macrophage Derived Foam Cell Differentiation (GO:0010745)","Adipose Tissue Development (GO:0060612)","Negative Regulation Of Voltage-Gated Calcium Channel Activity (GO:1901386)","Primary miRNA Processing (GO:0031053)","Regulation Of Nitric-Oxide Synthase Biosynthetic Process (GO:0051769)","Positive Regulation Of Gene Expression (GO:0010628)","Regulation Of Feeding Behavior (GO:0060259)","Negative Regulation Of Receptor Signaling Pathway Via STAT (GO:1904893)","Regulation Of Macrophage Differentiation (GO:0045649)","Lipoprotein Biosynthetic Process (GO:0042158)","Mammary Gland Epithelium Development (GO:0061180)","Regulation Of MHC Class II Biosynthetic Process (GO:0045346)","Positive Regulation Of Cellular Biosynthetic Process (GO:0031328)","Positive Regulation Of Glucose Metabolic Process (GO:0010907)","Activation Of NF-kappaB-inducing Kinase Activity (GO:0007250)","Negative Regulation Of Type I Interferon-Mediated Signaling Pathway (GO:0060339)","Platelet-Derived Growth Factor Receptor Signaling Pathway (GO:0048008)","Cellular Response To Sterol (GO:0036315)","Positive Regulation Of Cell Activation (GO:0050867)","Regulation Of Carbohydrate Metabolic Process (GO:0006109)","Positive Regulation Of Homotypic Cell-Cell Adhesion (GO:0034112)","Extrinsic Apoptotic Signaling Pathway In Absence Of Ligand (GO:0097192)","Negative Regulation Of Receptor Signaling Pathway Via JAK-STAT (GO:0046426)","Activin Receptor Signaling Pathway (GO:0032924)","Protein Localization To Mitochondrion (GO:0070585)","Negative Regulation Of Macromolecule Metabolic Process (GO:0010605)","Positive Regulation Of Transcription By RNA Polymerase II (GO:0045944)","Positive Regulation Of Leukocyte Proliferation (GO:0070665)","Positive Regulation Of Small GTPase Mediated Signal Transduction (GO:0051057)","Regulation Of Receptor Signaling Pathway Via STAT (GO:1904892)","Regulation Of Vacuole Organization (GO:0044088)","Positive Regulation Of SMAD Protein Signal Transduction (GO:0060391)","Regulation Of Signal Transduction (GO:0009966)","Regulation Of Cell Differentiation (GO:0045595)","Regulation Of Transcription By RNA Polymerase II (GO:0006357)","T-helper Cell Differentiation (GO:0042093)","Associative Learning (GO:0008306)","Negative Regulation Of ERBB Signaling Pathway (GO:1901185)","Negative Regulation Of T Cell Receptor Signaling Pathway (GO:0050860)","Positive Regulation Of Cytokine Production Involved In Inflammatory Response (GO:1900017)","Positive Regulation Of Extracellular Matrix Organization (GO:1903055)"],["3/379","4/763","2/130","5/1243","2/146","1/11","1/11","1/11","1/11","1/11","1/11","1/12","1/12","1/12","1/12","1/12","1/12","1/12","1/13","3/480","1/14","1/14","1/14","1/14","1/14","1/14","2/174","1/15","1/15","1/15","1/15","1/15","1/15","1/15","1/16","1/16","1/16","1/16","1/16","2/186","4/938","1/17","1/17","1/17","1/17","1/17","2/193","2/193","6/2028","1/18","1/18","1/18","1/18","1/18","1/18"],[0.00795718062611596,0.00888313754259764,0.00889480097267258,0.00996457736920374,0.0111101460750924,0.0120365474388125,0.0120365474388125,0.0120365474388125,0.0120365474388125,0.0120365474388125,0.0120365474388125,0.0131238993688259,0.0131238993688259,0.0131238993688259,0.0131238993688259,0.0131238993688259,0.0131238993688259,0.0131238993688259,0.0142101091631312,0.0150717979131515,0.0152951779293368,0.0152951779293368,0.0152951779293368,0.0152951779293368,0.0152951779293368,0.0152951779293368,0.0155076314157012,0.0163791067947741,0.0163791067947741,0.0163791067947741,0.0163791067947741,0.0163791067947741,0.0163791067947741,0.0163791067947741,0.0174618968734111,0.0174618968734111,0.0174618968734111,0.0174618968734111,0.0174618968734111,0.0175868771046072,0.0178975605538794,0.0185435492893687,0.0185435492893687,0.0185435492893687,0.0185435492893687,0.0185435492893687,0.0188518227411103,0.0188518227411103,0.0193413029411517,0.0196240651712844,0.0196240651712844,0.0196240651712844,0.0196240651712844,0.0196240651712844,0.0196240651712844],[0.0333379941967886,0.0333379941967886,0.0333379941967886,0.0359328699071286,0.037774496655314,0.0392424423347587,0.0392424423347587,0.0392424423347587,0.0392424423347587,0.0392424423347587,0.0392424423347587,0.0408299091474583,0.0408299091474583,0.0408299091474583,0.0408299091474583,0.0408299091474583,0.0408299091474583,0.0408299091474583,0.0439221555951329,0.0452205260519523,0.0452205260519523,0.0452205260519523,0.0452205260519523,0.0452205260519523,0.0452205260519523,0.0452205260519523,0.0455656330486034,0.0461328688420858,0.0461328688420858,0.0461328688420858,0.0461328688420858,0.0461328688420858,0.0461328688420858,0.0461328688420858,0.0477693270789867,0.0477693270789867,0.0477693270789867,0.0477693270789867,0.0477693270789867,0.0478363057245315,0.0484047660434466,0.0487664611145829,0.0487664611145829,0.0487664611145829,0.0487664611145829,0.0487664611145829,0.0490353422118498,0.0490353422118498,0.0491634474817441,0.0491634474817441,0.0491634474817441,0.0491634474817441,0.0491634474817441,0.0491634474817441,0.0491634474817441],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8.23152295632699,5.62699458351632,15.5078125,4.45215242801482,13.7736111111111,95.0857142857143,95.0857142857143,95.0857142857143,95.0857142857143,95.0857142857143,95.0857142857143,86.4372294372294,86.4372294372294,86.4372294372294,86.4372294372294,86.4372294372294,86.4372294372294,86.4372294372294,79.2301587301587,6.45514730221781,73.1318681318681,73.1318681318681,73.1318681318681,73.1318681318681,73.1318681318681,73.1318681318681,11.5151162790698,67.9047619047619,67.9047619047619,67.9047619047619,67.9047619047619,67.9047619047619,67.9047619047619,67.9047619047619,63.3746031746032,63.3746031746032,63.3746031746032,63.3746031746032,63.3746031746032,10.7576086956522,4.53104925053533,59.4107142857143,59.4107142857143,59.4107142857143,59.4107142857143,59.4107142857143,10.3596858638743,10.3596858638743,3.33011869436202,55.9131652661064,55.9131652661064,55.9131652661064,55.9131652661064,55.9131652661064,55.9131652661064],[39.7885522838301,26.5796741891276,73.2323620438479,20.5187183182519,61.9798248078744,420.260566245477,420.260566245477,420.260566245477,420.260566245477,420.260566245477,420.260566245477,374.560203730456,374.560203730456,374.560203730456,374.560203730456,374.560203730456,374.560203730456,374.560203730456,337.029380313891,27.0788908745501,305.707127278926,305.707127278926,305.707127278926,305.707127278926,305.707127278926,305.707127278926,47.9768456216383,279.207318677518,279.207318677518,279.207318677518,279.207318677518,279.207318677518,279.207318677518,279.207318677518,256.523541924451,256.523541924451,256.523541924451,256.523541924451,256.523541924451,43.4672181596009,18.2288228139353,236.908142520545,236.908142520545,236.908142520545,236.908142520545,236.908142520545,41.1398216880464,41.1398216880464,13.1390246718416,219.79457723602,219.79457723602,219.79457723602,219.79457723602,219.79457723602,219.79457723602],["SMAD3;JAK2;PTPN2","SMAD3;STAT3;RORC;IRF8","SMAD3;STAT3","SMAD3;STAT3;RORC;IRF8;FOSL2","STAT3;PTPN2","PTPN2","RGS14","JAK2","ATG16L1","PTPN2","RORC","ATG16L1","STAT3","PTPN2","ITGAV","RORC","GPR35","SMAD3","JAK2","SMAD3;STAT3;BRD7","STAT3","PTPN2","PTPN2","ATG16L1","JAK2","JAK2","SMAD3;PTPN2","PTPN2","TNFSF15","PTPN2","RGS14","RORC","JAK2","RORC","JAK2","ITGAV","PTPN2","SMAD3","RNF186","SMAD3;STAT3","SMAD3;STAT3;IRF8;FOSL2","JAK2","ITGAV","JAK2","RNF186","JAK2","RGS14;PTPN2","RORC;BRD7","SMAD3;STAT3;RORC;IRF8;BRD7;FOSL2","RORC","RGS14","PTPN2","PTPN2","STAT3","SMAD3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Term<\/th>\n      <th>Overlap<\/th>\n      <th>P.value<\/th>\n      <th>Adjusted.P.value<\/th>\n      <th>Old.P.value<\/th>\n      <th>Old.Adjusted.P.value<\/th>\n      <th>Odds.Ratio<\/th>\n      <th>Combined.Score<\/th>\n      <th>Genes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-37e46cf92521dcbed3f8" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"font-size: 150%; caption-side: topleft; text-align = left; color:black;\">Unique GO terms at p.adjust &lt; 0.05 -- IBD-ebi-a-GCST004131--CS_NOT_filtered<\/caption>","data":[["53","76","104","124"],["Cellular Response To Chemokine (GO:1990869)","Regulation Of Establishment Of T Cell Polarity (GO:1903903)","Intracellular pH Elevation (GO:0051454)","Regulation Of Small GTPase Mediated Signal Transduction (GO:0051056)"],["2/59","1/5","1/7","2/118"],[0.00307824866090805,0.00698104063423331,0.00976029245561303,0.01180359006217],[0.0288077610530263,0.0422267823729234,0.0444138078714134,0.0472143602486801],[0,0,0,0],[0,0,0,0],[26.8758434547908,184.888888888889,123.246913580247,13.1671087533157],[155.43360415256,917.891480308298,570.56331831325,58.4534246691948],["GPR35;DOCK8","DOCK8","SLC26A3","DOCK8;ITGAV"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Term<\/th>\n      <th>Overlap<\/th>\n      <th>P.value<\/th>\n      <th>Adjusted.P.value<\/th>\n      <th>Old.P.value<\/th>\n      <th>Old.Adjusted.P.value<\/th>\n      <th>Odds.Ratio<\/th>\n      <th>Combined.Score<\/th>\n      <th>Genes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] eulerr_6.1.1  ggplot2_3.5.1 tidyr_1.3.0   scales_1.3.0  dplyr_1.1.4  

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   xfun_0.41          bslib_0.3.1        purrr_1.0.2       
 [5] colorspace_2.0-3   vctrs_0.6.5        generics_0.1.2     htmltools_0.5.2   
 [9] yaml_2.3.5         utf8_1.2.2         rlang_1.1.2        jquerylib_0.1.4   
[13] later_1.3.0        pillar_1.9.0       glue_1.6.2         withr_2.5.0       
[17] RColorBrewer_1.1-3 lifecycle_1.0.4    stringr_1.5.1      munsell_0.5.0     
[21] gtable_0.3.0       workflowr_1.7.0    htmlwidgets_1.5.4  evaluate_0.15     
[25] labeling_0.4.2     knitr_1.39         fastmap_1.1.0      httpuv_1.6.5      
[29] crosstalk_1.2.0    fansi_1.0.3        highr_0.9          Rcpp_1.0.12       
[33] promises_1.2.0.1   DT_0.22            jsonlite_1.8.0     farver_2.1.0      
[37] fs_1.5.2           digest_0.6.29      stringi_1.7.6      polyclip_1.10-0   
[41] rprojroot_2.0.3    grid_4.2.0         cli_3.6.1          tools_4.2.0       
[45] magrittr_2.0.3     sass_0.4.1         tibble_3.2.1       whisker_0.4       
[49] pkgconfig_2.0.3    polylabelr_0.2.0   rmarkdown_2.25     rstudioapi_0.13   
[53] R6_2.5.1           git2r_0.30.1       compiler_4.2.0    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
