<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-04-07" />

<title>Comparing Munro st &amp; Susie st models in multi-group analysis</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparing Munro st &amp; Susie st models in
multi-group analysis</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-04-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreef119060f44db91a3ac699de3d62eafc5cd05431ftargetblankf119060a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/f119060f44db91a3ac699de3d62eafc5cd05431f" target="_blank">f119060</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreef119060f44db91a3ac699de3d62eafc5cd05431ftargetblankf119060a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/f119060f44db91a3ac699de3d62eafc5cd05431f" target="_blank">f119060</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    cv/

Untracked files:
    Untracked:  analysis/comparing_munro_susie_st.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The Susie st models are trained here <a
href="https://sq-96.github.io/multigroup_ctwas_analysis/data_weight_training_fusionscale_rsqupdated_allhaveqtl.html"
class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/data_weight_training_fusionscale_rsqupdated_allhaveqtl.html</a></p>
<p>The Munro’s model are from this paper <a
href="https://www.nature.com/articles/s41467-024-54840-8"
class="uri">https://www.nature.com/articles/s41467-024-54840-8</a></p>
<pre class="r"><code>library(ctwas)
library(ggplot2)
library(tidyverse)

source(&quot;/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R&quot;)

folder_results_munroST &lt;- &quot;/project/xinhe/shengqian/ctwas_GWAS_analysis/snakemake_outputs/&quot;
folder_results_susieST &lt;- &quot;/project/xinhe/xsun/multi_group_ctwas/15.susie_weights/snakemake_outputs/&quot;
folder_results_woST &lt;- &quot;/project/xinhe/shengqian/ctwas_GWAS_analysis/snakemake_outputs/&quot;

# mapping_predictdb &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS&quot;)
# mapping_munro &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS&quot;)
# mapping_two &lt;- rbind(mapping_predictdb,mapping_munro)

colors &lt;- c(  &quot;#1f77b4&quot;, &quot;#ff7f0e&quot;, &quot;#2ca02c&quot;, &quot;#d62728&quot;,  &quot;#9467bd&quot;, &quot;#8c564b&quot;, &quot;#e377c2&quot;, &quot;#7f7f7f&quot;,  &quot;#bcbd22&quot;,  &quot;#17becf&quot;,  &quot;#f7b6d2&quot;,  &quot;#c5b0d5&quot;,  &quot;#9edae5&quot;, &quot;#ffbb78&quot;,  &quot;#98df8a&quot;,  &quot;#ff9896&quot; )

plot_piechart &lt;- function(ctwas_parameters, colors, by, title) {
  # Create the initial data frame
  data &lt;- data.frame(
    category = names(ctwas_parameters$prop_heritability),
    percentage = ctwas_parameters$prop_heritability
  )
  
  # Split the category into context and type
  data &lt;- data %&gt;%
    mutate(
      context = sub(&quot;\\|.*&quot;, &quot;&quot;, category),
      type = sub(&quot;.*\\|&quot;, &quot;&quot;, category)
    )
  
  # Aggregate the data based on the &#39;by&#39; parameter
  if (by == &quot;type&quot;) {
    data &lt;- data %&gt;%
      group_by(type) %&gt;%
      summarize(percentage = sum(percentage)) %&gt;%
      mutate(category = type)  # Use type as the new category
  } else if (by == &quot;context&quot;) {
    data &lt;- data %&gt;%
      group_by(context) %&gt;%
      summarize(percentage = sum(percentage)) %&gt;%
      mutate(category = context)  # Use context as the new category
  } else {
    stop(&quot;Invalid &#39;by&#39; parameter. Use &#39;type&#39; or &#39;context&#39;.&quot;)
  }
  
  # Calculate percentage labels for the chart
  data$percentage_label &lt;- paste0(round(data$percentage * 100, 1), &quot;%&quot;)
  
  # Create the pie chart
  pie &lt;- ggplot(data, aes(x = &quot;&quot;, y = percentage, fill = category)) +
    geom_bar(stat = &quot;identity&quot;, width = 1) +
    coord_polar(&quot;y&quot;, start = 0) +
    theme_void() +  # Remove background and axes
    geom_text(aes(label = percentage_label), 
              position = position_stack(vjust = 0.5), size = 3) +  # Adjust size as needed
    scale_fill_manual(values = colors) +  # Custom colors
    labs(fill = &quot;Category&quot;) +  # Legend title
    ggtitle(title)  # Title
  
  return(pie)
}</code></pre>
</div>
<div id="ldl-ukb-d-30780_irnt" class="section level1">
<h1>LDL-ukb-d-30780_irnt</h1>
<div id="setting-shared_all-thin-0.1" class="section level2">
<h2>Setting: shared_all, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-0.5" class="section level2">
<h2>Setting: shared_all, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-1" class="section level2">
<h2>Setting: shared_all, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-16-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.1" class="section level2">
<h2>Setting: shared_type, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-21-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.5" class="section level2">
<h2>Setting: shared_type, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-26-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-1" class="section level2">
<h2>Setting: shared_type, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-31-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ibd-ebi-a-gcst004131" class="section level1">
<h1>IBD-ebi-a-GCST004131</h1>
<div id="setting-shared_all-thin-0.1-1" class="section level2">
<h2>Setting: shared_all, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;IBD-ebi-a-GCST004131&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-36-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-0.5-1" class="section level2">
<h2>Setting: shared_all, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-41-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-1-1" class="section level2">
<h2>Setting: shared_all, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-46-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.1-1" class="section level2">
<h2>Setting: shared_type, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-51-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.5-1" class="section level2">
<h2>Setting: shared_type, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-56-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-1-1" class="section level2">
<h2>Setting: shared_type, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-61-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="t2d-panukb" class="section level1">
<h1>T2D-panukb</h1>
<div id="setting-shared_all-thin-0.1-2" class="section level2">
<h2>Setting: shared_all, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;T2D-panukb&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-66-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-0.5-2" class="section level2">
<h2>Setting: shared_all, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-71-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-1-2" class="section level2">
<h2>Setting: shared_all, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-76-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.1-2" class="section level2">
<h2>Setting: shared_type, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-81-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.5-2" class="section level2">
<h2>Setting: shared_type, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-86-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-1-2" class="section level2">
<h2>Setting: shared_type, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-91-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="bmi-panukb" class="section level1">
<h1>BMI-panukb</h1>
<div id="setting-shared_all-thin-0.1-3" class="section level2">
<h2>Setting: shared_all, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;BMI-panukb&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-96-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-0.5-3" class="section level2">
<h2>Setting: shared_all, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-101-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_all-thin-1-3" class="section level2">
<h2>Setting: shared_all, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_all&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-106-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.1-3" class="section level2">
<h2>Setting: shared_type, thin = 0.1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-111-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-0.5-3" class="section level2">
<h2>Setting: shared_type, thin = 0.5</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 0.5
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-116-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="setting-shared_type-thin-1-3" class="section level2">
<h2>Setting: shared_type, thin = 1</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
thin &lt;- 1
var_struc &lt;- &quot;shared_type&quot;

gwas_n &lt;- samplesize[trait]</code></pre>
<pre class="r"><code>st &lt;- &quot;without_ST&quot;

param_woST &lt;- readRDS(paste0(folder_results_woST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_woST &lt;- summarize_param(param_woST, gwas_n)
total_nonSNPpve_woST &lt;- 1- ctwas_parameters_woST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_woST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_woST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;withoutST - nonSNP %h2g:&quot;,round(total_nonSNPpve_woST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_susieST&quot;

param_susieST &lt;- readRDS(paste0(folder_results_susieST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))


ctwas_parameters_susieST &lt;- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST &lt;- 1- ctwas_parameters_susieST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_susieST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;susieST - nonSNP %h2g:&quot;,round(total_nonSNPpve_susieST, digits = 4)))</code></pre>
<pre class="r"><code>st &lt;- &quot;with_ST&quot;
param_munroST &lt;- readRDS(paste0(folder_results_munroST,&quot;/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,st,&quot;.thin&quot;,thin,&quot;.&quot;,var_struc,&quot;.param.RDS&quot;))

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)

ctwas_parameters_munroST &lt;- summarize_param(param_munroST, gwas_n)
total_nonSNPpve_munroST &lt;- 1- ctwas_parameters_munroST$prop_heritability[&quot;SNP&quot;]
pve_pie_by_type_munroST &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_munroST, colors = colors, by = &quot;type&quot;, title = paste0(&quot;munroST - nonSNP %h2g:&quot;,round(total_nonSNPpve_munroST, digits = 4)))</code></pre>
<pre class="r"><code>gridExtra::grid.arrange(pve_pie_by_type_woST, pve_pie_by_type_susieST,pve_pie_by_type_munroST, 
                        ncol = 3, top = paste0(trait,&quot;-thin&quot;,thin,&quot;-&quot;,var_struc))        </code></pre>
<p><img src="figure/comparing_munro_susie_st.Rmd/unnamed-chunk-121-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1    stringr_1.5.1    dplyr_1.1.4      purrr_1.0.2     
 [5] readr_2.1.2      tidyr_1.3.0      tibble_3.2.1     tidyverse_1.3.1 
 [9] ggplot2_3.5.1    ctwas_0.5.4.9000

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] GenomicRanges_1.48.0        base64enc_0.1-3            
  [9] fs_1.5.2                    rstudioapi_0.13            
 [11] farver_2.1.0                ggrepel_0.9.1              
 [13] bit64_4.0.5                 lubridate_1.8.0            
 [15] AnnotationDbi_1.58.0        fansi_1.0.3                
 [17] xml2_1.3.3                  codetools_0.2-18           
 [19] logging_0.10-108            cachem_1.0.6               
 [21] knitr_1.39                  jsonlite_1.8.0             
 [23] workflowr_1.7.0             Rsamtools_2.12.0           
 [25] broom_0.8.0                 dbplyr_2.1.1               
 [27] png_0.1-7                   compiler_4.2.0             
 [29] httr_1.4.3                  backports_1.4.1            
 [31] assertthat_0.2.1            Matrix_1.5-3               
 [33] fastmap_1.1.0               lazyeval_0.2.2             
 [35] cli_3.6.1                   later_1.3.0                
 [37] htmltools_0.5.2             prettyunits_1.1.1          
 [39] tools_4.2.0                 gtable_0.3.0               
 [41] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [43] rappdirs_0.3.3              Rcpp_1.0.12                
 [45] Biobase_2.56.0              cellranger_1.1.0           
 [47] jquerylib_0.1.4             vctrs_0.6.5                
 [49] Biostrings_2.64.0           rtracklayer_1.56.0         
 [51] xfun_0.41                   rvest_1.0.2                
 [53] irlba_2.3.5                 lifecycle_1.0.4            
 [55] restfulr_0.0.14             ensembldb_2.20.2           
 [57] XML_3.99-0.14               zlibbioc_1.42.0            
 [59] zoo_1.8-10                  scales_1.3.0               
 [61] gggrid_0.2-0                hms_1.1.1                  
 [63] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [65] ProtGenerics_1.28.0         parallel_4.2.0             
 [67] SummarizedExperiment_1.26.1 AnnotationFilter_1.20.0    
 [69] LDlinkR_1.2.3               yaml_2.3.5                 
 [71] curl_4.3.2                  gridExtra_2.3              
 [73] memoise_2.0.1               sass_0.4.1                 
 [75] biomaRt_2.54.1              stringi_1.7.6              
 [77] RSQLite_2.3.1               highr_0.9                  
 [79] S4Vectors_0.34.0            BiocIO_1.6.0               
 [81] GenomicFeatures_1.48.3      BiocGenerics_0.42.0        
 [83] filelock_1.0.2              BiocParallel_1.30.3        
 [85] repr_1.1.4                  GenomeInfoDb_1.39.9        
 [87] rlang_1.1.2                 pkgconfig_2.0.3            
 [89] matrixStats_0.62.0          bitops_1.0-7               
 [91] evaluate_0.15               lattice_0.20-45            
 [93] labeling_0.4.2              GenomicAlignments_1.32.0   
 [95] htmlwidgets_1.5.4           cowplot_1.1.1              
 [97] bit_4.0.4                   tidyselect_1.2.0           
 [99] magrittr_2.0.3              AMR_2.1.1                  
[101] R6_2.5.1                    IRanges_2.30.0             
[103] generics_0.1.2              DelayedArray_0.22.0        
[105] DBI_1.2.2                   haven_2.5.0                
[107] withr_2.5.0                 pgenlibr_0.3.3             
[109] pillar_1.9.0                KEGGREST_1.36.3            
[111] RCurl_1.98-1.7              mixsqp_0.3-43              
[113] modelr_0.1.8                crayon_1.5.1               
[115] utf8_1.2.2                  BiocFileCache_2.4.0        
[117] plotly_4.10.0               tzdb_0.4.0                 
[119] rmarkdown_2.25              progress_1.2.2             
[121] readxl_1.4.0                grid_4.2.0                 
[123] data.table_1.14.2           blob_1.2.3                 
[125] git2r_0.30.1                reprex_2.0.1               
[127] digest_0.6.29               httpuv_1.6.5               
[129] stats4_4.2.0                munsell_0.5.0              
[131] viridisLite_0.4.0           skimr_2.1.4                
[133] bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
