<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-12-25" />

<title>GWAS tissue selection</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GWAS tissue selection</h1>
<h4 class="date">2024-12-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree5d4e1ee826e9da2168801c21b839a80ef322212ctargetblank5d4e1eea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/5d4e1ee826e9da2168801c21b839a80ef322212c" target="_blank">5d4e1ee</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree5d4e1ee826e9da2168801c21b839a80ef322212ctargetblank5d4e1eea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/5d4e1ee826e9da2168801c21b839a80ef322212c" target="_blank">5d4e1ee</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/GWAS_tissue_selection.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/GWAS_tissue_selection.Rmd</code>) and HTML (<code>docs/GWAS_tissue_selection.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/5d4e1ee826e9da2168801c21b839a80ef322212c/analysis/GWAS_tissue_selection.Rmd" target="_blank">5d4e1ee</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/5d4e1ee826e9da2168801c21b839a80ef322212c/docs/GWAS_tissue_selection.html" target="_blank">5d4e1ee</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/b0609b8dca50e4b80334b214c5137191b38c4dec/analysis/GWAS_tissue_selection.Rmd" target="_blank">b0609b8</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/b0609b8dca50e4b80334b214c5137191b38c4dec/docs/GWAS_tissue_selection.html" target="_blank">b0609b8</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/631921f6008a5e2c2fa1fc04d38ddbbc76a49fa7/analysis/GWAS_tissue_selection.Rmd" target="_blank">631921f</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/631921f6008a5e2c2fa1fc04d38ddbbc76a49fa7/docs/GWAS_tissue_selection.html" target="_blank">631921f</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/e0136fbc1fab82ca0df5ae317f9b80f76cf3bb28/analysis/GWAS_tissue_selection.Rmd" target="_blank">e0136fb</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/e0136fbc1fab82ca0df5ae317f9b80f76cf3bb28/docs/GWAS_tissue_selection.html" target="_blank">e0136fb</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/1d8b9ae6ba6534b8f4e7d17b45a8aeca17dd0cbd/docs/GWAS_tissue_selection.html" target="_blank">1d8b9ae</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/1fadcfe06a8d30e0deb36aa3dc59e05e61372b7d/analysis/GWAS_tissue_selection.Rmd" target="_blank">1fadcfe</a>
</td>
<td>
sq-96
</td>
<td>
2024-12-25
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>[1] “WBC-ieu-b-30” group %h2g h2g enrichment 45 Whole_Blood|eQTL 0.0941 0.0156 101.0520 17 Cells_Cultured_fibroblasts|eQTL 0.0349 0.0058 33.5420 23 Esophagus_Muscularis|eQTL 0.0334 0.0056 31.8354 33 Pancreas|eQTL 0.0246 0.0041 25.5033 31 Nerve_Tibial|eQTL 0.0236 0.0039 21.1199 4 Artery_Aorta|eQTL 0.0225 0.0037 21.7025 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0220 0.0037 20.5634 8 Brain_Cerebellar_Hemisphere|eQTL 0.0208 0.0035 22.1681 27 Liver|eQTL 0.0197 0.0033 22.9943 25 Heart_Left_Ventricle|eQTL 0.0186 0.0031 19.8063 pvalue 45 5.6807314184356e-64 17 2.90766830681651e-19 23 1.01650815898354e-17 33 1.67571255560782e-12 31 1.14847648350927e-11 4 1.10962840492442e-10 36 1.37733210260063e-10 8 7.76433473103528e-10 27 3.69791286290999e-10 25 2.03407031058494e-08 [1] “SBP-ukb-a-360” group %h2g h2g enrichment 4 Artery_Aorta|eQTL 0.0853 0.0048 64.9022 10 Brain_Cortex|eQTL 0.0290 0.0016 23.7171 6 Artery_Tibial|eQTL 0.0267 0.0015 19.8120 17 Cells_Cultured_fibroblasts|eQTL 0.0245 0.0014 18.6432 8 Brain_Cerebellar_Hemisphere|eQTL 0.0200 0.0011 16.6311 22 Esophagus_Mucosa|eQTL 0.0182 0.0010 13.6421 43 Uterus|eQTL 0.0179 0.0010 16.3428 31 Nerve_Tibial|eQTL 0.0176 0.0010 12.4256 44 Vagina|eQTL 0.0158 0.0009 14.9101 45 Whole_Blood|eQTL 0.0147 0.0008 12.3999 pvalue 4 7.85832758430999e-54 10 3.50720394943851e-14 6 3.20390446995539e-12 17 4.44078330099994e-11 8 5.4247352446862e-09 22 2.44137638683189e-07 43 4.54598674854416e-08 31 3.73974100107046e-07 44 5.59329467260843e-07 45 1.44973432875299e-05 [1] “LDL-ukb-d-30780_irnt” group %h2g h2g enrichment 27 Liver|eQTL 0.1855 0.0108 53.2274 39 Spleen|eQTL 0.0794 0.0046 20.5074 21 Esophagus_Gastroesophageal_Junction|eQTL 0.0479 0.0028 11.9923 23 Esophagus_Muscularis|eQTL 0.0413 0.0024 9.8514 22 Esophagus_Mucosa|eQTL 0.0404 0.0024 9.6216 41 Testis|eQTL 0.0377 0.0022 8.3416 5 Artery_Coronary|eQTL 0.0372 0.0022 9.8611 20 Colon_Transverse|eQTL 0.0361 0.0021 8.8942 9 Brain_Cerebellum|eQTL 0.0279 0.0016 7.2625 18 Cells_EBV-transformed_lymphocytes|eQTL 0.0249 0.0015 7.3576 pvalue 27 2.54298863039887e-40 39 2.88552476459251e-13 21 5.48469484638457e-07 23 9.5367457496114e-06 22 9.90605613804312e-06 41 0.000158547374821604 5 5.57853787838304e-05 20 7.95001364092926e-05 9 0.00055008336344716 18 0.00248772271233422 [1] “aFib-ebi-a-GCST006414” group %h2g h2g enrichment 24 Heart_Atrial_Appendage|eQTL 0.1319 0.0048 121.9800 30 Muscle_Skeletal|eQTL 0.0475 0.0017 43.9387 25 Heart_Left_Ventricle|eQTL 0.0452 0.0016 44.1891 42 Thyroid|eQTL 0.0420 0.0015 35.1272 18 Cells_EBV-transformed_lymphocytes|eQTL 0.0227 0.0008 24.6914 6 Artery_Tibial|eQTL 0.0212 0.0008 18.4574 23 Esophagus_Muscularis|eQTL 0.0175 0.0006 15.3095 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0163 0.0006 14.0232 4 Artery_Aorta|eQTL 0.0143 0.0005 12.7071 11 Brain_Frontal_Cortex_BA9|eQTL 0.0143 0.0005 13.9260 pvalue 24 4.22806053942042e-103 30 9.16937676593836e-30 25 3.07519054413411e-27 42 3.45917186396761e-24 18 3.42943532137214e-11 6 4.04040213182331e-10 23 1.09959794874964e-07 36 1.23477154464656e-07 4 1.52388873427118e-06 11 7.2441465464183e-07 [1] “IBD-ebi-a-GCST004131” group %h2g h2g enrichment 45 Whole_Blood|eQTL 0.0812 0.0034 55.7811 20 Colon_Transverse|eQTL 0.0690 0.0029 43.4906 23 Esophagus_Muscularis|eQTL 0.0300 0.0013 18.2639 5 Artery_Coronary|eQTL 0.0272 0.0011 18.5824 14 Brain_Nucleus_accumbens_basal_ganglia|eQTL 0.0269 0.0011 18.4098 25 Heart_Left_Ventricle|eQTL 0.0239 0.0010 16.2275 1 Adipose_Subcutaneous|eQTL 0.0226 0.0009 13.6953 28 Lung|eQTL 0.0186 0.0008 11.1695 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0168 0.0007 10.0033 27 Liver|eQTL 0.0165 0.0007 12.2731 pvalue 45 4.99505898316512e-41 20 6.2571247112006e-33 23 5.76247354623694e-12 5 3.19278202953527e-11 14 4.94663623607433e-10 25 1.04491130949366e-08 1 2.90198097587644e-08 28 4.75865121712891e-06 36 6.15299251065155e-05 27 1.6200342069326e-05 [1] “SCZ-ieu-b-5102” group %h2g h2g enrichment 6 Brain_Hippocampus|eQTL 0.0364 0.0021 30.2148 3 Brain_Cerebellum|eQTL 0.0340 0.0020 26.1635 2 Brain_Cerebellar_Hemisphere|eQTL 0.0269 0.0015 21.1399 8 Brain_Nucleus_accumbens_basal_ganglia|eQTL 0.0262 0.0015 20.7130 4 Brain_Cortex|eQTL 0.0227 0.0013 17.5502 1 Brain_Caudate_basal_ganglia|eQTL 0.0202 0.0012 15.9689 9 Brain_Putamen_basal_ganglia|eQTL 0.0176 0.0010 14.1077 7 Brain_Hypothalamus|eQTL 0.0145 0.0008 12.0616 5 Brain_Frontal_Cortex_BA9|eQTL 0.0035 0.0002 2.7686 pvalue 6 1.53881085901605e-20 3 2.05648338264769e-18 2 2.85577289581999e-13 8 3.0916072222875e-13 4 8.62054947522828e-12 1 2.09800137719681e-09 9 2.4680210139392e-08 7 4.01241940649027e-06 5 0.132331287686693 [1] “BMI-panukb” group %h2g h2g enrichment 17 Cells_Cultured_fibroblasts|eQTL 0.0253 0.0038 26.8057 8 Brain_Cerebellar_Hemisphere|eQTL 0.0157 0.0023 17.9658 22 Esophagus_Mucosa|eQTL 0.0152 0.0023 15.6915 2 Adipose_Visceral_Omentum|eQTL 0.0111 0.0017 11.6797 1 Adipose_Subcutaneous|eQTL 0.0110 0.0016 11.3628 3 Adrenal_Gland|eQTL 0.0110 0.0016 12.7761 26 Kidney_Cortex|eQTL 0.0104 0.0016 15.8748 4 Artery_Aorta|eQTL 0.0103 0.0015 10.8118 39 Spleen|eQTL 0.0103 0.0015 11.5282 33 Pancreas|eQTL 0.0096 0.0014 10.7796 pvalue 17 1.51479837581904e-13 8 1.75843118191658e-08 22 5.12787641123677e-08 2 0.000120776681274031 1 0.000130692559843376 3 7.53766382414686e-05 26 2.260936125254e-05 4 0.000129766237360201 39 8.10123587793894e-05 33 8.42801496997152e-05 [1] “Height-panukb” group %h2g h2g enrichment 17 Cells_Cultured_fibroblasts|eQTL 0.0562 0.0244 34.2110 6 Artery_Tibial|eQTL 0.0418 0.0181 24.6922 24 Heart_Atrial_Appendage|eQTL 0.0384 0.0166 24.1014 42 Thyroid|eQTL 0.0292 0.0127 16.6113 30 Muscle_Skeletal|eQTL 0.0274 0.0118 17.2069 9 Brain_Cerebellum|eQTL 0.0248 0.0107 16.0555 32 Ovary|eQTL 0.0238 0.0103 15.9644 19 Colon_Sigmoid|eQTL 0.0220 0.0096 13.7929 8 Brain_Cerebellar_Hemisphere|eQTL 0.0179 0.0078 11.8111 41 Testis|eQTL 0.0174 0.0075 9.5804 pvalue 17 7.04313093608084e-16 6 2.81224073990845e-11 24 3.21408475330862e-10 42 3.41101054860809e-07 30 1.69259502706613e-07 9 2.51102816213239e-06 32 1.93085966370715e-06 19 4.8179676575039e-05 8 0.000505092009694556 41 0.00109142280233462 [1] “PLT-panukb” group %h2g h2g enrichment 45 Whole_Blood|eQTL 0.0515 0.0113 45.9724 27 Liver|eQTL 0.0342 0.0075 32.4906 39 Spleen|eQTL 0.0313 0.0069 26.7610 11 Brain_Frontal_Cortex_BA9|eQTL 0.0274 0.0060 24.0827 6 Artery_Tibial|eQTL 0.0252 0.0055 19.7647 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0230 0.0050 17.7827 22 Esophagus_Mucosa|eQTL 0.0188 0.0041 14.7911 42 Thyroid|eQTL 0.0187 0.0041 14.0762 33 Pancreas|eQTL 0.0184 0.0040 15.7045 41 Testis|eQTL 0.0183 0.0040 13.3913 pvalue 45 1.25335025173107e-22 27 7.23553156385814e-15 39 1.81446888744998e-11 11 3.66915998804606e-10 6 1.54460854551567e-09 36 2.67818996198416e-08 22 5.0455639344779e-07 42 6.34876370950653e-07 33 3.83161781823002e-06 41 1.19925379331179e-05 [1] “RBC-panukb” group %h2g h2g enrichment 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0282 0.0044 27.5902 42 Thyroid|eQTL 0.0237 0.0037 22.5576 30 Muscle_Skeletal|eQTL 0.0225 0.0035 23.7573 21 Esophagus_Gastroesophageal_Junction|eQTL 0.0224 0.0035 23.4399 45 Whole_Blood|eQTL 0.0209 0.0032 23.5443 17 Cells_Cultured_fibroblasts|eQTL 0.0192 0.0030 19.5979 24 Heart_Atrial_Appendage|eQTL 0.0192 0.0030 20.2034 23 Esophagus_Muscularis|eQTL 0.0165 0.0026 16.3759 9 Brain_Cerebellum|eQTL 0.0160 0.0025 17.3485 37 Skin_Sun_Exposed_Lower_leg|eQTL 0.0150 0.0023 14.3446 pvalue 36 8.92714712105978e-14 42 8.65177203212483e-11 30 3.1417094729532e-11 21 3.07372373302249e-11 45 2.79668312009139e-10 17 2.10465920399433e-08 24 1.53773690932433e-08 23 4.3786635344099e-07 9 2.31325275814238e-07 37 7.96331852655667e-06 [1] “HB-panukb” group %h2g h2g enrichment 36 Skin_Not_Sun_Exposed_Suprapubic|eQTL 0.0399 0.0049 37.5308 16 Breast_Mammary_Tissue|eQTL 0.0283 0.0035 28.0139 45 Whole_Blood|eQTL 0.0274 0.0034 29.7257 6 Artery_Tibial|eQTL 0.0260 0.0032 24.8308 21 Esophagus_Gastroesophageal_Junction|eQTL 0.0248 0.0030 24.9326 1 Adipose_Subcutaneous|eQTL 0.0211 0.0026 20.2175 24 Heart_Atrial_Appendage|eQTL 0.0186 0.0023 18.7974 30 Muscle_Skeletal|eQTL 0.0183 0.0023 18.5992 28 Lung|eQTL 0.0173 0.0021 16.3151 17 Cells_Cultured_fibroblasts|eQTL 0.0166 0.0020 16.2702 pvalue 36 3.03958883961412e-22 16 5.19191705454531e-15 45 3.40931447375258e-14 6 2.82913700576883e-13 21 3.22907287493894e-12 1 1.31664595870116e-10 24 1.60803256350194e-09 30 3.23357879022541e-08 28 4.29398782333495e-08 17 4.1797528932889e-08 [1] “HTN-panukb” group %h2g h2g enrichment 41 Testis|eQTL 0.0673 0.0034 54.0009 30 Muscle_Skeletal|eQTL 0.0450 0.0023 41.3485 23 Esophagus_Muscularis|eQTL 0.0434 0.0022 37.6366 1 Adipose_Subcutaneous|eQTL 0.0364 0.0018 31.6459 45 Whole_Blood|eQTL 0.0252 0.0013 24.7075 4 Artery_Aorta|eQTL 0.0209 0.0011 18.3616 37 Skin_Sun_Exposed_Lower_leg|eQTL 0.0194 0.0010 16.1093 22 Esophagus_Mucosa|eQTL 0.0135 0.0007 11.7229 42 Thyroid|eQTL 0.0122 0.0006 10.0806 8 Brain_Cerebellar_Hemisphere|eQTL 0.0105 0.0005 10.0618 pvalue 41 9.05094966574421e-42 30 1.0561338876218e-26 23 1.13863649188553e-24 1 5.31964996369031e-20 45 2.95203823557443e-12 4 3.00158278702788e-10 37 8.15977246574473e-09 22 2.14266513243146e-05 42 2.68295917263992e-05 8 0.000135865574448767 [1] “RA-panukb” group %h2g h2g enrichment 45 Whole_Blood|eQTL 0.0991 7e-04 74.9993 39 Spleen|eQTL 0.0429 3e-04 31.0569 20 Colon_Transverse|eQTL 0.0328 2e-04 22.6594 7 Brain_Caudate_basal_ganglia|eQTL 0.0091 1e-04 6.7329 6 Artery_Tibial|eQTL 0.0088 1e-04 5.8696 28 Lung|eQTL 0.0083 1e-04 5.4898 18 Cells_EBV-transformed_lymphocytes|eQTL 0.0062 0e+00 5.1258 34 Pituitary|eQTL 0.0057 0e+00 4.0393 31 Nerve_Tibial|eQTL 0.0053 0e+00 3.3902 4 Artery_Aorta|eQTL 0.0052 0e+00 3.5109 pvalue 45 1.24224693566232e-44 39 2.06453127968359e-15 20 8.06212017936469e-11 7 0.00927710824291692 6 0.0127420894939191 28 0.0125271140036791 18 0.0532817243609152 34 0.0731186591804842 31 0.0904173323230918 4 0.0806219662162502 [1] “T1D-GCST90014023” group %h2g h2g enrichment 17 Cells_Cultured_fibroblasts|eQTL 0.0680 0.0020 212.0288 33 Pancreas|eQTL 0.0417 0.0012 137.8416 45 Whole_Blood|eQTL 0.0367 0.0011 126.4116 4 Artery_Aorta|eQTL 0.0366 0.0011 113.2641 18 Cells_EBV-transformed_lymphocytes|eQTL 0.0304 0.0009 114.8542 26 Kidney_Cortex|eQTL 0.0210 0.0006 94.0463 42 Thyroid|eQTL 0.0204 0.0006 59.4375 25 Heart_Left_Ventricle|eQTL 0.0173 0.0005 59.0918 37 Skin_Sun_Exposed_Lower_leg|eQTL 0.0161 0.0005 47.1809 38 Small_Intestine_Terminal_Ileum|eQTL 0.0154 0.0005 52.4535 pvalue 17 2.81065495401488e-94 33 4.41459517501271e-52 45 1.89157212394078e-46 4 6.54278434169991e-45 18 3.44998542292686e-37 26 1.87846221528631e-24 42 1.15333237587574e-21 25 4.37466047322857e-19 37 1.51825068972398e-16 38 1.25155624476807e-15 [1] “ATH_gtexukb” group %h2g h2g enrichment 39 Spleen|eQTL 0.0451 0.0012 59.7855 24 Heart_Atrial_Appendage|eQTL 0.0355 0.0009 45.7973 23 Esophagus_Muscularis|eQTL 0.0299 0.0008 36.5359 20 Colon_Transverse|eQTL 0.0297 0.0008 37.4784 6 Artery_Tibial|eQTL 0.0285 0.0008 34.7341 30 Muscle_Skeletal|eQTL 0.0265 0.0007 34.3453 37 Skin_Sun_Exposed_Lower_leg|eQTL 0.0253 0.0007 29.6975 25 Heart_Left_Ventricle|eQTL 0.0247 0.0007 33.7974 1 Adipose_Subcutaneous|eQTL 0.0179 0.0005 21.9811 27 Liver|eQTL 0.0124 0.0003 18.2111 pvalue 39 2.56371011056005e-39 24 1.42409403923027e-28 23 2.58255298487574e-23 20 1.35960173644423e-23 6 1.09989921176953e-21 30 1.4639570548215e-20 37 1.76148037899858e-18 25 5.12088446293589e-18 1 1.20284776096614e-11 27 1.80489862985216e-08 [1] “T2D-panukb” group %h2g h2g enrichment pvalue 34 Pituitary|eQTL 0.0882 0.0027 62.4205 1.07894043295238e-46 6 Artery_Tibial|eQTL 0.0430 0.0013 28.5893 4.30840161781444e-18 40 Stomach|eQTL 0.0371 0.0011 26.7158 8.79450013439535e-16 31 Nerve_Tibial|eQTL 0.0328 0.0010 20.8512 2.88607004792791e-12 10 Brain_Cortex|eQTL 0.0246 0.0008 17.8221 4.8886960440155e-09 42 Thyroid|eQTL 0.0199 0.0006 12.7329 2.32459358276808e-07 1 Adipose_Subcutaneous|eQTL 0.0192 0.0006 12.8371 2.33714126336666e-06 41 Testis|eQTL 0.0167 0.0005 10.3572 4.83558653825038e-05 9 Brain_Cerebellum|eQTL 0.0145 0.0004 10.5893 0.00019852374133577 2 Adipose_Visceral_Omentum|eQTL 0.0136 0.0004 9.2114 0.000268120934367265 [1] “MDD-ieu-b-102” group %h2g h2g enrichment 3 Brain_Cerebellum|eQTL 0.0290 9e-04 23.4849 5 Brain_Frontal_Cortex_BA9|eQTL 0.0220 7e-04 18.3540 4 Brain_Cortex|eQTL 0.0160 5e-04 12.9112 8 Brain_Nucleus_accumbens_basal_ganglia|eQTL 0.0147 4e-04 12.1412 1 Brain_Caudate_basal_ganglia|eQTL 0.0088 3e-04 7.2213 6 Brain_Hippocampus|eQTL 0.0079 2e-04 6.8489 7 Brain_Hypothalamus|eQTL 0.0061 2e-04 5.3045 9 Brain_Putamen_basal_ganglia|eQTL 0.0019 1e-04 1.5749 2 Brain_Cerebellar_Hemisphere|eQTL 0.0013 0e+00 1.0337 pvalue 3 1.00444405159175e-20 5 2.23454176119812e-14 4 2.04645718759749e-09 8 2.14150440448464e-08 1 0.000240153531408593 6 0.00131078024194488 7 0.00887988614306831 9 0.55798165690736 2 0.55989908903174 [1] “BIP-ieu-b-5110” group %h2g h2g enrichment 8 Brain_Nucleus_accumbens_basal_ganglia|eQTL 0.0400 0.0013 31.0540 7 Brain_Hypothalamus|eQTL 0.0350 0.0011 28.6168 3 Brain_Cerebellum|eQTL 0.0305 0.0010 23.0447 5 Brain_Frontal_Cortex_BA9|eQTL 0.0301 0.0010 23.5398 9 Brain_Putamen_basal_ganglia|eQTL 0.0217 0.0007 17.1441 1 Brain_Caudate_basal_ganglia|eQTL 0.0180 0.0006 13.9739 4 Brain_Cortex|eQTL 0.0120 0.0004 9.1046 2 Brain_Cerebellar_Hemisphere|eQTL 0.0058 0.0002 4.4965 6 Brain_Hippocampus|eQTL 0.0012 0.0000 0.9968 pvalue 8 2.55673516040731e-25 7 3.10921052044146e-21 3 7.7748105025584e-18 5 9.27897271646794e-17 9 1.30625972979217e-11 1 4.60536330398576e-09 4 1.11481556307573e-05 2 0.0338022949887545 6 0.483693933143771</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
  [1] colorspace_2.1-1            rjson_0.2.23               
  [3] rprojroot_2.0.3             XVector_0.38.0             
  [5] locuszoomr_0.3.5            GenomicRanges_1.50.2       
  [7] fs_1.6.4                    rstudioapi_0.14            
  [9] ggrepel_0.9.6               bit64_4.5.2                
 [11] AnnotationDbi_1.60.2        fansi_1.0.6                
 [13] xml2_1.3.3                  codetools_0.2-18           
 [15] logging_0.10-108            cachem_1.1.0               
 [17] knitr_1.48                  jsonlite_1.8.9             
 [19] Rsamtools_2.14.0            dbplyr_2.5.0               
 [21] png_0.1-7                   readr_2.1.5                
 [23] compiler_4.2.0              httr_1.4.7                 
 [25] Matrix_1.5-3                fastmap_1.2.0              
 [27] lazyeval_0.2.2              cli_3.6.3                  
 [29] later_1.3.2                 htmltools_0.5.8.1          
 [31] prettyunits_1.2.0           tools_4.2.0                
 [33] gtable_0.3.5                glue_1.7.0                 
 [35] GenomeInfoDbData_1.2.9      dplyr_1.1.4                
 [37] rappdirs_0.3.3              Rcpp_1.0.13                
 [39] Biobase_2.58.0              jquerylib_0.1.4            
 [41] vctrs_0.6.5                 Biostrings_2.66.0          
 [43] rtracklayer_1.58.0          xfun_0.47                  
 [45] stringr_1.5.1               ps_1.7.1                   
 [47] lifecycle_1.0.4             irlba_2.3.5.1              
 [49] restfulr_0.0.15             ensembldb_2.22.0           
 [51] XML_3.99-0.14               getPass_0.2-2              
 [53] zlibbioc_1.44.0             zoo_1.8-12                 
 [55] scales_1.3.0                gggrid_0.2-0               
 [57] hms_1.1.3                   promises_1.3.0             
 [59] MatrixGenerics_1.10.0       ProtGenerics_1.30.0        
 [61] parallel_4.2.0              SummarizedExperiment_1.28.0
 [63] AnnotationFilter_1.22.0     LDlinkR_1.4.0              
 [65] ctwas_0.4.15                yaml_2.3.10                
 [67] curl_5.2.3                  memoise_2.0.1              
 [69] ggplot2_3.5.1               sass_0.4.9                 
 [71] biomaRt_2.54.1              stringi_1.8.4              
 [73] RSQLite_2.3.7               S4Vectors_0.36.2           
 [75] BiocIO_1.8.0                GenomicFeatures_1.50.4     
 [77] BiocGenerics_0.44.0         filelock_1.0.3             
 [79] BiocParallel_1.32.6         GenomeInfoDb_1.34.9        
 [81] rlang_1.1.4                 pkgconfig_2.0.3            
 [83] matrixStats_1.4.1           bitops_1.0-8               
 [85] evaluate_1.0.0              lattice_0.20-45            
 [87] purrr_1.0.2                 GenomicAlignments_1.34.1   
 [89] htmlwidgets_1.6.4           cowplot_1.1.3              
 [91] bit_4.5.0                   processx_3.7.0             
 [93] tidyselect_1.2.1            magrittr_2.0.3             
 [95] R6_2.5.1                    IRanges_2.32.0             
 [97] generics_0.1.3              DelayedArray_0.24.0        
 [99] DBI_1.2.3                   pgenlibr_0.3.7             
[101] pillar_1.9.0                whisker_0.4                
[103] KEGGREST_1.38.0             RCurl_1.98-1.16            
[105] mixsqp_0.3-54               tibble_3.2.1               
[107] crayon_1.5.3                utf8_1.2.4                 
[109] BiocFileCache_2.6.1         plotly_4.10.4              
[111] tzdb_0.4.0                  rmarkdown_2.28             
[113] progress_1.2.3              grid_4.2.0                 
[115] data.table_1.16.0           blob_1.2.4                 
[117] callr_3.7.2                 git2r_0.30.1               
[119] digest_0.6.37               tidyr_1.3.1                
[121] httpuv_1.6.5                stats4_4.2.0               
[123] munsell_0.5.1               viridisLite_0.4.2          
[125] bslib_0.8.0                </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
