<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-08-25" />

<title>Comparing single-group &amp; multi-group for LDL – old PIP_combine_function</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparing single-group &amp; multi-group
for LDL – old PIP_combine_function</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-08-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree184fb61061cc2097e8edaf7f0744793291fd0b4atargetblank184fb61a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/184fb61061cc2097e8edaf7f0744793291fd0b4a" target="_blank">184fb61</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree184fb61061cc2097e8edaf7f0744793291fd0b4atargetblank184fb61a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/184fb61061cc2097e8edaf7f0744793291fd0b4a" target="_blank">184fb61</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/figure/
    Ignored:    results/

Untracked files:
    Untracked:  analysis/LDL_single_multi_compare_oldpip.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ctwas)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(RColorBrewer)
library(dplyr)

load(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/E_S_A_mapping_updated.RData&quot;)
ens_db &lt;- EnsDb.Hsapiens.v86

trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
gwas_n &lt;- 343621

source(&quot;/project/xinhe/xsun/r_functions/combine_pip_old_ctwas.R&quot;)

sum_pve_across_types &lt;- function(ctwas_parameters) {
  # Round the group_pve values
  pve &lt;- round(ctwas_parameters$group_pve, 4)
  pve &lt;- as.data.frame(pve)

  # Extract SNP PVE for later use
  SNP_pve &lt;- pve[&quot;SNP&quot;, ]

  # Add type and context columns
  pve$type &lt;- sapply(rownames(pve), function(x) { unlist(strsplit(x, &quot;[|]&quot;))[1] })
  pve$context &lt;- sapply(rownames(pve), function(x) { unlist(strsplit(x, &quot;[|]&quot;))[2] })

  # Remove rows with NA values and sort
  pve &lt;- na.omit(pve)
  pve &lt;- pve[order(rownames(pve)), ]

  # Aggregate PVE by type
  df_pve &lt;- aggregate(pve$pve, by = list(pve$type), FUN = sum)
  colnames(df_pve) &lt;- c(&quot;type&quot;, &quot;total_pve&quot;)
  df_pve$total_pve &lt;- round(df_pve$total_pve, 4)

  # Add context-specific columns
  for (context in unique(pve$context)) {
    context_pve &lt;- aggregate(pve$pve, by = list(pve$type, pve$context), FUN = sum)
    context_pve &lt;- context_pve[context_pve$Group.2 == context, ]
    colnames(context_pve)[3] &lt;- context
    df_pve &lt;- merge(df_pve, context_pve[, c(&quot;Group.1&quot;, context)], by.x = &quot;type&quot;, by.y = &quot;Group.1&quot;, all.x = TRUE)
  }

  # Insert SNP PVE
  SNP_row &lt;- c(&quot;SNP&quot;, SNP_pve, rep(0, ncol(df_pve) - 2))
  df_pve &lt;- rbind(df_pve, SNP_row)

  # Convert to numeric except for the type column
  df_pve[, -1] &lt;- lapply(df_pve[, -1], as.numeric)

  # Sum all rows and add a sum_pve row
  sum_row &lt;- colSums(df_pve[, -1], na.rm = TRUE)
  sum_row &lt;- c(&quot;total_pve&quot;, sum_row)
  df_pve &lt;- rbind(df_pve, sum_row)

  # Clean up row names and return
  row.names(df_pve) &lt;- NULL
  return(df_pve)
}

palette &lt;- c(brewer.pal(12, &quot;Paired&quot;), brewer.pal(12, &quot;Set3&quot;), brewer.pal(6, &quot;Dark2&quot;))

pve_pie_chart &lt;- function(pve_vector, palette=NULL, title) {
  # Create data frame for plotting
  data &lt;- data.frame(
    Group = names(pve_vector),
    Value = pve_vector
  )
  
  # Calculate percentages
  data$Percentage &lt;- round(100 * data$Value / sum(data$Value), 1)
  
  # Set palette if not specified
  if (is.null(palette)) {
    palette &lt;- brewer.pal(min(8, length(data$Group)), &quot;Set3&quot;)
  }
  
  # Create pie chart
  ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Group)) +
    geom_bar(stat = &quot;identity&quot;, width = 1, color = &quot;white&quot;) +
    coord_polar(theta = &quot;y&quot;) +  # This transforms the bar chart into a pie chart
    scale_fill_manual(values = palette, name = &quot;Group&quot;) +
    labs(title = title, x = NULL, y = NULL) +
    theme_void() +
    theme(plot.title = element_text(hjust = 0.5), 
          legend.position = &quot;right&quot;, 
          legend.title = element_text(face = &quot;bold&quot;), 
          legend.text = element_text(size = 12)) +
    geom_text(aes(label = paste(Percentage, &quot;%&quot;, sep=&quot;&quot;)), position = position_stack(vjust = 0.5))
}</code></pre>
<div id="settings" class="section level1">
<h1>Settings</h1>
<p>Weights:</p>
<p>single group: Liver, eQTL</p>
<p>multi group: Liver, Spleen, Adipose_Subcutaneous, Adrenal_Gland,
Esophagus_Mucosa; eQTL + sQTL</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = TRUE,</li>
<li>load_predictdb_LD = F,</li>
</ul>
<p>Main function</p>
<ul>
<li>niter_prefit = default,</li>
<li>niter = default,</li>
<li>pre-estimate L</li>
</ul>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="single-group" class="section level2">
<h2>Single group</h2>
<div id="parameters" class="section level3">
<h3>Parameters</h3>
<pre class="r"><code>results_dir_single &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/xxxintalk/results_predictdb_main_single/&quot;,trait,&quot;/&quot;)


finemap.res.single &lt;- readRDS(paste0(results_dir_single,trait,&quot;.ctwas.res.RDS&quot;))
snp_map.single &lt;- readRDS(paste0(results_dir_single,trait,&quot;.snp_map.RDS&quot;))
res.single &lt;- finemap.res.single$finemap_res


param.single &lt;- finemap.res.single$param
make_convergence_plots(param.single, gwas_n)</code></pre>
<p><img src="figure/LDL_single_multi_compare_oldpip.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_parameters.single &lt;- summarize_param(param.single, gwas_n)


group_size.single &lt;- data.frame(group = names(ctwas_parameters.single$group_size),
                         group_size = as.vector(ctwas_parameters.single$group_size))
group_size.single &lt;- t(group_size.single)
DT::datatable(group_size.single,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group size&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group size<\/caption>","data":[["group","group_size"],["eQTL|Liver","   8775"],["SNP","6027930"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>V1<\/th>\n      <th>V2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ctwas-results" class="section level3">
<h3>ctwas results</h3>
<pre class="r"><code>annotated_finemap_res.single &lt;- anno_finemap_res(finemap_res = res.single,
                                          snp_map = snp_map.single,
                                          gene_annot = E_S_A_mapping,
                                          use_gene_pos = &quot;mid&quot;,
                                          filter_protein_coding_genes = T,
                                          drop_unannotated_genes = T,
                                          filter_cs = T)</code></pre>
<pre><code>2024-08-29 10:44:10 INFO::Annotating ctwas finemapping result ...
2024-08-29 10:44:20 INFO::keep only protein coding genes
2024-08-29 10:44:21 INFO::keep only results in credible sets
2024-08-29 10:44:21 INFO::add gene_name and gene_type
2024-08-29 10:44:21 INFO::use gene mid positions
2024-08-29 10:44:21 INFO::add SNP positions</code></pre>
<pre class="r"><code>res_gene.single &lt;- annotated_finemap_res.single[annotated_finemap_res.single$type != &quot;SNP&quot;,]

combined_pip_by_context.single &lt;- combine_gene_pips_old(finemap_res = annotated_finemap_res.single,
                                             by = &quot;type&quot;, digits = 4)
highpip.single &lt;- combined_pip_by_context.single[combined_pip_by_context.single$combined_pip &gt; 0.8,]

DT::datatable(highpip.single,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Genes with PIP &gt; 0.8&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Genes with PIP &gt; 0.8<\/caption>","data":[["44","69","32","17","1","67","38","92","36","83","48","84","23","45","94","13","2","47","20","60","4","25","98","80","82","8","35","65","22","51","10","11","53"],["HPR","PSRC1","FADS1","CNIH4","ABCA1","PRKD2","GSK3B","TNKS","GAS6","SRRT","KDSR","STAT5B","CYP2A6","INHBB","TRIM39","C10orf88","ACVR1C","INSIG2","CSNK1G3","PARP9","ALLC","DDX56","USP53","SP4","SPTY2D1","ASAP3","FUT2","POP7","CWF19L1","LRRK2","BMPR2","BRI3","MZF1"],[1,1,0.998,0.991,0.9907,0.9887,0.9861,0.9757,0.9753,0.9731,0.9716,0.9692,0.9572,0.9546,0.9537,0.9503,0.9471,0.9459,0.9314,0.9138,0.9072,0.9052,0.8939,0.8866,0.8704,0.8648,0.856,0.8415,0.8389,0.8348,0.8324,0.819,0.8137],[1,1,0.998,0.991,0.9907,0.9887,0.9861,0.9757,0.9753,0.9731,0.9716,0.9692,0.9572,0.9546,0.9537,0.9503,0.9471,0.9459,0.9314,0.9138,0.9072,0.9052,0.8939,0.8866,0.8704,0.8648,0.856,0.8415,0.8389,0.8348,0.8324,0.819,0.8137]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="multi-group" class="section level2">
<h2>Multi group</h2>
<div id="parameters-1" class="section level3">
<h3>Parameters</h3>
<pre class="r"><code>results_dir_multi &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/xxxintalk/results_predictdb_main_multi/&quot;,trait,&quot;/&quot;)


finemap.res.multi &lt;- readRDS(paste0(results_dir_multi,trait,&quot;.ctwas.res.RDS&quot;))
snp_map.multi &lt;- readRDS(paste0(results_dir_multi,trait,&quot;.snp_map.RDS&quot;))
res.multi &lt;- finemap.res.multi$finemap_res


param.multi &lt;- finemap.res.multi$param
make_convergence_plots(param.multi, gwas_n,colors = palette)</code></pre>
<p><img src="figure/LDL_single_multi_compare_oldpip.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ctwas_parameters.multi &lt;- summarize_param(param.multi, gwas_n)


group_size.multi &lt;- data.frame(group = names(ctwas_parameters.multi$group_size),
                         group_size = as.vector(ctwas_parameters.multi$group_size))
group_size.multi &lt;- t(group_size.multi)
DT::datatable(group_size.multi,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group size&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group size<\/caption>","data":[["group","group_size"],["eQTL|Liver","   8775"],["eQTL|Spleen","   9749"],["eQTL|Adipose_Subcutaneous","  10538"],["eQTL|Adrenal_Gland","   9433"],["eQTL|Esophagus_Mucosa","  10580"],["sQTL|Liver","  18136"],["sQTL|Spleen","  27105"],["sQTL|Adipose_Subcutaneous","  30032"],["sQTL|Adrenal_Gland","  24941"],["sQTL|Esophagus_Mucosa","  25674"],["SNP","6721210"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>V1<\/th>\n      <th>V2<\/th>\n      <th>V3<\/th>\n      <th>V4<\/th>\n      <th>V5<\/th>\n      <th>V6<\/th>\n      <th>V7<\/th>\n      <th>V8<\/th>\n      <th>V9<\/th>\n      <th>V10<\/th>\n      <th>V11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>para.multi &lt;- sum_pve_across_types(ctwas_parameters.multi)
DT::datatable(para.multi,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Heritability contribution by contexts&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-93928d62269dcef60a51" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4"],["eQTL","sQTL","SNP","total_pve"],["0.0221","0.0163","0.0254","0.0638"],["0","7e-04","0","7e-04"],["1e-04","3e-04","0","4e-04"],["0.0023","0.0028","0","0.0051"],["0.0098","0.0056","0","0.0154"],["0.0099","0.0069","0","0.0168"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Adrenal_Gland<\/th>\n      <th>Esophagus_Mucosa<\/th>\n      <th>Liver<\/th>\n      <th>Spleen<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>pve_vector.multi &lt;- as.numeric(para.multi$total_pve[1:3])
names(pve_vector.multi) &lt;- para.multi$type[1:3]
pve_pie_chart(pve_vector.multi, title = &quot;Pie Chart of PVE across Types&quot;, palette)</code></pre>
<p><img src="figure/LDL_single_multi_compare_oldpip.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ctwas-results-1" class="section level3">
<h3>ctwas results</h3>
<pre class="r"><code>annotated_finemap_res.multi &lt;- anno_finemap_res(finemap_res = res.multi,
                                          snp_map = snp_map.multi,
                                          gene_annot = E_S_A_mapping,
                                          use_gene_pos = &quot;mid&quot;,
                                          filter_protein_coding_genes = T,
                                          drop_unannotated_genes = T,
                                          filter_cs = T)</code></pre>
<pre><code>2024-08-29 10:44:41 INFO::Annotating ctwas finemapping result ...
2024-08-29 10:44:46 INFO::keep only protein coding genes
2024-08-29 10:44:46 INFO::keep only results in credible sets
2024-08-29 10:44:46 INFO::add gene_name and gene_type
2024-08-29 10:44:46 INFO::split PIPs for traits mapped to multiple genes
2024-08-29 10:44:46 INFO::use gene mid positions
2024-08-29 10:44:46 INFO::add SNP positions</code></pre>
<pre class="r"><code>res_gene.multi &lt;- annotated_finemap_res.multi[annotated_finemap_res.multi$type != &quot;SNP&quot;,]


combined_pip_by_context.multi &lt;- combine_gene_pips_old(finemap_res = annotated_finemap_res.multi,
                                             by = &quot;type&quot;, digits = 4)
highpip.multi &lt;- combined_pip_by_context.multi[combined_pip_by_context.multi$combined_pip &gt; 0.8,]

DT::datatable(highpip.multi,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Genes with PIP &gt; 0.8&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-16707cd634e93674753d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Genes with PIP &gt; 0.8<\/caption>","data":[["1","60","252","15","134","185","251","125","118","272","31","240","68","163","39","20","174","248","275","117","270","89","16","168","123","222","8","98","42","137","111","161","47","2","228","75","171","265","128","146","155","245","69","170","85","53","141","51","190","181","239","135","208","223","108","263","216","24","9","273","107","178","19","13","266"],["ABCA1","DNAJC13","TRIM39","APOB","LDLR","PSRC1","TRAK2","INSIG2","HPR","ZGPAT","C10orf88","TIMD4","FADS1","PCSK9","CETP","ASGR1","PKN3","TNKS","ZNF227","HP","ZDHHC18","GAS6","APOC1","PELO","INHBB","SP4","ADH1B","GSK3B","CNIH4","LRCH4","HMGCR","PARP9","CSNK1G3","ABCA8","STAT5B","FCGRT","PHC1","USP53","KDSR","MYPOP","NPEPPS","TMEM57","FAM117B","PGS1","FUT2","CYP4F12","MITF","CYP2A6","R3HDM2","PRKD2","THOP1","LIPC","SEPT2","SPHK2","HLA-DQB2","USP1","SLC22A1","BCAT2","ADRB1","ZHX3","HLA-DQB1","POP7","ASAP3","ALLC","XPNPEP3"],[1.317,1.2751,1.1333,1,1,1,1,1,1,0.9999,0.9999,0.9998,0.9992,0.9985,0.9981,0.9968,0.9927,0.9917,0.988,0.9876,0.9868,0.9867,0.9867,0.9803,0.9797,0.9749,0.9733,0.9719,0.9687,0.9669,0.9669,0.9649,0.9625,0.9613,0.9587,0.958,0.957,0.9553,0.9537,0.9535,0.9524,0.9361,0.9198,0.9162,0.9156,0.9148,0.9137,0.911,0.9098,0.9015,0.8846,0.8842,0.8729,0.8623,0.8596,0.857,0.8441,0.8333,0.8307,0.8286,0.8241,0.8223,0.8213,0.8095,0.8062],[null,1.0248,0.9889,1,null,1,1,1,1,0.9999,0.9999,0.9998,0.9992,0.9985,0.9981,null,0.9927,0.9917,null,null,null,0.9867,0.9867,0.9803,0.9797,0.9749,null,0.9719,0.1097,null,null,null,0.9625,null,0.9587,0.0967,null,0.2043,0.9537,null,0.2682,0.073,0.9198,null,0.9156,0.3154,0.9137,0.911,null,0.8627,null,null,null,0.0052,0.0369,0.857,0.8441,null,0.8307,null,0.0014,0.8223,0.8213,0.8095,0.5096],[1.317,0.2503,0.1444,null,1,null,null,null,null,null,null,null,null,null,null,0.9968,null,null,0.988,0.9876,0.9868,null,null,null,null,null,0.9733,null,0.8589,0.9669,0.9669,0.9649,null,0.9613,null,0.8612,0.957,0.7511,null,0.9535,0.6842,0.8631,null,0.9162,null,0.5993,null,null,0.9098,0.0389,0.8846,0.8842,0.8729,0.8571,0.8227,null,null,0.8333,null,0.8286,0.8227,null,null,null,0.2967]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="comparing-the-results" class="section level2">
<h2>Comparing the results</h2>
<pre class="r"><code>overlap &lt;- highpip.multi[highpip.multi$gene_name %in% highpip.single$gene_name,]

sprintf(&quot;the number of genes reported by single group analysis: %s&quot;, nrow(highpip.single))</code></pre>
<pre><code>[1] &quot;the number of genes reported by single group analysis: 33&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;the number of genes reported by multi group analysis: %s&quot;, nrow(highpip.multi))</code></pre>
<pre><code>[1] &quot;the number of genes reported by multi group analysis: 65&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;the number of overlapped gene: %s&quot;, nrow(overlap))</code></pre>
<pre><code>[1] &quot;the number of overlapped gene: 24&quot;</code></pre>
<pre class="r"><code>DT::datatable(overlap,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;overlapped genes&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-37e46cf92521dcbed3f8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">overlapped genes<\/caption>","data":[["1","252","185","125","118","31","68","248","89","123","222","98","42","161","47","228","265","128","85","51","181","178","19","13"],["ABCA1","TRIM39","PSRC1","INSIG2","HPR","C10orf88","FADS1","TNKS","GAS6","INHBB","SP4","GSK3B","CNIH4","PARP9","CSNK1G3","STAT5B","USP53","KDSR","FUT2","CYP2A6","PRKD2","POP7","ASAP3","ALLC"],[1.317,1.1333,1,1,1,0.9999,0.9992,0.9917,0.9867,0.9797,0.9749,0.9719,0.9687,0.9649,0.9625,0.9587,0.9553,0.9537,0.9156,0.911,0.9015,0.8223,0.8213,0.8095],[null,0.9889,1,1,1,0.9999,0.9992,0.9917,0.9867,0.9797,0.9749,0.9719,0.1097,null,0.9625,0.9587,0.2043,0.9537,0.9156,0.911,0.8627,0.8223,0.8213,0.8095],[1.317,0.1444,null,null,null,null,null,null,null,null,null,null,0.8589,0.9649,null,null,0.7511,null,null,null,0.0389,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>multi_unique &lt;- combined_pip_by_context.multi[!combined_pip_by_context.multi$gene_name %in% overlap$gene_name,]
multi_unique &lt;- multi_unique[multi_unique$combined_pip &gt; 0.8,]
DT::datatable(multi_unique,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Unique genes -- multi group&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-99e959167faff8d1fc37" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-99e959167faff8d1fc37">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Unique genes -- multi group<\/caption>","data":[["60","15","134","251","272","240","163","39","20","174","275","117","270","16","168","8","137","111","2","75","171","146","155","245","69","170","53","141","190","239","135","208","223","108","263","216","24","9","273","107","266"],["DNAJC13","APOB","LDLR","TRAK2","ZGPAT","TIMD4","PCSK9","CETP","ASGR1","PKN3","ZNF227","HP","ZDHHC18","APOC1","PELO","ADH1B","LRCH4","HMGCR","ABCA8","FCGRT","PHC1","MYPOP","NPEPPS","TMEM57","FAM117B","PGS1","CYP4F12","MITF","R3HDM2","THOP1","LIPC","SEPT2","SPHK2","HLA-DQB2","USP1","SLC22A1","BCAT2","ADRB1","ZHX3","HLA-DQB1","XPNPEP3"],[1.2751,1,1,1,0.9999,0.9998,0.9985,0.9981,0.9968,0.9927,0.988,0.9876,0.9868,0.9867,0.9803,0.9733,0.9669,0.9669,0.9613,0.958,0.957,0.9535,0.9524,0.9361,0.9198,0.9162,0.9148,0.9137,0.9098,0.8846,0.8842,0.8729,0.8623,0.8596,0.857,0.8441,0.8333,0.8307,0.8286,0.8241,0.8062],[1.0248,1,null,1,0.9999,0.9998,0.9985,0.9981,null,0.9927,null,null,null,0.9867,0.9803,null,null,null,null,0.0967,null,null,0.2682,0.073,0.9198,null,0.3154,0.9137,null,null,null,null,0.0052,0.0369,0.857,0.8441,null,0.8307,null,0.0014,0.5096],[0.2503,null,1,null,null,null,null,null,0.9968,null,0.988,0.9876,0.9868,null,null,0.9733,0.9669,0.9669,0.9613,0.8612,0.957,0.9535,0.6842,0.8631,null,0.9162,0.5993,null,0.9098,0.8846,0.8842,0.8729,0.8571,0.8227,null,null,0.8333,null,0.8286,0.8227,0.2967]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>unique_detail &lt;- res_gene.multi[!res_gene.multi$gene_name %in%overlap$gene_name,]
unique_detail &lt;- unique_detail[unique_detail$gene_name %in% highpip.multi$gene_name,]
save(unique_detail, file = &quot;/project/xinhe/xsun/multi_group_ctwas/xxxintalk/unique_detail.rdata&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] dplyr_1.1.4               RColorBrewer_1.1-3       
 [3] ggplot2_3.5.1             EnsDb.Hsapiens.v86_2.99.0
 [5] ensembldb_2.20.2          AnnotationFilter_1.20.0  
 [7] GenomicFeatures_1.48.3    AnnotationDbi_1.58.0     
 [9] Biobase_2.56.0            GenomicRanges_1.48.0     
[11] GenomeInfoDb_1.39.9       IRanges_2.30.0           
[13] S4Vectors_0.34.0          BiocGenerics_0.42.0      
[15] ctwas_0.4.7              

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                DT_0.22                    
 [11] ggrepel_0.9.1               bit64_4.0.5                
 [13] fansi_1.0.3                 xml2_1.3.3                 
 [15] codetools_0.2-18            logging_0.10-108           
 [17] cachem_1.0.6                knitr_1.39                 
 [19] jsonlite_1.8.0              workflowr_1.7.0            
 [21] Rsamtools_2.12.0            dbplyr_2.1.1               
 [23] png_0.1-7                   readr_2.1.2                
 [25] compiler_4.2.0              httr_1.4.3                 
 [27] assertthat_0.2.1            Matrix_1.5-3               
 [29] fastmap_1.1.0               lazyeval_0.2.2             
 [31] cli_3.6.1                   later_1.3.0                
 [33] htmltools_0.5.2             prettyunits_1.1.1          
 [35] tools_4.2.0                 gtable_0.3.0               
 [37] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [39] rappdirs_0.3.3              Rcpp_1.0.12                
 [41] jquerylib_0.1.4             vctrs_0.6.5                
 [43] Biostrings_2.64.0           rtracklayer_1.56.0         
 [45] crosstalk_1.2.0             xfun_0.41                  
 [47] stringr_1.5.1               lifecycle_1.0.4            
 [49] irlba_2.3.5                 restfulr_0.0.14            
 [51] XML_3.99-0.14               zlibbioc_1.42.0            
 [53] zoo_1.8-10                  scales_1.3.0               
 [55] gggrid_0.2-0                hms_1.1.1                  
 [57] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [59] ProtGenerics_1.28.0         parallel_4.2.0             
 [61] SummarizedExperiment_1.26.1 LDlinkR_1.2.3              
 [63] yaml_2.3.5                  curl_4.3.2                 
 [65] memoise_2.0.1               sass_0.4.1                 
 [67] biomaRt_2.54.1              stringi_1.7.6              
 [69] RSQLite_2.3.1               highr_0.9                  
 [71] BiocIO_1.6.0                filelock_1.0.2             
 [73] BiocParallel_1.30.3         rlang_1.1.2                
 [75] pkgconfig_2.0.3             matrixStats_0.62.0         
 [77] bitops_1.0-7                evaluate_0.15              
 [79] lattice_0.20-45             purrr_1.0.2                
 [81] labeling_0.4.2              GenomicAlignments_1.32.0   
 [83] htmlwidgets_1.5.4           cowplot_1.1.1              
 [85] bit_4.0.4                   tidyselect_1.2.0           
 [87] magrittr_2.0.3              R6_2.5.1                   
 [89] generics_0.1.2              DelayedArray_0.22.0        
 [91] DBI_1.2.2                   withr_2.5.0                
 [93] pgenlibr_0.3.3              pillar_1.9.0               
 [95] KEGGREST_1.36.3             RCurl_1.98-1.7             
 [97] mixsqp_0.3-43               tibble_3.2.1               
 [99] crayon_1.5.1                utf8_1.2.2                 
[101] BiocFileCache_2.4.0         plotly_4.10.0              
[103] tzdb_0.4.0                  rmarkdown_2.25             
[105] progress_1.2.2              grid_4.2.0                 
[107] data.table_1.14.2           blob_1.2.3                 
[109] git2r_0.30.1                digest_0.6.29              
[111] tidyr_1.3.0                 httpuv_1.6.5               
[113] munsell_0.5.0               viridisLite_0.4.0          
[115] bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
