<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-09-22" />

<title>An example for LDL ~ Predictdb eQTL&amp;sQTL from Liver</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">An example for LDL ~ Predictdb
eQTL&amp;sQTL from Liver</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-09-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed7cd32f03668398183172e5fb41ab69462705fcbtargetblankd7cd32fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d7cd32f03668398183172e5fb41ab69462705fcb" target="_blank">d7cd32f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed7cd32f03668398183172e5fb41ab69462705fcbtargetblankd7cd32fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d7cd32f03668398183172e5fb41ab69462705fcb" target="_blank">d7cd32f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/figure/
    Ignored:    results/

Untracked files:
    Untracked:  analysis/LDL_predictdb_esQTL.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>We present a sample cTWAS report based on real data analysis. The
analyzed trait is LDL cholesterol, the prediction models are liver gene
expression and splicing models trained on GTEx v8 in the PredictDB
format.</p>
<div id="analysis-settings" class="section level1">
<h1>Analysis settings</h1>
<div id="input-data" class="section level2">
<h2>Input data</h2>
<ul>
<li>GWAS Z-scores</li>
</ul>
<p>The analyzed trait is LDL cholesterol. The summary statistics are
downloaded from <a href="https://gwas.mrcieu.ac.uk"
class="uri">https://gwas.mrcieu.ac.uk</a>, using dataset ID:
<code>ukb-d-30780_irnt</code>. The number of SNPs it contains is
13,586,016.</p>
<p>The sample size is</p>
<pre><code>[1] &quot;gwas_n = 343621&quot;</code></pre>
<ul>
<li>Prediction models</li>
</ul>
<p>The prediction models used in this analysis are liver gene expression
and splicing models, trained on GTEx v8 in the PredictDB format. These
models can be downloaded from <a
href="https://predictdb.org/post/2021/07/21/gtex-v8-models-on-eqtl-and-sqtl/"
class="uri">https://predictdb.org/post/2021/07/21/gtex-v8-models-on-eqtl-and-sqtl/</a></p>
<pre><code>[1] &quot;The number of eQTLs per gene = 1.5078&quot;</code></pre>
<pre><code>[1] &quot;Total genes = 12714&quot;</code></pre>
<pre><code>[1] &quot;The number of sQTLs per intron = 1.2151&quot;</code></pre>
<pre><code>[1] &quot;Total introns = 29250&quot;</code></pre>
<ul>
<li>Reference data</li>
</ul>
<p>The reference data include genomic region definitions and an LD
reference. We use the genomic regions provided by the package and the LD
reference in b38, located at
<code>/project2/mstephens/wcrouse/UKB_LDR_0.1/</code>. Alternatively,
the LD reference can be downloaded from this link:<a
href="https://uchicago.app.box.com/s/jqocacd2fulskmhoqnasrknbt59x3xkn"
class="uri">https://uchicago.app.box.com/s/jqocacd2fulskmhoqnasrknbt59x3xkn</a>.</p>
</div>
<div id="data-processing-and-harmonization" class="section level2">
<h2>Data processing and harmonization</h2>
<p>We map the reference SNPs and LD matrices to regions following the
instructions from the cTWAS tutorial.</p>
<p>When processing z-scores, we exclude multi-allelic and
strand-ambiguous variants by setting
<code>drop_multiallelic = TRUE</code> and
<code>drop_strand_ambig = TRUE</code>.</p>
<p>Users can expand the code snippets below to view the exact code
used.</p>
<pre class="r"><code>## input data

weight_files &lt;- c(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/expression_models/expression_Liver.db&quot;,&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/splicing_models/splicing_Liver.db&quot;)
z_snp_file &lt;- &quot;/project2/xinhe/shared_data/multigroup_ctwas/gwas/ctwas_inputs_zsnp/LDL-ukb-d-30780_irnt.z_snp.RDS&quot;
region_file &lt;- system.file(&quot;extdata/ldetect&quot;, paste0(&quot;EUR.&quot;, genome_version, &quot;.ldetect.regions.RDS&quot;), package = &quot;ctwas&quot;)
region_info &lt;- readRDS(region_file)
genome_version &lt;- &quot;b38&quot;
LD_dir &lt;- &quot;/project2/mstephens/wcrouse/UKB_LDR_0.1/&quot;

## output dir
outputdir &lt;- &quot;/project/xinhe/xsun/multi_group_ctwas/examples/results_predictdb_main/LDL-ukb-d-30780_irnt/&quot;
dir.create(outputdir, showWarnings=F, recursive=T)

## other parameters
ncore &lt;- 5

## process inputs

### Preprocess LD_map &amp; SNP_map
region_metatable &lt;- region_info
region_metatable$LD_file &lt;- file.path(LD_dir, paste0(LD_filestem, &quot;.RDS&quot;))
region_metatable$SNP_file &lt;- file.path(LD_dir, paste0(LD_filestem, &quot;.Rvar&quot;))
  
res &lt;- create_snp_LD_map(region_metatable)
  
region_info &lt;- res$region_info
snp_map &lt;- res$snp_map
LD_map &lt;- res$LD_map

### Preprocess GWAS z-scores
z_snp &lt;- readRDS(z_snp_file)
z_snp &lt;- preprocess_z_snp(z_snp = z_snp, 
                          snp_map = snp_map, 
                          drop_multiallelic = TRUE, 
                          drop_strand_ambig = TRUE)

### Preprocess weights
weights_expression1 &lt;- preprocess_weights(weight_file = weight_files[1],
                                              region_info = region_info,
                                              gwas_snp_ids = z_snp$id,
                                              snp_map = snp_map,
                                              LD_map = LD_map,
                                              type = &quot;eQTL&quot;,
                                              context = tissue,
                                              weight_format = &quot;PredictDB&quot;,
                                              drop_strand_ambig = TRUE,
                                              scale_predictdb_weights = T,
                                              load_predictdb_LD = F, #### F for fusion converted weights
                                              filter_protein_coding_genes = TRUE,
                                              ncore = ncore)
    
weights_splicing1 &lt;- preprocess_weights(weight_file = weight_files[2],
                                            region_info = region_info,
                                            gwas_snp_ids = z_snp$id,
                                            snp_map = snp_map,
                                            LD_map = LD_map,
                                            type = &quot;sQTL&quot;,
                                            context = tissue,
                                            weight_format = &quot;PredictDB&quot;,
                                            drop_strand_ambig = TRUE,
                                            scale_predictdb_weights = T, #### F for fusion converted weights
                                            load_predictdb_LD = F, 
                                            filter_protein_coding_genes = TRUE,
                                            ncore = ncore)    
    
weights &lt;- c(weights_expression1,weights_splicing1)    </code></pre>
</div>
</div>
<div id="running-ctwas-analysis" class="section level1">
<h1>Running cTWAS analysis</h1>
<p>We use the function ctwas_sumstats() to run cTWAS analysis with LD.
Details are shown in the tutorial. <a
href="https://xinhe-lab.github.io/multigroup_ctwas/articles/running_ctwas_analysis.html#running-ctwas-main-function"
class="uri">https://xinhe-lab.github.io/multigroup_ctwas/articles/running_ctwas_analysis.html#running-ctwas-main-function</a></p>
<p>The arguments are all in defaul settings, more specifically,</p>
<ul>
<li>we set <code>group_prior_var_structure = "shared_type"</code> to
allow all groups in one molecular QTL type to share the same variance
parameter</li>
<li>we first estimate the number of causal signals (L) for each region
by setting <code>filter_L = TRUE</code></li>
<li>The package will not compute the non-SNP PIP by setting
<code>filter_nonSNP_PIP = TRUE</code></li>
<li>We select regions with non-SNP PIP &gt; 0.5 by setting
<code>min_nonSNP_PIP = 0.5</code></li>
</ul>
<p>We use the function <code>ctwas_sumstats()</code> to run the cTWAS
analysis with LD. For more details on this function, refer to the cTWAS
tutorial: <a
href="https://xinhe-lab.github.io/multigroup_ctwas/articles/running_ctwas_analysis.html#running-ctwas-main-function"
class="uri">https://xinhe-lab.github.io/multigroup_ctwas/articles/running_ctwas_analysis.html#running-ctwas-main-function</a></p>
<p>All arguments are set to their default values, with the following
specific settings:</p>
<ul>
<li><code>group_prior_var_structure = "shared_type"</code>: Allows all
groups within a molecular QTL type to share the same variance
parameter.</li>
<li><code>filter_L = TRUE</code>: Estimates the number of causal signals
(L) for each region.</li>
<li><code>filter_nonSNP_PIP = TRUE</code>: Prevents the computation of
non-SNP PIP values.</li>
<li><code>min_nonSNP_PIP = 0.5</code>: Selects regions where the non-SNP
PIP is greater than 0.5.</li>
</ul>
<p>Users can expand the code snippets below to view the exact code
used.</p>
<pre class="r"><code>thin &lt;- 0.1
maxSNP &lt;- 20000

ctwas_res &lt;- ctwas_sumstats(z_snp, 
                            weights, 
                            region_info, 
                            LD_map, 
                            snp_map, 
                            thin = thin,
                            maxSNP = maxSNP,
                            group_prior_var_structure = &quot;shared_type&quot;, 
                            filter_L = TRUE,
                            filter_nonSNP_PIP = FALSE,
                            min_nonSNP_PIP = 0.5,
                            ncore = ncore, 
                            ncore_LD = ncore,
                            save_cor = TRUE, 
                            cor_dir = paste0(outputdir,&quot;/cor_matrix&quot;),
                            verbose = T)</code></pre>
</div>
<div id="ctwas-results" class="section level1">
<h1>cTWAS results</h1>
<p><code>ctwas_res</code> is the object contains the outputs of
cTWAS</p>
<div id="parameter-estimation" class="section level2">
<h2>Parameter estimation</h2>
<p>We extract the estimated parameters by
<code>param &lt;- ctwas_res$param</code></p>
<p>we make plots using the function
<code>make_convergence_plots(param, gwas_n)</code> to see how estimated
parameters converge during the execution of the program:</p>
<p><img src="figure/LDL_predictdb_esQTL.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>These plots show the estimated prior inclusion probability, prior
effect size variance, enrichment and proportion of variance explained
(PVE) over the iterations of parameter estimation. The enrichment is
defined as the ratio of the prior inclusion probability of molecular
traits over the prior inclusion probability of variants. We generally
expect molecular traits to have higher prior inclusion probability than
variants. Enrichment values typically range from 20 - 100 for expression
traits.</p>
<p>Then, we use <code>summarize_param(param, gwas_n)</code> to obtain
estimated parameters (from the last iteration) and to compute the PVE by
variants and molecular traits.</p>
<pre><code>[1] &quot;The number of genes/introns/SNPs used in the analysis is:&quot;</code></pre>
<pre><code>Liver|eQTL Liver|sQTL        SNP 
      8775      18136    7405450 </code></pre>
<p><code>ctwas_parameters$attributable_pve</code> contains the
proportion of heritability mediated by molecular traits and variants, we
visualize it using pie chart.</p>
<pre class="r"><code>data &lt;- data.frame(
  category = names(ctwas_parameters$attributable_pve),
  percentage = ctwas_parameters$attributable_pve
)

# Calculate percentage labels for the chart
data$percentage_label &lt;- paste0(round(data$percentage * 100, 1), &quot;%&quot;)

ggplot(data, aes(x = &quot;&quot;, y = percentage, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() +  # Remove background and axes
  geom_text(aes(label = percentage_label), 
            position = position_stack(vjust = 0.5), size = 5) +
  scale_fill_manual(values = c(&quot;#FF9999&quot;, &quot;#66B2FF&quot;, &quot;#99FF99&quot;)) +  # Custom colors
  labs(fill = &quot;Category&quot;) +
  ggtitle(&quot;Attributable PVE&quot;)</code></pre>
<p><img src="figure/LDL_predictdb_esQTL.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fine-mapping-results" class="section level2">
<h2>Fine-mapping results</h2>
<p>We process the fine-mapping results here.</p>
<p>We first add gene annotations to cTWAS results</p>
<pre class="r"><code>load(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/E_S_A_mapping_updated.RData&quot;)
colnames(E_S_A_mapping)[1] &lt;- &quot;molecular_id&quot;

finemap_res &lt;- ctwas_res$finemap_res 
finemap_res$molecular_id &lt;- get_molecular_ids(finemap_res)

snp_map &lt;- readRDS(paste0(results_dir,trait,&quot;.snp_map.RDS&quot;))

finemap_res &lt;- anno_finemap_res(finemap_res,
                                         snp_map = snp_map,
                                         mapping_table = E_S_A_mapping,
                                         add_gene_annot = TRUE,
                                         map_by = &quot;molecular_id&quot;,
                                         drop_unmapped = TRUE,
                                         add_position = TRUE,
                                         use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-09-23 16:25:07 INFO::Annotating fine-mapping result ...
2024-09-23 16:25:08 INFO::Map molecular traits to genes
2024-09-23 16:25:11 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-09-23 16:25:16 INFO::Add gene positions
2024-09-23 16:25:16 INFO::Add SNP positions</code></pre>
<pre class="r"><code>DT::datatable(finemap_res[finemap_res$susie_pip &gt; 0.8,],caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;The annotated fine-mapping results, ones with susie pip &gt; 0.8 are shown&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">The annotated fine-mapping results, ones with susie pip &gt; 0.8 are shown<\/caption>","data":[["47","294","586","639","854","892","913","1372","1833","1997","2704","2779","3076","3083","3086","3158","4028","4121","4223","4269","4806","4817","5331","5816","6137","6156","6526","6952","7108","7250","7562","7776","7900","8531","8565","8097100","16184","16235","91103","91264","91707","101720","105386","271338","287248","357335","368606","400090","454446","454539","455843","455856","519962","553801","581215","581217","601098","721198","760082","807584","949033","991969","992195","992216","1011888","1027155","1027319","1027347","1027602","1027838","1027937","1027957","1045175","1058211","1058582","1058655","1060904","1060905","1060907","1060943","1060952","1062916","1063366","1112278","1116947","1127288","1132500","1143936"],["ENSG00000088280.18|Liver_eQTL","ENSG00000134222.16|Liver_eQTL","ENSG00000143771.11|Liver_eQTL","ENSG00000151360.9|Liver_eQTL","ENSG00000125629.14|Liver_eQTL","ENSG00000163083.5|Liver_eQTL","ENSG00000123612.15|Liver_eQTL","ENSG00000082701.14|Liver_eQTL","ENSG00000151292.17|Liver_eQTL","ENSG00000204599.14|Liver_eQTL","ENSG00000105866.13|Liver_eQTL","ENSG00000136271.10|Liver_eQTL","intron_7_100575304_100575705|Liver_sQTL","ENSG00000172336.4|Liver_eQTL","ENSG00000087087.18|Liver_eQTL","ENSG00000173273.15|Liver_eQTL","ENSG00000095485.16|Liver_eQTL","ENSG00000119965.12|Liver_eQTL","ENSG00000179119.14|Liver_eQTL","ENSG00000149485.18|Liver_eQTL","intron_12_8921750_8922633|Liver_sQTL","ENSG00000188906.14|Liver_eQTL","ENSG00000183087.14|Liver_eQTL","intron_15_58431776_58431993|Liver_sQTL","ENSG00000261701.6|Liver_eQTL","intron_16_72057466_72058254|Liver_sQTL","ENSG00000173757.9|Liver_eQTL","ENSG00000119537.15|Liver_eQTL","intron_19_11120522_11123174|Liver_sQTL","intron_19_15685197_15696430|Liver_sQTL","ENSG00000255974.7|Liver_eQTL","ENSG00000105287.12|Liver_eQTL","intron_19_48800297_48810984|Liver_sQTL","intron_20_63737360_63737533|Liver_sQTL","intron_21_39178647_39180285|Liver_sQTL","rs75460349","rs11591147","rs499883","rs13027175","rs113206087","rs533617","rs1260326","rs75331444","rs9884390","rs1229984","rs78912080","rs3129059","rs1049110","rs12208357","rs2297374","rs140570886","rs73596816","rs763798411","rs117139027","rs112875651","rs75164422","rs3780181","rs11601507","rs11604424","rs568773603","rs2908806","rs77542162","rs60066663","rs72853625","rs77960347","rs17248748","rs201091555","rs147540853","rs379309","rs146898772","rs36022659","rs148356565","rs58542926","rs55840997","rs62115559","rs185920692","rs429358","rs7412","rs1065853","rs157595","rs8106813","rs12981080","rs73045960","rs117113213","rs1800961","rs11480320","rs202143810","rs34079499"],["eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","sQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","eQTL","sQTL","eQTL","eQTL","sQTL","eQTL","sQTL","eQTL","eQTL","sQTL","sQTL","eQTL","eQTL","sQTL","sQTL","sQTL","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP"],["Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP"],["Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|sQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|eQTL","Liver|sQTL","Liver|eQTL","Liver|eQTL","Liver|sQTL","Liver|eQTL","Liver|sQTL","Liver|eQTL","Liver|eQTL","Liver|sQTL","Liver|sQTL","Liver|eQTL","Liver|eQTL","Liver|sQTL","Liver|sQTL","Liver|sQTL","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP","SNP"],["1_22760390_23594100","1_107867043_109761309","1_222056878_224750818","2_1777250_3926189","2_117609890_120546207","2_120546207_122545330","2_156704123_157676706","3_118811404_120804208","5_123221211_124462407","6_29770194_30830391","7_20085285_22468010","7_43119475_44724229","7_99117851_100599028","7_100599028_101555972","7_100599028_101555972","8_9297184_9783277","10_98908643_101189482","10_122141030_124109830","11_17556855_19547988","11_59013076_62456299","12_8878799_10328935","12_38833367_40422383","13_112918174_114344378","15_58149167_59401917","16_71020125_72901251","16_71020125_72901251","17_41743558_43694719","18_62610482_64153018","19_9127717_11173352","19_15537799_16263605","19_40478694_41625199","19_46646825_48778970","19_48778970_51029311","20_63558827_64333810","21_39110976_40017600","1_25190354_27075376","1_53760589_55947444","1_53760589_55947444","2_19492643_20850730","2_19492643_20850730","2_20850730_23118512","2_26672117_28375910","2_43082452_44086664","4_67989047_70183435","4_98502916_99757203","6_24852047_25684359","6_28949831_29770194","6_32714887_33268720","6_157797687_160159465","6_157797687_160159465","6_160159465_161748532","6_160159465_161748532","7_104519078_106042459","8_28304875_29470379","8_125398675_127646866","8_125398675_127646866","9_1916877_3190064","11_4725479_6301639","11_116512631_117876395","12_114598797_115065411","17_7414079_8403107","17_66804312_69862629","17_66804312_69862629","17_66804312_69862629","18_48413361_50204214","19_9127717_11173352","19_9127717_11173352","19_9127717_11173352","19_11173352_13360313","19_11173352_13360313","19_11173352_13360313","19_11173352_13360313","19_18299052_19766662","19_43358303_44239955","19_43358303_44239955","19_43358303_44239955","19_44239955_45599439","19_44239955_45599439","19_44239955_45599439","19_44239955_45599439","19_44239955_45599439","19_45599439_46646825","19_45599439_46646825","20_39808119_40982216","20_44051536_46210417","20_53856010_55438728","20_63558827_64333810","21_39110976_40017600"],[5.2832247749124,-41.6873361207573,6.14553515481309,4.91906561985035,-8.98270179154181,-8.51893560227002,-4.68737018702948,6.83567550303477,9.11629087525006,8.8401635359729,10.6931912889074,9.64186135796621,5.29421762596956,-5.84525842771095,5.42499614580627,11.0385643985833,5.74066764441767,-6.78784967909072,-5.55712270803787,12.9263513117373,6.15579232441244,5.03364215570486,-8.92368839427658,-7.78263607107692,-17.9627704587969,21.8686484537068,5.42625212751111,-4.52628668210626,26.8978244543332,-5.86783602356772,5.4070280028537,5.07221668872336,-5.19744901635721,5.52831225948544,2.93096388502247,-7.38009256904532,39.1649339702164,-16.1075268817204,10.2778229562492,9.3149045743714,19.8793679278201,14.5534972586808,20.5689117653213,-8.97642569278505,-6.83342737132468,9.50288862018003,-6.18257929479955,-7.87884690299961,-12.2823366985935,12.1550343014419,-11.7119638451811,-5.31376286694625,-3.27214912280702,6.61766007345458,24.2936163238548,5.19593593452339,6.02814104324678,-7.24133431914061,14.1908204323808,5.41258969341161,6.0263590430389,-16.6257239228664,8.90324774698181,6.59500553821367,-6.97832058073377,6.71887503067419,29.0261292801639,11.4190458955691,8.42698061883713,-3.66066684808667,9.8735131799952,10.8754182853187,24.2286893890098,8.3318256674321,14.9460446936868,7.0968885387948,-59.2713178294574,110.748596528316,110.915451621235,-30.7695101283189,-4.44395987788923,9.71456344998233,12.8179108941561,-15.4868569394173,8.89695748180862,-5.86967903024445,5.16302276428398,-2.56872977346278],[0.852900550934959,0.999981986518198,0.972398420814004,0.911144989402156,0.959101765677439,0.956550205798665,0.945929810744478,0.981761971313114,0.927992629728818,0.961704863961651,0.867610769869974,0.916734127710682,0.978530305398574,0.848094478950196,0.851637143229038,0.976921005162492,0.842896288602277,0.949096511261486,0.874179062575899,0.933526948393775,0.842171608660238,0.8386836776066,0.974752972467129,0.916701126594213,0.999998874018692,0.985476371248342,0.966860619106787,0.96830115209781,1,0.872715393621341,0.95573910726706,0.973813319632806,0.862072076670389,0.999999994148804,0.961117881730878,0.901572033089439,1,0.998170464652193,0.999932614105255,0.996116124691665,0.996579077330143,0.880972498430026,0.844397413273583,0.937244933691764,0.946858493956244,0.999917411879216,0.971265807764492,0.843657988185609,0.996745193155753,0.993452951220781,0.954911714686923,0.999326702789982,1,0.993541365187144,1,0.999614145415628,0.977965669493387,0.999719514344415,0.999996630886902,0.927812628714357,0.917805577091638,0.999999999999999,0.999999984760916,0.929792402581132,0.96746456030383,0.907884792679081,0.825124983090491,0.802332611234229,0.999952573621309,0.915909341153808,0.999446203226608,0.999999999998685,0.999700913573795,0.992811178969743,0.99800565630558,0.965154645972753,1,0.852589141367018,0.999999956694112,0.894917704897637,1,0.928312915834236,0.99988809856019,0.998607660483333,0.999999924652878,0.824023481377446,1,1],[28.2342385024315,1497.18006943862,33.4453018669802,21.7613698721591,48.1536125641755,63.4116569058534,19.8450135446288,41.1587599405085,72.4817187852845,70.9987656745198,88.3174211624068,50.5431773250404,32.4629434478467,37.2274549281903,29.2204597468525,75.2448810234474,29.3019428132306,40.5977785345225,27.5165555749813,144.790663346015,36.1967936379853,22.7431934275594,69.4901815129855,65.357933995547,161.246897798172,292.151318742873,26.2789725577986,18.5671643253436,859.285090452782,32.9776655910382,26.0996631590447,25.2999824381184,26.0806006256436,3950.92914485967,926.094447495508,47.2554469038574,1225.3720545905,197.212615041162,89.9820433405661,85.5001777263637,281.550565352055,178.493422518528,291.601391393014,81.1977219819348,40.0655796813747,76.6277213035148,42.5980612184037,40.6148464495248,216.666392566262,141.819153246043,145.239800520292,41.9256293502064,2535.44129438759,37.6322727165854,592.379819203159,75.7544023074187,31.3820131031816,44.2057184603432,280.123845341867,25.4776652926618,33.9207905351783,228.624081129316,66.8144937981684,41.2241283565996,42.141893939372,65.0045346281805,666.465530131294,142.621884003769,67.5724604909836,65.5767203240987,137.625033358184,95.9590400520823,514.450519601179,63.5170594140945,192.805018641378,50.8083233838882,1831.92851435218,5367.34340725778,5398.29578256723,391.072612477874,244.8425476373,56.7103714534458,114.038712182789,168.939419969987,67.2803102320155,29.8013240488505,3572.36067537328,880.334643017219],[1,1,1,1,1,1,1,1,1,1,1,2,1,3,2,1,1,1,1,1,1,1,1,1,2,1,1,1,2,1,1,2,1,1,1,2,1,2,4,2,3,1,2,1,1,1,2,2,1,2,2,3,1,1,1,2,1,1,1,0,1,1,2,3,1,0,4,3,3,4,2,1,1,2,1,3,2,5,5,4,3,4,1,1,1,1,2,2],["ENSG00000088280.18","ENSG00000134222.16","ENSG00000143771.11","ENSG00000151360.9","ENSG00000125629.14","ENSG00000163083.5","ENSG00000123612.15","ENSG00000082701.14","ENSG00000151292.17","ENSG00000204599.14","ENSG00000105866.13","ENSG00000136271.10","intron_7_100575304_100575705","ENSG00000172336.4","ENSG00000087087.18","ENSG00000173273.15","ENSG00000095485.16","ENSG00000119965.12","ENSG00000179119.14","ENSG00000149485.18","intron_12_8921750_8922633","ENSG00000188906.14","ENSG00000183087.14","intron_15_58431776_58431993","ENSG00000261701.6","intron_16_72057466_72058254","ENSG00000173757.9","ENSG00000119537.15","intron_19_11120522_11123174","intron_19_15685197_15696430","ENSG00000255974.7","ENSG00000105287.12","intron_19_48800297_48810984","intron_20_63737360_63737533","intron_21_39178647_39180285","rs75460349","rs11591147","rs499883","rs13027175","rs113206087","rs533617","rs1260326","rs75331444","rs9884390","rs1229984","rs78912080","rs3129059","rs1049110","rs12208357","rs2297374","rs140570886","rs73596816","rs763798411","rs117139027","rs112875651","rs75164422","rs3780181","rs11601507","rs11604424","rs568773603","rs2908806","rs77542162","rs60066663","rs72853625","rs77960347","rs17248748","rs201091555","rs147540853","rs379309","rs146898772","rs36022659","rs148356565","rs58542926","rs55840997","rs62115559","rs185920692","rs429358","rs7412","rs1065853","rs157595","rs8106813","rs12981080","rs73045960","rs117113213","rs1800961","rs11480320","rs202143810","rs34079499"],["ASAP3","PSRC1","CNIH4","ALLC","INSIG2","INHBB","ACVR1C","GSK3B","CSNK1G3","TRIM39","SP4","DDX56","LRCH4","POP7","SRRT","TNKS","CWF19L1","C10orf88","SPTY2D1","FADS1","PHC1","LRRK2","GAS6","LIPC","HPR","HP","STAT5B","KDSR","LDLR","CYP4F12","CYP2A6","PRKD2","BCAT2","LIME1","PSMG1",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["23428563","109279556","224356850","3658195","118088452","120346143","157526767","119821323","123512099","30327259","21428034","44565417","100575304","100706053","100875126","9556521","100232298","122930903","18606401","61799625","8921750","40196744","113820558","58431776","72063224","72057466","42199168","63327726","11120522","15685197","40843538","46674275","48800297","63737360","39178647",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["23484568","109283186","224379459","3702584","118110997","120351808","157629005","120094417","123617045","30343729","21514822","44575051","100575705","100707495","100888664","9782346","100267680","122954380","18634791","61829318","8922633","40369285","113864067","58431993","72077246","72058254","42276707","63367327","11123174","15696430","40850447","46717127","48810984","63737533","39180285",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["1","1","1","2","2","2","2","3","5","6","7","7","7","7","7","8","10","10","11","11","12","12","13","15","16","16","17","18","19","19","19","19","19","20","21","1","1","1","2","2","2","2","2","4","4","6","6","6","6","6","6","6","7","8","8","8","9","11","11","12","17","17","17","17","18","19","19","19","19","19","19","19","19","19","19","19","19","19","19","19","19","19","19","20","20","20","20","21"],[23456566,109281371,224368154,3680390,118099724,120348976,157577886,119957870,123564572,30335494,21471428,44570234,100575504,100706774,100881895,9669434,100249989,122942642,18620596,61814472,8922192,40283014,113842312,58431884,72070235,72057860,42237938,63347526,11121848,15690814,40846992,46695701,48805640,63737446,39179466,26853597,55039974,55053501,20732983,20848277,21011100,27508073,43842633,68507689,99318162,25616225,29679910,32759026,160122116,160154953,160591981,160596331,104982257,29167426,125494452,125494856,2640759,5679844,116780399,114780030,7660509,69085137,69200002,69209833,49583585,11095364,11134395,11137428,11173626,11263017,11282627,11287746,19268740,43976827,44145584,44214466,44908684,44908822,44909976,44922203,44928401,45614274,45791965,40537052,44413724,53916157,63718338,39301893]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>type<\/th>\n      <th>context<\/th>\n      <th>group<\/th>\n      <th>region_id<\/th>\n      <th>z<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>cs_index<\/th>\n      <th>molecular_id<\/th>\n      <th>gene_name<\/th>\n      <th>gene_type<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>chrom<\/th>\n      <th>pos<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,16]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>For all genes analyzed, we compare the z-scores and fine-mapping
PIPs</p>
<pre class="r"><code>ggplot(data = finemap_res[finemap_res$type!=&quot;SNP&quot;,], aes(x = abs(z), y = susie_pip)) +
  geom_point() +
  labs(x = &quot;abs(z-scores)&quot;, y = &quot;PIPs&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/LDL_predictdb_esQTL.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Next, we compute gene PIPs across different types of molecular
traits</p>
<pre class="r"><code>combined_pip_by_type &lt;- combine_gene_pips(finemap_res,
                                          group_by = &quot;gene_name&quot;,
                                          by = &quot;type&quot;,
                                          method = &quot;combine_cs&quot;,
                                          filter_cs = TRUE)</code></pre>
<pre><code>2024-09-23 16:25:30 INFO::Limit gene results to credible sets</code></pre>
<pre class="r"><code>DT::datatable(combined_pip_by_type,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Gene PIPs, only genes within credible sets are shown&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Gene PIPs, only genes within credible sets are shown<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391"],["LDLR","LIME1","HPR","PSRC1","ASGR1","HP","DOCK7","GSK3B","LRCH4","TNKS","GAS6","PRKD2","PARP9","CNIH4","KDSR","STAT5B","TRIM39","PSMG1","INSIG2","INHBB","CYP2A6","C10orf88","ACVR1C","TMEM57","USP53","FADS1","SRRT","CSNK1G3","DDX56","LIPC","ALLC","SPTY2D1","CYP4F12","SP4","BCAT2","ASAP3","POP7","CWF19L1","PHC1","BRI3","LRRK2","MZF1","FUT2","THOP1","SPHK2","PELO","PRMT6","PGS1","VIL1","R3HDM2","ERGIC3","FURIN","INPP5E","LRCOL1","ACMSD","CCDC57","SLC22A18","TMEM199","PPP6R2","XPNPEP3","HLA-B","HLA-C","SCAMP3","DENND4C","ABHD12","LRRC8A","TCTN3","PLCG1-AS1","ZHX3","MDH2","NPC1L1","DPP3","GSN","MOSPD3","STAU1","GEMIN2","NYNRIN","DHRS13","FLOT2","CARF","IFT122","PXK","POR","HOXB6","AGPAT5","MRPS12","SARS2","HMGXB4","PEX6","CYP3A5","HEMK1","CLK2","SEC16A","C5orf56","RAD50","YDJC","NINL","TOM1","SPTLC3","PHLDB1","PRR13","FMO2","ELOC","C5","BANF1","TFR2","SGMS1","FES","DMTN","BTBD19","TNK2","UBE2L3","MBD4","PELI3","RAB34","COBLL1","FMO3","PGAM5","ZMIZ2","PTK2","PLXND1","RAC1","RPS6","PAQR9","AKR1C4","FBXO17","SARM1","ARFGEF2","NBEAL1","CSE1L","AC005154.6","GGCT","ARPC1A","ARPC1B","EIF1AD","SNX1","DAGLB","LAMC1","USP37","RAB2A","TNPO1","SCAMP2","N4BP2L2","ATP5J2","SIPA1","BMPR2","ZNF554","UGT1A1","UGT1A10","UGT1A2P","UGT1A3","UGT1A4","UGT1A6","UGT1A8","UGT1A9","SULT1C4","FDPS","FAM228A","FGF7","SYNJ2BP","FAM228B","MRPS14","CYP2C8","TANGO2","RPP14","SLC50A1","SIRT2","FASN","TMEM70","RABGAP1L","EVI5","ZNF329","SBF1","STK36","TCTEX1D4","PDAP1","TMED4","GIGYF1","ZNF446","CACNA2D2","RBM6","TARBP2","PARP10","CTSF","CENPW","RP11-527F13.1","FAM219B","EFEMP2","CC2D2B","ENTPD1","MYRF","PPP2R5D","DDX27","TPM1","SLC17A7","CSNK1G1","CNPY3","GARS","MUC20","ZNFX1","PLCD4","GBA","ABCG5","ZNF789","SGTA","ULK3","ZNF584","RASSF1","HOXB5","HINFP","BCS1L","PLEC","EVI2B","NAP1L4","RPAP2","DPAGT1","ZNF740","GNMT","ZDHHC19","ZNF132","HLA-A","HLA-H","IFT46","PIH1D1","SEMA3F","EVI2A","HMCES","SLC51A","EHBP1L1","ABCG4","USP4","CNOT9","MUC4","ZDHHC8","TASP1","HOXB7","NLRX1","MUTYH","CCER2","NFKBIB","AMT","PPP1R2","FBXO18","H2AFV","SOAT2","ANKLE2","SEC24B","ELOVL2","PI4KA","AC051649.12","LSP1","MUS81","ARVCF","TBRG4","RNASEH2C","XXYLT1","CFI","KRT8","PDZD3","COPG1","CASP6","ATP5G2","ACAP2","COMMD7","KMT2A","FBXO31","CCDC93","ISM1","AAAS","C16orf95","TRAPPC4","EFCAB12","APOD","ANKRD16","CCDC84","CYP2U1","CYP2C9","TNS2","MAP1LC3B","C3orf62","ERVFRD-1","PHLDA2","ENTPD6","RHCE","SMIM13","ARAP2","RP11-141C7.2","CALCOCO1","TMPRSS13","ENPEP","PLA2G12A","MST1","HMBS","ESPL1","PCYT1A","UBE4A","VPS11","HYAL3","ZCCHC14","SECISBP2L","SLC37A4","SEMA3B","MCAM","RAMP3","KLHL22","TCTA","TMPRSS4-AS1","MAST2","PPIA","HECTD3","FXYD6","CD3E","TRMT2A","RPL34","TTC36","DDX6","NOP58","SCN4B","PRR33","LRIT3","SPRYD3","HYOU1","CCM2","MPZL3","AKR1E2","NAT6","NOL4L","RP5-1184F4.5","HYAL2","HADH","BEST4","CSAD","RHOA","RNF25","RNF26","DTHD1","MOB2","ETNPPL","NEDD9","TMEM170B","IGFBP1","CBL","TREH","MPZL2","CYP2C18","ASB13","MAK","MAPKAPK3","APEH","TEX264","MYO1G","IFRD2","TMEM14C","TH","GPX1","CYP2C19","TMEM14B","SMUG1","FAM208B","KRT18","GMPPB","TRAIP","MFSD5","UBA7","FAM212A","OSTC","HNRNPA1","CBX5","MST1R","KRT1","IGFBP6","SGMS2","KRT78","ITGB7","GRM2","EIF4B","CALML3","TMEM25","NET1","AKR1C1","GDI2","MANF","CCDC153","RNF123","SLC38A3","C2CD2L","JAML","BSN","AKR1C2","CDHR4","IP6K1"],[null,null,0.999998874018692,0.999981986518198,null,null,null,0.981761971313114,null,0.976921005162492,0.974752972467129,0.973813319632806,0.261522997327423,0.972398420814004,0.96830115209781,0.966860619106787,0.961704863961651,null,0.959101765677439,0.956550205798665,0.95573910726706,0.949096511261486,0.945929810744478,0.00737577137158385,0.550904779889946,0.933526948393775,0.851637143229038,0.927992629728818,0.916734127710682,null,0.911144989402156,0.874179062575899,null,0.867610769869974,null,0.852900550934959,0.848094478950196,0.842896288602277,null,0.752384685371724,0.8386836776066,0.687214423748729,0.798509677163151,null,0.0217445570263542,0.770386553438614,0.762587099053716,null,0.708337460760004,null,null,0.410922280767311,0.124209250846037,null,null,null,null,0.386634505899983,null,null,null,null,null,null,null,null,0.438815805725003,null,null,0.366761074908189,0.426896539827531,0.424027694496832,0.417118342270856,null,null,0.000610441036658194,0.391436342105011,null,null,0.161871078785424,0.116965139810749,null,null,0.366151021267387,0.353092902889958,0.345034186914012,0.345034186914012,0.0831240126078187,0.332453496159421,null,0.137238575408882,0.227680786723551,0.0509686626416539,null,null,null,0.00347556903120294,0.180734435186444,0.295683535341309,0.000765336525645144,0.273332522979267,0.269939699617874,0.0656719080619679,null,0.123270067622859,null,0.233430758008754,null,null,0.22603364430521,null,null,0.116965139810749,0.209487397746437,null,null,null,null,0.173068333688446,null,null,null,null,0.154796074678764,0.153714834733202,null,null,0.0418338183136532,null,0.128072118078033,null,null,0.122245921889763,0.119794451104776,0.122728319712771,null,0.0376344704771164,0.099620856052448,0.0957068158292588,0.0816511361923984,null,0.0731343677250189,null,0.0413583088294267,null,0.0807296040069906,0.0242932480958045,null,null,null,null,null,null,null,null,0.0777705264408883,null,0.0110674210707523,null,0.0689676598682171,null,null,null,null,null,0.0521292638955235,0.0510008402396813,0.0316748803900917,0.0490271124510792,null,0.0395385558170661,0.0402709410779151,0.0390807751613153,0.0210001772047869,0.0381585525119732,null,0.0324706709786717,null,0.0307558203965381,0.0300808985476078,0.000631776545442664,0.0255492471623412,0.0232007218281233,null,null,null,0.0130715784047897,0.0159176426576854,null,null,null,0.0194091778265356,0.0184961540116337,null,0.0170983713808645,0.0167769762089058,null,null,0.0134558715004985,0.0147209380075902,0.0137545445450525,0.0136202996557037,null,0.00827946546183478,0.0128914234380683,0.012847034209766,null,null,0.0116782884308286,0.00519992667511304,null,0.00635912573927155,0.010169509368214,0.000902473226717548,0.00992495630476087,null,0.00748414190056435,null,0.00279655332220596,0.00900297642864223,null,null,0.0081938853872161,null,0.000259628091852759,0.00750635255991883,0.00738495972123054,0.00253279171003995,null,0.00637686557149031,0.00613761359730125,null,0.00592874768920226,0.00590761852585264,0.00217533040793783,0.00540600752809473,0.00281801590914943,null,0.00495747421284898,0.00495747421284898,0.00492719117366947,0.00235001736146678,0.000704069155054055,0.00297537972567863,0.00180072099689144,null,0.000971582713256591,0.00412034861008326,0.00188112937980145,null,null,null,null,0.00098087586638107,0.0033099553221807,0.0032192953877106,0.00236202674233277,0.00291500131228073,0.00310633776165725,0.00310173146701997,0.00102953892658009,0.00289723605830039,0.0028408790106631,0.00139012050715814,0.00270933601658685,null,null,0.00258837401396106,null,0.00139653743229828,0.000944963877125726,0.00237474799736659,0.00236998901856467,0.00189993723515192,null,0.0013659614289302,null,0.000748103575737846,0.00207394534551086,0.00200342470664527,0.00196557556399979,0.00192167476788485,null,0.00189659664839148,0.00188252385019272,null,null,0.000927649064900282,0.00169592047319889,0.00164264932858593,0.000840376509337881,0.00026237185614908,0.00123009375065031,0.0015763645390382,null,null,0.000885762478319285,0.00125334271689603,0.00148227606134266,0.00144618885250813,0.000742239940536793,0.00133720615613764,null,0.000732552231101358,0.00132477059879099,0.00106742869737975,0.00128519121967163,null,null,0.00123177768974991,0.00122112670616426,0.00119165341086336,null,0.00114182191768697,0.00082342183065176,null,null,0.0011157608156015,0.00111179961800456,0.00109189745734994,0.000843466750496513,null,0.000755731794905956,0.00076025003589042,0.000581373969500643,0.000733065863408933,null,null,0.000957139866127998,0.000953057726116535,0.000933279579970803,0.000699641149770835,0.000656714182307394,null,0.000898861229987835,0.000898372571719097,0.000891382021297349,0.000864510263771101,0.000595498887474677,0.000806082384420148,0.000800422828724168,0.000795189512582506,0.00075362366616083,0.000753119954275183,0.000746157910061274,0.0004331810120205,0.00072618984323547,0.000364422263924191,0.00051985692990697,null,0.000713607749258216,null,0.000700453690663072,0.000681863325342369,0.000656435971586711,0.000638711372256906,0.000618883074824383,0.000395061835024246,0.000339581413290935,0.000605698823554768,0.000582316405454586,0.000568172957495006,0.000554236142337294,null,0.000488974630298844,null,null,0.00044750094787227,0.000436886400289938,0.000427136557572205,0.000414777724235926,null,0.000388705667860645,0.000378921791058806,0.00037681970793435,0.000375719276985431,0.000358117196879393,null,0.000330842724442504,0.00033017735035179,null,0.000288934090845072,null,null,0.000262822178863753,null,null,0.000260530850480611,null,null,null],[1,0.999999994148804,null,null,0.99817769591598,0.985476371248342,0.981903311450483,null,0.978530305398574,null,null,null,0.712273155422864,null,null,null,null,0.961117881730878,null,null,null,null,null,0.932443327530994,0.384919222408995,null,0.0790127984047668,null,null,0.916701126594213,null,null,0.872715393621341,null,0.862072076670389,null,null,null,0.842171608660238,0.0878119329371567,null,0.134972025040551,null,0.796087523553277,0.764662190843972,null,null,0.748552718041942,null,0.693218744569119,0.645576502168772,0.209499651437603,0.481062784889215,0.583716764882127,0.580800660261371,0.553902043399055,0.531247560500094,0.142310761391858,0.520178043302548,0.515119919171418,0.499801821654769,0.499801821654769,0.47959012080811,0.479224512606568,0.460584470244505,0.460554095187634,null,0.436112346038886,0.436112346038886,0.0662292970411933,null,null,null,0.413391213767353,0.411388815517301,0.407597312646631,null,0.376922137978766,0.376922137978766,0.213833684605639,0.257878464854871,0.370516664452816,0.37027517845835,null,null,null,null,0.256890963740152,null,0.332190714685118,0.191432814232931,0.100138205807742,0.266018448152427,0.31340763892741,0.31340763892741,0.310476657964342,0.306934372517545,0.119900013323236,0.000453962224807425,0.283264695957402,null,null,0.201366717534681,0.261254738004252,0.132566681834535,0.255340918268942,null,0.231783047428254,0.226630156618636,null,0.220938596098816,0.216864579572824,0.0953888049795435,null,0.188457786498399,0.183035752434221,0.182828447924776,0.182693521949339,0.00131054779136197,0.171466157477074,0.165162726288777,0.162932547915054,0.156195362671973,null,null,0.146554122007523,0.142310761391858,0.0995792592846128,0.137159605671363,null,0.126286658506935,0.126286658506935,0.00393597539940616,0.00393597539940616,null,0.108150586356925,0.0645698032288284,null,null,0.0107271943267602,0.0889104632295744,0.0137952134179427,0.0865145519395252,0.0450855353298614,0.0843958876929944,0.0013374229043086,0.0558691158392456,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,null,0.0760095844987588,0.0647890101615882,0.0713485185375567,null,0.0647890101615882,0.0575213189904328,0.0564696941104702,0.0538930773429064,0.0538490935774176,null,null,0.0186699182648297,null,0.0452794036622566,0.00506806170838825,null,null,0.0173994349648032,null,0.0363185764087443,null,0.0309217114977492,null,null,0.0269590351241576,null,null,0.0227563448054738,0.0223394079836969,0.0223394079836969,0.00911186095578531,0.00611324472965979,0.0215802644388751,0.0215802644388751,0.0214054176704254,null,null,0.0183424418081871,null,null,0.0165135964683738,0.0154331283167679,0.00171700460322333,null,null,null,0.0135327477634933,0.00492476227050276,null,null,0.0127522956451115,0.0121764846034262,null,0.0060862989329854,0.0108473373360839,0.00434385102211898,null,0.0091960401320228,null,0.00980474932009956,0.00220580068531273,0.00931199445066355,0.00626871447690391,null,0.00867590822823705,0.00867590822823705,null,0.00814282672650157,0.00763227282354184,null,null,0.00445880869416904,0.0065521017059379,null,null,0.00605721186669128,null,null,0.00372119290978523,null,0.00221624227231221,0.00497973420696973,null,null,null,0.00257570372756477,0.00421195030536048,0.00167551737818317,0.0025292329418527,0.00419250293286133,0.0032075649637584,null,0.00223630522943297,0.003581025211989,0.003581025211989,0.00356406671064824,0.00349110562434107,0.00237253250712866,null,null,0.000800947389816287,0.000206205857589903,null,null,0.00197102483703548,null,null,0.00139900956537831,null,0.00263791142384884,0.00260230777325776,null,0.00257136608259867,0.001048274092049,0.00145825091657181,null,null,0.000440892167159923,0.00229079254223308,0.00080636666179057,0.00213249486999334,0.00137869075268948,null,null,null,null,0.00190416269063698,null,null,0.00184329439955844,0.00171700460322333,0.000774627302705388,null,null,0.000800947389816287,0.00135914466741949,0.000346447899208124,null,0.00153383742557467,0.00152729779423388,0.000617367348536679,0.000245785918150609,null,null,0.000694491866052416,null,0.00133630384335293,0.000594048092267152,null,0.000251128862849725,null,0.00124696782350908,0.00123777708850104,null,null,null,0.00114329798170854,null,0.000315830625966629,0.00113470081876765,0.0011310485455801,null,null,null,0.000230461666743009,0.00105559580954129,0.000267184920692776,0.000261648357403277,0.000428263483120417,0.000245785918150609,0.000977550676121708,0.000977550676121708,null,null,null,0.000216433975084418,0.000252777801804216,0.00090377945604414,null,null,null,null,0.000254902963160997,null,null,null,null,null,null,0.000295777554478405,null,0.000355283381343607,0.000199106662409188,0.000718829014160094,null,0.000705950954109635,null,null,null,null,null,0.000217786011859733,0.000273020869431817,null,null,null,null,0.000546685093151145,null,0.000463942075406676,0.000458047147642304,null,null,null,null,0.000401379635342147,null,null,null,null,null,0.000344291895433169,null,null,0.000304577857608512,null,0.000280095961652127,0.00027521875004044,null,0.000262605267515537,0.000262025585317627,null,0.000241024042835214,0.000140778034987021,0.000140778034987021],[1,0.999999994148804,0.999998874018692,0.999981986518198,0.99817769591598,0.985476371248342,0.981903311450483,0.981761971313114,0.978530305398574,0.976921005162492,0.974752972467129,0.973813319632806,0.973796152750287,0.972398420814004,0.96830115209781,0.966860619106787,0.961704863961651,0.961117881730878,0.959101765677439,0.956550205798665,0.95573910726706,0.949096511261486,0.945929810744478,0.939819098902578,0.935824002298941,0.933526948393775,0.930649941633804,0.927992629728818,0.916734127710682,0.916701126594213,0.911144989402156,0.874179062575899,0.872715393621341,0.867610769869974,0.862072076670389,0.852900550934959,0.848094478950196,0.842896288602277,0.842171608660238,0.840196618308881,0.8386836776066,0.82218644878928,0.798509677163151,0.796087523553277,0.786406747870326,0.770386553438614,0.762587099053716,0.748552718041942,0.708337460760004,0.693218744569119,0.645576502168772,0.620421932204914,0.605272035735252,0.583716764882127,0.580800660261371,0.553902043399055,0.531247560500094,0.528945267291841,0.520178043302548,0.515119919171418,0.499801821654769,0.499801821654769,0.47959012080811,0.479224512606568,0.460584470244505,0.460554095187634,0.438815805725003,0.436112346038886,0.436112346038886,0.432990371949383,0.426896539827531,0.424027694496832,0.417118342270856,0.413391213767353,0.411388815517301,0.40820775368329,0.391436342105011,0.376922137978766,0.376922137978766,0.375704763391064,0.37484360466562,0.370516664452816,0.37027517845835,0.366151021267387,0.353092902889958,0.345034186914012,0.345034186914012,0.340014976347971,0.332453496159421,0.332190714685118,0.328671389641813,0.327818992531293,0.316987110794081,0.31340763892741,0.31340763892741,0.310476657964342,0.310409941548748,0.30063444850968,0.296137497566116,0.284030032483047,0.273332522979267,0.269939699617874,0.267038625596649,0.261254738004252,0.255836749457394,0.255340918268942,0.233430758008754,0.231783047428254,0.226630156618636,0.22603364430521,0.220938596098816,0.216864579572824,0.212353944790293,0.209487397746437,0.188457786498399,0.183035752434221,0.182828447924776,0.182693521949339,0.174378881479808,0.171466157477074,0.165162726288777,0.162932547915054,0.156195362671973,0.154796074678764,0.153714834733202,0.146554122007523,0.142310761391858,0.141413077598266,0.137159605671363,0.128072118078033,0.126286658506935,0.126286658506935,0.12618189728917,0.123730426504182,0.122728319712771,0.108150586356925,0.102204273705945,0.099620856052448,0.0957068158292588,0.0923783305191586,0.0889104632295744,0.0869295811429616,0.0865145519395252,0.0864438441592881,0.0843958876929944,0.0820670269112992,0.0801623639350501,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0788651631126294,0.0777705264408883,0.0760095844987588,0.0758564312323405,0.0713485185375567,0.0689676598682171,0.0647890101615882,0.0575213189904328,0.0564696941104702,0.0538930773429064,0.0538490935774176,0.0521292638955235,0.0510008402396813,0.0503447986549214,0.0490271124510792,0.0452794036622566,0.0446066175254544,0.0402709410779151,0.0390807751613153,0.0383996121695901,0.0381585525119732,0.0363185764087443,0.0324706709786717,0.0309217114977492,0.0307558203965381,0.0300808985476078,0.0275908116696003,0.0255492471623412,0.0232007218281233,0.0227563448054738,0.0223394079836969,0.0223394079836969,0.022183439360575,0.0220308873873452,0.0215802644388751,0.0215802644388751,0.0214054176704254,0.0194091778265356,0.0184961540116337,0.0183424418081871,0.0170983713808645,0.0167769762089058,0.0165135964683738,0.0154331283167679,0.0151728761037219,0.0147209380075902,0.0137545445450525,0.0136202996557037,0.0135327477634933,0.0132042277323375,0.0128914234380683,0.012847034209766,0.0127522956451115,0.0121764846034262,0.0116782884308286,0.0112862256080984,0.0108473373360839,0.0107029767613905,0.010169509368214,0.0100985133587403,0.00992495630476087,0.00980474932009956,0.00968994258587708,0.00931199445066355,0.00906526779910988,0.00900297642864223,0.00867590822823705,0.00867590822823705,0.0081938853872161,0.00814282672650157,0.0078919009153946,0.00750635255991883,0.00738495972123054,0.00699160040420899,0.0065521017059379,0.00637686557149031,0.00613761359730125,0.00605721186669128,0.00592874768920226,0.00590761852585264,0.00589652331772306,0.00540600752809473,0.00503425818146164,0.00497973420696973,0.00495747421284898,0.00495747421284898,0.00492719117366947,0.00492572108903155,0.00491601946041453,0.0046508971038618,0.00432995393874414,0.00419250293286133,0.00417914767701499,0.00412034861008326,0.00411743460923442,0.003581025211989,0.003581025211989,0.00356406671064824,0.00349110562434107,0.00335340837350973,0.0033099553221807,0.0032192953877106,0.00316297413214905,0.00312120716987063,0.00310633776165725,0.00310173146701997,0.00300056376361557,0.00289723605830039,0.0028408790106631,0.00278913007253645,0.00270933601658685,0.00263791142384884,0.00260230777325776,0.00258837401396106,0.00257136608259867,0.00244481152434739,0.00240321479369754,0.00237474799736659,0.00236998901856467,0.00234082940231184,0.00229079254223308,0.00217232809072077,0.00213249486999334,0.00212679432842733,0.00207394534551086,0.00200342470664527,0.00196557556399979,0.00192167476788485,0.00190416269063698,0.00189659664839148,0.00188252385019272,0.00184329439955844,0.00171700460322333,0.00170227636760567,0.00169592047319889,0.00164264932858593,0.00164132389915417,0.00162151652356857,0.00157654164985843,0.0015763645390382,0.00153383742557467,0.00152729779423388,0.00150312982685596,0.00149912863504653,0.00148227606134266,0.00144618885250813,0.00143673180658921,0.00133720615613764,0.00133630384335293,0.00132660032336851,0.00132477059879099,0.00131855756022947,0.00128519121967163,0.00124696782350908,0.00123777708850104,0.00123177768974991,0.00122112670616426,0.00119165341086336,0.00114329798170854,0.00114182191768697,0.00113925245661839,0.00113470081876765,0.0011310485455801,0.0011157608156015,0.00111179961800456,0.00109189745734994,0.00107392841723952,0.00105559580954129,0.00102291671559873,0.0010218983932937,0.00100963745262106,0.000978851781559431,0.000977550676121708,0.000977550676121708,0.000957139866127998,0.000953057726116535,0.000933279579970803,0.000916075124855253,0.00090949198411161,0.00090377945604414,0.000898861229987835,0.000898372571719097,0.000891382021297349,0.000864510263771101,0.000850401850635674,0.000806082384420148,0.000800422828724168,0.000795189512582506,0.00075362366616083,0.000753119954275183,0.000746157910061274,0.000728958566498905,0.00072618984323547,0.000719705645267799,0.000718963592316157,0.000718829014160094,0.000713607749258216,0.000705950954109635,0.000700453690663072,0.000681863325342369,0.000656435971586711,0.000638711372256906,0.000618883074824383,0.000612847846883979,0.000612602282722752,0.000605698823554768,0.000582316405454586,0.000568172957495006,0.000554236142337294,0.000546685093151145,0.000488974630298844,0.000463942075406676,0.000458047147642304,0.00044750094787227,0.000436886400289938,0.000427136557572205,0.000414777724235926,0.000401379635342147,0.000388705667860645,0.000378921791058806,0.00037681970793435,0.000375719276985431,0.000358117196879393,0.000344291895433169,0.000330842724442504,0.00033017735035179,0.000304577857608512,0.000288934090845072,0.000280095961652127,0.00027521875004044,0.000262822178863753,0.000262605267515537,0.000262025585317627,0.000260530850480611,0.000241024042835214,0.000140778034987021,0.000140778034987021]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n      <th>combined_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization" class="section level2">
<h2>Visualization</h2>
<!-- We make locuz plot for the region(19_9127717_11173352) containing the top 1 gene (LDLR). we limit PIP results to credible sets in the PIP track by setting `filter_cs = TRUE`.   -->
<!-- ```{r} -->
<!-- weights <- readRDS(paste0(results_dir,trait,".preprocessed.weights.RDS")) -->
<!-- make_locusplot(finemap_res = finemap_res, -->
<!--                region_id = "19_9127717_11173352", -->
<!--                ens_db = ens_db, -->
<!--                weights = weights, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = T, -->
<!--                filter_cs = T, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs") -->
<!-- ``` -->
<ul>
<li>The top one shows -log10(p-value) of the association of variants
(from LDL GWAS) and molecular traits (from the package computed
z-scores) with the phenotype</li>
<li>The next track shows the PIPs of variants and molecular traits. By
default, we limit PIP results to credible sets in the PIP track
(<code>filter_cs = TRUE</code>)</li>
<li>The next track shows the QTLs of the focal gene.</li>
<li>The bottom is the gene track.</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplot2_3.5.1             EnsDb.Hsapiens.v86_2.99.0
 [3] ensembldb_2.20.2          AnnotationFilter_1.20.0  
 [5] GenomicFeatures_1.48.3    AnnotationDbi_1.58.0     
 [7] Biobase_2.56.0            GenomicRanges_1.48.0     
 [9] GenomeInfoDb_1.39.9       IRanges_2.30.0           
[11] S4Vectors_0.34.0          BiocGenerics_0.42.0      
[13] ctwas_0.4.14             

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                DT_0.22                    
 [11] ggrepel_0.9.1               bit64_4.0.5                
 [13] fansi_1.0.3                 xml2_1.3.3                 
 [15] codetools_0.2-18            logging_0.10-108           
 [17] cachem_1.0.6                knitr_1.39                 
 [19] jsonlite_1.8.0              workflowr_1.7.0            
 [21] Rsamtools_2.12.0            dbplyr_2.1.1               
 [23] png_0.1-7                   readr_2.1.2                
 [25] compiler_4.2.0              httr_1.4.3                 
 [27] assertthat_0.2.1            Matrix_1.5-3               
 [29] fastmap_1.1.0               lazyeval_0.2.2             
 [31] cli_3.6.1                   later_1.3.0                
 [33] htmltools_0.5.2             prettyunits_1.1.1          
 [35] tools_4.2.0                 gtable_0.3.0               
 [37] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [39] dplyr_1.1.4                 rappdirs_0.3.3             
 [41] Rcpp_1.0.12                 jquerylib_0.1.4            
 [43] vctrs_0.6.5                 Biostrings_2.64.0          
 [45] rtracklayer_1.56.0          crosstalk_1.2.0            
 [47] xfun_0.41                   stringr_1.5.1              
 [49] lifecycle_1.0.4             irlba_2.3.5                
 [51] restfulr_0.0.14             XML_3.99-0.14              
 [53] zlibbioc_1.42.0             zoo_1.8-10                 
 [55] scales_1.3.0                gggrid_0.2-0               
 [57] hms_1.1.1                   promises_1.2.0.1           
 [59] MatrixGenerics_1.8.0        ProtGenerics_1.28.0        
 [61] parallel_4.2.0              SummarizedExperiment_1.26.1
 [63] LDlinkR_1.2.3               yaml_2.3.5                 
 [65] curl_4.3.2                  memoise_2.0.1              
 [67] sass_0.4.1                  biomaRt_2.54.1             
 [69] stringi_1.7.6               RSQLite_2.3.1              
 [71] highr_0.9                   BiocIO_1.6.0               
 [73] filelock_1.0.2              BiocParallel_1.30.3        
 [75] rlang_1.1.2                 pkgconfig_2.0.3            
 [77] matrixStats_0.62.0          bitops_1.0-7               
 [79] evaluate_0.15               lattice_0.20-45            
 [81] purrr_1.0.2                 labeling_0.4.2             
 [83] GenomicAlignments_1.32.0    htmlwidgets_1.5.4          
 [85] cowplot_1.1.1               bit_4.0.4                  
 [87] tidyselect_1.2.0            magrittr_2.0.3             
 [89] R6_2.5.1                    generics_0.1.2             
 [91] DelayedArray_0.22.0         DBI_1.2.2                  
 [93] withr_2.5.0                 pgenlibr_0.3.3             
 [95] pillar_1.9.0                KEGGREST_1.36.3            
 [97] RCurl_1.98-1.7              mixsqp_0.3-43              
 [99] tibble_3.2.1                crayon_1.5.1               
[101] utf8_1.2.2                  BiocFileCache_2.4.0        
[103] plotly_4.10.0               tzdb_0.4.0                 
[105] rmarkdown_2.25              progress_1.2.2             
[107] grid_4.2.0                  data.table_1.14.2          
[109] blob_1.2.3                  git2r_0.30.1               
[111] digest_0.6.29               tidyr_1.3.0                
[113] httpuv_1.6.5                munsell_0.5.0              
[115] viridisLite_0.4.0           bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
