<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-10-22" />

<title>methylation_analysis</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">methylation_analysis</h1>
<h4 class="date">2024-10-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreecacd12af28e68bc17a8bebc2698369def91d21b1targetblankcacd12aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/cacd12af28e68bc17a8bebc2698369def91d21b1" target="_blank">cacd12a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreecacd12af28e68bc17a8bebc2698369def91d21b1targetblankcacd12aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/cacd12af28e68bc17a8bebc2698369def91d21b1" target="_blank">cacd12a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/methylation_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/methylation_analysis.Rmd</code>) and HTML (<code>docs/methylation_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/cacd12af28e68bc17a8bebc2698369def91d21b1/analysis/methylation_analysis.Rmd" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/cacd12af28e68bc17a8bebc2698369def91d21b1/docs/methylation_analysis.html" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/a901ca74061a5fea287f4ce8858c764288c4e7da/analysis/methylation_analysis.Rmd" target="_blank">a901ca7</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/770965a822de5471a8f1b51e0719f06fb70cf73e/analysis/methylation_analysis.Rmd" target="_blank">770965a</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/analysis/methylation_analysis.Rmd" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/methylation_analysis.html" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/2db1bdc25b5fb37d8910a50297eaaea42cb91705/analysis/methylation_analysis.Rmd" target="_blank">2db1bdc</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/2db1bdc25b5fb37d8910a50297eaaea42cb91705/docs/methylation_analysis.html" target="_blank">2db1bdc</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/f8b659fd7299d1753a4563d52b46106dc53451e1/analysis/methylation_analysis.Rmd" target="_blank">f8b659f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/f8b659fd7299d1753a4563d52b46106dc53451e1/docs/methylation_analysis.html" target="_blank">f8b659f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/b3ff8424e105409f127ee88a81bbfa4b66c92266/analysis/methylation_analysis.Rmd" target="_blank">b3ff842</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/b3ff8424e105409f127ee88a81bbfa4b66c92266/docs/methylation_analysis.html" target="_blank">b3ff842</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/919465c452be1dbc97b59012db7290a51f7a886e/analysis/methylation_analysis.Rmd" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/919465c452be1dbc97b59012db7290a51f7a886e/docs/methylation_analysis.html" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="fusion-lasso-model-of-dna-methylation" class="section level3">
<h3>Fusion Lasso model of DNA methylation</h3>
<p>I built lasso model of DNA methylation with FUSION for Whole Blood and Colon Transverse. Similar to meQTL mapping, for each CpG site, I extracted surrounding 50kb genptypes and train lasso models with cross validation. With heritability cutoff p&lt;0.0001, I have about 16,000 and 48,000 CpG sites in whole blood and colon transverse. Among which, 5,000 and 40,000 CpG sites are also in QTL mapping. Colon have more overlaps than whole blood. The average cross-validation R2 for lasso in whole blood and colon transverse are 0.393 and 0.248 In the single QTL approach (qval &lt; 0.001), we have 7,720 and 91,466 CpG sites.</p>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/figure/methylation_analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/919465c452be1dbc97b59012db7290a51f7a886e/docs/figure/methylation_analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ibd-results" class="section level2">
<h2>IBD results</h2>
<div id="ctwas-parameters-50kb-h2-pvalue0.00001" class="section level3">
<h3>cTWAS parameters (50kb, h2 pvalue&lt;0.00001)</h3>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/cacd12af28e68bc17a8bebc2698369def91d21b1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/f8b659fd7299d1753a4563d52b46106dc53451e1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">f8b659f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/b3ff8424e105409f127ee88a81bbfa4b66c92266/docs/figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">b3ff842</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/919465c452be1dbc97b59012db7290a51f7a886e/docs/figure/methylation_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="meqtls-explained-10-heritability" class="section level3">
<h3>meQTLs explained 10% heritability</h3>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/cacd12af28e68bc17a8bebc2698369def91d21b1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/figure/methylation_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/2db1bdc25b5fb37d8910a50297eaaea42cb91705/docs/figure/methylation_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">2db1bdc</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/919465c452be1dbc97b59012db7290a51f7a886e/docs/figure/methylation_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ctwas-with-eqtls-and-meqtls-identifies-28-genes-with-pip-0.8" class="section level3">
<h3>cTWAS with eQTLs and meQTLs identifies 28 genes with PIP &gt; 0.8</h3>
</div>
<div id="top-ctwas-genes-from-single-group-models-have-very-little-overlap" class="section level3">
<h3>Top cTWAS genes from single group models have very little overlap</h3>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/figure/methylation_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/f8b659fd7299d1753a4563d52b46106dc53451e1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">f8b659f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/b3ff8424e105409f127ee88a81bbfa4b66c92266/docs/figure/methylation_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">b3ff842</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/919465c452be1dbc97b59012db7290a51f7a886e/docs/figure/methylation_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">919465c</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="adding-eqtl-to-meqtl-identifies-an-additional-16-high-pip-genes" class="section level3">
<h3>Adding eQTL to meQTL identifies an additional 16 high PIP genes</h3>
<ol style="list-style-type: decimal">
<li>12/17 meQTL genes still have combined PIP &gt; 0.8 after adding eQTL</li>
<li>3/17 meQTL genes have decreased combined PIP &lt; 0.8 after adding eQTL</li>
<li>2/17 meQTL genes are lost, due to region selection after adding eQTL</li>
<li>Among the 12 overalpped genes:</li>
</ol>
<ul>
<li><p>One gene (BRD7) is mediation (meQTL pip decreases from 0.96 to 0.09, eQTL pip=0.90).</p></li>
<li><p>Two genes (TNFSF15 and ATG16L1) are competition (eQTL pip = 0.2 and meQTL pip is decreased by 0.2).</p></li>
<li><p>Nine genes are meQTL alone (no eQTL pip). <img src="figure/methylation_analysis.Rmd/unnamed-chunk-10-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
<p>Past versions of unnamed-chunk-10-1.png</p>
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/cacd12af28e68bc17a8bebc2698369def91d21b1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c3e7a9fc6eeeffe7f629c42146bc6e4a462b92c7/docs/figure/methylation_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">c3e7a9f</a>
</td>
<td>
sq-96
</td>
<td>
2024-10-25
</td>
</tr>
</tbody>
</table>
</div>
</div></li>
</ul>
</div>
<div id="adding-meqtl-to-eqtl-identifies-an-additional-11-high-pip-genes" class="section level3">
<h3>Adding meQTL to eQTL identifies an additional 11 high PIP genes</h3>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-11-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/cacd12af28e68bc17a8bebc2698369def91d21b1/docs/figure/methylation_analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">cacd12a</a>
</td>
<td>
sq-96
</td>
<td>
2024-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="genes-that-identified-by-meqtl-not-eqtl" class="section level3">
<h3>Genes that identified by meQTL not eQTL</h3>
<pre class="r"><code>finemap_res_multi[!is.na(finemap_res_multi$gene_name) &amp; finemap_res_multi$gene_name == &quot;ETS1&quot;, ]</code></pre>
<pre><code>                                           id       molecular_id  type
8377 ENSG00000134954.14|Colon_Transverse_eQTL ENSG00000134954.14  eQTL
8392             cg04583498|Whole_Blood_meQTL         cg04583498 meQTL
8419        cg27625507|Colon_Transverse_meQTL         cg27625507 meQTL
8423        cg13448797|Colon_Transverse_meQTL         cg13448797 meQTL
              context                  group              region_id         z
8377 Colon_Transverse  Colon_Transverse|eQTL 11_128318642_130472680  1.123077
8392      Whole_Blood      Whole_Blood|meQTL 11_128318642_130472680 -5.401685
8419 Colon_Transverse Colon_Transverse|meQTL 11_128318642_130472680  1.140970
8423 Colon_Transverse Colon_Transverse|meQTL 11_128318642_130472680 -2.524429
        susie_pip       mu2   cs gene_name      gene_type chrom     start
8377 3.765461e-06  1.967371 &lt;NA&gt;      ETS1 protein_coding    11 128458761
8392 8.471599e-01 25.715822   L1      ETS1 protein_coding    11 128556018
8419 7.231902e-07  2.028024 &lt;NA&gt;      ETS1 protein_coding    11 128583165
8423 7.486851e-06  6.337017 &lt;NA&gt;      ETS1 protein_coding    11 128512888
           end       pos
8377 128587558 128523160
8392 128556020 128556019
8419 128583167 128583166
8423 128512890 128512889</code></pre>
<pre class="r"><code>#pdf(file = &quot;/project/xinhe/shengqian/TNFSF15.pdf&quot;,width = 12,height = 10)#
make_locusplot(finemap_res_multi,
               region_id = &quot;9_114257521_115159681&quot;,
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;,
               point.sizes = c(2, 8),
               label.text.size = 4,
               axis.text.size = 18,
               axis.title.size = 18,
               legend.text.size = 18,
               panel.heights = c(4, 4, 0.8, 2))</code></pre>
<pre><code>2024-11-25 11:54:32 INFO::Limit to protein coding genes
2024-11-25 11:54:32 INFO::focal id: cg06629146|Whole_Blood_meQTL
2024-11-25 11:54:32 INFO::focal molecular trait: TNFSF15 Whole_Blood meQTL
2024-11-25 11:54:32 INFO::Range of locus: chr9:114257421-115159692</code></pre>
<pre><code>chromosome 9, position 114257421 to 115159692</code></pre>
<pre><code>2376 SNPs/datapoints</code></pre>
<pre><code>2024-11-25 11:54:35 INFO::focal molecular trait QTL positions: 114779085,114785492,114790605,114802595,114804160,114804843,114806486,114806680,114806766,114807377,114817177,114823090,114841815
2024-11-25 11:54:35 INFO::Limit PIPs to credible sets</code></pre>
<pre><code>Warning: ggrepel: 24 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<pre><code>Warning: ggrepel: 13 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/methylation_analysis.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.22.0         
 [3] AnnotationFilter_1.22.0   GenomicFeatures_1.50.4   
 [5] AnnotationDbi_1.60.2      Biobase_2.58.0           
 [7] GenomicRanges_1.50.2      GenomeInfoDb_1.34.9      
 [9] IRanges_2.32.0            S4Vectors_0.36.2         
[11] BiocGenerics_0.44.0       VennDiagram_1.7.3        
[13] futile.logger_1.4.3       RColorBrewer_1.1-3       
[15] pheatmap_1.0.12           magrittr_2.0.3           
[17] RSQLite_2.3.7             lubridate_1.9.3          
[19] forcats_1.0.0             stringr_1.5.1            
[21] dplyr_1.1.4               purrr_1.0.2              
[23] readr_2.1.5               tidyr_1.3.1              
[25] tibble_3.2.1              tidyverse_2.0.0          
[27] ctwas_0.4.15              data.table_1.16.0        
[29] gridExtra_2.3             ggVennDiagram_1.5.2      
[31] ggplot2_3.5.1             workflowr_1.7.0          

loaded via a namespace (and not attached):
  [1] BiocFileCache_2.6.1         lazyeval_0.2.2             
  [3] crosstalk_1.2.1             BiocParallel_1.32.6        
  [5] LDlinkR_1.4.0               digest_0.6.37              
  [7] yulab.utils_0.1.7           htmltools_0.5.8.1          
  [9] fansi_1.0.6                 memoise_2.0.1              
 [11] tzdb_0.4.0                  Biostrings_2.66.0          
 [13] matrixStats_1.4.1           locuszoomr_0.3.5           
 [15] timechange_0.3.0            prettyunits_1.2.0          
 [17] colorspace_2.1-1            blob_1.2.4                 
 [19] rappdirs_0.3.3              ggrepel_0.9.6              
 [21] xfun_0.47                   callr_3.7.2                
 [23] crayon_1.5.3                RCurl_1.98-1.16            
 [25] jsonlite_1.8.9              zoo_1.8-12                 
 [27] glue_1.7.0                  gtable_0.3.5               
 [29] zlibbioc_1.44.0             XVector_0.38.0             
 [31] DelayedArray_0.24.0         scales_1.3.0               
 [33] futile.options_1.0.1        DBI_1.2.3                  
 [35] Rcpp_1.0.13                 viridisLite_0.4.2          
 [37] progress_1.2.3              gridGraphics_0.5-1         
 [39] bit_4.5.0                   DT_0.22                    
 [41] htmlwidgets_1.6.4           httr_1.4.7                 
 [43] pkgconfig_2.0.3             XML_3.99-0.14              
 [45] farver_2.1.2                sass_0.4.9                 
 [47] dbplyr_2.5.0                utf8_1.2.4                 
 [49] ggplotify_0.1.2             tidyselect_1.2.1           
 [51] labeling_0.4.3              rlang_1.1.4                
 [53] later_1.3.2                 munsell_0.5.1              
 [55] pgenlibr_0.3.7              tools_4.2.0                
 [57] cachem_1.1.0                cli_3.6.3                  
 [59] generics_0.1.3              evaluate_1.0.0             
 [61] fastmap_1.2.0               yaml_2.3.10                
 [63] processx_3.7.0              knitr_1.48                 
 [65] bit64_4.5.2                 fs_1.6.4                   
 [67] KEGGREST_1.38.0             whisker_0.4                
 [69] formatR_1.14                aplot_0.2.3                
 [71] xml2_1.3.3                  biomaRt_2.54.1             
 [73] compiler_4.2.0              rstudioapi_0.14            
 [75] plotly_4.10.4               filelock_1.0.3             
 [77] curl_5.2.3                  png_0.1-7                  
 [79] bslib_0.8.0                 stringi_1.8.4              
 [81] highr_0.11                  ps_1.7.1                   
 [83] lattice_0.20-45             ProtGenerics_1.30.0        
 [85] Matrix_1.5-3                vctrs_0.6.5                
 [87] pillar_1.9.0                lifecycle_1.0.4            
 [89] jquerylib_0.1.4             cowplot_1.1.3              
 [91] bitops_1.0-8                irlba_2.3.5.1              
 [93] httpuv_1.6.5                patchwork_1.3.0            
 [95] rtracklayer_1.58.0          R6_2.5.1                   
 [97] BiocIO_1.8.0                promises_1.3.0             
 [99] codetools_0.2-18            lambda.r_1.2.4             
[101] SummarizedExperiment_1.28.0 rprojroot_2.0.3            
[103] rjson_0.2.23                withr_3.0.1                
[105] GenomicAlignments_1.34.1    Rsamtools_2.14.0           
[107] GenomeInfoDbData_1.2.9      parallel_4.2.0             
[109] hms_1.1.3                   ggfun_0.1.6                
[111] gggrid_0.2-0                rmarkdown_2.28             
[113] MatrixGenerics_1.10.0       logging_0.10-108           
[115] git2r_0.30.1                mixsqp_0.3-54              
[117] getPass_0.2-2               restfulr_0.0.15            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
