<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-08-08" />

<title>Comparing results from Munro weights &amp; predictdb weights</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparing results from Munro weights &amp;
predictdb weights</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-08-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreecf3486775e08c06085931b2bdcf21d724652069btargetblankcf34867a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/cf3486775e08c06085931b2bdcf21d724652069b" target="_blank">cf34867</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreecf3486775e08c06085931b2bdcf21d724652069btargetblankcf34867a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/cf3486775e08c06085931b2bdcf21d724652069b" target="_blank">cf34867</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/figure/
    Ignored:    results/

Untracked files:
    Untracked:  analysis/multi_group_compare_predictdb_munro.Rmd
    Untracked:  analysis/multi_group_compare_predictdb_munro_updated.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>We compare the results from Munro weights &amp; predictdb weights
here. We are figuring out how the number of high PIP genes compare with
PredictDB results with the same tissues?</p>
<div id="settings" class="section level1">
<h1>Settings</h1>
<div id="modalities-from-munro" class="section level2">
<h2>6 modalities from Munro</h2>
<ol style="list-style-type: decimal">
<li>Weight processing:</li>
</ol>
<p>PredictDB:</p>
<p>all the PredictDB are converted from FUSION weights</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = F (FUSION converted weights)</li>
<li>load_predictdb_LD = F,</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Parameter estimation and fine-mapping</li>
</ol>
<ul>
<li>niter_prefit = 5,</li>
<li>niter = 60,</li>
<li>L: determined by uniform susie,</li>
<li>group_prior_var_structure = “shared_type”,</li>
<li>maxSNP = 20000,</li>
<li>min_nonSNP_PIP = 0.5,</li>
</ul>
</div>
<div id="weights-from-predictdb" class="section level2">
<h2>weights from predictdb</h2>
<ol style="list-style-type: decimal">
<li>Weight processing:</li>
</ol>
<p>PredictDB (eqtl, sqtl)</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = T</li>
<li>load_predictdb_LD = F,</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Parameter estimation and fine-mapping</li>
</ol>
<ul>
<li>niter_prefit = 5,</li>
<li>niter = 60,</li>
<li>L: determined by uniform susie,</li>
<li>group_prior_var_structure = “shared_type”,</li>
<li>maxSNP = 20000,</li>
<li>min_nonSNP_PIP = 0.5,</li>
</ul>
<p>mem: 150g 5cores</p>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="ldl---liver" class="section level2">
<h2>LDL - Liver</h2>
<div id="predictdb-eqtl-and-sqtl" class="section level3">
<h3>Predictdb: eqtl and sqtl</h3>
<p><img src="figure/multi_group_compare_predictdb_munro_updated.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4"],["eQTL","sQTL","SNP","total_pve"],["0.007","0.0109","0.0378","0.0557"],["0.007","0.0109","0","0.0179"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Liver<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre><code>2024-08-09 11:44:45 INFO::Annotating ctwas finemapping result ...
2024-08-09 11:44:55 INFO::add gene_name and gene_type
2024-08-09 11:44:55 INFO::split PIPs for traits mapped to multiple genes
2024-08-09 11:44:56 INFO::use gene mid positions
2024-08-09 11:44:56 INFO::add SNP positions</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Combined PIP by omics, credible set filtered<\/caption>","data":[["53","49","9","1","42","96","48","79","4","16","22","2","113","68","13","105","6","39","35","52","50","95","11","56","12","8","107","21","23","94","76","73","61","19","80","54","100","24"],["LDLR","HPR","ASGR1","ABCA1","GSK3B","STAT5B","HP","PRKD2","ACVR1C","CNIH4","CYP2A6","ABCA8","USP53","PARP9","C10orf88","TNKS","ALLC","GAS6","FADS1","INSIG2","INHBB","SPTY2D1","BMPR2","LRRK2","BRI3","ASAP3","TRIM39","CWF19L1","CYP4F12","SPHK2","POP7","PHC1","MZF1","CSNK1G3","PRMT6","LIPC","THOP1","DDX56"],[1,1,0.992,0.9868,0.9841,0.9822,0.9768,0.9746,0.9735,0.9728,0.9726,0.9702,0.969,0.967,0.9548,0.9545,0.9527,0.9509,0.938,0.9356,0.9238,0.9217,0.913,0.9118,0.9099,0.907,0.898,0.8968,0.8951,0.8866,0.8835,0.8805,0.8731,0.8598,0.8539,0.8464,0.8464,0.8328],[null,1,null,0.9868,0.9841,0.9822,null,0.9746,0.9735,0.9728,0.9726,null,0.5538,null,0.9548,0.9545,0.9527,0.9509,0.938,0.9356,0.9238,0.9217,0.913,0.9118,0.8063,0.907,0.898,0.8968,null,0.0268,0.8835,null,0.7174,0.8598,0.8539,null,null,0.8328],[1,null,0.992,null,null,null,0.9768,null,null,null,null,0.9702,0.4153,0.967,null,null,null,null,null,null,null,null,null,null,0.1037,null,null,null,0.8951,0.8598,null,0.8805,0.1557,null,null,0.8464,0.8464,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="munro-et-al-6-modalities" class="section level3">
<h3>Munro et al : 6 modalities</h3>
<p><img src="figure/multi_group_compare_predictdb_munro_updated.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5","6","7","8"],["apaQTL","eQTL","isoQTL","rsQTL","sQTL","tssQTL","SNP","total_pve"],["4e-04","0.005","0.0028","0.0014","0.0019","4e-04","0.0388","0.0507"],["4e-04","0.005","0.0028","0.0014","0.0019","4e-04","0","0.0119"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Liver<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":8,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[8,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre><code>2024-08-09 11:45:17 INFO::Annotating ctwas finemapping result ...
2024-08-09 11:45:22 INFO::add gene_name and gene_type
2024-08-09 11:45:22 INFO::use gene mid positions
2024-08-09 11:45:22 INFO::add SNP positions</code></pre>
<div id="htmlwidget-93928d62269dcef60a51" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Combined PIP by omics, credible set filtered<\/caption>","data":[["68","41","61","42","34","29","82","37","76","74","49","57","5","12","81","38","4","64","20","44","75","73","67","33","14","90","11","43","24","63","1","3","66","15","22","54","92"],["RMDN1","LDLR","PNPO","LIPC","HLA-DMA","GIGYF1","TRPS1","HP","ST3GAL4","SLC39A8","NYNRIN","PKN3","C18orf32","CNIH4","TNKS","IL1RN","C10orf88","PXK","EBAG9","MICA","SPTY2D1","SERPINF2","RAC1","HLA-A","CWF19L1","XPNPEP3","CHMP1A","LRRK2","EVI5","PRMT6","ABHD12","BRI3","RAB2A","CYP4F12","EFNA1","PELI3","ZNF554"],[1.9572,1,1,1,1,0.9996,0.9995,0.9986,0.9965,0.9964,0.9959,0.9943,0.9933,0.9932,0.9918,0.9898,0.9889,0.9828,0.9788,0.9766,0.9746,0.9736,0.9647,0.9638,0.9612,0.9603,0.9513,0.942,0.9366,0.9207,0.913,0.9019,0.8609,0.86,0.841,0.834,0.8064],[0.8464,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.9766,null,0.3894,null,0.9638,null,0.2912,null,null,null,null,0.3177,null,null,0.0184,null,null,null],[0.9073,null,1,null,null,null,null,0.9986,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.3256,null,null,0.4217,null,null,null,null,null,null,null,0.8416,0.841,null,null],[null,null,null,null,null,null,null,null,null,0.9964,null,0.2207,0.9933,null,null,null,0.9889,null,0.9788,null,null,0.5841,0.3246,null,null,null,null,null,0.9366,null,null,null,0.1223,null,null,null,null],[0.2034,1,null,1,1,0.9996,0.9995,null,0.9965,null,0.9959,0.7736,null,0.9932,0.9918,0.9898,null,0.9828,null,null,0.9746,null,0.2946,null,0.9612,0.2014,null,0.942,null,0.9207,0.5953,0.9019,0.7385,null,null,0.834,0.8064],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.046,0.9513,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.0199,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>sQTL_pip<\/th>\n      <th>isoQTL_pip<\/th>\n      <th>rsQTL_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>tssQTL_pip<\/th>\n      <th>apaQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="compare-the-results-from-predictdb-munro-weights"
class="section level3">
<h3>Compare the results from Predictdb &amp; Munro weights</h3>
<div id="htmlwidget-16707cd634e93674753d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Overlapped genes, not filtered by combined_pip, filtered by credible sets<\/caption>","data":[["15","16","13","18","5","34","3","27","31","7","35","17","26","2","8","22","36","12","21","10","24","30","33","11","9","28","29","23","19","32","4","14","6","25","20","1"],["LDLR","LIPC","HP","NYNRIN","CNIH4","TNKS","C10orf88","PXK","SPTY2D1","CWF19L1","XPNPEP3","LRRK2","PRMT6","BRI3","CYP4F12","PELI3","ZNF554","GNMT","PARP9","ERGIC3","PGS1","SEC16A","TMEM57","FDPS","DMTN","RPS6","SCAMP3","PEX6","PAQR9","THOP1","C5","INPP5E","CNPY3","PLEC","PARP10","ACMSD"],[1,0.8464,0.9768,0.3673,0.9728,0.9545,0.9548,0.2904,0.9217,0.8968,0.7064,0.9118,0.8539,0.9099,0.8951,0.2352,0.0851,0.0157,0.967,0.5829,0.7184,0.3278,0.6571,0.0849,0.3694,0.1988,0.5152,0.4756,0.2187,0.8464,0.3337,0.6174,0.0366,0.0131,0.0277,0.6605],[null,null,null,0.3673,0.9728,0.9545,0.9548,null,0.9217,0.8968,null,0.9118,0.8539,0.8063,null,0.2352,0.0249,null,null,null,null,0.0537,0.0097,null,null,null,null,0.4756,0.2187,null,null,0.1296,null,0.0077,0.0277,null],[1,0.8464,0.9768,null,null,null,null,0.2904,null,null,0.7064,null,null,0.1037,0.8951,null,0.0602,0.0157,0.967,0.5829,0.7184,0.2741,0.6474,0.0849,0.3694,0.1988,0.5152,null,null,0.8464,0.3337,0.4878,0.0366,0.0054,null,0.6605],[1,1,0.9986,0.9959,0.9932,0.9918,0.9889,0.9828,0.9746,0.9612,0.9603,0.942,0.9207,0.9019,0.86,0.834,0.8064,0.7756,0.6995,0.6614,0.6243,0.5941,0.5428,0.4867,0.4825,0.4779,0.4236,0.1659,0.1394,0.1148,0.0228,0.0187,0.0166,0.0078,0.0046,0.0045],[null,null,null,null,null,null,null,null,null,null,0.2912,null,null,null,0.0184,null,null,null,0.3604,null,0.6243,0.2816,null,0.2032,null,null,null,null,null,null,null,0.0187,null,null,null,0.0045],[null,null,0.9986,null,null,null,null,null,null,null,0.4217,null,null,null,0.8416,null,null,null,0.3392,null,null,null,0.5428,0.2835,null,0.4779,null,null,0.1394,null,null,null,null,null,0.0046,null],[null,null,null,null,null,null,0.9889,null,null,null,null,null,null,null,null,null,null,null,null,0.3798,null,null,null,null,null,null,0.4236,null,null,null,null,null,null,null,null,null],[1,1,null,0.9959,0.9932,0.9918,null,0.9828,0.9746,0.9612,0.2014,0.942,0.9207,0.9019,null,0.834,0.8064,0.7756,null,0.2816,null,0.0132,null,null,0.4825,null,null,0.1659,null,null,0.0228,null,0.0166,0.0078,null,null],[null,null,null,null,null,null,null,null,null,null,0.046,null,null,null,null,null,null,null,null,null,null,0.2993,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.1148,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip_predictdb<\/th>\n      <th>eQTL_predictdb<\/th>\n      <th>sQTL_predictdb<\/th>\n      <th>combined_pip_munro<\/th>\n      <th>sQTL_munro<\/th>\n      <th>isoQTL_munro<\/th>\n      <th>rsQTL_munro<\/th>\n      <th>eQTL_munro<\/th>\n      <th>tssQTL_munro<\/th>\n      <th>apaQTL_munro<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>If we filter by combined pip &gt;0.8 in both settings, we have</p>
<div id="htmlwidget-37e46cf92521dcbed3f8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Overlapped genes, combined_pip &gt; 0.8<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["BRI3","C10orf88","CNIH4","CWF19L1","CYP4F12","HP","LDLR","LIPC","LRRK2","PRMT6","SPTY2D1","TNKS"],[0.9099,0.9548,0.9728,0.8968,0.8951,0.9768,1,0.8464,0.9118,0.8539,0.9217,0.9545],[0.8063,0.9548,0.9728,0.8968,null,null,null,null,0.9118,0.8539,0.9217,0.9545],[0.1037,null,null,null,0.8951,0.9768,1,0.8464,null,null,null,null],[0.9019,0.9889,0.9932,0.9612,0.86,0.9986,1,1,0.942,0.9207,0.9746,0.9918],[null,null,null,null,0.0184,null,null,null,null,null,null,null],[null,null,null,null,0.8416,0.9986,null,null,null,null,null,null],[null,0.9889,null,null,null,null,null,null,null,null,null,null],[0.9019,null,0.9932,0.9612,null,null,1,1,0.942,0.9207,0.9746,0.9918],[null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip_predictdb<\/th>\n      <th>eQTL_predictdb<\/th>\n      <th>sQTL_predictdb<\/th>\n      <th>combined_pip_munro<\/th>\n      <th>sQTL_munro<\/th>\n      <th>isoQTL_munro<\/th>\n      <th>rsQTL_munro<\/th>\n      <th>eQTL_munro<\/th>\n      <th>tssQTL_munro<\/th>\n      <th>apaQTL_munro<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="checking-why-predicdb-results-missed-many-munro-genes"
class="section level3">
<h3>Checking why Predicdb results missed many Munro genes</h3>
<pre><code>[1] &quot;# of Unique munro genes = 25&quot;</code></pre>
<pre><code>[1] &quot;# of Unique munro genes included in predictdb data = 18&quot;</code></pre>
<div id="htmlwidget-99e959167faff8d1fc37" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-99e959167faff8d1fc37">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Unique munro genes in predictdb setting<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69"],[1,1,1,1,1,3,3,3,3,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,7,7,7,7,7,7,8,8,8,11,11,11,11,11,11,11,14,17,17,17,17,17,18,18,19,19,20,20,20,20,20,22,22,22,22,22],[92650550,92664696,92606492,155131158,155132568,58379503,58418858,58367678,58401620,29943572,29917052,29889656,29945162,29917765,29944448,29943147,29916432,29944838,29976880,31407550,29888346,31407122,31410934,31411653,31412275,30261533,32958854,32951876,32950204,32950153,32949505,32949071,32949070,6389250,6401028,100684394,100684197,100683969,100683725,60570242,60588336,115609084,66472041,66470073,126408882,126408250,126407866,126381199,126380969,24409037,1749052,1750663,1745552,1750862,1744034,49482251,49488962,2827824,2812744,25331328,25320202,25316009,25298624,25302948,40894956,40861206,40863122,40859432,40858962],["ENSG00000067208.14","intron_1_92663452_92665939","intron_1_92605402_92607581","ENSG00000169242.11","intron_1_155131634_155133503","ENSG00000168297.15","intron_3_58412963_58424752","intron_3_58365924_58369431","intron_3_58399377_58403862","ENSG00000206503.12","intron_6_29889604_29944500","intron_6_29889604_29889707","intron_6_29945091_29945234","intron_6_29890296_29945234","intron_6_29944397_29944500","intron_6_29943026_29943268","intron_6_29888742_29944122","intron_6_29944616_29945059","intron_6_29944616_30009145","ENSG00000204520.12","intron_6_29888224_29888467","intron_6_31403702_31410543","intron_6_31410797_31411072","intron_6_31411359_31411947","intron_6_31412225_31412325","intron_6_30261479_30261587","ENSG00000204257.14","intron_6_32950803_32952949","intron_6_32949889_32950519","intron_6_32949787_32950519","intron_6_32949399_32949611","intron_6_32948868_32949274","intron_6_32948868_32949271","ENSG00000136238.17","intron_7_6400188_6401868","intron_7_100684337_100684450","intron_7_100684157_100684237","intron_7_100683918_100684020","intron_7_100683632_100683818","ENSG00000104388.14","intron_8_60584815_60591858","ENSG00000104447.12","ENSG00000174516.14","intron_11_66468904_66471242","intron_11_126408496_126409268","intron_11_126408194_126408307","intron_11_126407634_126408099","intron_11_126356302_126406096","intron_11_126355842_126406096","ENSG00000205978.5","ENSG00000167711.13","intron_17_1748740_1752586","intron_17_1745395_1745708","intron_17_1749137_1752586","intron_17_1743076_1744992","ENSG00000177576.11","intron_18_49488566_49489359","ENSG00000172006.11","intron_19_2812372_2813115","intron_20_25323430_25339227","intron_20_25317078_25323325","intron_20_25314970_25317048","intron_20_25295030_25302219","intron_20_25302346_25303550","ENSG00000196236.12","intron_22_40860793_40861618","intron_22_40857245_40868999","intron_22_40857245_40861618","intron_22_40857245_40860678"],["EVI5","EVI5","EVI5","EFNA1","EFNA1","PXK","PXK","PXK","PXK","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","HLA-A","MICA","HLA-A","MICA","MICA","MICA","MICA","HLA-A","HLA-DMA","HLA-DMA","HLA-DMA","HLA-DMA","HLA-DMA","HLA-DMA","HLA-DMA","RAC1","RAC1","GIGYF1","GIGYF1","GIGYF1","GIGYF1","RAB2A","RAB2A","TRPS1","PELI3","PELI3","ST3GAL4","ST3GAL4","ST3GAL4","ST3GAL4","ST3GAL4","NYNRIN","SERPINF2","SERPINF2","SERPINF2","SERPINF2","SERPINF2","C18orf32","C18orf32","ZNF554","ZNF554","ABHD12","ABHD12","ABHD12","ABHD12","ABHD12","XPNPEP3","XPNPEP3","XPNPEP3","XPNPEP3","XPNPEP3"],["protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding"],["ENSG00000067208.14|expression_Liver","intron_1_92663452_92665939|splicing_Liver","intron_1_92605402_92607581|splicing_Liver","ENSG00000169242.11|expression_Liver","intron_1_155131634_155133503|splicing_Liver","ENSG00000168297.15|expression_Liver","intron_3_58412963_58424752|splicing_Liver","intron_3_58365924_58369431|splicing_Liver","intron_3_58399377_58403862|splicing_Liver","ENSG00000206503.12|expression_Liver","intron_6_29889604_29944500|splicing_Liver","intron_6_29889604_29889707|splicing_Liver","intron_6_29945091_29945234|splicing_Liver","intron_6_29890296_29945234|splicing_Liver","intron_6_29944397_29944500|splicing_Liver","intron_6_29943026_29943268|splicing_Liver","intron_6_29888742_29944122|splicing_Liver","intron_6_29944616_29945059|splicing_Liver","intron_6_29944616_30009145|splicing_Liver","ENSG00000204520.12|expression_Liver","intron_6_29888224_29888467|splicing_Liver","intron_6_31403702_31410543|splicing_Liver","intron_6_31410797_31411072|splicing_Liver","intron_6_31411359_31411947|splicing_Liver","intron_6_31412225_31412325|splicing_Liver","intron_6_30261479_30261587|splicing_Liver","ENSG00000204257.14|expression_Liver","intron_6_32950803_32952949|splicing_Liver","intron_6_32949889_32950519|splicing_Liver","intron_6_32949787_32950519|splicing_Liver","intron_6_32949399_32949611|splicing_Liver","intron_6_32948868_32949274|splicing_Liver","intron_6_32948868_32949271|splicing_Liver","ENSG00000136238.17|expression_Liver","intron_7_6400188_6401868|splicing_Liver","intron_7_100684337_100684450|splicing_Liver","intron_7_100684157_100684237|splicing_Liver","intron_7_100683918_100684020|splicing_Liver","intron_7_100683632_100683818|splicing_Liver","ENSG00000104388.14|expression_Liver","intron_8_60584815_60591858|splicing_Liver","ENSG00000104447.12|expression_Liver","ENSG00000174516.14|expression_Liver","intron_11_66468904_66471242|splicing_Liver","intron_11_126408496_126409268|splicing_Liver","intron_11_126408194_126408307|splicing_Liver","intron_11_126407634_126408099|splicing_Liver","intron_11_126356302_126406096|splicing_Liver","intron_11_126355842_126406096|splicing_Liver","ENSG00000205978.5|expression_Liver","ENSG00000167711.13|expression_Liver","intron_17_1748740_1752586|splicing_Liver","intron_17_1745395_1745708|splicing_Liver","intron_17_1749137_1752586|splicing_Liver","intron_17_1743076_1744992|splicing_Liver","ENSG00000177576.11|expression_Liver","intron_18_49488566_49489359|splicing_Liver","ENSG00000172006.11|expression_Liver","intron_19_2812372_2813115|splicing_Liver","intron_20_25323430_25339227|splicing_Liver","intron_20_25317078_25323325|splicing_Liver","intron_20_25314970_25317048|splicing_Liver","intron_20_25295030_25302219|splicing_Liver","intron_20_25302346_25303550|splicing_Liver","ENSG00000196236.12|expression_Liver","intron_22_40860793_40861618|splicing_Liver","intron_22_40857245_40868999|splicing_Liver","intron_22_40857245_40861618|splicing_Liver","intron_22_40857245_40860678|splicing_Liver"],["eQTL","sQTL","sQTL","eQTL","sQTL","eQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","eQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","eQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL"],["Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver"],["eQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","eQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","eQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver","sQTL|Liver"],[-5.72477741763396,5.04694432868415,5.38497026860862,-2.30516877022838,0.594631789053508,1.29192127608954,3.30010717695098,6.85245275815915,6.84673276789248,-2.13363290205141,5.49032156978673,0.872572374530639,-1.47737675353953,-0.0798805196975272,-3.64594827863344,1.44104681121874,-3.09117045391798,-4.5813656040939,0.80043683721451,-1.92027111574566,5.40315678429939,-1.18384930158606,2.3152718858459,-1.98919087785103,1.16149820201457,2.38693874387358,-2.36210572751978,2.398087297696,-0.840899897685256,-2.70412572592414,-1.45503970241005,-1.3581011862863,1.35810065429975,2.28346607732718,4.6349219810053,-1.80146787863901,-1.80146787863901,-1.80146787863901,6.03870834729487,4.65200665783408,4.32258905582732,2.84693627986809,5.98819414079698,-1.9180969059048,0.812480102280258,-1.10561209501129,1.20364393377904,-1.08997325818776,-10.0756056164157,7.00995233066296,-0.333799425109881,4.09150410100454,1.8078610471818,-2.78355226337804,-4.02629821045771,-0.287364890048631,-1.06419007613577,4.39397670900371,4.87991686552405,4.65973163304322,-4.69341216216281,-4.6963682432422,0.7506442960078,4.17663938903698,-1.00920211714571,-5.20273961090799,5.20273961090799,5.20273961090799,-5.20273961090798],["1_91420760_93070902","1_91420760_93070902","1_91420760_93070902","1_154797927_156366342","1_154797927_156366342","3_58171792_60287177","3_58171792_60287177","3_58171792_60287177","3_58171792_60287177","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_29770194_30830391","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_32714887_33268720","6_32714887_33268720","6_32714887_33268720","6_32714887_33268720","6_32714887_33268720","6_32714887_33268720","6_32714887_33268720","7_5814895_6534226","7_5814895_6534226","7_100599028_101555972","7_100599028_101555972","7_100599028_101555972","7_100599028_101555972","8_58815541_61416540","8_58815541_61416540","8_115084266_118673218","11_66131160_68238358","11_66131160_68238358","11_124625632_126441425","11_124625632_126441425","11_124625632_126441425","11_124625632_126441425","11_124625632_126441425","14_22549721_24435917","17_1269105_2025437","17_1269105_2025437","17_1269105_2025437","17_1269105_2025437","17_1269105_2025437","18_48413361_50204214","18_48413361_50204214","19_2098397_3019662","19_2098397_3019662","20_24737088_25363595","20_24737088_25363595","20_24737088_25363595","20_24737088_25363595","20_24737088_25363595","22_40149793_42294812","22_40149793_42294812","22_40149793_42294812","22_40149793_42294812","22_40149793_42294812"],[0.0545615229702219,0.00119112738507343,0.00623288037108161,0.00331296085485688,4.55595539172737e-06,0.000118078753397866,0.138384368854326,0.147595179460601,0.142830807960907,2.32423912338575e-08,1.68986690640693e-08,1.62194020436601e-09,3.96102928235109e-09,3.84898679595125e-09,1.12355249548557e-09,5.42894945665395e-09,8.29865291670136e-08,2.09894878144823e-08,2.1445313302948e-09,0.000482701197341839,0.00868398282366156,3.83650562940652e-06,0.000113583275732909,0.00014344048009074,3.70833726637265e-06,1.41324213860372e-06,2.36660986241422e-06,5.95618082077642e-07,6.42894184554876e-07,3.22789215800867e-07,3.81170908364536e-06,8.5515374559364e-07,8.5515339554032e-07,0.000248594228391075,0.259178372507743,0.00031933489737479,0.00031933489737479,0.00031933489737479,0.0362456343442881,0.146663519504551,0.0219037490393514,1.34759980952026e-11,0.235162340115416,4.50557373632421e-08,0,0,0,0,9.72810276778091e-10,0.367284476622913,0.000226623638086898,0.265329963669056,0.000476300335374602,0.00389910694853945,0.206965161315239,2.67799074027408e-08,1.04656543986614e-08,0.0248958861653771,0.0602314244049832,0.190070360383538,0.220354699118856,0.223244005994494,9.29274536753955e-06,0.0256354308148951,0.0153215243761883,0.176605989428462,0.176605989428462,0.176605989428462,0.176605989428461],[28.5213100017262,23.3801712858351,26.4869445959424,13.6369492324719,1.94380278607512,9.970324446656,19.0000115781977,51.5018776434355,51.4464811066282,5.1748701661168,37.4792078777311,6.73439048868983,10.3236303798233,3.48335861484118,23.0340340813327,4.39950771171708,14.5136702804234,28.0827319595309,5.9001191807797,10.5743055514668,32.1535897863161,3.64242183044015,12.1781690601014,11.4849077244491,3.5481497915426,5.50844307547417,3.48579993088844,3.72693028388889,5.12733127693277,6.1200321688104,7.63852600525946,2.83051571697811,2.83051441050082,5.3094931762716,19.8655756346548,21.4547488392945,21.4547488392945,21.4547488392945,39.8901111240717,19.1453098253165,17.4006680903393,7.74433235033854,31.1200230742628,4.18006685805854,1.52023261283005,2.01560091406627,2.21505123905692,1.98534917911968,90.3798816256145,42.3060910330383,1.01159577636275,15.6876004932933,3.81819371487176,7.76507712462545,15.2212410071604,0.98730197371577,1.93641518582091,17.1793537941687,21.9193558549361,20.0687408702636,20.3462849899228,20.3707399687375,1.43507390102041,16.3077738837315,3.60527150607396,26.1319291627004,26.1319291627004,26.1319291627004,26.1319291627004],[0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>pos<\/th>\n      <th>gene_id<\/th>\n      <th>gene_name<\/th>\n      <th>gene_type<\/th>\n      <th>id<\/th>\n      <th>type<\/th>\n      <th>context<\/th>\n      <th>group<\/th>\n      <th>z<\/th>\n      <th>region_id<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>cs_index<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,10,12,13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="ibd-colon_transverse" class="section level2">
<h2>IBD – Colon_Transverse</h2>
<div id="predictdb-eqtl-and-sqtl-1" class="section level3">
<h3>Predictdb: eqtl and sqtl</h3>
<p><img src="figure/multi_group_compare_predictdb_munro_updated.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-4a6133f685aa29daac40" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4a6133f685aa29daac40">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4"],["eQTL","sQTL","SNP","total_pve"],["0.021","0.0306","0.2102","0.2618"],["0.021","0.0306","0","0.0516"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Colon_Transverse<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre><code>2024-08-09 11:45:43 INFO::Annotating ctwas finemapping result ...
2024-08-09 11:45:48 INFO::add gene_name and gene_type
2024-08-09 11:45:48 INFO::split PIPs for traits mapped to multiple genes
2024-08-09 11:45:48 INFO::use gene mid positions
2024-08-09 11:45:48 INFO::add SNP positions</code></pre>
<div id="htmlwidget-d72b6eff4afa454a8832" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d72b6eff4afa454a8832">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Combined PIP by omics, credible set filtered<\/caption>","data":[["38","65","53","33","34","21","36","24","48","54","4","58"],["KANSL1","TYK2","RNF186","IFNGR2","IL1R2","DOCK8","IRF8","ERI3","PTPN2","SBNO2","ATG16L1","SLC26A3"],[1,0.9832,0.9831,0.9742,0.9701,0.9444,0.9436,0.9421,0.9246,0.9157,0.9085,0.8472],[null,null,0.9831,0.9742,0.9701,null,0.9436,null,null,null,0.1555,0.8472],[1,0.9832,null,null,null,0.9444,null,0.9421,0.9246,0.9157,0.753,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL_pip<\/th>\n      <th>sQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="munro-et-al-6-modalities-1" class="section level3">
<h3>Munro et al : 6 modalities</h3>
<p><img src="figure/multi_group_compare_predictdb_munro_updated.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-73d12cd98287c4e0164c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-73d12cd98287c4e0164c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5","6","7","8"],["apaQTL","eQTL","isoQTL","rsQTL","sQTL","tssQTL","SNP","total_pve"],["0.0083","0.0016","0.0138","0.0073","0.0153","0","0.2014","0.2477"],["0.0083","0.0016","0.0138","0.0073","0.0153","0","0","0.0463"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Colon_Transverse<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":8,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[8,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre><code>2024-08-09 11:46:04 INFO::Annotating ctwas finemapping result ...
2024-08-09 11:46:09 INFO::add gene_name and gene_type
2024-08-09 11:46:10 INFO::use gene mid positions
2024-08-09 11:46:10 INFO::add SNP positions</code></pre>
<div id="htmlwidget-0796cbf5802287e6981e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0796cbf5802287e6981e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Combined PIP by omics, credible set filtered<\/caption>","data":[["18","51","50","49","40","35","21","22","38","2","47","24","48","41","30","23","42","37","12","46","16","5"],["GNA12","ZNF585A","ZNF568","WRB","SH3BGR","P4HTM","HLA-DQA1","HLA-DRB1","RTEL1","ARIH2","USP4","IP6K2","VPS25","SLC6A7","MICB","HMGN2","SRP54","RGS14","COQ8B","UBE2L3","EIF3C","AUH"],[1.9636,1.9501,1.859,1.5654,1.2401,1.0162,1,1,1,1,1,0.9996,0.9992,0.9983,0.9952,0.9915,0.959,0.949,0.8941,0.8489,0.8261,0.8091],[null,1.9327,0.8508,0.5655,1.2401,null,null,1,null,null,null,null,null,0.9983,null,null,null,0.2509,0.8941,null,0.1997,null],[null,0.0174,1,0.9999,null,0.0162,1,null,1,1,null,0.9996,null,null,0.9952,0.9915,0.959,0.6635,null,0.3092,null,0.8091],[null,null,null,null,null,null,null,null,null,null,null,null,0.9992,null,null,null,null,null,null,0.5397,0.6264,null],[0.9642,null,0.0082,null,null,1,null,null,null,null,1,null,null,null,null,null,null,null,null,null,null,null],[0.9994,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.0345,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene_name<\/th>\n      <th>combined_pip<\/th>\n      <th>sQTL_pip<\/th>\n      <th>isoQTL_pip<\/th>\n      <th>rsQTL_pip<\/th>\n      <th>apaQTL_pip<\/th>\n      <th>eQTL_pip<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="compare-the-results-from-predictdb-munro-weights-1"
class="section level3">
<h3>Compare the results from Predictdb &amp; Munro weights</h3>
<div id="htmlwidget-35c46d2ddbc1fab8b4dc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-35c46d2ddbc1fab8b4dc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Overlapped genes, not filtered by combined_pip, filtered by credible sets<\/caption>","data":[["8","9","13","16","2","15","12","7","11","3","6","14","5","10","4","1"],["GNA12","IP6K2","RGS14","UBE2L3","AUH","TNFRSF14","MST1","ERAP2","MAP3K8","CISD1","DLD","TMBIM1","DAP","LNPEP","CLN3","ATXN2L"],[0.1751,0.4706,0.4519,0.6802,0.7216,0.2099,0.177,0.3592,0.3721,0.0578,0.0059,0.0166,0.6238,0.12,0.2942,0.0367],[0.1751,null,0.0147,null,null,0.2099,null,0.2298,null,null,null,0.01,0.0237,null,0.2192,null],[null,0.4706,0.4372,0.6802,0.7216,null,0.177,0.1293,0.3721,0.0578,0.0059,0.0066,0.6001,0.12,0.075,0.0367],[1.9636,0.9996,0.949,0.8489,0.8091,0.7642,0.6014,0.4924,0.3628,0.2532,0.0677,0.0611,0.058,0.017,0.0074,0.007],[null,null,0.2509,null,null,0.5178,0.6014,0.037,null,null,null,0.0611,null,null,null,null],[null,0.9996,0.6635,0.3092,0.8091,0.2464,null,0.0846,0.0328,0.1352,0.0206,null,null,0.0113,null,0.007],[null,null,null,0.5397,null,null,null,0.1753,0.3071,0.118,null,null,0.058,null,null,null],[0.9642,null,null,null,null,null,null,0.1889,0.0228,null,0.0471,null,null,null,0.0074,null],[0.9994,null,0.0345,null,null,null,null,0.0066,null,null,null,null,null,0.0057,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip_predictdb<\/th>\n      <th>eQTL_predictdb<\/th>\n      <th>sQTL_predictdb<\/th>\n      <th>combined_pip_munro<\/th>\n      <th>sQTL_munro<\/th>\n      <th>isoQTL_munro<\/th>\n      <th>rsQTL_munro<\/th>\n      <th>apaQTL_munro<\/th>\n      <th>eQTL_munro<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>If we filter by combined pip &gt;0.8 in both settings, we have</p>
<div id="htmlwidget-2a1b6e0e6b8a75d77da4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a1b6e0e6b8a75d77da4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Overlapped genes, combined_pip &gt; 0.8<\/caption>","data":[[],[],[],[],[],[],[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip_predictdb<\/th>\n      <th>eQTL_predictdb<\/th>\n      <th>sQTL_predictdb<\/th>\n      <th>combined_pip_munro<\/th>\n      <th>sQTL_munro<\/th>\n      <th>isoQTL_munro<\/th>\n      <th>rsQTL_munro<\/th>\n      <th>eQTL_munro<\/th>\n      <th>apaQTL_munro<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p>There’s no overlapped genes at combined_pip &gt; 0.8.</p>
<p>We noticed that, when using Munro’s weights, we have GNA12 as the
top1 IBD risk gene, which has been reported by literature. <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10323775/"
class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10323775/</a></p>
<p>But when using predictdb weights, we missed this gene.</p>
</div>
<div id="checking-why-predicdb-results-missed-many-munro-genes-1"
class="section level3">
<h3>Checking why Predicdb results missed many Munro genes</h3>
<pre><code>[1] &quot;# of Unique munro genes = 22&quot;</code></pre>
<pre><code>[1] &quot;# of Unique munro genes included in predictdb data = 16&quot;</code></pre>
<div id="htmlwidget-f41426fe0b90d0c09df4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f41426fe0b90d0c09df4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Unique munro genes in predictdb setting<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112"],[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,5,5,5,5,5,5,5,5,6,6,6,6,6,6,6,6,7,7,7,9,9,9,9,9,16,16,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,22,22,22],[48714178,48953931,48998525,48694808,48694471,48694966,48694610,48694957,48694665,48694467,48694350,48694134,48689156,48701769,48706289,48705405,48705368,48705366,49003590,49002788,49003347,48716342,48706288,48967678,48953396,48947124,48926150,48921858,48996176,49309311,49312098,49300510,49299551,49324814,49324828,49294037,49279598,150206372,150207514,150207374,150204729,177365219,177371436,177371731,177371634,31503002,31405653,31407122,31410934,31411653,31412275,31463325,32639425,2732600,2786218,2764214,91358892,91349996,91340644,91326980,91215851,28712144,28722098,40689294,40685734,40715611,40714846,40714464,40714206,40714980,40707753,40705253,40704476,40703668,63689317,63689441,63689699,63695714,63695629,63694614,63695020,63687854,63685938,63686755,63661629,63661673,63676081,63676119,63676154,63683121,63684090,63676098,63676790,63681416,39404386,39480680,39395073,39395073,39457258,39457258,39454228,39454228,39391622,39391316,39389276,39392501,39392501,39510354,39513448,21616279,21612800,21621682],["ENSG00000068745.14","ENSG00000177479.19","ENSG00000178467.17","intron_3_48694525_48695090","intron_3_48694240_48694702","intron_3_48694841_48695090","intron_3_48694525_48694694","intron_3_48694824_48695090","intron_3_48694240_48695090","intron_3_48694240_48694694","intron_3_48694240_48694459","intron_3_48693179_48695090","intron_3_48688773_48689538","intron_3_48695418_48708120","intron_3_48695421_48717157","intron_3_48695421_48715389","intron_3_48695421_48715315","intron_3_48695418_48715315","intron_3_49003082_49004098","intron_3_49002596_49002981","intron_3_49002596_49004098","intron_3_48715527_48717157","intron_3_48695418_48717157","intron_3_48967275_48968080","intron_3_48945181_48961612","intron_3_48945181_48949066","intron_3_48924839_48927462","intron_3_48918998_48924718","intron_3_48990914_49001438","ENSG00000114316.12","intron_3_49305888_49318309","intron_3_49298635_49302384","intron_3_49298635_49300467","intron_3_49324763_49324865","intron_3_49324763_49324894","intron_3_49293667_49294407","intron_3_49278451_49280744","ENSG00000011083.8","intron_5_150205623_150209406","intron_5_150205623_150209126","intron_5_150204631_150204827","ENSG00000169220.17","intron_5_177371396_177371475","intron_5_177371589_177371873","intron_5_177371396_177371873","ENSG00000204516.9","intron_6_31400763_31410543","intron_6_31403702_31410543","intron_6_31410797_31411072","intron_6_31411359_31411947","intron_6_31412225_31412325","intron_6_31463279_31463371","intron_6_32637540_32641310","intron_7_2731750_2733451","ENSG00000146535.13","intron_7_2733501_2794928","intron_9_91356155_91361628","intron_9_91344110_91355883","intron_9_91325404_91355883","intron_9_91298076_91355883","intron_9_91214425_91217277","ENSG00000184110.14","intron_16_28721032_28723164","intron_19_40686995_40691592","intron_19_40684556_40686911","intron_19_40714635_40716587","intron_19_40714635_40715057","intron_19_40714397_40714531","intron_19_40714133_40714278","intron_19_40714635_40715325","intron_19_40705447_40710059","intron_19_40705181_40705325","intron_19_40703855_40705096","intron_19_40703622_40703715","intron_20_63689132_63689502","intron_20_63689132_63689750","intron_20_63689648_63689750","intron_20_63695650_63695778","intron_20_63695480_63695778","intron_20_63694488_63694741","intron_20_63694974_63695066","intron_20_63687770_63687937","intron_20_63685872_63686004","intron_20_63685872_63687638","intron_20_63661496_63661762","intron_20_63661496_63661850","intron_20_63674093_63678069","intron_20_63674093_63678145","intron_20_63674093_63678214","intron_20_63680719_63685523","intron_20_63682658_63685523","intron_20_63674051_63678145","intron_20_63675435_63678145","intron_20_63680719_63682113","ENSG00000182093.14","ENSG00000185437.13","intron_21_39393280_39396866","intron_21_39393280_39396866","intron_21_39452141_39462375","intron_21_39452141_39462375","intron_21_39446081_39462375","intron_21_39446081_39462375","intron_21_39391476_39391769","intron_21_39390863_39391769","intron_21_39387855_39390698","intron_21_39391836_39393166","intron_21_39391836_39393166","intron_21_39509027_39511680","intron_21_39511809_39515088","intron_22_21611043_21621515","intron_22_21611043_21614557","intron_22_21611043_21632320"],["IP6K2","ARIH2","P4HTM","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","IP6K2","P4HTM","P4HTM","P4HTM","IP6K2","IP6K2","ARIH2","ARIH2","ARIH2","ARIH2","ARIH2","P4HTM","USP4","USP4","USP4","USP4","USP4","USP4","USP4","USP4","SLC6A7","SLC6A7","SLC6A7","SLC6A7","RGS14","RGS14","RGS14","RGS14","MICB","MICB","MICB","MICB","MICB","MICB","MICB","HLA-DQA1","GNA12","GNA12","GNA12","AUH","AUH","AUH","AUH","AUH","EIF3C","EIF3C","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","COQ8B","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","RTEL1","WRB","SH3BGR","WRB","SH3BGR","WRB","SH3BGR","WRB","SH3BGR","WRB","WRB","WRB","WRB","SH3BGR","SH3BGR","SH3BGR","UBE2L3","UBE2L3","UBE2L3"],["protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding","protein_coding"],["ENSG00000068745.14|expression_Colon_Transverse","ENSG00000177479.19|expression_Colon_Transverse","ENSG00000178467.17|expression_Colon_Transverse","intron_3_48694525_48695090|splicing_Colon_Transverse","intron_3_48694240_48694702|splicing_Colon_Transverse","intron_3_48694841_48695090|splicing_Colon_Transverse","intron_3_48694525_48694694|splicing_Colon_Transverse","intron_3_48694824_48695090|splicing_Colon_Transverse","intron_3_48694240_48695090|splicing_Colon_Transverse","intron_3_48694240_48694694|splicing_Colon_Transverse","intron_3_48694240_48694459|splicing_Colon_Transverse","intron_3_48693179_48695090|splicing_Colon_Transverse","intron_3_48688773_48689538|splicing_Colon_Transverse","intron_3_48695418_48708120|splicing_Colon_Transverse","intron_3_48695421_48717157|splicing_Colon_Transverse","intron_3_48695421_48715389|splicing_Colon_Transverse","intron_3_48695421_48715315|splicing_Colon_Transverse","intron_3_48695418_48715315|splicing_Colon_Transverse","intron_3_49003082_49004098|splicing_Colon_Transverse","intron_3_49002596_49002981|splicing_Colon_Transverse","intron_3_49002596_49004098|splicing_Colon_Transverse","intron_3_48715527_48717157|splicing_Colon_Transverse","intron_3_48695418_48717157|splicing_Colon_Transverse","intron_3_48967275_48968080|splicing_Colon_Transverse","intron_3_48945181_48961612|splicing_Colon_Transverse","intron_3_48945181_48949066|splicing_Colon_Transverse","intron_3_48924839_48927462|splicing_Colon_Transverse","intron_3_48918998_48924718|splicing_Colon_Transverse","intron_3_48990914_49001438|splicing_Colon_Transverse","ENSG00000114316.12|expression_Colon_Transverse","intron_3_49305888_49318309|splicing_Colon_Transverse","intron_3_49298635_49302384|splicing_Colon_Transverse","intron_3_49298635_49300467|splicing_Colon_Transverse","intron_3_49324763_49324865|splicing_Colon_Transverse","intron_3_49324763_49324894|splicing_Colon_Transverse","intron_3_49293667_49294407|splicing_Colon_Transverse","intron_3_49278451_49280744|splicing_Colon_Transverse","ENSG00000011083.8|expression_Colon_Transverse","intron_5_150205623_150209406|splicing_Colon_Transverse","intron_5_150205623_150209126|splicing_Colon_Transverse","intron_5_150204631_150204827|splicing_Colon_Transverse","ENSG00000169220.17|expression_Colon_Transverse","intron_5_177371396_177371475|splicing_Colon_Transverse","intron_5_177371589_177371873|splicing_Colon_Transverse","intron_5_177371396_177371873|splicing_Colon_Transverse","ENSG00000204516.9|expression_Colon_Transverse","intron_6_31400763_31410543|splicing_Colon_Transverse","intron_6_31403702_31410543|splicing_Colon_Transverse","intron_6_31410797_31411072|splicing_Colon_Transverse","intron_6_31411359_31411947|splicing_Colon_Transverse","intron_6_31412225_31412325|splicing_Colon_Transverse","intron_6_31463279_31463371|splicing_Colon_Transverse","intron_6_32637540_32641310|splicing_Colon_Transverse","intron_7_2731750_2733451|splicing_Colon_Transverse","ENSG00000146535.13|expression_Colon_Transverse","intron_7_2733501_2794928|splicing_Colon_Transverse","intron_9_91356155_91361628|splicing_Colon_Transverse","intron_9_91344110_91355883|splicing_Colon_Transverse","intron_9_91325404_91355883|splicing_Colon_Transverse","intron_9_91298076_91355883|splicing_Colon_Transverse","intron_9_91214425_91217277|splicing_Colon_Transverse","ENSG00000184110.14|expression_Colon_Transverse","intron_16_28721032_28723164|splicing_Colon_Transverse","intron_19_40686995_40691592|splicing_Colon_Transverse","intron_19_40684556_40686911|splicing_Colon_Transverse","intron_19_40714635_40716587|splicing_Colon_Transverse","intron_19_40714635_40715057|splicing_Colon_Transverse","intron_19_40714397_40714531|splicing_Colon_Transverse","intron_19_40714133_40714278|splicing_Colon_Transverse","intron_19_40714635_40715325|splicing_Colon_Transverse","intron_19_40705447_40710059|splicing_Colon_Transverse","intron_19_40705181_40705325|splicing_Colon_Transverse","intron_19_40703855_40705096|splicing_Colon_Transverse","intron_19_40703622_40703715|splicing_Colon_Transverse","intron_20_63689132_63689502|splicing_Colon_Transverse","intron_20_63689132_63689750|splicing_Colon_Transverse","intron_20_63689648_63689750|splicing_Colon_Transverse","intron_20_63695650_63695778|splicing_Colon_Transverse","intron_20_63695480_63695778|splicing_Colon_Transverse","intron_20_63694488_63694741|splicing_Colon_Transverse","intron_20_63694974_63695066|splicing_Colon_Transverse","intron_20_63687770_63687937|splicing_Colon_Transverse","intron_20_63685872_63686004|splicing_Colon_Transverse","intron_20_63685872_63687638|splicing_Colon_Transverse","intron_20_63661496_63661762|splicing_Colon_Transverse","intron_20_63661496_63661850|splicing_Colon_Transverse","intron_20_63674093_63678069|splicing_Colon_Transverse","intron_20_63674093_63678145|splicing_Colon_Transverse","intron_20_63674093_63678214|splicing_Colon_Transverse","intron_20_63680719_63685523|splicing_Colon_Transverse","intron_20_63682658_63685523|splicing_Colon_Transverse","intron_20_63674051_63678145|splicing_Colon_Transverse","intron_20_63675435_63678145|splicing_Colon_Transverse","intron_20_63680719_63682113|splicing_Colon_Transverse","ENSG00000182093.14|expression_Colon_Transverse","ENSG00000185437.13|expression_Colon_Transverse","intron_21_39393280_39396866|splicing_Colon_Transverse","intron_21_39393280_39396866|splicing_Colon_Transverse","intron_21_39452141_39462375|splicing_Colon_Transverse","intron_21_39452141_39462375|splicing_Colon_Transverse","intron_21_39446081_39462375|splicing_Colon_Transverse","intron_21_39446081_39462375|splicing_Colon_Transverse","intron_21_39391476_39391769|splicing_Colon_Transverse","intron_21_39390863_39391769|splicing_Colon_Transverse","intron_21_39387855_39390698|splicing_Colon_Transverse","intron_21_39391836_39393166|splicing_Colon_Transverse","intron_21_39391836_39393166|splicing_Colon_Transverse","intron_21_39509027_39511680|splicing_Colon_Transverse","intron_21_39511809_39515088|splicing_Colon_Transverse","intron_22_21611043_21621515|splicing_Colon_Transverse","intron_22_21611043_21614557|splicing_Colon_Transverse","intron_22_21611043_21632320|splicing_Colon_Transverse"],["eQTL","eQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","eQTL","eQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL","sQTL"],["Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse","Colon_Transverse"],["eQTL|Colon_Transverse","eQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","eQTL|Colon_Transverse","eQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse","sQTL|Colon_Transverse"],[-1.22846441947636,-0.699999999999753,4.37931034483001,-1.22846441947636,1.22846441947636,1.22846441947636,-1.22846441947636,1.22846441947636,-3.12299656742608,1.22846441947636,-1.22846441947636,4.04464155711491,1.05426356589134,-1.04324324324316,0.309014404751288,-1.04324324324316,-1.04324324324316,-1.04324324324316,-4.37931034483,-4.37931034483001,4.37931034483001,-7.73369565217247,0.0815899581590048,-0.651162790697901,3.57736629840788,-0.915384615384442,2.04751131221784,-2.18396226415097,-0.674418604650838,-3.45390070922073,-6.70202020201998,-3.47999999999943,3.47999999999943,8.29113924050805,-8.29113924050805,-8.30188679245316,5.76213592233124,-2.00870707512983,0.547619047618886,-0.547619047618886,-2.32266666666687,-5.06886423626525,4.83687943262495,4.83687943262495,-5.89180800234971,-4.85330614609877,-0.453101110651638,-1.81690140845133,-1.49557522123867,-1.61306191154358,0.231597845601364,3.9444444444446,-3.03568613544094,-2.60000000000011,-4.4614742601732,2.71978021978088,-2.23859649122776,3.74550356594739,-2.95867768595152,3.53201970443559,5.10975609756247,6.16666666666469,4.1584507042246,0.893617021276951,0.272727272727328,3.55345911949643,-3.55345911949643,-3.55345911949643,-3.55345911949643,2.43728813559267,0.277611940298508,0.277611940298508,3.00735294117726,-3.00735294117726,7.87511801405971,-7.87999999999702,8.54158403994094,-3.92195121951099,3.93554269619764,-2.22957695037388,2.22947368420977,3.03643724696313,2.74576271186504,-2.74576271186504,-4.28205128205262,4.28205128205262,-0.291139240506376,-8.76470588235075,8.76470588235075,-7.86666666666687,-2.78099173553865,3.10410094637039,8.50793650793659,7.23197078945087,1.79852312492385,0.85232274663987,-1.70289855072478,-1.70289855072478,0.296610169491592,0.296610169491592,-0.0661157024793522,-0.0661157024793522,-1.79769260133869,-0.186936936936964,-2.16666666666763,2.17424242424173,2.17424242424173,0.995584795714167,-0.256410256410332,7.86363636363289,-7.86363636363289,-7.86363636363289],["3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_47685722_49279539","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","3_49279539_51797999","5_149283061_151181737","5_149283061_151181737","5_149283061_151181737","5_149283061_151181737","5_176023592_177824506","5_176023592_177824506","5_176023592_177824506","5_176023592_177824506","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_30830391_31603441","6_31603441_32714887","7_2022763_2732593","7_2732593_4533797","7_2732593_4533797","9_89827251_91404921","9_89827251_91404921","9_89827251_91404921","9_89827251_91404921","9_89827251_91404921","16_27434434_29025292","16_27434434_29025292","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","19_40478694_41625199","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","20_63558827_64333810","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","21_39110976_40017600","22_19924835_22002927","22_19924835_22002927","22_19924835_22002927"],[8.68185895064322e-07,1.99838588766932e-05,7.48558530756505e-05,3.15258352978631e-07,3.15258352978631e-07,3.15258352978631e-07,3.15258352978631e-07,3.15258352978631e-07,0.00145336746032743,3.15258352978631e-07,3.15258352978631e-07,0.00096390236619448,5.71541828231492e-06,3.20394497599352e-07,2.7822527037813e-07,3.20394497599352e-07,3.20394497599352e-07,3.20394497599352e-07,2.97776416768869e-05,2.97776416768869e-05,2.97776416768869e-05,0.470597733682621,3.00261080887765e-07,8.74269557515905e-06,3.90981037645677e-06,2.76689050449086e-07,2.23225925133352e-06,1.00191577712128e-06,8.21064290890128e-06,0,0,0,0,0,0,0,0,1.92288892113535e-05,1.67188987265643e-06,1.67188987265643e-06,1.41274109635159e-05,0.0146691013907916,0.00232839323281375,0.00232839323281375,0.437238287946096,0.000835218790457226,2.34517960606695e-12,1.3969381207346e-12,1.76025860554319e-12,1.56397117478946e-12,1.5351053761492e-12,3.79037357056689e-11,0,0.000839279492081069,0.175068946068094,0.000234181237190323,4.15190401605869e-05,0.00269005209314666,0.000234458701776297,0.00131115098029166,0.7188909501882,5.23697159282577e-06,1.8433821136199e-10,1.80805731697098e-05,1.29340381742615e-05,0.00859928227978013,0.00859928227978013,0.00859928227978013,0.00859928227978013,0.000390101711686075,2.59002626865623e-05,2.59002626865623e-05,0.00163944513556846,0.00163944513556846,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.00130984338459106,0.000420192026927557,0.000209287286302651,0.000209287286302651,5.69988128140686e-05,5.69988128140686e-05,5.48365621719227e-05,5.48365621719227e-05,0.000488016940620506,0.00011123519580114,0.000960029280147157,0.000487372313937307,0.000487372313937307,0.000173117546934165,0.000112831345424391,0.226743283200876,0.226743283200878,0.226743283200878],[2.58992398216323,11.3386696720282,19.0811487198656,2.65914582652051,2.65914582652051,2.65914582652051,2.65914582652051,2.65914582652051,17.7289713566877,2.65914582652051,2.65914582652051,17.7863717253314,11.6117004798789,2.55655483235235,1.89634330760263,2.55655483235235,2.55655483235235,2.55655483235235,19.8322342877722,19.8322342877722,19.8322342877722,58.0285251041685,2.14760510836437,11.8250473982187,12.4165812046885,1.99117960213026,8.29562066479948,7.05784081740635,11.7928667249857,10.7111340485142,39.3679529524375,11.2899613208855,11.2899613208855,59.7596521902365,59.7596521902365,59.9122818633714,29.3416437719151,6.27660330301687,1.98771731034201,1.98771731034201,8.43678472399353,22.0234508575595,20.9483680900668,20.9483680900667,30.6350150968917,49.5486090340929,7.00621491479482,3.62581639907654,3.56814720739284,3.34891887026478,3.30455263748149,14.1662487582424,31.5805863728682,6.71075640619045,17.2658908329309,7.2561155795652,5.21412243897187,12.9318446939332,8.41715183342697,11.6021483356718,23.2713532206926,32.1608399623312,15.7253062984702,1.60857809874553,0.988787356228155,11.7321610913219,11.7321610913219,11.7321610913219,11.7321610913219,6.00926905964453,0.991088105525901,0.991088105525901,8.66569254803065,8.66569254803065,5512.01076965322,5512.7211068019,4852.55237613582,389.537957336714,370.569366987693,55.9108712712813,55.7757430483106,158.921080867398,157.374708537193,157.374708537193,197.159775807978,197.159775807978,490.457755229865,3604.99017036342,3604.99017036342,5409.22942020314,174.880752668641,125.303128069937,1584.12373198456,1106.6854658464,3.56509864762148,1.50363384313447,3.40701161122311,3.40701161122311,1.00042487981113,1.00042487981113,0.928869405497278,0.928869405497278,3.69101681582446,0.955036648467086,4.94292556147471,4.9710711663332,4.9710711663332,1.773449421321,0.98139788879238,53.8488857573721,53.8488857573721,53.8488857573721],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,1,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chrom<\/th>\n      <th>pos<\/th>\n      <th>gene_id<\/th>\n      <th>gene_name<\/th>\n      <th>gene_type<\/th>\n      <th>id<\/th>\n      <th>type<\/th>\n      <th>context<\/th>\n      <th>group<\/th>\n      <th>z<\/th>\n      <th>region_id<\/th>\n      <th>susie_pip<\/th>\n      <th>mu2<\/th>\n      <th>cs_index<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,10,12,13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cowplot_1.1.1             ggrepel_0.9.1            
 [3] locuszoomr_0.2.1          logging_0.10-108         
 [5] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.20.2         
 [7] AnnotationFilter_1.20.0   GenomicFeatures_1.48.3   
 [9] AnnotationDbi_1.58.0      Biobase_2.56.0           
[11] GenomicRanges_1.48.0      GenomeInfoDb_1.39.9      
[13] IRanges_2.30.0            S4Vectors_0.34.0         
[15] BiocGenerics_0.42.0       gridExtra_2.3            
[17] forcats_0.5.1             stringr_1.5.1            
[19] dplyr_1.1.4               purrr_1.0.2              
[21] readr_2.1.2               tidyr_1.3.0              
[23] tibble_3.2.1              ggplot2_3.5.1            
[25] tidyverse_1.3.1           data.table_1.14.2        
[27] ctwas_0.4.5              

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              fs_1.5.2                   
  [7] rstudioapi_0.13             farver_2.1.0               
  [9] DT_0.22                     bit64_4.0.5                
 [11] lubridate_1.8.0             fansi_1.0.3                
 [13] xml2_1.3.3                  codetools_0.2-18           
 [15] cachem_1.0.6                knitr_1.39                 
 [17] jsonlite_1.8.0              workflowr_1.7.0            
 [19] Rsamtools_2.12.0            broom_0.8.0                
 [21] dbplyr_2.1.1                png_0.1-7                  
 [23] compiler_4.2.0              httr_1.4.3                 
 [25] backports_1.4.1             assertthat_0.2.1           
 [27] Matrix_1.5-3                fastmap_1.1.0              
 [29] lazyeval_0.2.2              cli_3.6.1                  
 [31] later_1.3.0                 htmltools_0.5.2            
 [33] prettyunits_1.1.1           tools_4.2.0                
 [35] gtable_0.3.0                glue_1.6.2                 
 [37] GenomeInfoDbData_1.2.8      rappdirs_0.3.3             
 [39] Rcpp_1.0.12                 cellranger_1.1.0           
 [41] jquerylib_0.1.4             vctrs_0.6.5                
 [43] Biostrings_2.64.0           rtracklayer_1.56.0         
 [45] crosstalk_1.2.0             xfun_0.41                  
 [47] rvest_1.0.2                 lifecycle_1.0.4            
 [49] irlba_2.3.5                 restfulr_0.0.14            
 [51] XML_3.99-0.14               zlibbioc_1.42.0            
 [53] zoo_1.8-10                  scales_1.3.0               
 [55] gggrid_0.2-0                hms_1.1.1                  
 [57] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [59] ProtGenerics_1.28.0         parallel_4.2.0             
 [61] SummarizedExperiment_1.26.1 LDlinkR_1.2.3              
 [63] yaml_2.3.5                  curl_4.3.2                 
 [65] memoise_2.0.1               sass_0.4.1                 
 [67] biomaRt_2.54.1              stringi_1.7.6              
 [69] RSQLite_2.3.1               highr_0.9                  
 [71] BiocIO_1.6.0                filelock_1.0.2             
 [73] BiocParallel_1.30.3         rlang_1.1.2                
 [75] pkgconfig_2.0.3             matrixStats_0.62.0         
 [77] bitops_1.0-7                evaluate_0.15              
 [79] lattice_0.20-45             labeling_0.4.2             
 [81] GenomicAlignments_1.32.0    htmlwidgets_1.5.4          
 [83] bit_4.0.4                   tidyselect_1.2.0           
 [85] magrittr_2.0.3              R6_2.5.1                   
 [87] generics_0.1.2              DelayedArray_0.22.0        
 [89] DBI_1.2.2                   withr_2.5.0                
 [91] haven_2.5.0                 pgenlibr_0.3.3             
 [93] pillar_1.9.0                KEGGREST_1.36.3            
 [95] RCurl_1.98-1.7              mixsqp_0.3-43              
 [97] modelr_0.1.8                crayon_1.5.1               
 [99] utf8_1.2.2                  BiocFileCache_2.4.0        
[101] plotly_4.10.0               tzdb_0.4.0                 
[103] rmarkdown_2.25              progress_1.2.2             
[105] readxl_1.4.0                grid_4.2.0                 
[107] blob_1.2.3                  git2r_0.30.1               
[109] reprex_2.0.1                digest_0.6.29              
[111] httpuv_1.6.5                munsell_0.5.0              
[113] viridisLite_0.4.0           bslib_0.3.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
