<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-05-27" />

<title>Multi group analysis: 6 traits, 5 tissues, eQTL + sQTL + apaQTL – Compute UKBB LD</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Multi group analysis: 6 traits, 5 tissues,
eQTL + sQTL + apaQTL – Compute UKBB LD</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-05-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree7d93861ab15960e9038ad6446d26ee4d2e5e75cdtargetblank7d93861a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/7d93861ab15960e9038ad6446d26ee4d2e5e75cd" target="_blank">7d93861</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree7d93861ab15960e9038ad6446d26ee4d2e5e75cdtargetblank7d93861a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/7d93861ab15960e9038ad6446d26ee4d2e5e75cd" target="_blank">7d93861</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    results/

Untracked files:
    Untracked:  analysis/multi_group_6traits_15weights_ukbb.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<div id="traits" class="section level2">
<h2>Traits</h2>
<p>aFib, IBD, LDL, SBP, SCZ, WBC</p>
<p><a
href="https://sq-96.github.io/multigroup_ctwas_analysis/data.html">details</a></p>
</div>
<div id="tissues" class="section level2">
<h2>Tissues</h2>
<p>The independent tissues are selected by <a
href="https://sq-96.github.io/multigroup_ctwas_analysis/matching_tissue_v2.html">single
tissue analysis</a></p>
</div>
<div id="omics" class="section level2">
<h2>Omics</h2>
<p>eQTL, sQTL weights are from GTEx PredictDB</p>
<p>apaQTL wetights are from <a
href="https://www.nature.com/articles/s41467-024-46064-7#Sec2"
class="uri">https://www.nature.com/articles/s41467-024-46064-7#Sec2</a>.
Top 10 SNPs with largest abs(weights) were selected after
harmonization</p>
</div>
<div id="settings" class="section level2">
<h2>Settings</h2>
<ol style="list-style-type: decimal">
<li>Weight processing:</li>
</ol>
<p>PredictDB:</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = TRUE,</li>
<li>load_predictdb_LD = F,</li>
</ul>
<p>FUSION:</p>
<ul>
<li>method_FUSION = “enet”,</li>
<li>fusion_top_n_snps = 10,</li>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = F,</li>
<li>load_predictdb_LD = F,</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Parameter estimation and fine-mapping</li>
</ol>
<ul>
<li>niter_prefit = 5,</li>
<li>niter = 60,</li>
<li>L = 3,</li>
<li>group_prior_var_structure = “shared_type”,</li>
<li>maxSNP = 20000,</li>
<li>min_nonSNP_PIP = 0.5,</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Memory requested</li>
</ol>
<ul>
<li>cpus-per-task=2</li>
<li>mem=200G</li>
</ul>
<p>100G 2core got killed</p>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<p><strong>Results from multi-group analysis</strong></p>
<p>The results are summarized by</p>
<ol style="list-style-type: decimal">
<li><p>Heritability contribution by contexts: we aggregate the PVE
values by omics and tissues, making it easier to understand the
distribution of PVE across different genetic contexts.</p></li>
<li><p>Combined PIP by omics: we aggregate the Susie PIPs by
omics</p></li>
<li><p>Combined PIP by contexts: we aggregate the Susie PIPs by tissues,
making it easier to understand the distribution of PIP across different
genetic contexts.</p></li>
<li><p>Specific molecular traits of top genes: we creates a pie chart to
visualize the proportion of genes classified into different categories
based on their PIPs contributed by each genetics contexts. The
categories are based on the proportion of each QTL type relative to the
combined PIP value:</p></li>
</ol>
<ul>
<li>by eQTL: Number of genes where the ratio of eQTL to combined PIP is
greater than 0.8.</li>
<li>by sQTL: Number of genes where the ratio of sQTL to combined PIP is
greater than 0.8.</li>
<li>by apaQTL: Number of genes where the ratio of apaQTL to combined PIP
is greater than 0.8.</li>
<li>by sQTL+apaQTL: Number of genes where the combined ratio of apaQTL
and sQTL to combined PIP is greater than 0.8, but neither apaQTL nor
sQTL individually exceed 0.8.</li>
<li>unspecified: Number of genes not classified into any of the above
categories.</li>
</ul>
<p><strong>Comparing with single group eQTL results</strong></p>
<p><em>Please not that the ealier single group eQTL analyses were
performed under L=5 but the current analyses were under L=3</em></p>
<p>We compared number of significant genes, overlapping genes and the
changes in PVE for eQTLs across five tissues reported by single eQTL
analysi</p>
<div id="afib" class="section level2">
<h2>aFib</h2>
<p>TO DO</p>
</div>
<div id="ibd" class="section level2">
<h2>IBD</h2>
<div id="results-from-multi-group-analysis" class="section level3">
<h3>Results from multi-group analysis</h3>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5"],["apaQTL","eQTL","sQTL","SNP","total_pve"],["0.0044","0.0508","0.0348","0.1612","0.2512"],["0.0019","0.0094","0.0018","0","0.0131"],["3e-04","0","0.0078","0","0.0081"],["0","0.003","0.0185","0","0.0215"],["0","0.0127","0.0014","0","0.0141"],["0.0022","0.0257","0.0053","0","0.0332"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Cells_Cultured_fibroblasts<\/th>\n      <th>Esophagus_Mucosa<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Whole_Blood<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by omics<\/caption>","data":[["63","47","68","105","58","67","90","59","86","106","9","29","3","23","46","74","97","111","50","96","26","95","76","113","49","123","32","115","89","125","45","2"],["IP6K2","ERRFI1","KANSL1","SPNS1","HLA-DQA1","JAK2","PTPN2","IFNGR2","PHF1","STAT3","ATG16L1","CDK11A","ADAM15","CARD9","ERI3","LSP1","SBNO2","TMEM151B","FGFR1OP","RTEL1","CCR5","RGS14","MAST2","TNFSF15","FCGR2A","ZFP36L2","CIITA","TSPAN14","PPP5C","ZNF300","ERAP2","ACBD3"],[1.244,1.055,1,0.999,0.997,0.997,0.982,0.98,0.969,0.964,0.963,0.963,0.962,0.962,0.962,0.949,0.947,0.947,0.94,0.932,0.927,0.924,0.911,0.902,0.895,0.889,0.851,0.837,0.831,0.825,0.82,0.817],[0,0,0,0.999,0,0,0,0,0.969,0,0,0.963,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.069,0],[0,0.908,0,0,0.997,0.997,0,0.98,0,0.964,0.018,0,0.011,0.93,0,0.505,0,0.947,0.633,0,0.927,0.794,0,0.902,0.895,0.889,0.824,0,0.831,0.802,0.329,0.817],[1.244,0.147,1,0,0,0,0.982,0,0,0,0.944,0,0.951,0.033,0.962,0.444,0.947,0,0.308,0.932,0,0.13,0.911,0,0,0,0.027,0.837,0,0.023,0.422,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>apaQTL<\/th>\n      <th>eQTL<\/th>\n      <th>sQTL<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by tissue<\/caption>","data":[["63","47","68","105","58","67","90","59","86","106","9","29","3","23","46","74","97","111","50","96","26","95","76","113","49","123","32","115","89","125","45","2"],["IP6K2","ERRFI1","KANSL1","SPNS1","HLA-DQA1","JAK2","PTPN2","IFNGR2","PHF1","STAT3","ATG16L1","CDK11A","ADAM15","CARD9","ERI3","LSP1","SBNO2","TMEM151B","FGFR1OP","RTEL1","CCR5","RGS14","MAST2","TNFSF15","FCGR2A","ZFP36L2","CIITA","TSPAN14","PPP5C","ZNF300","ERAP2","ACBD3"],[1.244,1.055,1,0.999,0.997,0.997,0.982,0.98,0.969,0.964,0.963,0.963,0.962,0.962,0.962,0.949,0.947,0.947,0.94,0.932,0.927,0.924,0.911,0.902,0.895,0.889,0.851,0.837,0.831,0.825,0.82,0.817],[0.07,0.01,1,0.999,0.997,0,0.068,0,0.969,0,0.039,0.963,0.088,0.228,0.05,0.522,0.06,0,0.566,0,0,0.055,0.05,0,0,0.159,0.824,0,0.009,0,0.092,0],[0.411,0.077,0,0,0,0,0.492,0.169,0,0.061,0.573,0,0.363,0,0.599,0.41,0.888,0,0,0,0,0.212,0.388,0,0.895,0,0.027,0.348,0,0.074,0.32,0],[0.549,0.908,0,0,0,0,0.214,0,0,0.606,0.018,0,0,0.734,0,0.017,0,0,0.015,0.932,0.927,0.029,0.169,0.902,0,0.49,0,0.382,0.499,0.728,0.237,0.817],[0.04,0.007,0,0,0,0.997,0,0.811,0,0.296,0.019,0,0.034,0,0.058,0,0,0.947,0.095,0,0,0.612,0.048,0,0,0.24,0,0.017,0.323,0.011,0.083,0],[0.174,0.053,0,0,0,0,0.209,0,0,0,0.313,0,0.476,0,0.255,0,0,0,0.265,0,0,0.016,0.255,0,0,0,0,0.091,0,0.012,0.088,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Esophagus_Mucosa<\/th>\n      <th>Whole_Blood<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Cells_Cultured_fibroblasts<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/multi_group_6traits_15weights_ukbb.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-with-single-group-eqtl-results"
class="section level3">
<h3>Comparing with single group eQTL results</h3>
<pre><code>[1] &quot;the top tissues from single group analyses are Cells_Cultured_fibroblasts,Whole_Blood,Adipose_Subcutaneous,Esophagus_Mucosa,Heart_Left_Ventricle&quot;</code></pre>
<div id="htmlwidget-93928d62269dcef60a51" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">#of overlap for single eqtl analysis and multi group analysis<\/caption>","data":[["Cells_Cultured_fibroblasts","Whole_Blood","Adipose_Subcutaneous","Esophagus_Mucosa","Heart_Left_Ventricle"],[16,14,12,8,8],[5,7,6,3,5],[32,32,32,32,32]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#of.sig.gene_single_group_cs_filtered<\/th>\n      <th>#of.overlap<\/th>\n      <th>#of.sig.gene_multi_group_cs_filtered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-16707cd634e93674753d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">pve changes<\/caption>","data":[["Cells_Cultured_fibroblasts","Whole_Blood","Adipose_Subcutaneous","Esophagus_Mucosa","Heart_Left_Ventricle"],[0.0103,0.0179,0.012,0.0155,0.0111],[0,0.0257,0.0094,0.003,0.0127]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pve_single_group<\/th>\n      <th>pve_eqtl_multi_group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="ldl" class="section level2">
<h2>LDL</h2>
<div id="results-from-multi-group-analysis-1" class="section level3">
<h3>Results from multi-group analysis</h3>
<div id="htmlwidget-37e46cf92521dcbed3f8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5"],["apaQTL","eQTL","sQTL","SNP","total_pve"],["7e-04","0.0165","0.0162","0.0286","0.062"],["1e-04","0","5e-04","0","6e-04"],["0","0.0014","2e-04","0","0.0016"],["0","0.0016","0.0022","0","0.0038"],["2e-04","0.0072","0.0068","0","0.0142"],["4e-04","0.0063","0.0065","0","0.0132"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Adrenal_Gland<\/th>\n      <th>Esophagus_Mucosa<\/th>\n      <th>Liver<\/th>\n      <th>Spleen<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-99e959167faff8d1fc37" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-99e959167faff8d1fc37">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by omics<\/caption>","data":[["45","192","1","182","18","141","8","23","78","82","85","90","93","191","206","213","11","183","52","26","113","131","81","167","189","215","211","177","69","133","4","79","83","106","35","120","29","126","147","96","168","114","63","87","174","188","49","130","205","57","38","88","138","186","193","39","94","129","102","214","144","72","122","61","162","21","41"],["DNAJC13","TRIM39","ABCA1","THOP1","C10orf88","PSRC1","APOB","CCNJ","HLA-DRB5","HPR","INSIG2","KMT2E","LDLR","TRAK2","VAMP5","ZGPAT","ASGR1","TIMD4","FADS1","CETP","NPC1L1","PKN3","HP","SMARCA4","TNKS","ZNF227","ZDHHC18","SUGP1","GSK3B","PLTP","ADH1B","HMGCR","INHBB","MYPOP","CSNK1G3","PARP9","CNIH4","PELO","RAC1","LRCH4","SP4","NPEPPS","GAS6","KDSR","STAT5B","TMEM57","ERGIC3","PHC1","USP53","FCGRT","CYP2A6","KIF13B","PRKD2","TMEM150A","TRIM5","CYP4F12","LIPC","PGS1","MITF","ZHX3","R3HDM2","HFE","PCSK9","FUT2","SIPA1","CCDC57","DDX56"],[1.16,1.112,1.103,1.01,1.001,1.001,1,1,1,1,1,1,1,1,1,1,0.999,0.999,0.997,0.995,0.995,0.991,0.989,0.988,0.985,0.983,0.981,0.98,0.978,0.976,0.975,0.973,0.968,0.968,0.966,0.965,0.964,0.962,0.961,0.958,0.958,0.952,0.95,0.95,0.949,0.949,0.945,0.942,0.939,0.93,0.913,0.909,0.907,0.897,0.896,0.894,0.891,0.89,0.875,0.875,0.872,0.855,0.84,0.833,0.816,0.813,0.804],[1.017,0.999,0,0,1.001,1.001,1,1,0,1,1,1,0,1,1,1,0,0.999,0.997,0.995,0.995,0.991,0,0,0.985,0,0,0,0.978,0.976,0,0,0.968,0,0.966,0.267,0.123,0.962,0,0,0.958,0.14,0.95,0.95,0.949,0.028,0,0.013,0.24,0.138,0.913,0,0.884,0.342,0,0.313,0,0,0.875,0,0,0.855,0.84,0.833,0.417,0.015,0.804],[0.143,0.113,1.103,0.533,0,0,0,0,0,0,0,0,1,0,0,0,0.999,0,0,0,0,0,0.989,0,0,0.983,0.981,0.98,0,0,0.975,0.973,0,0.968,0,0.697,0.841,0,0,0.958,0,0.812,0,0,0,0.921,0.945,0.928,0.699,0.792,0,0.909,0.022,0.555,0.896,0.581,0.891,0.89,0,0.875,0.872,0,0,0,0.399,0.601,0],[0,0,0,0.478,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.988,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.961,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.197,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL<\/th>\n      <th>sQTL<\/th>\n      <th>apaQTL<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-4a6133f685aa29daac40" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4a6133f685aa29daac40">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by tissue<\/caption>","data":[["45","192","1","182","18","141","8","23","78","82","85","90","93","191","206","213","11","183","52","26","113","131","81","167","189","215","211","177","69","133","4","79","83","106","35","120","29","126","147","96","168","114","63","87","174","188","49","130","205","57","38","88","138","186","193","39","94","129","102","214","144","72","122","61","162","21","41"],["DNAJC13","TRIM39","ABCA1","THOP1","C10orf88","PSRC1","APOB","CCNJ","HLA-DRB5","HPR","INSIG2","KMT2E","LDLR","TRAK2","VAMP5","ZGPAT","ASGR1","TIMD4","FADS1","CETP","NPC1L1","PKN3","HP","SMARCA4","TNKS","ZNF227","ZDHHC18","SUGP1","GSK3B","PLTP","ADH1B","HMGCR","INHBB","MYPOP","CSNK1G3","PARP9","CNIH4","PELO","RAC1","LRCH4","SP4","NPEPPS","GAS6","KDSR","STAT5B","TMEM57","ERGIC3","PHC1","USP53","FCGRT","CYP2A6","KIF13B","PRKD2","TMEM150A","TRIM5","CYP4F12","LIPC","PGS1","MITF","ZHX3","R3HDM2","HFE","PCSK9","FUT2","SIPA1","CCDC57","DDX56"],[1.16,1.112,1.103,1.01,1.001,1.001,1,1,1,1,1,1,1,1,1,1,0.999,0.999,0.997,0.995,0.995,0.991,0.989,0.988,0.985,0.983,0.981,0.98,0.978,0.976,0.975,0.973,0.968,0.968,0.966,0.965,0.964,0.962,0.961,0.958,0.958,0.952,0.95,0.95,0.949,0.949,0.945,0.942,0.939,0.93,0.913,0.909,0.907,0.897,0.896,0.894,0.891,0.89,0.875,0.875,0.872,0.855,0.84,0.833,0.816,0.813,0.804],[0.12,0.029,0,0.012,0,0.548,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0.988,0,0,0,0,0.129,0,0,0.794,0,0.257,0,0.094,0.802,0.085,0,0.026,0,0.345,0,0.086,0.052,0.09,0,0.162,0.114,0.804,0,0.082,0,0,0.204,0.456,0,0.027,0,0.157,0.153,0,0,0,0.028,0.05,0],[0.039,0.999,0,0.307,0,0.453,0,0,1,1,1,0,1,0,0,1,0.999,0,0.997,0,0.756,0,0.989,0,0.985,0,0,0,0.719,0.931,0,0,0.968,0,0.818,0.531,0.074,0,0.961,0.779,0.958,0,0.95,0.424,0.441,0.425,0.424,0.051,0.402,0,0.913,0.356,0.735,0.358,0.422,0.426,0.891,0.246,0,0.678,0.238,0,0,0.466,0,0.28,0.5],[0.977,0.079,0.733,0.551,0.388,0,0,0,0,0,0,1,0,0,0,0,0,0.999,0,0.995,0,0.991,0,0,0,0.983,0.981,0,0,0.045,0.975,0,0,0.711,0,0.315,0.049,0.876,0,0.153,0,0.568,0,0.382,0.374,0.403,0.445,0.621,0.402,0,0,0.455,0.171,0.505,0.27,0.012,0,0.561,0.875,0,0.452,0,0.84,0.367,0.788,0.455,0],[0.023,0.005,0.264,0.141,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.98,0,0,0,0.151,0,0,0,0,0,0,0,0,0,0.038,0,0,0,0.027,0.063,0.094,0.022,0.126,0,0.016,0,0.035,0,0,0,0.056,0,0.03,0.021,0,0,0,0,0.025,0],[0,0,0.106,0,0.613,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0.239,0,0,0,0,0,0,0,0.13,0,0,0.028,0,0,0.148,0.025,0.04,0,0,0,0,0,0,0.057,0.08,0.003,0.013,0.013,0,0,0,0,0,0,0,0,0,0,0,0.011,0.009,0.855,0,0,0,0.002,0.304]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Esophagus_Mucosa<\/th>\n      <th>Liver<\/th>\n      <th>Spleen<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Adrenal_Gland<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/multi_group_6traits_15weights_ukbb.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-with-single-group-eqtl-results-1"
class="section level3">
<h3>Comparing with single group eQTL results</h3>
<pre><code>[1] &quot;the top tissues from single group analyses are Liver,Spleen,Adipose_Subcutaneous,Adrenal_Gland,Esophagus_Mucosa&quot;</code></pre>
<div id="htmlwidget-d72b6eff4afa454a8832" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d72b6eff4afa454a8832">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">#of overlap for single eqtl analysis and multi group analysis<\/caption>","data":[["Liver","Spleen","Adipose_Subcutaneous","Adrenal_Gland","Esophagus_Mucosa"],[40,24,21,19,19],[26,15,6,13,8],[67,67,67,67,67]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#of.sig.gene_single_group_cs_filtered<\/th>\n      <th>#of.overlap<\/th>\n      <th>#of.sig.gene_multi_group_cs_filtered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-73d12cd98287c4e0164c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-73d12cd98287c4e0164c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">pve changes<\/caption>","data":[["Liver","Spleen","Adipose_Subcutaneous","Adrenal_Gland","Esophagus_Mucosa"],[0.0165,0.0199,0.0212,0.0103,0.0308],[0.0072,0.0063,0,0.0014,0.0016]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pve_single_group<\/th>\n      <th>pve_eqtl_multi_group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="sbp" class="section level2">
<h2>SBP</h2>
<div id="results-from-multi-group-analysis-2" class="section level3">
<h3>Results from multi-group analysis</h3>
<div id="htmlwidget-0796cbf5802287e6981e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0796cbf5802287e6981e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5"],["apaQTL","eQTL","sQTL","SNP","total_pve"],["0.0027","0.0139","0.005","0.0419","0.0635"],["7e-04","0.0014","0","0","0.0021"],["0.0011","0.0048","0.0022","0","0.0081"],["5e-04","0.0052","1e-04","0","0.0058"],["4e-04","1e-04","0.0026","0","0.0031"],["0","0.0024","1e-04","0","0.0025"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Artery_Tibial<\/th>\n      <th>Brain_Cortex<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Spleen<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-35c46d2ddbc1fab8b4dc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-35c46d2ddbc1fab8b4dc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by omics<\/caption>","data":[["94","111","154","30","38","220","150","223","17","213","125","12","15","20","23","32","46","47","84","42","186","190","219","128","1","205","82","178","151","40","129","75","140","194","197","139","222","142","115","173","155","62","132","185","80","171","13","168","5","65","179","36","202","174","209","224","73","218","14","141","134","199","191","100","79","4","105","91","2","165","98","164","50","188","7","119","39","131","187","221","57","127","120","121"],["IRAK1BP1","MCPH1","PTPN13","CCDC163","CFDP1","YWHAE","PRKAB1","ZNF76","ATG7","ULK4","NAA60","ARHGAP42","ASCC2","BAG6","C1orf21","CCNT2","CRLF3","CSE1L","HAUS8","CNNM2","STYX","TBRG4","WNK1","NDUFAF8","AATK-AS1","TRPM7","GSPT1","SLC9A3R2","PRKD3","CLCN6","NECAB1","GDF7","PAWR","THBS2","TMBIM1","PAQR5","ZNF692","PHACTR1","MORC3","SLC16A7","RAB34","EML1","NOTCH3","STAG1","GOLGB1","SGSM3","ARHGEF12","SEC61A2","ADHFE1","FAM214A","SMAD5","CEP68","TNNT3","SLC20A2","UBE2L3","ZNF827","FURIN","WLS","ARID1A","PGPEP1","NPR1","TMEM175","TBX2","KLHDC8B","GNPDA1","ADH1B","LRP4","HSCB","ACADVL","RTF1","KCNK3","RSRC1","CYBRD1","SYNPO2L","AGT","MRC2","CKB","NGF","SUGCT","ZNF467","DYNC1I2","NDUFAF1","MRPL21","MRVI1"],[1.865,1.668,1.606,1.513,1.488,1.375,1.165,1.161,1.112,1.003,1.001,1,1,1,1,1,1,1,1,0.999,0.999,0.999,0.999,0.997,0.996,0.996,0.995,0.994,0.987,0.986,0.985,0.983,0.983,0.983,0.983,0.982,0.982,0.981,0.979,0.979,0.978,0.977,0.976,0.973,0.97,0.97,0.968,0.968,0.966,0.964,0.963,0.962,0.962,0.96,0.96,0.956,0.949,0.939,0.938,0.936,0.934,0.932,0.925,0.923,0.919,0.913,0.91,0.904,0.903,0.902,0.898,0.89,0.883,0.862,0.861,0.858,0.853,0.848,0.848,0.839,0.829,0.827,0.82,0.818],[0.791,1.668,1.606,0.211,1.488,1.375,1.165,0.357,1.112,0,0.967,0,0,1,1,1,1,1,1,0.999,0.999,0.999,0.999,0,0,0.996,0,0,0.987,0.56,0.985,0,0,0,0.983,0,0,0,0,0.979,0.949,0,0.976,0,0.97,0.97,0.968,0.968,0.966,0.964,0.963,0.426,0.962,0,0.96,0,0.949,0.005,0,0,0.934,0.2,0.06,0,0,0.913,0,0.11,0,0,0,0,0.354,0.539,0,0.858,0,0,0,0,0.038,0.827,0.625,0.818],[1.074,0,0,1.302,0,0,0,0.804,0,1.003,0,1,1,0,0,0,0,0,0,0,0,0,0,0.997,0.996,0,0.995,0,0,0.426,0,0.983,0.006,0.983,0,0.982,0.982,0.981,0.979,0,0.03,0.977,0,0.973,0,0,0,0,0,0,0,0.536,0,0.96,0,0.956,0,0.933,0.938,0.936,0,0.733,0.865,0.923,0.919,0,0.91,0.795,0.903,0.902,0.898,0.89,0.529,0.322,0.861,0,0.853,0.848,0,0.839,0.249,0,0.195,0],[0,0,0,0,0,0,0,0,0,0,0.035,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.994,0,0,0,0,0.977,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.848,0,0.542,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>sQTL<\/th>\n      <th>eQTL<\/th>\n      <th>apaQTL<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-2a1b6e0e6b8a75d77da4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a1b6e0e6b8a75d77da4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by tissue<\/caption>","data":[["94","111","154","30","38","220","150","223","17","213","125","12","15","20","23","32","46","47","84","42","186","190","219","128","1","205","82","178","151","40","129","75","140","194","197","139","222","142","115","173","155","62","132","185","80","171","13","168","5","65","179","36","202","174","209","224","73","218","14","141","134","199","191","100","79","4","105","91","2","165","98","164","50","188","7","119","39","131","187","221","57","127","120","121"],["IRAK1BP1","MCPH1","PTPN13","CCDC163","CFDP1","YWHAE","PRKAB1","ZNF76","ATG7","ULK4","NAA60","ARHGAP42","ASCC2","BAG6","C1orf21","CCNT2","CRLF3","CSE1L","HAUS8","CNNM2","STYX","TBRG4","WNK1","NDUFAF8","AATK-AS1","TRPM7","GSPT1","SLC9A3R2","PRKD3","CLCN6","NECAB1","GDF7","PAWR","THBS2","TMBIM1","PAQR5","ZNF692","PHACTR1","MORC3","SLC16A7","RAB34","EML1","NOTCH3","STAG1","GOLGB1","SGSM3","ARHGEF12","SEC61A2","ADHFE1","FAM214A","SMAD5","CEP68","TNNT3","SLC20A2","UBE2L3","ZNF827","FURIN","WLS","ARID1A","PGPEP1","NPR1","TMEM175","TBX2","KLHDC8B","GNPDA1","ADH1B","LRP4","HSCB","ACADVL","RTF1","KCNK3","RSRC1","CYBRD1","SYNPO2L","AGT","MRC2","CKB","NGF","SUGCT","ZNF467","DYNC1I2","NDUFAF1","MRPL21","MRVI1"],[1.865,1.668,1.606,1.513,1.488,1.375,1.165,1.161,1.112,1.003,1.001,1,1,1,1,1,1,1,1,0.999,0.999,0.999,0.999,0.997,0.996,0.996,0.995,0.994,0.987,0.986,0.985,0.983,0.983,0.983,0.983,0.982,0.982,0.981,0.979,0.979,0.978,0.977,0.976,0.973,0.97,0.97,0.968,0.968,0.966,0.964,0.963,0.962,0.962,0.96,0.96,0.956,0.949,0.939,0.938,0.936,0.934,0.932,0.925,0.923,0.919,0.913,0.91,0.904,0.903,0.902,0.898,0.89,0.883,0.862,0.861,0.858,0.853,0.848,0.848,0.839,0.829,0.827,0.82,0.818],[0.678,0.706,1.554,0.465,0.268,0,0,0.422,0.995,0.329,0,0.098,0,1,0,0.381,0,0,0,0.999,0.999,0.999,0,0,0,0.245,0,0.378,0.376,0.153,0,0.983,0.977,0.983,0.09,0.633,0.78,0.981,0.101,0,0.172,0.382,0,0,0.534,0.911,0,0,0.368,0.368,0.266,0.426,0.962,0.96,0.611,0.956,0.949,0.005,0.938,0,0.934,0.622,0.347,0.056,0,0,0,0.11,0,0,0.517,0.224,0.31,0.322,0,0.858,0.388,0.848,0.848,0,0.489,0,0.06,0.556],[0.604,0.961,0,0,1.219,0,0,0.357,0,0,0,0,0,0,1,0.618,0,0,0,0,0,0,0.999,0,0,0.751,0,0.415,0.611,0.335,0,0,0,0,0.893,0,0,0,0,0.979,0.726,0,0.976,0,0.436,0,0,0,0.597,0.596,0.697,0,0,0,0.349,0,0,0,0,0,0,0.181,0.06,0,0.008,0.913,0,0.02,0,0,0,0,0.285,0.539,0,0,0,0,0,0,0,0.827,0.552,0.261],[0,0,0,0.14,0,0,0.771,0.246,0,0.181,0,0,0,0,0,0,0,0.338,0,0,0,0,0,0.997,0,0,0,0,0,0.03,0,0,0,0,0,0.349,0,0,0.277,0,0,0,0,0,0,0,0.968,0.968,0,0,0,0.536,0,0,0,0,0,0,0,0.294,0,0,0,0,0.809,0,0,0,0,0.57,0,0,0.074,0,0.861,0,0,0,0,0.256,0.093,0,0.195,0],[0.582,0,0.052,0.833,0,1.375,0.393,0,0.116,0.395,1.001,0,0,0,0,0,0,0.662,0,0,0,0,0,0,0.996,0,0.995,0,0,0.468,0.985,0,0,0,0,0,0,0,0.601,0,0.08,0,0,0.973,0,0.059,0,0,0,0,0,0,0,0,0,0,0,0.933,0,0.642,0,0,0.416,0.868,0,0,0.91,0.774,0.903,0,0,0.659,0.173,0,0,0,0.465,0,0,0.556,0.145,0,0.013,0],[0,0,0,0.075,0,0,0,0.136,0,0.097,0,0.902,1,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0.201,0,0,0,0,0.006,0,0,0,0.203,0,0,0,0,0.595,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.129,0.102,0,0.102,0,0,0,0,0.331,0.381,0.007,0.042,0,0,0,0,0,0,0.027,0.102,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Artery_Tibial<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Spleen<\/th>\n      <th>Brain_Cortex<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/multi_group_6traits_15weights_ukbb.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-with-single-group-eqtl-results-2"
class="section level3">
<h3>Comparing with single group eQTL results</h3>
<pre><code>[1] &quot;the top tissues from single group analyses are Artery_Tibial,Adipose_Subcutaneous,Brain_Cortex,Heart_Left_Ventricle,Spleen&quot;</code></pre>
<div id="htmlwidget-f41426fe0b90d0c09df4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f41426fe0b90d0c09df4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">#of overlap for single eqtl analysis and multi group analysis<\/caption>","data":[["Artery_Tibial","Adipose_Subcutaneous","Brain_Cortex","Heart_Left_Ventricle","Spleen"],[35,29,29,25,21],[19,10,18,8,11],[84,84,84,84,84]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#of.sig.gene_single_group_cs_filtered<\/th>\n      <th>#of.overlap<\/th>\n      <th>#of.sig.gene_multi_group_cs_filtered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-35e1d0e81f5ada6ecac0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-35e1d0e81f5ada6ecac0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">pve changes<\/caption>","data":[["Artery_Tibial","Adipose_Subcutaneous","Brain_Cortex","Heart_Left_Ventricle","Spleen"],[0.019,0.0142,0.0209,0.0136,0.0195],[0.0048,0.0014,0.0052,0.0001,0.0024]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pve_single_group<\/th>\n      <th>pve_eqtl_multi_group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="scz" class="section level2">
<h2>SCZ</h2>
<div id="results-from-multi-group-analysis-3" class="section level3">
<h3>Results from multi-group analysis</h3>
<div id="htmlwidget-a41f9b9304ecbba20fc2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a41f9b9304ecbba20fc2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5"],["apaQTL","eQTL","sQTL","SNP","total_pve"],["0.0014","0.0288","0.0206","0.1268","0.1776"],["0","0.005","0.0026","0","0.0076"],["0","0.0044","0.0065","0","0.0109"],["0","0.006","0.0082","0","0.0142"],["0.0014","0.0091","0.0033","0","0.0138"],["0","0.0043","0","0","0.0043"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adrenal_Gland<\/th>\n      <th>Artery_Coronary<\/th>\n      <th>Brain_Cerebellum<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Stomach<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-80db18273cfc4a9b8682" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-80db18273cfc4a9b8682">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by omics<\/caption>","data":[["220","128","241","83","2","106","131","199","1","112","213","38","174","81","221","54","143","99","76","193","159","236","214","118","210","150","15","173"],["TRIM38","MICB","ZNF823","FOXN2","ACTR1B","IRF3","MRPS33","SOHLH1","ACE","KLHL20","TMEM222","CHST2","R3HDM2","FEZF1","TRPV4","CUL9","NGEF","HLA-DMA","FAM120A","SF3B1","PFKFB2","ZDHHC17","TMEM56","LY6H","THYN1","NUTM1","ARFGAP2","PUF60"],[1,0.999,0.993,0.984,0.981,0.977,0.973,0.968,0.963,0.948,0.947,0.944,0.942,0.924,0.922,0.906,0.895,0.89,0.886,0.883,0.873,0.868,0.861,0.854,0.851,0.839,0.831,0.815],[0,0,0.993,0.843,0.11,0.669,0,0.968,0.676,0.948,0.947,0,0,0.924,0.922,0.444,0.895,0.872,0.886,0.57,0.822,0,0.861,0.854,0,0.839,0.831,0.815],[0,0.999,0,0.14,0.871,0.307,0.973,0,0.288,0,0,0.944,0.942,0,0,0.462,0,0.018,0,0.313,0.052,0.868,0,0,0,0,0,0],[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.851,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL<\/th>\n      <th>sQTL<\/th>\n      <th>apaQTL<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-6f19a3278480c8d45e45" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6f19a3278480c8d45e45">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by tissue<\/caption>","data":[["220","128","241","83","2","106","131","199","1","112","213","38","174","81","221","54","143","99","76","193","159","236","214","118","210","150","15","173"],["TRIM38","MICB","ZNF823","FOXN2","ACTR1B","IRF3","MRPS33","SOHLH1","ACE","KLHL20","TMEM222","CHST2","R3HDM2","FEZF1","TRPV4","CUL9","NGEF","HLA-DMA","FAM120A","SF3B1","PFKFB2","ZDHHC17","TMEM56","LY6H","THYN1","NUTM1","ARFGAP2","PUF60"],[1,0.999,0.993,0.984,0.981,0.977,0.973,0.968,0.963,0.948,0.947,0.944,0.942,0.924,0.922,0.906,0.895,0.89,0.886,0.883,0.873,0.868,0.861,0.854,0.851,0.839,0.831,0.815],[0,0.999,0.327,0.411,0.175,0.188,0.126,0,0.306,0.377,0.798,0,0.335,0,0.431,0.171,0.233,0.009,0.438,0.338,0.387,0,0,0.548,0.851,0,0.263,0],[1,0,0.177,0.23,0.111,0.104,0.16,0,0.218,0.211,0.015,0,0,0,0,0.057,0,0,0.245,0.022,0.208,0.265,0.861,0.306,0,0,0.147,0.301],[0,0,0.148,0.202,0.237,0.197,0,0,0.106,0.185,0.008,0.418,0,0,0.211,0.291,0.662,0,0,0.098,0.02,0.603,0,0,0,0,0.129,0.265],[0,0,0.153,0,0,0.434,0,0,0,0.175,0.126,0,0,0,0,0.079,0,0,0.203,0.125,0,0,0,0,0,0.839,0.122,0.25],[0,0,0.188,0.14,0.458,0.054,0.687,0.968,0.333,0,0,0.527,0.607,0.924,0.28,0.309,0,0.88,0,0.299,0.259,0,0,0,0,0,0.171,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Heart_Left_Ventricle<\/th>\n      <th>Adrenal_Gland<\/th>\n      <th>Artery_Coronary<\/th>\n      <th>Stomach<\/th>\n      <th>Brain_Cerebellum<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/multi_group_6traits_15weights_ukbb.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-with-single-group-eqtl-results-3"
class="section level3">
<h3>Comparing with single group eQTL results</h3>
<pre><code>[1] &quot;the top tissues from single group analyses are Heart_Left_Ventricle,Adrenal_Gland,Artery_Coronary,Brain_Cerebellum,Stomach&quot;</code></pre>
<div id="htmlwidget-81a148ea6c961cf45e2c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-81a148ea6c961cf45e2c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">#of overlap for single eqtl analysis and multi group analysis<\/caption>","data":[["Heart_Left_Ventricle","Adrenal_Gland","Artery_Coronary","Brain_Cerebellum","Stomach"],[23,22,16,16,15],[12,10,9,7,7],[28,28,28,28,28]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#of.sig.gene_single_group_cs_filtered<\/th>\n      <th>#of.overlap<\/th>\n      <th>#of.sig.gene_multi_group_cs_filtered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-425deb884d24b4ea5f7c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-425deb884d24b4ea5f7c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">pve changes<\/caption>","data":[["Heart_Left_Ventricle","Adrenal_Gland","Artery_Coronary","Brain_Cerebellum","Stomach"],[0.0234,0.0214,0.0213,0.0263,0.0189],[0.0091,0.005,0.0044,0.006,0.0043]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pve_single_group<\/th>\n      <th>pve_eqtl_multi_group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="wbc" class="section level2">
<h2>WBC</h2>
<div id="results-from-multi-group-analysis-4" class="section level3">
<h3>Results from multi-group analysis</h3>
<div id="htmlwidget-5674f84365e74fcb3e3c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5674f84365e74fcb3e3c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Heritability contribution by contexts<\/caption>","data":[["1","2","3","4","5"],["apaQTL","eQTL","sQTL","SNP","total_pve"],["0.0011","0.0345","0.0179","0.0597","0.1132"],["0","0.0059","0.0075","0","0.0134"],["0","0.0067","0.0024","0","0.0091"],["6e-04","0.0045","0.0012","0","0.0063"],["0","0.0059","0.0035","0","0.0094"],["5e-04","0.0115","0.0033","0","0.0153"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type<\/th>\n      <th>total_pve<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Artery_Aorta<\/th>\n      <th>Skin_Sun_Exposed_Lower_leg<\/th>\n      <th>Spleen<\/th>\n      <th>Whole_Blood<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-caaf43859fd0889ca800" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-caaf43859fd0889ca800">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by omics<\/caption>","data":[["33","98","391","352","430","185","322","136","90","261","132","227","344","215","537","22","26","71","97","130","143","209","229","255","286","296","314","325","343","353","475","528","531","564","599","139","319","392","405","417","467","688","259","421","627","105","544","8","32","80","303","99","125","147","464","653","144","387","641","133","142","270","413","484","350","262","56","231","321","473","511","643","102","330","480","624","35","77","368","279","114","675","548","48","84","205","529","647","254","264","67","436","85","101","422","304","522","547","150","471","667","43","439","164","78","188","297","483","349","658","218","310","615","223","411","454","401","487","579","460","145","225","328","148","562","59","663","351","170","576","23","339","100","213","273","315","482","568","370","210","318","240","514","381","92","266","272","415","268","331","423","444","559","694","50","672","309","208","609","79","83","283","553","452","435","63","284","70","448","655","396","76","162","385","111","103","661","141","600","538","468","340","183","488","580","329","560","629","496","669","53","200","265","120","410","68","263","135","617","700","393","587","121","465","634","563","509","113","380","42","27","651","129","604","323","486","299","662","253","34"],["ARHGAP15","CCZ1B","NOSTRIN","MICALL2","PIGL","ELMO1","LYZ","CPEB4","CCDC125","HTT","COQ8A","GCNT2","MED24","FN1","SFI1","ALDH2","ANKRD44","C4B","CCZ1","CNN2","CSF3R","FES","GGT5","HSD17B1","ITSN2","KIAA0391","LPAR1","MAP2K2","MED12L","MICB","RABGEF1","SEC22A","SELL","SLC4A7","TAGAP","CREB5","LSP1","NOXRED1","OAS1","PCM1","PTPRJ","ZNF516","HSF2","PCYT1A","TNKS2","CDC14A","SLC12A7","ACAP1","ARAP2","CANX","LAMP1","CD101","CLEC4M","CXCR2","PTK2B","VLDLR","CSGALNACT1","NLRC5","TSPAN32","COTL1","CSF1","IL1RN","PAQR9","RBPMS","MFSD13A","ID3","BAX","GLG1","LYN","RABEP1","RUNX3","TTC39C","CD33","MAP3K5","RAPGEF3","TNFRSF10C","ARHGAP24","CADM4","MYH10","IRF4","CERKL","ZFPM1","SLC23A3","ATP13A1","CARD6","FCER1G","SEC31A","UBE2F","HPN","IFITM2","C18orf25","PKN3","CARS2","CD300A","PDLIM2","LAPTM5","SCD","SLC22A5","DAP","RAB34","ZBTB22","ASL","PLCG2","DMXL1","CALCOCO2","ERAL1","KIAA1614","RBP4","MFN1","WDFY2","FRAT1","LIPA","TMC4","FZD6","P2RY6","PREX1","NUCB2","RERE","SPECC1L","PSD4","CTSC","GBAS","MAP3K10","CYTH1","SLC45A3","BCL2A1","YAP1","MICAL3","DOCK9","SNX32","AMPD3","MBNL1","CD226","FIP1L1","INPP5D","LRBA","RBM38","SMG1","MYO1G","FGD3","LSM4","GPR157","S100A8","NEK6","CCDC82","IFT52","ING5","PARP12","IL1B","MAP3K7CL","PECAM1","PM20D1","SLC39A11","ZNF713","AXL","ZC3HC1","LIMS2","FEM1C","TIE1","CALCRL","CAPZB","ITGAX","SLC2A4","PPP5C","PKD1","BTBD3","ITGB2","C3orf18","POLR3A","VSIR","NPM1","CACNA1H","DLG5","NKX2-3","CEP83","CD36","WWOX","CRLS1","TBC1D10C","SFXN2","PXYLP1","MCCC1","EIF3B","REST","SPHK2","MAP3K4","SLC39A9","TNS3","RNF139","ZC2HC1A","BACE2","FAM45A","IFNAR1","CITED4","P2RY13","C18orf8","IFIH1","COX7A2L","TMED8","ZRANB2","NPAS2","STARD3NL","CKMT1A","PTPN1","TRIM33","SLC45A4","RRP12","CEPT1","NECTIN2","ARVCF","AP3D1","UPP1","CNIH4","TEAD1","MANBA","REEP3","KLF5","XPO6","HOXC6","ARHGAP23"],[1.957,1.927,1.866,1.75,1.671,1.592,1.576,1.558,1.53,1.436,1.356,1.353,1.06,1.051,1.031,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.999,0.999,0.999,0.999,0.999,0.999,0.999,0.998,0.998,0.998,0.997,0.997,0.996,0.996,0.996,0.995,0.994,0.994,0.994,0.994,0.994,0.991,0.991,0.99,0.989,0.989,0.989,0.989,0.989,0.987,0.986,0.985,0.985,0.985,0.985,0.985,0.985,0.981,0.981,0.981,0.981,0.979,0.979,0.979,0.978,0.977,0.977,0.975,0.973,0.972,0.971,0.97,0.97,0.969,0.969,0.966,0.966,0.965,0.965,0.965,0.964,0.964,0.964,0.963,0.963,0.963,0.962,0.962,0.96,0.959,0.959,0.959,0.959,0.958,0.957,0.956,0.956,0.956,0.954,0.954,0.953,0.95,0.95,0.95,0.948,0.946,0.946,0.946,0.945,0.945,0.944,0.944,0.941,0.94,0.939,0.938,0.938,0.936,0.936,0.936,0.936,0.936,0.936,0.933,0.932,0.932,0.931,0.931,0.93,0.929,0.928,0.927,0.927,0.925,0.925,0.925,0.925,0.925,0.925,0.924,0.924,0.922,0.92,0.92,0.917,0.913,0.913,0.909,0.907,0.906,0.905,0.905,0.903,0.901,0.898,0.897,0.895,0.891,0.887,0.882,0.881,0.881,0.88,0.879,0.878,0.876,0.874,0.868,0.867,0.866,0.864,0.863,0.861,0.86,0.859,0.858,0.858,0.858,0.856,0.855,0.854,0.846,0.844,0.843,0.842,0.84,0.839,0.836,0.835,0.835,0.832,0.831,0.828,0.827,0.824,0.823,0.822,0.82,0.819,0.818,0.818,0.813,0.81,0.809,0.808],[1,0,0.895,0.98,0,0.597,0.588,1.558,0,0.959,1.356,1.353,0,0,0.978,1,1,1,1,1,0,1,1,0,0,1,1,1,1,0,1,1,1,1,1,0.999,0.999,0.999,0,0,0.999,0.999,0.998,0,0,0.997,0.997,0.996,0.996,0,0.995,0.316,0.994,0.994,0.994,0.994,0.852,0.991,0,0.989,0.989,0.989,0.989,0.989,0.331,0.986,0,0.985,0.985,0,0.985,0.985,0,0.981,0.981,0.981,0.979,0.979,0.979,0.978,0.977,0.977,0.975,0.973,0.972,0.971,0.022,0.97,0.969,0.969,0.311,0.966,0,0.059,0.693,0.964,0.964,0,0.963,0,0.963,0,0.962,0,0.2,0.959,0.296,0.959,0.958,0.09,0.956,0.94,0.633,0.954,0.954,0.953,0.593,0.879,0.95,0,0,0,0.696,0.945,0,0.8,0,0,0.94,0.939,0.494,0.938,0.936,0,0,0,0,0.895,0,0.932,0.932,0,0.931,0.93,0.112,0.889,0,0,0.925,0.307,0,0.901,0.925,0.925,0.924,0,0,0.88,0.92,0.88,0.035,0,0.909,0.795,0.213,0.905,0.687,0.903,0,0,0.825,0.519,0.193,0.887,0.377,0.079,0.596,0.177,0,0.878,0.876,0.011,0.786,0.867,0.486,0,0.863,0,0.86,0.217,0.838,0.218,0.858,0.856,0.855,0.78,0,0.689,0.843,0.372,0,0.561,0.836,0,0.835,0.832,0.424,0,0.227,0.009,0.823,0.822,0.813,0,0.798,0.818,0.813,0.164,0.809,0.808],[0.957,1,0.971,0.77,0.671,0.995,0.989,0,1.53,0.477,0,0,0.861,1.051,0.053,0,0,0,0,0,1,0,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.998,0,0,0,0,0.996,0,0,0,0,0,0,0.14,0,0.99,0,0,0,0,0,0.656,0,0.606,0,0,0,0,0,0.981,0,0,0,0,0,0,0,0,0,0,0,0,0,0.948,0,0,0,0.655,0,0.965,0.906,0.272,0,0,0.964,0,0.963,0,0.962,0,0.96,0.759,0,0.664,0,0,0.866,0,0.017,0,0,0,0,0.358,0.071,0,0.948,0.946,0.946,0.25,0,0.945,0.145,0.944,0.941,0,0,0.444,0,0,0.936,0.936,0.936,0.936,0.041,0.933,0,0,0.931,0,0,0.818,0.039,0.927,0.927,0,0.619,0,0.024,0,0,0,0.924,0.922,0,0,0.037,0.014,0.913,0,0.111,0.692,0,0.218,0,0.901,0.898,0.072,0.376,0.698,0,0.504,0.802,0.285,0.703,0.879,0,0,0.863,0.082,0,0.38,0.864,0,0.861,0,0.642,0.02,0.64,0,0,0,0.074,0.846,0.091,0,0.47,0.84,0.278,0,0.835,0,0,0.407,0.828,0.6,0.815,0,0,0.007,0.819,0.02,0,0,0.646,0,0],[0,0.927,0,0,1,0,0,0,0,0,0,0,0.199,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.999,0.999,0,0,0,0.998,0,0,0,0,0,0,0,0.679,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.379,0,0,0.985,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.323,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.925,0,0,0,0,0,0,0.039,0,0,0.864,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.064,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>eQTL<\/th>\n      <th>sQTL<\/th>\n      <th>apaQTL<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-28fc46a89962ff28ef30" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-28fc46a89962ff28ef30">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">Combined PIP by tissue<\/caption>","data":[["33","98","391","352","430","185","322","136","90","261","132","227","344","215","537","22","26","71","97","130","143","209","229","255","286","296","314","325","343","353","475","528","531","564","599","139","319","392","405","417","467","688","259","421","627","105","544","8","32","80","303","99","125","147","464","653","144","387","641","133","142","270","413","484","350","262","56","231","321","473","511","643","102","330","480","624","35","77","368","279","114","675","548","48","84","205","529","647","254","264","67","436","85","101","422","304","522","547","150","471","667","43","439","164","78","188","297","483","349","658","218","310","615","223","411","454","401","487","579","460","145","225","328","148","562","59","663","351","170","576","23","339","100","213","273","315","482","568","370","210","318","240","514","381","92","266","272","415","268","331","423","444","559","694","50","672","309","208","609","79","83","283","553","452","435","63","284","70","448","655","396","76","162","385","111","103","661","141","600","538","468","340","183","488","580","329","560","629","496","669","53","200","265","120","410","68","263","135","617","700","393","587","121","465","634","563","509","113","380","42","27","651","129","604","323","486","299","662","253","34"],["ARHGAP15","CCZ1B","NOSTRIN","MICALL2","PIGL","ELMO1","LYZ","CPEB4","CCDC125","HTT","COQ8A","GCNT2","MED24","FN1","SFI1","ALDH2","ANKRD44","C4B","CCZ1","CNN2","CSF3R","FES","GGT5","HSD17B1","ITSN2","KIAA0391","LPAR1","MAP2K2","MED12L","MICB","RABGEF1","SEC22A","SELL","SLC4A7","TAGAP","CREB5","LSP1","NOXRED1","OAS1","PCM1","PTPRJ","ZNF516","HSF2","PCYT1A","TNKS2","CDC14A","SLC12A7","ACAP1","ARAP2","CANX","LAMP1","CD101","CLEC4M","CXCR2","PTK2B","VLDLR","CSGALNACT1","NLRC5","TSPAN32","COTL1","CSF1","IL1RN","PAQR9","RBPMS","MFSD13A","ID3","BAX","GLG1","LYN","RABEP1","RUNX3","TTC39C","CD33","MAP3K5","RAPGEF3","TNFRSF10C","ARHGAP24","CADM4","MYH10","IRF4","CERKL","ZFPM1","SLC23A3","ATP13A1","CARD6","FCER1G","SEC31A","UBE2F","HPN","IFITM2","C18orf25","PKN3","CARS2","CD300A","PDLIM2","LAPTM5","SCD","SLC22A5","DAP","RAB34","ZBTB22","ASL","PLCG2","DMXL1","CALCOCO2","ERAL1","KIAA1614","RBP4","MFN1","WDFY2","FRAT1","LIPA","TMC4","FZD6","P2RY6","PREX1","NUCB2","RERE","SPECC1L","PSD4","CTSC","GBAS","MAP3K10","CYTH1","SLC45A3","BCL2A1","YAP1","MICAL3","DOCK9","SNX32","AMPD3","MBNL1","CD226","FIP1L1","INPP5D","LRBA","RBM38","SMG1","MYO1G","FGD3","LSM4","GPR157","S100A8","NEK6","CCDC82","IFT52","ING5","PARP12","IL1B","MAP3K7CL","PECAM1","PM20D1","SLC39A11","ZNF713","AXL","ZC3HC1","LIMS2","FEM1C","TIE1","CALCRL","CAPZB","ITGAX","SLC2A4","PPP5C","PKD1","BTBD3","ITGB2","C3orf18","POLR3A","VSIR","NPM1","CACNA1H","DLG5","NKX2-3","CEP83","CD36","WWOX","CRLS1","TBC1D10C","SFXN2","PXYLP1","MCCC1","EIF3B","REST","SPHK2","MAP3K4","SLC39A9","TNS3","RNF139","ZC2HC1A","BACE2","FAM45A","IFNAR1","CITED4","P2RY13","C18orf8","IFIH1","COX7A2L","TMED8","ZRANB2","NPAS2","STARD3NL","CKMT1A","PTPN1","TRIM33","SLC45A4","RRP12","CEPT1","NECTIN2","ARVCF","AP3D1","UPP1","CNIH4","TEAD1","MANBA","REEP3","KLF5","XPO6","HOXC6","ARHGAP23"],[1.957,1.927,1.866,1.75,1.671,1.592,1.576,1.558,1.53,1.436,1.356,1.353,1.06,1.051,1.031,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0.999,0.999,0.999,0.999,0.999,0.999,0.999,0.998,0.998,0.998,0.997,0.997,0.996,0.996,0.996,0.995,0.994,0.994,0.994,0.994,0.994,0.991,0.991,0.99,0.989,0.989,0.989,0.989,0.989,0.987,0.986,0.985,0.985,0.985,0.985,0.985,0.985,0.981,0.981,0.981,0.981,0.979,0.979,0.979,0.978,0.977,0.977,0.975,0.973,0.972,0.971,0.97,0.97,0.969,0.969,0.966,0.966,0.965,0.965,0.965,0.964,0.964,0.964,0.963,0.963,0.963,0.962,0.962,0.96,0.959,0.959,0.959,0.959,0.958,0.957,0.956,0.956,0.956,0.954,0.954,0.953,0.95,0.95,0.95,0.948,0.946,0.946,0.946,0.945,0.945,0.944,0.944,0.941,0.94,0.939,0.938,0.938,0.936,0.936,0.936,0.936,0.936,0.936,0.933,0.932,0.932,0.931,0.931,0.93,0.929,0.928,0.927,0.927,0.925,0.925,0.925,0.925,0.925,0.925,0.924,0.924,0.922,0.92,0.92,0.917,0.913,0.913,0.909,0.907,0.906,0.905,0.905,0.903,0.901,0.898,0.897,0.895,0.891,0.887,0.882,0.881,0.881,0.88,0.879,0.878,0.876,0.874,0.868,0.867,0.866,0.864,0.863,0.861,0.86,0.859,0.858,0.858,0.858,0.856,0.855,0.854,0.846,0.844,0.843,0.842,0.84,0.839,0.836,0.835,0.835,0.832,0.831,0.828,0.827,0.824,0.823,0.822,0.82,0.819,0.818,0.818,0.813,0.81,0.809,0.808],[1.575,0.927,0,0.182,0.449,0.646,1.04,0.998,0.04,0.959,0.965,0.825,0,0.292,0.013,1,0,0,0,0,0,1,0,0,0.72,1,1,0,1,0,1,0,1,0,0.074,0.999,0.999,0,0.999,0,0.999,0.999,0,0,0,0.074,0.522,0.807,0.304,0.996,0.493,0.825,0,0.994,0.994,0.994,0.14,0.991,0.99,0,0,0,0.989,0,0.248,0.531,0.69,0.985,0.985,0,0.431,0.985,0.382,0,0,0,0,0.673,0,0,0,0.977,0,0.201,0.65,0,0.022,0.949,0.969,0,0.262,0.78,0,0.083,0.076,0,0,0,0.714,0.949,0,0.26,0,0,0.234,0.959,0,0.424,0,0.419,0,0.94,0.238,0,0,0,0.715,0.906,0.494,0.312,0.412,0.217,0.319,0,0,0.843,0,0,0.617,0,0,0,0.936,0.111,0.562,0.57,0.563,0,0.562,0.579,0,0,0.931,0.93,0.238,0,0,0.414,0.925,0.307,0.763,0.185,0.925,0.149,0.279,0,0,0.92,0.92,0.058,0.754,0,0,0.575,0.075,0.384,0.369,0,0,0,0,0,0.238,0,0,0.308,0,0.144,0.432,0,0.876,0,0,0.65,0,0,0.863,0.418,0.425,0.075,0.838,0.133,0.858,0.856,0.066,0.018,0,0,0.471,0.372,0,0,0,0.636,0,0.832,0.695,0,0,0,0.823,0.224,0.006,0,0,0,0.813,0.144,0,0],[0,0,1.7,0.159,0,0.947,0.287,0.56,0.499,0.411,0,0,0.418,0.362,0.012,0,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0.478,0,0,0,0.999,0,0,0,0,0.4,0,0.483,0.395,0.32,0,0.691,0,0,0,0,0,0,0,0,0,0,0,0.569,0.989,0,0,0.351,0,0,0,0,0,0.195,0,0.323,0,0,0,0,0.305,0.979,0,0.977,0,0.179,0,0,0.971,0.194,0,0,0,0.524,0,0.636,0.711,0,0,0.964,0.731,0,0,0,0.099,0,0.611,0.269,0,0.633,0.012,0.424,0.177,0.457,0.017,0.395,0.367,0,0,0.155,0.035,0,0.221,0.193,0.297,0.086,0.945,0.422,0.054,0.245,0.451,0,0,0.876,0,0,0.557,0,0,0,0.033,0,0.022,0.932,0,0,0,0.3,0.012,0.927,0.28,0,0.331,0,0.396,0,0,0,0,0.493,0,0,0.406,0.007,0.45,0.909,0.269,0.284,0,0.155,0.903,0,0,0.333,0,0.244,0,0.335,0.228,0.596,0.286,0.447,0,0,0.541,0,0,0.58,0.696,0,0.319,0,0.298,0.016,0.242,0,0,0,0,0,0.016,0.214,0.231,0.475,0.505,0,0,0,0,0,0.491,0,0.066,0,0.074,0.244,0.819,0,0,0,0.223,0,0],[0,0,0.094,0.051,0.197,0,0.099,0,0.24,0,0.391,0.528,0.145,0.134,0.01,0,0,0,0,1,0,0,0,0,0.279,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0.507,0,0.053,0.455,0.054,0,0,0,0.258,0,0,0,0,0,0.852,0,0,0,0,0,0,0.382,0.103,0.277,0.134,0,0,0,0,0,0.055,0,0,0,0.979,0,0,0.978,0,0,0.709,0,0,0,0,0.021,0,0,0.086,0,0.11,0,0.484,0,0,0.233,0,0,0,0.321,0,0.07,0.158,0,0.236,0.523,0.533,0.122,0,0,0,0.016,0,0.953,0,0,0.076,0,0.181,0.095,0.226,0,0.146,0.047,0.438,0.18,0.323,0,0,0,0,0.205,0,0.25,0,0,0,0.131,0,0.461,0,0,0.137,0.888,0,0,0,0.114,0,0.344,0,0,0,0,0.17,0,0,0.44,0.028,0.155,0,0,0.386,0.201,0.187,0,0.358,0.898,0.03,0,0.176,0.455,0.09,0.089,0,0.303,0,0.878,0,0.12,0.858,0,0.151,0,0,0.125,0.223,0.13,0.004,0.31,0,0,0.79,0.762,0.846,0.028,0,0.08,0.164,0,0,0.2,0.835,0,0.016,0,0,0.287,0,0,0.439,0,0.011,0,0,0.148,0,0],[0,1,0.072,0.026,1.02,0,0,0,0.725,0.066,0,0,0.266,0.064,0.978,0,0,0,0,0,0,0,1,1,0,0,0,1,0,0,0,0,0,0,0.926,0,0,0,0,0.999,0,0,0,0.998,0,0,0,0.188,0,0,0,0,0,0,0,0,0,0,0,0,0.42,0,0,0.245,0.059,0.178,0,0,0,0.985,0.144,0,0.052,0,0,0.981,0,0,0,0,0,0,0.087,0,0,0,0,0,0,0.969,0.027,0,0.051,0,0.042,0.964,0,0,0.249,0,0,0,0.962,0.066,0.111,0,0.09,0,0,0.205,0,0,0.323,0.537,0,0,0,0,0.136,0.05,0.059,0.048,0.121,0,0.068,0,0.207,0.073,0,0,0.062,0.938,0,0.015,0,0.116,0,0.862,0,0,0,0.121,0,0,0.069,0,0,0,0,0,0.162,0,0,0.595,0,0,0,0,0,0,0.123,0.072,0,0,0.061,0.129,0,0,0,0,0.216,0.895,0.091,0,0.177,0.052,0.067,0.046,0,0,0,0.023,0.01,0.218,0,0.168,0,0,0,0.2,0,0.034,0,0,0,0.017,0,0.075,0.158,0.037,0.077,0.239,0.836,0,0,0,0.03,0.079,0.141,0,0,0.524,0.13,0,0.807,0,0,0.078,0,0.808],[0.382,0,0,1.333,0.006,0,0.15,0,0.027,0,0,0,0.231,0.2,0.018,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0.522,0,0,0,0,0,0,0,0,0.091,0,0.462,0.072,0.101,0,0,0,0.244,0.169,0.994,0,0,0,0,0,0,0.989,0,0,0,0.362,0.225,0,0.16,0,0,0,0.214,0,0.169,0.981,0.981,0,0,0,0,0,0,0,0,0.771,0.323,0,0.753,0,0,0,0.068,0.186,0.167,0.171,0.364,0,0,0,0,0.014,0.963,0.282,0,0.213,0.187,0,0,0,0,0.033,0.499,0,0,0.034,0.954,0,0.081,0.009,0.245,0.365,0.102,0.289,0.194,0,0.309,0,0.054,0.236,0,0.939,0,0,0,0.049,0.373,0,0.374,0.041,0.371,0.2,0,0.35,0,0,0.185,0.028,0,0.233,0,0.174,0,0,0,0.181,0.645,0.924,0.259,0,0,0.013,0,0.236,0,0.062,0.099,0.191,0.194,0,0.543,0,0.318,0,0.142,0.432,0.28,0.203,0.218,0.101,0,0,0,0.19,0,0,0.135,0,0,0,0.212,0.156,0,0.14,0,0,0,0.057,0,0.726,0,0.121,0.124,0.096,0,0,0,0,0.091,0.258,0.686,0.472,0,0,0,0,0,0.818,0,0.217,0.809,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genename<\/th>\n      <th>combined_pip<\/th>\n      <th>Whole_Blood<\/th>\n      <th>Adipose_Subcutaneous<\/th>\n      <th>Artery_Aorta<\/th>\n      <th>Skin_Sun_Exposed_Lower_leg<\/th>\n      <th>Spleen<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/multi_group_6traits_15weights_ukbb.Rmd/unnamed-chunk-10-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparing-with-single-group-eqtl-results-4"
class="section level3">
<h3>Comparing with single group eQTL results</h3>
<pre><code>[1] &quot;the top tissues from single group analyses are Whole_Blood,Adipose_Subcutaneous,Artery_Aorta,Skin_Sun_Exposed_Lower_leg,Spleen&quot;</code></pre>
<div id="htmlwidget-018aa8438ee3bad3ceee" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-018aa8438ee3bad3ceee">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">#of overlap for single eqtl analysis and multi group analysis<\/caption>","data":[["Whole_Blood","Adipose_Subcutaneous","Artery_Aorta","Skin_Sun_Exposed_Lower_leg","Spleen"],[109,68,56,54,52],[73,40,38,31,34],[220,220,220,220,220]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>#of.sig.gene_single_group_cs_filtered<\/th>\n      <th>#of.overlap<\/th>\n      <th>#of.sig.gene_multi_group_cs_filtered<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-de3a36f80a2aa43a84ec" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de3a36f80a2aa43a84ec">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;  font-size:150% ;\">pve changes<\/caption>","data":[["Whole_Blood","Adipose_Subcutaneous","Artery_Aorta","Skin_Sun_Exposed_Lower_leg","Spleen"],[0.0279,0.019,0.0201,0.0169,0.018],[0.0115,0.0059,0.0067,0.0045,0.0059]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pve_single_group<\/th>\n      <th>pve_eqtl_multi_group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.5.1      dplyr_1.1.4       
 [5] purrr_1.0.2        readr_2.1.2        tidyr_1.3.0        tibble_3.2.1      
 [9] ggplot2_3.5.1      tidyverse_1.3.1    data.table_1.14.2  ctwas_0.2.1.9000  

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] GenomicRanges_1.48.0        fs_1.5.2                   
  [9] rstudioapi_0.13             farver_2.1.0               
 [11] DT_0.22                     ggrepel_0.9.1              
 [13] bit64_4.0.5                 lubridate_1.8.0            
 [15] AnnotationDbi_1.58.0        fansi_1.0.3                
 [17] xml2_1.3.3                  codetools_0.2-18           
 [19] logging_0.10-108            cachem_1.0.6               
 [21] knitr_1.39                  jsonlite_1.8.0             
 [23] workflowr_1.7.0             Rsamtools_2.12.0           
 [25] broom_0.8.0                 dbplyr_2.1.1               
 [27] png_0.1-7                   compiler_4.2.0             
 [29] httr_1.4.3                  backports_1.4.1            
 [31] assertthat_0.2.1            Matrix_1.5-3               
 [33] fastmap_1.1.0               lazyeval_0.2.2             
 [35] cli_3.6.1                   later_1.3.0                
 [37] htmltools_0.5.2             prettyunits_1.1.1          
 [39] tools_4.2.0                 gtable_0.3.0               
 [41] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [43] rappdirs_0.3.3              Rcpp_1.0.8.3               
 [45] Biobase_2.56.0              cellranger_1.1.0           
 [47] jquerylib_0.1.4             vctrs_0.6.5                
 [49] Biostrings_2.64.0           rtracklayer_1.56.0         
 [51] crosstalk_1.2.0             xfun_0.41                  
 [53] rvest_1.0.2                 lifecycle_1.0.4            
 [55] irlba_2.3.5                 restfulr_0.0.14            
 [57] ensembldb_2.20.2            XML_3.99-0.14              
 [59] zlibbioc_1.42.0             zoo_1.8-10                 
 [61] scales_1.3.0                gggrid_0.2-0               
 [63] hms_1.1.1                   promises_1.2.0.1           
 [65] MatrixGenerics_1.8.0        ProtGenerics_1.28.0        
 [67] parallel_4.2.0              SummarizedExperiment_1.26.1
 [69] AnnotationFilter_1.20.0     LDlinkR_1.2.3              
 [71] yaml_2.3.5                  curl_4.3.2                 
 [73] memoise_2.0.1               sass_0.4.1                 
 [75] biomaRt_2.54.1              stringi_1.7.6              
 [77] RSQLite_2.3.1               highr_0.9                  
 [79] S4Vectors_0.34.0            BiocIO_1.6.0               
 [81] GenomicFeatures_1.48.3      BiocGenerics_0.42.0        
 [83] filelock_1.0.2              BiocParallel_1.30.3        
 [85] GenomeInfoDb_1.39.9         rlang_1.1.2                
 [87] pkgconfig_2.0.3             matrixStats_0.62.0         
 [89] bitops_1.0-7                evaluate_0.15              
 [91] lattice_0.20-45             labeling_0.4.2             
 [93] GenomicAlignments_1.32.0    htmlwidgets_1.5.4          
 [95] cowplot_1.1.1               bit_4.0.4                  
 [97] tidyselect_1.2.0            plyr_1.8.7                 
 [99] magrittr_2.0.3              R6_2.5.1                   
[101] IRanges_2.30.0              generics_0.1.2             
[103] DelayedArray_0.22.0         DBI_1.2.2                  
[105] withr_2.5.0                 haven_2.5.0                
[107] pgenlibr_0.3.3              pillar_1.9.0               
[109] KEGGREST_1.36.3             RCurl_1.98-1.7             
[111] mixsqp_0.3-43               modelr_0.1.8               
[113] crayon_1.5.1                utf8_1.2.2                 
[115] BiocFileCache_2.4.0         plotly_4.10.0              
[117] tzdb_0.4.0                  rmarkdown_2.25             
[119] progress_1.2.2              readxl_1.4.0               
[121] grid_4.2.0                  blob_1.2.3                 
[123] git2r_0.30.1                reprex_2.0.1               
[125] digest_0.6.29               httpuv_1.6.5               
[127] stats4_4.2.0                munsell_0.5.0              
[129] viridisLite_0.4.0           bslib_0.3.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
