<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-01-07" />

<title>Downstream analysis for multi-group analysis</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Downstream analysis for multi-group
analysis</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-01-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree1ddb625263916a6ad9357de8918ae5788c0b84d8targetblank1ddb625a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/1ddb625263916a6ad9357de8918ae5788c0b84d8" target="_blank">1ddb625</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree1ddb625263916a6ad9357de8918ae5788c0b84d8targetblank1ddb625a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/1ddb625263916a6ad9357de8918ae5788c0b84d8" target="_blank">1ddb625</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/LDL_silver_standard.Rmd
    Untracked:  analysis/multi_group_downstream_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ctwas)
library(ggplot2)
library(dplyr)
library(tidyr)
library(gridExtra)
library(pheatmap)
 library(VennDiagram)

source(&quot;/project/xinhe/xsun/multi_group_ctwas/data/selected_tissues.R&quot;)
source(&quot;/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R&quot;)

traits &lt;- names(tissues_alltraits)
traits &lt;- traits[order(traits)]

brain_traits &lt;- c(&quot;ASD-ieu-a-1185&quot;,&quot;BIP-ieu-b-5110&quot;,&quot;MDD-ieu-b-102&quot;,&quot;NS-ukb-a-230&quot;,&quot;PD-ieu-b-7&quot;,&quot;SCZ-ieu-b-5102&quot;)


create_bubble_plot &lt;- function(trait, param, gwas_n, tissue_order) {

  ctwas_parameters &lt;- summarize_param(param, gwas_n)
  
  # Extract and process PVE data
  prop_pve &lt;- ctwas_parameters$prop_heritability
  prop_pve_df &lt;- data.frame(
    Tissue = sapply(strsplit(names(prop_pve), &quot;\\|&quot;), `[`, 1),
    QTL = sapply(strsplit(names(prop_pve), &quot;\\|&quot;), `[`, 2),
    Value = prop_pve
  )
  prop_pve_matrix &lt;- as.data.frame(pivot_wider(prop_pve_df, names_from = QTL, values_from = Value))
  prop_pve_matrix &lt;- prop_pve_matrix[-which(prop_pve_matrix$Tissue == &quot;SNP&quot;),]
  prop_pve_matrix &lt;- prop_pve_matrix[,-which(colnames(prop_pve_matrix) == &quot;NA&quot;)]
  
  # Extract and process enrichment data
  enrich &lt;- ctwas_parameters$enrichment
  enrich_df &lt;- data.frame(
    Tissue = sapply(strsplit(names(enrich), &quot;\\|&quot;), `[`, 1),
    QTL = sapply(strsplit(names(enrich), &quot;\\|&quot;), `[`, 2),
    Value = enrich
  )
  enrich_matrix &lt;- as.data.frame(pivot_wider(enrich_df, names_from = QTL, values_from = Value))
  
  # Convert matrices to long format and merge
  pve_long &lt;- prop_pve_matrix %&gt;%
    pivot_longer(cols = -Tissue, names_to = &quot;Trait&quot;, values_to = &quot;prop_PVE&quot;)
  enrich_long &lt;- enrich_matrix %&gt;%
    pivot_longer(cols = -Tissue, names_to = &quot;Trait&quot;, values_to = &quot;Enrichment&quot;)
  
  plot_data &lt;- left_join(pve_long, enrich_long, by = c(&quot;Tissue&quot;, &quot;Trait&quot;))
  plot_data$Tissue &lt;- gsub(pattern = &quot;_&quot;, replacement = &quot; &quot;, x = plot_data$Tissue)
  plot_data &lt;- plot_data %&gt;%
    mutate(prop_PVE = prop_PVE * 100)
  #plot_data$Tissue &lt;- factor(plot_data$Tissue, levels = unique(plot_data$Tissue))
  plot_data$Tissue &lt;- factor(plot_data$Tissue, levels = rev(unique(plot_data$Tissue)))
  
  # Create the bubble plot
  p &lt;- ggplot(plot_data, aes(x = Trait, y = Tissue, size = prop_PVE, color = Enrichment)) +
    geom_point(alpha = 0.7) +
    scale_size(range = c(1, 20), name = &quot;Percentage of Heritability (%)&quot;) +
    scale_color_gradient(low = &quot;lightblue&quot;, high = &quot;darkblue&quot;, name = &quot;Enrichment&quot;) +
    labs(x = &quot;Modalities&quot;, y = &quot;Tissues&quot;) +
    guides(size = guide_legend(override.aes = list(color = &quot;lightblue&quot;))) +
    ggtitle(trait) +
    theme_minimal() +
    theme(
      axis.text.x = element_text(size = 16, angle = 45, hjust = 1),
      axis.text.y = element_text(size = 16),
      axis.title.x = element_text(size = 16),
      axis.title.y = element_text(size = 16),
      legend.text = element_text(size = 16),
      legend.title = element_text(size = 18)
    )
  
  return(p)
}

plot_heatmap_bytissue &lt;- function(heatmap_data, main, tissues) {
  
  rownames(heatmap_data) &lt;- heatmap_data$gene_name
  heatmap_data &lt;- heatmap_data %&gt;% dplyr::select(-gene_name, -combined_pip)
  
  pip_types &lt;- c(&quot;|eQTL_pip&quot;, &quot;|sQTL_pip&quot;, &quot;|stQTL_pip&quot;)

  combinations &lt;- expand.grid(pip_types, tissues)
  order &lt;- paste0(combinations$Var2, combinations$Var1)
  
  heatmap_data &lt;- heatmap_data[,order]
  
  if(nrow(heatmap_data) ==1){
    
    heatmap_data &lt;- rbind(heatmap_data,rep(0,ncol(heatmap_data)))
    rownames(heatmap_data)[2] &lt;- &quot;fake_gene_for_plotting&quot;
    
  }
  
  heatmap_matrix &lt;- as.matrix(heatmap_data)
  
  p &lt;- pheatmap(heatmap_matrix,
                cluster_rows = F,   # Cluster the rows (genes)
                cluster_cols = F,   # Cluster the columns (QTL types)
                color = colorRampPalette(c(&quot;white&quot;, &quot;red&quot;))(50), # Color gradient
                display_numbers = TRUE, # Display numbers in cells
                main = main,labels_row = rownames(heatmap_data), silent = T)
  
  return(p)
}

plot_heatmap_byomics &lt;- function(heatmap_data, main) {

  rownames(heatmap_data) &lt;- heatmap_data$gene_name
  heatmap_data &lt;- heatmap_data %&gt;% dplyr::select(-gene_name, -combined_pip)

  if(nrow(heatmap_data) ==1){

    heatmap_data &lt;- rbind(heatmap_data,rep(0,ncol(heatmap_data)))
    rownames(heatmap_data)[2] &lt;- &quot;fake_gene_for_plotting&quot;

  }

  heatmap_matrix &lt;- as.matrix(heatmap_data)

  p &lt;- pheatmap(heatmap_matrix,
                cluster_rows = F,   # Cluster the rows (genes)
                cluster_cols = F,   # Cluster the columns (QTL types)
                color = colorRampPalette(c(&quot;white&quot;, &quot;red&quot;))(50), # Color gradient
                display_numbers = TRUE, # Display numbers in cells
                main = main,labels_row = rownames(heatmap_data), silent = T)

  return(p)
}</code></pre>
<div id="genetic-architecture-of-complex-traits" class="section level1">
<h1>Genetic architecture of complex traits</h1>
<p>Bubble plot: h2g partition across tissues and omics</p>
<p>Tissue order is from: <a
href="https://sq-96.github.io/multigroup_ctwas_analysis/GWAS_tissue_selection.html"
class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/GWAS_tissue_selection.html</a></p>
<pre class="r"><code>folder_results &lt;- &quot;/project/xinhe/shengqian/ctwas_GWAS_analysis/results/&quot;

p &lt;- list()
for(trait in traits[!traits %in% brain_traits]){

  param &lt;- readRDS(paste0(folder_results, &quot;/&quot;, trait, &quot;/&quot;, trait, &quot;.param.RDS&quot;))
  gwas_n &lt;- samplesize[trait]
  tissue_order &lt;- tissues_alltraits[[trait]]
  p[[length(p)+1]] &lt;- create_bubble_plot(trait = trait,param = param,gwas_n = gwas_n, tissue_order = tissue_order)

}

print(&quot;Non-psychiatric&quot;)</code></pre>
<pre><code>[1] &quot;Non-psychiatric&quot;</code></pre>
<pre class="r"><code>grid.arrange(grobs = p, ncol = 3, nrow = 5)</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-2-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>folder_results &lt;- &quot;/project/xinhe/shengqian/ctwas_GWAS_analysis/results/&quot;

p &lt;- list()
for(trait in brain_traits){

  param &lt;- readRDS(paste0(folder_results, &quot;/&quot;, trait, &quot;/&quot;, trait, &quot;.param.RDS&quot;))
  gwas_n &lt;- samplesize[trait]
  tissue_order &lt;- tissues_alltraits[[trait]]
  p[[length(p)+1]] &lt;- create_bubble_plot(trait = trait,param = param,gwas_n = gwas_n, tissue_order = tissue_order)

}

print(&quot;psychiatric&quot;)</code></pre>
<pre><code>[1] &quot;psychiatric&quot;</code></pre>
<pre class="r"><code>grid.arrange(grobs = p, ncol = 3, nrow = 2)</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-3-1.png" width="4800" style="display: block; margin: auto;" /></p>
</div>
<div id="the-power-of-gene-discovery" class="section level1">
<h1>The power of gene discovery</h1>
<pre class="r"><code>load(&quot;/project/xinhe/xsun/multi_group_ctwas/13.post_processing_0103/results_downstream/compare_multi_single_genenum.rdata&quot;)

sum &lt;- sum[!sum$trait %in% brain_traits,]

sum$num_multi &lt;- as.numeric(sum$num_multi)
sum$num_single &lt;- as.numeric(sum$num_single)
sum$overlap &lt;- as.numeric(sum$overlap)
#sum$overlap_adj &lt;- as.numeric(sum$overlap) * 1.001   # Adjust the value to slightly offset behind the main bars

data_long &lt;- pivot_longer(sum, cols = c(num_single, num_multi), names_to = &quot;category&quot;, values_to = &quot;count&quot;)

print(&quot;Non-psychiatric&quot;)</code></pre>
<pre><code>[1] &quot;Non-psychiatric&quot;</code></pre>
<pre class="r"><code># Facet by trait, with tissues as the bars
ggplot(data_long, aes(x = tissue_single, y = count, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.8), width = 0.8) +
  geom_bar(data = sum, aes(x = tissue_single, y = overlap), stat = &quot;identity&quot;, position = position_dodge(width = 0.8), fill = &quot;grey&quot;, alpha = 0.7, width = 0.8) +
  facet_wrap(~ trait, nrow = 1, scales = &quot;free_x&quot;) +  # Display all facets in one row with free scales on x
  labs(x = &quot;Tissue&quot;, y = &quot;Number of Significant Genes&quot;) +
  scale_fill_manual(values = c(&quot;num_single&quot; = &quot;skyblue&quot;, &quot;num_multi&quot; = &quot;orange&quot;)) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, angle = 45, vjust = 0.7, hjust = 0.6),  # Adjusted hjust here
        axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14),
        strip.background = element_blank(),
        strip.text.x = element_text(size = 12, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-4-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>load(&quot;/project/xinhe/xsun/multi_group_ctwas/13.post_processing_0103/results_downstream/compare_multi_single_genenum.rdata&quot;)

sum &lt;- sum[sum$trait %in% brain_traits,]

sum$num_multi &lt;- as.numeric(sum$num_multi)
sum$num_single &lt;- as.numeric(sum$num_single)
sum$overlap &lt;- as.numeric(sum$overlap)
#sum$overlap_adj &lt;- as.numeric(sum$overlap) * 1.001   # Adjust the value to slightly offset behind the main bars

data_long &lt;- pivot_longer(sum, cols = c(num_single, num_multi), names_to = &quot;category&quot;, values_to = &quot;count&quot;)

print(&quot;psychiatric&quot;)</code></pre>
<pre><code>[1] &quot;psychiatric&quot;</code></pre>
<pre class="r"><code># Facet by trait, with tissues as the bars
ggplot(data_long, aes(x = tissue_single, y = count, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.8), width = 0.8) +
  geom_bar(data = sum, aes(x = tissue_single, y = overlap), stat = &quot;identity&quot;, position = position_dodge(width = 0.8), fill = &quot;grey&quot;, alpha = 0.7, width = 0.8) +
  facet_wrap(~ trait, nrow = 1, scales = &quot;free_x&quot;) +  # Display all facets in one row with free scales on x
  labs(x = &quot;Tissue&quot;, y = &quot;Number of Significant Genes&quot;) +
  scale_fill_manual(values = c(&quot;num_single&quot; = &quot;skyblue&quot;, &quot;num_multi&quot; = &quot;orange&quot;)) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, angle = 45, vjust = 0.7, hjust = 0.6),  # Adjusted hjust here
        axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14),
        strip.background = element_blank(),
        strip.text.x = element_text(size = 12, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="highlight-the-causal-context-and-modality"
class="section level1">
<h1>Highlight the causal context and modality</h1>
<p>For all: <a
href="https://drive.google.com/drive/folders/1COItzR1y_Em6UXb_J8XCzqP-PuXIth6J?usp=share_link"
class="uri">https://drive.google.com/drive/folders/1COItzR1y_Em6UXb_J8XCzqP-PuXIth6J?usp=share_link</a></p>
<div id="mdd-ieu-b-102" class="section level2">
<h2>MDD-ieu-b-102</h2>
<pre class="r"><code>trait &lt;- &quot;MDD-ieu-b-102&quot;

combined_pip_multi &lt;- readRDS(paste0(&quot;/project/xinhe/shengqian/ctwas_GWAS_analysis/results/&quot;,trait,&quot;/&quot;,trait,&quot;.combined_pip_bygroup_final.RDS&quot;))
combined_pip_sig_multi &lt;- combined_pip_multi[combined_pip_multi$combined_pip &gt; 0.8,]

# plot_heatmap_bytissue(heatmap_data = combined_pip_sig_multi, main = &quot;PIP partition for genes with PIP &gt; 0.8 from multi-group analysis&quot;,tissues = tissues_alltraits[[trait]])</code></pre>
<pre class="r"><code>plot_heatmap_byomics(heatmap_data = combined_pip_sig_multi, main = &quot;PIP partition for genes with PIP &gt; 0.8 from multi-group analysis&quot;)</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-7-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="explore-allelic-heterogeneity-ah" class="section level1">
<h1>Explore allelic heterogeneity (AH)</h1>
<pre class="r"><code>load(&quot;/project/xinhe/xsun/multi_group_ctwas/13.post_processing_0103/results_downstream/pip_per_cs_alltraits.rdata&quot;)
DT::datatable(pip_per_cs_alltraits,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;PIP per CS, for genes with AH&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">PIP per CS, for genes with AH<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181"],["aFib-ebi-a-GCST006414","aFib-ebi-a-GCST006414","aFib-ebi-a-GCST006414","aFib-ebi-a-GCST006414","aFib-ebi-a-GCST006414","aFib-ebi-a-GCST006414","ASD-ieu-a-1185","ATH_gtexukb","ATH_gtexukb","ATH_gtexukb","ATH_gtexukb","BIP-ieu-b-5110","BMI-panukb","BMI-panukb","BMI-panukb","BMI-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","HB-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","Height-panukb","HTN-panukb","IBD-ebi-a-GCST004131","IBD-ebi-a-GCST004131","IBD-ebi-a-GCST004131","IBD-ebi-a-GCST004131","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","LDL-ukb-d-30780_irnt","MDD-ieu-b-102","MDD-ieu-b-102","NS-ukb-a-230","PD-ieu-b-7","PD-ieu-b-7","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","PLT-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","RBC-panukb","SBP-ukb-a-360","SBP-ukb-a-360","SBP-ukb-a-360","SBP-ukb-a-360","SBP-ukb-a-360","SBP-ukb-a-360","SCZ-ieu-b-5102","T1D-GCST90014023","T1D-GCST90014023","T2D-panukb","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30","WBC-ieu-b-30"],["PAFAH1B2","PAFAH1B2","PCCB","PCCB","DTWD1","DTWD1",null,"LGALS8","LGALS8","RAD50","RAD50",null,"ADGRB2","ADGRB2","MRAS","MRAS","ABHD12","ABHD12","VEGFA","VEGFA","VEGFA","PFKM","PFKM","PFKM","LRRC47","LRRC47","TUBD1","TUBD1","ZMIZ2","ZMIZ2","SLC20A2","SLC20A2","EFNA3","EFNA3","TNK2","TNK2","CRISPLD2","CRISPLD2","SPIN1","SPIN1","CCDC6","CCDC6","ACHE","ACHE","ACP1","ACP1","TRIM41","TRIM41","RRBP1","RRBP1","ORM2","ORM2","UVSSA","UVSSA","SERPINH1","SERPINH1","DICER1","DICER1","HMGN1","HMGN1","H1FX","H1FX","CHST6","CHST6","CKAP4","CKAP4","CCND3","CCND3","SLC22A3","SLC22A3","USP47","USP47","ZRANB1","ZRANB1","PLA2R1","PLA2R1","ZZEF1","ZZEF1","MICAL3","MICAL3","PPP2R5E","PPP2R5E","CHCHD2","CHCHD2","KLC1","KLC1","PCSK5","PCSK5",null,"TNFSF15","TNFSF15","FCGR2A","FCGR2A","SNX17","SNX17","DNAJC13","DNAJC13","TRIM39","TRIM39","SP4","SP4","N4BP2L1","N4BP2L1","ABCA1","ABCA1","ESYT2","ESYT2",null,"SNCA","SNCA","CBL","CBL","ARHGAP15","ARHGAP15","TBC1D22A","TBC1D22A","CD33","CD33","GRTP1","GRTP1","FARP2","FARP2","RTEL1","RTEL1","RAB11FIP3","RAB11FIP3","NRF1","NRF1","PACSIN2","PACSIN2","RAD23A","RAD23A","ZFPM1","ZFPM1","PFKM","PFKM","VEGFA","VEGFA","SLC25A37","SLC25A37","SKAP1","SKAP1","TTC13","TTC13","EHBP1L1","EHBP1L1","CPEB4","CPEB4","FBXO7","FBXO7","PPP2R2A","PPP2R2A","BAG6","BAG6","CLCN6","CLCN6","PIGV","PIGV",null,"PLEKHM1","PLEKHM1",null,"ARHGAP15","ARHGAP15","ARHGAP15","NLRC5","NLRC5","CPEB4","CPEB4","VLDLR","VLDLR","ELMO1","ELMO1","CXCL12","CXCL12","COQ8A","COQ8A","SLC45A4","SLC45A4","DNAJC13","DNAJC13"],["L1","L3","L1","L3","L1","L3",null,"L1","L3","L1","L4",null,"L1","L3","L1","L3","L1","L3","L1","L2","L3","L1","L2","L4","L1","L2","L2","L4","L1","L2","L1","L2","L4","L5","L1","L2","L2","L4","L1","L2","L2","L5","L1","L4","L1","L2","L1","L2","L1","L4","L2","L3","L1","L5","L1","L4","L2","L4","L1","L2","L1","L3","L1","L3","L1","L2","L1","L4","L1","L3","L1","L2","L3","L4","L3","L4","L1","L2","L1","L3","L3","L4","L1","L3","L1","L4","L2","L5",null,"L1","L3","L1","L3","L1","L2","L1","L2","L1","L2","L1","L2","L1","L2","L1","L2","L1","L3",null,"L1","L2","L1","L5","L1","L2","L1","L3","L1","L3","L1","L2","L2","L3","L1","L2","L1","L3","L2","L4","L1","L4","L1","L3","L1","L3","L2","L3","L1","L3","L1","L3","L1","L2","L3","L5","L2","L4","L1","L2","L1","L2","L1","L2","L1","L3","L1","L2","L1","L3",null,"L1","L3",null,"L1","L2","L3","L1","L2","L1","L3","L1","L2","L1","L4","L1","L3","L1","L2","L1","L2","L1","L2"],[0.882,0.894,0.448,0.966,0.015,0.931,null,1,0.953,0.958,0.015,null,0.934,0.645,0.808,0.015,1,0.016,0.967,0.981,0.963,0.948,0.027,0.974,0.975,0.231,0.234,0.965,0.926,0.39,0.007,0.942,0.866,0.334,0.679,0.689,1,0.999,1,0.969,1,0.906,0.99,0.98,0.972,0.975,0.993,0.761,0.9,0.959,0.914,0.925,0.993,0.021,0.978,0.615,0.962,0.778,0.962,0.754,0.968,0.705,0.034,0.99,0.984,0.02,0.978,0.115,0.643,0.934,0.218,0.955,0.218,0.948,0.945,0.239,0.017,0.94,1.321,0.004,0.814,0.4,0.842,0.149,0.047,0.846,0.8,0.025,null,1,0.061,0.934,0.969,1,0.122,0.988,0.947,0.999,0.088,0.986,0.024,0.905,0.344,0.684,0.595,0.924,0.265,null,0.987,0.874,1,0.468,1,0.85,0.997,0.881,0.954,0.97,0.996,0.039,0.738,0.971,0.106,0.987,0.862,0.49,0.709,0.648,0.042,0.872,0.822,0.013,0.514,0.61,0.99,0.712,0.888,0.969,0.979,0.38,0.976,0.175,0.94,0.56,0.967,0.115,0.751,0.845,0.308,0.898,0.468,0.785,0.967,0.997,0.964,0.81,0.589,0.762,null,0.989,0.989,null,1,0.737,0.508,0.991,1,0.994,0.507,0.949,0.884,0.641,0.934,0.175,0.959,0.946,0.105,0.854,0.18,0.252,0.77],[1,1,3,1,1,1,null,1,1,5,2,null,8,1,7,1,1,10,1,2,1,2,1,1,1,6,4,1,16,11,1,2,4,2,3,1,1,1,1,4,1,2,1,1,3,1,4,2,1,10,1,1,2,1,2,2,7,2,1,1,2,2,5,1,1,1,2,1,2,2,2,1,4,1,1,1,1,7,10,1,9,13,3,3,1,10,4,2,null,1,1,1,2,1,9,2,2,1,3,1,1,1,1,1,1,10,5,null,4,6,1,2,1,1,1,4,3,1,6,1,7,4,1,8,3,2,5,1,1,1,12,1,1,1,3,2,2,4,1,6,3,1,2,1,5,1,1,1,2,8,1,4,1,1,5,4,1,1,null,1,3,null,1,1,5,1,1,1,1,1,1,1,2,2,3,1,3,1,1,5,4],["ENSG00000168092.13|Heart_Left_Ventricle_eQTL(0.882)","ENSG00000168092.13|Heart_Left_Ventricle_eQTL(0.894)","ENSG00000114054.13|Muscle_Skeletal_eQTL(0.128), ENSG00000114054.13|Heart_Left_Ventricle_eQTL(0.181), ENSG00000114054.grp_2.downstream.ENST00000484181|Heart_Atrial_Appendage_stQTL(0.14)","ENSG00000114054.grp_2.downstream.ENST00000484181|Heart_Atrial_Appendage_stQTL(0.966)","ENSG00000104047|Heart_Left_Ventricle_stQTL(0.015)","ENSG00000104047|Heart_Left_Ventricle_stQTL(0.931)",null,"intron_1_236541710_236542761|Colon_Transverse_sQTL(1)","intron_1_236542787_236542915|Esophagus_Muscularis_sQTL(0.953)","intron_5_132638223_132640672|Esophagus_Muscularis_sQTL(0.02), intron_5_132580066_132587562|Esophagus_Muscularis_sQTL(0.25), intron_5_132579502_132579862|Esophagus_Muscularis_sQTL(0.25), intron_5_132580066_132587562|Colon_Transverse_sQTL(0.219), intron_5_132579502_132579862|Colon_Transverse_sQTL(0.219)","intron_5_132556473_132559284|Esophagus_Muscularis_sQTL(0.008), intron_5_132449689_132486192|Esophagus_Muscularis_sQTL(0.007)",null,"intron_1_31728081_31728182|Cells_Cultured_fibroblasts_sQTL(0.011), intron_1_31727605_31728182|Cells_Cultured_fibroblasts_sQTL(0.011), intron_1_31727605_31728025|Cells_Cultured_fibroblasts_sQTL(0.011), intron_1_31728081_31728182|Brain_Cerebellar_Hemisphere_sQTL(0.154), intron_1_31727605_31728182|Brain_Cerebellar_Hemisphere_sQTL(0.154), intron_1_31727605_31728025|Brain_Cerebellar_Hemisphere_sQTL(0.154), intron_1_31728081_31728182|Adipose_Subcutaneous_sQTL(0.219), intron_1_31727605_31728182|Adipose_Subcutaneous_sQTL(0.219)","intron_1_31732154_31732517|Brain_Cerebellar_Hemisphere_sQTL(0.645)","intron_3_138373076_138397324|Cells_Cultured_fibroblasts_sQTL(0.012), intron_3_138373076_138373221|Cells_Cultured_fibroblasts_sQTL(0.012), intron_3_138373076_138397324|Brain_Cerebellar_Hemisphere_sQTL(0.157), intron_3_138373076_138373221|Brain_Cerebellar_Hemisphere_sQTL(0.157), intron_3_138373076_138397324|Adipose_Subcutaneous_sQTL(0.224), intron_3_138373076_138373221|Adipose_Subcutaneous_sQTL(0.224), ENSG00000158186|Brain_Cerebellar_Hemisphere_stQTL(0.024)","ENSG00000158186.12|Adipose_Subcutaneous_eQTL(0.015)","ENSG00000100997|Artery_Tibial_stQTL(1)","intron_20_25323430_25339220|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.001), intron_20_25317078_25323325|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.001), intron_20_25314970_25317048|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.001), intron_20_25300884_25301332|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.001), intron_20_25323430_25339220|Breast_Mammary_Tissue_sQTL(0.001), intron_20_25300884_25302219|Whole_Blood_sQTL(0.001), intron_20_25317078_25323325|Artery_Tibial_sQTL(0.001), intron_20_25301833_25302219|Artery_Tibial_sQTL(0.001), intron_20_25300884_25301719|Artery_Tibial_sQTL(0.001), ENSG00000100997|Whole_Blood_stQTL(0.007)","ENSG00000112715.21|Breast_Mammary_Tissue_eQTL(0.967)","intron_6_43780803_43781956|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.667), intron_6_43780785_43781956|Artery_Tibial_sQTL(0.314)","ENSG00000112715.21|Artery_Tibial_eQTL(0.963)","intron_12_48119406_48122767|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.378), intron_12_48119406_48122767|Artery_Tibial_sQTL(0.57)","intron_12_48119406_48122767|Breast_Mammary_Tissue_sQTL(0.027)","ENSG00000152556.15|Esophagus_Gastroesophageal_Junction_eQTL(0.974)","ENSG00000130764.9|Whole_Blood_eQTL(0.975)","intron_1_3783911_3783996|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.048), intron_1_3782763_3783770|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.009), intron_1_3783911_3783996|Breast_Mammary_Tissue_sQTL(0.143), intron_1_3783841_3783996|Breast_Mammary_Tissue_sQTL(0.008), intron_1_3782763_3783996|Breast_Mammary_Tissue_sQTL(0.009), intron_1_3782763_3783770|Artery_Tibial_sQTL(0.013)","intron_17_59863847_59874539|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.054), intron_17_59874703_59878103|Breast_Mammary_Tissue_sQTL(0.041), intron_17_59863847_59874539|Breast_Mammary_Tissue_sQTL(0.058), intron_17_59863847_59874539|Artery_Tibial_sQTL(0.082)","ENSG00000108423.grp_2.downstream.ENST00000394239|Esophagus_Gastroesophageal_Junction_stQTL(0.965)","ENSG00000122515.grp_1.downstream.ENST00000265346|Whole_Blood_stQTL(0.035), ENSG00000122515.grp_1.downstream.ENST00000413916|Whole_Blood_stQTL(0.067), ENSG00000122515.grp_2.downstream.ENST00000265346|Whole_Blood_stQTL(0.074), ENSG00000122515.grp_1.downstream.ENST00000265346|Artery_Tibial_stQTL(0.026), ENSG00000122515.grp_1.downstream.ENST00000413916|Artery_Tibial_stQTL(0.027), ENSG00000122515.grp_2.downstream.ENST00000265346|Artery_Tibial_stQTL(0.026), ENSG00000122515.grp_2.downstream.ENST00000413916|Artery_Tibial_stQTL(0.026), ENSG00000122515.grp_1.downstream.ENST00000413916|Esophagus_Gastroesophageal_Junction_stQTL(0.06), ENSG00000122515.grp_2.downstream.ENST00000265346|Esophagus_Gastroesophageal_Junction_stQTL(0.06), ENSG00000122515.grp_2.downstream.ENST00000413916|Esophagus_Gastroesophageal_Junction_stQTL(0.06), ENSG00000122515.grp_1.downstream.ENST00000413916|Skin_Not_Sun_Exposed_Suprapubic_stQTL(0.065), ENSG00000122515.grp_2.downstream.ENST00000413916|Skin_Not_Sun_Exposed_Suprapubic_stQTL(0.123), ENSG00000122515.grp_1.downstream.ENST00000265346|Esophagus_Gastroesophageal_Junction_stQTL(0.072), ENSG00000122515.grp_2.downstream.ENST00000413916|Whole_Blood_stQTL(0.097), ENSG00000122515.grp_1.downstream.ENST00000265346|Skin_Not_Sun_Exposed_Suprapubic_stQTL(0.077), ENSG00000122515.grp_2.downstream.ENST00000265346|Skin_Not_Sun_Exposed_Suprapubic_stQTL(0.032)","ENSG00000122515.14|Skin_Not_Sun_Exposed_Suprapubic_eQTL(0.124), ENSG00000122515.14|Breast_Mammary_Tissue_eQTL(0.109), intron_7_44759460_44760151|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.015), intron_7_44760228_44760425|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.015), intron_7_44759746_44760151|Breast_Mammary_Tissue_sQTL(0.015), intron_7_44759460_44760151|Breast_Mammary_Tissue_sQTL(0.015), intron_7_44759460_44760425|Breast_Mammary_Tissue_sQTL(0.015), intron_7_44760228_44760425|Breast_Mammary_Tissue_sQTL(0.015), intron_7_44759460_44760151|Artery_Tibial_sQTL(0.022), intron_7_44759460_44760425|Artery_Tibial_sQTL(0.022), intron_7_44760228_44760425|Artery_Tibial_sQTL(0.022)","ENSG00000168575.9|Esophagus_Gastroesophageal_Junction_eQTL(0.007)","ENSG00000168575.9|Artery_Tibial_eQTL(0.902), intron_8_42465917_42472102|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.04)","intron_1_155085942_155086128|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.171), intron_1_155085942_155086128|Breast_Mammary_Tissue_sQTL(0.06), intron_1_155085942_155086128|Artery_Tibial_sQTL(0.258), intron_1_155085942_155086413|Artery_Tibial_sQTL(0.378)","ENSG00000143590.13|Breast_Mammary_Tissue_eQTL(0.15), ENSG00000143590.13|Whole_Blood_eQTL(0.184)","intron_3_195869541_195870114|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.051), intron_3_195869541_195870114|Breast_Mammary_Tissue_sQTL(0.259), intron_3_195869541_195870114|Artery_Tibial_sQTL(0.368)","ENSG00000061938.16|Artery_Tibial_eQTL(0.689)","ENSG00000103196|Cells_Cultured_fibroblasts_stQTL(1)","ENSG00000103196.11|Cells_Cultured_fibroblasts_eQTL(0.999)","ENSG00000106723.16|Heart_Atrial_Appendage_eQTL(1)","intron_9_88448989_88462496|Cells_Cultured_fibroblasts_sQTL(0.297), intron_9_88448989_88462496|Artery_Tibial_sQTL(0.364), intron_9_88448989_88462496|Heart_Atrial_Appendage_sQTL(0.214), intron_9_88448989_88462496|Muscle_Skeletal_sQTL(0.095)","ENSG00000108091.10|Thyroid_eQTL(1)","ENSG00000108091|Artery_Tibial_stQTL(0.029), ENSG00000108091|Thyroid_stQTL(0.877)","ENSG00000087085.13|Artery_Tibial_eQTL(0.99)","ENSG00000087085|Heart_Atrial_Appendage_stQTL(0.98)","intron_2_272305_275140|Artery_Tibial_sQTL(0.045), intron_2_272150_275140|Artery_Tibial_sQTL(0.444), intron_2_272150_275140|Heart_Atrial_Appendage_sQTL(0.482)","ENSG00000143727.15|Muscle_Skeletal_eQTL(0.975)","ENSG00000146063.grp_2.downstream.ENST00000351937|Cells_Cultured_fibroblasts_stQTL(0.147), ENSG00000146063.grp_2.downstream.ENST00000515499|Cells_Cultured_fibroblasts_stQTL(0.47), ENSG00000146063.grp_2.downstream.ENST00000351937|Muscle_Skeletal_stQTL(0.312), ENSG00000146063|Thyroid_stQTL(0.065)","ENSG00000146063.19|Cells_Cultured_fibroblasts_eQTL(0.281), ENSG00000146063.19|Thyroid_eQTL(0.481)","ENSG00000125844.15|Muscle_Skeletal_eQTL(0.9)","intron_20_17615529_17615926|Artery_Tibial_sQTL(0.025), intron_20_17615523_17615926|Artery_Tibial_sQTL(0.025), intron_20_17615529_17615926|Heart_Atrial_Appendage_sQTL(0.014), intron_20_17615523_17615926|Heart_Atrial_Appendage_sQTL(0.014), ENSG00000125844.grp_1.downstream.ENST00000455029|Cells_Cultured_fibroblasts_stQTL(0.262), ENSG00000125844.grp_1.downstream.ENST00000468428|Cells_Cultured_fibroblasts_stQTL(0.269), ENSG00000125844.grp_2.downstream.ENST00000377813|Cells_Cultured_fibroblasts_stQTL(0.118), ENSG00000125844.grp_2.downstream.ENST00000455029|Cells_Cultured_fibroblasts_stQTL(0.114), ENSG00000125844.grp_1.downstream.ENST00000455029|Heart_Atrial_Appendage_stQTL(0.053), ENSG00000125844.grp_1.downstream.ENST00000468428|Heart_Atrial_Appendage_stQTL(0.064)","ENSG00000228278.3|Cells_Cultured_fibroblasts_eQTL(0.914)","ENSG00000228278.3|Muscle_Skeletal_eQTL(0.925)","ENSG00000163945.15|Artery_Tibial_eQTL(0.895), intron_4_1347229_1348090|Cells_Cultured_fibroblasts_sQTL(0.098)","ENSG00000163945.grp_1.downstream.ENST00000677428|Muscle_Skeletal_stQTL(0.021)","ENSG00000149257.13|Artery_Tibial_eQTL(0.605), ENSG00000149257.13|Thyroid_eQTL(0.374)","ENSG00000149257.grp_2.downstream.ENST00000530284|Artery_Tibial_stQTL(0.275), ENSG00000149257|Thyroid_stQTL(0.34)","ENSG00000100697.14|Muscle_Skeletal_eQTL(0.129), intron_14_95087229_95088849|Muscle_Skeletal_sQTL(0.016), intron_14_95087898_95088849|Muscle_Skeletal_sQTL(0.016), ENSG00000100697.grp_1.downstream.ENST00000529720|Muscle_Skeletal_stQTL(0.21), ENSG00000100697.grp_1.downstream.ENST00000696736|Muscle_Skeletal_stQTL(0.21), ENSG00000100697.grp_2.downstream.ENST00000527416|Muscle_Skeletal_stQTL(0.179), ENSG00000100697.grp_2.downstream.ENST00000529720|Muscle_Skeletal_stQTL(0.202)","intron_14_95117754_95124196|Muscle_Skeletal_sQTL(0.103), ENSG00000100697|Cells_Cultured_fibroblasts_stQTL(0.675)","ENSG00000205581.10|Artery_Tibial_eQTL(0.962)","intron_21_39345274_39347379|Cells_Cultured_fibroblasts_sQTL(0.754)","ENSG00000184897.5|Cells_Cultured_fibroblasts_eQTL(0.585), ENSG00000184897.5|Thyroid_eQTL(0.383)","ENSG00000184897.5|Artery_Tibial_eQTL(0.375), ENSG00000184897.5|Muscle_Skeletal_eQTL(0.33)","intron_16_75451839_75464468|Cells_Cultured_fibroblasts_sQTL(0.007), intron_16_75451839_75464232|Cells_Cultured_fibroblasts_sQTL(0.007), intron_16_75451839_75464468|Artery_Tibial_sQTL(0.008), intron_16_75451839_75464232|Artery_Tibial_sQTL(0.008), intron_16_75451839_75464468|Heart_Atrial_Appendage_sQTL(0.005)","ENSG00000183196.8|Heart_Atrial_Appendage_eQTL(0.99)","ENSG00000136026|Cells_Cultured_fibroblasts_stQTL(0.984)","ENSG00000136026.13|Cells_Cultured_fibroblasts_eQTL(0.02)","intron_6_41936107_41936559|Cells_Cultured_fibroblasts_sQTL(0.741), intron_6_41936107_41936559|Muscle_Skeletal_sQTL(0.236)","ENSG00000112576.12|Muscle_Skeletal_eQTL(0.115)","ENSG00000146477.5|Cells_Cultured_fibroblasts_eQTL(0.389), ENSG00000146477.5|Thyroid_eQTL(0.254)","intron_6_160410846_160425851|Artery_Tibial_sQTL(0.27), intron_6_160426076_160436780|Artery_Tibial_sQTL(0.664)","ENSG00000170242.17|Heart_Atrial_Appendage_eQTL(0.17), intron_11_11842224_11880177|Cells_Cultured_fibroblasts_sQTL(0.048)","ENSG00000170242.17|Artery_Tibial_eQTL(0.955)","ENSG00000019995.6|Muscle_Skeletal_eQTL(0.026), ENSG00000019995|Artery_Tibial_stQTL(0.144), ENSG00000019995|Thyroid_stQTL(0.041), ENSG00000019995|Muscle_Skeletal_stQTL(0.007)","ENSG00000019995.6|Thyroid_eQTL(0.948)","ENSG00000153246.12|Thyroid_eQTL(0.945)","intron_2_160028963_160032959|Cells_Cultured_fibroblasts_sQTL(0.239)","intron_17_4054195_4057994|Artery_Tibial_sQTL(0.017)","intron_17_4070924_4072608|Cells_Cultured_fibroblasts_sQTL(0.157), intron_17_4071446_4072608|Cells_Cultured_fibroblasts_sQTL(0.224), intron_17_4070924_4072608|Artery_Tibial_sQTL(0.189), intron_17_4071446_4072608|Artery_Tibial_sQTL(0.089), intron_17_4070924_4072608|Heart_Atrial_Appendage_sQTL(0.169), intron_17_4071446_4072608|Heart_Atrial_Appendage_sQTL(0.045), intron_17_4070924_4072608|Muscle_Skeletal_sQTL(0.067)","intron_22_17810813_17816690|Cells_Cultured_fibroblasts_sQTL(0.156), intron_22_17791301_17808844|Cells_Cultured_fibroblasts_sQTL(0.023), intron_22_17812557_17816690|Artery_Tibial_sQTL(0.072), intron_22_17810813_17816690|Artery_Tibial_sQTL(0.422), intron_22_17791301_17806239|Artery_Tibial_sQTL(0.048), intron_22_17812557_17816690|Heart_Atrial_Appendage_sQTL(0.042), intron_22_17810813_17812507|Heart_Atrial_Appendage_sQTL(0.042), intron_22_17866012_17871837|Artery_Tibial_sQTL(0.287), intron_22_17889230_17891485|Artery_Tibial_sQTL(0.038), ENSG00000243156|Heart_Atrial_Appendage_stQTL(0.192)","ENSG00000243156.7|Cells_Cultured_fibroblasts_eQTL(0.004)","ENSG00000154001.13|Thyroid_eQTL(0.22), intron_14_63539692_63542779|Cells_Cultured_fibroblasts_sQTL(0.09), intron_14_63539689_63542779|Cells_Cultured_fibroblasts_sQTL(0.09), intron_14_63539692_63542779|Artery_Tibial_sQTL(0.113), intron_14_63539689_63542779|Artery_Tibial_sQTL(0.113), intron_14_63539692_63542779|Heart_Atrial_Appendage_sQTL(0.065), intron_14_63539689_63542779|Heart_Atrial_Appendage_sQTL(0.065), intron_14_63539692_63542779|Muscle_Skeletal_sQTL(0.029), intron_14_63539689_63542779|Muscle_Skeletal_sQTL(0.029)","ENSG00000154001.13|Heart_Atrial_Appendage_eQTL(0.055), ENSG00000154001.grp_1.downstream.ENST00000422769|Cells_Cultured_fibroblasts_stQTL(0.036), ENSG00000154001.grp_1.downstream.ENST00000555899|Cells_Cultured_fibroblasts_stQTL(0.063), ENSG00000154001.grp_2.downstream.ENST00000337537|Cells_Cultured_fibroblasts_stQTL(0.014), ENSG00000154001.grp_2.downstream.ENST00000422769|Cells_Cultured_fibroblasts_stQTL(0.02), ENSG00000154001.grp_1.downstream.ENST00000337537|Artery_Tibial_stQTL(0.034), ENSG00000154001.grp_1.downstream.ENST00000422769|Artery_Tibial_stQTL(0.022), ENSG00000154001.grp_2.downstream.ENST00000337537|Artery_Tibial_stQTL(0.036), ENSG00000154001.grp_2.downstream.ENST00000422769|Artery_Tibial_stQTL(0.037), ENSG00000154001|Thyroid_stQTL(0.014), ENSG00000154001.grp_2.downstream.ENST00000337537|Muscle_Skeletal_stQTL(0.02), ENSG00000154001.grp_2.downstream.ENST00000422769|Muscle_Skeletal_stQTL(0.015), ENSG00000154001.grp_1.downstream.ENST00000422769|Muscle_Skeletal_stQTL(0.034)","ENSG00000106153|Artery_Tibial_stQTL(0.672), ENSG00000106153|Heart_Atrial_Appendage_stQTL(0.164), ENSG00000106153|Thyroid_stQTL(0.006)","intron_7_56114897_56116015|Artery_Tibial_sQTL(0.03), intron_7_56021231_56031929|Artery_Tibial_sQTL(0.095), intron_7_56021231_56031929|Muscle_Skeletal_sQTL(0.025)","ENSG00000126214.grp_2.downstream.ENST00000334553|Heart_Atrial_Appendage_stQTL(0.047)","intron_14_103679545_103684986|Cells_Cultured_fibroblasts_sQTL(0.029), intron_14_103679518_103684986|Artery_Tibial_sQTL(0.248), intron_14_103679518_103700655|Artery_Tibial_sQTL(0.036), intron_14_103679545_103684986|Artery_Tibial_sQTL(0.226), intron_14_103679545_103687081|Artery_Tibial_sQTL(0.226), intron_14_103679518_103700655|Heart_Atrial_Appendage_sQTL(0.021), intron_14_103679545_103684986|Heart_Atrial_Appendage_sQTL(0.021), intron_14_103679545_103687081|Heart_Atrial_Appendage_sQTL(0.021), intron_14_103679518_103700655|Muscle_Skeletal_sQTL(0.009), intron_14_103679545_103687081|Muscle_Skeletal_sQTL(0.009)","intron_9_76134212_76157045|Cells_Cultured_fibroblasts_sQTL(0.2), intron_9_76134212_76157045|Artery_Tibial_sQTL(0.2), intron_9_76134212_76157045|Heart_Atrial_Appendage_sQTL(0.2), intron_9_76134212_76157045|Thyroid_sQTL(0.2)","intron_9_76071898_76093110|Artery_Tibial_sQTL(0.013), intron_9_76071898_76095890|Thyroid_sQTL(0.013)",null,"ENSG00000181634.7|Esophagus_Muscularis_eQTL(1)","ENSG00000181634.7|Whole_Blood_eQTL(0.061)","ENSG00000143226.13|Brain_Nucleus_accumbens_basal_ganglia_eQTL(0.934)","ENSG00000143226|Colon_Transverse_stQTL(0.175), ENSG00000143226|Esophagus_Muscularis_stQTL(0.794)","intron_2_27372740_27373247|Esophagus_Muscularis_sQTL(1)","ENSG00000115234.10|Esophagus_Gastroesophageal_Junction_eQTL(0.016), intron_2_27376183_27376313|Spleen_sQTL(0.037), intron_2_27376216_27376313|Spleen_sQTL(0.037), intron_2_27376183_27376313|Esophagus_Gastroesophageal_Junction_sQTL(0.002), intron_2_27376216_27376313|Esophagus_Gastroesophageal_Junction_sQTL(0.002), intron_2_27376183_27376313|Esophagus_Muscularis_sQTL(0.01), intron_2_27376216_27376313|Esophagus_Muscularis_sQTL(0.01), intron_2_27376183_27376313|Esophagus_Mucosa_sQTL(0.004), intron_2_27376216_27376313|Esophagus_Mucosa_sQTL(0.004)","ENSG00000138246.15|Spleen_eQTL(0.893), ENSG00000138246.15|Esophagus_Gastroesophageal_Junction_eQTL(0.095)","intron_3_132467313_132472557|Esophagus_Muscularis_sQTL(0.003), ENSG00000138246|Esophagus_Mucosa_stQTL(0.944)","ENSG00000204599.14|Liver_eQTL(0.999)","intron_6_30335975_30339908|Spleen_sQTL(0.064), intron_6_30335975_30339908|Esophagus_Muscularis_sQTL(0.017), intron_6_30335975_30339908|Esophagus_Mucosa_sQTL(0.007)","ENSG00000105866.13|Liver_eQTL(0.986)","ENSG00000105866.13|Esophagus_Muscularis_eQTL(0.024)","ENSG00000139597.17|Esophagus_Gastroesophageal_Junction_eQTL(0.905)","ENSG00000139597.17|Esophagus_Mucosa_eQTL(0.344)","intron_9_104903771_104927935|Spleen_sQTL(0.684)","intron_9_104827169_104828916|Esophagus_Muscularis_sQTL(0.595)","ENSG00000117868.15|Brain_Frontal_Cortex_BA9_eQTL(0.011), ENSG00000117868.15|Brain_Cortex_eQTL(0.014), ENSG00000117868.15|Brain_Nucleus_accumbens_basal_ganglia_eQTL(0.016), ENSG00000117868.15|Brain_Caudate_basal_ganglia_eQTL(0.003), intron_7_158737179_158737288|Brain_Cerebellum_sQTL(0.008), intron_7_158739121_158739975|Brain_Cerebellum_sQTL(0.716), intron_7_158737179_158739023|Brain_Cerebellum_sQTL(0.007), intron_7_158739121_158739975|Brain_Frontal_Cortex_BA9_sQTL(0.049), intron_7_158739121_158741523|Brain_Frontal_Cortex_BA9_sQTL(0.047), intron_7_158739121_158739975|Brain_Caudate_basal_ganglia_sQTL(0.053)","ENSG00000117868.15|Brain_Cortex_eQTL(0.039), ENSG00000117868.15|Brain_Nucleus_accumbens_basal_ganglia_eQTL(0.065), ENSG00000117868.15|Brain_Caudate_basal_ganglia_eQTL(0.017), intron_7_158737179_158737288|Brain_Cerebellum_sQTL(0.077), intron_7_158737179_158739023|Brain_Cerebellum_sQTL(0.067)",null,"ENSG00000145335.grp_1.downstream.ENST00000336904|Brain_Cerebellum_stQTL(0.389), ENSG00000145335.grp_2.downstream.ENST00000336904|Brain_Cerebellum_stQTL(0.454), ENSG00000145335.grp_1.downstream.ENST00000336904|Brain_Frontal_Cortex_BA9_stQTL(0.073), ENSG00000145335.grp_2.downstream.ENST00000336904|Brain_Frontal_Cortex_BA9_stQTL(0.071)","intron_4_89835692_89836962|Brain_Hypothalamus_sQTL(0.098), intron_4_89835692_89836743|Brain_Cerebellum_sQTL(0.128), intron_4_89835692_89836962|Brain_Cortex_sQTL(0.164), intron_4_89835692_89836743|Brain_Cortex_sQTL(0.164), intron_4_89835692_89836962|Brain_Frontal_Cortex_BA9_sQTL(0.14), intron_4_89835692_89836743|Brain_Cerebellar_Hemisphere_sQTL(0.181)","ENSG00000110395.6|Artery_Tibial_eQTL(1)","ENSG00000110395.6|Whole_Blood_eQTL(0.234), ENSG00000110395.6|Liver_eQTL(0.234)","ENSG00000075884.13|Whole_Blood_eQTL(1)","intron_2_143129466_143155477|Spleen_sQTL(0.85)","ENSG00000054611|Whole_Blood_stQTL(0.997)","ENSG00000054611.13|Whole_Blood_eQTL(0.437), ENSG00000054611.13|Liver_eQTL(0.103), ENSG00000054611.13|Spleen_eQTL(0.095), ENSG00000054611.13|Artery_Tibial_eQTL(0.245)","intron_19_51225155_51225218|Spleen_sQTL(0.35), intron_19_51225598_51225803|Artery_Tibial_sQTL(0.302), intron_19_51225155_51225218|Artery_Tibial_sQTL(0.302)","ENSG00000105383.14|Whole_Blood_eQTL(0.97)","ENSG00000139835.13|Liver_eQTL(0.25), ENSG00000139835.13|Spleen_eQTL(0.23), ENSG00000139835.13|Brain_Frontal_Cortex_BA9_eQTL(0.177), ENSG00000139835.13|Artery_Tibial_eQTL(0.072), intron_13_113360284_113363762|Spleen_sQTL(0.143), intron_13_113360284_113363762|Artery_Tibial_sQTL(0.123)","intron_13_113363910_113364072|Liver_sQTL(0.039)","ENSG00000006607.13|Liver_eQTL(0.004), intron_2_241468377_241475857|Spleen_sQTL(0.011), intron_2_241475987_241483465|Brain_Frontal_Cortex_BA9_sQTL(0.014), intron_2_241475987_241483465|Artery_Tibial_sQTL(0.011), intron_2_241468377_241475857|Artery_Tibial_sQTL(0.007), ENSG00000006607.grp_1.downstream.ENST00000373287|Brain_Frontal_Cortex_BA9_stQTL(0.654), ENSG00000006607.grp_1.downstream.ENST00000373287|Artery_Tibial_stQTL(0.036)","intron_2_241404841_241407537|Liver_sQTL(0.16), intron_2_241404841_241407537|Spleen_sQTL(0.268), intron_2_241404841_241407537|Brain_Frontal_Cortex_BA9_sQTL(0.312), intron_2_241404841_241407537|Artery_Tibial_sQTL(0.231)","ENSG00000258366|Spleen_stQTL(0.106)","intron_20_63694974_63695066|Liver_sQTL(0.086), intron_20_63694488_63694741|Liver_sQTL(0.086), intron_20_63694488_63694741|Spleen_sQTL(0.139), intron_20_63694974_63695066|Spleen_sQTL(0.137), intron_20_63694974_63695066|Brain_Frontal_Cortex_BA9_sQTL(0.169), intron_20_63694488_63694741|Brain_Frontal_Cortex_BA9_sQTL(0.134), intron_20_63694488_63694741|Artery_Tibial_sQTL(0.118), intron_20_63694974_63695066|Artery_Tibial_sQTL(0.118)","ENSG00000090565.15|Whole_Blood_eQTL(0.646), intron_16_477656_482525|Liver_sQTL(0.081), intron_16_477656_482525|Spleen_sQTL(0.135)","intron_16_477656_482525|Artery_Tibial_sQTL(0.195), intron_16_471389_482525|Artery_Tibial_sQTL(0.295)","intron_7_129641822_129657343|Spleen_sQTL(0.063), intron_7_129611824_129657343|Spleen_sQTL(0.194), intron_7_129641822_129657343|Brain_Frontal_Cortex_BA9_sQTL(0.075), intron_7_129641822_129657343|Artery_Tibial_sQTL(0.055), intron_7_129611824_129657346|Artery_Tibial_sQTL(0.322)","ENSG00000106459.14|Whole_Blood_eQTL(0.648)","ENSG00000100266.grp_2.downstream.ENST00000263246|Spleen_stQTL(0.042)","ENSG00000100266.18|Spleen_eQTL(0.872)","intron_19_12948552_12948686|Liver_sQTL(0.045), intron_19_12948358_12948514|Liver_sQTL(0.045), intron_19_12948358_12948497|Liver_sQTL(0.045), intron_19_12948552_12948686|Spleen_sQTL(0.076), intron_19_12948358_12948514|Spleen_sQTL(0.076), intron_19_12948358_12948497|Spleen_sQTL(0.076), intron_19_12948552_12948686|Brain_Frontal_Cortex_BA9_sQTL(0.088), intron_19_12948358_12948514|Brain_Frontal_Cortex_BA9_sQTL(0.088), intron_19_12948358_12948497|Brain_Frontal_Cortex_BA9_sQTL(0.088), intron_19_12948552_12948686|Artery_Tibial_sQTL(0.065), intron_19_12948358_12948514|Artery_Tibial_sQTL(0.065), intron_19_12948358_12948497|Artery_Tibial_sQTL(0.065)","ENSG00000179262.9|Whole_Blood_eQTL(0.013)","ENSG00000179588.8|Whole_Blood_eQTL(0.514)","ENSG00000179588|Artery_Tibial_stQTL(0.61)","ENSG00000152556.15|Thyroid_eQTL(0.359), ENSG00000152556.15|Muscle_Skeletal_eQTL(0.241), ENSG00000152556.15|Esophagus_Gastroesophageal_Junction_eQTL(0.389)","intron_12_48119406_48122767|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.65), intron_12_48119406_48122767|Esophagus_Gastroesophageal_Junction_sQTL(0.061)","ENSG00000112715.21|Muscle_Skeletal_eQTL(0.871), intron_6_43771312_43774341|Muscle_Skeletal_sQTL(0.017)","intron_6_43780803_43781956|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.198), intron_6_43780803_43781956|Thyroid_sQTL(0.019), intron_6_43778918_43781956|Thyroid_sQTL(0.731), intron_6_43780803_43781956|Esophagus_Gastroesophageal_Junction_sQTL(0.02)","ENSG00000147454.grp_1.downstream.ENST00000523930|Muscle_Skeletal_stQTL(0.979)","ENSG00000147454.13|Muscle_Skeletal_eQTL(0.164), intron_8_23529212_23566108|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.004), intron_8_23529212_23543088|Thyroid_sQTL(0.075), intron_8_23529212_23566108|Thyroid_sQTL(0.019), intron_8_23529212_23566108|Muscle_Skeletal_sQTL(0.015), ENSG00000147454|Whole_Blood_stQTL(0.103)","ENSG00000141293.grp_1.downstream.ENST00000336915|Esophagus_Gastroesophageal_Junction_stQTL(0.037), ENSG00000141293.grp_2.downstream.ENST00000336915|Esophagus_Gastroesophageal_Junction_stQTL(0.63), ENSG00000141293.grp_2.downstream.ENST00000584924|Esophagus_Gastroesophageal_Junction_stQTL(0.309)","ENSG00000141293.15|Skin_Not_Sun_Exposed_Suprapubic_eQTL(0.175)","ENSG00000143643.12|Thyroid_eQTL(0.371), ENSG00000143643.12|Whole_Blood_eQTL(0.569)","intron_1_230936264_230939386|Thyroid_sQTL(0.56)","intron_11_65583765_65584241|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.038), intron_11_65581965_65584241|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.035), intron_11_65583765_65584241|Thyroid_sQTL(0.779), intron_11_65581965_65584241|Thyroid_sQTL(0.094), intron_11_65583765_65584241|Esophagus_Gastroesophageal_Junction_sQTL(0.022)","ENSG00000173442.12|Esophagus_Gastroesophageal_Junction_eQTL(0.115)","ENSG00000113742.12|Thyroid_eQTL(0.751)","ENSG00000113742.12|Whole_Blood_eQTL(0.845)","intron_22_32487828_32491086|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.07), intron_22_32487828_32491086|Muscle_Skeletal_sQTL(0.238)","ENSG00000100225.17|Whole_Blood_eQTL(0.008), intron_22_32475398_32478981|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.075), intron_22_32475124_32478981|Skin_Not_Sun_Exposed_Suprapubic_sQTL(0.017), intron_22_32493281_32498144|Thyroid_sQTL(0.205), intron_22_32475398_32479133|Thyroid_sQTL(0.165), intron_22_32475398_32478981|Thyroid_sQTL(0.165), intron_22_32479275_32483897|Thyroid_sQTL(0.008), intron_22_32475398_32479133|Muscle_Skeletal_sQTL(0.254)","ENSG00000221914|Esophagus_Gastroesophageal_Junction_stQTL(0.468)","intron_8_26293740_26338890|Thyroid_sQTL(0.384), intron_8_26293740_26338893|Thyroid_sQTL(0.384), intron_8_26293740_26338890|Esophagus_Gastroesophageal_Junction_sQTL(0.009), intron_8_26293740_26338893|Esophagus_Gastroesophageal_Junction_sQTL(0.009)","intron_6_31642403_31642829|Brain_Cerebellar_Hemisphere_sQTL(0.967)","intron_6_31643977_31644082|Artery_Tibial_sQTL(0.997)","ENSG00000011021.21|Brain_Cortex_eQTL(0.064), intron_1_11819554_11820362|Artery_Tibial_sQTL(0.029), intron_1_11819554_11822695|Cells_Cultured_fibroblasts_sQTL(0.175), intron_1_11819554_11820362|Cells_Cultured_fibroblasts_sQTL(0.175), intron_1_11822801_11823707|Brain_Cerebellar_Hemisphere_sQTL(0.522)","intron_1_11838660_11840143|Cells_Cultured_fibroblasts_sQTL(0.187), intron_1_11827221_11828106|Cells_Cultured_fibroblasts_sQTL(0.205), intron_1_11838660_11840143|Brain_Cerebellar_Hemisphere_sQTL(0.191), intron_1_11827221_11828106|Brain_Cerebellar_Hemisphere_sQTL(0.227)","intron_1_26788394_26790759|Cells_Cultured_fibroblasts_sQTL(0.589)","intron_1_26788394_26790759|Cells_Cultured_fibroblasts_sQTL(0.762)",null,"intron_17_45439634_45440163|Cells_Cultured_fibroblasts_sQTL(0.989)","intron_17_45458439_45475100|Pancreas_sQTL(0.241), intron_17_45458439_45468209|Artery_Aorta_sQTL(0.374), intron_17_45458439_45475100|Artery_Aorta_sQTL(0.374)",null,"ENSG00000075884.13|Whole_Blood_eQTL(1)","intron_2_143129466_143155477|Whole_Blood_sQTL(0.737)","intron_2_143435699_143436913|Esophagus_Muscularis_sQTL(0.112), intron_2_143435289_143435601|Esophagus_Muscularis_sQTL(0.096), intron_2_143437042_143487373|Esophagus_Muscularis_sQTL(0.114), intron_2_143556485_143624133|Nerve_Tibial_sQTL(0.079), intron_2_143435289_143435601|Nerve_Tibial_sQTL(0.106)","ENSG00000140853.15|Whole_Blood_eQTL(0.991)","ENSG00000140853.15|Esophagus_Muscularis_eQTL(1)","ENSG00000113742.12|Whole_Blood_eQTL(0.994)","ENSG00000113742.12|Pancreas_eQTL(0.507)","ENSG00000147852|Whole_Blood_stQTL(0.949)","ENSG00000147852.15|Whole_Blood_eQTL(0.884)","ENSG00000155849.15|Whole_Blood_eQTL(0.641)","intron_7_37429756_37448675|Whole_Blood_sQTL(0.208), intron_7_37429756_37448675|Nerve_Tibial_sQTL(0.726)","ENSG00000107562.grp_1.downstream.ENST00000374429|Pancreas_stQTL(0.138), ENSG00000107562.grp_2.downstream.ENST00000374429|Pancreas_stQTL(0.037)","intron_10_44376032_44378637|Esophagus_Muscularis_sQTL(0.309), intron_10_44373343_44378637|Esophagus_Muscularis_sQTL(0.309), intron_10_44376032_44378637|Nerve_Tibial_sQTL(0.341)","ENSG00000163050.16|Whole_Blood_eQTL(0.946)","ENSG00000163050.16|Nerve_Tibial_eQTL(0.025), intron_1_226899555_226903502|Esophagus_Muscularis_sQTL(0.023), intron_1_226899555_226903502|Nerve_Tibial_sQTL(0.057)","ENSG00000022567.9|Whole_Blood_eQTL(0.854)","ENSG00000022567.grp_1.downstream.ENST00000519067|Pancreas_stQTL(0.18)","intron_3_132467313_132473145|Cells_Cultured_fibroblasts_sQTL(0.06), intron_3_132467313_132472557|Esophagus_Muscularis_sQTL(0.046), intron_3_132467313_132473145|Esophagus_Muscularis_sQTL(0.046), intron_3_132467313_132472557|Nerve_Tibial_sQTL(0.05), intron_3_132467313_132473145|Nerve_Tibial_sQTL(0.05)","intron_3_132483577_132484588|Cells_Cultured_fibroblasts_sQTL(0.308), intron_3_132483577_132484588|Esophagus_Muscularis_sQTL(0.234), ENSG00000138246|Cells_Cultured_fibroblasts_stQTL(0.155), ENSG00000138246|Esophagus_Muscularis_stQTL(0.073)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pip_per_cs_alltraits$trait<\/th>\n      <th>gene_name<\/th>\n      <th>CS<\/th>\n      <th>total_PIP_CS<\/th>\n      <th>num_molecular_traits_CS<\/th>\n      <th>ids_pip_CS<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="reduce-fp-findings" class="section level1">
<h1>Reduce FP findings</h1>
<div id="silver-standard-for-ldl" class="section level2">
<h2>Silver standard for LDL</h2>
<p><a
href="https://sq-96.github.io/multigroup_ctwas_analysis/LDL_silver_standard.html"
class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/LDL_silver_standard.html</a></p>
</div>
</div>
<div id="enrichment-analysis-fractional-model" class="section level1">
<h1>Enrichment analysis – fractional model</h1>
<p>Methods for enrichment analsis can be found here: <a
href="https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_enrichment_genesymbol.html#Fractional_model"
class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_enrichment_genesymbol.html#Fractional_model</a></p>
<div id="ldl-ukb-d-30780_irnt" class="section level2">
<h2>LDL-ukb-d-30780_irnt</h2>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
db &lt;- &quot;GO_Biological_Process_2023&quot;

enrich_multi &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/13.post_processing_0103/results_downstream/enrich_fractional/enrichment_fractional_calibrated_blgeneset_summary_multigroup_&quot;, trait, &quot;_&quot;, db, &quot;.RDS&quot;))
enrich_single &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/13.post_processing_0103/results_downstream/enrich_fractional/enrichment_fractional_calibrated_blgeneset_summary_singlegroup_&quot;, trait, &quot;_&quot;, db, &quot;.RDS&quot;))

print(&quot;FDR_adjust &lt; 0.05&quot;)</code></pre>
<pre><code>[1] &quot;FDR_adjust &lt; 0.05&quot;</code></pre>
<pre class="r"><code>enrich_multi_sig &lt;- enrich_multi[enrich_multi$fdr_calibrated &lt; 0.05,]
enrich_single_sig &lt;- enrich_multi[enrich_single$fdr_calibrated &lt; 0.05,]

venn.plot &lt;- draw.pairwise.venn(
     area1 = nrow(enrich_multi_sig),          # Size of Group A
     area2 = nrow(enrich_single_sig),          # Size of Group B
     cross.area = sum(enrich_multi_sig$GO %in% enrich_single_sig$GO),     # Overlap between Group A and Group B
     category = c(&quot;Multigroup&quot;, &quot;Singlegroup&quot;),  # Labels for the groups
     fill = c(&quot;red&quot;, &quot;blue&quot;),             # Colors for the groups
     lty = &quot;blank&quot;,                       # Line type for the circles
     cex = 2,                             # Font size for the numbers
     cat.cex = 2                          # Font size for the labels
   )</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-9-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>enrich_multi_unique &lt;- enrich_multi_sig[!enrich_multi_sig$GO %in% enrich_single_sig$GO,]

DT::datatable(enrich_multi_unique,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Enrichment results -- unique GO terms found by multi-group analysis&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Enrichment results -- unique GO terms found by multi-group analysis<\/caption>","data":[["16018","839","11011","5220","1712","1841","8713","3619","2712","15217","15618","10619","869","3719","14519","2431","2191"],[2.25234279012133e-12,1.44691765385768e-08,4.92196033582439e-09,1.00237085599952e-08,1.54235294860089e-08,4.61489516511912e-07,2.80379867573521e-08,2.99744346403665e-08,1.5366763829185e-06,8.54601023578103e-08,5.05781412346534e-08,4.03086486200594e-08,9.75544633645496e-07,8.43723725569794e-08,2.4687126696864e-07,6.36464236862261e-08,2.38741756694588e-06],[3.95350135891477e-09,8.33950239308499e-06,3.80186279082893e-06,6.77477402298678e-06,8.33950239308499e-06,0.00011129282432248,1.35059806750385e-05,1.35059806750385e-05,0.000307733674164457,2.56712651915934e-05,1.82317339770514e-05,1.55677759349044e-05,0.000210990793364848,2.56712651915934e-05,6.67416470249718e-05,2.1508513304464e-05,0.000431797145715168],[9.99990000099999e-06,1.9999800002e-05,2.9999700003e-05,2.9999700003e-05,3.9999600004e-05,4.99995000049999e-05,5.99994000059999e-05,5.99994000059999e-05,6.99993000069999e-05,6.99993000069999e-05,8.99991000089999e-05,0.000119998800012,0.000169998300017,0.000169998300017,0.000179998200018,0.000189998100019,0.000209997900021],[0.00772420847220099,0.0135173648263517,0.0162208377916221,0.0162208377916221,0.0196616215656025,0.0225289413772529,0.023172625416603,0.023172625416603,0.0236553884461155,0.0236553884461155,0.0286250078675684,0.0360463062036046,0.0459590404095959,0.0459590404095959,0.046345250833206,0.046696351218306,0.0493677671918933],["Sterol Transport (GO:0015918)","Peptidyl-Serine Phosphorylation (GO:0018105)","Negative Regulation Of Lipoprotein Particle Clearance (GO:0010985)","Secondary Alcohol Biosynthetic Process (GO:1902653)","Cholesterol Biosynthetic Process (GO:0006695)","Cellular Response To Insulin Stimulus (GO:0032869)","Protein poly-ADP-ribosylation (GO:0070212)","Triglyceride Metabolic Process (GO:0006641)","Endocytosis (GO:0006897)","Regulation Of Low-Density Lipoprotein Particle Clearance (GO:0010988)","Sterol Biosynthetic Process (GO:0016126)","Very-Low-Density Lipoprotein Particle Remodeling (GO:0034372)","Peptidyl-Threonine Phosphorylation (GO:0018107)","Triglyceride-Rich Lipoprotein Particle Remodeling (GO:0034370)","Xenobiotic Transport (GO:0042908)","Cellular Response To Sterol Depletion (GO:0071501)","Cellular Response To Nutrient Levels (GO:0031669)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pvalue_origin<\/th>\n      <th>fdr_origin<\/th>\n      <th>pvalue_calibrated<\/th>\n      <th>fdr_calibrated<\/th>\n      <th>GO<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># enrich_single_unique &lt;- enrich_single_sig[!enrich_single_sig$GO %in% enrich_multi_sig$GO,]
# DT::datatable(enrich_single_unique,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Enrichment results -- unique GO terms found by single-group analysis, FDR &lt; 0.05&#39;),options = list(pageLength = 10) )</code></pre>
<pre class="r"><code>print(&quot;FDR_adjust &lt; 0.1&quot;)</code></pre>
<pre><code>[1] &quot;FDR_adjust &lt; 0.1&quot;</code></pre>
<pre class="r"><code>enrich_multi_sig &lt;- enrich_multi[enrich_multi$fdr_calibrated &lt; 0.1,]
enrich_single_sig &lt;- enrich_multi[enrich_single$fdr_calibrated &lt; 0.1,]

venn.plot &lt;- draw.pairwise.venn(
     area1 = nrow(enrich_multi_sig),          # Size of Group A
     area2 = nrow(enrich_single_sig),          # Size of Group B
     cross.area = sum(enrich_multi_sig$GO %in% enrich_single_sig$GO),     # Overlap between Group A and Group B
     category = c(&quot;Multigroup&quot;, &quot;Singlegroup&quot;),  # Labels for the groups
     fill = c(&quot;red&quot;, &quot;blue&quot;),             # Colors for the groups
     lty = &quot;blank&quot;,                       # Line type for the circles
     cex = 2,                             # Font size for the numbers
     cat.cex = 2                          # Font size for the labels
   )</code></pre>
<p><img src="figure/multi_group_downstream_analysis.Rmd/unnamed-chunk-10-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>enrich_multi_unique &lt;- enrich_multi_sig[!enrich_multi_sig$GO %in% enrich_single_sig$GO,]

DT::datatable(enrich_multi_unique,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Enrichment results -- unique GO terms found by multi-group analysis&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Enrichment results -- unique GO terms found by multi-group analysis<\/caption>","data":[["11011","5220","1712","1841","8713","3619","2712","15217","15618","10619","869","3719","14519","2431","2191","1916","20911","859","13018","1934","684","17716","1179","12918","14113","14912","545","1189","11415","3112","1435","6519","1234"],[4.92196033582439e-09,1.00237085599952e-08,1.54235294860089e-08,4.61489516511912e-07,2.80379867573521e-08,2.99744346403665e-08,1.5366763829185e-06,8.54601023578103e-08,5.05781412346534e-08,4.03086486200594e-08,9.75544633645496e-07,8.43723725569794e-08,2.4687126696864e-07,6.36464236862261e-08,2.38741756694588e-06,1.27592289792554e-07,4.03038195705471e-07,3.95243169507654e-06,4.71439464451713e-06,4.73411311155361e-07,1.12401507868779e-06,3.6717204138305e-06,1.87610315676074e-06,7.12366993164551e-07,9.65863539023086e-06,3.00928694446181e-05,1.59236128044373e-05,5.73817898601525e-06,1.42769849759714e-05,2.39576740733402e-06,2.52554699765235e-05,2.21010990895691e-05,3.24369535942743e-05],[3.80186279082893e-06,6.77477402298678e-06,8.33950239308499e-06,0.00011129282432248,1.35059806750385e-05,1.35059806750385e-05,0.000307733674164457,2.56712651915934e-05,1.82317339770514e-05,1.55677759349044e-05,0.000210990793364848,2.56712651915934e-05,6.67416470249718e-05,2.1508513304464e-05,0.000431797145715168,3.63100795214915e-05,0.000103772739246642,0.000647599944705419,0.000749727407144238,0.00011129282432248,0.000233751905017881,0.000620406008674422,0.000362288920307333,0.000160490347168364,0.00137432214618364,0.00347413152731469,0.00209997498618518,0.000886466679353841,0.00197937584013019,0.000431797145715168,0.00310355286734233,0.00284525339945953,0.00357931853233145],[2.9999700003e-05,2.9999700003e-05,3.9999600004e-05,4.99995000049999e-05,5.99994000059999e-05,5.99994000059999e-05,6.99993000069999e-05,6.99993000069999e-05,8.99991000089999e-05,0.000119998800012,0.000169998300017,0.000169998300017,0.000179998200018,0.000189998100019,0.000209997900021,0.000239997600024,0.000249997500025,0.000279997200028,0.000279997200028,0.000289997100029,0.000309996900031,0.000319996800032,0.000369996300037,0.000559994400055999,0.000569994300056999,0.000569994300056999,0.000589994100058999,0.000619993800061999,0.000649993500064999,0.000669993300066999,0.000709992900070999,0.000719992800071999,0.000749992500074999],[0.0162208377916221,0.0162208377916221,0.0196616215656025,0.0225289413772529,0.023172625416603,0.023172625416603,0.0236553884461155,0.0236553884461155,0.0286250078675684,0.0360463062036046,0.0459590404095959,0.0459590404095959,0.046345250833206,0.046696351218306,0.0493677671918933,0.0540694593054069,0.054069459305407,0.0560005114234572,0.0560005114234572,0.0560005114234572,0.0576740899257674,0.0576740899257674,0.0645345159451631,0.0906458582472999,0.0906458582472999,0.0906458582472999,0.0911456599719717,0.093119624359312,0.0949868879689581,0.0953329940384807,0.0973250267497325,0.0973250267497325,0.0989075475098908],["Negative Regulation Of Lipoprotein Particle Clearance (GO:0010985)","Secondary Alcohol Biosynthetic Process (GO:1902653)","Cholesterol Biosynthetic Process (GO:0006695)","Cellular Response To Insulin Stimulus (GO:0032869)","Protein poly-ADP-ribosylation (GO:0070212)","Triglyceride Metabolic Process (GO:0006641)","Endocytosis (GO:0006897)","Regulation Of Low-Density Lipoprotein Particle Clearance (GO:0010988)","Sterol Biosynthetic Process (GO:0016126)","Very-Low-Density Lipoprotein Particle Remodeling (GO:0034372)","Peptidyl-Threonine Phosphorylation (GO:0018107)","Triglyceride-Rich Lipoprotein Particle Remodeling (GO:0034370)","Xenobiotic Transport (GO:0042908)","Cellular Response To Sterol Depletion (GO:0071501)","Cellular Response To Nutrient Levels (GO:0031669)","Intestinal Cholesterol Absorption (GO:0030299)","Positive Regulation Of Protein Export From Nucleus (GO:0046827)","Peptidyl-Threonine Modification (GO:0018210)","Sphingolipid Biosynthetic Process (GO:0030148)","Intestinal Lipid Absorption (GO:0098856)","High-Density Lipoprotein Particle Remodeling (GO:0034375)","Regulation Of Receptor Recycling (GO:0001919)","Phosphatidic Acid Biosynthetic Process (GO:0006654)","Sphingoid Metabolic Process (GO:0046519)","Phospholipid Biosynthetic Process (GO:0008654)","Positive Regulation Of Wnt Signaling Pathway (GO:0030177)","Long-Term Memory (GO:0007616)","Phosphatidic Acid Metabolic Process (GO:0046473)","Regulation Of Cholesterol Transport (GO:0032374)","Negative Regulation Of Signaling (GO:0023057)","Membrane Lipid Biosynthetic Process (GO:0046467)","Unsaturated Fatty Acid Metabolic Process (GO:0033559)","Icosanoid Metabolic Process (GO:0006690)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pvalue_origin<\/th>\n      <th>fdr_origin<\/th>\n      <th>pvalue_calibrated<\/th>\n      <th>fdr_calibrated<\/th>\n      <th>GO<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># enrich_single_unique &lt;- enrich_single_sig[!enrich_single_sig$GO %in% enrich_multi_sig$GO,]
# DT::datatable(enrich_single_unique,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Enrichment results -- unique GO terms found by single-group analysis, FDR &lt; 0.1&#39;),options = list(pageLength = 10) )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] VennDiagram_1.7.3   futile.logger_1.4.3 pheatmap_1.0.12    
[4] gridExtra_2.3       tidyr_1.3.0         dplyr_1.1.4        
[7] ggplot2_3.5.1       ctwas_0.4.20.9001  

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] GenomicRanges_1.48.0        fs_1.5.2                   
  [9] rstudioapi_0.13             farver_2.1.0               
 [11] DT_0.22                     ggrepel_0.9.1              
 [13] bit64_4.0.5                 AnnotationDbi_1.58.0       
 [15] fansi_1.0.3                 xml2_1.3.3                 
 [17] codetools_0.2-18            logging_0.10-108           
 [19] cachem_1.0.6                knitr_1.39                 
 [21] jsonlite_1.8.0              workflowr_1.7.0            
 [23] Rsamtools_2.12.0            dbplyr_2.1.1               
 [25] png_0.1-7                   readr_2.1.2                
 [27] compiler_4.2.0              httr_1.4.3                 
 [29] assertthat_0.2.1            Matrix_1.5-3               
 [31] fastmap_1.1.0               lazyeval_0.2.2             
 [33] cli_3.6.1                   formatR_1.12               
 [35] later_1.3.0                 htmltools_0.5.2            
 [37] prettyunits_1.1.1           tools_4.2.0                
 [39] gtable_0.3.0                glue_1.6.2                 
 [41] GenomeInfoDbData_1.2.8      rappdirs_0.3.3             
 [43] Rcpp_1.0.12                 Biobase_2.56.0             
 [45] jquerylib_0.1.4             vctrs_0.6.5                
 [47] Biostrings_2.64.0           rtracklayer_1.56.0         
 [49] crosstalk_1.2.0             xfun_0.41                  
 [51] stringr_1.5.1               lifecycle_1.0.4            
 [53] irlba_2.3.5                 restfulr_0.0.14            
 [55] ensembldb_2.20.2            XML_3.99-0.14              
 [57] zlibbioc_1.42.0             zoo_1.8-10                 
 [59] scales_1.3.0                gggrid_0.2-0               
 [61] hms_1.1.1                   promises_1.2.0.1           
 [63] MatrixGenerics_1.8.0        ProtGenerics_1.28.0        
 [65] parallel_4.2.0              SummarizedExperiment_1.26.1
 [67] lambda.r_1.2.4              RColorBrewer_1.1-3         
 [69] AnnotationFilter_1.20.0     LDlinkR_1.2.3              
 [71] yaml_2.3.5                  curl_4.3.2                 
 [73] memoise_2.0.1               sass_0.4.1                 
 [75] biomaRt_2.54.1              stringi_1.7.6              
 [77] RSQLite_2.3.1               highr_0.9                  
 [79] S4Vectors_0.34.0            BiocIO_1.6.0               
 [81] GenomicFeatures_1.48.3      BiocGenerics_0.42.0        
 [83] filelock_1.0.2              BiocParallel_1.30.3        
 [85] GenomeInfoDb_1.39.9         rlang_1.1.2                
 [87] pkgconfig_2.0.3             matrixStats_0.62.0         
 [89] bitops_1.0-7                evaluate_0.15              
 [91] lattice_0.20-45             purrr_1.0.2                
 [93] labeling_0.4.2              GenomicAlignments_1.32.0   
 [95] htmlwidgets_1.5.4           cowplot_1.1.1              
 [97] bit_4.0.4                   tidyselect_1.2.0           
 [99] magrittr_2.0.3              R6_2.5.1                   
[101] IRanges_2.30.0              generics_0.1.2             
[103] DelayedArray_0.22.0         DBI_1.2.2                  
[105] pgenlibr_0.3.3              pillar_1.9.0               
[107] withr_2.5.0                 KEGGREST_1.36.3            
[109] RCurl_1.98-1.7              mixsqp_0.3-43              
[111] tibble_3.2.1                crayon_1.5.1               
[113] futile.options_1.0.1        utf8_1.2.2                 
[115] BiocFileCache_2.4.0         plotly_4.10.0              
[117] tzdb_0.4.0                  rmarkdown_2.25             
[119] progress_1.2.2              data.table_1.14.2          
[121] blob_1.2.3                  git2r_0.30.1               
[123] digest_0.6.29               httpuv_1.6.5               
[125] stats4_4.2.0                munsell_0.5.0              
[127] viridisLite_0.4.0           bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
