<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-10-13" />

<title>Comparing results from predictdb eQTL + sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / predictdb eQTL + sQTL + Munro stQTL for 4 traits</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparing results from predictdb eQTL +
sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / predictdb eQTL +
sQTL + Munro stQTL for 4 traits</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-10-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree04a6e93bd12a1de5778c701cb6d4f946805ea3cdtargetblank04a6e93a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/04a6e93bd12a1de5778c701cb6d4f946805ea3cd" target="_blank">04a6e93</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree04a6e93bd12a1de5778c701cb6d4f946805ea3cdtargetblank04a6e93a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/04a6e93bd12a1de5778c701cb6d4f946805ea3cd" target="_blank">04a6e93</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/fractional_enrichment_LDL_es.Rmd
    Untracked:  analysis/multi_group_compare_decidingweights_4traits_ess.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="settings" class="section level1">
<h1>Settings</h1>
<div id="modalities-from-munro" class="section level2">
<h2>6 modalities from Munro</h2>
<p>The stQTL was a combination of Munro apa + rs QTL, if a gene has both
rs-QTL and APA-QTL, we use rs-QTL.</p>
<ol style="list-style-type: decimal">
<li>Weight processing:</li>
</ol>
<p>PredictDB:</p>
<p>all the PredictDB are converted from FUSION weights</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = F (FUSION converted weights)</li>
<li>load_predictdb_LD = F,</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Parameter estimation and fine-mapping</li>
</ol>
<ul>
<li>niter_prefit = 5,</li>
<li>niter = 30(default),</li>
<li>L: determined by uniform susie,</li>
<li>group_prior_var_structure = “shared_type”,</li>
<li>maxSNP = 20000,</li>
<li>min_nonSNP_PIP = 0.5,</li>
</ul>
</div>
<div id="weights-from-predictdb" class="section level2">
<h2>weights from predictdb</h2>
<ol style="list-style-type: decimal">
<li>Weight processing:</li>
</ol>
<p>PredictDB (eqtl, sqtl)</p>
<ul>
<li>drop_strand_ambig = TRUE,</li>
<li>scale_by_ld_variance = T</li>
<li>load_predictdb_LD = F,</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Parameter estimation and fine-mapping</li>
</ol>
<ul>
<li>group_prior_var_structure = “shared_type”,</li>
<li>filter_L = TRUE,</li>
<li>filter_nonSNP_PIP = FALSE,</li>
<li>min_nonSNP_PIP = 0.5,</li>
<li>min_abs_corr = 0.1,</li>
</ul>
<p>mem: 100g 10cores</p>
<pre class="r"><code>library(ctwas)
library(EnsDb.Hsapiens.v86)
library(VennDiagram)
library(ggplot2)
library(gridExtra)
library(pheatmap)
library(dplyr)

ens_db &lt;- EnsDb.Hsapiens.v86

mapping_predictdb &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS&quot;)
mapping_munro &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS&quot;)
mapping_two &lt;- rbind(mapping_predictdb,mapping_munro)

load(&quot;/project2/xinhe/shared_data/multigroup_ctwas/gwas/samplesize.rdata&quot;)


colors &lt;- c(&quot;#1b9e77&quot;, &quot;#d95f02&quot;, &quot;#7570b3&quot;, &quot;#e7298a&quot;, &quot;#66a61e&quot;, 
               &quot;#e6ab02&quot;, &quot;#a6761d&quot;, &quot;#666666&quot;, &quot;#a6cee3&quot;)

source(&quot;/project/xinhe/xsun/r_functions/ctwas_combine_pip_nocs.R&quot;)

plot_piechart &lt;- function(ctwas_parameters, colors) {
  data &lt;- data.frame(
    category = names(ctwas_parameters$prop_heritability),
    percentage = ctwas_parameters$prop_heritability
  )
  
  # Calculate percentage labels for the chart
  data$percentage_label &lt;- paste0(round(data$percentage * 100, 1), &quot;%&quot;)
  
  pie &lt;- ggplot(data, aes(x = &quot;&quot;, y = percentage, fill = category)) +
    geom_bar(stat = &quot;identity&quot;, width = 1) +
    coord_polar(&quot;y&quot;, start = 0) +
    theme_void() +  # Remove background and axes
    geom_text(aes(label = percentage_label), 
              position = position_stack(vjust = 0.5), size = 2) +
    scale_fill_manual(values = colors) +  # Custom colors
    labs(fill = &quot;Category&quot;) +  # Legend title
    ggtitle(&quot;Percent of heritability&quot;)  # Title
  
  return(pie)
}

rename_heatmap_columns &lt;- function(heatmap_data, column_order) {
  # Select columns that are in the column_order and exist in heatmap_data
  selected_columns &lt;- intersect(column_order, colnames(heatmap_data))
  heatmap_data &lt;- heatmap_data[, selected_columns]
  
  # Initialize new_column_names as a copy of selected_columns
  new_column_names &lt;- selected_columns

  # Assign the new column names to heatmap_data
  colnames(heatmap_data) &lt;- new_column_names
  
  return(heatmap_data)
}

plot_heatmap &lt;- function(heatmap_data, main) {
  
  rownames(heatmap_data) &lt;- heatmap_data$gene_name
  heatmap_data &lt;- heatmap_data %&gt;% dplyr::select(-gene_name, -combined_pip)
  
  if(nrow(heatmap_data) ==1){
    
    heatmap_data &lt;- rbind(heatmap_data,rep(0,ncol(heatmap_data)))
    rownames(heatmap_data)[2] &lt;- &quot;fake_gene_for_plotting&quot;
    
  }
  
  heatmap_matrix &lt;- as.matrix(heatmap_data)
  
  p &lt;- pheatmap(heatmap_matrix,
         cluster_rows = F,   # Cluster the rows (genes)
         cluster_cols = F,   # Cluster the columns (QTL types)
         color = colorRampPalette(c(&quot;white&quot;, &quot;red&quot;))(50), # Color gradient
         display_numbers = TRUE, # Display numbers in cells
         main = main,labels_row = rownames(heatmap_data), silent = T)
  
  return(p)
}


plot_3venn &lt;- function(es, esra, ess) {
  
  venn.plot &lt;- draw.triple.venn(
      area1 = length(es),
      area2 = length(esra),
      area3 = length(ess),
      n12 = length(intersect(es, esra)),
      n23 = length(intersect(esra, ess)),
      n13 = length(intersect(es, ess)),
      n123 = length(Reduce(intersect, list(es, esra, ess))),
      category = c(&quot;e+s&quot;, &quot;e+s+rs+apa&quot;, &quot;e+e+st&quot;),
      fill = c(&quot;red&quot;, &quot;green&quot;, &quot;blue&quot;),
      lty = &quot;dashed&quot;,
      cex = 2,
      cat.cex = 1.5,
      cat.col = c(&quot;red&quot;, &quot;green&quot;, &quot;blue&quot;),
      scaled = T
      )
  
  return(venn.plot)
  
}</code></pre>
</div>
</div>
<div id="ldl-ukb-d-30780_irnt" class="section level1">
<h1>LDL-ukb-d-30780_irnt</h1>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;
tissue &lt;- &quot;Liver&quot;

gwas_n &lt;- samplesize[trait]

results_dir_ess &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/ess/&quot;)

snp_map_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.ctwas.res.RDS&quot;))

param_ess &lt;- ctwas_res_ess$param
finemap_res_ess &lt;- ctwas_res_ess$finemap_res


p_conv_ess &lt;- make_convergence_plots(param_ess, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_ess &lt;- summarize_param(param_ess, gwas_n)
pve_pie_ess &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_ess, colors = colors)

susie_alpha_res_ess &lt;- ctwas_res_ess$susie_alpha_res

susie_alpha_res_ess &lt;- anno_susie_alpha_res(susie_alpha_res_ess,
                                        mapping_table = mapping_two,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-13 23:17:04 INFO::Annotating susie alpha result ...
2024-10-13 23:17:04 INFO::Map molecular traits to genes
2024-10-13 23:17:05 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type_ess &lt;- combine_gene_pips(susie_alpha_res_ess, 
                                             group_by = &quot;gene_name&quot;,
                                             by = &quot;type&quot;,
                                             method = &quot;combine_cs&quot;,
                                             filter_cs = TRUE,
                                             include_cs_id = F)</code></pre>
<pre class="r"><code>results_dir_espred &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/espred/&quot;)

snp_map_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.ctwas.res.RDS&quot;))

param_espred &lt;- ctwas_res_espred$param
finemap_res_espred &lt;- ctwas_res_espred$finemap_res


p_conv_espred &lt;- make_convergence_plots(param_espred, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_espred &lt;- summarize_param(param_espred, gwas_n)
pve_pie_espred &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_espred, colors = colors)


finemap_res_espred$molecular_id &lt;- get_molecular_ids(finemap_res_espred)
finemap_res_espred &lt;- anno_finemap_res(finemap_res_espred,
                              snp_map = snp_map_espred,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:17:26 INFO::Annotating fine-mapping result ...
2024-10-13 23:17:29 INFO::Map molecular traits to genes
2024-10-13 23:17:30 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:17:35 INFO::Add gene positions
2024-10-13 23:17:35 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_espred &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_espred,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:17:47 INFO::Limit gene results to credible sets</code></pre>
<pre class="r"><code>results_dir_4W &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/4W/&quot;)

snp_map_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.ctwas.res.RDS&quot;))

param_4W &lt;- ctwas_res_4W$param
finemap_res_4W &lt;- ctwas_res_4W$finemap_res

p_conv_4W &lt;- make_convergence_plots(param_4W, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_4W &lt;- summarize_param(param_4W, gwas_n)
pve_pie_4W &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_4W, colors = colors)

finemap_res_4W$molecular_id &lt;- get_molecular_ids(finemap_res_4W)
finemap_res_4W &lt;- anno_finemap_res(finemap_res_4W,
                              snp_map = snp_map_4W,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:18:03 INFO::Annotating fine-mapping result ...
2024-10-13 23:18:03 INFO::Map molecular traits to genes
2024-10-13 23:18:04 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:18:09 INFO::Add gene positions
2024-10-13 23:18:09 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_4W &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_4W,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:18:17 INFO::Limit gene results to credible sets</code></pre>
<div id="parameters" class="section level2">
<h2>Parameters</h2>
<pre class="r"><code>print(&quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(p_conv_espred,p_conv_4W,p_conv_ess, ncol = 3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>######pve
group_pve_espred &lt;- ctwas_parameters_espred$group_pve
group_pve_espred &lt;- group_pve_espred[-length(group_pve_espred)]
group_pve_espred &lt;- c(group_pve_espred, rep(NA,3))
group_pve_espred &lt;- c(group_pve_espred, ctwas_parameters_espred$total_pve)
names(group_pve_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_4W &lt;- ctwas_parameters_4W$group_pve
group_pve_4W &lt;- group_pve_4W[-length(group_pve_4W)]
group_pve_4W &lt;- c(group_pve_4W, rep(NA,1))
group_pve_4W &lt;- c(group_pve_4W, ctwas_parameters_4W$total_pve)
names(group_pve_4W) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_ess &lt;- ctwas_parameters_ess$group_pve
group_pve_ess &lt;- group_pve_ess[-length(group_pve_ess)]
group_pve_ess &lt;- c(group_pve_ess[1:2], rep(NA,2),group_pve_ess[3])
group_pve_ess &lt;- c(group_pve_ess, ctwas_parameters_ess$total_pve)
names(group_pve_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

grouppve &lt;- cbind(group_pve_espred,group_pve_4W,group_pve_ess)
grouppve &lt;- round(grouppve,digits = 4)

######size
group_size_espred &lt;- ctwas_parameters_espred$group_size
group_size_espred &lt;- group_size_espred[-length(group_size_espred)]
group_size_espred &lt;- c(group_size_espred, rep(NA,3))
names(group_size_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_4W &lt;- ctwas_parameters_4W$group_size
group_size_4W &lt;- group_size_4W[-length(group_size_4W)]
group_size_4W &lt;- c(group_size_4W, rep(NA,1))
names(group_size_4W) &lt;-  c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_ess &lt;- ctwas_parameters_ess$group_size
group_size_ess &lt;- group_size_ess[-length(group_size_ess)]
group_size_ess &lt;- c(group_size_ess[1:2], rep(NA,2),group_size_ess[3])
names(group_size_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

groupsize &lt;- cbind(group_size_espred,group_size_4W,group_size_ess)

group_info &lt;- cbind(grouppve,rbind(groupsize,c(rep(NA,3))))

DT::datatable(group_info,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group PVE and Group Size&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group PVE and Group Size<\/caption>","data":[["eQTL_pred","sQTL_pred","apaQTL_munro","rsQTL_munro","stQTL_munro","TOTAL"],[0.0051,0.0115,null,null,null,0.0519],[0.006,0.0102,0.001,0.0022,null,0.0554],[0.0049,0.0114,null,null,0.0025,0.0513],[8775,18136,null,null,null,null],[8775,18136,2700,1188,null,null],[8775,18136,null,null,3307,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_pve_espred<\/th>\n      <th>group_pve_4W<\/th>\n      <th>group_pve_ess<\/th>\n      <th>group_size_espred<\/th>\n      <th>group_size_4W<\/th>\n      <th>group_size_ess<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>print(&quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(pve_pie_espred,pve_pie_4W,pve_pie_ess, ncol =3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-5-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fine-mapping-results" class="section level2">
<h2>Fine-mapping results</h2>
<pre class="r"><code>combined_sig_espred &lt;- combined_pip_by_type_espred[combined_pip_by_type_espred$combined_pip &gt; 0.8,]
combined_sig_4W &lt;- combined_pip_by_type_4W[combined_pip_by_type_4W$combined_pip &gt; 0.8,]
combined_sig_ess &lt;- combined_pip_by_type_ess[combined_pip_by_type_ess$combined_pip &gt; 0.8,]

sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e +s&quot;, nrow(combined_sig_espred))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 42 -- predictdb e +s&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e + s + Munro apa + rs&quot;, nrow(combined_sig_4W))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 60 -- predictdb e + s + Munro apa + rs&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e+s + Munro st QTL&quot;, nrow(combined_sig_ess))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 58 -- predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>venn.plot &lt;- plot_3venn(es = combined_sig_espred$gene_name,esra = combined_sig_4W$gene_name,ess = combined_sig_ess$gene_name)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###1
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p1 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e+s setting&quot;)

###2
heatmap_data &lt;- combined_sig_4W[!combined_sig_4W$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;rsQTL_pip&quot;,&quot;apaQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p2 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + apa + rs setting but not by e+s setting&quot;)

###3
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_4W$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)


p3 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e + s + apa + rs setting&quot;)

g1 &lt;- p1$gtable
g2 &lt;- p2$gtable
g3 &lt;- p3$gtable
grid.arrange(g1, g2, g3, ncol=3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-7-1.png" width="2304" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ibd-ebi-a-gcst004131" class="section level1">
<h1>IBD-ebi-a-GCST004131</h1>
<pre class="r"><code>trait &lt;- &quot;IBD-ebi-a-GCST004131&quot;
tissue &lt;- &quot;Colon_Transverse&quot;

gwas_n &lt;- samplesize[trait]

results_dir_ess &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/ess/&quot;)

snp_map_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.ctwas.res.RDS&quot;))

param_ess &lt;- ctwas_res_ess$param
finemap_res_ess &lt;- ctwas_res_ess$finemap_res


p_conv_ess &lt;- make_convergence_plots(param_ess, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_ess &lt;- summarize_param(param_ess, gwas_n)
pve_pie_ess &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_ess, colors = colors)

susie_alpha_res_ess &lt;- ctwas_res_ess$susie_alpha_res

susie_alpha_res_ess &lt;- anno_susie_alpha_res(susie_alpha_res_ess,
                                        mapping_table = mapping_two,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-13 23:18:39 INFO::Annotating susie alpha result ...
2024-10-13 23:18:41 INFO::Map molecular traits to genes
2024-10-13 23:18:42 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type_ess &lt;- combine_gene_pips(susie_alpha_res_ess, 
                                             group_by = &quot;gene_name&quot;,
                                             by = &quot;type&quot;,
                                             method = &quot;combine_cs&quot;,
                                             filter_cs = TRUE,
                                             include_cs_id = F)</code></pre>
<pre class="r"><code>results_dir_espred &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/espred/&quot;)

snp_map_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.ctwas.res.RDS&quot;))

param_espred &lt;- ctwas_res_espred$param
finemap_res_espred &lt;- ctwas_res_espred$finemap_res


p_conv_espred &lt;- make_convergence_plots(param_espred, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_espred &lt;- summarize_param(param_espred, gwas_n)
pve_pie_espred &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_espred, colors = colors)


finemap_res_espred$molecular_id &lt;- get_molecular_ids(finemap_res_espred)
finemap_res_espred &lt;- anno_finemap_res(finemap_res_espred,
                              snp_map = snp_map_espred,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:18:57 INFO::Annotating fine-mapping result ...
2024-10-13 23:18:57 INFO::Map molecular traits to genes
2024-10-13 23:18:57 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:18:59 INFO::Add gene positions
2024-10-13 23:18:59 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_espred &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_espred,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:19:04 INFO::Limit gene results to credible sets</code></pre>
<pre class="r"><code>results_dir_4W &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/4W/&quot;)

snp_map_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.ctwas.res.RDS&quot;))

param_4W &lt;- ctwas_res_4W$param
finemap_res_4W &lt;- ctwas_res_4W$finemap_res

p_conv_4W &lt;- make_convergence_plots(param_4W, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_4W &lt;- summarize_param(param_4W, gwas_n)
pve_pie_4W &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_4W, colors = colors)

finemap_res_4W$molecular_id &lt;- get_molecular_ids(finemap_res_4W)
finemap_res_4W &lt;- anno_finemap_res(finemap_res_4W,
                              snp_map = snp_map_4W,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:19:21 INFO::Annotating fine-mapping result ...
2024-10-13 23:19:21 INFO::Map molecular traits to genes
2024-10-13 23:19:22 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:19:23 INFO::Add gene positions
2024-10-13 23:19:23 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_4W &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_4W,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:19:30 INFO::Limit gene results to credible sets</code></pre>
<div id="parameters-1" class="section level2">
<h2>Parameters</h2>
<pre class="r"><code>print(&quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(p_conv_espred,p_conv_4W,p_conv_ess, ncol = 3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>######pve
group_pve_espred &lt;- ctwas_parameters_espred$group_pve
group_pve_espred &lt;- group_pve_espred[-length(group_pve_espred)]
group_pve_espred &lt;- c(group_pve_espred, rep(NA,3))
group_pve_espred &lt;- c(group_pve_espred, ctwas_parameters_espred$total_pve)
names(group_pve_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_4W &lt;- ctwas_parameters_4W$group_pve
group_pve_4W &lt;- group_pve_4W[-length(group_pve_4W)]
group_pve_4W &lt;- c(group_pve_4W, rep(NA,1))
group_pve_4W &lt;- c(group_pve_4W, ctwas_parameters_4W$total_pve)
names(group_pve_4W) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_ess &lt;- ctwas_parameters_ess$group_pve
group_pve_ess &lt;- group_pve_ess[-length(group_pve_ess)]
group_pve_ess &lt;- c(group_pve_ess[1:2], rep(NA,2),group_pve_ess[3])
group_pve_ess &lt;- c(group_pve_ess, ctwas_parameters_ess$total_pve)
names(group_pve_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

grouppve &lt;- cbind(group_pve_espred,group_pve_4W,group_pve_ess)
grouppve &lt;- round(grouppve,digits = 4)

######size
group_size_espred &lt;- ctwas_parameters_espred$group_size
group_size_espred &lt;- group_size_espred[-length(group_size_espred)]
group_size_espred &lt;- c(group_size_espred, rep(NA,3))
names(group_size_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_4W &lt;- ctwas_parameters_4W$group_size
group_size_4W &lt;- group_size_4W[-length(group_size_4W)]
group_size_4W &lt;- c(group_size_4W, rep(NA,1))
names(group_size_4W) &lt;-  c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_ess &lt;- ctwas_parameters_ess$group_size
group_size_ess &lt;- group_size_ess[-length(group_size_ess)]
group_size_ess &lt;- c(group_size_ess[1:2], rep(NA,2),group_size_ess[3])
names(group_size_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

groupsize &lt;- cbind(group_size_espred,group_size_4W,group_size_ess)

group_info &lt;- cbind(grouppve,rbind(groupsize,c(rep(NA,3))))

DT::datatable(group_info,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group PVE and Group Size&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group PVE and Group Size<\/caption>","data":[["eQTL_pred","sQTL_pred","apaQTL_munro","rsQTL_munro","stQTL_munro","TOTAL"],[0.0182,0.0387,null,null,null,0.2691],[0.0164,0.0325,0.0107,0.008,null,0.2675],[0.0162,0.0323,null,null,0.0186,0.2709],[9577,27048,null,null,null,null],[9577,27048,5318,2761,null,null],[9577,27048,null,null,6237,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_pve_espred<\/th>\n      <th>group_pve_4W<\/th>\n      <th>group_pve_ess<\/th>\n      <th>group_size_espred<\/th>\n      <th>group_size_4W<\/th>\n      <th>group_size_ess<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>print(&quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(pve_pie_espred,pve_pie_4W,pve_pie_ess, ncol =3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-11-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fine-mapping-results-1" class="section level2">
<h2>Fine-mapping results</h2>
<pre class="r"><code>combined_sig_espred &lt;- combined_pip_by_type_espred[combined_pip_by_type_espred$combined_pip &gt; 0.8,]
combined_sig_4W &lt;- combined_pip_by_type_4W[combined_pip_by_type_4W$combined_pip &gt; 0.8,]
combined_sig_ess &lt;- combined_pip_by_type_ess[combined_pip_by_type_ess$combined_pip &gt; 0.8,]

sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e +s&quot;, nrow(combined_sig_espred))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 17 -- predictdb e +s&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e + s + Munro apa + rs&quot;, nrow(combined_sig_4W))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 24 -- predictdb e + s + Munro apa + rs&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e+s + Munro st QTL&quot;, nrow(combined_sig_ess))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 24 -- predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>venn.plot &lt;- plot_3venn(es = combined_sig_espred$gene_name,esra = combined_sig_4W$gene_name,ess = combined_sig_ess$gene_name)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-12-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###1
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p1 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e+s setting&quot;)

###2
heatmap_data &lt;- combined_sig_4W[!combined_sig_4W$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;rsQTL_pip&quot;,&quot;apaQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p2 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + apa + rs setting but not by e+s setting&quot;)

###3
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_4W$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)


p3 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e + s + apa + rs setting&quot;)

g1 &lt;- p1$gtable
g2 &lt;- p2$gtable
g3 &lt;- p3$gtable
grid.arrange(g1, g2, g3, ncol=3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-13-1.png" width="2304" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sbp-ukb-a-360" class="section level1">
<h1>SBP-ukb-a-360</h1>
<pre class="r"><code>trait &lt;- &quot;SBP-ukb-a-360&quot;
tissue &lt;- &quot;Artery_Tibial&quot;

gwas_n &lt;- samplesize[trait]

results_dir_ess &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/ess/&quot;)

snp_map_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.ctwas.res.RDS&quot;))

param_ess &lt;- ctwas_res_ess$param
finemap_res_ess &lt;- ctwas_res_ess$finemap_res


p_conv_ess &lt;- make_convergence_plots(param_ess, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_ess &lt;- summarize_param(param_ess, gwas_n)
pve_pie_ess &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_ess, colors = colors)

susie_alpha_res_ess &lt;- ctwas_res_ess$susie_alpha_res

susie_alpha_res_ess &lt;- anno_susie_alpha_res(susie_alpha_res_ess,
                                        mapping_table = mapping_two,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-13 23:19:53 INFO::Annotating susie alpha result ...
2024-10-13 23:19:53 INFO::Map molecular traits to genes
2024-10-13 23:19:53 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type_ess &lt;- combine_gene_pips(susie_alpha_res_ess, 
                                             group_by = &quot;gene_name&quot;,
                                             by = &quot;type&quot;,
                                             method = &quot;combine_cs&quot;,
                                             filter_cs = TRUE,
                                             include_cs_id = F)</code></pre>
<pre class="r"><code>results_dir_espred &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/espred/&quot;)

snp_map_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.ctwas.res.RDS&quot;))

param_espred &lt;- ctwas_res_espred$param
finemap_res_espred &lt;- ctwas_res_espred$finemap_res


p_conv_espred &lt;- make_convergence_plots(param_espred, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_espred &lt;- summarize_param(param_espred, gwas_n)
pve_pie_espred &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_espred, colors = colors)


finemap_res_espred$molecular_id &lt;- get_molecular_ids(finemap_res_espred)
finemap_res_espred &lt;- anno_finemap_res(finemap_res_espred,
                              snp_map = snp_map_espred,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:20:13 INFO::Annotating fine-mapping result ...
2024-10-13 23:20:16 INFO::Map molecular traits to genes
2024-10-13 23:20:16 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:20:19 INFO::Add gene positions
2024-10-13 23:20:20 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_espred &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_espred,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:20:26 INFO::Limit gene results to credible sets</code></pre>
<pre class="r"><code>results_dir_4W &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/4W/&quot;)

snp_map_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.ctwas.res.RDS&quot;))

param_4W &lt;- ctwas_res_4W$param
finemap_res_4W &lt;- ctwas_res_4W$finemap_res

p_conv_4W &lt;- make_convergence_plots(param_4W, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_4W &lt;- summarize_param(param_4W, gwas_n)
pve_pie_4W &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_4W, colors = colors)

finemap_res_4W$molecular_id &lt;- get_molecular_ids(finemap_res_4W)
finemap_res_4W &lt;- anno_finemap_res(finemap_res_4W,
                              snp_map = snp_map_4W,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:20:46 INFO::Annotating fine-mapping result ...
2024-10-13 23:20:46 INFO::Map molecular traits to genes
2024-10-13 23:20:46 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:20:49 INFO::Add gene positions
2024-10-13 23:20:50 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_4W &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_4W,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:20:59 INFO::Limit gene results to credible sets</code></pre>
<div id="parameters-2" class="section level2">
<h2>Parameters</h2>
<pre class="r"><code>print(&quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(p_conv_espred,p_conv_4W,p_conv_ess, ncol = 3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-17-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>######pve
group_pve_espred &lt;- ctwas_parameters_espred$group_pve
group_pve_espred &lt;- group_pve_espred[-length(group_pve_espred)]
group_pve_espred &lt;- c(group_pve_espred, rep(NA,3))
group_pve_espred &lt;- c(group_pve_espred, ctwas_parameters_espred$total_pve)
names(group_pve_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_4W &lt;- ctwas_parameters_4W$group_pve
group_pve_4W &lt;- group_pve_4W[-length(group_pve_4W)]
group_pve_4W &lt;- c(group_pve_4W, rep(NA,1))
group_pve_4W &lt;- c(group_pve_4W, ctwas_parameters_4W$total_pve)
names(group_pve_4W) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_ess &lt;- ctwas_parameters_ess$group_pve
group_pve_ess &lt;- group_pve_ess[-length(group_pve_ess)]
group_pve_ess &lt;- c(group_pve_ess[1:2], rep(NA,2),group_pve_ess[3])
group_pve_ess &lt;- c(group_pve_ess, ctwas_parameters_ess$total_pve)
names(group_pve_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

grouppve &lt;- cbind(group_pve_espred,group_pve_4W,group_pve_ess)
grouppve &lt;- round(grouppve,digits = 4)

######size
group_size_espred &lt;- ctwas_parameters_espred$group_size
group_size_espred &lt;- group_size_espred[-length(group_size_espred)]
group_size_espred &lt;- c(group_size_espred, rep(NA,3))
names(group_size_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_4W &lt;- ctwas_parameters_4W$group_size
group_size_4W &lt;- group_size_4W[-length(group_size_4W)]
group_size_4W &lt;- c(group_size_4W, rep(NA,1))
names(group_size_4W) &lt;-  c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_ess &lt;- ctwas_parameters_ess$group_size
group_size_ess &lt;- group_size_ess[-length(group_size_ess)]
group_size_ess &lt;- c(group_size_ess[1:2], rep(NA,2),group_size_ess[3])
names(group_size_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

groupsize &lt;- cbind(group_size_espred,group_size_4W,group_size_ess)

group_info &lt;- cbind(grouppve,rbind(groupsize,c(rep(NA,3))))

DT::datatable(group_info,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group PVE and Group Size&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group PVE and Group Size<\/caption>","data":[["eQTL_pred","sQTL_pred","apaQTL_munro","rsQTL_munro","stQTL_munro","TOTAL"],[0.0083,0.0075,null,null,null,0.0673],[0.0072,0.0057,0.0006,0.0046,null,0.0675],[0.0077,0.0076,null,null,0.0027,0.0675],[10137,27192,null,null,null,null],[10137,27192,6667,3561,null,null],[10137,27192,null,null,7369,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_pve_espred<\/th>\n      <th>group_pve_4W<\/th>\n      <th>group_pve_ess<\/th>\n      <th>group_size_espred<\/th>\n      <th>group_size_4W<\/th>\n      <th>group_size_ess<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>print(&quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(pve_pie_espred,pve_pie_4W,pve_pie_ess, ncol =3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-17-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fine-mapping-results-2" class="section level2">
<h2>Fine-mapping results</h2>
<pre class="r"><code>combined_sig_espred &lt;- combined_pip_by_type_espred[combined_pip_by_type_espred$combined_pip &gt; 0.8,]
combined_sig_4W &lt;- combined_pip_by_type_4W[combined_pip_by_type_4W$combined_pip &gt; 0.8,]
combined_sig_ess &lt;- combined_pip_by_type_ess[combined_pip_by_type_ess$combined_pip &gt; 0.8,]

sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e +s&quot;, nrow(combined_sig_espred))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 38 -- predictdb e +s&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e + s + Munro apa + rs&quot;, nrow(combined_sig_4W))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 46 -- predictdb e + s + Munro apa + rs&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e+s + Munro st QTL&quot;, nrow(combined_sig_ess))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 41 -- predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>venn.plot &lt;- plot_3venn(es = combined_sig_espred$gene_name,esra = combined_sig_4W$gene_name,ess = combined_sig_ess$gene_name)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-18-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###1
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p1 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e+s setting&quot;)

###2
heatmap_data &lt;- combined_sig_4W[!combined_sig_4W$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;rsQTL_pip&quot;,&quot;apaQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p2 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + apa + rs setting but not by e+s setting&quot;)

###3
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_4W$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)


p3 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e + s + apa + rs setting&quot;)

g1 &lt;- p1$gtable
g2 &lt;- p2$gtable
g3 &lt;- p3$gtable
grid.arrange(g1, g2, g3, ncol=3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-19-1.png" width="2304" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="wbc-ieu-b-30" class="section level1">
<h1>WBC-ieu-b-30</h1>
<pre class="r"><code>trait &lt;- &quot;WBC-ieu-b-30&quot;
tissue &lt;- &quot;Whole_Blood&quot;

gwas_n &lt;- samplesize[trait]

results_dir_ess &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/ess/&quot;)

snp_map_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_ess &lt;- readRDS(paste0(results_dir_ess,trait,&quot;.ctwas.res.RDS&quot;))

param_ess &lt;- ctwas_res_ess$param
finemap_res_ess &lt;- ctwas_res_ess$finemap_res


p_conv_ess &lt;- make_convergence_plots(param_ess, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_ess &lt;- summarize_param(param_ess, gwas_n)
pve_pie_ess &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_ess, colors = colors)

susie_alpha_res_ess &lt;- ctwas_res_ess$susie_alpha_res

susie_alpha_res_ess &lt;- anno_susie_alpha_res(susie_alpha_res_ess,
                                        mapping_table = mapping_two,
                                        map_by = &quot;molecular_id&quot;,
                                        drop_unmapped = TRUE)</code></pre>
<pre><code>2024-10-13 23:21:27 INFO::Annotating susie alpha result ...
2024-10-13 23:21:27 INFO::Map molecular traits to genes
2024-10-13 23:21:27 INFO::Split PIPs for molecular traits mapped to multiple genes</code></pre>
<pre class="r"><code>combined_pip_by_type_ess &lt;- combine_gene_pips(susie_alpha_res_ess, 
                                             group_by = &quot;gene_name&quot;,
                                             by = &quot;type&quot;,
                                             method = &quot;combine_cs&quot;,
                                             filter_cs = TRUE,
                                             include_cs_id = F)</code></pre>
<pre class="r"><code>results_dir_espred &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/espred/&quot;)

snp_map_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_espred &lt;- readRDS(paste0(results_dir_espred,trait,&quot;.ctwas.res.RDS&quot;))

param_espred &lt;- ctwas_res_espred$param
finemap_res_espred &lt;- ctwas_res_espred$finemap_res


p_conv_espred &lt;- make_convergence_plots(param_espred, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_espred &lt;- summarize_param(param_espred, gwas_n)
pve_pie_espred &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_espred, colors = colors)


finemap_res_espred$molecular_id &lt;- get_molecular_ids(finemap_res_espred)
finemap_res_espred &lt;- anno_finemap_res(finemap_res_espred,
                              snp_map = snp_map_espred,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:21:54 INFO::Annotating fine-mapping result ...
2024-10-13 23:21:54 INFO::Map molecular traits to genes
2024-10-13 23:21:54 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:22:01 INFO::Add gene positions
2024-10-13 23:22:01 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_espred &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_espred,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:22:13 INFO::Limit gene results to credible sets</code></pre>
<pre class="r"><code>results_dir_4W &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/9.deciding_weights_4traits/results/&quot;,trait,&quot;/4W/&quot;)

snp_map_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.snp_map.RDS&quot;))
ctwas_res_4W &lt;- readRDS(paste0(results_dir_4W,trait,&quot;.ctwas.res.RDS&quot;))

param_4W &lt;- ctwas_res_4W$param
finemap_res_4W &lt;- ctwas_res_4W$finemap_res

p_conv_4W &lt;- make_convergence_plots(param_4W, gwas_n, ncol = 1, colors = colors)
ctwas_parameters_4W &lt;- summarize_param(param_4W, gwas_n)
pve_pie_4W &lt;- plot_piechart(ctwas_parameters = ctwas_parameters_4W, colors = colors)

finemap_res_4W$molecular_id &lt;- get_molecular_ids(finemap_res_4W)
finemap_res_4W &lt;- anno_finemap_res(finemap_res_4W,
                              snp_map = snp_map_4W,
                              mapping_table = mapping_two,
                              add_gene_annot = TRUE,
                              map_by = &quot;molecular_id&quot;,
                              drop_unmapped = TRUE,
                              add_position = TRUE,
                              use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-10-13 23:22:35 INFO::Annotating fine-mapping result ...
2024-10-13 23:22:36 INFO::Map molecular traits to genes
2024-10-13 23:22:36 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-10-13 23:22:47 INFO::Add gene positions
2024-10-13 23:22:47 INFO::Add SNP positions</code></pre>
<pre class="r"><code>combined_pip_by_type_4W &lt;- combine_gene_pips_nocs(finemap_res =finemap_res_4W,
                                  group_by = &quot;gene_name&quot;,
                                  by = &quot;type&quot;, 
                                  method = &quot;combine_cs&quot;,
                                  filter_cs = T )</code></pre>
<pre><code>2024-10-13 23:22:54 INFO::Limit gene results to credible sets</code></pre>
<div id="parameters-3" class="section level2">
<h2>Parameters</h2>
<pre class="r"><code>print(&quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each column represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(p_conv_espred,p_conv_4W,p_conv_ess, ncol = 3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-23-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>######pve
group_pve_espred &lt;- ctwas_parameters_espred$group_pve
group_pve_espred &lt;- group_pve_espred[-length(group_pve_espred)]
group_pve_espred &lt;- c(group_pve_espred, rep(NA,3))
group_pve_espred &lt;- c(group_pve_espred, ctwas_parameters_espred$total_pve)
names(group_pve_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_4W &lt;- ctwas_parameters_4W$group_pve
group_pve_4W &lt;- group_pve_4W[-length(group_pve_4W)]
group_pve_4W &lt;- c(group_pve_4W, rep(NA,1))
group_pve_4W &lt;- c(group_pve_4W, ctwas_parameters_4W$total_pve)
names(group_pve_4W) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

group_pve_ess &lt;- ctwas_parameters_ess$group_pve
group_pve_ess &lt;- group_pve_ess[-length(group_pve_ess)]
group_pve_ess &lt;- c(group_pve_ess[1:2], rep(NA,2),group_pve_ess[3])
group_pve_ess &lt;- c(group_pve_ess, ctwas_parameters_ess$total_pve)
names(group_pve_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;,&quot;TOTAL&quot;)

grouppve &lt;- cbind(group_pve_espred,group_pve_4W,group_pve_ess)
grouppve &lt;- round(grouppve,digits = 4)

######size
group_size_espred &lt;- ctwas_parameters_espred$group_size
group_size_espred &lt;- group_size_espred[-length(group_size_espred)]
group_size_espred &lt;- c(group_size_espred, rep(NA,3))
names(group_size_espred) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_4W &lt;- ctwas_parameters_4W$group_size
group_size_4W &lt;- group_size_4W[-length(group_size_4W)]
group_size_4W &lt;- c(group_size_4W, rep(NA,1))
names(group_size_4W) &lt;-  c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

group_size_ess &lt;- ctwas_parameters_ess$group_size
group_size_ess &lt;- group_size_ess[-length(group_size_ess)]
group_size_ess &lt;- c(group_size_ess[1:2], rep(NA,2),group_size_ess[3])
names(group_size_ess) &lt;- c(&quot;eQTL_pred&quot;,&quot;sQTL_pred&quot;,&quot;apaQTL_munro&quot;,&quot;rsQTL_munro&quot;,&quot;stQTL_munro&quot;)

groupsize &lt;- cbind(group_size_espred,group_size_4W,group_size_ess)

group_info &lt;- cbind(grouppve,rbind(groupsize,c(rep(NA,3))))

DT::datatable(group_info,caption = htmltools::tags$caption( style = &#39;caption-side: topleft; text-align = left; color:black;&#39;,&#39;Group PVE and Group Size&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-93928d62269dcef60a51" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: topleft; text-align = left; color:black;\">Group PVE and Group Size<\/caption>","data":[["eQTL_pred","sQTL_pred","apaQTL_munro","rsQTL_munro","stQTL_munro","TOTAL"],[0.0151,0.0106,null,null,null,0.1129],[0.0205,0.0092,0.0027,0.0144,null,0.1237],[0.0207,0.0092,null,null,0.0139,0.121],[8455,15248,null,null,null,null],[8455,15248,4700,2668,null,null],[8455,15248,null,null,5629,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group_pve_espred<\/th>\n      <th>group_pve_4W<\/th>\n      <th>group_pve_ess<\/th>\n      <th>group_size_espred<\/th>\n      <th>group_size_4W<\/th>\n      <th>group_size_ess<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>print(&quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;)</code></pre>
<pre><code>[1] &quot;each pie chart represents one setting: predictdb e+s, predictdb e+s + Munro apa+rs, predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>grid.arrange(pve_pie_espred,pve_pie_4W,pve_pie_ess, ncol =3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-23-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="fine-mapping-results-3" class="section level2">
<h2>Fine-mapping results</h2>
<pre class="r"><code>combined_sig_espred &lt;- combined_pip_by_type_espred[combined_pip_by_type_espred$combined_pip &gt; 0.8,]
combined_sig_4W &lt;- combined_pip_by_type_4W[combined_pip_by_type_4W$combined_pip &gt; 0.8,]
combined_sig_ess &lt;- combined_pip_by_type_ess[combined_pip_by_type_ess$combined_pip &gt; 0.8,]

sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e +s&quot;, nrow(combined_sig_espred))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 121 -- predictdb e +s&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e + s + Munro apa + rs&quot;, nrow(combined_sig_4W))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 147 -- predictdb e + s + Munro apa + rs&quot;</code></pre>
<pre class="r"><code>sprintf(&quot;# of genes with PIP &gt; 0.8 = %s -- predictdb e+s + Munro st QTL&quot;, nrow(combined_sig_ess))</code></pre>
<pre><code>[1] &quot;# of genes with PIP &gt; 0.8 = 135 -- predictdb e+s + Munro st QTL&quot;</code></pre>
<pre class="r"><code>venn.plot &lt;- plot_3venn(es = combined_sig_espred$gene_name,esra = combined_sig_4W$gene_name,ess = combined_sig_ess$gene_name)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-24-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###1
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p1 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e+s setting&quot;)

###2
heatmap_data &lt;- combined_sig_4W[!combined_sig_4W$gene_name %in%combined_sig_espred$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;rsQTL_pip&quot;,&quot;apaQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)
p2 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + apa + rs setting but not by e+s setting&quot;)

###3
heatmap_data &lt;- combined_sig_ess[!combined_sig_ess$gene_name %in%combined_sig_4W$gene_name, ]
column_order &lt;- c(&quot;gene_name&quot;,&quot;combined_pip&quot;,
                  &quot;eQTL_pip&quot;, &quot;sQTL_pip&quot;, &quot;stQTL_pip&quot;)
heatmap_data &lt;- rename_heatmap_columns(heatmap_data = heatmap_data, column_order = column_order)


p3 &lt;- plot_heatmap(heatmap_data = heatmap_data,main = &quot;PIP partition for the genes reported by e + s + st setting but not by e + s + apa + rs setting&quot;)

g1 &lt;- p1$gtable
g2 &lt;- p2$gtable
g3 &lt;- p3$gtable
grid.arrange(g1, g2, g3, ncol=3)</code></pre>
<p><img src="figure/multi_group_compare_decidingweights_4traits_ess.Rmd/unnamed-chunk-25-1.png" width="2304" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] logging_0.10-108          dplyr_1.1.4              
 [3] pheatmap_1.0.12           gridExtra_2.3            
 [5] ggplot2_3.5.1             VennDiagram_1.7.3        
 [7] futile.logger_1.4.3       EnsDb.Hsapiens.v86_2.99.0
 [9] ensembldb_2.20.2          AnnotationFilter_1.20.0  
[11] GenomicFeatures_1.48.3    AnnotationDbi_1.58.0     
[13] Biobase_2.56.0            GenomicRanges_1.48.0     
[15] GenomeInfoDb_1.39.9       IRanges_2.30.0           
[17] S4Vectors_0.34.0          BiocGenerics_0.42.0      
[19] ctwas_0.4.15             

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                DT_0.22                    
 [11] ggrepel_0.9.1               bit64_4.0.5                
 [13] fansi_1.0.3                 xml2_1.3.3                 
 [15] codetools_0.2-18            cachem_1.0.6               
 [17] knitr_1.39                  jsonlite_1.8.0             
 [19] workflowr_1.7.0             Rsamtools_2.12.0           
 [21] dbplyr_2.1.1                png_0.1-7                  
 [23] readr_2.1.2                 compiler_4.2.0             
 [25] httr_1.4.3                  assertthat_0.2.1           
 [27] Matrix_1.5-3                fastmap_1.1.0              
 [29] lazyeval_0.2.2              cli_3.6.1                  
 [31] formatR_1.12                later_1.3.0                
 [33] htmltools_0.5.2             prettyunits_1.1.1          
 [35] tools_4.2.0                 gtable_0.3.0               
 [37] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [39] rappdirs_0.3.3              Rcpp_1.0.12                
 [41] jquerylib_0.1.4             vctrs_0.6.5                
 [43] Biostrings_2.64.0           rtracklayer_1.56.0         
 [45] crosstalk_1.2.0             xfun_0.41                  
 [47] stringr_1.5.1               lifecycle_1.0.4            
 [49] irlba_2.3.5                 restfulr_0.0.14            
 [51] XML_3.99-0.14               zlibbioc_1.42.0            
 [53] zoo_1.8-10                  scales_1.3.0               
 [55] gggrid_0.2-0                hms_1.1.1                  
 [57] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [59] ProtGenerics_1.28.0         parallel_4.2.0             
 [61] SummarizedExperiment_1.26.1 RColorBrewer_1.1-3         
 [63] lambda.r_1.2.4              LDlinkR_1.2.3              
 [65] yaml_2.3.5                  curl_4.3.2                 
 [67] memoise_2.0.1               sass_0.4.1                 
 [69] biomaRt_2.54.1              stringi_1.7.6              
 [71] RSQLite_2.3.1               highr_0.9                  
 [73] BiocIO_1.6.0                filelock_1.0.2             
 [75] BiocParallel_1.30.3         rlang_1.1.2                
 [77] pkgconfig_2.0.3             matrixStats_0.62.0         
 [79] bitops_1.0-7                evaluate_0.15              
 [81] lattice_0.20-45             purrr_1.0.2                
 [83] labeling_0.4.2              GenomicAlignments_1.32.0   
 [85] htmlwidgets_1.5.4           cowplot_1.1.1              
 [87] bit_4.0.4                   tidyselect_1.2.0           
 [89] magrittr_2.0.3              R6_2.5.1                   
 [91] generics_0.1.2              DelayedArray_0.22.0        
 [93] DBI_1.2.2                   withr_2.5.0                
 [95] pgenlibr_0.3.3              pillar_1.9.0               
 [97] KEGGREST_1.36.3             RCurl_1.98-1.7             
 [99] mixsqp_0.3-43               tibble_3.2.1               
[101] crayon_1.5.1                futile.options_1.0.1       
[103] utf8_1.2.2                  BiocFileCache_2.4.0        
[105] plotly_4.10.0               tzdb_0.4.0                 
[107] rmarkdown_2.25              progress_1.2.2             
[109] data.table_1.14.2           blob_1.2.3                 
[111] git2r_0.30.1                digest_0.6.29              
[113] tidyr_1.3.0                 httpuv_1.6.5               
[115] munsell_0.5.0               viridisLite_0.4.0          
[117] bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
