<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2024-12-17" />

<title>6 Traits, 5 tissues, eQTL + sQTL + stQTL – compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP only</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">6 Traits, 5 tissues, eQTL + sQTL + stQTL – compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP only</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2024-12-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree922307556daa022d63e87e2320be53f3cfd7f170targetblank9223075a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/922307556daa022d63e87e2320be53f3cfd7f170" target="_blank">9223075</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree922307556daa022d63e87e2320be53f3cfd7f170targetblank9223075a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/922307556daa022d63e87e2320be53f3cfd7f170" target="_blank">9223075</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/multi_group_6traits_15weights_ess_postprocessing_compare_pipz.Rmd

Unstaged changes:
    Modified:   analysis/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd</code>) and HTML (<code>docs/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/922307556daa022d63e87e2320be53f3cfd7f170/analysis/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd" target="_blank">9223075</a>
</td>
<td>
XSun
</td>
<td>
2024-12-18
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We compare post-processed results with the original results: <a href="https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess.html" class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess.html</a></p>
<p>The post-processing steps include the following:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Region Merging</strong></p>
<p>For the regions with <code>susie_pip &gt; 0.5</code></p></li>
<li><p><strong>LD Mismatch Fixing</strong></p></li>
</ol>
<ul>
<li>Regions were selected where nonSNP_PIP &gt; 0.5.</li>
<li>For genes with <code>susie_pip &gt; thresholds</code> (0.5 and 0.2), we performed LD mismatch diagnosis.</li>
<li>To address LD mismatches, two strategies were employed:
<ul>
<li>Fine-mapping the region without LD.</li>
<li>Removing mismatched SNPs for all genes in the problematic regions, updating gene Z-scores, re-estimated L, and re-fine-mapping with LD.</li>
</ul></li>
</ul>
<p>The problematic regions here are the regions 1) reported by <code>diagnose_LD_mismatch_susie</code> function &amp; 2) containing at least 1 problematic genes reported by <code>get_problematic_genes</code> function.</p>
<pre class="r"><code>library(ctwas)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(gridExtra)
library(dplyr)

ens_db &lt;- EnsDb.Hsapiens.v86

mapping_predictdb &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS&quot;)
mapping_munro &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS&quot;)
mapping_two &lt;- rbind(mapping_predictdb,mapping_munro)</code></pre>
</div>
<div id="afib-ebi-a-gcst006414" class="section level1">
<h1>aFib-ebi-a-GCST006414</h1>
<pre class="r"><code>trait &lt;- &quot;aFib-ebi-a-GCST006414&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9"],["ENSG00000078674.grp_1.downstream.ENST00000325083|Heart_Atrial_Appendage_stQTL","ENSG00000078674.grp_1.downstream.ENST00000524226|Heart_Atrial_Appendage_stQTL","ENSG00000104763|Thyroid_stQTL","ENSG00000114450|Heart_Atrial_Appendage_stQTL","ENSG00000146535.grp_2.downstream.ENST00000407904|Artery_Tibial_stQTL","ENSG00000169902|Artery_Tibial_stQTL","ENSG00000170873.18|Heart_Atrial_Appendage_eQTL","ENSG00000172650.13|Heart_Atrial_Appendage_eQTL","ENSG00000177791|Heart_Atrial_Appendage_stQTL"],[0.999998629090069,0.9859562771073552,0.9999999999999797,0.5509053259805465,0.5617376568938596,1,0.6405805952717825,1,1],["L4","L3","L1","L1","L1","L1","L1","L1","L1"],[4.420128534998735e-05,0.0001164351697535704,0.999999999999894,0.1690747127350973,0.1085287714937646,0.616687169919746,0.03593048497046891,2.330445847409379e-08,1],[null,null,"L1","L1","L1",null,"L1",null,"L1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

load(file_pipthreshold02)
pip_02 &lt;- data.frame(
&quot;PIP Threshold&quot; = &quot;0.2&quot;,
&quot;Number of Selected Regions&quot; = length(selected_region_ids),
&quot;Number of Problematic Genes&quot; = length(problematic_genes),
&quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
&quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
&quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)


load(file_pipthreshold05)
pip_05 &lt;- data.frame(
&quot;PIP Threshold&quot; = &quot;0.5&quot;,
&quot;Number of Selected Regions&quot; = length(selected_region_ids),
&quot;Number of Problematic Genes&quot; = length(problematic_genes),
&quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
&quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
&quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-20575149739a71fc930f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[101,101],[4,4],[4,4],[2175,2175],[24,24]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {
  
  load(file_pipthreshold02)
  
  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res
  
  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
  p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()
  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{
  
  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)
  
}</code></pre>
<pre><code>[1] &quot;Total genes in problematic regions = 1516&quot;
[1] &quot;Number of genes disappeared after removing prblematic SNPs = 8&quot;
[1] &quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-5-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  
  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  
  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()
  

  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  
  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{
  
  print(&quot;There&#39;s no problematic genes&quot;)
  
  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-6-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;L - estimated in region merge step&quot;
 3_110794923_113096852 10_110801735_113568673 11_116512631_117876395 
                     3                      3                      3 
12_121569746_124493434 
                     5 
[1] &quot;L - re-estimated after updating z_scores, region data&quot;
 3_110794923_113096852 10_110801735_113568673 11_116512631_117876395 
                     2                      1                      1 
12_121569746_124493434 
                     3 </code></pre>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-6-2.png" width="1632" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ldl-ukb-d-30780_irnt" class="section level1">
<h1>LDL-ukb-d-30780_irnt</h1>
<pre class="r"><code>trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge-1" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-a23c348805048a1f64c7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17"],["ENSG00000004975|Esophagus_Gastroesophageal_Junction_stQTL","ENSG00000067208|Liver_stQTL","ENSG00000100994|Skin_Not_Sun_Exposed_Suprapubic_stQTL","ENSG00000100997.grp_2.downstream.ENST00000465694|Esophagus_Gastroesophageal_Junction_stQTL","ENSG00000108424.grp_2.downstream.ENST00000540627|Spleen_stQTL","ENSG00000112033.grp_1.downstream.ENST00000360694|Skin_Not_Sun_Exposed_Suprapubic_stQTL","ENSG00000117500.grp_1.downstream.ENST00000370280|Spleen_stQTL","ENSG00000130733|Esophagus_Mucosa_stQTL","ENSG00000131242|Esophagus_Mucosa_stQTL","ENSG00000134201.grp_1.downstream.ENST00000648974|Spleen_stQTL","ENSG00000136238|Liver_stQTL","ENSG00000167711|Liver_stQTL","ENSG00000196735|Esophagus_Gastroesophageal_Junction_stQTL","ENSG00000204287|Esophagus_Mucosa_stQTL","ENSG00000204482.grp_1.downstream.ENST00000339530|Liver_stQTL","intron_6_32661457_32661967|Skin_Not_Sun_Exposed_Suprapubic_sQTL","intron_6_32756466_32758850|Skin_Not_Sun_Exposed_Suprapubic_sQTL"],[0.9999008608354507,0.6748046017236357,0.9433763113101934,0.5388451350829,1,0.882607203029726,1,0.9999999065472096,0.9999997062481353,1,0.6633849666240283,0.8298475003164851,0.9138928034171887,1,0.9987278449804673,0.9999999999742228,1],["L2","L1","L1","L1","L1","L3","L1","L5","L1","L1","L1","L1","L2","L3","L2","L1","L5"],[1.542041151325613e-05,0.1002641964238559,0.01376757382367289,0.5338625615914399,3.994186371647501e-07,1.821018479342129e-07,4.406021183456232e-07,1.110223024625157e-15,0.0001301691974008312,0,0.05456546046260335,0.1253280836183581,0.0001427680582100077,4.234280703840909e-10,1.472405837699142e-06,0.5939130290592542,2.594609904704726e-08],[null,"L1",null,"L1",null,null,null,null,null,null,"L1","L1",null,null,null,"L5",null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch-1" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis-1" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold02)) {
  load(file_pipthreshold02)
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_&quot;, trait, &quot;.rdata&quot;))
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold05)) {
  load(file_pipthreshold05)
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_&quot;, trait, &quot;.rdata&quot;))
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}


results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-93928d62269dcef60a51" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[121,121],[1,1],[1,1],[405,405],[10,10]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods-1" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {

  load(file_pipthreshold02)

  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res

  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
    p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()
  
  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  print(&quot;Notes: 2 intron overlapped&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{

  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)

}</code></pre>
<pre><code>[1] &quot;Total genes in problematic regions = 1643&quot;
[1] &quot;Number of genes disappeared after removing prblematic SNPs = 4&quot;
[1] &quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;
[1] &quot;Notes: 2 intron overlapped&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-10-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip-1" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)

  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-11-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;L - estimated in region merge step&quot;
19_9127717_13360313 
                  5 
[1] &quot;L - re-estimated after updating z_scores, region data&quot;
19_9127717_13360313 
                  5 </code></pre>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-11-2.png" width="1632" style="display: block; margin: auto;" /></p>
</div>
<div id="examples-for-ld-mismatch-fixing" class="section level2">
<h2>Examples for LD-mismatch fixing</h2>
<pre class="r"><code>weights_origin &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;,trait,&quot;.preprocessed.weights.RDS&quot;))

load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_weights_updated_nozfilter_&quot;,trait,&quot;.rdata&quot;))

region_id &lt;- &quot;19_9127717_13360313&quot;

finemap_res_rm &lt;- anno_finemap_res(finemap_res_rm,
                                          snp_map = updated_data_res_regionmerge[[&quot;updated_snp_map&quot;]],
                                          mapping_table = mapping_two,
                                          add_gene_annot = TRUE,
                                          map_by = &quot;molecular_id&quot;,
                                          drop_unmapped = TRUE,
                                          add_position = TRUE,
                                          use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-12-19 09:53:11 INFO::Annotating fine-mapping result ...
2024-12-19 09:53:11 INFO::Map molecular traits to genes
2024-12-19 09:53:12 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-12-19 09:53:19 INFO::Add gene positions
2024-12-19 09:53:19 INFO::Add SNP positions</code></pre>
<pre class="r"><code>finemap_res_ldmm_nold &lt;- anno_finemap_res(finemap_res_ldmm_nold,
                                          snp_map = updated_data_res_regionmerge[[&quot;updated_snp_map&quot;]],
                                          mapping_table = mapping_two,
                                          add_gene_annot = TRUE,
                                          map_by = &quot;molecular_id&quot;,
                                          drop_unmapped = TRUE,
                                          add_position = TRUE,
                                          use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-12-19 09:53:28 INFO::Annotating fine-mapping result ...
2024-12-19 09:53:28 INFO::Map molecular traits to genes
2024-12-19 09:53:29 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-12-19 09:53:35 INFO::Add gene positions
2024-12-19 09:53:36 INFO::Add SNP positions</code></pre>
<pre class="r"><code>finemap_res_ldmm_removesnp &lt;- anno_finemap_res(finemap_res_ldmm_removesnp,
                                   snp_map = updated_data_res_regionmerge[[&quot;updated_snp_map&quot;]],
                                   mapping_table = mapping_two,
                                   add_gene_annot = TRUE,
                                   map_by = &quot;molecular_id&quot;,
                                   drop_unmapped = TRUE,
                                   add_position = TRUE,
                                   use_gene_pos = &quot;mid&quot;)</code></pre>
<pre><code>2024-12-19 09:53:40 INFO::Annotating fine-mapping result ...
2024-12-19 09:53:40 INFO::Map molecular traits to genes
2024-12-19 09:53:41 INFO::Split PIPs for molecular traits mapped to multiple genes
2024-12-19 09:53:47 INFO::Add gene positions
2024-12-19 09:53:47 INFO::Add SNP positions</code></pre>
<pre class="r"><code>finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]



print(&quot;locus plot -- after region merge&quot;)</code></pre>
<pre><code>[1] &quot;locus plot -- after region merge&quot;</code></pre>
<pre class="r"><code>make_locusplot(finemap_res_rm,
               region_id = region_id,
               ens_db = ens_db,
               weights = weights_origin,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;,panel.heights = c(4,4,1,1))</code></pre>
<pre><code>2024-12-19 09:53:55 INFO::Limit to protein coding genes
2024-12-19 09:53:55 INFO::focal id: intron_19_11120522_11123174|Liver_sQTL
2024-12-19 09:53:55 INFO::focal molecular trait: LDLR Liver sQTL
2024-12-19 09:53:55 INFO::Range of locus: chr19:9127860-13930432
2024-12-19 09:53:57 INFO::focal molecular trait QTL positions: 11120205,11120527
2024-12-19 09:53:57 INFO::Limit PIPs to credible sets</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-12-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(&quot;locus plot -- LD mismatch: no LD&quot;)</code></pre>
<pre><code>[1] &quot;locus plot -- LD mismatch: no LD&quot;</code></pre>
<pre class="r"><code>make_locusplot(finemap_res_ldmm_nold,
               region_id = region_id,
               ens_db = ens_db,
               weights = weights_origin,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;,panel.heights = c(4,4,1,1))</code></pre>
<pre><code>2024-12-19 09:54:02 INFO::Limit to protein coding genes
2024-12-19 09:54:02 INFO::focal id: ENSG00000197256.10|Liver_eQTL
2024-12-19 09:54:02 INFO::focal molecular trait: KANK2 Liver eQTL
2024-12-19 09:54:02 INFO::Range of locus: chr19:9127860-13930432
2024-12-19 09:54:03 INFO::focal molecular trait QTL positions: 11197621
2024-12-19 09:54:03 INFO::Limit PIPs to credible sets</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-12-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(&quot;locus plot -- LD mismatch: snp removed&quot;)</code></pre>
<pre><code>[1] &quot;locus plot -- LD mismatch: snp removed&quot;</code></pre>
<pre class="r"><code>make_locusplot(finemap_res_ldmm_removesnp,
               region_id = region_id,
               ens_db = ens_db,
               weights = weights_updated,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = &quot;cs&quot;,
               color_pip_by = &quot;cs&quot;,panel.heights = c(4,4,1,1))</code></pre>
<pre><code>2024-12-19 09:54:08 INFO::Limit to protein coding genes
2024-12-19 09:54:08 INFO::focal id: intron_19_11116212_11120092|Liver_sQTL
2024-12-19 09:54:08 INFO::focal molecular trait: LDLR Liver sQTL
2024-12-19 09:54:08 INFO::Range of locus: chr19:9127860-13930432
2024-12-19 09:54:09 INFO::focal molecular trait QTL positions: 11116394
2024-12-19 09:54:09 INFO::Limit PIPs to credible sets</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-12-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>finemap_res_rm_gene_region &lt;- finemap_res_rm_gene[finemap_res_rm_gene$region_id == region_id,]
finemap_res_ldmm_removesnp_gene_region &lt;- finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$region_id == region_id,]
merged_region_gene &lt;- merge(finemap_res_rm_gene_region,finemap_res_ldmm_removesnp_gene_region,by = &quot;id&quot;)
merged_region_gene &lt;- merged_region_gene[,c(&quot;id&quot;,&quot;gene_name.x&quot;,&quot;z.x&quot;,&quot;susie_pip.x&quot;,&quot;cs.x&quot;,&quot;z.y&quot;,&quot;susie_pip.y&quot;,&quot;cs.y&quot;)]
colnames(merged_region_gene) &lt;- c(&quot;id&quot;,&quot;gene_name&quot;,&quot;z_regionmerge&quot;,&quot;susie_pip_regionmerge&quot;,&quot;cs_regionmerge&quot;,&quot;z_ldmismatch&quot;,&quot;susie_pip_ldmismatch&quot;,&quot;cs_ldmismatch&quot;)

merged_region_gene$highlight &lt;- ifelse(merged_region_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

ggplot(data = merged_region_gene, aes(x= z_regionmerge, y= z_ldmismatch, color = highlight, alpha = highlight)) +
  geom_point() +
  scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
  scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.3)) +
  ggtitle(&quot;Comparing z-scores before/after removing the problematic SNPs&quot;) +
  theme_minimal()</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-12-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DT::datatable(merged_region_gene[merged_region_gene$z_ldmismatch != merged_region_gene$z_regionmerge,],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Genes with different z before / after removing the problematic SNPs&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-16707cd634e93674753d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Genes with different z before / after removing the problematic SNPs<\/caption>","data":[["240","350","402","404","405","415","418","918"],["ENSG00000130818|Esophagus_Gastroesophageal_Junction_stQTL","ENSG00000174652|Spleen_stQTL","ENSG00000188000.grp_1.downstream.ENST00000641288|Esophagus_Mucosa_stQTL","ENSG00000188000.grp_1.downstream.ENST00000642043|Esophagus_Gastroesophageal_Junction_stQTL","ENSG00000188000.grp_1.downstream.ENST00000642043|Esophagus_Mucosa_stQTL","ENSG00000188321.grp_1.downstream.ENST00000587557|Esophagus_Mucosa_stQTL","ENSG00000188321|Skin_Not_Sun_Exposed_Suprapubic_stQTL","intron_19_11120522_11123174|Liver_sQTL"],["ZNF426","ZNF266","OR7D2","OR7D2","OR7D2","ZNF559","ZNF559","LDLR"],[0.3385219166978995,-1.373705858525579,-0.3851316091593578,1.237843991154348,0.05981593521616021,0.8026369843687174,1.49170281785308,26.89782445433318],[0,0,0,0,0,0,0,1],[null,null,null,null,null,null,null,"L2"],[0.4446578098620101,-1.841388585357525,-0.6566879007295703,1.441182672370958,0.3669892162501076,0.8018715481271792,1.946308308760859,13.644536360949],[0,0,0,0,0,0,0,0],[null,null,null,null,null,null,null,null],["good genes","good genes","good genes","good genes","good genes","good genes","good genes","problematic genes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>gene_name<\/th>\n      <th>z_regionmerge<\/th>\n      <th>susie_pip_regionmerge<\/th>\n      <th>cs_regionmerge<\/th>\n      <th>z_ldmismatch<\/th>\n      <th>susie_pip_ldmismatch<\/th>\n      <th>cs_ldmismatch<\/th>\n      <th>highlight<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[3,4,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="ibd-ebi-a-gcst004131" class="section level1">
<h1>IBD-ebi-a-GCST004131</h1>
<pre class="r"><code>trait &lt;- &quot;IBD-ebi-a-GCST004131&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge-2" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-37e46cf92521dcbed3f8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["ENSG00000075426|Cells_Cultured_fibroblasts_stQTL","ENSG00000114316.grp_1.downstream.ENST00000265560|Spleen_stQTL","ENSG00000138796|Testis_stQTL","ENSG00000159128|Testis_stQTL","ENSG00000164403.grp_2.downstream.ENST00000606676|Whole_Blood_stQTL","ENSG00000173421.grp_2.downstream.ENST00000438782|Testis_stQTL","ENSG00000174945.grp_2.downstream.ENST00000407112|Spleen_stQTL","ENSG00000196367.grp_2.downstream.ENST00000482799|Adipose_Subcutaneous_stQTL","ENSG00000197050|Adipose_Subcutaneous_stQTL","ENSG00000198742|Spleen_stQTL","ENSG00000198742|Testis_stQTL","ENSG00000204498|Cells_Cultured_fibroblasts_stQTL","ENSG00000232629|Spleen_stQTL","ENSG00000233276.grp_1.downstream.ENST00000419783|Spleen_stQTL"],[0.8824260803068251,1,0.7474772969237748,0.8300300409594512,0.9999999999194221,0.7996577225074121,0.9953330808060462,0.9813245556398729,0.5161986540730998,0.9446921393974248,1,0.99995310092348,0.9999999999983545,1],["L1","L1","L1","L1","L2","L1","L1","L2","L1","L2","L1","L1","L4","L3"],[0.0056128385431925,5.574125207630232e-06,0.04282433714642897,0.06553262137204385,4.30699414344371e-06,0.6811979067313946,0.03092137006899576,2.54955945244717e-10,0.03142785018496874,4.111235760217991e-05,0.988961259206468,1.110223024625157e-14,0,1.963319360198668e-06],[null,null,"L1","L1",null,"L1","L1",null,"L1",null,"L1",null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch-2" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis-2" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold02)) {
  load(file_pipthreshold02)
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_&quot;, trait, &quot;.rdata&quot;))
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold05)) {
  load(file_pipthreshold05)
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_&quot;, trait, &quot;.rdata&quot;))
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}


results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-99e959167faff8d1fc37" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-99e959167faff8d1fc37">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[68,68],[1,0],[1,0],[538,538],[5,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods-2" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {

  load(file_pipthreshold02)

  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res

  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
  p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{

  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)

}</code></pre>
<pre><code>[1] &quot;Total genes in problematic regions = 201&quot;
[1] &quot;Number of genes disappeared after removing prblematic SNPs = 1&quot;
[1] &quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-16-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip-2" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)

  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-17-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;L - estimated in region merge step&quot;
5_96627815_97979897 
                  1 
[1] &quot;L - re-estimated after updating z_scores, region data&quot;
5_96627815_97979897 
                  1 </code></pre>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-17-2.png" width="1632" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sbp-ukb-a-360" class="section level1">
<h1>SBP-ukb-a-360</h1>
<pre class="r"><code>trait &lt;- &quot;SBP-ukb-a-360&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge-3" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4a6133f685aa29daac40" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4a6133f685aa29daac40">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10"],["ENSG00000002822.15|Heart_Atrial_Appendage_eQTL","ENSG00000025434|Adipose_Subcutaneous_stQTL","ENSG00000067560|Heart_Atrial_Appendage_stQTL","ENSG00000108883|Brain_Cortex_stQTL","ENSG00000121068|Heart_Atrial_Appendage_stQTL","ENSG00000128829|Brain_Cortex_stQTL","ENSG00000139436.grp_1.downstream.ENST00000320063|Skin_Sun_Exposed_Lower_leg_stQTL","ENSG00000145020.grp_1.downstream.ENST00000636188|Heart_Atrial_Appendage_stQTL","ENSG00000146966|Brain_Cortex_stQTL","ENSG00000177791|Heart_Atrial_Appendage_stQTL"],[0.523274888061062,0.9999968037592195,1,1,0.831050625033969,0.6477569036522894,0.8046738978056319,0.9999999582357118,0.9845774227830817,0.9999998888410815],["L1","L1","L3","L3","L1","L1","L1","L1","L1","L1"],[0,0,8.186049213931845e-06,0,0.5054638839330887,0.0519501546422485,1.024735851729019e-13,0.0008791260610928608,0.6806266396314091,0.9999994498619506],[null,null,null,null,"L1","L1",null,null,"L1","L1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch-3" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis-3" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold02)) {
  load(file_pipthreshold02)
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_&quot;, trait, &quot;.rdata&quot;))
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold05)) {
  load(file_pipthreshold05)
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_&quot;, trait, &quot;.rdata&quot;))
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}


results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d72b6eff4afa454a8832" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d72b6eff4afa454a8832">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[125,125],[3,0],[1,0],[1073,1073],[193,193]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods-3" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {

  load(file_pipthreshold02)

  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res

  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
  p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{

  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)

}</code></pre>
<pre><code>[1] &quot;Total genes in problematic regions = 229&quot;
[1] &quot;Number of genes disappeared after removing prblematic SNPs = 4&quot;
[1] &quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-21-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip-3" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)

  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-22-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;L - estimated in region merge step&quot;
3_133533329_135738064 
                    2 
[1] &quot;L - re-estimated after updating z_scores, region data&quot;
3_133533329_135738064 
                    2 </code></pre>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-22-2.png" width="1632" style="display: block; margin: auto;" /></p>
<!-- ## Examples for LD-mismatch fixing -->
<!-- ### Genes identified after LD mismatch fixing but not before -->
<!-- ```{r message=FALSE, warning=FALSE, fig.width=12, fig.height=9} -->
<!-- print("Two genes have PIP == 0 after region merging but PIP > 0.8 after LD mismatch fixed (remove snp method)") -->
<!-- finemap_res_ldmm_removesnp_problematic_gene[finemap_res_ldmm_removesnp_problematic_gene$id %in% c("ENSG00000103415.11|Artery_Tibial_eQTL","ENSG00000130592.15|Heart_Atrial_Appendage_eQTL"),] -->
<!-- weights_origin <- readRDS(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/",trait,".preprocessed.weights.RDS")) -->
<!-- load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres05_removesnp_weights_updated_",trait,".rdata")) -->
<!-- finemap_res_rm <- anno_finemap_res(finemap_res_rm, -->
<!--                                           snp_map = updated_data_res_regionmerge[["updated_snp_map"]], -->
<!--                                           mapping_table = mapping_two, -->
<!--                                           add_gene_annot = TRUE, -->
<!--                                           map_by = "molecular_id", -->
<!--                                           drop_unmapped = TRUE, -->
<!--                                           add_position = TRUE, -->
<!--                                           use_gene_pos = "mid") -->
<!-- finemap_res_ldmm_nold <- anno_finemap_res(finemap_res_ldmm_nold, -->
<!--                                           snp_map = updated_data_res_regionmerge[["updated_snp_map"]], -->
<!--                                           mapping_table = mapping_two, -->
<!--                                           add_gene_annot = TRUE, -->
<!--                                           map_by = "molecular_id", -->
<!--                                           drop_unmapped = TRUE, -->
<!--                                           add_position = TRUE, -->
<!--                                           use_gene_pos = "mid") -->
<!-- finemap_res_ldmm_removesnp <- anno_finemap_res(finemap_res_ldmm_removesnp, -->
<!--                                    snp_map = updated_data_res_regionmerge[["updated_snp_map"]], -->
<!--                                    mapping_table = mapping_two, -->
<!--                                    add_gene_annot = TRUE, -->
<!--                                    map_by = "molecular_id", -->
<!--                                    drop_unmapped = TRUE, -->
<!--                                    add_position = TRUE, -->
<!--                                    use_gene_pos = "mid") -->
<!-- finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",] -->
<!-- finemap_res_ldmm_removesnp_gene <- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !="SNP",] -->
<!-- region_id <- "11_1192365_3644251" -->
<!-- print("locus plot -- after region merge") -->
<!-- make_locusplot(finemap_res_rm, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_origin, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- print("locus plot -- LD mismatch: no LD") -->
<!-- make_locusplot(finemap_res_ldmm_nold, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_origin, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- print("locus plot -- LD mismatch: snp removed") -->
<!-- make_locusplot(finemap_res_ldmm_removesnp, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_updated, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- finemap_res_rm_gene_region <- finemap_res_rm_gene[finemap_res_rm_gene$region_id == region_id,] -->
<!-- finemap_res_ldmm_removesnp_gene_region <- finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$region_id == region_id,] -->
<!-- merged_region_gene <- merge(finemap_res_rm_gene_region,finemap_res_ldmm_removesnp_gene_region,by = "id") -->
<!-- merged_region_gene <- merged_region_gene[,c("id","gene_name.x","z.x","susie_pip.x","cs.x","z.y","susie_pip.y","cs.y")] -->
<!-- colnames(merged_region_gene) <- c("id","gene_name","z_regionmerge","susie_pip_regionmerge","cs_regionmerge","z_ldmismatch","susie_pip_ldmismatch","cs_ldmismatch") -->
<!-- ggplot(data = merged_region_gene, aes(x= z_regionmerge, y= z_ldmismatch)) +  -->
<!--   geom_point() + -->
<!--   ggtitle("Comparing z-scores before/after removing the problematic SNPs") + -->
<!--   theme_minimal() -->
<!-- DT::datatable(merged_region_gene[merged_region_gene$z_ldmismatch != merged_region_gene$z_regionmerge,],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with different z before / after removing the problematic SNPs'),options = list(pageLength = 10) ) -->
<!-- region_id <- "16_3951195_5068344" -->
<!-- print("locus plot -- after region merge") -->
<!-- make_locusplot(finemap_res_rm, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_origin, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- print("locus plot -- LD mismatch: no LD") -->
<!-- make_locusplot(finemap_res_ldmm_nold, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_origin, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- print("locus plot -- LD mismatch: snp removed") -->
<!-- make_locusplot(finemap_res_ldmm_removesnp, -->
<!--                region_id = region_id, -->
<!--                ens_db = ens_db, -->
<!--                weights = weights_updated, -->
<!--                highlight_pip = 0.8, -->
<!--                filter_protein_coding_genes = TRUE, -->
<!--                filter_cs = TRUE, -->
<!--                color_pval_by = "cs", -->
<!--                color_pip_by = "cs",panel.heights = c(4,4,1,1)) -->
<!-- finemap_res_rm_gene_region <- finemap_res_rm_gene[finemap_res_rm_gene$region_id == region_id,] -->
<!-- finemap_res_ldmm_removesnp_gene_region <- finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$region_id == region_id,] -->
<!-- merged_region_gene <- merge(finemap_res_rm_gene_region,finemap_res_ldmm_removesnp_gene_region,by = "id") -->
<!-- merged_region_gene <- merged_region_gene[,c("id","gene_name.x","z.x","susie_pip.x","cs.x","z.y","susie_pip.y","cs.y")] -->
<!-- colnames(merged_region_gene) <- c("id","gene_name","z_regionmerge","susie_pip_regionmerge","cs_regionmerge","z_ldmismatch","susie_pip_ldmismatch","cs_ldmismatch") -->
<!-- ggplot(data = merged_region_gene, aes(x= z_regionmerge, y= z_ldmismatch)) +  -->
<!--   geom_point() + -->
<!--   ggtitle("Comparing z-scores before/after removing the problematic SNPs") + -->
<!--   theme_minimal() -->
<!-- DT::datatable(merged_region_gene[merged_region_gene$z_ldmismatch != merged_region_gene$z_regionmerge,],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with different z before / after removing the problematic SNPs'),options = list(pageLength = 10) ) -->
<!-- ``` -->
</div>
</div>
<div id="scz-ieu-b-5102" class="section level1">
<h1>SCZ-ieu-b-5102</h1>
<pre class="r"><code>trait &lt;- &quot;SCZ-ieu-b-5102&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge-4" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-73d12cd98287c4e0164c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-73d12cd98287c4e0164c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["ENSG00000086189|Heart_Left_Ventricle_stQTL","ENSG00000106268|Spleen_stQTL","ENSG00000110583.grp_2.downstream.ENST00000542163|Spleen_stQTL","ENSG00000134569|Brain_Hippocampus_stQTL","ENSG00000146966|Spleen_stQTL","ENSG00000148356|Brain_Spinal_cord_cervical_c-1_stQTL","ENSG00000149179.grp_2.downstream.ENST00000528488|Adrenal_Gland_stQTL","ENSG00000166888.grp_2.downstream.ENST00000300134|Brain_Hippocampus_stQTL","ENSG00000178149|Brain_Hippocampus_stQTL","ENSG00000183066.grp_2.downstream.ENST00000436265|Adrenal_Gland_stQTL","ENSG00000204632.grp_1.downstream.ENST00000360323|Adrenal_Gland_stQTL","ENSG00000206503|Spleen_stQTL","ENSG00000239521.grp_2.downstream.ENST00000414739|Brain_Hippocampus_stQTL","ENSG00000242866.grp_2.downstream.ENST00000450892|Spleen_stQTL","intron_6_31463279_31463371|Heart_Left_Ventricle_sQTL"],[0.600420224101996,0.9949345609969291,0.7438515035211307,0.9999999999999998,0.9950212906596942,0.6374366777132472,0.979156127243807,0.5629172784568016,0.6821721739112203,0.6997564305782613,1,1,0.9998536661766755,0.7611410134715674,0.9996055652852529],["L1","L2","L1","L1","L1","L1","L3","L1","L1","L1","L3","L2","L1","L1","L2"],[0.09181788877586428,2.352658602355895e-05,0.3101597470751897,0.0002782331469236565,0.02779865198913489,0.1337243646419074,2.798361542488692e-10,0.008811468940875145,0.001577978094805865,0.006931035264025498,0,0,0.008163793885427983,1.850111845302926e-05,0],["L1",null,null,null,"L1","L1",null,null,"L1",null,null,null,"L1",null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch-4" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis-4" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold02)) {
  load(file_pipthreshold02)
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_&quot;, trait, &quot;.rdata&quot;))
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold05)) {
  load(file_pipthreshold05)
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_&quot;, trait, &quot;.rdata&quot;))
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}


results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-0796cbf5802287e6981e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0796cbf5802287e6981e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[95,95],[0,0],[0,0],[345,345],[4,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods-4" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {

  load(file_pipthreshold02)

  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res

  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
  p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()
  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{

  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)

}</code></pre>
<pre><code>[1] &quot;There&#39;s no problematic genes, no need to compare&quot;</code></pre>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip-4" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)

  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<pre><code>[1] &quot;There&#39;s no problematic genes&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-27-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<pre><code>[1] &quot;There&#39;s no problematic genes&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-27-2.png" width="1632" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="wbc-ieu-b-30" class="section level1">
<h1>WBC-ieu-b-30</h1>
<pre class="r"><code>trait &lt;- &quot;WBC-ieu-b-30&quot;

results_dir_origin &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/&quot;,trait,&quot;/&quot;)
ctwas_res_origin &lt;- readRDS(paste0(results_dir_origin,trait,&quot;.ctwas.res.RDS&quot;))

finemap_res_origin &lt;- ctwas_res_origin$finemap_res</code></pre>
<div id="region-merge-5" class="section level2">
<h2>Region merge</h2>
<pre class="r"><code>load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_&quot;,trait,&quot;.rdata&quot;))

finemap_res_rm &lt;- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes &lt;- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip &lt;- finemap_res_rm_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]


finemap_res_origin_boundary_genes &lt;- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip &lt;- finemap_res_origin_boundary_genes[,c(&quot;id&quot;,&quot;susie_pip&quot;,&quot;cs&quot;)]

finemap_res_compare_regionmerge &lt;- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = &quot;id&quot;)
colnames(finemap_res_compare_regionmerge) &lt;- c(&quot;id&quot;,&quot;susie_pip_origin&quot;,&quot;cs_origin&quot;,&quot;susie_pip_reginmerge&quot;,&quot;cs_reginmerge&quot;)

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Selected boundary genes (susie_pip &gt; 0.5)&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-35c46d2ddbc1fab8b4dc" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-35c46d2ddbc1fab8b4dc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Selected boundary genes (susie_pip &gt; 0.5)<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48"],["ENSG00000065413.18|Adipose_Subcutaneous_eQTL","ENSG00000066056.13|Whole_Blood_eQTL","ENSG00000071054.grp_1.downstream.ENST00000427603|Whole_Blood_stQTL","ENSG00000075426|Cells_Cultured_fibroblasts_stQTL","ENSG00000081665.grp_2.downstream.ENST00000545006|Esophagus_Muscularis_stQTL","ENSG00000081665|Thyroid_stQTL","ENSG00000082153|Cells_Cultured_fibroblasts_stQTL","ENSG00000090382|Whole_Blood_stQTL","ENSG00000100304|Thyroid_stQTL","ENSG00000101425.12|Esophagus_Muscularis_eQTL","ENSG00000104643|Cells_Cultured_fibroblasts_stQTL","ENSG00000107798.17|Whole_Blood_eQTL","ENSG00000107882|Cells_Cultured_fibroblasts_stQTL","ENSG00000130589|Whole_Blood_stQTL","ENSG00000137413.grp_1.downstream.ENST00000472818|Whole_Blood_stQTL","ENSG00000138735|Thyroid_stQTL","ENSG00000146374|Esophagus_Muscularis_stQTL","ENSG00000148795|Thyroid_stQTL","ENSG00000152784.15|Adipose_Subcutaneous_eQTL","ENSG00000152784.15|Thyroid_eQTL","ENSG00000157617.16|Cells_Cultured_fibroblasts_eQTL","ENSG00000160007|Thyroid_stQTL","ENSG00000162885|Thyroid_stQTL","ENSG00000163959|Thyroid_stQTL","ENSG00000164096|Cells_Cultured_fibroblasts_stQTL","ENSG00000164403.grp_2.downstream.ENST00000606676|Whole_Blood_stQTL","ENSG00000166225|Esophagus_Muscularis_stQTL","ENSG00000166579.grp_1.downstream.ENST00000582812|Whole_Blood_stQTL","ENSG00000167703|Cells_Cultured_fibroblasts_stQTL","ENSG00000171206|Cells_Cultured_fibroblasts_stQTL","ENSG00000171840|Whole_Blood_stQTL","ENSG00000177646|Esophagus_Muscularis_stQTL","ENSG00000178397|Cells_Cultured_fibroblasts_stQTL","ENSG00000186417|Cells_Cultured_fibroblasts_stQTL","ENSG00000187775|Esophagus_Muscularis_stQTL","ENSG00000197062.grp_1.downstream.ENST00000623276|Thyroid_stQTL","ENSG00000198315|Thyroid_stQTL","ENSG00000204257|Esophagus_Muscularis_stQTL","ENSG00000204287|Adipose_Subcutaneous_stQTL","ENSG00000204387|Whole_Blood_stQTL","ENSG00000204642.grp_1.downstream.ENST00000465459|Thyroid_stQTL","ENSG00000204642.grp_2.downstream.ENST00000465459|Thyroid_stQTL","ENSG00000206503|Whole_Blood_stQTL","ENSG00000237441|Thyroid_stQTL","ENSG00000242574|Thyroid_stQTL","ENSG00000243335.grp_1.downstream.ENST00000640234|Adipose_Subcutaneous_stQTL","ENSG00000243335|Esophagus_Muscularis_stQTL","ENSG00000267795.grp_1.downstream.ENST00000591004|Whole_Blood_stQTL"],[0.9999915952546875,0.9202233211459653,0.9997435263013689,0.9999999762646118,0.501060465433235,1,0.9997303723671751,0.9999999999898939,0.556866053415955,0.9979142677527093,1,0.9590292203386498,0.9999992899366762,0.9859538784519345,0.9900277591435825,1,1,0.9999987743082592,1,0.9999999999930851,0.9626421743206931,0.9887526058114098,1,0.9476618577932618,1,1,0.9961305360217276,0.9453277514839149,1,0.999999215478263,0.976319136431963,0.9958462077514962,1,1,1,1,0.9989760166537506,0.9999740880636784,1,1,1,1,1,1,1,0.9999999999999996,1,0.9689004485026682],["L3","L1","L2","L1","L4","L2","L1","L2","L1","L1","L2","L2","L3","L1","L3","L1","L2","L2","L1","L2","L2","L3","L3","L1","L1","L1","L1","L1","L4","L1","L1","L2","L1","L3","L3","L4","L5","L5","L2","L1","L3","L1","L2","L3","L1","L3","L2","L3"],[0.0001628612194669588,0.0750113635686912,0.0008928216575725401,0,9.264705377320936e-07,5.479844908840015e-06,6.363625548733154e-08,1.848887831723545e-08,0.1259733057416983,4.331669274260097e-05,8.694060227298905e-08,0.2446379727694531,5.440092820663267e-15,0.2286822588074312,0.7497645032668033,0.9998018762997448,9.459100169806334e-14,1.185884723753361e-11,2.441031821120987e-10,1.099041968544157e-10,0.1003100803195184,0.1929331481473338,1.054621275597967e-05,0.005943657593373786,1.892706765960206e-05,1.884912559368956e-09,0.002784145563732965,0.0008762217309377096,1.500242152729925e-10,0.007853246390417401,0.6798061254854602,3.323217256934186e-10,0.0005376924001871952,0,0.0001428333996236208,0,0,0,0,1,0,0,0,0,0,0.004280762817373662,0,0.7552759461252312],[null,"L3",null,null,null,null,null,null,"L1",null,null,"L2",null,"L1",null,"L1",null,null,null,null,null,null,null,null,null,null,null,null,null,null,"L1",null,null,null,null,null,null,null,null,"L1",null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>susie_pip_origin<\/th>\n      <th>cs_origin<\/th>\n      <th>susie_pip_reginmerge<\/th>\n      <th>cs_reginmerge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ld-mismatch-5" class="section level2">
<h2>LD-mismatch</h2>
<div id="diagnosis-5" class="section level3">
<h3>Diagnosis</h3>
<pre class="r"><code>file_pipthreshold02 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold02)) {
  load(file_pipthreshold02)
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres02_&quot;, trait, &quot;.rdata&quot;))
  pip_02 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.2&quot;,
  &quot;Number of Selected Regions Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}

file_pipthreshold05 &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_nozfilter_&quot;, trait, &quot;.rdata&quot;)

if (file.exists(file_pipthreshold05)) {
  load(file_pipthreshold05)
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = length(problematic_genes),
  &quot;Number of Problematic Regions&quot; = length(problematic_region_ids),
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)
}else{

  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_diagnosis_pipthres05_&quot;, trait, &quot;.rdata&quot;))
  pip_05 &lt;- data.frame(
  &quot;PIP Threshold&quot; = &quot;0.5&quot;,
  &quot;Number of Selected Regions&quot; = length(selected_region_ids),
  &quot;Number of Problematic Genes&quot; = 0,
  &quot;Number of Problematic Regions&quot; = 0,
  &quot;Number of Problematic SNPs&quot; = length(res_ldmismatch$problematic_snps),
  &quot;Number of Flipped SNPs&quot; = length(res_ldmismatch$flipped_snps)
)

}


results_table &lt;- rbind(pip_02, pip_05)

DT::datatable(results_table,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;LD mismatch diagnosis table for different gene cutoff&#39;),options = list(pageLength = 10) )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-2a1b6e0e6b8a75d77da4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2a1b6e0e6b8a75d77da4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">LD mismatch diagnosis table for different gene cutoff<\/caption>","data":[["1","2"],["0.2","0.5"],[357,357],[4,3],[3,3],[1473,1473],[25,25]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PIP.Threshold<\/th>\n      <th>Number.of.Selected.Regions<\/th>\n      <th>Number.of.Problematic.Genes<\/th>\n      <th>Number.of.Problematic.Regions<\/th>\n      <th>Number.of.Problematic.SNPs<\/th>\n      <th>Number.of.Flipped.SNPs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="comparing-2-ld-mismatch-fixing-methods-5" class="section level3">
<h3>Comparing 2 LD mismatch fixing methods</h3>
<pre class="r"><code>file_ldmismatch_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_nold_nozfilter_&quot;,trait,&quot;.rdata&quot;)

if(file.exists(file_ldmismatch_results)) {

  load(file_pipthreshold02)

  load(file_ldmismatch_results)
  finemap_res_ldmm_nold &lt;- res_ldmm_nold$finemap_res
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  finemap_res_ldmm_removesnp &lt;- res_ldmm_removesnp$finemap_res

  finemap_res_ldmm_nold_problematic_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_nold$type != &quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_problematic_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$region_id %in% problematic_region_ids &amp; finemap_res_ldmm_removesnp$type != &quot;SNP&quot;,]

  merge_2method &lt;- merge(finemap_res_ldmm_nold_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_2method$highlight &lt;- ifelse(merge_2method$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_2method$susie_pip.y[is.na(merge_2method$susie_pip.y)] &lt;- 1.5
  p1 &lt;- ggplot(data = merge_2method, aes(x = susie_pip.x, y = susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    labs(x = &quot;PIP_noLD&quot;, y = &quot;PIP_removesnp&quot;) + 
    geom_abline(slope = 1, intercept = 0, col = &quot;red&quot;) + 
    ggtitle(&quot;Problematic regions only, genes only&quot;) +
    theme_minimal()

  finemap_res_rm_problematic_gene &lt;- finemap_res_rm[finemap_res_rm$region_id %in% problematic_region_ids &amp; finemap_res_rm$type != &quot;SNP&quot;,]

  merge_rm_ldmm_nold &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_nold_problematic_gene, by =&quot;id&quot;,all.x=T)
  merge_rm_ldmm_nold$highlight &lt;- ifelse(merge_rm_ldmm_nold$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_nold$susie_pip.y[is.na(merge_rm_ldmm_nold$susie_pip.y)] &lt;- 1.5
  p2 &lt;- ggplot(data = merge_rm_ldmm_nold, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_noLD&quot;) + 
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +  
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()

  merge_rm_ldmm_removesnp &lt;-  merge(finemap_res_rm_problematic_gene,finemap_res_ldmm_removesnp_problematic_gene, by =&quot;id&quot;,all.x =T)
  merge_rm_ldmm_removesnp$highlight &lt;- ifelse(merge_rm_ldmm_removesnp$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)
  merge_rm_ldmm_removesnp$susie_pip.y[is.na(merge_rm_ldmm_removesnp$susie_pip.y)] &lt;- 1.5
  p3 &lt;- ggplot(data = merge_rm_ldmm_removesnp, aes(x= susie_pip.x, y= susie_pip.y, color = highlight, alpha = highlight)) + 
    geom_point() +
    labs(x=&quot;PIP_after_regionmerge&quot;, y=&quot;PIP_removesnp&quot;) + 
    scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.1)) +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    geom_abline(slope = 1, intercept = 0, col =&quot;red&quot;) + 
    ggtitle(&quot;problematic regions only, genes only&quot;) +
    theme_minimal()
  print(sprintf(&quot;Total genes in problematic regions = %s&quot;,nrow(merge_rm_ldmm_removesnp)))
  print(sprintf(&quot;Number of genes disappeared after removing prblematic SNPs = %s&quot;, sum(merge_rm_ldmm_removesnp$susie_pip.y == 1.5)))
  print(&quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;)
  grid.arrange(p1,p2,p3, ncol = 3)



}else{

  print(&quot;There&#39;s no problematic genes, no need to compare&quot;)

}</code></pre>
<pre><code>[1] &quot;Total genes in problematic regions = 1309&quot;
[1] &quot;Number of genes disappeared after removing prblematic SNPs = 6&quot;
[1] &quot;The dots showing PIP =1.5 means: these genes were removed since the only QTLs of them are problematic&quot;</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-31-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="comparing-z-scores-and-susie_pip-5" class="section level2">
<h2>Comparing z-scores and susie_pip</h2>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]
  finemap_res_origin_gene$highlight &lt;- ifelse(finemap_res_origin_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]
  finemap_res_rm_gene$highlight &lt;- ifelse(finemap_res_rm_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()


  finemap_res_ldmm_nold_gene &lt;- finemap_res_ldmm_nold[finemap_res_ldmm_nold$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_nold_gene$highlight &lt;- ifelse(finemap_res_ldmm_nold_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
    theme_minimal()

  finemap_res_ldmm_removesnp_gene &lt;- finemap_res_ldmm_removesnp[finemap_res_ldmm_removesnp$type !=&quot;SNP&quot;,]
  finemap_res_ldmm_removesnp_gene$highlight &lt;- ifelse(finemap_res_ldmm_removesnp_gene$id %in% problematic_genes, &quot;problematic genes&quot;, &quot;good genes&quot;)

  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene, aes(x= abs(z), y= susie_pip, color = highlight)) +
    geom_point() +
    scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
    ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
    theme_minimal()


  grid.arrange(p1,p2,p3,p4, ncol = 4)

  print(&quot;L - estimated in region merge step&quot;)
  print(updated_data_res_regionmerge$updated_region_L[problematic_region_ids])
  load(paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/ldmismatch_pipthres02_removesnp_rescreenregion_nozfilter_&quot;,trait,&quot;.rdata&quot;))
  print(&quot;L - re-estimated after updating z_scores, region data&quot;)
  print(screen_res$screened_region_L)
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-32-1.png" width="1632" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;L - estimated in region merge step&quot;
2_217530757_219589829   5_68555033_71944629  19_43358303_44239955 
                    3                     1                     3 
[1] &quot;L - re-estimated after updating z_scores, region data&quot;
2_217530757_219589829   5_68555033_71944629  19_43358303_44239955 
                    4                     1                     4 </code></pre>
<pre class="r"><code>print(&quot;Zoom in the z&lt;15 part&quot;)</code></pre>
<pre><code>[1] &quot;Zoom in the z&lt;15 part&quot;</code></pre>
<pre class="r"><code>if(file.exists(file_ldmismatch_results)){
  

  finemap_res_origin_gene_prob &lt;- finemap_res_origin_gene[finemap_res_origin_gene$highlight == &quot;problematic genes&quot;,]
  p1 &lt;- ggplot(data = finemap_res_origin_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;Original ctwas results&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_rm_gene_prob &lt;- finemap_res_rm_gene[finemap_res_rm_gene$highlight == &quot;problematic genes&quot;,]
  p2 &lt;- ggplot(data = finemap_res_rm_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After region merge&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_nold_gene_prob &lt;- finemap_res_ldmm_nold_gene[finemap_res_ldmm_nold_gene$highlight == &quot;problematic genes&quot;,]
  p3 &lt;- ggplot(data = finemap_res_ldmm_nold_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- noLD&quot;) +
        theme_minimal() +
        xlim(0, 15)

  finemap_res_ldmm_removesnp_gene_prob &lt;-   finemap_res_ldmm_removesnp_gene[finemap_res_ldmm_removesnp_gene$highlight == &quot;problematic genes&quot;,]
  p4 &lt;- ggplot(data = finemap_res_ldmm_removesnp_gene_prob, 
        aes(x = abs(z), y = susie_pip, color = highlight, alpha = highlight)) + 
        geom_point() +
        scale_color_manual(values = c(&quot;problematic genes&quot; = &quot;red&quot;, &quot;good genes&quot; = &quot;black&quot;)) +
        scale_alpha_manual(values = c(&quot;problematic genes&quot; = 1, &quot;good genes&quot; = 0.01)) +
        ggtitle(&quot;After LD mismatch fixed -- SNP removed&quot;) +
        theme_minimal() +
        xlim(0, 15)
  
  grid.arrange(p1,p2,p3,p4, ncol = 4)
  
}else{

  print(&quot;There&#39;s no problematic genes&quot;)

  finemap_res_origin &lt;- ctwas_res_origin$finemap_res
  finemap_res_origin_gene &lt;- finemap_res_origin[finemap_res_origin$type != &quot;SNP&quot;,]

  p1 &lt;- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;Original ctwas results&quot;) +
    theme_minimal()+
    xlim(0, 15)


  finemap_res_rm_gene &lt;- finemap_res_rm[finemap_res_rm$type != &quot;SNP&quot;,]

  p2 &lt;- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
    geom_point() +
    ggtitle(&quot;After region merge&quot;) +
    theme_minimal()+
    xlim(0, 15)

  grid.arrange(p1,p2, ncol = 2)
}</code></pre>
<p><img src="figure/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.Rmd/unnamed-chunk-32-2.png" width="1632" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 8

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el8-x86_64/lib/libopenblas_skylakexp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] dplyr_1.1.2               gridExtra_2.3            
 [3] ggplot2_3.4.2             EnsDb.Hsapiens.v86_2.99.0
 [5] ensembldb_2.22.0          AnnotationFilter_1.22.0  
 [7] GenomicFeatures_1.50.4    AnnotationDbi_1.60.2     
 [9] Biobase_2.58.0            GenomicRanges_1.50.2     
[11] GenomeInfoDb_1.34.9       IRanges_2.32.0           
[13] S4Vectors_0.36.2          BiocGenerics_0.44.0      
[15] ctwas_0.4.20.9001        

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.38.0              locuszoomr_0.1.5           
  [7] fs_1.5.2                    rstudioapi_0.14            
  [9] farver_2.1.0                ggrepel_0.9.3              
 [11] DT_0.22                     bit64_4.0.5                
 [13] fansi_1.0.3                 xml2_1.3.3                 
 [15] codetools_0.2-18            logging_0.10-108           
 [17] cachem_1.0.6                knitr_1.42                 
 [19] jsonlite_1.8.7              workflowr_1.7.1            
 [21] Rsamtools_2.14.0            dbplyr_2.3.2               
 [23] png_0.1-7                   readr_2.1.4                
 [25] compiler_4.2.0              httr_1.4.7                 
 [27] Matrix_1.6-1.1              fastmap_1.1.0              
 [29] lazyeval_0.2.2              cli_3.6.2                  
 [31] later_1.3.0                 htmltools_0.5.7            
 [33] prettyunits_1.1.1           tools_4.2.0                
 [35] gtable_0.3.0                glue_1.6.2                 
 [37] GenomeInfoDbData_1.2.9      rappdirs_0.3.3             
 [39] Rcpp_1.0.11                 jquerylib_0.1.4            
 [41] vctrs_0.6.1                 Biostrings_2.66.0          
 [43] rtracklayer_1.58.0          crosstalk_1.2.0            
 [45] xfun_0.38                   stringr_1.5.0              
 [47] lifecycle_1.0.4             irlba_2.3.5                
 [49] restfulr_0.0.15             XML_3.99-0.9               
 [51] zlibbioc_1.44.0             scales_1.2.0               
 [53] gggrid_0.2-0                hms_1.1.3                  
 [55] promises_1.2.0.1            MatrixGenerics_1.10.0      
 [57] ProtGenerics_1.30.0         parallel_4.2.0             
 [59] SummarizedExperiment_1.28.0 LDlinkR_1.3.0              
 [61] yaml_2.3.5                  curl_4.3.2                 
 [63] memoise_2.0.1               sass_0.4.1                 
 [65] biomaRt_2.54.1              stringi_1.7.6              
 [67] RSQLite_2.3.1               highr_0.9                  
 [69] BiocIO_1.8.0                filelock_1.0.2             
 [71] BiocParallel_1.32.6         rlang_1.1.2                
 [73] pkgconfig_2.0.3             matrixStats_1.2.0          
 [75] bitops_1.0-7                evaluate_0.15              
 [77] lattice_0.20-45             purrr_1.0.1                
 [79] labeling_0.4.2              GenomicAlignments_1.34.1   
 [81] htmlwidgets_1.6.2           cowplot_1.1.1              
 [83] bit_4.0.4                   tidyselect_1.2.0           
 [85] magrittr_2.0.3              R6_2.5.1                   
 [87] generics_0.1.3              DelayedArray_0.24.0        
 [89] DBI_1.1.2                   pgenlibr_0.3.6             
 [91] pillar_1.9.0                whisker_0.4                
 [93] withr_2.5.0                 KEGGREST_1.38.0            
 [95] RCurl_1.98-1.12             mixsqp_0.3-48              
 [97] tibble_3.2.1                crayon_1.5.1               
 [99] utf8_1.2.2                  BiocFileCache_2.6.1        
[101] plotly_4.10.0               tzdb_0.3.0                 
[103] rmarkdown_2.21              progress_1.2.2             
[105] grid_4.2.0                  data.table_1.14.4          
[107] blob_1.2.3                  git2r_0.30.1               
[109] digest_0.6.29               tidyr_1.3.0                
[111] httpuv_1.6.5                munsell_0.5.0              
[113] viridisLite_0.4.0           bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
