<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-08-21" />

<title>PVE share pattern, tissue ranked by PHE</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">PVE share pattern, tissue ranked by
PHE</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-08-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreea2b260dd04dccad8a08e206e6e6ad7917d5c132btargetblanka2b260da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/a2b260dd04dccad8a08e206e6e6ad7917d5c132b" target="_blank">a2b260d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreea2b260dd04dccad8a08e206e6e6ad7917d5c132btargetblanka2b260da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/a2b260dd04dccad8a08e206e6e6ad7917d5c132b" target="_blank">a2b260d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    cv/
    Ignored:    figures/lz/IBD-ebi-a-GCST004131/single/
    Ignored:    figures/lz_ld/IBD-ebi-a-GCST004131/single/

Unstaged changes:
    Modified:   analysis/realdata_final_PVE_share_pattern_rankPHE.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/realdata_final_PVE_share_pattern_rankPHE.Rmd</code>) and
HTML (<code>docs/realdata_final_PVE_share_pattern_rankPHE.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/analysis/realdata_final_PVE_share_pattern_rankPHE.Rmd" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/fef2c966c1b38536af83c4909ba01818af93de90/docs/realdata_final_PVE_share_pattern_rankPHE.html" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/8996a4df3bcd8dc696e6be1244bd2f730aa1c74c/analysis/realdata_final_PVE_share_pattern_rankPHE.Rmd" target="_blank">8996a4d</a>
</td>
<td>
XSun
</td>
<td>
2025-08-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/13195f626640fc3b11bcbab08fa81fd269506f5d/analysis/realdata_final_PVE_share_pattern_rankPHE.Rmd" target="_blank">13195f6</a>
</td>
<td>
XSun
</td>
<td>
2025-08-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/13195f626640fc3b11bcbab08fa81fd269506f5d/docs/realdata_final_PVE_share_pattern_rankPHE.html" target="_blank">13195f6</a>
</td>
<td>
XSun
</td>
<td>
2025-08-21
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We ranked tissues by the PHE in single-tissue eQTL analyses and
selected the top 10.</p>
<p>From the 10 selected tissues, we obtained 45 unique tissue pairs. For
each pair, we</p>
<p>For each tissue pair <span class="math inline">\((A, B)\)</span>, we
first estimated the PHE of tissue <span class="math inline">\(A\)</span>
when analyzed alone (in single-tissue eQTL/sQTL analysis), and then
conducted both two-tissue eQTL and two-tissue sQTL analyses to estimate
the PHE of each tissue in the joint analysis.</p>
<p>For tissue A, <span
class="math display">\[\text{PHE}_{\text{single}}\]</span> is the PHE
from single tissue analysis, <span
class="math display">\[\text{PHE}_{\text{joint}}\]</span> is the PHE
from 2 tissue joint analysis, we then calculated the relative change
as:</p>
<p><span class="math display">\[
\frac{\text{PHE}_{\text{single}} -
\text{PHE}_{\text{joint}}}{\text{PHE}_{\text{single}}}
\]</span></p>
<pre class="r"><code>library(ctwas)
library(ComplexHeatmap)
library(grid)
library(circlize)

trait_nopsy &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;IBD-ebi-a-GCST004131&quot;,&quot;aFib-ebi-a-GCST006414&quot;,&quot;SBP-ukb-a-360&quot;,
                 &quot;T1D-GCST90014023&quot;,&quot;T2D-panukb&quot;,&quot;ATH_gtexukb&quot;,&quot;BMI-panukb&quot;,&quot;HB-panukb&quot;,
                 &quot;Height-panukb&quot;,&quot;HTN-panukb&quot;,&quot;PLT-panukb&quot;,&quot;RA-panukb&quot;,&quot;RBC-panukb&quot;,
                 &quot;WBC-ieu-b-30&quot;
                 )
trait_psy &lt;- c(&quot;SCZ-ieu-b-5102&quot;,&quot;BIP-ieu-b-5110&quot;,&quot;MDD-ieu-b-102&quot;,&quot;PD-ieu-b-7&quot;,
               &quot;NS-ukb-a-230&quot;,&quot;ASD-ieu-a-1185&quot;,&quot;ADHD-ieu-a-1183&quot;)
traits &lt;- c(trait_nopsy,trait_psy)
#

#traits &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;IBD-ebi-a-GCST004131&quot;,&quot;SCZ-ieu-b-5102&quot;)


source(&quot;/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R&quot;)

get_top_tissues &lt;- function(trait, n_top = 10, folder_results) {
 
  
  # Find files matching pattern
  finalfiles &lt;- list.files(folder_results, pattern = &quot;_csinclude&quot;)
  
  # Collect tissue summaries
  trait_sum &lt;- do.call(rbind, lapply(finalfiles, function(file) {
    gene_pip &lt;- readRDS(file.path(folder_results, file))
    
    tissue &lt;- gsub(pattern = paste0(trait, &quot;_&quot;), replacement = &quot;&quot;, x = file)
    tissue &lt;- gsub(pattern = &quot;.combined_pip_bygroup_final_csinclude.RDS&quot;, replacement = &quot;&quot;, x = tissue)
    
    data.frame(tissue = tissue,
               num_gene_pip08 = sum(gene_pip$combined_pip &gt; 0.8),
               stringsAsFactors = FALSE)
  }))
  
  # Filter, order, and select top tissues
  
  # trait_sum &lt;- trait_sum[trait_sum$num_gene_pip08 &gt; 0, ]
  trait_sum &lt;- trait_sum[order(trait_sum$num_gene_pip08, decreasing = TRUE), ]
  
  # Return top tissues
  head(trait_sum$tissue, n_top)
}

get_top_tissues_phe &lt;- function(trait, n_top = 10, folder_results, gwas_n) {
  
  # Find files matching pattern
  finalfiles &lt;- list.files(folder_results, pattern = &quot;param.RDS&quot;)
  
  # Collect tissue summaries
  trait_sum &lt;- do.call(rbind, lapply(finalfiles, function(file) {
    param_single &lt;- readRDS(file.path(folder_results, file))
    ctwas_parameters_single &lt;- summarize_param(param_single, gwas_n, enrichment_test = &quot;fisher&quot;)
    phe &lt;- ctwas_parameters_single$prop_heritability[1]
    
    tissue &lt;- gsub(pattern = paste0(trait, &quot;_&quot;), replacement = &quot;&quot;, x = file)
    tissue &lt;- gsub(pattern = &quot;.thin1.shared_all.param.RDS&quot;, replacement = &quot;&quot;, x = tissue)
    
    data.frame(tissue = tissue,
               PHE_single_eQTL = phe,
               stringsAsFactors = FALSE)
  }))
  
  # Order and select top tissues
  trait_sum &lt;- trait_sum[order(trait_sum$PHE_single_eQTL, decreasing = TRUE), ]
  
  # Return top tissues
  head(trait_sum$tissue, n_top)
}</code></pre>
</div>
<div id="tissue-ranking" class="section level1">
<h1>Tissue ranking</h1>
<pre class="r"><code>for (trait  in traits){

  print(trait)

  gwas_n &lt;- samplesize[trait]

  folder_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/&quot;,trait,&quot;/&quot;)

  top_tissues &lt;- get_top_tissues(
    trait = trait,
    n_top = 10,
    folder_results = folder_results
  )


  top_tissues_phe &lt;- get_top_tissues_phe(trait,
                                     n_top = 10,
                                     folder_results,
                                     gwas_n)

  tissues &lt;- cbind( top_tissues, top_tissues_phe)
  colnames(tissues) &lt;- c(&quot;by gene number&quot;, &quot;by PHE&quot;)
  tissues &lt;- as.data.frame(tissues)
  print(tissues)
}</code></pre>
<pre><code>[1] &quot;LDL-ukb-d-30780_irnt&quot;
                        by gene number                              by PHE
1                                Liver                               Liver
2                     Esophagus_Mucosa                                Lung
3  Esophagus_Gastroesophageal_Junction                         Whole_Blood
4                               Spleen                    Esophagus_Mucosa
5                          Whole_Blood Esophagus_Gastroesophageal_Junction
6             Adipose_Visceral_Omentum                              Spleen
7                 Adipose_Subcutaneous                Adipose_Subcutaneous
8                        Artery_Tibial                Esophagus_Muscularis
9                Breast_Mammary_Tissue              Heart_Atrial_Appendage
10              Heart_Atrial_Appendage     Skin_Not_Sun_Exposed_Suprapubic
[1] &quot;IBD-ebi-a-GCST004131&quot;
               by gene number                          by PHE
1        Adipose_Subcutaneous                     Whole_Blood
2  Cells_Cultured_fibroblasts Skin_Not_Sun_Exposed_Suprapubic
3            Esophagus_Mucosa      Cells_Cultured_fibroblasts
4                   Pituitary                Esophagus_Mucosa
5        Heart_Left_Ventricle            Adipose_Subcutaneous
6                      Testis                          Spleen
7               Colon_Sigmoid                       Pituitary
8        Esophagus_Muscularis                            Lung
9      Heart_Atrial_Appendage  Small_Intestine_Terminal_Ileum
10            Muscle_Skeletal            Heart_Left_Ventricle
[1] &quot;aFib-ebi-a-GCST006414&quot;
                by gene number                               by PHE
1       Heart_Atrial_Appendage               Heart_Atrial_Appendage
2         Heart_Left_Ventricle                 Heart_Left_Ventricle
3                Artery_Tibial                 Esophagus_Muscularis
4             Brain_Cerebellum                        Artery_Tibial
5   Cells_Cultured_fibroblasts                         Artery_Aorta
6         Esophagus_Muscularis                      Muscle_Skeletal
7                 Artery_Aorta                              Thyroid
8  Brain_Cerebellar_Hemisphere           Cells_Cultured_fibroblasts
9              Muscle_Skeletal Brain_Anterior_cingulate_cortex_BA24
10                   Pituitary           Skin_Sun_Exposed_Lower_leg
[1] &quot;SBP-ukb-a-360&quot;
             by gene number                     by PHE
1             Artery_Tibial               Artery_Aorta
2              Artery_Aorta              Artery_Tibial
3           Muscle_Skeletal            Muscle_Skeletal
4                  Pancreas                     Spleen
5             Adrenal_Gland              Adrenal_Gland
6                    Spleen Skin_Sun_Exposed_Lower_leg
7      Adipose_Subcutaneous                    Thyroid
8  Adipose_Visceral_Omentum     Heart_Atrial_Appendage
9             Colon_Sigmoid               Brain_Cortex
10     Esophagus_Muscularis       Minor_Salivary_Gland
[1] &quot;T1D-GCST90014023&quot;
                         by gene number                              by PHE
1              Adipose_Visceral_Omentum                         Whole_Blood
2                          Artery_Aorta                                Lung
3                         Artery_Tibial   Cells_EBV-transformed_lymphocytes
4  Brain_Anterior_cingulate_cortex_BA24                        Artery_Aorta
5                              Pancreas      Small_Intestine_Terminal_Ileum
6       Skin_Not_Sun_Exposed_Suprapubic                            Pancreas
7                                Spleen                        Nerve_Tibial
8                           Whole_Blood          Skin_Sun_Exposed_Lower_leg
9                  Adipose_Subcutaneous Esophagus_Gastroesophageal_Junction
10             Brain_Frontal_Cortex_BA9                       Artery_Tibial
[1] &quot;T2D-panukb&quot;
                by gene number                              by PHE
1         Adipose_Subcutaneous                       Artery_Tibial
2              Muscle_Skeletal                             Thyroid
3                       Testis                           Pituitary
4                 Artery_Aorta                        Artery_Aorta
5  Brain_Caudate_basal_ganglia                     Muscle_Skeletal
6             Brain_Cerebellum                              Testis
7       Heart_Atrial_Appendage                Adipose_Subcutaneous
8                    Pituitary              Heart_Atrial_Appendage
9     Adipose_Visceral_Omentum Esophagus_Gastroesophageal_Junction
10               Artery_Tibial                             Stomach
[1] &quot;ATH_gtexukb&quot;
               by gene number                              by PHE
1       Breast_Mammary_Tissue                              Spleen
2            Colon_Transverse                    Colon_Transverse
3        Heart_Left_Ventricle                                Lung
4                      Spleen                Esophagus_Muscularis
5        Esophagus_Muscularis              Heart_Atrial_Appendage
6                    Pancreas                Heart_Left_Ventricle
7  Skin_Sun_Exposed_Lower_leg                       Artery_Tibial
8                     Thyroid                            Pancreas
9    Adipose_Visceral_Omentum Esophagus_Gastroesophageal_Junction
10               Artery_Aorta          Skin_Sun_Exposed_Lower_leg
[1] &quot;BMI-panukb&quot;
                by gene number                      by PHE
1         Adipose_Subcutaneous             Muscle_Skeletal
2             Brain_Cerebellum            Brain_Cerebellum
3   Cells_Cultured_fibroblasts  Cells_Cultured_fibroblasts
4                Artery_Tibial               Artery_Tibial
5                 Artery_Aorta Brain_Caudate_basal_ganglia
6              Muscle_Skeletal        Adipose_Subcutaneous
7  Brain_Caudate_basal_ganglia                Brain_Cortex
8  Brain_Cerebellar_Hemisphere                     Stomach
9                       Testis Brain_Cerebellar_Hemisphere
10    Adipose_Visceral_Omentum                Artery_Aorta
[1] &quot;HB-panukb&quot;
                        by gene number                              by PHE
1           Cells_Cultured_fibroblasts                Adipose_Subcutaneous
2                 Adipose_Subcutaneous          Cells_Cultured_fibroblasts
3                Breast_Mammary_Tissue              Heart_Atrial_Appendage
4             Adipose_Visceral_Omentum               Breast_Mammary_Tissue
5                        Artery_Tibial                       Artery_Tibial
6               Heart_Atrial_Appendage Esophagus_Gastroesophageal_Junction
7                              Stomach            Adipose_Visceral_Omentum
8  Esophagus_Gastroesophageal_Junction     Skin_Not_Sun_Exposed_Suprapubic
9                          Whole_Blood                             Stomach
10     Skin_Not_Sun_Exposed_Suprapubic                     Muscle_Skeletal
[1] &quot;Height-panukb&quot;
                    by gene number                          by PHE
1       Cells_Cultured_fibroblasts                           Ovary
2                    Artery_Tibial      Cells_Cultured_fibroblasts
3                        Pituitary        Adipose_Visceral_Omentum
4         Adipose_Visceral_Omentum                   Artery_Tibial
5                          Stomach                        Prostate
6             Adipose_Subcutaneous            Heart_Left_Ventricle
7  Skin_Not_Sun_Exposed_Suprapubic          Heart_Atrial_Appendage
8                     Artery_Aorta Skin_Not_Sun_Exposed_Suprapubic
9                            Ovary                     Whole_Blood
10      Skin_Sun_Exposed_Lower_leg                    Artery_Aorta
[1] &quot;HTN-panukb&quot;
               by gene number                          by PHE
1                     Thyroid        Adipose_Visceral_Omentum
2      Heart_Atrial_Appendage            Adipose_Subcutaneous
3                Nerve_Tibial          Heart_Atrial_Appendage
4  Skin_Sun_Exposed_Lower_leg                          Testis
5                     Stomach Skin_Not_Sun_Exposed_Suprapubic
6                      Testis                            Lung
7               Artery_Tibial                 Muscle_Skeletal
8        Heart_Left_Ventricle            Esophagus_Muscularis
9                      Spleen      Skin_Sun_Exposed_Lower_leg
10       Adipose_Subcutaneous                         Stomach
[1] &quot;PLT-panukb&quot;
               by gene number                          by PHE
1                        Lung                          Spleen
2                      Spleen                     Whole_Blood
3                     Thyroid                            Lung
4                 Whole_Blood  Small_Intestine_Terminal_Ileum
5        Adipose_Subcutaneous                         Thyroid
6       Breast_Mammary_Tissue                Esophagus_Mucosa
7            Esophagus_Mucosa           Breast_Mammary_Tissue
8            Colon_Transverse            Adipose_Subcutaneous
9  Skin_Sun_Exposed_Lower_leg Skin_Not_Sun_Exposed_Suprapubic
10              Colon_Sigmoid                           Ovary
[1] &quot;RA-panukb&quot;
                         by gene number                          by PHE
1                Heart_Atrial_Appendage                            Lung
2                                  Lung                     Whole_Blood
3       Skin_Not_Sun_Exposed_Suprapubic                    Artery_Aorta
4                                Spleen                          Spleen
5  Brain_Anterior_cingulate_cortex_BA24                    Nerve_Tibial
6                  Esophagus_Muscularis          Heart_Atrial_Appendage
7                       Muscle_Skeletal            Esophagus_Muscularis
8                          Nerve_Tibial Skin_Not_Sun_Exposed_Suprapubic
9                             Pituitary            Adipose_Subcutaneous
10       Small_Intestine_Terminal_Ileum  Small_Intestine_Terminal_Ileum
[1] &quot;RBC-panukb&quot;
                    by gene number                          by PHE
1             Adipose_Subcutaneous            Adipose_Subcutaneous
2       Cells_Cultured_fibroblasts            Esophagus_Muscularis
3                      Whole_Blood           Breast_Mammary_Tissue
4                 Brain_Cerebellum                   Colon_Sigmoid
5                    Colon_Sigmoid Skin_Not_Sun_Exposed_Suprapubic
6  Skin_Not_Sun_Exposed_Suprapubic                     Whole_Blood
7                          Thyroid        Adipose_Visceral_Omentum
8       Skin_Sun_Exposed_Lower_leg      Skin_Sun_Exposed_Lower_leg
9                    Artery_Tibial                            Lung
10            Esophagus_Muscularis                         Thyroid
[1] &quot;WBC-ieu-b-30&quot;
                    by gene number                          by PHE
1                      Whole_Blood                     Whole_Blood
2             Adipose_Subcutaneous      Cells_Cultured_fibroblasts
3                     Nerve_Tibial                            Lung
4                             Lung Skin_Not_Sun_Exposed_Suprapubic
5            Breast_Mammary_Tissue                    Nerve_Tibial
6         Adipose_Visceral_Omentum            Adipose_Subcutaneous
7                           Spleen                   Artery_Tibial
8  Skin_Not_Sun_Exposed_Suprapubic                          Spleen
9                     Artery_Aorta      Skin_Sun_Exposed_Lower_leg
10                   Artery_Tibial                    Artery_Aorta
[1] &quot;SCZ-ieu-b-5102&quot;
                          by gene number                                by PHE
1            Brain_Cerebellar_Hemisphere                    Brain_Hypothalamus
2                       Brain_Cerebellum           Brain_Cerebellar_Hemisphere
3                           Brain_Cortex           Brain_Putamen_basal_ganglia
4                     Brain_Hypothalamus                     Brain_Hippocampus
5         Brain_Spinal_cord_cervical_c-1                          Brain_Cortex
6                      Brain_Hippocampus           Brain_Caudate_basal_ganglia
7  Brain_Nucleus_accumbens_basal_ganglia                      Brain_Cerebellum
8            Brain_Putamen_basal_ganglia              Brain_Frontal_Cortex_BA9
9                 Brain_Substantia_nigra Brain_Nucleus_accumbens_basal_ganglia
10                        Brain_Amygdala        Brain_Spinal_cord_cervical_c-1
[1] &quot;BIP-ieu-b-5110&quot;
                          by gene number                                by PHE
1            Brain_Caudate_basal_ganglia                 Breast_Mammary_Tissue
2            Brain_Cerebellar_Hemisphere           Brain_Caudate_basal_ganglia
3                           Brain_Cortex                    Brain_Hypothalamus
4                      Brain_Hippocampus Brain_Nucleus_accumbens_basal_ganglia
5                     Brain_Hypothalamus           Brain_Cerebellar_Hemisphere
6  Brain_Nucleus_accumbens_basal_ganglia              Brain_Frontal_Cortex_BA9
7            Brain_Putamen_basal_ganglia           Brain_Putamen_basal_ganglia
8                       Brain_Cerebellum                          Brain_Cortex
9               Brain_Frontal_Cortex_BA9                      Brain_Cerebellum
10        Brain_Spinal_cord_cervical_c-1                Brain_Substantia_nigra
[1] &quot;MDD-ieu-b-102&quot;
                          by gene number                                by PHE
1            Brain_Caudate_basal_ganglia Brain_Nucleus_accumbens_basal_ganglia
2                      Brain_Hippocampus                 Breast_Mammary_Tissue
3                         Brain_Amygdala           Brain_Caudate_basal_ganglia
4   Brain_Anterior_cingulate_cortex_BA24                      Brain_Cerebellum
5            Brain_Cerebellar_Hemisphere                     Brain_Hippocampus
6                       Brain_Cerebellum              Brain_Frontal_Cortex_BA9
7                           Brain_Cortex                          Brain_Cortex
8               Brain_Frontal_Cortex_BA9           Brain_Putamen_basal_ganglia
9                     Brain_Hypothalamus  Brain_Anterior_cingulate_cortex_BA24
10 Brain_Nucleus_accumbens_basal_ganglia           Brain_Cerebellar_Hemisphere
[1] &quot;PD-ieu-b-7&quot;
                          by gene number                                by PHE
1                     Brain_Hypothalamus                    Brain_Hypothalamus
2                         Brain_Amygdala                      Brain_Cerebellum
3                       Brain_Cerebellum                          Brain_Cortex
4                           Brain_Cortex                        Brain_Amygdala
5               Brain_Frontal_Cortex_BA9  Brain_Anterior_cingulate_cortex_BA24
6            Brain_Putamen_basal_ganglia                 Breast_Mammary_Tissue
7   Brain_Anterior_cingulate_cortex_BA24                     Brain_Hippocampus
8            Brain_Cerebellar_Hemisphere              Brain_Frontal_Cortex_BA9
9                      Brain_Hippocampus           Brain_Caudate_basal_ganglia
10 Brain_Nucleus_accumbens_basal_ganglia Brain_Nucleus_accumbens_basal_ganglia
[1] &quot;NS-ukb-a-230&quot;
                          by gene number                                by PHE
1            Brain_Cerebellar_Hemisphere                          Brain_Cortex
2            Brain_Caudate_basal_ganglia           Brain_Cerebellar_Hemisphere
3                       Brain_Cerebellum           Brain_Caudate_basal_ganglia
4                           Brain_Cortex Brain_Nucleus_accumbens_basal_ganglia
5                     Brain_Hypothalamus              Brain_Frontal_Cortex_BA9
6               Brain_Frontal_Cortex_BA9                      Brain_Cerebellum
7            Brain_Putamen_basal_ganglia                Brain_Substantia_nigra
8                 Brain_Substantia_nigra                    Brain_Hypothalamus
9  Brain_Nucleus_accumbens_basal_ganglia           Brain_Putamen_basal_ganglia
10        Brain_Spinal_cord_cervical_c-1                 Breast_Mammary_Tissue
[1] &quot;ASD-ieu-a-1185&quot;
                          by gene number                      by PHE
1                         Brain_Amygdala    Brain_Frontal_Cortex_BA9
2   Brain_Anterior_cingulate_cortex_BA24            Brain_Cerebellum
3            Brain_Caudate_basal_ganglia          Brain_Hypothalamus
4            Brain_Cerebellar_Hemisphere Brain_Caudate_basal_ganglia
5                       Brain_Cerebellum Brain_Cerebellar_Hemisphere
6                           Brain_Cortex           Brain_Hippocampus
7               Brain_Frontal_Cortex_BA9 Brain_Putamen_basal_ganglia
8                      Brain_Hippocampus      Brain_Substantia_nigra
9                     Brain_Hypothalamus                Brain_Cortex
10 Brain_Nucleus_accumbens_basal_ganglia       Breast_Mammary_Tissue
[1] &quot;ADHD-ieu-a-1183&quot;
                          by gene number                                by PHE
1                         Brain_Amygdala                 Breast_Mammary_Tissue
2   Brain_Anterior_cingulate_cortex_BA24        Brain_Spinal_cord_cervical_c-1
3            Brain_Caudate_basal_ganglia                          Brain_Cortex
4            Brain_Cerebellar_Hemisphere  Brain_Anterior_cingulate_cortex_BA24
5                       Brain_Cerebellum                        Brain_Amygdala
6                           Brain_Cortex           Brain_Caudate_basal_ganglia
7               Brain_Frontal_Cortex_BA9 Brain_Nucleus_accumbens_basal_ganglia
8                      Brain_Hippocampus                    Brain_Hypothalamus
9                     Brain_Hypothalamus           Brain_Putamen_basal_ganglia
10 Brain_Nucleus_accumbens_basal_ganglia                Brain_Substantia_nigra</code></pre>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<pre class="r"><code>#trait &lt;- &quot;LDL-ukb-d-30780_irnt&quot;


for (trait  in traits){
  
  print(trait)
  
  gwas_n &lt;- samplesize[trait]
  
  folder_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/&quot;,trait,&quot;/&quot;)
  
  top_tissues_phe &lt;- get_top_tissues_phe(trait,
                                     n_top = 10,
                                     folder_results,
                                     gwas_n)
  
  
  ## single tissue eQTL
  prob_pve_alltissue &lt;- c()
  for (tissue in top_tissues_phe){
    
    file_param_single &lt;- paste0(folder_results,&quot;/&quot;,trait,&quot;_&quot;,tissue,&quot;.thin1.shared_all.param.RDS&quot;)
    param_single &lt;- readRDS(file_param_single)
    
    ctwas_parameters_single &lt;- summarize_param(param_single, gwas_n, enrichment_test = &quot;fisher&quot;)
    
    prob_pve &lt;- ctwas_parameters_single$prop_heritability[1]
    prob_pve_alltissue &lt;- c(prob_pve_alltissue, prob_pve)
    
  }
  
  names(prob_pve_alltissue) &lt;- gsub(pattern = &quot;\\|eQTL&quot;,x = names(prob_pve_alltissue), replacement = &quot;&quot;)
  
  
  ## pairwise
  
  tissue_combination &lt;- combn(top_tissues_phe, 2, simplify = FALSE)
  
  mat &lt;- matrix(NA, nrow = length(top_tissues_phe), ncol = length(top_tissues_phe),
                dimnames = list(top_tissues_phe, top_tissues_phe))
  
  for (i in 1:length(tissue_combination)){
    
    tissue1 &lt;- tissue_combination[[i]][1]
    pve_tissue1 &lt;- prob_pve_alltissue[tissue1]
    tissue2 &lt;- tissue_combination[[i]][2]
    pve_tissue2 &lt;- prob_pve_alltissue[tissue2]
    
    tissue_pair &lt;- paste0(tissue_combination[[i]], collapse = &quot;-&quot;)
    
    file_param_pair &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/pairwise_snakemake_outputs_byPHE/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,tissue_pair,&quot;.eqtlonly.thin1.shared_all.param.RDS&quot;)
    param_pair &lt;- readRDS(file_param_pair)
    ctwas_parameters_pair &lt;- summarize_param(param_pair, gwas_n, enrichment_test = &quot;fisher&quot;)
    
    pve_tissue1_joint &lt;- ctwas_parameters_pair$prop_heritability[paste0(tissue1,&quot;|eQTL&quot;)]
    pve_tissue2_joint &lt;- ctwas_parameters_pair$prop_heritability[paste0(tissue2,&quot;|eQTL&quot;)]
    
    pct_pve_shared_tissue1 &lt;- (pve_tissue1 - pve_tissue1_joint)/pve_tissue1 * 100
    pct_pve_shared_tissue2 &lt;- (pve_tissue2 - pve_tissue2_joint)/pve_tissue2 * 100
    
    mat[tissue1, tissue2] &lt;- pct_pve_shared_tissue1
    mat[tissue2, tissue1] &lt;- pct_pve_shared_tissue2
    
  
  }
  
  prob_pve_alltissue &lt;- prob_pve_alltissue[rownames(mat)]
  prob_pve_alltissue &lt;- round(prob_pve_alltissue*100,digits = 2)
  
  if (any(mat &lt; 0, na.rm = TRUE)) {
    
    mat_range &lt;- range(mat, na.rm = TRUE)  # includes negatives
    
    # create diverging color function
    col_fun &lt;- colorRamp2(
      c(mat_range[1], 0, mat_range[2]),  # min (neg), 0, max (pos)
      c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)          # colors
    )
    
    rownames(mat) &lt;- paste0(rownames(mat),&quot; &quot;,prob_pve_alltissue,&quot;%&quot;)
    
    ht1 &lt;- Heatmap(mat,
            name = &quot;%PVE decrease&quot;,
            cluster_rows = FALSE,
            cluster_columns = FALSE,
            col = col_fun,
            row_names_side = &quot;left&quot;,
            column_title = paste0(&quot;%PVE decreased after \n adding the second tissue(column) -- \n&quot;, trait,&quot; (eQTL)&quot;),
            rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
            cell_fun = function(j, i, x, y, width, height, fill) {
              if (!is.na(mat[i, j])) {
                grid.text(sprintf(&quot;%.1f&quot;, mat[i, j]), x, y, gp = gpar(fontsize = 8))
              }
            })
    
    
    
  }else{
    
    rownames(mat) &lt;- paste0(rownames(mat),&quot; &quot;,prob_pve_alltissue,&quot;%&quot;)
    
       ht1 &lt;- Heatmap(mat,
          name = &quot;%PVE decrease&quot;,
          cluster_rows = FALSE,
          cluster_columns = FALSE,
          col = colorRampPalette(c(&quot;white&quot;, &quot;red&quot;))(100),
          row_names_side = &quot;left&quot;,   # can also be &quot;right&quot;
          column_title = paste0(&quot;%PVE decreased after  \n adding the second tissue(column) -- \n&quot;, trait,&quot; (eQTL)&quot;),
          rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),   # draw grid lines
          cell_fun = function(j, i, x, y, width, height, fill) {
            grid.text(sprintf(&quot;%.1f&quot;, mat[i, j]), x, y, gp = gpar(fontsize = 8))
          })
    
  }

  
  
  
  ## single tissue sQTL
  
  folder_results &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/splicing/&quot;,trait,&quot;/&quot;)
  
  prob_pve_alltissue &lt;- c()
  for (tissue in top_tissues_phe){
    
    file_param_single &lt;- paste0(folder_results,&quot;/&quot;,trait,&quot;_&quot;,tissue,&quot;.thin1.shared_all.param.RDS&quot;)
    param_single &lt;- readRDS(file_param_single)
    
    ctwas_parameters_single &lt;- summarize_param(param_single, gwas_n, enrichment_test = &quot;fisher&quot;)
    
    prob_pve &lt;- ctwas_parameters_single$prop_heritability[1]
    prob_pve_alltissue &lt;- c(prob_pve_alltissue, prob_pve)
    
  }
  
  names(prob_pve_alltissue) &lt;- gsub(pattern = &quot;\\|sQTL&quot;,x = names(prob_pve_alltissue), replacement = &quot;&quot;)
  
    mat &lt;- matrix(NA, nrow = length(top_tissues_phe), ncol = length(top_tissues_phe),
                dimnames = list(top_tissues_phe, top_tissues_phe))
  
  for (i in 1:length(tissue_combination)){
    
    tissue1 &lt;- tissue_combination[[i]][1]
    pve_tissue1 &lt;- prob_pve_alltissue[tissue1]
    tissue2 &lt;- tissue_combination[[i]][2]
    pve_tissue2 &lt;- prob_pve_alltissue[tissue2]
    
    tissue_pair &lt;- paste0(tissue_combination[[i]], collapse = &quot;-&quot;)
    
    file_param_pair &lt;- paste0(&quot;/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/pairwise_snakemake_outputs_byPHE/&quot;,trait,&quot;/&quot;,trait,&quot;.&quot;,tissue_pair,&quot;.sqtlonly.thin1.shared_all.param.RDS&quot;)
    param_pair &lt;- readRDS(file_param_pair)
    ctwas_parameters_pair &lt;- summarize_param(param_pair, gwas_n, enrichment_test = &quot;fisher&quot;)
    
    pve_tissue1_joint &lt;- ctwas_parameters_pair$prop_heritability[paste0(tissue1,&quot;|sQTL&quot;)]
    pve_tissue2_joint &lt;- ctwas_parameters_pair$prop_heritability[paste0(tissue2,&quot;|sQTL&quot;)]
    
    pct_pve_shared_tissue1 &lt;- (pve_tissue1 - pve_tissue1_joint)/pve_tissue1 * 100
    pct_pve_shared_tissue2 &lt;- (pve_tissue2 - pve_tissue2_joint)/pve_tissue2 * 100
    
    mat[tissue1, tissue2] &lt;- pct_pve_shared_tissue1
    mat[tissue2, tissue1] &lt;- pct_pve_shared_tissue2
    
  
  }
  
  prob_pve_alltissue &lt;- prob_pve_alltissue[rownames(mat)]
  prob_pve_alltissue &lt;- round(prob_pve_alltissue*100,digits = 2)
  
  

  if (any(mat &lt; 0, na.rm = TRUE)) {
    
    mat_range &lt;- range(mat, na.rm = TRUE)  # includes negatives
    
    # create diverging color function
    col_fun &lt;- colorRamp2(
      c(mat_range[1], 0, mat_range[2]),  # min (neg), 0, max (pos)
      c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)          # colors
    )
    
    rownames(mat) &lt;- paste0(rownames(mat),&quot; &quot;,prob_pve_alltissue,&quot;%&quot;)
    
    ht2 &lt;- Heatmap(mat,
            name = &quot;%PVE decrease&quot;,
            cluster_rows = FALSE,
            cluster_columns = FALSE,
            col = col_fun,
            row_names_side = &quot;left&quot;,
            column_title = paste0(&quot;%PVE decreased after \n adding the second tissue(column) -- \n&quot;, trait,&quot; (sQTL)&quot;),
            rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),
            cell_fun = function(j, i, x, y, width, height, fill) {
              if (!is.na(mat[i, j])) {
                grid.text(sprintf(&quot;%.1f&quot;, mat[i, j]), x, y, gp = gpar(fontsize = 8))
              }
            })
    
    
    
  }else{
    
    rownames(mat) &lt;- paste0(rownames(mat),&quot; &quot;,prob_pve_alltissue,&quot;%&quot;)
    
       ht2 &lt;- Heatmap(mat,
          name = &quot;%PVE decrease&quot;,
          cluster_rows = FALSE,
          cluster_columns = FALSE,
          col = colorRampPalette(c(&quot;white&quot;, &quot;red&quot;))(100),
          row_names_side = &quot;left&quot;,   # can also be &quot;right&quot;
          column_title = paste0(&quot;%PVE decreased after  \n adding the second tissue(column) -- \n&quot;, trait,&quot; (sQTL)&quot;),
          rect_gp = gpar(col = &quot;black&quot;, lwd = 0.5),   # draw grid lines
          cell_fun = function(j, i, x, y, width, height, fill) {
            grid.text(sprintf(&quot;%.1f&quot;, mat[i, j]), x, y, gp = gpar(fontsize = 8))
          })
    
  }
  

  
  
  draw(ht1 + ht2)
  
}</code></pre>
<pre><code>[1] &quot;LDL-ukb-d-30780_irnt&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-1.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;IBD-ebi-a-GCST004131&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-2.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;aFib-ebi-a-GCST006414&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-3.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SBP-ukb-a-360&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-4">
Past versions of unnamed-chunk-3-4.png
</button>
</p>
<div id="fig-unnamed-chunk-3-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-4.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;T1D-GCST90014023&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-5">
Past versions of unnamed-chunk-3-5.png
</button>
</p>
<div id="fig-unnamed-chunk-3-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-5.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;T2D-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-6">
Past versions of unnamed-chunk-3-6.png
</button>
</p>
<div id="fig-unnamed-chunk-3-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-6.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ATH_gtexukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-7.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-7">
Past versions of unnamed-chunk-3-7.png
</button>
</p>
<div id="fig-unnamed-chunk-3-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-7.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;BMI-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-8.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-8">
Past versions of unnamed-chunk-3-8.png
</button>
</p>
<div id="fig-unnamed-chunk-3-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-8.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;HB-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-9.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-9">
Past versions of unnamed-chunk-3-9.png
</button>
</p>
<div id="fig-unnamed-chunk-3-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-9.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Height-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-10.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-10">
Past versions of unnamed-chunk-3-10.png
</button>
</p>
<div id="fig-unnamed-chunk-3-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-10.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;HTN-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-11.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-11">
Past versions of unnamed-chunk-3-11.png
</button>
</p>
<div id="fig-unnamed-chunk-3-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-11.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;PLT-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-12.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-12">
Past versions of unnamed-chunk-3-12.png
</button>
</p>
<div id="fig-unnamed-chunk-3-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-12.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;RA-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-13.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-13">
Past versions of unnamed-chunk-3-13.png
</button>
</p>
<div id="fig-unnamed-chunk-3-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-13.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;RBC-panukb&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-14.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-14">
Past versions of unnamed-chunk-3-14.png
</button>
</p>
<div id="fig-unnamed-chunk-3-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-14.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;WBC-ieu-b-30&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-15.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-15">
Past versions of unnamed-chunk-3-15.png
</button>
</p>
<div id="fig-unnamed-chunk-3-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-15.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SCZ-ieu-b-5102&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-16.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-16">
Past versions of unnamed-chunk-3-16.png
</button>
</p>
<div id="fig-unnamed-chunk-3-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-16.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;BIP-ieu-b-5110&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-17.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-17">
Past versions of unnamed-chunk-3-17.png
</button>
</p>
<div id="fig-unnamed-chunk-3-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-17.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;MDD-ieu-b-102&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-18.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-18">
Past versions of unnamed-chunk-3-18.png
</button>
</p>
<div id="fig-unnamed-chunk-3-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-18.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;PD-ieu-b-7&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-19.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-19">
Past versions of unnamed-chunk-3-19.png
</button>
</p>
<div id="fig-unnamed-chunk-3-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-19.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;NS-ukb-a-230&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-20.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-20">
Past versions of unnamed-chunk-3-20.png
</button>
</p>
<div id="fig-unnamed-chunk-3-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-20.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ASD-ieu-a-1185&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-21.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-21">
Past versions of unnamed-chunk-3-21.png
</button>
</p>
<div id="fig-unnamed-chunk-3-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-21.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ADHD-ieu-a-1183&quot;</code></pre>
<p><img src="figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-22.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-22">
Past versions of unnamed-chunk-3-22.png
</button>
</p>
<div id="fig-unnamed-chunk-3-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/fef2c966c1b38536af83c4909ba01818af93de90/docs/figure/realdata_final_PVE_share_pattern_rankPHE.Rmd/unnamed-chunk-3-22.png" target="_blank">fef2c96</a>
</td>
<td>
XSun
</td>
<td>
2025-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] circlize_0.4.15       ComplexHeatmap_2.12.0 ctwas_0.5.32         

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] GlobalOptions_0.1.2         GenomicRanges_1.48.0       
  [9] base64enc_0.1-3             fs_1.5.2                   
 [11] clue_0.3-61                 rstudioapi_0.13            
 [13] ggrepel_0.9.1               bit64_4.0.5                
 [15] AnnotationDbi_1.58.0        fansi_1.0.3                
 [17] xml2_1.3.3                  codetools_0.2-18           
 [19] logging_0.10-108            doParallel_1.0.17          
 [21] cachem_1.0.6                knitr_1.39                 
 [23] jsonlite_1.8.0              workflowr_1.7.0            
 [25] Rsamtools_2.12.0            cluster_2.1.3              
 [27] dbplyr_2.1.1                png_0.1-7                  
 [29] readr_2.1.2                 compiler_4.2.0             
 [31] httr_1.4.3                  assertthat_0.2.1           
 [33] Matrix_1.5-3                fastmap_1.1.0              
 [35] lazyeval_0.2.2              cli_3.6.1                  
 [37] later_1.3.0                 htmltools_0.5.2            
 [39] prettyunits_1.1.1           tools_4.2.0                
 [41] gtable_0.3.0                glue_1.6.2                 
 [43] GenomeInfoDbData_1.2.8      dplyr_1.1.4                
 [45] rappdirs_0.3.3              Rcpp_1.0.12                
 [47] Biobase_2.56.0              jquerylib_0.1.4            
 [49] vctrs_0.6.5                 Biostrings_2.64.0          
 [51] rtracklayer_1.56.0          iterators_1.0.14           
 [53] xfun_0.41                   stringr_1.5.1              
 [55] irlba_2.3.5                 lifecycle_1.0.4            
 [57] restfulr_0.0.14             ensembldb_2.20.2           
 [59] XML_3.99-0.14               zlibbioc_1.42.0            
 [61] zoo_1.8-10                  scales_1.3.0               
 [63] gggrid_0.2-0                hms_1.1.1                  
 [65] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [67] ProtGenerics_1.28.0         parallel_4.2.0             
 [69] SummarizedExperiment_1.26.1 RColorBrewer_1.1-3         
 [71] AnnotationFilter_1.20.0     LDlinkR_1.2.3              
 [73] yaml_2.3.5                  curl_4.3.2                 
 [75] memoise_2.0.1               ggplot2_3.5.1              
 [77] sass_0.4.1                  biomaRt_2.54.1             
 [79] stringi_1.7.6               RSQLite_2.3.1              
 [81] highr_0.9                   S4Vectors_0.34.0           
 [83] BiocIO_1.6.0                foreach_1.5.2              
 [85] GenomicFeatures_1.48.3      BiocGenerics_0.42.0        
 [87] filelock_1.0.2              BiocParallel_1.30.3        
 [89] shape_1.4.6                 repr_1.1.4                 
 [91] GenomeInfoDb_1.39.9         rlang_1.1.2                
 [93] pkgconfig_2.0.3             matrixStats_0.62.0         
 [95] bitops_1.0-7                evaluate_0.15              
 [97] lattice_0.20-45             purrr_1.0.2                
 [99] GenomicAlignments_1.32.0    htmlwidgets_1.5.4          
[101] cowplot_1.1.1               bit_4.0.4                  
[103] tidyselect_1.2.0            magrittr_2.0.3             
[105] AMR_2.1.1                   R6_2.5.1                   
[107] IRanges_2.30.0              generics_0.1.2             
[109] DelayedArray_0.22.0         DBI_1.2.2                  
[111] pgenlibr_0.3.3              pillar_1.9.0               
[113] whisker_0.4                 mixsqp_0.3-43              
[115] KEGGREST_1.36.3             RCurl_1.98-1.7             
[117] tibble_3.2.1                crayon_1.5.1               
[119] utf8_1.2.2                  BiocFileCache_2.4.0        
[121] plotly_4.10.0               tzdb_0.4.0                 
[123] rmarkdown_2.25              GetoptLong_1.0.5           
[125] progress_1.2.2              data.table_1.14.2          
[127] blob_1.2.3                  git2r_0.30.1               
[129] digest_0.6.29               tidyr_1.3.0                
[131] httpuv_1.6.5                stats4_4.2.0               
[133] munsell_0.5.0               viridisLite_0.4.0          
[135] skimr_2.1.4                 bslib_0.3.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
