<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-05-21" />

<title>Tissue selection – expression, tissue filtered by sample size, irrelevant tissues filtered out, brian tissue removed from non-psychiatric traits</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tissue selection – expression, tissue
filtered by sample size, irrelevant tissues filtered out, brian tissue
removed from non-psychiatric traits</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-05-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreec791833a53a54625de7b3345b9ddee4069882335targetblankc791833a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/c791833a53a54625de7b3345b9ddee4069882335" target="_blank">c791833</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreec791833a53a54625de7b3345b9ddee4069882335targetblankc791833a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/c791833a53a54625de7b3345b9ddee4069882335" target="_blank">c791833</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    cv/

Unstaged changes:
    Modified:   analysis/realdata_final_tissueselection_mingene0_exclude_brainprocessed.Rmd
    Modified:   analysis/realdata_final_tissueselection_mingene0_splicing_exclude_brainprocessed.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/realdata_final_tissueselection_mingene0_exclude_brainprocessed.Rmd</code>)
and HTML
(<code>docs/realdata_final_tissueselection_mingene0_exclude_brainprocessed.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c791833a53a54625de7b3345b9ddee4069882335/analysis/realdata_final_tissueselection_mingene0_exclude_brainprocessed.Rmd" target="_blank">c791833</a>
</td>
<td>
XSun
</td>
<td>
2025-05-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/c791833a53a54625de7b3345b9ddee4069882335/docs/realdata_final_tissueselection_mingene0_exclude_brainprocessed.html" target="_blank">c791833</a>
</td>
<td>
XSun
</td>
<td>
2025-05-21
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<ul>
<li><p>For non-psychiatric traits, we ran a cTWAS analysis using sQTL
data from all 49 GTEx tissues.</p></li>
<li><p>For psychiatric traits, we used only the 9 GTEx brain
tissues.</p></li>
<li><p>We ranked tissues by their G-test p-values &amp; Fisher p-values
and selected those passing the Bonferroni threshold (0.05 divided by the
number of tissues) for the multi-group cTWAS analysis.</p></li>
<li><p>Default settings were used for computing z-scores, assembling
input data, and estimating model parameters.</p></li>
</ul>
<p>Mem: 50g/node got killed, 100g/node</p>
<ol style="list-style-type: decimal">
<li>Remove tissues with &lt;200 samples. (2) Remove some tissues that
are not related to our traits, including Testis, Nerve Tibial, Ovary,
Prostate, Uterus, Vagina, Breast Mammary Tissue.</li>
</ol>
<pre><code>z_gene &lt;- compute_gene_z(z_snp, weights, ncore=10)</code></pre>
<pre><code>res &lt;- assemble_region_data(region_info, 
                                z_snp, 
                                z_gene, 
                                weights,
                                snp_map,
                                maxSNP = Inf,
                                min_group_size = 100,
                                thin = 1,
                                adjust_boundary_genes = TRUE,
                                ncore = 15)</code></pre>
<pre><code>param &lt;- est_param(region_data, 
                       group_prior_var_structure = &quot;shared_all&quot;,
                       null_method = &quot;ctwas&quot;,
                       niter_prefit = 3,
                       min_gene = 0,
                       min_var = 2,                          
                       min_p_single_effect = 0.8,
                       niter = 200, 
                       ncore = 15,
                       verbose=TRUE)</code></pre>
<pre class="r"><code>library(ctwas)</code></pre>
<pre><code>Warning: replacing previous import &#39;utils::download.file&#39; by
&#39;restfulr::download.file&#39; when loading &#39;rtracklayer&#39;</code></pre>
<pre class="r"><code>source(&quot;/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R&quot;)

trait_nopsy &lt;- c(&quot;LDL-ukb-d-30780_irnt&quot;,&quot;aFib-ebi-a-GCST006414&quot;,&quot;ATH_gtexukb&quot;,&quot;BMI-panukb&quot;,&quot;HB-panukb&quot;,
             &quot;Height-panukb&quot;,&quot;HTN-panukb&quot;,&quot;IBD-ebi-a-GCST004131&quot;,&quot;PLT-panukb&quot;,&quot;RA-panukb&quot;,&quot;RBC-panukb&quot;,
             &quot;SBP-ukb-a-360&quot;,&quot;T1D-GCST90014023&quot;,&quot;WBC-ieu-b-30&quot;,&quot;T2D-panukb&quot;)

trait_psy &lt;- c(&quot;SCZ-ieu-b-5102&quot;,&quot;ASD-ieu-a-1185&quot;,&quot;BIP-ieu-b-5110&quot;,&quot;MDD-ieu-b-102&quot;,&quot;PD-ieu-b-7&quot;,&quot;ADHD-ieu-a-1183&quot;,&quot;NS-ukb-a-230&quot;)

DT::datatable(matrix())</code></pre>
<div id="htmlwidget-4de263013227ca78a293" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de263013227ca78a293">{"x":{"filter":"none","vertical":false,"data":[[null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>folder_results &lt;- &quot;/project/xinhe/xsun/multi_group_ctwas/21.tissue_selection_0511/results/E_thin1_shared_all_mingene0_exclude_brainprocess/&quot;</code></pre>
</div>
<div id="non-psychiatric-traits" class="section level1">
<h1>Non-psychiatric traits</h1>
<div id="tissue-selection" class="section level2">
<h2>Tissue selection</h2>
<pre class="r"><code>converge_df &lt;- c()
for (trait in trait_nopsy){

  param &lt;- readRDS(paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.thin1.shared_all.param.RDS&quot;))

  gwas_n &lt;- samplesize[trait]

  param_summarized_fisher &lt;- summarize_param(param = param,gwas_n = gwas_n,enrichment_test = &quot;fisher&quot;,alternative = &quot;greater&quot;)
  param_summarized_G &lt;- summarize_param(param = param,gwas_n = gwas_n,enrichment_test = &quot;G&quot;)

  param_df &lt;- data.frame(
    group = names(param_summarized_fisher$group_size),
    group_size = as.numeric(param_summarized_fisher$group_size[names(param_summarized_fisher$group_size)]),
    group_pve = as.numeric(param_summarized_fisher$group_pve[names(param_summarized_fisher$group_size)]),
    prop_heritability = as.numeric(param_summarized_fisher$prop_heritability[names(param_summarized_fisher$group_size)]),
    log_enrichment = as.numeric(param_summarized_fisher$log_enrichment[names(param_summarized_fisher$group_size)]),
    log_enrichment_se = as.numeric(param_summarized_fisher$log_enrichment_se[names(param_summarized_fisher$group_size)]),
    enrichment_pval_fisher = as.numeric(param_summarized_fisher$enrichment_pval[names(param_summarized_fisher$group_size)]),
    enrichment_pval_G = as.numeric(param_summarized_G$enrichment_pval[names(param_summarized_G$group_size)])
  )

  param_df$total_pve &lt;- param_summarized_fisher$total_pve

  param_df$prop_heritability &lt;- paste0(round(param_df$prop_heritability * 100, 5), &quot;%&quot;)

  param_df &lt;- param_df[order(param_df$enrichment_pval_fisher,decreasing = F),]

  param_df_qtl &lt;- param_df[-nrow(param_df),]
  threshold &lt;- 0.05/(nrow(param_df_qtl)-1)



  cat(&quot;&lt;br&gt;&quot;)
  cat(knitr::knit_print(DT::datatable(param_df, caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,trait,options = list(pageLength = 10)))))
  cat(&quot;&lt;br&gt;&quot;)


  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  print(paste0(&quot;p-value cutoff(0.05/num_tissue) = &quot;,threshold))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(&quot;Number of selected tissue -- fisher = &quot;,min(10,sum(param_df_qtl$enrichment_pval_fisher &lt; threshold)),&quot;\n&quot;))
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(
    head(param_df_qtl$group[param_df_qtl$enrichment_pval_fisher &lt; threshold], 10),
    collapse = &quot; &quot;
  ))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
    cat(paste0(&quot;Number of selected tissue -- G = &quot;,min(10,sum(param_df_qtl$enrichment_pval_G &lt; threshold)),&quot;\n&quot;))
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(
    head(param_df_qtl$group[param_df_qtl$enrichment_pval_G &lt; threshold], 10),
    collapse = &quot; &quot;
  ))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)

  EM_iter &lt;- length(param$loglik_iters)
  converge &lt;- param$converged
  converge_df &lt;- rbind(converge_df,c(trait,EM_iter,converge))
}</code></pre>
<p><br></p>
<div id="htmlwidget-20575149739a71fc930f" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-20575149739a71fc930f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">LDL-ukb-d-30780_irnt<\/caption>","data":[["15","25","16","22","10","11","9","5","7","18","4","23","17","2","1","8","13","24","12","3","6","14","19","20","21","26"],["Liver|eQTL","Whole_Blood|eQTL","Lung|eQTL","Spleen|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Mucosa|eQTL","Colon_Transverse|eQTL","Artery_Coronary|eQTL","Cells_Cultured_fibroblasts|eQTL","Pancreas|eQTL","Artery_Aorta|eQTL","Stomach|eQTL","Muscle_Skeletal|eQTL","Adipose_Visceral_Omentum|eQTL","Adipose_Subcutaneous|eQTL","Colon_Sigmoid|eQTL","Heart_Atrial_Appendage|eQTL","Thyroid|eQTL","Esophagus_Muscularis|eQTL","Adrenal_Gland|eQTL","Artery_Tibial|eQTL","Heart_Left_Ventricle|eQTL","Pituitary|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","SNP"],[8775,9336,10692,9749,10059,10580,10221,9504,10310,9739,10411,9799,9969,10384,10538,10025,10003,11040,10564,9433,10599,9447,9966,10760,10998,7410963],[0.0120879470723113,0.00506468998385855,0.00315597883257555,0.00294017107909328,0.00281863799852992,0.00268768644716527,0.00208883941267625,0.00180692656167652,0.00163392478558833,0.00128445592648764,0.00126749147949314,0.000920445939880802,0.000988212096904757,0.00104757305809072,0.00098658272295391,0.000651215200918539,0.000513347210315984,0.000421499402544133,0.000320563998786741,1.43644090842876e-12,3.4313604076e-10,1.62349850941388e-08,4.5233362200087e-06,1.52068788057465e-07,3.68583195455899e-11,0.0630986501065242],["11.42641%","4.78752%","2.98326%","2.77926%","2.66438%","2.5406%","1.97452%","1.70804%","1.54451%","1.21416%","1.19813%","0.87007%","0.93413%","0.99024%","0.93259%","0.61558%","0.48525%","0.39843%","0.30302%","0%","0%","2e-05%","0.00428%","0.00014%","0%","59.64546%"],[5.08631837004294,4.15443132627825,3.545819190608,3.56731934714376,3.49380232584658,3.39573163327216,3.1781804724168,3.10593090010925,2.9238869594633,2.74021307516149,2.66019308764011,2.40083894419705,2.45467808363225,2.47222624857985,2.39752041913967,2.0320191981405,1.79632840543759,1.50055433312369,1.27089096017697,-17.8392952608537,-12.4798822122016,-8.50802300836265,-2.93166846252302,-6.40099684824772,-14.7478816210543,null],[0.154370815383883,0.227382727965491,0.284117311926287,0.293875713213197,0.299880486483628,0.306809460635765,0.346466205024627,0.371718080544779,0.390405126763425,0.439156243662184,0.442036143359127,0.517346090396922,0.499551734972741,0.485414746386022,0.49996330225839,0.613801921640278,0.690600319890717,0.761608447435456,0.872638681350625,13003.9289324426,841.366668452212,122.318621219647,7.32831610989437,39.9667595899272,2567.14680232684,null],[6.97449024649081e-89,2.243988196207e-30,4.60833595664846e-16,5.20197375829456e-15,7.49264501214462e-15,4.17998225599105e-13,2.30356047639489e-10,7.51435643823963e-08,1.29494454316137e-07,3.07932882003377e-05,4.21368022959074e-05,0.00045809487244452,0.000488363485798595,0.000568188793593119,0.00060003174787339,0.00569364808549101,0.0493558893343618,0.0587083104498586,0.310513177223887,1,1,1,1,1,1,null],[1.37052784766108e-88,3.06907607841405e-30,3.89996680361701e-16,2.95550542428385e-15,2.50617535516804e-14,2.8433933808239e-13,7.39478599147538e-10,1.70907921892911e-08,2.7090384171323e-07,1.3134654502552e-05,2.25177076990497e-05,0.000793344731256249,0.000411178485731475,0.000256322913598751,0.000521374848269758,0.0126671377572486,0.0440439092931585,0.113559980141073,0.228204254258516,0.416259259670406,0.388879420714696,0.416361687218692,0.457587893775716,0.388227481274334,0.380091379164645,null],[0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378,0.1057895313378]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Liver|eQTL Whole_Blood|eQTL Lung|eQTL Spleen|eQTL
Esophagus_Gastroesophageal_Junction|eQTL Esophagus_Mucosa|eQTL
Colon_Transverse|eQTL Artery_Coronary|eQTL
Cells_Cultured_fibroblasts|eQTL Pancreas|eQTL<br><br><br>Number of
selected tissue – G = 10 <br>Liver|eQTL Whole_Blood|eQTL Lung|eQTL
Spleen|eQTL Esophagus_Gastroesophageal_Junction|eQTL
Esophagus_Mucosa|eQTL Colon_Transverse|eQTL Artery_Coronary|eQTL
Cells_Cultured_fibroblasts|eQTL Pancreas|eQTL<br><br><br><br></p>
<div id="htmlwidget-a23c348805048a1f64c7" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a23c348805048a1f64c7">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">aFib-ebi-a-GCST006414<\/caption>","data":[["13","14","12","4","17","22","6","24","7","16","15","3","21","1","2","5","8","9","10","11","18","19","20","23","25","26"],["Heart_Atrial_Appendage|eQTL","Heart_Left_Ventricle|eQTL","Esophagus_Muscularis|eQTL","Artery_Aorta|eQTL","Muscle_Skeletal|eQTL","Spleen|eQTL","Artery_Tibial|eQTL","Thyroid|eQTL","Cells_Cultured_fibroblasts|eQTL","Lung|eQTL","Liver|eQTL","Adrenal_Gland|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Adipose_Subcutaneous|eQTL","Adipose_Visceral_Omentum|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Mucosa|eQTL","Pancreas|eQTL","Pituitary|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Stomach|eQTL","Whole_Blood|eQTL","SNP"],[9894,9352,10455,10287,9888,9643,10493,10944,10227,10578,8666,9324,10905,10432,10274,9393,9927,10116,9944,10469,9638,9839,10653,9690,9239,7059851],[0.00152053983662635,0.00106625110820558,0.000368409424686976,0.000320617039827865,0.000279598168091843,0.000213036816261834,0.000224477977991362,0.000220190863487978,0.000188938190681265,0.000122944161746579,8.18664255313537e-05,9.09734939877406e-05,7.27564989455673e-05,9.00502808454751e-06,2.9191527710986e-19,1.52337162713829e-21,3.47738724567016e-15,3.12720035853259e-13,4.83349369627473e-14,2.81441948644143e-11,2.78601187184716e-13,1.62331704323428e-07,1.92927358644072e-08,2.65401290136284e-11,1.33686562113631e-12,0.00909784445703635],["10.95677%","7.68324%","2.6547%","2.31032%","2.01474%","1.53511%","1.61755%","1.58666%","1.36146%","0.88592%","0.58992%","0.65554%","0.52427%","0.06489%","0%","0%","0%","0%","0%","0%","0%","0.00117%","0.00014%","0%","0%","65.55762%"],[4.78127862751291,4.48270034784004,3.30850098778414,3.18575286027608,3.08841799158961,2.84161916740039,2.80945567447109,2.74808980452309,2.66277508017194,2.19934034396898,1.99206807267463,2.02436290191031,1.64428326884894,-0.400708473932887,-31.4455427798165,-36.6114286379801,-22.025856500371,-17.545689395764,-19.3956941255574,-13.0802015216395,-17.6128116987205,-4.35808112561453,-6.56748174277592,-13.0615606600177,-16.00222999348,null],[0.142832387798103,0.166971861611633,0.274485128479265,0.293509694471931,0.313633620125058,0.358066155725965,0.349040767689857,0.352349400090723,0.379748775242954,0.469145663224362,0.573662395096111,0.544458458533867,0.608247200353481,1.72308038548597,9565592.11206038,132415046.936979,87642.2950375819,9241.92450380329,23507.6757571142,974.194910170192,9791.49009096056,12.8275092384305,37.2086408603454,1003.20284675687,4469.8854395924,null],[8.76482745698993e-94,1.29148189564609e-61,6.73251017409504e-16,3.71086963347496e-13,1.28323030117515e-10,4.00718114965615e-08,7.58672989489524e-08,1.04144873494553e-07,9.96922291937231e-07,0.000197378165480432,0.00909199084278667,0.0110602242431542,0.0167225959222679,1,1,1,1,1,1,1,1,1,1,1,1,null],[7.78121143347149e-94,1.25541775843802e-61,1.14599876354542e-15,3.22031981550391e-13,2.70942040586995e-11,4.02532150367397e-08,2.28145171461529e-08,5.04711436721021e-08,7.90849384991516e-07,0.000528876992581194,0.00863787417661125,0.00507924668641166,0.032964844679476,0.803259326920864,0.319591661520864,0.341250599218519,0.327897727267978,0.323338859311314,0.327484202148315,0.315052591656761,0.335035713050965,0.347219287886571,0.313421420103367,0.333743315516849,0.345237354729927,null],[0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977,0.0138776311722977]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Heart_Atrial_Appendage|eQTL Heart_Left_Ventricle|eQTL
Esophagus_Muscularis|eQTL Artery_Aorta|eQTL Muscle_Skeletal|eQTL
Spleen|eQTL Artery_Tibial|eQTL Thyroid|eQTL
Cells_Cultured_fibroblasts|eQTL Lung|eQTL<br><br><br>Number of selected
tissue – G = 10 <br>Heart_Atrial_Appendage|eQTL
Heart_Left_Ventricle|eQTL Esophagus_Muscularis|eQTL Artery_Aorta|eQTL
Muscle_Skeletal|eQTL Spleen|eQTL Artery_Tibial|eQTL Thyroid|eQTL
Cells_Cultured_fibroblasts|eQTL Lung|eQTL<br><br><br><br></p>
<div id="htmlwidget-93928d62269dcef60a51" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93928d62269dcef60a51">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">ATH_gtexukb<\/caption>","data":[["22","24","14","12","13","9","6","15","1","20","11","21","17","7","2","3","4","5","8","10","16","18","19","23","25","26"],["Spleen|eQTL","Thyroid|eQTL","Heart_Left_Ventricle|eQTL","Esophagus_Muscularis|eQTL","Heart_Atrial_Appendage|eQTL","Colon_Transverse|eQTL","Artery_Tibial|eQTL","Liver|eQTL","Adipose_Subcutaneous|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Esophagus_Mucosa|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Muscle_Skeletal|eQTL","Cells_Cultured_fibroblasts|eQTL","Adipose_Visceral_Omentum|eQTL","Adrenal_Gland|eQTL","Artery_Aorta|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Lung|eQTL","Pancreas|eQTL","Pituitary|eQTL","Stomach|eQTL","Whole_Blood|eQTL","SNP"],[9753,11047,9449,10569,10004,10229,10614,8778,10541,10761,10583,11000,9977,10315,10388,9433,10415,9510,10030,10064,10700,9747,9969,9800,9342,6601263],[0.00153119917089804,0.00105604252838122,0.000949008358494988,0.000858363588973063,0.000723215797089053,0.000714135374140562,0.000522845531030379,0.000471033657703793,0.000484238637095191,0.000428851213486719,0.000268835702348162,0.000163207977581386,0.000149974437682048,6.54704987688014e-05,6.92240800986285e-05,7.0317653677703e-09,1.78899535387427e-05,6.65117137976342e-10,3.20299982119182e-09,4.61535215147958e-07,9.18685528248816e-08,3.22356099375931e-06,4.28094935368294e-09,4.5667223078036e-12,1.64790159060366e-05,0.0183800864400433],["5.69772%","3.92962%","3.53134%","3.19404%","2.69115%","2.65736%","1.94555%","1.75276%","1.80189%","1.59579%","1.00036%","0.60731%","0.55807%","0.24362%","0.25759%","3e-05%","0.06657%","0%","1e-05%","0.00172%","0.00034%","0.012%","2e-05%","0%","0.06132%","68.39383%"],[4.03222472374088,3.53611801315765,3.58550186496192,3.37309575957452,3.25671581364832,3.22183889503386,2.87310535878018,2.95867413479445,2.80329859628727,2.66117472434817,2.21084465545741,1.67312322528006,1.68617559469375,0.823993668685158,0.872690661100836,-8.22553849067744,-0.483014599743273,-10.5918980875408,-9.07325489164994,-4.10616844755815,-5.78164706420964,-2.13047972366698,-8.77706758591315,-15.6031040408542,-0.45643937931685,null],[0.205806650789755,0.244955770966278,0.257677082782706,0.27034566018038,0.293507671763512,0.295303381192882,0.343443749898029,0.361330276653153,0.356516826372737,0.378302327665089,0.475821464873995,0.609004967029226,0.635078120345401,0.959072696797688,0.932800564684215,92.3832117703029,1.83242239292615,300.38351866998,136.882125830137,11.4032337455337,25.5589364733237,4.31513212896388,118.400944259009,3625.12394518178,1.90918379605616,null],[5.30072915779456e-36,9.60775302050345e-22,8.91082579149494e-20,4.61531612167795e-16,1.71333859759234e-13,2.21483922439807e-13,3.42684885854742e-09,1.45019611512732e-08,5.81778099229955e-08,1.07138409432372e-06,0.0016804468487304,0.0153836536120901,0.0796232314962209,0.380828457425379,0.382923624970994,1,1,1,1,1,1,1,1,1,1,null],[2.86686189538174e-35,3.36007077374488e-21,1.45799198429375e-19,1.45223817074509e-16,1.28557550948436e-13,2.72958944073537e-13,7.51170226408838e-09,2.12652936003505e-08,4.57781709383879e-08,7.08241583360169e-07,0.000599673723275417,0.0307743165807518,0.0371969084755534,0.449525142493374,0.413507800389046,0.350269802476058,0.774439911390721,0.347786105514033,0.335154483421419,0.355257178895232,0.324364533025084,0.451218915601459,0.336712692286567,0.340474987426204,0.795879979794139,null],[0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203,0.0268738941134203]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Spleen|eQTL Thyroid|eQTL Heart_Left_Ventricle|eQTL
Esophagus_Muscularis|eQTL Heart_Atrial_Appendage|eQTL
Colon_Transverse|eQTL Artery_Tibial|eQTL Liver|eQTL
Adipose_Subcutaneous|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL<br><br><br>Number of selected
tissue – G = 10 <br>Spleen|eQTL Thyroid|eQTL Heart_Left_Ventricle|eQTL
Esophagus_Muscularis|eQTL Heart_Atrial_Appendage|eQTL
Colon_Transverse|eQTL Artery_Tibial|eQTL Liver|eQTL
Adipose_Subcutaneous|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL<br><br><br><br></p>
<div id="htmlwidget-16707cd634e93674753d" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16707cd634e93674753d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">BMI-panukb<\/caption>","data":[["15","8","4","10","19","22","9","25","14","30","3","1","2","12","13","24","5","6","7","11","16","17","18","20","21","23","26","27","28","29","31","32","33","34"],["Cells_Cultured_fibroblasts|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Artery_Aorta|eQTL","Brain_Cortex|eQTL","Esophagus_Mucosa|eQTL","Heart_Left_Ventricle|eQTL","Brain_Cerebellum|eQTL","Muscle_Skeletal|eQTL","Brain_Putamen_basal_ganglia|eQTL","Spleen|eQTL","Adrenal_Gland|eQTL","Adipose_Subcutaneous|eQTL","Adipose_Visceral_Omentum|eQTL","Brain_Hypothalamus|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Lung|eQTL","Artery_Coronary|eQTL","Artery_Tibial|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Muscularis|eQTL","Heart_Atrial_Appendage|eQTL","Liver|eQTL","Pancreas|eQTL","Pituitary|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Stomach|eQTL","Thyroid|eQTL","Whole_Blood|eQTL","SNP"],[10317,9534,10419,9744,10587,9454,9689,9978,9401,9758,9437,10547,10391,9130,9510,10702,9514,10614,9570,9481,10034,10234,10067,10574,10009,8781,9748,9972,10767,11005,9805,11052,9343,8021061],[0.00275908207788294,0.00162539398025109,0.00137202688676944,0.00132455878776237,0.0012462529546519,0.000888563558958296,0.000897115058737491,0.000814045778484186,0.000769612995489407,0.000758188469564052,0.000691729060028046,0.000723856771267351,0.000638066585480058,0.000573798032246404,0.000265797805527135,0.000159770790252405,1.40895742981435e-16,1.58927559066791e-07,1.60241294686404e-05,4.42080386328832e-23,1.2248278777507e-20,2.42448118876585e-15,1.9197689008331e-08,8.42674111114386e-13,2.70462023186048e-14,5.15264354338603e-11,1.7978705585842e-09,2.23967164087672e-14,3.25110705838976e-23,6.23003313544828e-11,8.30831661254089e-13,1.13886218371259e-08,6.36676459496702e-23,0.125896398161623],["1.95098%","1.14933%","0.97018%","0.93661%","0.88124%","0.62831%","0.63436%","0.57562%","0.5442%","0.53612%","0.48913%","0.51185%","0.45118%","0.40574%","0.18795%","0.11298%","0%","0.00011%","0.01133%","0%","0%","0%","1e-05%","0%","0%","0%","0%","0%","0%","0%","0%","1e-05%","0%","89.02275%"],[2.83547167827439,2.38525253345617,2.12702471791028,2.15879435710694,2.0148812342218,1.78977962326563,1.7748042785161,1.64824530938219,1.65168309411924,1.59945590940439,1.54116758963431,1.47536345925491,1.36411397246052,1.38732315507325,0.577003389057964,-0.0500790371984027,-27.6891539330634,-6.95486929286408,-2.23792615198768,-42.6603034522832,-37.092758926337,-24.9167524131301,-9.01561674361367,-19.0984723183322,-22.4825925491473,-14.7993944934995,-11.3516030977001,-22.6675212679105,-43.0948255345666,-14.8352821529961,-19.0371197204858,-9.63114053804768,-42.2808711775592,null],[0.16000474505019,0.207686136052364,0.225887954639548,0.229849961166334,0.23691883759455,0.28023323557405,0.278906622399775,0.292718577982259,0.300996467480759,0.303251094354279,0.317411448942917,0.310337293535275,0.330447539076059,0.348370287692213,0.511356951401964,0.659339778256648,701746.677327533,20.8944008092807,2.08096352973775,1252791151.28099,75264775.31242,169168.653031983,60.1180453731365,9074.0146262722,50649.6157010094,1160.41745060274,196.449122385569,55659.2212047836,1460877045.95298,1055.31988772408,9138.45508507141,78.0537424447302,1043926457.29606,null],[1.05346638116241e-34,2.69989320823018e-16,1.35805595530754e-12,3.72010807334249e-12,1.18473266830511e-10,4.54091033582842e-07,5.9476659627781e-07,4.76013150221889e-06,1.49584178882779e-05,2.09425674747232e-05,8.08546717320358e-05,0.000196241395060724,0.000757986318400578,0.00135730887336619,0.171436047353436,0.696051342915187,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,null],[1.97293252174679e-34,1.0252846022722e-16,2.44367407790253e-12,3.3601884933548e-12,1.45225506050052e-10,7.70054093651567e-07,8.05601690118815e-07,9.21741860409271e-06,1.56416916441952e-05,2.92445868602092e-05,0.000106388959459828,0.000129526059232809,0.000747256102482654,0.00118629247728328,0.303233423275476,0.938962310369822,0.0380810854568522,0.0290954550917996,0.0924187930697769,0.0384164907706896,0.0331823076258644,0.03147786573093,0.0330012065354319,0.0287867738952778,0.0334020437522625,0.0463134775714444,0.0358025033012106,0.0337300475581454,0.0273669017711864,0.0257161156074509,0.0352529622798589,0.0254533201271824,0.0398530333410075,null],[0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737,0.141420473311737]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) = 0.0015625”
<br><br>Number of selected tissue – fisher = 10
<br>Cells_Cultured_fibroblasts|eQTL Brain_Cerebellar_Hemisphere|eQTL
Artery_Aorta|eQTL Brain_Cortex|eQTL Esophagus_Mucosa|eQTL
Heart_Left_Ventricle|eQTL Brain_Cerebellum|eQTL Muscle_Skeletal|eQTL
Brain_Putamen_basal_ganglia|eQTL Spleen|eQTL<br><br><br>Number of
selected tissue – G = 10 <br>Cells_Cultured_fibroblasts|eQTL
Brain_Cerebellar_Hemisphere|eQTL Artery_Aorta|eQTL Brain_Cortex|eQTL
Esophagus_Mucosa|eQTL Heart_Left_Ventricle|eQTL Brain_Cerebellum|eQTL
Muscle_Skeletal|eQTL Brain_Putamen_basal_ganglia|eQTL
Spleen|eQTL<br><br><br><br></p>
<div id="htmlwidget-37e46cf92521dcbed3f8" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-37e46cf92521dcbed3f8">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">HB-panukb<\/caption>","data":[["1","7","13","25","20","24","10","16","15","8","3","2","21","12","6","22","17","4","5","9","11","14","18","19","23","26"],["Adipose_Subcutaneous|eQTL","Cells_Cultured_fibroblasts|eQTL","Heart_Atrial_Appendage|eQTL","Whole_Blood|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Thyroid|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Lung|eQTL","Liver|eQTL","Colon_Sigmoid|eQTL","Adrenal_Gland|eQTL","Adipose_Visceral_Omentum|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Esophagus_Muscularis|eQTL","Artery_Tibial|eQTL","Spleen|eQTL","Muscle_Skeletal|eQTL","Artery_Aorta|eQTL","Artery_Coronary|eQTL","Colon_Transverse|eQTL","Esophagus_Mucosa|eQTL","Heart_Left_Ventricle|eQTL","Pancreas|eQTL","Pituitary|eQTL","Stomach|eQTL","SNP"],[10547,10317,10009,9343,10767,11052,10067,10702,8781,10034,9437,10391,11005,10574,10614,9758,9978,10419,9514,10234,10587,9454,9748,9972,9805,8021061],[0.0040282308969548,0.00362095367874206,0.00289636027706746,0.00233105443096943,0.00199791466118437,0.00150336294241248,0.00146091605177989,0.00130117590244883,0.0012500337480191,0.00127196329256783,0.00119283498658952,0.00102560739183788,0.00108505831818002,0.000940547680101429,0.000936946810161185,0.000461274126652234,0.000138926465857192,6.54053320686778e-06,9.19830289954023e-21,9.37749972641281e-15,3.24551985752907e-12,2.62443012067869e-17,1.69658426940489e-06,8.9889020765263e-19,2.42062041530692e-12,0.0945828030069401],["3.3009%","2.96716%","2.3734%","1.91016%","1.63718%","1.23192%","1.19714%","1.06624%","1.02433%","1.0423%","0.97746%","0.84043%","0.88914%","0.77072%","0.76777%","0.37799%","0.11384%","0.00536%","0%","0%","0%","0%","0.00139%","0%","0%","77.50516%"],[3.47783615013488,3.39329473142773,3.20032049685173,3.0520436241175,2.75596837959874,2.4454434664475,2.51015123718338,2.33318806931163,2.49093056622527,2.37493259363339,2.37204499571727,2.1246952808561,2.1236343332823,2.02065922044466,2.01304766437409,1.38850007395418,0.166157135355834,-2.93301733592628,-37.0399267590673,-23.2780731392743,-17.4652668171162,-29.0774151454593,-4.21585012499341,-32.5048016264965,-17.6817841060837,null],[0.157366324668936,0.165662082281017,0.184599363987022,0.205211642624499,0.221351299459251,0.254584683488898,0.2581875029198,0.273382335135431,0.278814390673131,0.276454298130792,0.285359100027038,0.307520048936755,0.299069968421033,0.320997760479869,0.32160905002792,0.457313036242495,0.832028019264946,3.83221809179624,102185516.96468,101204.464141673,5440.02711592036,1913047.23372085,7.52417664486338,10336889.532748,6299.1195945538,null],[2.66022445019454e-47,9.57593712749112e-42,8.98519016435737e-31,1.9677438718008e-23,2.44925763799221e-18,2.01010696637755e-12,6.48811675125897e-12,1.70475467314153e-10,1.72027271738901e-10,8.42635490224994e-10,4.53393402662026e-09,1.21025285658949e-07,2.12284631044137e-07,1.21685038821373e-06,1.25802463650918e-06,0.00778520624667602,0.706490335101269,1,1,1,1,1,1,1,1,null],[2.74571791993075e-47,2.24865671138299e-41,4.84837916143373e-31,7.88883350756544e-24,4.08656196438616e-18,4.87163389064455e-12,3.8583392356981e-12,5.24306531381549e-10,1.68281728425319e-10,4.89097817704286e-10,1.731100788054e-09,2.67274055843418e-07,1.22523511905543e-07,2.09464049401731e-06,2.34200824692532e-06,0.0134184633049873,0.845934000448732,0.155135836596459,0.126531618281493,0.113046247362513,0.107021115915714,0.127733573724529,0.137321965859405,0.117759953633121,0.120877789231808,null],[0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618,0.122034201791618]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Adipose_Subcutaneous|eQTL Cells_Cultured_fibroblasts|eQTL
Heart_Atrial_Appendage|eQTL Whole_Blood|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Thyroid|eQTL
Esophagus_Gastroesophageal_Junction|eQTL Lung|eQTL Liver|eQTL
Colon_Sigmoid|eQTL<br><br><br>Number of selected tissue – G = 10
<br>Adipose_Subcutaneous|eQTL Cells_Cultured_fibroblasts|eQTL
Heart_Atrial_Appendage|eQTL Whole_Blood|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Thyroid|eQTL
Esophagus_Gastroesophageal_Junction|eQTL Lung|eQTL Liver|eQTL
Colon_Sigmoid|eQTL<br><br><br><br></p>
<div id="htmlwidget-99e959167faff8d1fc37" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-99e959167faff8d1fc37">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">Height-panukb<\/caption>","data":[["6","7","8","20","2","13","25","16","17","19","18","1","24","4","3","5","9","10","11","12","14","15","21","22","23","26"],["Artery_Tibial|eQTL","Cells_Cultured_fibroblasts|eQTL","Colon_Sigmoid|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Adipose_Visceral_Omentum|eQTL","Heart_Atrial_Appendage|eQTL","Whole_Blood|eQTL","Lung|eQTL","Muscle_Skeletal|eQTL","Pituitary|eQTL","Pancreas|eQTL","Adipose_Subcutaneous|eQTL","Thyroid|eQTL","Artery_Aorta|eQTL","Adrenal_Gland|eQTL","Artery_Coronary|eQTL","Colon_Transverse|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Mucosa|eQTL","Esophagus_Muscularis|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Spleen|eQTL","Stomach|eQTL","SNP"],[10614,10317,10034,10767,10391,10009,9343,10702,9978,9972,9748,10547,11052,10419,9437,9514,10234,10067,10587,10574,9454,8781,11005,9758,9805,8021061],[0.00843556082547345,0.00806422010868674,0.00611732263036315,0.0044085470240454,0.00354800892146392,0.00324896844894391,0.0031743264842523,0.00340555591647191,0.00246035947521901,0.0018788963950724,0.00172817687443953,0.00181763923947811,0.00169537651095658,0.00136503565844861,1.12111489519842e-06,2.37232352587426e-10,2.63711839671266e-14,4.80454545270721e-16,8.1030037294819e-06,8.32411224427654e-19,2.89853191160829e-11,4.08520710071938e-16,2.01362643993024e-17,1.24036888845272e-12,3.01322008208209e-07,0.368088847695247],["2.01112%","1.92259%","1.45843%","1.05104%","0.84588%","0.77458%","0.75679%","0.81192%","0.58657%","0.44795%","0.41201%","0.43334%","0.40419%","0.32544%","0.00027%","0%","0%","0%","0.00193%","0%","0%","0%","0%","0%","7e-05%","87.75588%"],[2.85178396067825,2.83514562945418,2.58664683943348,2.18856090472331,2.00694817547058,1.95635450078953,2.0019697224786,1.93647919452745,1.68142644135943,1.41240505830876,1.35150653066581,1.32319872203037,1.20679508285368,1.04905113266311,-5.956568136305,-14.4254885400343,-23.6029630853019,-27.5918048741427,-4.09364564521363,-33.9991016094699,-16.521412661497,-27.6173224370312,-30.8531020780739,-19.7044428288851,-7.30872188658276,null],[0.145557704850488,0.148803031857955,0.170486865992087,0.200480606101059,0.223258499968104,0.233223986862677,0.235916232655205,0.227851290556819,0.267781372418172,0.306237601180759,0.319257425720252,0.311343582657835,0.322338508682544,0.359092804783173,12.5115795954194,860.101554355727,81577.7787749586,604380.630069958,4.65390488183519,14520037.1122994,2460.637550126,655435.033739304,2952209.93693158,11894.9126729738,24.133560715308,null],[1.81834019877494e-41,1.49148089233335e-39,4.71571137406833e-27,8.64218265718275e-16,1.63607539306797e-11,6.48893993131783e-11,1.60827512664071e-10,1.94644536607461e-10,7.18142199233481e-07,9.0873371017974e-05,0.000328057871068558,0.000598540897758855,0.000849450039834276,0.00703221338179545,1,1,1,1,1,1,1,1,1,1,1,null],[1.44644847756664e-41,1.65227640684563e-39,6.89240405908649e-27,7.25665560674877e-16,1.19845006199441e-11,1.96736284524329e-10,1.53015456505804e-10,1.01882460176962e-10,8.46295571358515e-07,0.000182056720662807,0.000534904645059285,0.00048367144187123,0.00179385441374129,0.0127946865313206,0.0277606720548799,0.0257622648671106,0.020746373839965,0.0218115069397587,0.0244545585727924,0.0187409275301209,0.0262328779252025,0.0321759935692397,0.0164830450188367,0.0239343821450287,0.0239893582310751,null],[0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668,0.41944636791668]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Artery_Tibial|eQTL Cells_Cultured_fibroblasts|eQTL
Colon_Sigmoid|eQTL Skin_Not_Sun_Exposed_Suprapubic|eQTL
Adipose_Visceral_Omentum|eQTL Heart_Atrial_Appendage|eQTL
Whole_Blood|eQTL Lung|eQTL Muscle_Skeletal|eQTL
Pituitary|eQTL<br><br><br>Number of selected tissue – G = 10
<br>Artery_Tibial|eQTL Cells_Cultured_fibroblasts|eQTL
Colon_Sigmoid|eQTL Skin_Not_Sun_Exposed_Suprapubic|eQTL
Adipose_Visceral_Omentum|eQTL Heart_Atrial_Appendage|eQTL
Whole_Blood|eQTL Lung|eQTL Muscle_Skeletal|eQTL
Pituitary|eQTL<br><br><br><br></p>
<div id="htmlwidget-4a6133f685aa29daac40" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4a6133f685aa29daac40">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">HTN-panukb<\/caption>","data":[["2","1","17","13","18","4","24","25","20","21","22","6","16","12","3","23","7","5","8","9","10","11","14","15","19","26"],["Adipose_Visceral_Omentum|eQTL","Adipose_Subcutaneous|eQTL","Muscle_Skeletal|eQTL","Heart_Atrial_Appendage|eQTL","Pancreas|eQTL","Artery_Aorta|eQTL","Thyroid|eQTL","Whole_Blood|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Spleen|eQTL","Artery_Tibial|eQTL","Lung|eQTL","Esophagus_Muscularis|eQTL","Adrenal_Gland|eQTL","Stomach|eQTL","Cells_Cultured_fibroblasts|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Mucosa|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Pituitary|eQTL","SNP"],[10391,10547,9978,10009,9748,10419,11052,9343,10767,11005,9758,10614,10702,10574,9437,9805,10317,9514,10034,10234,10067,10587,9454,8781,9972,8021061],[0.00122148527084334,0.00112075734657357,0.0010754584574131,0.000896403198178173,0.00079004750532602,0.000695639322793918,0.000695356168126519,0.000650926344276653,0.000497848626840837,0.000487915872440174,0.000302958268960659,0.000312013450854123,0.00029545521589884,0.000217686581004615,0.000115501639202694,0.000128202540252328,0.000123024729521611,1.03696548829228e-07,1.69930417102834e-12,2.27256968628584e-13,1.49457547212086e-08,3.84546491448727e-10,2.97046354409584e-07,2.81558502456141e-08,5.55580120122813e-08,0.02773599609092],["3.26922%","2.99963%","2.87839%","2.39916%","2.11451%","1.86183%","1.86107%","1.74216%","1.33246%","1.30587%","0.81085%","0.83508%","0.79077%","0.58262%","0.30913%","0.34313%","0.32927%","0.00028%","0%","0%","4e-05%","0%","8e-04%","8e-05%","0.00015%","74.23351%"],[3.52622242544986,3.42525811871952,3.43945929217055,3.25424526143483,3.15437058116304,2.96053988915713,2.90115241930791,3.00310873351327,2.593149807767,2.55113292342158,2.19484724837694,2.14021205413522,2.07742627177859,1.78399780862603,1.26398622900185,1.33005853935149,1.23793200801574,-5.75971120269707,-16.8319313879659,-18.8635603097608,-7.75325109240622,-11.4637330556996,-4.70096536932706,-6.98325077841742,-6.43076906836513,null],[0.189450406933756,0.197461001638158,0.201414532929075,0.219973419031963,0.233898999636214,0.248893602268742,0.248954150606883,0.257088676657344,0.293253453232004,0.296177790062212,0.374690208961172,0.369279907486595,0.379382532683074,0.441407450419886,0.604935045521255,0.574313774607284,0.586228441741367,20.150321192214,4977.68955310224,13611.4616600494,53.0767781876372,330.894040201899,11.9056584368842,38.6704520964012,27.5289994052397,null],[9.70036695212704e-34,1.4641734840762e-30,1.08804591137289e-29,2.06867877816498e-22,7.49175303622914e-20,1.11380313774499e-16,2.88450181768168e-16,8.4070753753439e-15,2.36582204595615e-10,3.01937513376319e-10,2.18113801146886e-05,3.69429090772045e-05,3.88939299066618e-05,0.00205102058652133,0.0442227618152379,0.0485285920214708,0.0548457670152621,1,1,1,1,1,1,1,1,null],[7.76327451830875e-34,4.55299197325023e-30,4.41254785006255e-29,7.07592359525261e-23,1.40804257498451e-19,4.28186743983701e-16,1.11143997177914e-15,1.94277938675776e-15,3.72425328102606e-10,8.79652844902654e-10,1.50462773746036e-05,1.6345097804657e-05,4.11314245461617e-05,0.00173199401595302,0.0836699559054083,0.0573511164363806,0.0793532352959777,0.21634827585762,0.199364633647761,0.194946786194387,0.199473611853335,0.187454821339822,0.225116500022287,0.231635824376805,0.203476111293522,null],[0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204,0.0373631764184204]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Adipose_Visceral_Omentum|eQTL Adipose_Subcutaneous|eQTL
Muscle_Skeletal|eQTL Heart_Atrial_Appendage|eQTL Pancreas|eQTL
Artery_Aorta|eQTL Thyroid|eQTL Whole_Blood|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL<br><br><br>Number of selected tissue – G
= 10 <br>Adipose_Visceral_Omentum|eQTL Adipose_Subcutaneous|eQTL
Muscle_Skeletal|eQTL Heart_Atrial_Appendage|eQTL Pancreas|eQTL
Artery_Aorta|eQTL Thyroid|eQTL Whole_Blood|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL<br><br><br><br></p>
<div id="htmlwidget-d72b6eff4afa454a8832" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d72b6eff4afa454a8832">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">IBD-ebi-a-GCST004131<\/caption>","data":[["25","7","20","1","9","11","15","14","12","24","5","6","17","2","16","18","3","4","8","10","13","19","21","22","23","26"],["Whole_Blood|eQTL","Cells_Cultured_fibroblasts|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Adipose_Subcutaneous|eQTL","Colon_Transverse|eQTL","Esophagus_Mucosa|eQTL","Liver|eQTL","Heart_Left_Ventricle|eQTL","Esophagus_Muscularis|eQTL","Thyroid|eQTL","Artery_Coronary|eQTL","Artery_Tibial|eQTL","Muscle_Skeletal|eQTL","Adipose_Visceral_Omentum|eQTL","Lung|eQTL","Pancreas|eQTL","Adrenal_Gland|eQTL","Artery_Aorta|eQTL","Colon_Sigmoid|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Heart_Atrial_Appendage|eQTL","Pituitary|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Spleen|eQTL","Stomach|eQTL","SNP"],[8784,9791,10145,9944,9577,9984,8130,8896,9900,10450,8833,9991,9382,9711,10049,9180,8750,9763,9407,9421,9364,9309,10397,9108,9121,6427836],[0.00801450249633005,0.00744617890805996,0.00723078352827696,0.00445941878007008,0.00427294838299404,0.00393975010920658,0.00366885393135509,0.00375253684246642,0.00380630023208103,0.00335242553029128,0.00219792305681729,0.00210128341904755,0.00180486991827273,0.00140495850835365,0.00153784515879142,0.000254266193344316,1.86513165391144e-12,1.59640298583798e-08,1.85360445924459e-06,1.62854571048682e-09,6.10169776746506e-09,2.79325276780435e-07,8.57502343556455e-09,7.89796052474729e-07,3.66705952259398e-06,0.158415903477377],["3.682%","3.4209%","3.32194%","2.04873%","1.96306%","1.80999%","1.68553%","1.72398%","1.74868%","1.54016%","1.00976%","0.96536%","0.82919%","0.64546%","0.70651%","0.11681%","0%","1e-05%","0.00085%","0%","0%","0.00013%","0%","0.00036%","0.00168%","72.77889%"],[3.61149016586472,3.42940674426669,3.36453576691508,2.90121842303289,2.89610902431608,2.77330282368358,2.90748774752384,2.8399997031528,2.74729223058255,2.56625148240081,2.31218747967956,2.14403296030006,2.05486489991411,1.76991784538798,1.82607840003113,0.116768577993856,-18.5658185567306,-9.62060289259458,-4.8289079340463,-11.8675950455453,-10.5406467145315,-6.71094592470397,-10.30500479683,-5.64971961738705,-4.11577526719217,null],[0.220668917088822,0.228587788949927,0.231830760388561,0.29282252140009,0.298973061372995,0.311060765123807,0.322046849318475,0.318530327514091,0.31634071809895,0.336631279996281,0.414297894525579,0.423611810446712,0.456663413542001,0.516978031440111,0.4943371374388,1.21103880872308,14129.1332888602,152.721093049745,14.1730774840256,478.156753483529,247.027287955094,36.5103095377571,208.378363000609,21.7126994616194,10.0766478282717,null],[5.88204747161601e-27,5.97414004119882e-23,3.38359417530042e-21,8.81779557034885e-12,1.06410874919919e-10,1.63960411959778e-10,3.85168873136574e-10,9.03744131337085e-10,2.47435494515277e-09,5.85011538654911e-08,3.49810146522782e-05,6.85689126675928e-05,0.000472029218945553,0.00434141500501829,0.0048909001566192,0.455566602203082,1,1,1,1,1,1,1,1,1,null],[2.74496268359641e-26,5.2242526552494e-23,7.3250940349443e-22,8.94826691677637e-12,2.54209031911111e-11,5.1768146897815e-10,5.60546517153367e-10,6.9108997315766e-10,1.30090486254145e-09,5.82834451857107e-08,4.63544239077768e-05,0.00016659773384598,0.000724651746762349,0.00783031433086808,0.0043664995367881,0.924649611717022,0.281984268848251,0.255956686972968,0.276071111849473,0.264300918768102,0.265801219126167,0.26939417113408,0.241009381113962,0.278044616003457,0.292995218522235,null],[0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608,0.217667370529608]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Whole_Blood|eQTL Cells_Cultured_fibroblasts|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Adipose_Subcutaneous|eQTL
Colon_Transverse|eQTL Esophagus_Mucosa|eQTL Liver|eQTL
Heart_Left_Ventricle|eQTL Esophagus_Muscularis|eQTL
Thyroid|eQTL<br><br><br>Number of selected tissue – G = 10
<br>Whole_Blood|eQTL Cells_Cultured_fibroblasts|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Adipose_Subcutaneous|eQTL
Colon_Transverse|eQTL Esophagus_Mucosa|eQTL Liver|eQTL
Heart_Left_Ventricle|eQTL Esophagus_Muscularis|eQTL
Thyroid|eQTL<br><br><br><br></p>
<div id="htmlwidget-73d12cd98287c4e0164c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-73d12cd98287c4e0164c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">PLT-panukb<\/caption>","data":[["22","25","6","16","15","24","11","18","20","7","17","1","23","2","12","13","19","21","14","10","3","4","5","8","9","26"],["Spleen|eQTL","Whole_Blood|eQTL","Artery_Tibial|eQTL","Lung|eQTL","Liver|eQTL","Thyroid|eQTL","Esophagus_Mucosa|eQTL","Pancreas|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Cells_Cultured_fibroblasts|eQTL","Muscle_Skeletal|eQTL","Adipose_Subcutaneous|eQTL","Stomach|eQTL","Adipose_Visceral_Omentum|eQTL","Esophagus_Muscularis|eQTL","Heart_Atrial_Appendage|eQTL","Pituitary|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Heart_Left_Ventricle|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Adrenal_Gland|eQTL","Artery_Aorta|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","SNP"],[9758,9343,10614,10702,8781,11052,10587,9748,10767,10317,9978,10547,9805,10391,10574,10009,9972,11005,9454,10067,9437,10419,9514,10034,10234,8021061],[0.00622607866061505,0.00515506943910872,0.00487104181688208,0.00433858808483603,0.00370895993140976,0.00351059668424732,0.00324933524532554,0.00289498329423617,0.00249031973911884,0.00222992415098202,0.00169139908513165,0.00154893344945688,0.00142608462763484,0.00133340571429824,0.00140457162425743,0.00112968186485551,0.00103209634652202,0.00102914821998125,0.000227963977139367,0.000156881064998654,1.41996698850927e-17,1.60679464686911e-06,7.91473589958574e-06,4.13808317643224e-13,4.95644040457613e-12,0.180592151113673],["2.70397%","2.23884%","2.11548%","1.88424%","1.61079%","1.52464%","1.41118%","1.25728%","1.08154%","0.96845%","0.73457%","0.6727%","0.61935%","0.5791%","0.61%","0.49062%","0.44824%","0.44696%","0.099%","0.06813%","0%","7e-04%","0.00344%","0%","0%","78.43078%"],[3.34424405115324,3.19893799420505,3.01471876164131,2.89070313561204,2.93174600539376,2.64675941787456,2.61240840855293,2.57950201108139,2.32951001707744,2.26175935815628,2.01875818442635,1.8753099440581,1.86562501314386,1.74038111622503,1.77491901110943,1.6120361757162,1.52539566246684,1.42396658063556,0.0685792165094519,-0.367945249851206,-30.3366110047481,-4.98356024306898,-3.29820854489943,-20.1180127698516,-17.6547087182082,null],[0.152276135245674,0.16691623799541,0.171637396910147,0.181636715907065,0.196102604890776,0.201532012078582,0.209336324488342,0.221571661543965,0.238687623427765,0.252072872712395,0.289053554823288,0.301960305679914,0.314593051667326,0.325279192650505,0.316988529469372,0.353215449740237,0.369449747137334,0.36998864132533,0.784587096353677,0.945620396508071,3141944.92617249,9.34026497458212,4.20850361661833,18405.095882961,5318.05406114986,null],[6.92464952811297e-47,9.81747289118765e-38,3.85070796870208e-33,6.5859424409435e-28,6.40144836684659e-24,2.52851903719905e-20,1.86670709208338e-18,6.39801025561726e-17,6.56326240507364e-13,3.57928734027934e-11,1.44496046705305e-07,1.83598580703442e-06,6.44602463374947e-06,1.0576562901467e-05,1.22632608397168e-05,0.000274077916577586,0.00136309397544852,0.00235729733712966,0.448543208444369,0.80146755772605,1,1,1,1,1,null],[1.53191921014584e-47,1.80693657354065e-37,7.60672623887498e-33,7.2170445103407e-28,1.57908606015463e-24,2.23932092281513e-20,1.16708630148001e-18,1.59072056855293e-16,1.20665443956172e-12,4.73990758082698e-11,1.4204685257832e-07,1.937327576341e-06,5.37768450675349e-06,3.07653361569152e-05,1.40227659807872e-05,0.000304312947760013,0.000955881935432753,0.00180923079720489,0.931141955887258,0.678581457875206,0.0817651595912183,0.0733078389034727,0.108980867203643,0.072699645090843,0.0699092610610964,null],[0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628,0.230256735670628]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Spleen|eQTL Whole_Blood|eQTL Artery_Tibial|eQTL Lung|eQTL Liver|eQTL
Thyroid|eQTL Esophagus_Mucosa|eQTL Pancreas|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL
Cells_Cultured_fibroblasts|eQTL<br><br><br>Number of selected tissue – G
= 10 <br>Spleen|eQTL Whole_Blood|eQTL Artery_Tibial|eQTL Lung|eQTL
Liver|eQTL Thyroid|eQTL Esophagus_Mucosa|eQTL Pancreas|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL
Cells_Cultured_fibroblasts|eQTL<br><br><br><br></p>
<div id="htmlwidget-0796cbf5802287e6981e" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0796cbf5802287e6981e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">RA-panukb<\/caption>","data":[["22","13","1","2","3","4","5","6","7","8","9","10","11","12","14","15","16","17","18","19","20","21","23","24","25","26"],["Spleen|eQTL","Heart_Atrial_Appendage|eQTL","Adipose_Subcutaneous|eQTL","Adipose_Visceral_Omentum|eQTL","Adrenal_Gland|eQTL","Artery_Aorta|eQTL","Artery_Coronary|eQTL","Artery_Tibial|eQTL","Cells_Cultured_fibroblasts|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Mucosa|eQTL","Esophagus_Muscularis|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Lung|eQTL","Muscle_Skeletal|eQTL","Pancreas|eQTL","Pituitary|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Stomach|eQTL","Thyroid|eQTL","Whole_Blood|eQTL","SNP"],[9758,10009,10547,10391,9437,10419,9514,10614,10317,10034,10234,10067,10587,10574,9454,8781,10702,9978,9748,9972,10767,11005,9805,11052,9343,8021061],[0.000774503958109454,0.000566863339548211,1.78437918632275e-14,4.27679232166821e-21,2.91071705131614e-17,1.16328152138713e-07,2.81402583866912e-18,2.86232591298232e-18,4.80234890904638e-12,3.48005880608269e-11,9.33287694993363e-12,5.76236149829181e-17,3.38255971374211e-21,4.86065183748229e-05,1.74566616443594e-15,3.56238401535066e-20,1.432285073094e-08,4.221564925443e-17,7.4691860549137e-23,1.06473537480666e-12,3.22002622026472e-12,9.53671127899447e-15,3.85648044835935e-20,1.07955508483596e-16,3.50084770692968e-07,0.00274457766203204],["18.7303%","13.7088%","0%","0%","0%","0.00281%","0%","0%","0%","0%","0%","0%","0%","1.17548%","0%","0%","0.00035%","0%","0%","0%","0%","0%","0%","0%","0.00847%","66.37379%"],[5.44657881043695,5.1090770641671,-19.1250081418354,-34.3540696692737,-25.4322311943055,-3.42252774490765,-27.7767258716518,-27.8691169482771,-13.5077549223614,-11.4994120976374,-12.8352392448054,-24.8139078961998,-34.6073269921832,2.59780298848987,-21.340124028938,-32.0659081088727,-5.54388620230995,-25.1161692962032,-38.3377791838611,-14.9801219282474,-13.9501634548837,-19.7940233541111,-32.0968852935231,-24.2794698377773,-2.21176385020414,null],[0.419011969766977,0.475142166597568,77116.0245823087,157517614.712012,1909361.72513018,30.2033401328636,6140788.88664483,6088757.34333477,4700.68720204152,1746.20375948643,3371.94465919544,1357025.34034628,177119130.467364,1.4905402902896,246551.478034122,54578028.1267573,86.0745191158373,1585446.06004778,1191932244.68309,9983.14469830655,5740.61608724407,105484.558661101,52455694.4041906,991438.555309845,17.4112205944705,null],[1.62955309120386e-14,4.97991457648607e-10,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,null],[1.9701054526109e-15,5.63914315813952e-11,0.794310427800265,0.795802379071035,0.805198238568504,0.810563594681045,0.804421627728342,0.793673264697632,0.796516254349798,0.799274791190567,0.797319695899297,0.798939779586376,0.793929765970678,0.216858729561536,0.805026492759679,0.811955669858237,0.795449586138573,0.799811001755774,0.802081838143896,0.799870373336862,0.792227620791116,0.789997189435339,0.801516430485697,0.789560047004673,0.843381552904126,null],[0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804,0.00413503226708804]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 2
<br>Spleen|eQTL Heart_Atrial_Appendage|eQTL<br><br><br>Number of
selected tissue – G = 2 <br>Spleen|eQTL
Heart_Atrial_Appendage|eQTL<br><br><br><br></p>
<div id="htmlwidget-35c46d2ddbc1fab8b4dc" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-35c46d2ddbc1fab8b4dc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">RBC-panukb<\/caption>","data":[["1","25","8","7","24","12","21","15","17","13","14","2","6","16","20","3","23","9","18","10","4","5","11","19","22","26"],["Adipose_Subcutaneous|eQTL","Whole_Blood|eQTL","Colon_Sigmoid|eQTL","Cells_Cultured_fibroblasts|eQTL","Thyroid|eQTL","Esophagus_Muscularis|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Liver|eQTL","Muscle_Skeletal|eQTL","Heart_Atrial_Appendage|eQTL","Heart_Left_Ventricle|eQTL","Adipose_Visceral_Omentum|eQTL","Artery_Tibial|eQTL","Lung|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Adrenal_Gland|eQTL","Stomach|eQTL","Colon_Transverse|eQTL","Pancreas|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Artery_Aorta|eQTL","Artery_Coronary|eQTL","Esophagus_Mucosa|eQTL","Pituitary|eQTL","Spleen|eQTL","SNP"],[10547,9343,10034,10317,11052,10574,11005,8781,9978,10009,9454,10391,10614,10702,10767,9437,9805,10234,9748,10067,10419,9514,10587,9972,9758,8021061],[0.0051439971969443,0.0045751924986178,0.00364846311244668,0.00354736664774379,0.00281559541448818,0.00274789135886498,0.00278067683090336,0.00237978866780454,0.00207437241479528,0.001822369759306,0.00119382881092486,0.00115406484611593,0.00118920067880607,0.000691147568768692,0.000556124569792374,0.000497247854808841,0.000399647926213607,0.000273500749243242,8.15403926380814e-05,0.000156941631512123,6.84579111077667e-10,1.73141667606558e-16,7.6568851434829e-08,7.68895086696709e-10,3.54215154899423e-07,0.139539422166025],["2.90181%","2.58093%","2.05815%","2.00112%","1.58832%","1.55013%","1.56862%","1.34247%","1.17018%","1.02803%","0.67346%","0.65103%","0.67085%","0.38989%","0.31372%","0.2805%","0.22545%","0.15429%","0.046%","0.08853%","0%","0%","4e-05%","0%","2e-04%","78.71628%"],[3.3334678115871,3.33750020309602,3.03980551664036,2.98389134760319,2.68404117594415,2.70391451000952,2.67582347630913,2.74590022118682,2.48075481665923,2.34813186031625,1.9822065267494,1.85382914097539,1.86258632740887,1.31164626923598,1.0882299645933,1.10817402045744,0.851415065285674,0.429312400820571,-0.732240073219823,-0.109665235450346,-12.4866138574767,-27.5859518167657,-7.7854682076472,-12.3266136417681,-6.17221467978327,null],[0.15458448704982,0.163587246024574,0.182691272640212,0.185225688881687,0.20744952978534,0.20993513537461,0.208724140867062,0.225284885705755,0.241103096431184,0.257031948503467,0.316932227830049,0.322320896124152,0.317561971407781,0.415897601302819,0.463446928091292,0.490011636926274,0.546417179603929,0.660257824876622,1.20848574536985,0.871290362514618,416.969729484465,829116.485634377,39.4266963576027,393.443891265154,18.3308778849557,null],[9.99698123589742e-46,1.3357698128262e-40,6.38399895093334e-30,2.90353543861564e-28,4.33503112612472e-20,2.45254525223198e-19,5.77571671061562e-19,2.11864245458176e-17,4.88738545567117e-13,6.41012255719556e-11,2.08198301854666e-06,4.72971258979229e-06,5.67810126679264e-06,0.00549052181366466,0.0224102696564218,0.0516370556360258,0.174879468894327,0.440973503079976,0.75942516553716,0.770376365274021,1,1,1,1,1,null],[4.28212816347511e-46,2.74626663984375e-41,1.69473837403223e-29,4.87400611801065e-28,1.08358877295038e-19,1.90930661369353e-19,2.17196370730091e-19,2.10196976893544e-17,1.73072011049462e-13,3.23957008564176e-11,2.2188200926233e-06,9.89331715595731e-06,6.71690329159934e-06,0.00947627126623421,0.046514461700288,0.0559012686436082,0.172062880903996,0.543940670953203,0.490422634634629,0.898002138117452,0.0810681881981206,0.0954895277792178,0.079199572831335,0.0878767002773728,0.0938033708029954,null],[0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245,0.177268813334245]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Adipose_Subcutaneous|eQTL Whole_Blood|eQTL Colon_Sigmoid|eQTL
Cells_Cultured_fibroblasts|eQTL Thyroid|eQTL Esophagus_Muscularis|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL Liver|eQTL Muscle_Skeletal|eQTL
Heart_Atrial_Appendage|eQTL<br><br><br>Number of selected tissue – G =
10 <br>Adipose_Subcutaneous|eQTL Whole_Blood|eQTL Colon_Sigmoid|eQTL
Cells_Cultured_fibroblasts|eQTL Thyroid|eQTL Esophagus_Muscularis|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL Liver|eQTL Muscle_Skeletal|eQTL
Heart_Atrial_Appendage|eQTL<br><br><br><br></p>
<div id="htmlwidget-2a1b6e0e6b8a75d77da4" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a1b6e0e6b8a75d77da4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">SBP-ukb-a-360<\/caption>","data":[["6","17","3","25","22","4","18","21","13","2","20","1","9","5","11","16","19","24","7","8","10","12","14","15","23","26"],["Artery_Tibial|eQTL","Muscle_Skeletal|eQTL","Adrenal_Gland|eQTL","Whole_Blood|eQTL","Spleen|eQTL","Artery_Aorta|eQTL","Pancreas|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Heart_Atrial_Appendage|eQTL","Adipose_Visceral_Omentum|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Adipose_Subcutaneous|eQTL","Colon_Transverse|eQTL","Artery_Coronary|eQTL","Esophagus_Mucosa|eQTL","Lung|eQTL","Pituitary|eQTL","Thyroid|eQTL","Cells_Cultured_fibroblasts|eQTL","Colon_Sigmoid|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Esophagus_Muscularis|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Stomach|eQTL","SNP"],[10137,9543,8881,8903,9234,9879,9230,10466,9496,9855,10220,10071,9698,8972,10051,10168,9435,10528,9874,9527,9550,10037,8977,8253,9221,6396527],[0.0024917039167889,0.00147824244460351,0.00126519660258686,0.00131682174082966,0.0011754343020032,0.00118444773292575,0.000990377908751069,0.00092505133122909,0.000813895533476088,0.000653332023091318,0.000671511294852013,0.000391864294867858,0.000327476275571615,0.00026849904272105,0.000263764421340038,0.000125274126523949,3.07200253528143e-05,4.80147880673522e-05,9.62322572476442e-06,2.22680674443054e-16,7.98761390147187e-10,6.66070385583124e-11,3.50235002957523e-10,1.95782694957966e-06,2.72801538268549e-11,0.0426530038184387],["4.36481%","2.58949%","2.21629%","2.30672%","2.05905%","2.07484%","1.73488%","1.62045%","1.42573%","1.14447%","1.17631%","0.68644%","0.57365%","0.47034%","0.46205%","0.21945%","0.05381%","0.08411%","0.01686%","0%","0%","0%","0%","0.00343%","0%","74.71682%"],[3.60718744882121,3.14545870807228,3.06172612873459,3.09924552740618,2.94915813878803,2.88927809056938,2.77828503364682,2.58437480710396,2.55361884906689,2.29676400333789,2.2878417372375,1.76391307248707,1.62215341862608,1.50139621786048,1.37004171517998,0.613916300325958,-0.716853959393851,-0.379871840526317,-1.92306803107919,-26.3767547107274,-11.2863319610411,-13.8203210809144,-12.0489144566863,-3.33608412576317,-14.628177091887,null],[0.15772988472497,0.202612070924933,0.218499513448774,0.21429009966929,0.226485113271443,0.225658798452151,0.246257819471491,0.254653845930584,0.271145992746412,0.302124942588944,0.298071369373553,0.389017962003305,0.425246944881419,0.469327009600346,0.47350914888882,0.686048754221349,1.3839735713521,1.10722076545611,2.47217563191914,513869.700638905,271.322460461446,939.580971956176,409.745792586503,5.48044992443981,1468.15148175361,null],[2.73227165350231e-49,2.51278848006442e-25,5.52890435815407e-22,5.82347870773384e-22,4.99438343090277e-19,1.79188889818547e-18,2.75810130828194e-15,2.86458147041527e-13,1.34696575311014e-11,3.30458300086472e-08,4.74422745015446e-08,0.000191825412273036,0.000985855428298251,0.00398685616753428,0.0289922974999016,0.320702019951818,0.657035267903902,0.696987693403238,1,1,1,1,1,1,1,null],[6.07028883709818e-50,1.47640002429196e-25,2.24701578924517e-21,1.58171264811963e-22,4.84035662182572e-19,1.1274381961783e-18,4.14935711189223e-15,5.79767342334355e-13,2.05027380854716e-11,2.71062194238165e-08,1.91626474246459e-08,0.000424543773729936,0.00255637704315731,0.0102259824504837,0.0181370599841364,0.4166854034024,0.556648841954554,0.714244945579636,0.257650801216134,0.141735746296302,0.141286870100005,0.131520515958127,0.153784638810781,0.208497057994554,0.14829498040426,null],[0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579,0.057086213919579]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Artery_Tibial|eQTL Muscle_Skeletal|eQTL Adrenal_Gland|eQTL
Whole_Blood|eQTL Spleen|eQTL Artery_Aorta|eQTL Pancreas|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL Heart_Atrial_Appendage|eQTL
Adipose_Visceral_Omentum|eQTL<br><br><br>Number of selected tissue – G =
10 <br>Artery_Tibial|eQTL Muscle_Skeletal|eQTL Adrenal_Gland|eQTL
Whole_Blood|eQTL Spleen|eQTL Artery_Aorta|eQTL Pancreas|eQTL
Skin_Sun_Exposed_Lower_leg|eQTL Heart_Atrial_Appendage|eQTL
Adipose_Visceral_Omentum|eQTL<br><br><br><br></p>
<div id="htmlwidget-f41426fe0b90d0c09df4" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f41426fe0b90d0c09df4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">T1D-GCST90014023<\/caption>","data":[["25","18","7","2","16","4","24","6","20","12","21","9","19","10","1","3","5","8","11","13","14","15","17","22","23","26"],["Whole_Blood|eQTL","Pancreas|eQTL","Cells_Cultured_fibroblasts|eQTL","Adipose_Visceral_Omentum|eQTL","Lung|eQTL","Artery_Aorta|eQTL","Thyroid|eQTL","Artery_Tibial|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Esophagus_Muscularis|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Colon_Transverse|eQTL","Pituitary|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Adipose_Subcutaneous|eQTL","Adrenal_Gland|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Esophagus_Mucosa|eQTL","Heart_Atrial_Appendage|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Muscle_Skeletal|eQTL","Spleen|eQTL","Stomach|eQTL","SNP"],[9332,9737,10312,10385,10689,10410,11045,10609,10763,10565,10999,10225,9966,10061,10539,9422,9506,10026,10577,10004,9444,8774,9971,9748,9794,7539915],[0.00217073856913821,0.00201214918200805,0.00146673694470523,0.00151045922554393,0.00144375173489684,0.00120491257558369,0.00124710391635918,0.00116576011806381,0.00107835523708222,0.000864264921601958,0.000975750329954882,0.000741635102868937,0.000206247448177784,0.000131351487116694,1.87362511457658e-05,1.08174549032579e-13,2.00069980699909e-06,2.59247227505556e-17,2.93052238834806e-17,2.03611240412556e-18,6.96506012334467e-11,5.29807390345523e-10,4.03165502784968e-05,7.10066681259822e-10,4.70052073454107e-14,0.0345488465821666],["4.27066%","3.95865%","2.88562%","2.97164%","2.8404%","2.37052%","2.45352%","2.29349%","2.12153%","1.70033%","1.91967%","1.45908%","0.40577%","0.25842%","0.03686%","0%","0.00394%","0%","0%","0%","0%","0%","0.07932%","0%","0%","67.97058%"],[3.92721013201814,3.80886235571027,3.43532392440654,3.45764329223037,3.38362194374641,3.22923215239176,3.20443818153138,3.177262660728,3.08491464748268,2.88216943607022,2.96323912774952,2.76185838827691,1.50773409591637,1.04704701765302,-0.946785707628092,-19.8047303568825,-3.08058941222559,-28.2031690318918,-28.1341004154959,-30.7451270803577,-13.33956210168,-11.2369530915469,-0.125082075204228,-11.0493770677258,-20.6769406569619,null],[0.299207187697651,0.310122908975867,0.360573635951127,0.35552944404287,0.363323311140753,0.396403225411585,0.389872962490409,0.402790317577897,0.418293844276272,0.465855004040982,0.439116441619837,0.502037641814676,0.944881813460452,1.18275508243594,3.12664108136329,41137.8143306423,9.56588421324084,2657337.40111036,2499373.94747872,9482064.07124109,1621.21766040561,587.820540332379,2.13211454152233,507.754986887228,62406.6361796614,null],[6.71902230522886e-17,5.12714525429333e-15,4.70553208942408e-10,4.97015596140051e-10,6.21589371782022e-10,1.49586360197448e-08,2.23130077902086e-08,4.3396111184116e-07,4.71531870085083e-07,9.37220521986704e-06,1.13555015930387e-05,0.000153835194639762,0.221971135348758,0.223828963539454,1,1,1,1,1,1,1,1,1,1,1,null],[9.02522182242017e-17,4.16103794298644e-15,3.78548081464908e-10,1.74241921052147e-10,7.74435244664605e-10,5.16695105456624e-08,3.58801371835379e-08,1.17481226610893e-07,5.93634203920937e-07,1.99465349713296e-05,3.8409833618195e-06,0.000125380159826491,0.200488787003662,0.450259956476618,0.719528449984583,0.492360530616534,0.5341986298805,0.478815286638616,0.466983200248938,0.479297817281341,0.491861011285011,0.507657006015482,0.952227146628678,0.485013850899858,0.483944293917934,null],[0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791,0.0508291181861791]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Whole_Blood|eQTL Pancreas|eQTL Cells_Cultured_fibroblasts|eQTL
Adipose_Visceral_Omentum|eQTL Lung|eQTL Artery_Aorta|eQTL Thyroid|eQTL
Artery_Tibial|eQTL Skin_Not_Sun_Exposed_Suprapubic|eQTL
Esophagus_Muscularis|eQTL<br><br><br>Number of selected tissue – G = 10
<br>Whole_Blood|eQTL Pancreas|eQTL Cells_Cultured_fibroblasts|eQTL
Adipose_Visceral_Omentum|eQTL Lung|eQTL Artery_Aorta|eQTL Thyroid|eQTL
Artery_Tibial|eQTL Skin_Not_Sun_Exposed_Suprapubic|eQTL
Esophagus_Muscularis|eQTL<br><br><br><br></p>
<div id="htmlwidget-35e1d0e81f5ada6ecac0" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-35e1d0e81f5ada6ecac0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">WBC-ieu-b-30<\/caption>","data":[["25","7","20","12","16","17","15","4","13","22","9","1","6","24","23","11","2","14","18","3","5","8","10","19","21","26"],["Whole_Blood|eQTL","Cells_Cultured_fibroblasts|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Esophagus_Muscularis|eQTL","Lung|eQTL","Muscle_Skeletal|eQTL","Liver|eQTL","Artery_Aorta|eQTL","Heart_Atrial_Appendage|eQTL","Spleen|eQTL","Colon_Transverse|eQTL","Adipose_Subcutaneous|eQTL","Artery_Tibial|eQTL","Thyroid|eQTL","Stomach|eQTL","Esophagus_Mucosa|eQTL","Adipose_Visceral_Omentum|eQTL","Heart_Left_Ventricle|eQTL","Pancreas|eQTL","Adrenal_Gland|eQTL","Artery_Coronary|eQTL","Colon_Sigmoid|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Pituitary|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","SNP"],[8455,9443,9736,9541,9701,9110,7782,9402,9025,8721,9218,9589,9648,10070,8728,9570,9372,8514,8757,8428,8460,8999,9049,8865,9943,6013866],[0.00723359693363048,0.00322660755688645,0.00273885693688684,0.00242998745212589,0.00243487898431959,0.00224413241444895,0.00170975637405617,0.00160639221134053,0.00149047424316112,0.00133923691078483,0.00125378390278345,0.00105939322019709,0.00113193584863736,0.000877159635481841,0.000768046990417414,0.00070527884942851,0.000609250163963477,0.000447096256099781,0.000157214575947216,2.54217138641907e-05,2.98414027492962e-21,1.83159210259552e-15,1.55209038916343e-10,1.49061692920582e-20,2.1259352812426e-06,0.079742610545827],["6.38823%","2.84952%","2.41877%","2.146%","2.15032%","1.98187%","1.50994%","1.41866%","1.31629%","1.18272%","1.10726%","0.93559%","0.99965%","0.77465%","0.67829%","0.62285%","0.53805%","0.39485%","0.13884%","0.02245%","0%","0%","0%","0%","0.00188%","70.42332%"],[4.16699743361847,3.24917656934221,3.05472930367116,2.95530679415254,2.94068708620634,2.92196528518645,2.80755651680735,2.55608737170081,2.52211485141489,2.44938507426236,2.32802700905086,2.12009870459066,2.18019766969258,1.88239201690714,1.89257839151968,1.71522377358506,1.58976616431839,1.37632576539172,0.303021940802296,-1.48069208723816,-38.1655501549323,-24.8999299836233,-13.5581282425045,-36.603848437397,-4.12739322476674,null],[0.112121445938263,0.164413301035355,0.17798826483345,0.188635616437195,0.188455694336585,0.196078251599301,0.223939236281174,0.230950522736594,0.239599979455802,0.252548264227861,0.260901787897794,0.283532202877597,0.274408608722568,0.311286683970257,0.332436987474076,0.346808639010463,0.372936796245793,0.434936347576665,0.732288284853722,1.81972648237327,167933557.438755,214354.558588488,736.357306700197,75138769.8084017,6.29183495461529,null],[3.72319314519978e-119,3.61370999306353e-39,6.4285494520632e-32,5.73324317757392e-27,9.04989337457919e-27,5.97383078949685e-25,9.48629020561506e-18,4.29279685116258e-15,2.74724481242966e-14,2.44527310447181e-12,5.99441291879412e-11,9.27502671836748e-09,9.95753548454465e-09,6.9908535606257e-06,1.47958246332033e-05,0.000181322072864716,0.00086733353425122,0.0122190643099667,0.401840669910899,1,1,1,1,1,1,null],[3.24767856058895e-119,1.04817860115676e-39,2.31300897662954e-31,7.62705941575825e-27,1.01618012222811e-26,1.46655535196572e-24,3.71206411174049e-18,3.41721756613847e-15,5.76057229157949e-14,3.08849230253824e-12,7.96854146348361e-11,2.47595917328233e-08,3.99965615753347e-09,3.58008714093996e-06,1.32639368498459e-05,0.000110463920677487,0.000713438887046798,0.00984351467325784,0.693647612145195,0.282321587653075,0.102654526852414,0.0923227392689746,0.0914264176896116,0.0947808736812099,0.0901652434878077,null],[0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078,0.11323323781078]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00208333333333333” <br><br>Number of selected tissue – fisher = 10
<br>Whole_Blood|eQTL Cells_Cultured_fibroblasts|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Esophagus_Muscularis|eQTL Lung|eQTL
Muscle_Skeletal|eQTL Liver|eQTL Artery_Aorta|eQTL
Heart_Atrial_Appendage|eQTL Spleen|eQTL<br><br><br>Number of selected
tissue – G = 10 <br>Whole_Blood|eQTL Cells_Cultured_fibroblasts|eQTL
Skin_Not_Sun_Exposed_Suprapubic|eQTL Esophagus_Muscularis|eQTL Lung|eQTL
Muscle_Skeletal|eQTL Liver|eQTL Artery_Aorta|eQTL
Heart_Atrial_Appendage|eQTL Spleen|eQTL<br><br><br><br></p>
<div id="htmlwidget-a41f9b9304ecbba20fc2" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a41f9b9304ecbba20fc2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">T2D-panukb<\/caption>","data":[["27","6","9","1","32","21","25","15","2","33","3","10","4","31","12","11","18","5","7","8","13","14","16","17","19","20","22","23","24","26","28","29","30","34"],["Pituitary|eQTL","Artery_Tibial|eQTL","Brain_Cerebellum|eQTL","Adipose_Subcutaneous|eQTL","Thyroid|eQTL","Heart_Atrial_Appendage|eQTL","Muscle_Skeletal|eQTL","Cells_Cultured_fibroblasts|eQTL","Adipose_Visceral_Omentum|eQTL","Whole_Blood|eQTL","Adrenal_Gland|eQTL","Brain_Cortex|eQTL","Artery_Aorta|eQTL","Stomach|eQTL","Brain_Hypothalamus|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Esophagus_Gastroesophageal_Junction|eQTL","Artery_Coronary|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Putamen_basal_ganglia|eQTL","Colon_Sigmoid|eQTL","Colon_Transverse|eQTL","Esophagus_Mucosa|eQTL","Esophagus_Muscularis|eQTL","Heart_Left_Ventricle|eQTL","Liver|eQTL","Lung|eQTL","Pancreas|eQTL","Skin_Not_Sun_Exposed_Suprapubic|eQTL","Skin_Sun_Exposed_Lower_leg|eQTL","Spleen|eQTL","SNP"],[9961,10605,9679,10538,11040,10000,9969,10311,10381,9334,9431,9733,10411,9797,9123,9470,10058,9507,9561,9523,9500,9389,10025,10224,10578,10562,9446,8774,10693,9738,10758,10995,9750,8015736],[0.000778090462010502,0.000667446493182989,0.000587731716899257,0.000565790328525671,0.00054597138364687,0.000461570579442022,0.000398118571028904,0.000386649171063559,0.000393502856866974,0.000348209701683727,0.000167279162125824,0.000171432538011853,0.00013826507863875,0.000120675947792489,4.32015426637488e-05,4.16969678739119e-05,2.68932072994009e-05,2.23348383570483e-12,2.93358788940818e-06,9.40842716136579e-10,6.87262961110605e-06,1.01310313517041e-05,7.39989262441454e-07,5.31715464285062e-13,2.15794361556534e-14,1.81034426727871e-05,2.44417510841771e-09,8.34455990806663e-17,4.64811021031468e-11,8.51410935702625e-11,7.89258591270711e-14,1.64285451020557e-09,6.86866211967367e-08,0.0145230383780562],["3.81334%","3.27109%","2.88041%","2.77288%","2.67575%","2.26211%","1.95114%","1.89493%","1.92852%","1.70654%","0.81982%","0.84017%","0.67762%","0.59142%","0.21173%","0.20435%","0.1318%","0%","0.01438%","0%","0.03368%","0.04965%","0.00363%","0%","0%","0.08872%","1e-05%","0%","0%","0%","0%","1e-05%","0.00034%","71.17595%"],[3.76383569180913,3.54780400649471,3.51198235731034,3.38890616095914,3.30671189009472,3.2377202515999,3.09293996750884,3.02997682809882,3.04078148324021,3.02481160928239,2.28133227820889,2.27433801888664,1.99198013034918,1.91670298672011,0.960747874533574,0.887969873582659,0.389175767694828,-15.8583059352859,-1.77579574709698,-9.81677413553039,-0.918074630869166,-0.518265299103488,-3.20053120882065,-17.3662248624127,-20.6046310107237,-0.0554901637073994,-8.85396859594915,-25.9729486396799,-12.9403836886595,-12.2415661435548,-19.3247362135768,-9.40309025168439,-5.54979775865851,null],[0.222453260681713,0.23936064133102,0.254413231864218,0.259134757948977,0.263637348960737,0.285937161820965,0.307253506790382,0.311667464227512,0.309010758708987,0.327996795173447,0.470460600123805,0.464793118458824,0.516992961091787,0.553064063811813,0.921995467197156,0.93843826752046,1.16795482197269,4049.2076607546,3.53349061154867,197.289038668521,2.30886440813877,1.90186484615903,7.03492340729159,8298.92444973839,41194.7062355344,1.42311900725128,122.403921928134,662460.118858165,887.612171331757,655.830856057506,21540.302483342,149.300728947646,23.0901293161111,null],[7.76354843493667e-27,1.33288951233622e-21,3.30127628905003e-19,3.63992352410209e-17,7.13732581135e-17,1.36283804284281e-14,7.93293044949955e-12,1.1312650297286e-11,1.21465472278904e-11,9.20963799725135e-11,0.000130089195097295,0.000150397585630246,0.0019984687291422,0.0134544032563337,0.364128672941087,0.374977746926162,0.392939821793378,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,null],[2.43664838288359e-27,4.93343479995797e-22,2.30200988735413e-19,6.76081493518882e-18,8.27976427141235e-17,4.31706428289248e-14,1.06041826588539e-11,3.97808420624103e-11,2.39968618471047e-11,3.79294314105024e-10,0.000386724792939783,0.000337796790161417,0.00356483095319431,0.00819464650716514,0.368497139204556,0.409321926390696,0.75430022541171,0.332240159293362,0.479087239189701,0.331976061865922,0.639124736610363,0.765575109568052,0.364703263548331,0.314661684989936,0.306432412217027,0.968608865834104,0.334135202619082,0.351592347636332,0.303827506889788,0.326453087729903,0.302355283137736,0.297300682373073,0.332381009515087,null],[0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816,0.0204044186165816]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) = 0.0015625”
<br><br>Number of selected tissue – fisher = 10 <br>Pituitary|eQTL
Artery_Tibial|eQTL Brain_Cerebellum|eQTL Adipose_Subcutaneous|eQTL
Thyroid|eQTL Heart_Atrial_Appendage|eQTL Muscle_Skeletal|eQTL
Cells_Cultured_fibroblasts|eQTL Adipose_Visceral_Omentum|eQTL
Whole_Blood|eQTL<br><br><br>Number of selected tissue – G = 10
<br>Pituitary|eQTL Artery_Tibial|eQTL Brain_Cerebellum|eQTL
Adipose_Subcutaneous|eQTL Thyroid|eQTL Heart_Atrial_Appendage|eQTL
Muscle_Skeletal|eQTL Cells_Cultured_fibroblasts|eQTL
Adipose_Visceral_Omentum|eQTL Whole_Blood|eQTL<br><br><br></p>
</div>
<div id="em-convergence" class="section level2">
<h2>EM convergence</h2>
<pre class="r"><code>colnames(converge_df) &lt;- c(&quot;trait&quot;,&quot;num_EM_iter&quot;,&quot;converge&quot;)
cat(&quot;&lt;br&gt;&quot;)</code></pre>
<p><br></p>
<pre class="r"><code>DT::datatable(converge_df,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;EM convergence &#39;),options = list(pageLength = 30) )</code></pre>
<div id="htmlwidget-80db18273cfc4a9b8682" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-80db18273cfc4a9b8682">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">EM convergence <\/caption>","data":[["LDL-ukb-d-30780_irnt","aFib-ebi-a-GCST006414","ATH_gtexukb","BMI-panukb","HB-panukb","Height-panukb","HTN-panukb","IBD-ebi-a-GCST004131","PLT-panukb","RA-panukb","RBC-panukb","SBP-ukb-a-360","T1D-GCST90014023","WBC-ieu-b-30","T2D-panukb"],["73","97","100","74","100","52","80","94","65","70","67","100","100","90","100"],["TRUE","TRUE","FALSE","TRUE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","FALSE","TRUE","FALSE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>trait<\/th>\n      <th>num_EM_iter<\/th>\n      <th>converge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":30,"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,30,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;&lt;br&gt;&quot;)</code></pre>
<p><br></p>
</div>
</div>
<div id="psychiatric-traits" class="section level1">
<h1>Psychiatric traits</h1>
<div id="tissue-selection-1" class="section level2">
<h2>Tissue selection</h2>
<pre class="r"><code>converge_df &lt;- c()
for (trait in trait_psy){
  
  param &lt;- readRDS(paste0(folder_results,trait,&quot;/&quot;,trait,&quot;.thin1.shared_all.param.RDS&quot;))
  
  gwas_n &lt;- samplesize[trait]

  param_summarized_fisher &lt;- summarize_param(param = param,gwas_n = gwas_n,enrichment_test = &quot;fisher&quot;,alternative = &quot;greater&quot;)
  param_summarized_G &lt;- summarize_param(param = param,gwas_n = gwas_n,enrichment_test = &quot;G&quot;)
  
  param_df &lt;- data.frame(
    group = names(param_summarized_fisher$group_size),
    group_size = as.numeric(param_summarized_fisher$group_size[names(param_summarized_fisher$group_size)]),
    group_pve = as.numeric(param_summarized_fisher$group_pve[names(param_summarized_fisher$group_size)]),
    prop_heritability = as.numeric(param_summarized_fisher$prop_heritability[names(param_summarized_fisher$group_size)]),
    log_enrichment = as.numeric(param_summarized_fisher$log_enrichment[names(param_summarized_fisher$group_size)]),
    log_enrichment_se = as.numeric(param_summarized_fisher$log_enrichment_se[names(param_summarized_fisher$group_size)]),
    enrichment_pval_fisher = as.numeric(param_summarized_fisher$enrichment_pval[names(param_summarized_fisher$group_size)]),
    enrichment_pval_G = as.numeric(param_summarized_G$enrichment_pval[names(param_summarized_G$group_size)])
  )

  param_df$total_pve &lt;- param_summarized_fisher$total_pve
  
  param_df$prop_heritability &lt;- paste0(round(param_df$prop_heritability * 100, 5), &quot;%&quot;)
  
  param_df &lt;- param_df[order(param_df$enrichment_pval_fisher,decreasing = F),]
  
  param_df_qtl &lt;- param_df[-nrow(param_df),]
  threshold &lt;- 0.05/(nrow(param_df_qtl)-1)
  
  
  
  cat(&quot;&lt;br&gt;&quot;)
  cat(knitr::knit_print(DT::datatable(param_df, caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,trait,options = list(pageLength = 10)))))
  cat(&quot;&lt;br&gt;&quot;)
  
  
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  print(paste0(&quot;p-value cutoff(0.05/num_tissue) = &quot;,threshold))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(&quot;Number of selected tissue -- fisher = &quot;,min(10,sum(param_df_qtl$enrichment_pval_fisher &lt; threshold)),&quot;\n&quot;))
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(
    head(param_df_qtl$group[param_df_qtl$enrichment_pval_fisher &lt; threshold], 10), 
    collapse = &quot; &quot;
  ))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
    cat(paste0(&quot;Number of selected tissue -- G = &quot;,min(10,sum(param_df_qtl$enrichment_pval_G &lt; threshold)),&quot;\n&quot;))
  cat(&quot;&lt;br&gt;&quot;)
  cat(paste0(
    head(param_df_qtl$group[param_df_qtl$enrichment_pval_G &lt; threshold], 10), 
    collapse = &quot; &quot;
  ))
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
  cat(&quot;&lt;br&gt;&quot;)
 
  EM_iter &lt;- length(param$loglik_iters)
  converge &lt;- param$converged
  converge_df &lt;- rbind(converge_df,c(trait,EM_iter,converge))
}</code></pre>
<p><br></p>
<div id="htmlwidget-6f19a3278480c8d45e45" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6f19a3278480c8d45e45">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">SCZ-ieu-b-5102<\/caption>","data":[["2","3","8","6","4","7","5","1","9"],["Brain_Cerebellar_Hemisphere|eQTL","Brain_Cerebellum|eQTL","Brain_Putamen_basal_ganglia|eQTL","Brain_Hypothalamus|eQTL","Brain_Cortex|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Caudate_basal_ganglia|eQTL","SNP"],[8499,8675,8325,8037,8663,8450,8391,8458,5919307],[0.00409364392792245,0.00390377258598358,0.00314340642803043,0.00292365991894997,0.00292941131255254,0.00238589986515747,0.00222540680103113,0.000988070416950602,0.132907978621402],["2.63255%","2.51044%","2.02147%","1.88015%","1.88385%","1.53433%","1.43112%","0.63541%","85.47068%"],[3.06580466133712,2.99781575100452,2.82236168724506,2.78509801062851,2.71205806424122,2.53172757528793,2.46909768946703,1.64920361091727,null],[0.185157234080109,0.18949614048352,0.210640178253467,0.218246699109046,0.218057033355279,0.241184236889751,0.249596759685984,0.373063866707202,null],[3.58648363353904e-29,1.31357859696345e-27,1.87585160014387e-20,2.43083299771377e-18,1.06043387526305e-17,2.34835197368483e-13,2.56029617836889e-12,0.00061095803014381,null],[3.63333191351185e-29,4.31855319255529e-27,1.74223280261767e-20,7.66692628658329e-19,3.11660297583313e-18,7.30779731684732e-14,1.27596961006264e-12,0.00049570732365394,null],[0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798,0.15550124987798]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 8
<br>Brain_Cerebellar_Hemisphere|eQTL Brain_Cerebellum|eQTL
Brain_Putamen_basal_ganglia|eQTL Brain_Hypothalamus|eQTL
Brain_Cortex|eQTL Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Frontal_Cortex_BA9|eQTL
Brain_Caudate_basal_ganglia|eQTL<br><br><br>Number of selected tissue –
G = 8 <br>Brain_Cerebellar_Hemisphere|eQTL Brain_Cerebellum|eQTL
Brain_Putamen_basal_ganglia|eQTL Brain_Hypothalamus|eQTL
Brain_Cortex|eQTL Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Frontal_Cortex_BA9|eQTL
Brain_Caudate_basal_ganglia|eQTL<br><br><br><br></p>
<div id="htmlwidget-81a148ea6c961cf45e2c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-81a148ea6c961cf45e2c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">ASD-ieu-a-1185<\/caption>","data":[["5","3","8","1","2","4","6","7","9"],["Brain_Frontal_Cortex_BA9|eQTL","Brain_Cerebellum|eQTL","Brain_Putamen_basal_ganglia|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Cortex|eQTL","Brain_Hypothalamus|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","SNP"],[9047,9264,8981,9117,9142,9306,8656,9060,6618275],[0.0012060908617328,0.00113483894598657,0.000501560590407519,6.17149072884962e-06,1.51735468315218e-05,5.24430668943733e-07,1.57316183295214e-05,5.35888595291684e-07,0.0326080547566788],["3.39852%","3.19775%","1.4133%","0.01739%","0.04276%","0.00148%","0.04433%","0.00151%","91.88297%"],[3.29798193050509,3.21338552690316,2.42788860947112,-1.98492495895846,-1.08805018254389,-4.47082597265483,-0.997304702905783,-4.42242267847804,null],[0.337157247961749,0.347227476605569,0.517499118889071,4.63118387503428,2.95394105069558,15.8857031165072,2.90109622693706,15.7149606203426,null],[1.33189475383993e-10,1.63605952717584e-10,0.000411366248160805,1,1,1,1,1,null],[8.06516042853789e-11,5.84939289646397e-10,0.000717884953591324,0.527042741121671,0.653368719748126,0.420446651989986,0.680560234919293,0.427274656415875,null],[0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598,0.0354886821299598]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 3
<br>Brain_Frontal_Cortex_BA9|eQTL Brain_Cerebellum|eQTL
Brain_Putamen_basal_ganglia|eQTL<br><br><br>Number of selected tissue –
G = 3 <br>Brain_Frontal_Cortex_BA9|eQTL Brain_Cerebellum|eQTL
Brain_Putamen_basal_ganglia|eQTL<br><br><br><br></p>
<div id="htmlwidget-425deb884d24b4ea5f7c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-425deb884d24b4ea5f7c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">BIP-ieu-b-5110<\/caption>","data":[["7","1","2","5","4","6","3","8","9"],["Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Cortex|eQTL","Brain_Hypothalamus|eQTL","Brain_Cerebellum|eQTL","Brain_Putamen_basal_ganglia|eQTL","SNP"],[8437,8441,8487,8374,8644,8024,8664,8313,5893416],[0.000915619174813809,0.00069827460949207,0.0005642306041909,0.000447466954484746,0.000447417328142515,0.000390866391196667,0.000367290008521879,0.000247881428599749,0.0189630990911364],["3.97367%","3.03042%","2.44869%","1.94195%","1.94173%","1.69631%","1.59399%","1.07577%","82.29745%"],[3.51831459297238,3.24685252167996,3.02826831757253,2.80981188061325,2.77796719837124,2.71726959838371,2.57831609820606,2.22647082474866,null],[0.184471558409915,0.210164849448735,0.233061564363508,0.26097952841624,0.261000838568402,0.278850446254166,0.287515374540589,0.348968985900508,null],[5.99782560161398e-36,2.89793184063915e-24,8.67299989870039e-19,8.59754247913507e-14,1.34526077491793e-13,1.26798224821971e-11,4.41015054566717e-10,4.9443539159098e-06,null],[1.43099397353385e-35,6.74205995413948e-25,1.01699380567505e-18,8.12437544638355e-14,1.28482789330371e-13,9.22357289778394e-12,1.94287392114281e-10,2.60618442263203e-06,null],[0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787,0.0230421455905787]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 8
<br>Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Caudate_basal_ganglia|eQTL Brain_Cerebellar_Hemisphere|eQTL
Brain_Frontal_Cortex_BA9|eQTL Brain_Cortex|eQTL Brain_Hypothalamus|eQTL
Brain_Cerebellum|eQTL Brain_Putamen_basal_ganglia|eQTL<br><br><br>Number
of selected tissue – G = 8
<br>Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Caudate_basal_ganglia|eQTL Brain_Cerebellar_Hemisphere|eQTL
Brain_Frontal_Cortex_BA9|eQTL Brain_Cortex|eQTL Brain_Hypothalamus|eQTL
Brain_Cerebellum|eQTL
Brain_Putamen_basal_ganglia|eQTL<br><br><br><br></p>
<div id="htmlwidget-5674f84365e74fcb3e3c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5674f84365e74fcb3e3c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">MDD-ieu-b-102<\/caption>","data":[["7","1","3","8","2","4","5","6","9"],["Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellum|eQTL","Brain_Putamen_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Cortex|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Hypothalamus|eQTL","SNP"],[8364,8380,8519,8228,8389,8536,8276,7954,6004761],[0.000696036454462074,0.00015401288290682,0.000175791455914721,2.08383184042149e-05,1.14814293151762e-05,6.39751129831744e-06,5.28934434908137e-06,2.72867730700426e-10,0.0163210500179893],["4.0023%","0.88559%","1.01082%","0.11982%","0.06602%","0.03679%","0.03041%","0%","93.84823%"],[3.4215622222666,1.91128530742934,2.02709638761113,-0.0706524811683383,-0.686093468147009,-1.28828649112772,-1.44756849549748,-11.2801013621336,null],[0.18767490223344,0.393082408623802,0.368154179854811,1.06466490168426,1.43394546057291,1.9207161067786,2.11229254538522,294.045425653914,null],[9.65081668191069e-34,6.94745764312294e-05,7.68697452055594e-05,0.612868607974806,1,1,1,1,null],[4.45214063626271e-33,0.00020548708930549,3.35800539241597e-05,0.946468274750396,0.589272535242041,0.39451135282563,0.368537808345428,0.176000149509175,null],[0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074,0.0173908976875074]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 3
<br>Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Caudate_basal_ganglia|eQTL Brain_Cerebellum|eQTL<br><br><br>Number
of selected tissue – G = 3
<br>Brain_Nucleus_accumbens_basal_ganglia|eQTL
Brain_Caudate_basal_ganglia|eQTL
Brain_Cerebellum|eQTL<br><br><br><br></p>
<div id="htmlwidget-caaf43859fd0889ca800" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-caaf43859fd0889ca800">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">PD-ieu-b-7<\/caption>","data":[["3","6","4","1","2","5","7","8","9"],["Brain_Cerebellum|eQTL","Brain_Hypothalamus|eQTL","Brain_Cortex|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Putamen_basal_ganglia|eQTL","SNP"],[8587,7956,8565,8355,8416,8292,8375,8211,5925344],[0.000628979988667058,0.000609754208199508,0.000161459619633413,1.1725130111871e-07,1.66985868029406e-11,1.40247061926683e-07,1.02082209058163e-07,3.41342818063317e-10,0.00325417206810477],["13.51272%","13.09968%","3.46872%","0.00252%","0%","0.00301%","0.00219%","1e-05%","69.91114%"],[4.89315086804252,4.938430435655,3.53587180445177,-3.66699507874509,-12.5310202626377,-3.48034002014334,-3.80792703881257,-9.48879711479246,null],[0.222158236023741,0.225060491650813,0.411580577561999,14.9131805036241,1249.6302933562,13.6358976961281,15.9828080306623,276.392364050618,null],[2.01723010273708e-41,3.9403615746844e-40,4.85386126898852e-08,1,1,1,1,1,null],[7.21147554638019e-42,3.98602925135264e-41,2.01976599627166e-08,0.577849428305794,0.551728127500066,0.583348111272425,0.574694690046627,0.556757132538257,null],[0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826,0.00465472582321826]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 3
<br>Brain_Cerebellum|eQTL Brain_Hypothalamus|eQTL
Brain_Cortex|eQTL<br><br><br>Number of selected tissue – G = 3
<br>Brain_Cerebellum|eQTL Brain_Hypothalamus|eQTL
Brain_Cortex|eQTL<br><br><br><br></p>
<div id="htmlwidget-28fc46a89962ff28ef30" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-28fc46a89962ff28ef30">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">ADHD-ieu-a-1183<\/caption>","data":[["4","7","1","2","3","5","6","8","9"],["Brain_Cortex|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Cerebellum|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Hypothalamus|eQTL","Brain_Putamen_basal_ganglia|eQTL","SNP"],[8646,8367,8453,8468,8632,8370,7973,8308,5998545],[0.00264252375087995,0.000109885443892244,8.04826844707598e-05,1.20926389035845e-11,9.12226395718652e-09,2.39025006017979e-08,2.15302393020853e-05,7.08812073231789e-06,0.0591126836513525],["4.26391%","0.17731%","0.12986%","0%","1e-05%","4e-05%","0.03474%","0.01144%","95.38269%"],[3.43446434453682,0.287214431708039,-0.0344079213574304,-15.7471365737887,-9.14044206769461,-8.14635465728464,-1.29453098234113,-2.44672723447137,null],[0.238787672192201,1.14783385262967,1.34089714970828,3457.04483095048,125.867691906824,77.7578006805913,2.59129072003654,4.51569658402628,null],[9.90922903207941e-21,0.435341408696524,0.438646055545592,1,1,1,1,1,null],[3.19905712151473e-21,0.811249009169202,0.979411853077405,0.282720869246442,0.278372211873211,0.286160802227236,0.525414169738708,0.372826883827579,null],[0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487,0.061974226927487]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 1
<br>Brain_Cortex|eQTL<br><br><br>Number of selected tissue – G = 1
<br>Brain_Cortex|eQTL<br><br><br><br></p>
<div id="htmlwidget-018aa8438ee3bad3ceee" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-018aa8438ee3bad3ceee">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\" options=\"10\">NS-ukb-a-230<\/caption>","data":[["4","2","1","3","6","5","7","8","9"],["Brain_Cortex|eQTL","Brain_Cerebellar_Hemisphere|eQTL","Brain_Caudate_basal_ganglia|eQTL","Brain_Cerebellum|eQTL","Brain_Hypothalamus|eQTL","Brain_Frontal_Cortex_BA9|eQTL","Brain_Nucleus_accumbens_basal_ganglia|eQTL","Brain_Putamen_basal_ganglia|eQTL","SNP"],[9201,9029,9003,9166,8571,8919,8942,8876,6396527],[0.00149502234791659,0.00114387536875829,0.000986136290596589,0.000727943588745648,0.00016932174020556,0.000137240236379246,0.000160402083449793,5.76633676594426e-09,0.0324171628360251],["4.01487%","3.07187%","2.64826%","1.95489%","0.45471%","0.36856%","0.43076%","2e-05%","87.05606%"],[3.46765144746752,3.21880279040203,3.07330390279668,2.75178975823615,1.36048346841241,1.11061629479884,1.26399159374024,-8.96200209395558,null],[0.179704672103353,0.204450831510331,0.219716365960352,0.254818955062483,0.524210893448677,0.582006885631428,0.538525680685318,89.6135088912077,null],[3.78398062396499e-36,2.92095266221045e-26,7.71758334864147e-21,2.35062862400353e-14,0.0155766394922189,0.076356594457579,0.0768106539966259,1,null],[1.20858077958757e-36,6.75120766052203e-26,2.84737088368884e-21,4.94397724418352e-14,0.0338773854280799,0.106741008302096,0.0519728740250821,0.16379017258399,null],[0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135,0.0372371102584135]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group<\/th>\n      <th>group_size<\/th>\n      <th>group_pve<\/th>\n      <th>prop_heritability<\/th>\n      <th>log_enrichment<\/th>\n      <th>log_enrichment_se<\/th>\n      <th>enrichment_pval_fisher<\/th>\n      <th>enrichment_pval_G<\/th>\n      <th>total_pve<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br><br><br>[1] “p-value cutoff(0.05/num_tissue) =
0.00714285714285714” <br><br>Number of selected tissue – fisher = 4
<br>Brain_Cortex|eQTL Brain_Cerebellar_Hemisphere|eQTL
Brain_Caudate_basal_ganglia|eQTL Brain_Cerebellum|eQTL<br><br><br>Number
of selected tissue – G = 4 <br>Brain_Cortex|eQTL
Brain_Cerebellar_Hemisphere|eQTL Brain_Caudate_basal_ganglia|eQTL
Brain_Cerebellum|eQTL<br><br><br></p>
</div>
<div id="em-convergence-1" class="section level2">
<h2>EM convergence</h2>
<pre class="r"><code>colnames(converge_df) &lt;- c(&quot;trait&quot;,&quot;num_EM_iter&quot;,&quot;converge&quot;)
cat(&quot;&lt;br&gt;&quot;)</code></pre>
<p><br></p>
<pre class="r"><code>DT::datatable(converge_df,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;EM convergence &#39;),options = list(pageLength = 30) )</code></pre>
<div id="htmlwidget-de3a36f80a2aa43a84ec" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de3a36f80a2aa43a84ec">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">EM convergence <\/caption>","data":[["SCZ-ieu-b-5102","ASD-ieu-a-1185","BIP-ieu-b-5110","MDD-ieu-b-102","PD-ieu-b-7","ADHD-ieu-a-1183","NS-ukb-a-230"],["63","100","66","100","91","100","67"],["TRUE","FALSE","TRUE","FALSE","TRUE","FALSE","TRUE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>trait<\/th>\n      <th>num_EM_iter<\/th>\n      <th>converge<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":30,"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,30,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;&lt;br&gt;&quot;)</code></pre>
<p><br></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ctwas_0.5.19

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              locuszoomr_0.2.1           
  [7] GenomicRanges_1.48.0        base64enc_0.1-3            
  [9] fs_1.5.2                    rstudioapi_0.13            
 [11] DT_0.22                     ggrepel_0.9.1              
 [13] bit64_4.0.5                 AnnotationDbi_1.58.0       
 [15] fansi_1.0.3                 xml2_1.3.3                 
 [17] codetools_0.2-18            logging_0.10-108           
 [19] cachem_1.0.6                knitr_1.39                 
 [21] jsonlite_1.8.0              workflowr_1.7.0            
 [23] Rsamtools_2.12.0            dbplyr_2.1.1               
 [25] png_0.1-7                   readr_2.1.2                
 [27] compiler_4.2.0              httr_1.4.3                 
 [29] assertthat_0.2.1            Matrix_1.5-3               
 [31] fastmap_1.1.0               lazyeval_0.2.2             
 [33] cli_3.6.1                   later_1.3.0                
 [35] htmltools_0.5.2             prettyunits_1.1.1          
 [37] tools_4.2.0                 gtable_0.3.0               
 [39] glue_1.6.2                  GenomeInfoDbData_1.2.8     
 [41] dplyr_1.1.4                 rappdirs_0.3.3             
 [43] Rcpp_1.0.12                 Biobase_2.56.0             
 [45] jquerylib_0.1.4             vctrs_0.6.5                
 [47] Biostrings_2.64.0           rtracklayer_1.56.0         
 [49] crosstalk_1.2.0             xfun_0.41                  
 [51] stringr_1.5.1               irlba_2.3.5                
 [53] lifecycle_1.0.4             restfulr_0.0.14            
 [55] ensembldb_2.20.2            XML_3.99-0.14              
 [57] zlibbioc_1.42.0             zoo_1.8-10                 
 [59] scales_1.3.0                gggrid_0.2-0               
 [61] hms_1.1.1                   promises_1.2.0.1           
 [63] MatrixGenerics_1.8.0        ProtGenerics_1.28.0        
 [65] parallel_4.2.0              SummarizedExperiment_1.26.1
 [67] AnnotationFilter_1.20.0     LDlinkR_1.2.3              
 [69] yaml_2.3.5                  curl_4.3.2                 
 [71] memoise_2.0.1               ggplot2_3.5.1              
 [73] sass_0.4.1                  biomaRt_2.54.1             
 [75] stringi_1.7.6               RSQLite_2.3.1              
 [77] S4Vectors_0.34.0            BiocIO_1.6.0               
 [79] GenomicFeatures_1.48.3      BiocGenerics_0.42.0        
 [81] filelock_1.0.2              BiocParallel_1.30.3        
 [83] repr_1.1.4                  GenomeInfoDb_1.39.9        
 [85] rlang_1.1.2                 pkgconfig_2.0.3            
 [87] matrixStats_0.62.0          bitops_1.0-7               
 [89] evaluate_0.15               lattice_0.20-45            
 [91] purrr_1.0.2                 GenomicAlignments_1.32.0   
 [93] htmlwidgets_1.5.4           cowplot_1.1.1              
 [95] bit_4.0.4                   tidyselect_1.2.0           
 [97] magrittr_2.0.3              AMR_2.1.1                  
 [99] R6_2.5.1                    IRanges_2.30.0             
[101] generics_0.1.2              DelayedArray_0.22.0        
[103] DBI_1.2.2                   pgenlibr_0.3.3             
[105] pillar_1.9.0                whisker_0.4                
[107] mixsqp_0.3-43               KEGGREST_1.36.3            
[109] RCurl_1.98-1.7              tibble_3.2.1               
[111] crayon_1.5.1                utf8_1.2.2                 
[113] BiocFileCache_2.4.0         plotly_4.10.0              
[115] tzdb_0.4.0                  rmarkdown_2.25             
[117] progress_1.2.2              grid_4.2.0                 
[119] data.table_1.14.2           blob_1.2.3                 
[121] git2r_0.30.1                digest_0.6.29              
[123] tidyr_1.3.0                 httpuv_1.6.5               
[125] stats4_4.2.0                munsell_0.5.0              
[127] viridisLite_0.4.0           skimr_2.1.4                
[129] bslib_0.3.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
