<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-05-29" />

<title>Gene functions</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene functions</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-05-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree8b0a5539e9df1f97d9e68ca19a649d93ea8c0e07targetblank8b0a553a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/8b0a5539e9df1f97d9e68ca19a649d93ea8c0e07" target="_blank">8b0a553</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree8b0a5539e9df1f97d9e68ca19a649d93ea8c0e07targetblank8b0a553a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/8b0a5539e9df1f97d9e68ca19a649d93ea8c0e07" target="_blank">8b0a553</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    cv/

Unstaged changes:
    Modified:   analysis/realdata_final_genefunctions.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/realdata_final_genefunctions.Rmd</code>) and HTML
(<code>docs/realdata_final_genefunctions.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/379c37cf254582bba536b6ad9b574e53d774a5b5/analysis/realdata_final_genefunctions.Rmd" target="_blank">379c37c</a>
</td>
<td>
XSun
</td>
<td>
2025-06-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/379c37cf254582bba536b6ad9b574e53d774a5b5/docs/realdata_final_genefunctions.html" target="_blank">379c37c</a>
</td>
<td>
XSun
</td>
<td>
2025-06-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/db13c36c51b19c4cae101db42eeef0d964700872/analysis/realdata_final_genefunctions.Rmd" target="_blank">db13c36</a>
</td>
<td>
XSun
</td>
<td>
2025-06-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/db13c36c51b19c4cae101db42eeef0d964700872/docs/realdata_final_genefunctions.html" target="_blank">db13c36</a>
</td>
<td>
XSun
</td>
<td>
2025-06-06
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/00be92fbc4cf62b58cfb5a5be5bcc55f48920bbe/analysis/realdata_final_genefunctions.Rmd" target="_blank">00be92f</a>
</td>
<td>
XSun
</td>
<td>
2025-06-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/00be92fbc4cf62b58cfb5a5be5bcc55f48920bbe/docs/realdata_final_genefunctions.html" target="_blank">00be92f</a>
</td>
<td>
XSun
</td>
<td>
2025-06-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a2aa7f800d51e4b3203c8b02a7508c2c6d41dae/analysis/realdata_final_genefunctions.Rmd" target="_blank">0a2aa7f</a>
</td>
<td>
XSun
</td>
<td>
2025-05-30
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/0a2aa7f800d51e4b3203c8b02a7508c2c6d41dae/docs/realdata_final_genefunctions.html" target="_blank">0a2aa7f</a>
</td>
<td>
XSun
</td>
<td>
2025-05-30
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/28a04c55a29db4fd4d5e1670e11a1cdb2db1435c/docs/realdata_final_genefunctions.html" target="_blank">28a04c5</a>
</td>
<td>
XSun
</td>
<td>
2025-05-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/bd07e5175c4f5da600653789987a199e8becdbd4/analysis/realdata_final_genefunctions.Rmd" target="_blank">bd07e51</a>
</td>
<td>
XSun
</td>
<td>
2025-05-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/bd07e5175c4f5da600653789987a199e8becdbd4/docs/realdata_final_genefunctions.html" target="_blank">bd07e51</a>
</td>
<td>
XSun
</td>
<td>
2025-05-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/1cec5356bc465f19e1b6a4db67a1ae112d1621c6/analysis/realdata_final_genefunctions.Rmd" target="_blank">1cec535</a>
</td>
<td>
XSun
</td>
<td>
2025-05-29
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/1cec5356bc465f19e1b6a4db67a1ae112d1621c6/docs/realdata_final_genefunctions.html" target="_blank">1cec535</a>
</td>
<td>
XSun
</td>
<td>
2025-05-29
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><a
href="https://sq-96.github.io/multigroup_ctwas_analysis/realdata_final_multigroup_summary.html"
class="uri">https://sq-96.github.io/multigroup_ctwas_analysis/realdata_final_multigroup_summary.html</a></p>
<div id="ldl-ukb-d-30780_irnt" class="section level1">
<h1>LDL-ukb-d-30780_irnt</h1>
<div id="new-genes-identified-by-multi-group" class="section level2">
<h2>New genes identified by multi-group</h2>
<p>LDLR, PCSK9, CETP, CCNJ, PKN3, ABCA8, FCGRT, DNAJC13, LRCH4, HMGCR,
ASGR1, APOB, ADH1B, FLT3, ZDHHC18, USP39, TIMD4, ZFYVE1, ACP6, NPC1L1,
ERGIC3, MITF, PSRC1, SNX17, GABBR1, HMGN1, KIF13B, R3HDM2, MYPOP, SIPA1,
USP3, PGS1, FAM117B, ADRB1, PHC1, WASHC4, TPD52, THOP1, ZFP28, DMTN,
PDE4C, XPNPEP3</p>
<p><strong>Genes with Strong Evidence of Association with LDL
Cholesterol</strong></p>
<p>LDLR (Low-Density Lipoprotein Receptor): Mutations in LDLR are a
primary cause of familial hypercholesterolemia, leading to elevated LDL
cholesterol levels due to impaired clearance of LDL particles from the
bloodstream.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/LDLR.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/LDLR.png" width="150%">
</figure>
<p>PCSK9 (Proprotein Convertase Subtilisin/Kexin Type 9): PCSK9
regulates LDL receptor degradation. Gain-of-function mutations increase
LDL cholesterol levels, while loss-of-function mutations are associated
with reduced LDL cholesterol and a lower risk of coronary heart
disease.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/PCSK9.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/PCSK9.png" width="150%">
</figure>
<p>APOB (Apolipoprotein B): APOB encodes the primary protein component
of LDL particles. Mutations can impair LDL binding to its receptor,
resulting in elevated LDL cholesterol levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/APOB.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/APOB.png" width="150%">
</figure>
<p>CETP (Cholesteryl Ester Transfer Protein): CETP facilitates the
transfer of cholesteryl esters from HDL to LDL and VLDL particles.
Variants in CETP can influence HDL and LDL cholesterol levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/CETP.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/CETP.png" width="150%">
</figure>
<p>HMGCR (3-Hydroxy-3-Methylglutaryl-CoA Reductase): This enzyme is the
rate-limiting step in cholesterol biosynthesis and is the target of
statin therapy. Variants in HMGCR can affect LDL cholesterol levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/HMGCR.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/HMGCR.png" width="150%">
</figure>
<p>NPC1L1 (Niemann-Pick C1-Like 1): NPC1L1 is critical for intestinal
cholesterol absorption. Inhibitors like ezetimibe target this protein to
reduce LDL cholesterol levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/NPC1L1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/NPC1L1.png" width="150%">
</figure>
<p>ASGR1 (Asialoglycoprotein Receptor 1): Loss-of-function mutations in
ASGR1 have been associated with reduced LDL cholesterol levels and a
lower risk of coronary artery disease.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/ASGR1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/ASGR1.png" width="150%">
</figure>
<p>PSRC1 (Proline and Serine Rich Coiled-Coil 1): Variants near PSRC1
have been linked to LDL cholesterol levels in genome-wide association
studies (GWAS).</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/PSRC1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/PSRC1.png" width="150%">
</figure>
<p>TIMD4 (T-cell Immunoglobulin and Mucin Domain Containing 4): Variants
in TIMD4 have been associated with decreased serum triglyceride levels
and a reduced risk of coronary heart disease and ischemic stroke in
certain populations.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/multi/TIMD4.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/multi/TIMD4.png" width="150%">
</figure>
</div>
<div
id="genes-reported-by-single-eqtl-analysis-but-not-by-multi-group-analysis"
class="section level2">
<h2>Genes reported by single eQTL analysis but not by multi-group
analysis</h2>
<p>PARP9，DDX56，MZF1，CLDN23，FUT2，VIL1，KLHDC7A，WBP1L</p>
<p><strong>Genes with No Strong or Established Evidence for LDL
Association</strong></p>
<p>DDX56 RNA helicase with roles in ribosome biogenesis; no known links
to lipid metabolism or LDL levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/single/DDX56.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/single/DDX56.png" width="150%">
</figure>
<p>MZF1 A transcription factor; not directly linked to lipid or
cholesterol traits.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/single/MZF1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/single/MZF1.png" width="150%">
</figure>
<p>VIL1 Villin 1, an actin-binding protein in the intestinal brush
border; no direct LDL link known.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/single/VIL1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/single/VIL1.png" width="150%">
</figure>
<p>WBP1L Not currently known to be involved in lipid metabolism or LDL
levels.</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/LDL-ukb-d-30780_irnt/single/WBP1L.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/LDL-ukb-d-30780_irnt/single/WBP1L.png" width="150%">
</figure>
</div>
</div>
<div id="sbp-ukb-a-360" class="section level1">
<h1>SBP-ukb-a-360</h1>
<div id="new-genes-identified-by-multi-group-1" class="section level2">
<h2>New genes identified by multi-group</h2>
<p>BAG6, PKN2, PKP4, ENPEP, FES, RERE, NUDT5, RAB34, SLC9A3R2, TMBIM1,
HOXA11, DMWD, UVSSA, TMEM175, MTMR9, ADH1B, SP140L, PPP3R1, NAA60,
CCDC163, FHOD3, SGSM3, NPR1, PAQR5, CLCN6, THAP3, CAMK1D, TBX2, FBXO38,
REXO1</p>
<p><strong>Genes with Strong Evidence of Association with systolic blood
pressure</strong></p>
<p>ENPEP: A GWAS in East Asians identified genome-wide significant SBP
associations at the ENPEP locus. Kidney transcriptome and protein-level
Mendelian randomisation linked genetically predicted ENPEP expression to
elevated SBP/DBP/PP</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/SBP-ukb-a-360/multi/ENPEP.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/SBP-ukb-a-360/multi/ENPEP.png" width="150%">
</figure>
<p>NPR1 Rare/low-frequency coding variants show association with SBP in
&gt;300k sample-size meta-analyses. It encodes the receptor for
atrial/brain natriuretic peptides, central in blood volume and BP
regulation</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/SBP-ukb-a-360/multi/NPR1.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/SBP-ukb-a-360/multi/NPR1.png" width="150%">
</figure>
<p>FES A TWAS reported inverse association between predicted FES
expression in vascular tissues and SBP</p>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz/SBP-ukb-a-360/multi/FES.png" width="150%">
</figure>
<figure class="half">
<img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/figures/lz_ld/SBP-ukb-a-360/multi/FES.png" width="150%">
</figure>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
