<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="shengqian" />


<title>Simulation Results using simulated (correlated)</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation Results using simulated (correlated)</h1>
<h4 class="author">shengqian</h4>
<h4 class="date">2023-12-3</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed2ccdfd598074f65c43296454f791bb474b0b74ctargetblankd2ccdfda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d2ccdfd598074f65c43296454f791bb474b0b74c" target="_blank">d2ccdfd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistreed2ccdfd598074f65c43296454f791bb474b0b74ctargetblankd2ccdfda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/d2ccdfd598074f65c43296454f791bb474b0b74c" target="_blank">d2ccdfd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/simulation_2tissues_same.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>source(&quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/analysis/simulation_help_functions.R&quot;)</code></pre>
<div id="simulation-1-liver-and-three-95-correlated-tissues" class="section level2">
<h2>Simulation 1: Liver and three 95% correlated tissues</h2>
<div id="shared-effect-size-parameters" class="section level3">
<h3>Shared effect size parameters</h3>
<p>30% PVE and 2.5e-4 prior inclusion for SNPs, 5% PVE and 0.015 prior inclusion for Liver, 1% PVE and 0.003 prior inclusion for other three mixed groups.</p>
<pre class="r"><code>results_dir &lt;- &quot;/project/xinhe/shengqian/cTWAS_simulation/simulation_same_two_tissues/&quot;
runtag = &quot;ukb-s80.45-3_corr&quot;
configtag &lt;- 1
simutags &lt;- paste(1, 1:5, sep = &quot;-&quot;)
thin &lt;- 0.1
sample_size &lt;- 45000
PIP_threshold &lt;- 0.8</code></pre>
<pre class="r"><code>#results using PIP threshold (gene+tissue)
results_df[,c(&quot;simutag&quot;, &quot;n_causal&quot;, &quot;n_detected_pip&quot;, &quot;n_detected_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal n_detected_pip n_detected_pip_in_causal
1     1-1      118              0                        0
2     1-2       95              0                        0
3     1-3      116              0                        0
4     1-4      117              0                        0
5     1-5      113              0                        0</code></pre>
<pre class="r"><code>#mean percent causal using PIP &gt; 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)</code></pre>
<pre><code>[1] NaN</code></pre>
<pre class="r"><code>#results using combined PIP threshold
results_df[,c(&quot;simutag&quot;, &quot;n_causal_combined&quot;, &quot;n_detected_comb_pip&quot;, &quot;n_detected_comb_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal_combined n_detected_comb_pip n_detected_comb_pip_in_causal
1     1-1               118                  31                            28
2     1-2                95                  32                            28
3     1-3               116                  26                            26
4     1-4               117                  41                            36
5     1-5               113                  40                            33</code></pre>
<pre class="r"><code>#mean percent causal using combined PIP &gt; 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)</code></pre>
<pre><code>[1] 0.8882353</code></pre>
<pre class="r"><code>#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df))))]</code></pre>
<pre><code>  simutag    prior_snp prior_weight1 prior_weight2
1     1-1 0.0002108796   0.005154730   0.005154730
2     1-2 0.0002573964   0.007027990   0.007027990
3     1-3 0.0002761674   0.006451763   0.006451763
4     1-4 0.0002896721   0.012034873   0.012034873
5     1-5 0.0002807151   0.009465010   0.009465010</code></pre>
<pre class="r"><code>colMeans(results_df[,setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df)))])</code></pre>
<pre><code>    prior_snp prior_weight1 prior_weight2 
 0.0002629661  0.0080268734  0.0080268734 </code></pre>
<pre class="r"><code>#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;prior_var&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag prior_var_snp prior_var_weight1 prior_var_weight2
1     1-1      9.443700          32.95046          32.95046
2     1-2      8.605591          22.28596          22.28596
3     1-3      7.982523          26.83210          26.83210
4     1-4      7.486181          14.84881          14.84881
5     1-5      7.210972          21.07834          21.07834</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;prior_var&quot;, names(results_df))])</code></pre>
<pre><code>    prior_var_snp prior_var_weight1 prior_var_weight2 
         8.145793         23.599134         23.599134 </code></pre>
<pre class="r"><code>#PVE and mean PVE
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;pve&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag   pve_snp pve_weight1 pve_weight2
1     1-1 0.2344277  0.02791214  0.02791214
2     1-2 0.2607447  0.02573879  0.02573879
3     1-3 0.2595045  0.02844846  0.02844846
4     1-4 0.2552697  0.02936695  0.02936695
5     1-5 0.2382823  0.03278560  0.03278560</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;pve&quot;, names(results_df))])</code></pre>
<pre><code>    pve_snp pve_weight1 pve_weight2 
 0.24964577  0.02885039  0.02885039 </code></pre>
<pre class="r"><code>#TWAS results
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;twas&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag n_detected_twas n_detected_twas_in_causal n_detected_comb_twas
1     1-1             136                        34                   68
2     1-2             146                        34                   73
3     1-3             150                        37                   75
4     1-4             168                        44                   84
5     1-5             146                        38                   73
  n_detected_comb_twas_in_causal
1                             34
2                             34
3                             37
4                             44
5                             38</code></pre>
<pre class="r"><code>sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)</code></pre>
<pre><code>[1] 0.5013405</code></pre>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="separate-effect-size-parameters" class="section level3">
<h3>Separate effect size parameters</h3>
<p>For the cTWAS analysis, each tissue had its own prior inclusion parameter end effect size parameter.</p>
<pre class="r"><code>results_dir &lt;- &quot;/project/xinhe/shengqian/cTWAS_simulation/simulation_same_two_tissues/&quot;
runtag = &quot;ukb-s80.45-3_corr&quot;
configtag &lt;- 2
simutags &lt;- paste(1, 1:5, sep = &quot;-&quot;)
thin &lt;- 0.1
sample_size &lt;- 45000
PIP_threshold &lt;- 0.8</code></pre>
<pre class="r"><code>results_df &lt;- data.frame(simutag=as.character(),
                              n_causal=as.integer(),
                              n_causal_combined=as.integer(),
                              n_detected_pip=as.integer(),
                              n_detected_pip_in_causal=as.integer(),
                              n_detected_comb_pip=as.integer(),
                              n_detected_comb_pip_in_causal=as.integer(),
                              pve_snp=as.numeric(),
                              pve_weight1=as.numeric(),
                              pve_weight2=as.numeric(),
                              prior_weight1=as.numeric(),
                              prior_weight2=as.numeric(),
                              prior_var_snp=as.numeric(),
                              prior_var_weight1=as.numeric(),
                              prior_var_weight2=as.numeric(),
                              n_detected_twas=as.integer(),
                              n_detected_twas_in_causal=as.integer(),
                              n_detected_comb_twas=as.integer(),
                              n_detected_comb_twas_in_causal=as.integer())

for (i in 1:length(simutags)){
  simutag &lt;- simutags[i]
  
  #load genes with true simulated effect
  load(paste0(results_dir, runtag, &quot;_simu&quot;, simutag, &quot;-pheno.Rd&quot;))
  true_genes &lt;- unlist(sapply(1:22, function(x){phenores$batch[[x]]$id.cgene}))
  true_genes_combined &lt;- unique(sapply(true_genes, function(x){unlist(strsplit(x, &quot;[|]&quot;))[1]}))
  
  #load cTWAS results
  ctwas_res &lt;- data.table::fread(paste0(results_dir, runtag, &quot;_simu&quot;, simutag, &quot;_config&quot;, configtag, &quot;_LDR.susieIrss.txt&quot;))
  ctwas_gene_res &lt;- ctwas_res[ctwas_res$type!=&quot;SNP&quot;,]
  
  #number of causal genes
  n_causal &lt;- length(true_genes)
  n_causal_combined &lt;- length(true_genes_combined)
  
  #number of gene+tissue combinations with cTWAS PIP &gt; threshold
  n_ctwas_genes &lt;- sum(ctwas_gene_res$susie_pip &gt; PIP_threshold)
  
  #number of cTWAS genes that are causal
  n_causal_detected &lt;- sum(ctwas_gene_res$id[ctwas_gene_res$susie_pip &gt; PIP_threshold] %in% true_genes)
  
  #collapse gene+tissues to genes and compute combined PIP
  ctwas_gene_res$gene &lt;- sapply(ctwas_gene_res$id, function(x){unlist(strsplit(x,&quot;[|]&quot;))[1]})
  ctwas_gene_res_combined &lt;- aggregate(ctwas_gene_res$susie_pip, by=list(ctwas_gene_res$gene), FUN=sum)
  colnames(ctwas_gene_res_combined) &lt;- c(&quot;gene&quot;, &quot;pip_combined&quot;)
  
  #number of genes with combined PIP &gt; threshold
  n_ctwas_genes_combined &lt;- sum(ctwas_gene_res_combined$pip_combined &gt; PIP_threshold)
  
  #number of cTWAS genes using combined PIP that are causal
  n_causal_detected_combined &lt;- sum(ctwas_gene_res_combined$gene[ctwas_gene_res_combined$pip_combined &gt; PIP_threshold] %in% true_genes_combined)
  
  #collect number of SNPs analyzed by cTWAS
  ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, runtag, &quot;_simu&quot;, simutag, &quot;_config&quot;, configtag, &quot;_LDR.s1.susieIrss.txt&quot;))
  n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
  rm(ctwas_res_s1)
  
  #load estimated parameters
  load(paste0(results_dir, runtag, &quot;_simu&quot;, simutag, &quot;_config&quot;, configtag, &quot;_LDR.s2.susieIrssres.Rd&quot;))
  
  #estimated group prior (all iterations)
  estimated_group_prior_all &lt;- group_prior_rec
  estimated_group_prior_all[&quot;SNP&quot;,] &lt;- estimated_group_prior_all[&quot;SNP&quot;,]*thin #adjust parameter to account for thin argument
  
  #estimated group prior variance (all iterations)
  estimated_group_prior_var_all &lt;- group_prior_var_rec
  
  #set group size
  group_size &lt;- c(table(ctwas_gene_res$type), structure(n_snps, names=&quot;SNP&quot;))
  group_size &lt;- group_size[rownames(estimated_group_prior_all)]
  
  #estimated group PVE (all iterations)
  estimated_group_pve_all &lt;- estimated_group_prior_var_all*estimated_group_prior_all*group_size/sample_size #check PVE calculation
  
  #multitissue TWAS analysis with bonferroni adjusted threshold for z scores
  load(paste0(results_dir, runtag, &quot;_simu&quot;, simutag, &quot;_config&quot;, configtag,&quot;_LDR_z_gene.Rd&quot;))
  alpha &lt;- 0.05
  sig_thresh &lt;- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes &lt;- z_gene$id[abs(z_gene$z)&gt;sig_thresh]
  twas_genes_combined &lt;- unique(sapply(twas_genes, function(x){unlist(strsplit(x, &quot;[|]&quot;))[1]}))
  
  n_twas_genes &lt;- length(twas_genes)
  n_twas_genes_combined &lt;- length(twas_genes_combined)
  
  n_twas_genes_in_causal &lt;- sum(twas_genes %in% true_genes)
  n_twas_genes_in_causal_combined &lt;- sum(twas_genes_combined %in% true_genes_combined)

  results_current &lt;- data.frame(simutag=as.character(simutag),
                                n_causal=as.integer(n_causal),
                                n_causal_combined=as.integer(n_causal_combined),
                                n_detected_pip=as.integer(n_ctwas_genes),
                                n_detected_pip_in_causal=as.integer(n_causal_detected),
                                n_detected_comb_pip=as.integer(n_ctwas_genes_combined),
                                n_detected_comb_pip_in_causal=as.integer(n_causal_detected_combined),
                                pve_snp=as.numeric(rev(estimated_group_pve_all[&quot;SNP&quot;,])[1]),
                                pve_weight1=as.numeric(rev(estimated_group_pve_all[&quot;Liver&quot;,])[1]),
                                pve_weight2=as.numeric(rev(estimated_group_pve_all[&quot;Liver_copy&quot;,])[1]),
                                prior_snp=as.numeric(rev(estimated_group_prior_all[&quot;SNP&quot;,])[1]),
                                prior_weight1=as.numeric(rev(estimated_group_prior_all[&quot;Liver&quot;,])[1]),
                                prior_weight2=as.numeric(rev(estimated_group_prior_all[&quot;Liver_copy&quot;,])[1]),
                                prior_var_snp=as.numeric(rev(estimated_group_prior_var_all[&quot;SNP&quot;,])[1]),
                                prior_var_weight1=as.numeric(rev(estimated_group_prior_var_all[&quot;Liver&quot;,])[1]),
                                prior_var_weight2=as.numeric(rev(estimated_group_prior_var_all[&quot;Liver_copy&quot;,])[1]),
                                n_detected_twas=as.integer(n_twas_genes),
                                n_detected_twas_in_causal=as.integer(n_twas_genes_in_causal),
                                n_detected_comb_twas=as.integer(n_twas_genes_combined),
                                n_detected_comb_twas_in_causal=as.integer(n_twas_genes_in_causal_combined))
  
  results_df &lt;- rbind(results_df, results_current)
}</code></pre>
<pre class="r"><code>#results using PIP threshold (gene+tissue)
results_df[,c(&quot;simutag&quot;, &quot;n_causal&quot;, &quot;n_detected_pip&quot;, &quot;n_detected_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal n_detected_pip n_detected_pip_in_causal
1     1-1      118              0                        0
2     1-2       95              0                        0
3     1-3      116              0                        0
4     1-4      117              0                        0
5     1-5      113              0                        0</code></pre>
<pre class="r"><code>#mean percent causal using PIP &gt; 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)</code></pre>
<pre><code>[1] NaN</code></pre>
<pre class="r"><code>#results using combined PIP threshold
results_df[,c(&quot;simutag&quot;, &quot;n_causal_combined&quot;, &quot;n_detected_comb_pip&quot;, &quot;n_detected_comb_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal_combined n_detected_comb_pip n_detected_comb_pip_in_causal
1     1-1               118                  31                            28
2     1-2                95                  32                            28
3     1-3               116                  26                            26
4     1-4               117                  41                            36
5     1-5               113                  40                            33</code></pre>
<pre class="r"><code>#mean percent causal using combined PIP &gt; 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)</code></pre>
<pre><code>[1] 0.8882353</code></pre>
<pre class="r"><code>#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df))))]</code></pre>
<pre><code>  simutag    prior_snp prior_weight1 prior_weight2
1     1-1 0.0002108796   0.005154730   0.005154730
2     1-2 0.0002573964   0.007027990   0.007027990
3     1-3 0.0002761674   0.006451763   0.006451763
4     1-4 0.0002896721   0.012034873   0.012034873
5     1-5 0.0002807151   0.009465010   0.009465010</code></pre>
<pre class="r"><code>colMeans(results_df[,setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df)))])</code></pre>
<pre><code>    prior_snp prior_weight1 prior_weight2 
 0.0002629661  0.0080268734  0.0080268734 </code></pre>
<pre class="r"><code>#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;prior_var&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag prior_var_snp prior_var_weight1 prior_var_weight2
1     1-1      9.443700          32.95046          32.95046
2     1-2      8.605591          22.28596          22.28596
3     1-3      7.982523          26.83210          26.83210
4     1-4      7.486181          14.84881          14.84881
5     1-5      7.210972          21.07834          21.07834</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;prior_var&quot;, names(results_df))])</code></pre>
<pre><code>    prior_var_snp prior_var_weight1 prior_var_weight2 
         8.145793         23.599134         23.599134 </code></pre>
<pre class="r"><code>#PVE and mean PVE
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;pve&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag   pve_snp pve_weight1 pve_weight2
1     1-1 0.2344277  0.02791214  0.02791214
2     1-2 0.2607447  0.02573879  0.02573879
3     1-3 0.2595045  0.02844846  0.02844846
4     1-4 0.2552697  0.02936695  0.02936695
5     1-5 0.2382823  0.03278560  0.03278560</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;pve&quot;, names(results_df))])</code></pre>
<pre><code>    pve_snp pve_weight1 pve_weight2 
 0.24964577  0.02885039  0.02885039 </code></pre>
<pre class="r"><code>#TWAS results
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;twas&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag n_detected_twas n_detected_twas_in_causal n_detected_comb_twas
1     1-1             136                        34                   68
2     1-2             146                        34                   73
3     1-3             150                        37                   75
4     1-4             168                        44                   84
5     1-5             146                        38                   73
  n_detected_comb_twas_in_causal
1                             34
2                             34
3                             37
4                             44
5                             38</code></pre>
<pre class="r"><code>sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)</code></pre>
<pre><code>[1] 0.5013405</code></pre>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/simulation_2tissues_same.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggpubr_0.6.0    ggplot2_3.4.0   workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0  xfun_0.35         bslib_0.4.1       purrr_1.0.2      
 [5] carData_3.0-4     colorspace_2.0-3  vctrs_0.6.3       generics_0.1.3   
 [9] htmltools_0.5.4   yaml_2.3.6        utf8_1.2.2        rlang_1.1.1      
[13] jquerylib_0.1.4   later_1.3.0       pillar_1.8.1      glue_1.6.2       
[17] withr_2.5.0       DBI_1.1.3         lifecycle_1.0.3   stringr_1.5.0    
[21] munsell_0.5.0     ggsignif_0.6.4    gtable_0.3.1      evaluate_0.19    
[25] labeling_0.4.2    knitr_1.41        callr_3.7.3       fastmap_1.1.0    
[29] httpuv_1.6.7      ps_1.7.2          fansi_1.0.3       highr_0.9        
[33] broom_1.0.2       Rcpp_1.0.9        backports_1.2.1   promises_1.2.0.1 
[37] scales_1.2.1      cachem_1.0.6      jsonlite_1.8.4    abind_1.4-5      
[41] farver_2.1.0      fs_1.5.2          digest_0.6.31     stringi_1.7.8    
[45] rstatix_0.7.2     processx_3.8.0    dplyr_1.0.10      getPass_0.2-2    
[49] rprojroot_2.0.3   grid_4.1.0        cli_3.6.1         tools_4.1.0      
[53] magrittr_2.0.3    sass_0.4.4        tibble_3.1.8      car_3.1-1        
[57] tidyr_1.3.0       whisker_0.4.1     pkgconfig_2.0.3   data.table_1.14.6
[61] assertthat_0.2.1  rmarkdown_2.19    httr_1.4.4        rstudioapi_0.14  
[65] R6_2.5.1          git2r_0.30.1      compiler_4.1.0   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
