---
title: "cTWAS paper figures"
output: html_document
date: '2025-8-18'
editor_options: 
  chunk_output_type: console
---


```{r echo=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(ctwas)
library(data.table)
library(ggplot2)
library(ggrepel)
library(egg)
library(grid)
library(dplyr)
library(pheatmap)
library(EnsDb.Hsapiens.v86)
library(ComplexHeatmap)
library(GenomicRanges)
library(mapgen)
library(RSQLite)
library(RColorBrewer)
source("/project/xinhe/shengqian/cTWAS_paper_figures/plot_function.R")
source("/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R")
```

## Figure 1: Workflow of M-cTWAS

<figure class="half">
    <img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/data/figures/multi_cTWAS_workflow.png" width="150%">
</figure>

## Figure 2: cTWAS simulations (Correlated Brain Tissues)
```{r echo=FALSE}
pve_gene <- c(0.02, 0.02, 0, 0, 0,
              0.01, 0.01, 0, 0, 0,
              0.003, 0.003, 0, 0, 0)

group_size <- c(8798,8772,8478,8596,8542, 
                22802,19188,17467,17104,22142, 
                3561,2935,2578,2659,3142)
pve_snp <- 0.3
sigma_theta <- 0.02
sigma_beta <- 0.02
gene_prior <- pve_gene/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/group_size
snp_prior <- pve_snp/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/5000000

thins <- c(1)

prior_true <- gene_prior
  
enrichment_true_correlated <- log(gene_prior/snp_prior,base = exp(1))

pve_true <- pve_gene

phe_true_correlated <- pve_gene/(sum(pve_gene)+pve_snp)
```

```{r echo=FALSE}
pve_gene <- c(0.02, 0.02, 0, 0, 0,
              0.01, 0.01, 0, 0, 0,
              0.003, 0.003, 0, 0, 0)

group_size <- c(8170,7395,8274,8561,9774,
                13273, 14245, 15620, 19358, 24842,
                3479, 1930, 2834, 3271, 4751)
pve_snp <- 0.3
sigma_theta <- 0.02
sigma_beta <- 0.02
gene_prior <- pve_gene/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/group_size
snp_prior <- pve_snp/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/5000000

thins <- c(1)

prior_true <- gene_prior
  
#enrichment_true <- log(gene_prior/snp_prior,base = exp(1))
enrichment_true_uncorrelated <- gene_prior/snp_prior
pve_true <- pve_gene

phe_true_uncorrelated <- pve_gene/(sum(pve_gene)+pve_snp)
```

### Figure 2a: Inflated PHE by single-group analysis
```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
make_phe_plot_correlated()$plot
```

```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
make_phe_plot_uncorrelated()$plot
```

### Figure 2b: PHE
```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag = "five_correlated_brain_tissues_three_omics"
plot_PHE_correlated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",phe_true)
```

```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/"
runtag = "five_uncorrelated_tissues_three_omics"
plot_PHE_uncorrelated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",phe_true)
```

```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag = "five_correlated_brain_tissues_three_omics"
plot_PHE(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",phe_true_correlated)
```

```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/"
runtag = "five_uncorrelated_tissues_three_omics"
plot_PHE(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",phe_true_uncorrelated)
```

### Figure S: Enrichment (original scale)
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag = "five_correlated_brain_tissues_three_omics"
plot_enrichment_correlated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",enrichment_true_correlated)
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/"
runtag = "five_uncorrelated_tissues_three_omics"
plot_enrichment_uncorrelated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",enrichment_true_uncorrelated)
```

### Figure 2c: PIP calibration
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
df1 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/molecular_PIP_calibration_df.RDS")
df2 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/gene_PIP_calibration_df.RDS")
fig1 <- cp_plot(df1$susie_pip, df1$ifcausal, df1$runtag, mode ="PIP", main = "MT level")
fig2 <- cp_plot(df2$susie_pip, df2$ifcausal, df2$runtag, mode ="PIP", main = "Gene level")
grid.arrange(fig1, fig2, ncol = 2, top = "Correlated tissues")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
df1 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/molecular_PIP_calibration_df.RDS")
df2 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/gene_PIP_calibration_df.RDS")
fig1 <- cp_plot(df1$susie_pip, df1$ifcausal, df1$runtag, mode ="PIP", main = "MT level")
fig2 <- cp_plot(df2$susie_pip, df2$ifcausal, df2$runtag, mode ="PIP", main = "Gene level")
grid.arrange(fig1, fig2, ncol = 2, top = "Uncorrelated tissues")
```

### Figure 2d: Power and False discovery rate
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/multi_ctwas_simu_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/single_ctwas_simu_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/coloc_simu_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/smr_simu_res.RDS"
p1 <- plot_molecular_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)

multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/multi_ctwas_simu_gene_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/single_ctwas_simu_gene_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/coloc_simu_gene_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/smr_simu_gene_res.RDS"
p2 <- plot_gene_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)
grid.arrange(p1, p2, ncol = 2, top = "Correlated tissues")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/multi_ctwas_simu_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/single_ctwas_simu_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/coloc_simu_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/smr_simu_res.RDS"
p1 <- plot_molecular_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)

multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/multi_ctwas_simu_gene_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/single_ctwas_simu_gene_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/coloc_simu_gene_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_uncorrelated_tissues_three_omics/smr_simu_gene_res.RDS"
p2 <- plot_gene_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)
grid.arrange(p1, p2, ncol = 2, top = "Uncorrelated tissues")
```

## Figure 3: cTWAS estimates genetic architecture of complex traits from GTEx

### Figure 3a: Modalities are mostly independent
```{r echo=FALSE}
folder_results_single_E <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_single_S <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/splicing/"
folder_results_single_ST <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/stability/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/STMM_snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
top_tissue <- "Liver"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)$prop_heritability
param_single_E <- readRDS(paste0(folder_results_single_E, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single_E <- summarize_param(param_single_E, gwas_n)$prop_heritability
param_single_S <- readRDS(paste0(folder_results_single_S, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single_S <- summarize_param(param_single_S, gwas_n)$prop_heritability
param_single_ST <- readRDS(paste0(folder_results_single_ST, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single_ST <- summarize_param(param_single_ST, gwas_n)$prop_heritability

library(ggplot2)
library(dplyr)

# Extract values from your objects
bar1_eQTL <- ctwas_parameters_single_E[1]
bar1_sQTL <- ctwas_parameters_single_S[1]
bar1_stQTL <- ctwas_parameters_single_ST[1]

bar2_eQTL <- ctwas_parameters_multi[1]
bar2_sQTL <- ctwas_parameters_multi[2]
bar2_stQTL <- ctwas_parameters_multi[3]

# Build data frame
df <- data.frame(
  group = c("Single-modality", "Single-modality", "Single-modality", "Multi-modality", "Multi-modality", "Multi-modality"),
  QTL   = c("eQTL", "sQTL", "stQTL",
            "eQTL", "sQTL", "stQTL"),
  value = c(bar1_eQTL, bar1_sQTL, bar1_stQTL,
            bar2_eQTL, bar2_sQTL, bar2_stQTL)
)


# Plot
ggplot(df, aes(x = group, y = value, fill = QTL)) +
  geom_bar(stat = "identity") +
  geom_text(aes(label = round(value, 3)),
            position = position_stack(vjust = 0.5), size = 4, color = "black") +
  ylab("Proportion of PVE") +
  xlab("") +
  ggtitle(trait) +
  scale_fill_manual(values = c("eQTL" = "#1f77b4",
                               "sQTL" = "#ff7f0e",
                               "stQTL" = "#2ca02c")) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "right")

```

### Figure 3b: Tissues are often correlated: but one tissue is not enough.
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
thin <- 1
vgs <- "shared_all"
L <-5
colors <- c("#ff7f0e", "#2ca02c", "#d62728",  "#9467bd", "#8c564b", "#e377c2", "#7f7f7f",  "#bcbd22",  "#17becf",  "#f7b6d2",  "#c5b0d5",  "#9edae5", "#ffbb78",  "#98df8a",  "#ff9896" )

fix_panel_size <- function(plot, width = 2.1, height = 2) {
  set_panel_size(plot, 
                 width = unit(width, "in"), 
                 height = unit(height, "in"))
}

folder_results_multiqtl <- "/project/xinhe/xsun/multi_group_ctwas/24.diff_qtls_0519/snakemake_outputs/"

trait <- "IBD-ebi-a-GCST004131"

gwas_n <- samplesize[trait]


file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Whole_Blood.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p1 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)


file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Colon_Transverse.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p2 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)

file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/24.diff_qtls_0519/snakemake_outputs/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131.eonly.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p3 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)

pie1 <- fix_panel_size(p1)
pie2 <- fix_panel_size(p2)
pie3 <- fix_panel_size(p3)

# Calculate widths of each gtable (plot + legend)
widths <- unit.c(grobWidth(pie1), grobWidth(pie2),grobWidth(pie3))

# Arrange plots with their natural widths
p <- grid.arrange(pie1, pie2, pie3,
                  ncol = 3, 
                  widths = widths,
                  top = paste0(trait))
```

### Figure 3b: Shared pattern
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=6, fig.width=10}
trait <- "IBD-ebi-a-GCST004131"
out <- plot_pairwise_PVE(trait, samplesize, n_top = 10)
```

### Figure 3c: Partition of h2g (Multiple tissues contribute and Contribution of each modality)
```{r echo=FALSE, message=FALSE, warning=FALSE}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
p <- generate_piecharts_for_trait(ctwas_parameters_single, ctwas_parameters_multi, title_top = title)
grid::grid.draw(p)
invisible(NULL)
```


### Figure 3d: Existing methods may overestimate the contribution of molQTL.
```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE}
gwas <- readRDS("/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.3qtls.thin1.shared_all.L5.regionmerge_with_mappingtable_finemap_res.RDS")
gwas <- anno_finemap_res(gwas,
                       snp_map = snp_map,
                       mapping_table = mapping_table,
                       add_gene_annot = TRUE,
                       map_by = "molecular_id",
                       drop_unmapped = TRUE,
                       add_position = TRUE,
                       use_gene_pos = "mid")
gwas <- gwas[gwas$type=="SNP",]
gwas$p <- z2p(gwas$z)
gwas_loci <- count_loci_500kb(gwas,chr_col = "chrom",pos_col = "pos",p_col   = "p",id_col  = "id",p_thresh = 5e-8, window_bp = 5e5)$loci

coloc_genes <- readRDS("/project/xinhe/shengqian/coloc_GWAS_analysis/results/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.coloc_res.RDS")
coloc_genes <- coloc_genes[coloc_genes$group=="expression_Skin_Not_Sun_Exposed_Suprapubic",]
coloc_genes <- coloc_genes[coloc_genes$PP4>0.8,]
coloc_genes <- unique(coloc_genes$gene_name)

coloc_genes <- AnnotationDbi::select(
  EnsDb.Hsapiens.v86,
  keys = coloc_genes,
  keytype = "SYMBOL",
  columns = c("GENEID", "SYMBOL","GENESEQSTART","GENESEQEND")
)
coloc_genes$GENESEQSTART <- as.numeric(coloc_genes$GENESEQSTART)
coloc_genes$GENESEQEND   <- as.numeric(coloc_genes$GENESEQEND)

out <- map_loci_to_genes(
  loci_df   = gwas_loci,
  genes_df  = coloc_genes,
  loci_chr   = "chr",
  loci_start = "locus_start",
  loci_end   = "locus_end",
  loci_id    = "lead_id",
  gene_start = "GENESEQSTART",
  gene_end   = "GENESEQEND",
  gene_chr   = NULL,          # change to your gene chr column if you have it
  gene_symbol= "SYMBOL",
  gene_id    = "GENEID"
)

out$n_loci
out$n_loci_with_genes
```

```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE}
plot_fraction_of_coloc_signals_in_TWAS_regions("LDL-ukb-d-30780_irnt", pretty_labels = TRUE)
```

```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE}
plot_MESC_eQTL_heritability("LDL-ukb-d-30780_irnt")$plot
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
make_eqtl_prop_plot("LDL-ukb-d-30780_irnt")$plot
```

## Figure 4: Multi-cTWAS improves the discovery of candidate genes

### Figure 4a: Incorporating multiple modality and tissues improves discovery power
```{r echo=FALSE, message=FALSE, warning=FALSE}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
folder_results_multi_E <- "/project/xinhe/xsun/multi_group_ctwas/26.multi_tissue_eQTL_0805/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
PIP_cutoff <- 0.8
combined_pip_by_group_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi_E <- readRDS(paste0(folder_results_multi_E,trait,"/",trait,".eqtlonly.combined_pip_rmmapping_bygroup_final.RDS"))
#print(plot_overlap_barplot_multi_vs_single(combined_pip_by_group_multi = combined_pip_by_group_multi, combined_pip_by_group_single = combined_pip_by_group_single, PIP_cutoff = PIP_cutoff,tissue = top_tissue,trait = trait,return_unique_genes = T)$plot)
print(plot_overlap_barplot_multi_vs_multi_E_vs_single(combined_pip_by_group_multi = combined_pip_by_group_multi, combined_pip_by_group_single = combined_pip_by_group_single, combined_pip_by_group_multi_E = combined_pip_by_group_multi_E, PIP_cutoff = PIP_cutoff,tissue = top_tissue,trait = trait,return_unique_genes = T)$plot)
```

```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE}
traits <- c(
  "SBP-ukb-a-360", "LDL-ukb-d-30780_irnt", "aFib-ebi-a-GCST006414",
  "WBC-ieu-b-30", "IBD-ebi-a-GCST004131", "BMI-panukb",
  "PLT-panukb", "RBC-panukb", "HB-panukb", "HTN-panukb", "Height-panukb",
  "T2D-panukb", "MDD-ieu-b-102", "SCZ-ieu-b-5102", "T1D-GCST90014023", "BIP-ieu-b-5110", 
  "ASD-ieu-a-1185", "PD-ieu-b-7", "NS-ukb-a-230","ATH_gtexukb"
)
for(i in traits){
  combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,i,"/",i,".3qtls.combined_pip_bytype_final.RDS"))
  combined_pip_by_group_multi <- combined_pip_by_group_multi[combined_pip_by_group_multi$combined_pip>0.8,]
  write.table(combined_pip_by_group_multi$gene_name,file = paste0("/project/xinhe/shengqian/cTWAS_paper_figures/data/ctwas_genes_PIP0.8/",i,".txt"),col.names = F,row.names = F,quote = F,sep = "\t")
}
```

### Figure 4b: M-cTWAS identified genes with higher POPS scores
Averaged POPS scores across all traits stratified by M-cTWAS PIPs 
```{r echo=FALSE, message=FALSE, warning=FALSE}
traits <- c(
  "SBP-ukb-a-360", "LDL-ukb-d-30780_irnt", "aFib-ebi-a-GCST006414",
  "WBC-ieu-b-30", "IBD-ebi-a-GCST004131", "BMI-panukb",
  "PLT-panukb", "RBC-panukb", "HB-panukb", "HTN-panukb", "Height-panukb",
  "T2D-panukb", "MDD-ieu-b-102", "SCZ-ieu-b-5102", "T1D-GCST90014023", "BIP-ieu-b-5110", 
  "ASD-ieu-a-1185", "PD-ieu-b-7", "NS-ukb-a-230","ATH_gtexukb"
)

plot_df <- data.frame()
for(i in traits){
  pops <- fread(paste0("/project/xinhe/shengqian/pops/pops_full_features_output/",i,"_pops/",i,".preds"))
  pop_anno <- fread("/project/xinhe/shengqian/pops/example/data/utils/gene_annot_jun10.txt")
  pops_pred <- merge(pops, pop_anno, by = "ENSGID", all.x = TRUE)
  pops_pred <- pops_pred[,c("PoPS_Score","NAME")]
  ctwas_res <- readRDS(paste0("/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/",i,"/",i,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_res <- ctwas_res[,c("gene_name","combined_pip")]
  merged_df <- merge(ctwas_res, pops_pred, by.x = "gene_name", by.y = "NAME", all.x = TRUE)
  plot_df <- rbind(plot_df,merged_df)
}
plot_df <- na.omit(plot_df)
plot_pops_by_pip_bin(plot_df,"POPs score across all traits")
```

### Figure 4c: Validation with Silver Standand Genes
```{r echo=FALSE, message=FALSE, warning=FALSE}
trait <- "LDL-ukb-d-30780_irnt"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
traits_silver <- c("T2D","LDL","BMI","RBC","IBD","SCZ","aFib")
names(traits_silver) <- c("T2D-panukb","LDL-ukb-d-30780_irnt","BMI-panukb","RBC-panukb","IBD-ebi-a-GCST004131","SCZ-ieu-b-5102","aFib-ebi-a-GCST006414")
known <- readRDS(paste0("/project/xinhe/xsun/multi_group_ctwas/data/silverstandard/known_annotations_",traits_silver[trait],".RDS"))
bystander <- readRDS(paste0("/project/xinhe/xsun/multi_group_ctwas/data/silverstandard/bystanders_",traits_silver[trait],".RDS"))

ctwas_genes <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
ctwas_genes <- ctwas_genes[,c("gene_name","combined_pip")]
ctwas_genes <- ctwas_genes$gene_name[ctwas_genes$combined_pip > 0.8]

folder_results <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/results_downstream/enrichr_compare/"
coloc_genes <- readRDS(paste0(folder_results,trait,".genes.coloc.RDS"))
coloc_genes <- coloc_genes$gene_name[coloc_genes$PP4 > 0.8]
coloc_genes <- unique(coloc_genes)
twas_genes <- readRDS(paste0(folder_results,trait,".genes.twas_bonf.RDS"))
twas_genes <- unique(twas_genes)
df <- data.frame(
  Method = c("All genes", "M-cTWAS", "Coloc", "TWAS"),
  Proportion = c(length(known)/(length(known)+length(bystander)),sum(ctwas_genes%in%known) / (sum(ctwas_genes%in%known) + sum(ctwas_genes%in%bystander)), sum(coloc_genes%in%known) / (sum(coloc_genes%in%known) + sum(coloc_genes%in%bystander)), sum(twas_genes%in%known) / (sum(twas_genes%in%known) + sum(twas_genes%in%bystander)))
)

# Ensure order
df$Method <- factor(df$Method, levels = c("All genes", "M-cTWAS", "Coloc", "TWAS"))

# Plot
ggplot(df, aes(x = Method, y = Proportion, fill = Method)) +
  geom_col(width = 0.6, alpha = 0.8) +
  geom_text(aes(label = round(Proportion, 2)), 
            vjust = -0.5, size = 5) +
  scale_y_continuous(limits = c(0, 1)) +
  labs(
    title = "",
    y = "Detected genes in silver standard (%)",
    x = ""
  ) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "none")
```

### Figure 4d: Comparison of number of signals per region between M-cTWAS, Coloc and TWAS
cTWAS tends to report single genes per locus, while coloc or TWAS report many. The number of signals of Coloc and TWAS are calcuated in regions with TWAS signals (after bonferroni correction). M-cTWAS signals are culated in regions selected by screen region step. 
```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_munro     <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
mapping_table <- rbind(mapping_predictdb, mapping_munro)

plots <- plot_number_of_signals_per_region_distributions(
  coloc_file = "/project/xinhe/shengqian/coloc_GWAS_analysis/results/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.coloc_res.RDS",
  twas_file  = "/project/xinhe/shengqian/coloc_GWAS_analysis/results/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.twas_res.RDS",
  ctwas_id_file = "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.3qtls.thin1.shared_all.L5.regionmerge_with_mappingtable_finemap_res.RDS",
  ctwas_gene_alpha_file = "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.3qtls.thin1.shared_all.L5.regionmerge_with_mappingtable_susie_alpha_res.RDS",
  mapping_table = mapping_table,
  x_max = 10
)
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(plots)
```

### Figure 4e: Compare with TGFM: compare with genes found in the paper. Show that unique genes by cTWAS are valid
For unique genes identified by M-cTWAS and TGFM, I plotted the distribution of POPS scores and showed that M-cTWAS unique genes have higher POPS score than TGFM unique genes.
```{r echo=FALSE, message=FALSE, warning=FALSE}
library(grex)
library(ggplot2)
trait_list <- list("LDL-ukb-d-30780_irnt" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_biochemistry_LDLdirect.txt",
                   "HTN-panukb" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_disease_HYPERTENSION_DIAGNOSED.txt")
for(i in names(trait_list)){
  ctwas_folder_results <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
  #ctwas_folder_results <- "/project/xinhe/xsun/multi_group_ctwas/26.multi_tissue_eQTL_0805/snakemake_outputs/"
  ctwas_genes <-  readRDS(paste0(ctwas_folder_results,i,"/",i,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
  #ctwas_genes <-  readRDS(paste0(ctwas_folder_results,i,"/",i,".eqtlonly.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_genes_highpip <- ctwas_genes$gene_name[ctwas_genes$combined_pip > 0.8]
  tgfm_genes <- data.table::fread(trait_list[[i]])
  tgfm_genes_highpip <- tgfm_genes[tgfm_genes$TGFM_PIP > 0.8,]
  tgfm_genes_highpip <- tgfm_genes_highpip[tgfm_genes_highpip$genetic_element_class != "variant",]
  tgfm_genes_highpip$ensg_id <- sub("\\..*", "", tgfm_genes_highpip$genetic_element_name)
  tgfm_genes_highpip_ensg <- unique(tgfm_genes_highpip$ensg_id)
  library(EnsDb.Hsapiens.v86)
  ensg_to_symbol <- AnnotationDbi::select(
    EnsDb.Hsapiens.v86,
    keys = tgfm_genes_highpip_ensg,
    keytype = "GENEID",
    columns = c("GENEID", "SYMBOL")
  )

  tgfm_genes_highpip <- ensg_to_symbol$SYMBOL
  ctwas_unique_genes <- setdiff(ctwas_genes_highpip,tgfm_genes_highpip)
  tgfm_unique_genes <- setdiff(tgfm_genes_highpip,ctwas_genes_highpip)
  p1 <- plot_overlap_barplot_cTWAS_vs_TGFM(ctwas_genes_highpip = ctwas_genes_highpip, tgfm_genes_highpip_symbol = tgfm_genes_highpip,trait = i,return_unique_genes = F)
  
  pops <- fread(paste0("/project/xinhe/shengqian/pops/pops_full_features_output/",i,"_pops/",i,".preds"))
  df <- grex(cleanid(pops$ENSGID))
  pops <- cbind(pops,df)
  tgfm_gene <- pops[pops$hgnc_symbol %in% tgfm_unique_genes,]
  ctwas_gene <- pops[pops$hgnc_symbol %in% ctwas_unique_genes,]

  # Combine into one data frame
  df <- data.frame(
    score = c(tgfm_gene$PoPS_Score, ctwas_gene$PoPS_Score),
    group = c(rep("TGFM", length(tgfm_gene$PoPS_Score)),
            rep("cTWAS", length(ctwas_gene$PoPS_Score)))
  )

  # Boxplot
  p2 <- ggplot(df, aes(x = group, y = score, fill = group)) +
    geom_boxplot(alpha = 0.5) +
    scale_fill_manual(values = c("blue", "red")) +
    theme_minimal(base_size = 14) +
    labs(title = i, x = "", y = "PoPS Score",fill = "")
  
  combined_plot <- (p1 | p2) + plot_layout(guides = "collect") & theme(legend.position = "bottom")
  print(combined_plot)
}
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(grex)
library(ggplot2)
trait_list <- list("LDL-ukb-d-30780_irnt" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_biochemistry_LDLdirect.txt",
                   "HTN-panukb" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_disease_HYPERTENSION_DIAGNOSED.txt")
for(i in names(trait_list)){
  #ctwas_folder_results <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
  ctwas_folder_results <- "/project/xinhe/xsun/multi_group_ctwas/26.multi_tissue_eQTL_0805/snakemake_outputs/"
  #ctwas_genes <-  readRDS(paste0(ctwas_folder_results,i,"/",i,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_genes <-  readRDS(paste0(ctwas_folder_results,i,"/",i,".eqtlonly.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_genes_highpip <- ctwas_genes$gene_name[ctwas_genes$combined_pip > 0.8]
  tgfm_genes <- data.table::fread(trait_list[[i]])
  tgfm_genes_highpip <- tgfm_genes[tgfm_genes$TGFM_PIP > 0.8,]
  tgfm_genes_highpip <- tgfm_genes_highpip[tgfm_genes_highpip$genetic_element_class != "variant",]
  tgfm_genes_highpip$ensg_id <- sub("\\..*", "", tgfm_genes_highpip$genetic_element_name)
  tgfm_genes_highpip_ensg <- unique(tgfm_genes_highpip$ensg_id)
  library(EnsDb.Hsapiens.v86)
  ensg_to_symbol <- AnnotationDbi::select(
    EnsDb.Hsapiens.v86,
    keys = tgfm_genes_highpip_ensg,
    keytype = "GENEID",
    columns = c("GENEID", "SYMBOL")
  )

  tgfm_genes_highpip <- ensg_to_symbol$SYMBOL
  ctwas_unique_genes <- setdiff(ctwas_genes_highpip,tgfm_genes_highpip)
  tgfm_unique_genes <- setdiff(tgfm_genes_highpip,ctwas_genes_highpip)
  p1 <- plot_overlap_barplot_cTWAS_vs_TGFM(ctwas_genes_highpip = ctwas_genes_highpip, tgfm_genes_highpip_symbol = tgfm_genes_highpip,trait = i,return_unique_genes = F)
  
  pops <- fread(paste0("/project/xinhe/shengqian/pops/pops_full_features_output/",i,"_pops/",i,".preds"))
  df <- grex(cleanid(pops$ENSGID))
  pops <- cbind(pops,df)
  tgfm_gene <- pops[pops$hgnc_symbol %in% tgfm_unique_genes,]
  ctwas_gene <- pops[pops$hgnc_symbol %in% ctwas_unique_genes,]

  # Combine into one data frame
  df <- data.frame(
    score = c(tgfm_gene$PoPS_Score, ctwas_gene$PoPS_Score),
    group = c(rep("TGFM", length(tgfm_gene$PoPS_Score)),
            rep("cTWAS", length(ctwas_gene$PoPS_Score)))
  )

  # Boxplot
  p2 <- ggplot(df, aes(x = group, y = score, fill = group)) +
    geom_boxplot(alpha = 0.5) +
    scale_fill_manual(values = c("blue", "red")) +
    theme_minimal(base_size = 14) +
    labs(title = i, x = "", y = "PoPS Score",fill = "")
  
  combined_plot <- (p1 | p2) + plot_layout(guides = "collect") & theme(legend.position = "bottom")
  print(combined_plot)
}
```

## Figure 5: cTWAS discovers candidate genes for complex traits and provides insights on their molecular mechanisms 
```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=8, fig.width=10}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
PIP_cutoff <- 0.9
combined_pip_by_group_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bytype_final.RDS"))
combined_pip_by_context_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bycontext_final.RDS"))
#combined_pip_by_group_multi_sig <- combined_pip_by_group_multi[combined_pip_by_group_multi$combined_pip > PIP_cutoff,]
#combined_pip_by_group_single_sig <- combined_pip_by_group_single[combined_pip_by_group_single$combined_pip > PIP_cutoff,]
#combined_pip_by_context_multi_sig <- combined_pip_by_context_multi[combined_pip_by_context_multi$combined_pip > PIP_cutoff,]
  
combined_pip_by_group_multi_sig <- combined_pip_by_group_multi[1:30,]
combined_pip_by_context_multi_sig <- combined_pip_by_context_multi[1:30,]
#combined_pip_by_group_multi_unique <- combined_pip_by_group_multi_sig[!combined_pip_by_group_multi_sig$gene_name %in% combined_pip_by_group_single_sig$gene_name, ]
#combined_pip_by_context_multi_unique <- combined_pip_by_context_multi_sig[!combined_pip_by_context_multi_sig$gene_name %in% combined_pip_by_group_single_sig$gene_name, ]

p1 <- plot_heatmap(heatmap_data = combined_pip_by_group_multi_sig, main = paste0("By modality"),showPIP = T,show_heatmap_legend=TRUE)
  
p2 <- plot_heatmap(heatmap_data = combined_pip_by_context_multi_sig, main = paste0("By context"),showPIP = T,show_heatmap_legend=FALSE)
```

### Figure 5a: cTWAS helps identify causal modality and context
```{r, eval=F, echo=FALSE, message=FALSE, warning=FALSE,fig.height=15, fig.width=12}
grid.newpage()
print(p1)
```

```{r, eval=F, echo=FALSE, message=FALSE, warning=FALSE,fig.height=15, fig.width=12}
grid.newpage()
print(p2)
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.height=25, fig.width=30}
draw(p1 + p2,heatmap_legend_side = "right",
  annotation_legend_side = "right")
```


### Figure 5b: Gene set enrichment analysis
<figure class="half">
    <img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/data/figures/LDL_pathway_enrichment.png" width="150%">
</figure>

### Figure 5c: Network analysis of candidate genes suggests novel pathways
<figure class="half">
    <img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/data/figures/LDL_gene_clusters.png" width="150%">
</figure>

### Figure 5d: A substantial fraction of cTWAS genes are novel
```{r,  echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_munro     <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
mapping_table <- rbind(mapping_predictdb, mapping_munro)
snp_map <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/LD_region_info/snp_map.RDS")
aa <- make_ctwas_category_barplot(trait = "LDL-ukb-d-30780_irnt",
                            mapping_table = mapping_table,
                            snp_map = snp_map)
```

### Figure 5e: locus examples
<figure class="half">
    <img src="https://github.com/sq-96/multigroup_ctwas_analysis/raw/master/data/figures/LDL-ukb-d-30780_irnt/multi/LDLR.png" width="150%">
</figure>

## Figure 6: Brain epiQTLs explain a large fraction of missing heritability by eQTLs

### Figure 6a: EpiQTLs explain a larger fraction of h2g than eQTLs
```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
prop_heritability <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.parameters.RDS")

# Extract values
log_enrichment <- prop_heritability$log_enrichment
log_enrichment_se <- prop_heritability$log_enrichment_se

# Build data frame
df <- data.frame(
  context = names(log_enrichment),
  value = as.numeric(log_enrichment),
  se = as.numeric(log_enrichment_se)
)

# Plot
ggplot(df, aes(x = context, y = value)) +
  geom_col(fill = "steelblue", alpha = 0.8) +
  geom_errorbar(aes(ymin = value - se, ymax = value + se), width = 0.2) +
  theme_minimal(base_size = 14) +
  labs(x = "", y = "Log Enrichment") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r,echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
prop_heritability <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.parameters.RDS")
prop_heritability <- prop_heritability$prop_heritability
prop_heritability <- c(sum(prop_heritability[1:5]),sum(prop_heritability[6:7]),prop_heritability[8],prop_heritability[9])
colors <- brewer.pal(n = 4, name = "Set2")  # Alternatives: "Pastel1", "Dark2", "Set3"

# Draw pie chart
pie(
  prop_heritability,
  labels = paste0(c("eQTL","caQTL","mQTL","SNP"), "\n", round(100 * prop_heritability, 1), "%"),
  col = colors,
  border = "white"
)
```

### Figure 6b: Using epiQTL discovered novel CREs not explained by eQTLs
```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
library(RColorBrewer)
# Example: Load three different heritability proportion datasets
prop1 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL/SCZ_eQTL.parameters.RDS")$prop_heritability
prop1 <- c(sum(prop1[1:5]),prop1[6])
prop2 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL/SCZ_caQTL.parameters.RDS")$prop_heritability
prop2 <- c(prop2[1:2],prop2[3])
prop3 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_mQTL/SCZ_mQTL.parameters.RDS")$prop_heritability

# Set up layout for 3 plots in one row
par(mfrow = c(1, 3), mar = c(2, 2, 2, 1))  # Adjust margins if needed

# Choose consistent color palette
colors <- brewer.pal(n = max(lengths(list(prop1, prop2, prop3))), name = "Set2")

# Plot each pie chart
pie(
  prop1,
  labels = paste0(c("eQTL","SNP"), "\n", round(100 * prop1, 1), "%"),
  col = colors,
  border = "white"
)

pie(
  prop2,
  labels = paste0(c("caQTL","SNP"), "\n", round(100 * prop2, 1), "%"),
  col = colors,
  border = "white"
)

pie(
  prop3,
  labels = paste0(c("mQTL","SNP"), "\n", round(100 * prop3, 1), "%"),
  col = colors,
  border = "white"
)

```

```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_table <- readRDS("/project/xinhe/shengqian/cTWAS_epigenetic/data/weights/gene_mapping.RDS")
predictdb_mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_table <- rbind(mapping_table,predictdb_mapping_table)
snp_map <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/LD_region_info/snp_map.RDS")
suppressMessages({
  suppressWarnings({
    full_cre_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
    full_gene_res <- get_gene_res(mapping_table, "/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS", snp_map)
  })
})

# Values from your results
df <- data.frame(
  category = c("Gene", "CRE"),
  count = c(dim(full_gene_res$result_sig)[1], dim(full_cre_res$result_sig)[1])
)

# Create bar plot with ggplot2
ggplot(df, aes(x = category, y = count, fill = category)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "Count (PIP>0.8)") +
  theme_minimal() +
  theme(legend.position = "none")
```

### Figure 6c: Using epiQTL discovered novel CREs missed by eQTLs
```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_table <- readRDS("/project/xinhe/shengqian/cTWAS_epigenetic/data/weights/gene_mapping.RDS")
predictdb_mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_table <- rbind(mapping_table,predictdb_mapping_table)
snp_map <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/LD_region_info/snp_map.RDS")
epi_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL_mQTL/SCZ_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
full_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
suppressMessages({
  suppressWarnings({
    epi_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL_mQTL/SCZ_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
    full_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
  })
})
epi_result <- epi_res$result_sig
epi_result <- epi_result[,c("region_id","start_pos","end_pos","total_susie_pip")]
full_result <- full_res$result
full_result <- full_result[,c("region_id","start_pos","end_pos","total_susie_pip")]
setDT(epi_result)
setDT(full_result)
setnames(full_result, "total_susie_pip", "total_susie_pip_full")
merged_result <- full_result[epi_result, 
                             on = .(region_id, start_pos, end_pos)]
merged_result[is.na(total_susie_pip_full), total_susie_pip_full := 0]
p1 <- ggplot(merged_result, aes(x = total_susie_pip, 
                          y = total_susie_pip_full)) +
    geom_point(alpha = 0.6) +
    geom_abline(intercept = 0, slope = 1, linetype = "dashed", color = "red") +
    labs(x = "CRE PIP in EpiQTL results",
       y = "CRE PIP in full results",
       title = "") +
    coord_cartesian(xlim = c(0, 1), ylim = c(0, 1)) +
    theme_minimal(base_size = 14)
```

```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
suppressMessages({
  suppressWarnings({
    eQTL_res <- get_gene_res(mapping_table, "/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL/SCZ_eQTL.finemap_regions_res.RDS", snp_map)
    full_res <- get_gene_res(mapping_table, "/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS", snp_map)
  })
})
eQTL_result <- eQTL_res$result_sig
eQTL_result <- eQTL_result[,c("gene_name","combined_pip")]
full_result <- full_res$result
full_result <- full_result[,c("gene_name","combined_pip")]
setDT(eQTL_result)
setDT(full_result)
setnames(full_result, "combined_pip", "combined_pip_full")
merged_result <- full_result[eQTL_result, 
                             on = .(gene_name)]
merged_result[is.na(combined_pip_full), combined_pip_full := 0]
p2 <- ggplot(merged_result, aes(x = combined_pip, 
                          y = combined_pip_full)) +
  geom_point(alpha = 0.6) +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed", color = "red") +
  labs(x = "Gene PIP in eQTL results",
       y = "Gene PIP in full results",
       title = "") +
  coord_cartesian(xlim = c(0, 1), ylim = c(0, 1)) +
  theme_minimal(base_size = 14)
```

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=10}
combined_plot <- (p1 | p2) + plot_layout(guides = "collect") & theme(legend.position = "bottom")
print(combined_plot)
```

```{r echo=FALSE}
# ---- settings ----
workdir <- "/project/xinhe/shengqian/cTWAS_epigenetic/data/all_cre_cpgs_peaks/"
bed_files <- list.files(workdir, pattern = "^all_cre_cpgs\\.[^.]+\\.PeakCalls\\.bed$", full.names = TRUE)
stopifnot(length(bed_files) > 0)

# Extract clean cell type names
cell_types <- sub("^all_cre_cpgs\\.(.*?)\\.PeakCalls\\.bed$", "\\1",
                  basename(bed_files))

# ---- read CpGs and build long table ----
read_cpgs <- function(fp) {
  dt <- fread(fp, header = FALSE, sep = "\t", showProgress = FALSE)
  unique(dt[[ncol(dt)]])
}

cpg_list <- lapply(bed_files, read_cpgs)

records <- rbindlist(
  lapply(seq_along(cpg_list), function(i) {
    data.table(cpg = cpg_list[[i]], cell = cell_types[i], present = 1L)
  })
)

records <- unique(records, by = c("cpg", "cell"))

# ---- cast to wide matrix ----
mat_dt <- dcast(records, cpg ~ cell, value.var = "present", fill = 0L)
presence_count <- rowSums(as.matrix(mat_dt[, -1, with = FALSE]))
mat_dt <- mat_dt[order(presence_count, decreasing = TRUE)]

cpg_ids <- mat_dt$cpg
mat <- as.matrix(mat_dt[, -1, with = FALSE])
rownames(mat) <- cpg_ids

# ---- plot heatmap ----
cols <- c("white", "#FF9999")
brks <- c(-0.5, 0.5, 1.5)

pheatmap(
  mat,
  color = cols,
  breaks = brks,
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = FALSE,
  show_colnames = TRUE,
  legend = FALSE,
  border_color = NA
)

# ---- proportions: CpG present in exactly k cell types (k = 1..7) ----
k_per_cpg <- rowSums(mat)                # how many cell types have a peak for each CpG
n_cpg     <- length(k_per_cpg)
n_cells   <- ncol(mat)                    # should be 7

stat_dt <- data.table(
  peaks = 1:n_cells,
  n = as.integer(table(factor(k_per_cpg, levels = 1:n_cells)))
)
stat_dt[, proportion := n / n_cpg]

cat("\nProportion of CpGs present in exactly k cell-type peaks:\n")
print(stat_dt)
# ---- optional: quick bar plot to screen (not saved) ----
ggplot(stat_dt, aes(x = factor(peaks), y = proportion)) +
  geom_col(fill = "#FF9999") +
  geom_text(aes(label = sprintf("%.2f", proportion)), vjust = -0.3, size = 3) +
  labs(x = "k (number of cell types with a peak)", y = "Proportion of CpGs",
       title = "CpGs present in exactly k cell-type peaks") +
  ylim(0, max(stat_dt$proportion) * 1.15) +
  theme_classic()
```

```{r echo=FALSE}
driver <- dbDriver('SQLite')
db <- dbConnect(drv = driver, "/project/xinhe/shengqian/cTWAS_epigenetic/data/weights/brain_mQTL.db")
query <- function(...) RSQLite::dbGetQuery(db, ...)
weights <- query("select * from weights")
RSQLite::dbDisconnect(db)
MetaBrain_eQTL <- readRDS("/project/xinhe/shengqian/cTWAS_epigenetic/data/MetaBrain_eQTL/MetaBrain_eQTL_qvalue005.RDS")
cre_cpgs <- fread("/project/xinhe/shengqian/cTWAS_epigenetic/data/all_cre_cpgs.bed")
weights <- weights[weights$gene %in% cre_cpgs$V4,]
length(intersect(weights$rsid,MetaBrain_eQTL$rsid))
```

```{r echo=FALSE}
#library(annotatr)
#dm_regions = read_regions(con = "/project/xinhe/shengqian/cTWAS_epigenetic/data/all_cre_cpgs.bed", genome = 'hg38',format = 'bed')
#annots = c('hg38_basicgenes', 'hg38_genes_intergenic')
#annotations = build_annotations(genome = 'hg38', annotations = annots)
#dm_annotated = annotate_regions(regions = dm_regions,annotations = annotations,ignore.strand = TRUE,quiet = FALSE)
#df_dm_annotated = data.frame(dm_annotated)
#saveRDS(df_dm_annotated,file = "/project/xinhe/shengqian/cTWAS_epigenetic/data/all_cre_cpgs_annotation.RDS")
cpg_annot <- readRDS("/project/xinhe/shengqian/cTWAS_epigenetic/data/all_cre_cpgs_annotation.RDS")
cpg_annot$annotation <- sapply(cpg_annot$annot.id, function(x){unlist(strsplit(x, split="[:]"))[1]})
cpg_annot <- cpg_annot[,c("name","annotation","annot.symbol")]
cpg_annot <- cpg_annot[cpg_annot$annotation=="intergenic",]
cpg_annot <- unique(cpg_annot[,c("name","annotation")])
table(cpg_annot$annotation)
```

