---
title: "cTWAS paper figures"
output: html_document
date: '2025-8-18'
editor_options: 
  chunk_output_type: console
---


```{r echo=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(ctwas)
library(data.table)
library(ggplot2)
library(ggrepel)
library(egg)
library(grid)
library(dplyr)
library(pheatmap)
source("/project/xinhe/shengqian/cTWAS_paper_figures/plot_function.R")
source("/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R")
```

## Figure 2: cTWAS simulations (Correlated Brain Tissues)
```{r echo=FALSE}
pve_gene <- c(0.02, 0.02, 0, 0, 0,
              0.01, 0.01, 0, 0, 0,
              0.003, 0.003, 0, 0, 0)

group_size <- c(8798,8772,8478,8596,8542, 
                22802,19188,17467,17104,22142, 
                3561,2935,2578,2659,3142)
pve_snp <- 0.3
sigma_theta <- 0.02
sigma_beta <- 0.02
gene_prior <- pve_gene/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/group_size
snp_prior <- pve_snp/sigma_beta^2/(1-sum(pve_gene)-pve_snp)/5000000

thins <- c(1)

prior_true <- gene_prior
  
enrichment_true <- log(gene_prior/snp_prior,base = exp(1))

pve_true <- pve_gene

phe_true <- pve_gene/(sum(pve_gene)+pve_snp)
```

### Figure 2a: Enrichment
```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag = "five_correlated_brain_tissues_three_omics"
plot_enrichment_correlated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",enrichment_true)
```

### Figure 2b: PHE
```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag = "five_correlated_brain_tissues_three_omics"
plot_PHE_correlated(results_dir,runtag,paste(1, 1:10, sep = "-"),c(1),"shared_all","ctwas",".minp08.mingene0.parameters.RDS",phe_true)
```

### Figure 2c: Inflated single group PHE
```{r,echo=FALSE, message=FALSE, warning=FALSE, fig.width= 12, fig.height= 4}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/"
runtag_list <- c(paste0(c("Cerebellum","Cortex","Nucleus_accumbens","Caudate","Cerebellar_Hemisphere"),"_expression"),
                  paste0(c("Cerebellum","Cortex","Nucleus_accumbens","Caudate","Cerebellar_Hemisphere"),"_splicing"),
                  paste0(c("Cerebellum","Cortex","Nucleus_accumbens","Caudate","Cerebellar_Hemisphere"),"_stability"))
plot_single_PHE_correlated(results_dir,runtag_list,c(1),phe_true)
```

### Figure 2d: PIP calibration
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
df1 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/molecular_PIP_calibration_df.RDS")
df2 <- readRDS("/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/gene_PIP_calibration_df.RDS")
fig1 <- cp_plot(df1$susie_pip, df1$ifcausal, df1$runtag, mode ="PIP", main = "MT level")
fig2 <- cp_plot(df2$susie_pip, df2$ifcausal, df2$runtag, mode ="PIP", main = "Gene level")
grid.arrange(fig1, fig2, ncol = 2)
```

### Figure 2e: Power and False discovery rate
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.width=8, fig.height=4}
multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/multi_ctwas_simu_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/single_ctwas_simu_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/coloc_simu_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/smr_simu_res.RDS"
p1 <- plot_molecular_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)

multi_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/multi_ctwas_simu_gene_res.RDS"
single_ctwas_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/single_ctwas_simu_gene_res.RDS"
coloc_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/coloc_simu_gene_res.RDS"
smr_res <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_five_correlated_brain_tissues_three_omics/smr_simu_gene_res.RDS"
p2 <- plot_gene_power_compare_method(multi_ctwas_res,single_ctwas_res,coloc_res,smr_res)
grid.arrange(p1, p2, ncol = 2)
```

## Figure 3: cTWAS estimates genetic architecture of complex traits from GTEx

### Figure 3a: Tissues are often correlated: but one tissue is not enough.
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
thin <- 1
vgs <- "shared_all"
L <-5
colors <- c("#ff7f0e", "#2ca02c", "#d62728",  "#9467bd", "#8c564b", "#e377c2", "#7f7f7f",  "#bcbd22",  "#17becf",  "#f7b6d2",  "#c5b0d5",  "#9edae5", "#ffbb78",  "#98df8a",  "#ff9896" )

fix_panel_size <- function(plot, width = 2.1, height = 2) {
  set_panel_size(plot, 
                 width = unit(width, "in"), 
                 height = unit(height, "in"))
}

folder_results_multiqtl <- "/project/xinhe/xsun/multi_group_ctwas/24.diff_qtls_0519/snakemake_outputs/"

trait <- "IBD-ebi-a-GCST004131"

gwas_n <- samplesize[trait]


file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Whole_Blood.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p1 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)


file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Colon_Transverse.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p2 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)

file_param <- paste0("/project/xinhe/xsun/multi_group_ctwas/24.diff_qtls_0519/snakemake_outputs/IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131.eonly.thin1.shared_all.param.RDS")
param <- readRDS(file_param)
ctwas_parameters <- summarize_param(param, gwas_n)

p3 <- plot_piechart_topn(ctwas_parameters = ctwas_parameters,colors = colors,by = "context",title = NULL)

pie1 <- fix_panel_size(p1)
pie2 <- fix_panel_size(p2)
pie3 <- fix_panel_size(p3)

# Calculate widths of each gtable (plot + legend)
widths <- unit.c(grobWidth(pie1), grobWidth(pie2),grobWidth(pie3))

# Arrange plots with their natural widths
p <- grid.arrange(pie1, pie2, pie3,
                  ncol = 3, 
                  widths = widths,
                  top = paste0(trait))
```

### Figure 3b: Shared pattern
```{r, eval=FALSE, echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=6, fig.width=10}
trait <- "LDL-ukb-d-30780_irnt"
out <- plot_pairwise_PVE(trait, samplesize, n_top = 10)
```

### Figure 3c: Partition of h2g (Multiple tissues contribute and Contribution of each modality)
```{r echo=FALSE, message=FALSE, warning=FALSE}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
p <- generate_piecharts_for_trait(ctwas_parameters_single, ctwas_parameters_multi, title_top = title)
grid::grid.draw(p)
invisible(NULL)
```

### Figure 3d: Percent of TWAS regions with Coloc signals
```{r echo=FALSE, message=FALSE, warning=FALSE}
plot_fraction_of_coloc_signals_in_TWAS_regions("LDL-ukb-d-30780_irnt", pretty_labels = TRUE)
```


### Figure 3e: MECS estimated eQTL heritability contributions
```{r echo=FALSE, message=FALSE, warning=FALSE}
plot_MESC_eQTL_heritability("LDL-ukb-d-30780_irnt")$plot
```

## Figure 4: Multi-cTWAS improves the discovery of candidate genes

### Figure 4a: Incorporating multiple modality and tissues improves discovery power
```{r echo=FALSE, message=FALSE, warning=FALSE}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
folder_results_multi_E <- "/project/xinhe/xsun/multi_group_ctwas/26.multi_tissue_eQTL_0805/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
PIP_cutoff <- 0.8
combined_pip_by_group_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi_E <- readRDS(paste0(folder_results_multi_E,trait,"/",trait,".eqtlonly.combined_pip_rmmapping_bygroup_final.RDS"))
#print(plot_overlap_barplot_multi_vs_single(combined_pip_by_group_multi = combined_pip_by_group_multi, combined_pip_by_group_single = combined_pip_by_group_single, PIP_cutoff = PIP_cutoff,tissue = top_tissue,trait = trait,return_unique_genes = T)$plot)
print(plot_overlap_barplot_multi_vs_multi_E_vs_single(combined_pip_by_group_multi = combined_pip_by_group_multi, combined_pip_by_group_single = combined_pip_by_group_single, combined_pip_by_group_multi_E = combined_pip_by_group_multi_E, PIP_cutoff = PIP_cutoff,tissue = top_tissue,trait = trait,return_unique_genes = T)$plot)
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
traits <- c(
  "SBP-ukb-a-360", "LDL-ukb-d-30780_irnt", "aFib-ebi-a-GCST006414",
  "WBC-ieu-b-30", "IBD-ebi-a-GCST004131", "BMI-panukb",
  "PLT-panukb", "RBC-panukb", "HB-panukb", "HTN-panukb", "Height-panukb",
  "T2D-panukb", "MDD-ieu-b-102", "SCZ-ieu-b-5102", "T1D-GCST90014023", "BIP-ieu-b-5110", 
  "ASD-ieu-a-1185", "PD-ieu-b-7", "NS-ukb-a-230","ATH_gtexukb"
)
for(i in traits){
  combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,i,"/",i,".3qtls.combined_pip_bytype_final.RDS"))
  combined_pip_by_group_multi <- combined_pip_by_group_multi[combined_pip_by_group_multi$combined_pip>0.8,]
  write.table(combined_pip_by_group_multi$gene_name,file = paste0("/project/xinhe/shengqian/cTWAS_paper_figures/data/ctwas_genes_PIP0.8/",i,".txt"),col.names = F,row.names = F,quote = F,sep = "\t")
}
```


### Figure 4b: M-cTWAS identified genes with higher POPS scores
Averaged POPS scores across all traits stratified by M-cTWAS PIPs 
```{r echo=FALSE, message=FALSE, warning=FALSE}
traits <- c(
  "SBP-ukb-a-360", "LDL-ukb-d-30780_irnt", "aFib-ebi-a-GCST006414",
  "WBC-ieu-b-30", "IBD-ebi-a-GCST004131", "BMI-panukb",
  "PLT-panukb", "RBC-panukb", "HB-panukb", "HTN-panukb", "Height-panukb",
  "T2D-panukb", "MDD-ieu-b-102", "SCZ-ieu-b-5102", "T1D-GCST90014023", "BIP-ieu-b-5110", 
  "ASD-ieu-a-1185", "PD-ieu-b-7", "NS-ukb-a-230","ATH_gtexukb"
)

plot_df <- data.frame()
for(i in traits){
  pops <- fread(paste0("/project/xinhe/shengqian/pops/pops_full_features_output/",i,"_pops/",i,".preds"))
  pop_anno <- fread("/project/xinhe/shengqian/pops/example/data/utils/gene_annot_jun10.txt")
  pops_pred <- merge(pops, pop_anno, by = "ENSGID", all.x = TRUE)
  pops_pred <- pops_pred[,c("PoPS_Score","NAME")]
  ctwas_res <- readRDS(paste0("/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/",i,"/",i,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_res <- ctwas_res[,c("gene_name","combined_pip")]
  merged_df <- merge(ctwas_res, pops_pred, by.x = "gene_name", by.y = "NAME", all.x = TRUE)
  plot_df <- rbind(plot_df,merged_df)
}
plot_df <- na.omit(plot_df)
plot_pops_by_pip_bin(plot_df,"POPs score across all traits")
```


### Figure 4c: Comparison of number of signals per region between M-cTWAS, Coloc and TWAS
cTWAS tends to report single genes per locus, while coloc or TWAS report many. The number of signals of Coloc and TWAS are calcuated in regions with TWAS signals (after bonferroni correction). M-cTWAS signals are culated in regions selected by screen region step. 
```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_munro     <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
mapping_table <- rbind(mapping_predictdb, mapping_munro)

plots <- plot_number_of_signals_per_region_distributions(
  coloc_file = "/project/xinhe/shengqian/coloc_GWAS_analysis/results/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.coloc_res.RDS",
  twas_file  = "/project/xinhe/shengqian/coloc_GWAS_analysis/results/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.twas_res.RDS",
  ctwas_id_file = "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.3qtls.thin1.shared_all.L5.regionmerge_with_mappingtable_finemap_res.RDS",
  ctwas_gene_alpha_file = "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.3qtls.thin1.shared_all.L5.regionmerge_with_mappingtable_susie_alpha_res.RDS",
  mapping_table = mapping_table,
  x_max = 50
)

```

```{r echo=FALSE, message=FALSE, warning=FALSE}
print(plots)
```

### Figure 4d: Compare with TGFM: compare with genes found in the paper. Show that unique genes by cTWAS are valid
For unique genes identified by M-cTWAS and TGFM, I plotted the distribution of POPS scores and showed that M-cTWAS unique genes have higher POPS score than TGFM unique genes.
```{r echo=FALSE, message=FALSE, warning=FALSE}
library(grex)
library(ggplot2)
trait_list <- list("LDL-ukb-d-30780_irnt" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_biochemistry_LDLdirect.txt",
                   "HTN-panukb" = "/project/xinhe/xsun/multi_group_ctwas/data/tgfm/GTEx_TGFM_PIPs_disease_HYPERTENSION_DIAGNOSED.txt")
for(i in names(trait_list)){
  ctwas_folder_results <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
  ctwas_genes <-  readRDS(paste0(ctwas_folder_results,i,"/",i,".3qtls.combined_pip_rmmapping_bygroup_final.RDS"))
  ctwas_genes_highpip <- ctwas_genes$gene_name[ctwas_genes$combined_pip > 0.8]
  tgfm_genes <- data.table::fread(trait_list[[i]])
  tgfm_genes_highpip <- tgfm_genes[tgfm_genes$TGFM_PIP > 0.8,]
  tgfm_genes_highpip <- tgfm_genes_highpip[tgfm_genes_highpip$genetic_element_class != "variant",]
  tgfm_genes_highpip$ensg_id <- sub("\\..*", "", tgfm_genes_highpip$genetic_element_name)
  tgfm_genes_highpip_ensg <- unique(tgfm_genes_highpip$ensg_id)
  library(EnsDb.Hsapiens.v86)
  ensg_to_symbol <- AnnotationDbi::select(
    EnsDb.Hsapiens.v86,
    keys = tgfm_genes_highpip_ensg,
    keytype = "GENEID",
    columns = c("GENEID", "SYMBOL")
  )

  tgfm_genes_highpip <- ensg_to_symbol$SYMBOL
  ctwas_unique_genes <- setdiff(ctwas_genes_highpip,tgfm_genes_highpip)
  tgfm_unique_genes <- setdiff(tgfm_genes_highpip,ctwas_genes_highpip)
  p1 <- plot_overlap_barplot_cTWAS_vs_TGFM(ctwas_genes_highpip = ctwas_genes_highpip, tgfm_genes_highpip_symbol = tgfm_genes_highpip,trait = i,return_unique_genes = F)
  
  pops <- fread(paste0("/project/xinhe/shengqian/pops/pops_full_features_output/",i,"_pops/",i,".preds"))
  df <- grex(cleanid(pops$ENSGID))
  pops <- cbind(pops,df)
  tgfm_gene <- pops[pops$hgnc_symbol %in% tgfm_unique_genes,]
  ctwas_gene <- pops[pops$hgnc_symbol %in% ctwas_unique_genes,]

  # Combine into one data frame
  df <- data.frame(
    score = c(tgfm_gene$PoPS_Score, ctwas_gene$PoPS_Score),
    group = c(rep("TGFM", length(tgfm_gene$PoPS_Score)),
            rep("cTWAS", length(ctwas_gene$PoPS_Score)))
  )

  # Boxplot
  p2 <- ggplot(df, aes(x = group, y = score, fill = group)) +
    geom_boxplot(alpha = 0.5) +
    scale_fill_manual(values = c("blue", "red")) +
    theme_minimal(base_size = 14) +
    labs(title = i, x = "", y = "PoPS Score",fill = "")
  
  combined_plot <- (p1 | p2) + plot_layout(guides = "collect") & theme(legend.position = "bottom")
  print(combined_plot)
}
```

## Figure 5: cTWAS discovers candidate genes for complex traits and provides insights on their molecular mechanisms 
```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=8, fig.width=10}
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/22.singlegroup_0515/ctwas_output/expression/"
folder_results_multi <- "/project/xinhe/xsun/multi_group_ctwas/23.multi_group_0515/snakemake_outputs/"
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]
param_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.thin1.shared_all.param.RDS"))
ctwas_parameters_multi <- summarize_param(param = param_multi,gwas_n = gwas_n)
top_tissue <- get_top_tissue(ctwas_parameters_multi$group_pve)
param_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin1.shared_all.param.RDS"))
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
title <- paste0(trait, ", top tissue: ", top_tissue)
PIP_cutoff <- 0.8
combined_pip_by_group_single <- readRDS(paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".combined_pip_rmmapping_bygroup_final.RDS"))
combined_pip_by_group_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bytype_final.RDS"))
combined_pip_by_context_multi <- readRDS(paste0(folder_results_multi,trait,"/",trait,".3qtls.combined_pip_rmmapping_bycontext_final.RDS"))
combined_pip_by_group_multi_sig <- combined_pip_by_group_multi[combined_pip_by_group_multi$combined_pip > PIP_cutoff,]
combined_pip_by_group_single_sig <- combined_pip_by_group_single[combined_pip_by_group_single$combined_pip > PIP_cutoff,]
combined_pip_by_context_multi_sig <- combined_pip_by_context_multi[combined_pip_by_context_multi$combined_pip > PIP_cutoff,]
  
combined_pip_by_group_multi_unique <- combined_pip_by_group_multi_sig[!combined_pip_by_group_multi_sig$gene_name %in% combined_pip_by_group_single_sig$gene_name, ]
combined_pip_by_context_multi_unique <- combined_pip_by_context_multi_sig[!combined_pip_by_context_multi_sig$gene_name %in% combined_pip_by_group_single_sig$gene_name, ]

if(nrow(combined_pip_by_group_multi_unique) > 0) {
  p1 <- plot_heatmap(heatmap_data = combined_pip_by_group_multi_unique, main = paste0("New genes identified by multigroup analysis, PIP>",PIP_cutoff),showPIP = T)
}
  
if(nrow(combined_pip_by_context_multi_unique) > 0) {
  p2 <- plot_heatmap(heatmap_data = combined_pip_by_context_multi_unique, main = paste0("New genes identified by multicontext analysis, PIP>",PIP_cutoff),showPIP = T)
}
```

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=8, fig.width=10}
grid.newpage()
print(p1)
```

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=8, fig.width=10}
grid.newpage()
print(p2)
```

## Figure 5: Brain epiQTLs explain a large fraction of missing heritability by eQTLs

### Figure 5a: Enrichment of epiQTLs vs. eQTLs
```{r,echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
prop_heritability <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.parameters.RDS")

# Extract values
log_enrichment <- prop_heritability$log_enrichment
log_enrichment_se <- prop_heritability$log_enrichment_se

# Build data frame
df <- data.frame(
  context = names(log_enrichment),
  value = as.numeric(log_enrichment),
  se = as.numeric(log_enrichment_se)
)

# Plot
ggplot(df, aes(x = context, y = value)) +
  geom_col(fill = "steelblue", alpha = 0.8) +
  geom_errorbar(aes(ymin = value - se, ymax = value + se), width = 0.2) +
  theme_minimal(base_size = 14) +
  labs(x = "", y = "Log Enrichment") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

### Figure 5b: epiQTLs explain a larger fraction of h2g than eQTLs and different QTLs have limited overlap
```{r,echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
library(RColorBrewer)
# Example: Load three different heritability proportion datasets
prop1 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL/SCZ_eQTL.parameters.RDS")$prop_heritability
prop1 <- c(sum(prop1[1:5]),prop1[6])
prop2 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL/SCZ_caQTL.parameters.RDS")$prop_heritability
prop2 <- c(prop2[1:2],prop2[3])
prop3 <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_mQTL/SCZ_mQTL.parameters.RDS")$prop_heritability

# Set up layout for 3 plots in one row
par(mfrow = c(1, 3), mar = c(2, 2, 2, 1))  # Adjust margins if needed

# Choose consistent color palette
colors <- brewer.pal(n = max(lengths(list(prop1, prop2, prop3))), name = "Set2")

# Plot each pie chart
pie(
  prop1,
  labels = paste0(c("eQTL","SNP"), "\n", round(100 * prop1, 1), "%"),
  col = colors,
  border = "white"
)

pie(
  prop2,
  labels = paste0(c("caQTL","SNP"), "\n", round(100 * prop2, 1), "%"),
  col = colors,
  border = "white"
)

pie(
  prop3,
  labels = paste0(c("mQTL","SNP"), "\n", round(100 * prop3, 1), "%"),
  col = colors,
  border = "white"
)

```

```{r,echo=FALSE, message=FALSE, warning=FALSE, results='hide', fig.height=4, fig.width=10}
prop_heritability <- readRDS("/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.parameters.RDS")
prop_heritability <- prop_heritability$prop_heritability
prop_heritability <- c(sum(prop_heritability[1:5]),sum(prop_heritability[6:7]),prop_heritability[8],prop_heritability[9])
colors <- brewer.pal(n = 4, name = "Set2")  # Alternatives: "Pastel1", "Dark2", "Set3"

# Draw pie chart
pie(
  prop_heritability,
  labels = paste0(c("eQTL","caQTL","mQTL","SNP"), "\n", round(100 * prop_heritability, 1), "%"),
  col = colors,
  border = "white"
)
```

### Figure 5c: Using epiQTL discovered novel CREs missed by eQTLs
```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
mapping_table <- readRDS("/project/xinhe/shengqian/cTWAS_epigenetic/data/weights/gene_mapping.RDS")
predictdb_mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_table <- rbind(mapping_table,predictdb_mapping_table)
snp_map <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/LD_region_info/snp_map.RDS")
epi_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL_mQTL/SCZ_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
full_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
suppressMessages({
  suppressWarnings({
    epi_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_caQTL_mQTL/SCZ_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
    full_res <- get_cre_res(mapping_table,"/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS",snp_map)
  })
})
epi_result <- epi_res$result_sig
epi_result <- epi_result[,c("region_id","start_pos","end_pos","total_susie_pip")]
full_result <- full_res$result
full_result <- full_result[,c("region_id","start_pos","end_pos","total_susie_pip")]
setDT(epi_result)
setDT(full_result)
setnames(full_result, "total_susie_pip", "total_susie_pip_full")
merged_result <- full_result[epi_result, 
                             on = .(region_id, start_pos, end_pos)]
merged_result[is.na(total_susie_pip_full), total_susie_pip_full := 0]
p1 <- ggplot(merged_result, aes(x = total_susie_pip, 
                          y = total_susie_pip_full)) +
    geom_point(alpha = 0.6) +
    geom_abline(intercept = 0, slope = 1, linetype = "dashed", color = "red") +
    labs(x = "CRE PIP in EpiQTL results",
       y = "CRE PIP in full results",
       title = "") +
    coord_cartesian(xlim = c(0, 1), ylim = c(0, 1)) +
    theme_minimal(base_size = 14)
```

```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
suppressMessages({
  suppressWarnings({
    eQTL_res <- get_gene_res(mapping_table, "/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL/SCZ_eQTL.finemap_regions_res.RDS", snp_map)
    full_res <- get_gene_res(mapping_table, "/project2/xinhe/shengqian/cTWAS_epigenetics/results/SCZ_eQTL_caQTL_mQTL/SCZ_eQTL_caQTL_mQTL.finemap_regions_res.RDS", snp_map)
  })
})
eQTL_result <- eQTL_res$result_sig
eQTL_result <- eQTL_result[,c("gene_name","combined_pip")]
full_result <- full_res$result
full_result <- full_result[,c("gene_name","combined_pip")]
setDT(eQTL_result)
setDT(full_result)
setnames(full_result, "combined_pip", "combined_pip_full")
merged_result <- full_result[eQTL_result, 
                             on = .(gene_name)]
merged_result[is.na(combined_pip_full), combined_pip_full := 0]
p2 <- ggplot(merged_result, aes(x = combined_pip, 
                          y = combined_pip_full)) +
  geom_point(alpha = 0.6) +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed", color = "red") +
  labs(x = "Gene PIP in eQTL results",
       y = "Gene PIP in full results",
       title = "") +
  coord_cartesian(xlim = c(0, 1), ylim = c(0, 1)) +
  theme_minimal(base_size = 14)
```

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=10}
combined_plot <- (p1 | p2) + plot_layout(guides = "collect") & theme(legend.position = "bottom")
print(combined_plot)
```

