---
title: "6 modality weights from Munro et al."
author: "XSun"
date: "2024-06-05"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r warning=F, message=F}
library(ggplot2)
library(tidyverse)
```

# Introduction

The data is from [Multimodal analysis of RNA sequencing data powers discovery of complex trait genetics](https://www.biorxiv.org/content/10.1101/2024.05.14.594051v1.full). It provides FUSION weights for gene expression, isoform ratios, splice junction usage, alternative TSS/polyA usage, and RNA stability. 


```{r}
dat_sum <- readxl::read_excel("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/Munro_data_summary.xlsx")

DT::datatable(dat_sum,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Table S1 from the paper: Counts of xGenes and xQTLs per GTEx tissue, modalities mapped separately'),options = list(pageLength = 10) )
```

genes_expression_only: Number of genes with at least one cis-QTL for expression (cis-eQTL) and none for any other modality

genes_expression_and_others:	Number of genes with at least one cis-QTL for expression and at least one for any other modality

genes_others_only:	Number of genes with no cis-QTLs for expression and at least one for any other modality

genes_total:	Number of genes with any cis-QTL, i.e. the sum of the previous three columns

genes_expression/isoforms/splicing/alt_TSS/alt_polyA/stability:	Number of genes with at least one cis-QTL for expression/isoforms/splicing/alt_TSS/alt_polyA/stability

qtls_expression/isoforms/splicing/alt_TSS/alt_polyA/stability	Number of conditionally independent cis-QTLs for expression/isoforms/splicing/alt_TSS/alt_polyA/stability

# Expression models

## Comparing with predictdb exression models

```{r}

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/summary/compare_expr_predictdb_munro.rdata")

DT::datatable(df_summary,caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%; ', ''),options = list(pageLength = 5))

```

## An example: LDL-Liver as an example (from the single group analysis)

### z-scores for the overlapped genes

We compared the imputed gene-zscores for the overlapped genes between these 2 datasets. 

Taking LDL-Liver as an example (from the single group analysis), 

```{r}

z_gene_earlier <- readRDS("/project/xinhe/xsun/multi_group_ctwas/1.single_tissue/results/LDL-ukb-d-30780_irnt/Liver/LDL-ukb-d-30780_irnt_Liver.z_gene.RDS")
z_gene_earlier$ensg_id <- sub("\\..*", "", z_gene_earlier$id)

z_gene_current <- readRDS("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results_ldl_liver/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.z_gene.RDS")
z_gene_current$ensg_id <- sub("\\|.*", "", z_gene_current$id)

overlap <- merge(z_gene_current,z_gene_earlier, by="ensg_id")
ggplot(data = overlap) + aes(x = z.x, y=z.y) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, col="red") +
  labs(x="Munro zscores (LDL-liver, single group)",y="Predictdb zscores (LDL-liver, single group)")+
  theme_minimal()

```



### Comparing results with the published ctwas paper

```{r}
mapping_data <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_mapping.RDS")
finemap_res <- readRDS("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results_ldl_liver/LDL-ukb-d-30780_irnt/LDL-ukb-d-30780_irnt.ctwas.res.RDS")

finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

finemap_gene_res <- finemap_gene_res[,c("genename","susie_pip","z","type","context","gene_type")]

DT::datatable(finemap_gene_res,caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%;','Gene PIPs for the LDL-liver single group analysis'),options = list(pageLength = 5))
```


# APA models

## Some stats

```{r}

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/apa.stat.rdata")

colnames(sum) <- c("tissue","#of genes","average # of non-zero weights_lasso","average # of non-zero weights_enet")

DT::datatable(sum,caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%;'),options = list(pageLength = 5))

```

## Comparing the APA model from Chen et al

Chen et al: https://sq-96.github.io/multigroup_ctwas_analysis/apa_weights_analysis_chenetal.html

```{r}


dat_chen <- get(load("/project2/xinhe/shared_data/multigroup_ctwas/weights/apa.stat.rdata"))
dat_chen$tissue <- rownames(dat_chen)

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/apa.stat.rdata")

overlap <- merge(sum,dat_chen, by="tissue")

ggplot(data = overlap, aes(x=ngene_alltissue.x, y=ngene_alltissue.y)) +
  geom_point()+
  geom_abline(slope = 1, intercept = 0, col="red") +
  labs(x="#of apa target in Munro et al (current study)", y="#of apa target in Chen et al", title = "Comparing apa targets reported by 2 studies") +
  theme_minimal()
  

```




