---
title: "6 modality weights from Munro et al."
author: "XSun"
date: "2024-06-05"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r}
library(ggplot2)
library(RSQLite)

```

# Introduction

The data is from [Multimodal analysis of RNA sequencing data powers discovery of complex trait genetics](https://www.biorxiv.org/content/10.1101/2024.05.14.594051v1.full). It provides FUSION weights for gene expression, isoform ratios, splice junction usage, alternative TSS/polyA usage, and RNA stability. 


```{r}
dat_sum <- readxl::read_excel("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/Munro_data_summary.xlsx")

DT::datatable(dat_sum,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Table S1 from the paper: Counts of xGenes and xQTLs per GTEx tissue, modalities mapped separately'),options = list(pageLength = 10) )
```

genes_expression_only: Number of genes with at least one cis-QTL for expression (cis-eQTL) and none for any other modality

genes_expression_and_others:	Number of genes with at least one cis-QTL for expression and at least one for any other modality

genes_others_only:	Number of genes with no cis-QTLs for expression and at least one for any other modality

genes_total:	Number of genes with any cis-QTL, i.e. the sum of the previous three columns

genes_expression/isoforms/splicing/alt_TSS/alt_polyA/stability:	Number of genes with at least one cis-QTL for expression/isoforms/splicing/alt_TSS/alt_polyA/stability

qtls_expression/isoforms/splicing/alt_TSS/alt_polyA/stability	Number of conditionally independent cis-QTLs for expression/isoforms/splicing/alt_TSS/alt_polyA/stability

# Expression models

## Comparing with predictdb exression models

```{r}

folder_weight_predictdb <- "/project2/xinhe/shared_data/multigroup_ctwas/weights/expression_models/"
files_predictdb <- list.files(path = folder_weight_predictdb, pattern = ".db")

folder_weight_munro <- "/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_expression_PredictDB/"
files_munro <- list.files(path = folder_weight_munro, pattern = ".db")

files_overlap <- files_predictdb[files_predictdb %in% files_munro]
tissue <- sub("expression_(.*?)\\.db", "\\1", files_overlap)

sqlite.driver <- dbDriver("SQLite")

gene_num_pred <- c()
gene_num_munro <- c()
overlap <- c()
for (i in 1:length(files_overlap)) {
  
  
  
  file_pred <- paste0(folder_weight_predictdb,files_overlap[i])
  
  db <- dbConnect(sqlite.driver,dbname = file_pred)
  extra <- dbReadTable(db,"extra")
  dbDisconnect(db)
  
  genes_pred <- extra$genename
  gene_num_pred <- c(gene_num_pred,nrow(extra))
  
  file_munro <- paste0(folder_weight_munro,files_overlap[i])
  
  db <- dbConnect(sqlite.driver,dbname = file_munro)
  extra <- dbReadTable(db,"extra")
  dbDisconnect(db)
  
  genes_munro <- extra$genename
  gene_num_munro <- c(gene_num_munro,nrow(extra))
  
  num_overlap <- sum(genes_pred%in%genes_munro)
  overlap <- c(overlap, num_overlap)
}

df_summary <- data.frame(
  tissue = tissue,
  num_gene_predictdb = gene_num_pred,
  num_gene_munro = gene_num_munro,
  num_gene_overlap = overlap
)

DT::datatable(df_summary,caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%; ', 'Number of genes in each data set and the number of overlaps'),options = list(pageLength = 5))


```



# APA models

## Some stats

```{r}

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/apa.stat.rdata")

colnames(sum) <- c("tissue","#of genes","average # of non-zero weights_lasso","average # of non-zero weights_enet")

DT::datatable(sum,caption = htmltools::tags$caption(style = 'caption-side: left; text-align: left; color:black; font-size:150%;'),options = list(pageLength = 5))

```

## Comparing the APA model from Chen et al

Chen et al: https://sq-96.github.io/multigroup_ctwas_analysis/apa_weights_analysis_chenetal.html

```{r}


dat_chen <- get(load("/project2/xinhe/shared_data/multigroup_ctwas/weights/apa.stat.rdata"))
dat_chen$tissue <- rownames(dat_chen)

load("/project2/xinhe/shared_data/multigroup_ctwas/weights/files_Munro/apa.stat.rdata")

overlap <- merge(sum,dat_chen, by="tissue")

ggplot(data = overlap, aes(x=ngene_alltissue.x, y=ngene_alltissue.y)) +
  geom_point()+
  geom_abline(slope = 1, intercept = 0, col="red") +
  labs(x="#of apa target in Munro et al (current study)", y="#of apa target in Chen et al", title = "Comparing apa targets reported by 2 studies") +
  theme_minimal()
  

```




