---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Data

  [GWAS, LD, expr_weights](data.html)
  
  [6 modality weights from Munro et al.](data_6modality_Munro.html)

  [Training stability weights using munro's RNA data -- FUSION pipeline](data_weight_training_fusionscale_rsqupdated.html)
  
  [Training stability weights using munro's RNA data -- FUSION pipeline, use top qtl for non-cs genes](data_weight_training_fusionscale_rsqupdated_allhaveqtl.html)


## Deciding weights  
  
  Comparing results from Munro weights & predictdb weights
  
  * [IBD: predictdb eQTL + sQTL VS Munro 6 modalities -- thin method V1](multi_group_compare_predictdb_munro_updated_cs_thin1.html)
  * [IBD: predictdb eQTL + sQTL + Munro rsQTL + apaQTL VS all 8 weights -- thin method V1](multi_group_compare_predictdb_munro_4weights_8weights_update_thin1.html)
  * [IBD, LDL, SBP, WBC: predictdb eQTL / Munro eQTL / predictdb eQTL + sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / all 8 weights](multi_group_compare_decidingweights_4traits.html) 
  * [IBD, LDL, SBP, WBC: predictdb eQTL / Munro eQTL / predictdb eQTL + sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / predictdb eQTL + sQTL + Munro stQTL](multi_group_compare_decidingweights_4traits_ess.html) 
  * [Enrichment analysis using fractional models, for IBD, LDL, SBP, WBC + predictdb eQTL + sQTL /  predictdb eQTL + sQTL + Munro stQTL](fractional_enrichment.html) 

# cTWAS epigenetics results
  * [cTWAS epigenetic analysis](ctwas_epigenetic_analysis.html)

# Simulation results
  * [Simulation five uncorrelated tissues three omics results](simulation_five_uncorrelated_tissues_three_omics.html)
  * [Simulation five correlated brain tissues three omics results](simulation_five_correlated_brain_tissues_three_omics.html)
  * [Simulation one tissue three omics](simulation_one_tissues_three_omics.html)
  * [Simulation two tissues two omics enrichment](simulation_two_tissues_two_omics.html)
  

# Real data application


## Real data final round 06/03/2025, e + s + st QTLs

### Exploring genetic architecture

[LDL, IBD: different qtl combination](realdata_final_combine_qtls_3qtls.html)

### Tissue selection

Tissue filtered by sample size, irrelevant tissues filtered out, brian tissue removed from non-psychiatric traits

* [sQTL - some tissues removed](realdata_final_tissueselection_mingene0_splicing_exclude_brainprocessed.html)
* [Summary](realdata_final_tissueselection_mingene0_summary.html)

### Multi group analysis results

Data analysis

region merge without mapping table

* [Summary for parameters, number of genes in single/multigroup analysis, heatmap for new genes -- regionmerge](realdata_final_multigroup_summary_3qtls.html)
* [Diagnosis](realdata_final_multigroup_examine_results_3qtls.html)

region merge with mapping table

* [Summary for parameters, number of genes in single/multigroup analysis, heatmap for new genes -- regionmerge](realdata_final_multigroup_summary_3qtls_rmmapping.html)
* [Diagnosis](realdata_final_multigroup_examine_results_3qtls_rmmapping.html)
* [Comparing region merge settings: with mapping table / without mapping table](realdata_final_compare_new_regionmerge.html)

Validation

* [Gene functions](realdata_final_genefunctions.html)
* [Overlap with silver standard genes](realdata_final_multigroup_validation_silver_3qtls.html)
* [Weighted pathway-based gene set enrichment](realdata_final_multigroup_validation_weighted_pwy_3qtls.html)
* [Enrichment analysis using chatgpt](realdata_final_downstream_enrichment_chatgpt.html)
* [Enrichment -- comparing coloc, twas and ctwas results](realdata_final_downstream_enrichment_compare_twasctwascoloc.html)


## Real data final round 05/14/2025 , e + s + st + ed QTLs

### Exploring genetic architecture

[LDL, IBD: different qtl combination](realdata_final_combine_qtls.html)

### Tissue selection

Tissue filtered by sample size, irrelevant tissues filtered out, brian tissue removed from non-psychiatric traits

* [eQTL - some tissues removed](realdata_final_tissueselection_mingene0_exclude_brainprocessed.html)
* [sQTL - some tissues removed](realdata_final_tissueselection_mingene0_splicing_exclude_brainprocessed.html)
* [Summary](realdata_final_tissueselection_mingene0_summary.html)


### Multi group analysis results

Data analysis

* [Summary for parameters, number of genes in single/multigroup analysis, heatmap for new genes](realdata_final_multigroup_summary.html)
* [Gene functions](realdata_final_genefunctions.html)

Validation

* [Comparing enriched terms found by fractional model/enrichR for multigroup&singlegroup](realdata_final_downstream_enrichment_compare.html)





## Omics

### Methylation analysis results

  * [Methylation_analysis results](methylation_analysis.html)


### Single QTL analysis

Package V1.0

  * APA QTL analysis [link](apa_weights_analysis.html)
  * histone QTL analysis [link](histone_weights_analysis.html)
  
Package V2.0

  * [APA QTL analysis (using APA FUSION models by Chen et al.)](apa_weights_analysis_chenetal.html)


## Tissues

### Deciding matching tissues

Package V1.0

  * [Deciding the matching tissue -- by gene numbers](matching_tissue.html)
  
Package V2.0

  * [Deciding the matching tissue -- by gene numbers](matching_tissue_v2.html)
  * [Deciding the matching tissue -- %h2g adaptive method](tissue_selection_extra_tissue.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, round1](real_data_all_tissues.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, G-tests](GWAS_tissue_selection.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, G-tests, scatter plots added](GWAS_tissue_selection_v2.html)

Finalized package

45 tissues jointly

* [eQTL](realdata_final_tissueselection_mingene0.html)
* [sQTL](realdata_final_tissueselection_mingene0_splicing.html)

Tissue filtered by sample size, irrelevant tissues filtered out

* [eQTL - some tissues removed](realdata_final_tissueselection_mingene0_exclude.html)
* [sQTL - some tissues removed](realdata_final_tissueselection_mingene0_splicing_exclude.html)

### Multi-tissue 

Package V1.0

  * [Multi-tissue analysis, merging the LD blocks](multi_tissue_ldmerge.html)
  * [Validation](multi_tissue_ldmerge_validation.html)
  * [Enrichment analysis -- enriched pathways](multi_tissue_ldmerge_enriched_pwy.html)
  * LDL ten tissues analysis [link](LDL_ten_tissues.html)
  
Package V2.0


## Multi omics and multi tissues analysis

  package v1.0

  * LDL - Liver and Adipose - Expression and Splicing [link](LDL_Liver_Adipose_E_S.html)
  * LDL - Multi contexts/types [link](LDL_multi_context_type.html)
  
  package v2.0
  
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- load predicdb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](multi_group_6traits_15weights.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](multi_group_6traits_15weights_ukbb.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](multi_group_6traits_15weights_ukbb_summary.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, all weights are from Munro et al](multi_group_6traits_15weights_ukbb_munro.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, all weights are from Munro et al](multi_group_6traits_15weights_ukbb_munro_summary.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing undo](multi_group_6traits_15weights_ess.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing](multi_group_6traits_15weights_ess_postprocessing.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP & z score](multi_group_6traits_15weights_ess_postprocessing_compare_pipz.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP only](multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.html)
  * [Enrichment analysis for 6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing undo](multi_group_6traits_15weights_ess_enrichment_genesymbol.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al](multi_group_6traits_15weights_ess_summary.html)
  
## Comparing with other methods

* [coloc results](real_data_coloc.html)


## Post - processing

### LD mismatch diagnosis

  * [LD mismatch diagnosis using SuSiE_RSS](detect_LD_mismatch_gwas_UKBBref_susie_rss.html)
  * [Detect LD mismatch for LDL and aFib using DENTIST vs. SuSiE_RSS](detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html)
  * [Compare DENTIST and SuSiE_RSS for LD mismatch detection on example locus with manually changed z-scores](test_LD_mismatch_DENTIST_susierss.html)
  * [Compare LD mismatch filterng on real data](compare_LD_mismatch_filtering_real_data.html)

### Region Merge

  * Test region merge [link](region_merge_test.html)

## Downstream analysis
  
  * [20 traits](multi_group_downstream_analysis.html)
  
  
## Comparing different settings in multi-group analysis

ctwas settings

* [Comparing ctwas running setting: %h2g/gene number/silver standard/go terms -- munro st only](comparing_differnt_settings.html)
* [Comparing ctwas running setting: %h2g/gene number/silver standard/go terms -- susie st added](comparing_differnt_settings_susieST.html)


QTL settings

* [Comparing susie & munro st: parameters only](comparing_munro_susie_st.html)
* [Comparing with susie st & apa model](parameters_apa_li.html)
* [RNA editing QTLs](edqtl.html)

postprocessing settings

* [Comparing ld-mismatch setting](comparing_differnt_settings_ldmm.html)
* [CS filtering or not](comparing_differnt_settings_csindex.html)
  
  
  

## LDL examples

package v2.0

* [Predictdb sQTL from Liver](LDL_predictdb_sQTL.html)
* [Predictdb eQTL&sQTL from Liver](LDL_predictdb_esQTL.html)
* [Comparing single-group & multi-group for LDL – old PIP combining function](LDL_single_multi_compare_oldpip.html)
* [Comparing single-group & multi-group for LDL - new PIP combining function](LDL_single_multi_compare.html)
* [Comparing different PIP combining functions](compare_pip_func_LDL.html)
* [New tissues comparing LD mismatch](LDL_newtissue_example.html)
* [Downstream analysis, predictdb eQTL, sQTL, Munro stQTL](LDL_silver_standard.html)



# Summary

[Exploring genetic architecture](combine_qtls.html)


# Old simulation results

  * seven uncorrelated tissues simulations [link](simulation_seven_tissues_uncorrelated.html)
  * seven correlated tissues simulations [link](simulation_seven_tissues_correlated.html)
  * seven correlated brain tissues simulations [link](simulation_seven_brain_tissues_correlated.html)
  * Single weight simulations [link](single_weight_simulations.html)
  * Four expression traits simulations [link](four_expression_traits_simulations.html)
  * E and S three tissues simulations [link](simulation_E_S_three_tissues_new.html)