---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# Data

  [GWAS, LD, expr_weights](https://sq-96.github.io/multigroup_ctwas_analysis/data.html)
  
  [6 modality weights from Munro et al.](https://sq-96.github.io/multigroup_ctwas_analysis/data_6modality_Munro.html)

  [Training stability weights using munro's RNA data -- FUSION pipeline](https://sq-96.github.io/multigroup_ctwas_analysis/data_weight_training_fusionscale_rsqupdated.html)
  
  [Training stability weights using munro's RNA data -- FUSION pipeline, use top qtl for non-cs genes](https://sq-96.github.io/multigroup_ctwas_analysis/data_weight_training_fusionscale_rsqupdated_allhaveqtl.html)


## Deciding weights  
  
  Comparing results from Munro weights & predictdb weights
  
  * [IBD: predictdb eQTL + sQTL VS Munro 6 modalities -- thin method V1](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_compare_predictdb_munro_updated_cs_thin1.html)
  * [IBD: predictdb eQTL + sQTL + Munro rsQTL + apaQTL VS all 8 weights -- thin method V1](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_compare_predictdb_munro_4weights_8weights_update_thin1.html)
  * [IBD, LDL, SBP, WBC: predictdb eQTL / Munro eQTL / predictdb eQTL + sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / all 8 weights](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_compare_decidingweights_4traits.html) 
  * [IBD, LDL, SBP, WBC: predictdb eQTL / Munro eQTL / predictdb eQTL + sQTL / predictdb eQTL + sQTL + Munro rsQTL + apaQTL / predictdb eQTL + sQTL + Munro stQTL](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_compare_decidingweights_4traits_ess.html) 
  * [Enrichment analysis using fractional models, for IBD, LDL, SBP, WBC + predictdb eQTL + sQTL /  predictdb eQTL + sQTL + Munro stQTL](https://sq-96.github.io/multigroup_ctwas_analysis/fractional_enrichment.html) 



# Simulation results
  * [Simulation results](simulation_summary.html)
  * [Simulation null weight results](simulation_null_weight_summary.html)
  * [Simulation Param PIPs](simulation_param_PIPs.html)
  * [Simulation no enrichment](simulation_no_enrichment.html)
  * [Simulation one tissue no enrichment](simulation_one_tissue_no_enrichment.html)
  * [Simulation two tissues no enrichment](simulation_two_tissues_no_enrichment.html)
  * [Simulation no enrichment four tissues combined](simulation_no_enrichment_four_tissues_combined.html)
  * [Simulation pip calibration](simulation_pip_calibration.html)
  

# Real data application

## Omics

### Methylation analysis results

  * [Methylation_analysis results](methylation_analysis.html)


### Single QTL analysis

Package V1.0

  * APA QTL analysis [link](https://sq-96.github.io/multigroup_ctwas_analysis/apa_weights_analysis.html)
  * histone QTL analysis [link](https://sq-96.github.io/multigroup_ctwas_analysis/histone_weights_analysis.html)
  
Package V2.0

  * [APA QTL analysis (using APA FUSION models by Chen et al.)](https://sq-96.github.io/multigroup_ctwas_analysis/apa_weights_analysis_chenetal.html)


## Tissues

### Deciding matching tissues

Package V1.0

  * [Deciding the matching tissue -- by gene numbers](https://sq-96.github.io/multigroup_ctwas_analysis/matching_tissue.html)
  
Package V2.0

  * [Deciding the matching tissue -- by gene numbers](https://sq-96.github.io/multigroup_ctwas_analysis/matching_tissue_v2.html)
  * [Deciding the matching tissue -- %h2g adaptive method](https://sq-96.github.io/multigroup_ctwas_analysis/tissue_selection_extra_tissue.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, round1](https://sq-96.github.io/multigroup_ctwas_analysis/real_data_all_tissues.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, G-tests](https://sq-96.github.io/multigroup_ctwas_analysis/GWAS_tissue_selection.html)
  * [Deciding the matching tissue -- run ctwas using all tissues, G-tests, scatter plots added](https://sq-96.github.io/multigroup_ctwas_analysis/GWAS_tissue_selection_v2.html)

### Multi-tissue 

Package V1.0

  * [Multi-tissue analysis, merging the LD blocks](https://sq-96.github.io/multigroup_ctwas_analysis/multi_tissue_ldmerge.html)
  * [Validation](https://sq-96.github.io/multigroup_ctwas_analysis/multi_tissue_ldmerge_validation.html)
  * [Enrichment analysis -- enriched pathways](https://sq-96.github.io/multigroup_ctwas_analysis/multi_tissue_ldmerge_enriched_pwy.html)
  * LDL ten tissues analysis [link](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_ten_tissues.html)
  
Package V2.0


## Multi omics and multi tissues analysis

  package v1.0

  * LDL - Liver and Adipose - Expression and Splicing [link](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_Liver_Adipose_E_S.html)
  * LDL - Multi contexts/types [link](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_multi_context_type.html)
  
  package v2.0
  
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- load predicdb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ukbb.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, eqtl, sqtl from predictdb; apa qtl from Chen et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ukbb_summary.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, all weights are from Munro et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ukbb_munro.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, all weights are from Munro et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ukbb_munro_summary.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing undo](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_postprocessing.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP & z score](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_postprocessing_compare_pipz.html)
  * [6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing, comparing 2 LD mismatch approaches, problematic genes filtered by PIP only](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_postprocessing_compare_nozfilter.html)
  * [Enrichment analysis for 6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing undo](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_enrichment_genesymbol.html)
  * [Summary for 6 Traits, 5 tissues, eQTL + sQTL + stQTL -- compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al](https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess_summary.html)
  
## Comparing with other methods

* [coloc results](real_data_coloc.html)


## Post - processing

### LD mismatch diagnosis

  * [LD mismatch diagnosis using SuSiE_RSS](detect_LD_mismatch_gwas_UKBBref_susie_rss.html)
  * [Detect LD mismatch for LDL and aFib using DENTIST vs. SuSiE_RSS](detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html)
  * [Compare DENTIST and SuSiE_RSS for LD mismatch detection on example locus with manually changed z-scores](test_LD_mismatch_DENTIST_susierss.html)
  * [Compare LD mismatch filterng on real data](compare_LD_mismatch_filtering_real_data.html)

### Region Merge

  * Test region merge [link](https://sq-96.github.io/multigroup_ctwas_analysis/region_merge_test.html)



## Downstream analysis
  
  * [20 traits](multi_group_downstream_analysis.html)
  
  
## Comparing different settings in multi-group analysis

* [Comparing ctwas running setting: %h2g/gene number/silver standard/go terms -- munro st only](https://sq-96.github.io/multigroup_ctwas_analysis/comparing_differnt_settings.html)
* [Comparing ctwas running setting: %h2g/gene number/silver standard/go terms -- susie st added](https://sq-96.github.io/multigroup_ctwas_analysis/comparing_differnt_settings_susieST.html)
* [Comparing ctwas running setting: parameters only -- susie + munro st](https://sq-96.github.io/multigroup_ctwas_analysis/comparing_munro_susie_st.html)

* [Comparing ld-mismatch setting](https://sq-96.github.io/multigroup_ctwas_analysis/comparing_differnt_settings_ldmm.html)

  
  
## LDL examples

package v2.0

* [Predictdb sQTL from Liver](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_predictdb_sQTL.html)
* [Predictdb eQTL&sQTL from Liver](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_predictdb_esQTL.html)
* [Comparing single-group & multi-group for LDL â€“ old PIP combining function](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_single_multi_compare_oldpip.html)
* [Comparing single-group & multi-group for LDL - new PIP combining function](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_single_multi_compare.html)
* [Comparing different PIP combining functions](https://sq-96.github.io/multigroup_ctwas_analysis/compare_pip_func_LDL.html)
* [New tissues comparing LD mismatch](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_newtissue_example.html)
* [Downstream analysis, predictdb eQTL, sQTL, Munro stQTL](https://sq-96.github.io/multigroup_ctwas_analysis/LDL_silver_standard.html)


# Old simulation results

  * seven uncorrelated tissues simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/simulation_seven_tissues_uncorrelated.html)
  * seven correlated tissues simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/simulation_seven_tissues_correlated.html)
  * seven correlated brain tissues simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/simulation_seven_brain_tissues_correlated.html)
  * Single weight simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/single_weight_simulations.html)
  * Four expression traits simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/four_expression_traits_simulations.html)
  * E and S three tissues simulations [link](https://sq-96.github.io/multigroup_ctwas_analysis/simulation_E_S_three_tissues_new.html)