---
title: "Deciding the matching tissue (using package V2)"
author: "XSun"
date: "2024-05-16"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Analysis settings

## parameters

- thin = 0.1
- niter_prefit = 3
- niter = 30
- L = 5
- group_prior_var_structure = “independent”
- maxSNP = 20000
- min_nonSNP_PIP = 0.5

## Traits

aFib, IBD, LDL, SBP, SCZ, WBC 

[details](https://sq-96.github.io/multigroup_ctwas_analysis/data.html)

## Weights 

PredictDB, 32 tissues with sample size (RNASeq.and.Genotyped.samples below) >= 200

```{r }
load("/project/xinhe/xsun/multi_group_ctwas/1.single_tissue/summary/para_pip08.rdata")
sum_all <- para_sum
colnames(sum_all)[ncol(sum_all)] <-"PVE(%)"
sum_all$`PVE(%)` <- as.numeric(sum_all$`PVE(%)`)*100

weights <- read.csv("/project/xinhe/xsun/ctwas/1.matching_tissue/results_data/gtex_samplesize.csv")
DT::datatable(weights,options = list(pageLength = 5))

num_top <- 10
```

##  Tissue-tissue correlation data 

From [MASH paper, PMID30478440](https://www.nature.com/articles/s41588-018-0268-8)

The MASH paper provides a matrix indicating the shared magnitude of eQTLs among tissues [by computing the proportion of effects significant in either tissue that are within 2-fold magnitude of one another](https://stephenslab.github.io/gtexresults/SharingMag.html).

[data download link (MASH matrix)](https://github.com/stephenslab/gtexresults)

```{r}
load("/project/xinhe/xsun/ctwas/1.matching_tissue/data/tissue_cor.rdata")
clrs <- colorRampPalette(rev(c("#D73027","#FC8D59","#FEE090","#FFFFBF",
                               "#E0F3F8","#91BFDB","#4575B4")))(64)
```

We filtered the correlation using 0.8 as cutoff.

## Functions used 

```{r }
library(lattice)
library(gridExtra)
library(ggplot2)
get_correlation <- function(tissue1, tissue2, cor_matrix) {
  # Check if tissues are in the matrix and find their positions
  pos1 <- match(tissue1, colnames(cor_matrix))
  pos2 <- match(tissue2, rownames(cor_matrix))
  
  return(cor_matrix[pos2, pos1]) 
}

r2cutoff <- 0.8
filter_tissues <- function(tissue_list, cor_matrix) {
  filtered_list <- c(tissue_list[1]) # Start with the first tissue
  
  for (i in 2:length(tissue_list)) {
    high_correlation_found <- FALSE
    for (j in 1:length(filtered_list)) {
      if (!is.na(get_correlation(filtered_list[j], tissue_list[i], cor_matrix)) &&
          get_correlation(filtered_list[j], tissue_list[i], cor_matrix) > r2cutoff) {
        high_correlation_found <- TRUE
        break # Break the inner loop as high correlation is found
      }
    }
    
    if (!high_correlation_found) {
      filtered_list <- c(filtered_list, tissue_list[i])
    }
  }
  return(filtered_list)
}

fill_upper_triangle <- function(cor_matrix) {
    for (i in 1:nrow(cor_matrix)) {
        for (j in 1:ncol(cor_matrix)) {
            # Check if the upper triangle element is NA and the lower triangle element is not NA
            if (is.na(cor_matrix[i, j]) && !is.na(cor_matrix[j, i])) {
                # Copy the value from the lower triangle to the upper triangle
                cor_matrix[i, j] <- cor_matrix[j, i]
            }
        }
    }
    cor_matrix[lower.tri(cor_matrix)] <- NA
    return(cor_matrix)
}

combine_vectors <- function(vec1, vec_cor1, vec2, vec_cor2, pad_with_na = TRUE) {
  # Check if padding with NAs is required
  if (pad_with_na) {
    # Pad the shorter primary vector and its corresponding vector with NAs
    if (length(vec1) < length(vec2)) {
      extra_length <- length(vec2) - length(vec1)
      vec1 <- c(vec1, rep(NA, extra_length))
      vec_cor1 <- c(vec_cor1, rep(NA, extra_length))
    } else {
      extra_length <- length(vec1) - length(vec2)
      vec2 <- c(vec2, rep(NA, extra_length))
      vec_cor2 <- c(vec_cor2, rep(NA, extra_length))
    }
    # Combine the vectors into a matrix/data frame
    return(cbind(vec1, vec_cor1, vec2, vec_cor2))
  } else {
    # Combine the vectors into a list
    return(list(vec1=vec1, vec_cor1=vec_cor1, vec2=vec2, vec_cor2=vec_cor2))
  }
}


```



# aFib

```{r }

sum_cat <- sum_all[sum_all$trait =="aFib-ebi-a-GCST006414",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```

# IBD

```{r }

sum_cat <- sum_all[sum_all$trait =="IBD-ebi-a-GCST004131",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```

# LDL

```{r }

sum_cat <- sum_all[sum_all$trait =="LDL-ukb-d-30780_irnt",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```

# SBP

```{r }

sum_cat <- sum_all[sum_all$trait =="SBP-ukb-a-360",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```

# SCZ

```{r }

sum_cat <- sum_all[sum_all$trait =="SCZ-ieu-b-5102",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```

# WBC

```{r }

sum_cat <- sum_all[sum_all$trait =="WBC-ieu-b-30",]

DT::datatable(sum_cat,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','summary for ctwas parameters and high pip genes (all tissues analyzed)'),options = list(pageLength = 5) )
```

## Filtering based on the number of high PIP genes

```{r }
sum_cat <- sum_cat[order(as.numeric(sum_cat$`#pip>0.8&incs`),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$`#pip>0.8&incs`[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","#highpip_gene","filtered","#highpip_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )

```

## Filtering based on PVE explained by genes 

```{r}
sum_cat <- sum_cat[order(as.numeric(sum_cat$group_pve),decreasing = T),]

tissue_select <- sum_cat$tissue[num_top:1]
tissue_select <- tissue_select[tissue_select%in%rownames(lat)]

##heatmap
cor <- lat[tissue_select,tissue_select]
cor <- fill_upper_triangle(cor)
print(levelplot(cor,col.regions = clrs,xlab = "",ylab = "",
                colorkey = TRUE,main = "heatmap showing the tissue-tissue correlation"))

##cor matrix
cor <- cor[rev(tissue_select),rev(tissue_select)]
DT::datatable(cor,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;  font-size:150% ;','tissue-tissue correlation matrix '),options = list(pageLength = 10) )


tissue_select_f1 <- sum_cat$tissue[1:num_top]
filtered_tissues <- filter_tissues(tissue_select_f1, cor)

tissue_select_f1_cor <- sum_cat$group_pve[sum_cat$tissue %in% tissue_select_f1 ]
filtered_tissues_cor <- sum_cat$group_pve[sum_cat$tissue %in% filtered_tissues ]
comb <- combine_vectors(tissue_select_f1, tissue_select_f1_cor, filtered_tissues, filtered_tissues_cor, pad_with_na = TRUE)
colnames(comb) <- c("without_filtering","PVE_gene","filtered","PVE_gene")


DT::datatable(comb,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Comparing the top tissue lists (with/without filtering by tissue correlation) '),options = list(pageLength = 10) )
```



# Selected Tissues 

| Trait | Number of Tissues | Tissue                               | # of high pip genes (in cs) | PVE (if same in number of high pip genes) |
|-------|-------------------|--------------------------------------|---------------------|-------------------------------------------|
| aFib  | 5                 | Heart_Atrial_Appendage               | 23                  |                                           |
|       |                   | Esophagus_Muscularis                 | 14                  |                                           |
|       |                   | Muscle_Skeletal                      | 11                  |                                           |
|       |                   | Thyroid                              | 12                  |                                           |
|       |                   | Brain_Cerebellum                     | 10                  |                                           |
| IBD   | 6                 | Cells_Cultured_fibroblasts           | 16                  |                                           |
|       |                   | Whole_Blood                          | 14                  |                                           |
|       |                   | Adipose_Subcutaneous                 | 12                  |                                           |
|       |                   | Esophagus_Mucosa                     | 8                   | 0.025728                                  |
|       |                   | Heart_Left_Ventricle                 | 8                   | 0.023978                                  |
|       |                   | Thyroid                              | 8                   | 0.016433                                  |
| LDL   | 5                 | Liver                                | 40                  |                                           |
|       |                   | Spleen                               | 24                  |                                           |
|       |                   | Adipose_Subcutaneous                 | 21                  |                                           |
|       |                   | Adrenal_Gland                        | 19                  |                                           |
|       |                   | Esophagus_Mucosa                     | 19                  |                                           |
| SBP   | 5                 | Artery_Tibial                        | 35                  |                                           |
|       |                   | Adipose_Subcutaneous                 | 29                  |                                           |
|       |                   | Brain_Cortex                         | 29                  |                                           |
|       |                   | Heart_Left_Ventricle                 | 25                  |                                           |
|       |                   | Spleen                               | 21                  |                                           |
| SCZ   | 6                 | Heart_Left_Ventricle                 | 23                  |                                           |
|       |                   | Adrenal_Gland                        | 22                  |                                           |
|       |                   | Artery_Coronary                      | 16                  |                                           |
|       |                   | Brain_Cerebellum                     | 16                  |                                           |
|       |                   | Spleen                               | 15                  | 0.01715                                   |
|       |                   | Stomach                              | 15                  | 0.01868                                   |
| WBC   | 5                 | Whole_Blood                          | 109                 |                                           |
|       |                   | Adipose_Subcutaneous                 | 68                  |                                           |
|       |                   | Artery_Aorta                         | 56                  |                                           |
|       |                   | Skin_Sun_Exposed_Lower_leg           | 54                  |                                           |
|       |                   | Spleen                               | 52                  |                                           |



# Comparing with the results from package V1

The only different setting running these analyses: 

- When running V1: we used weights for nolnc genes

- When running V2: we just used weights for protein-coding genes, which are a bit fewer.


```{r, fig.width=15, fig.height=5}
load("/project/xinhe/xsun/ctwas/1.matching_tissue/results_data/summary_table.rd")
colnames(sum_all)[8] <- "num_gene_susiepip08"
results_v1 <- sum_all

load("/project/xinhe/xsun/ctwas/1.matching_tissue/results_data/summary_table_cs.rdata")
colnames(sum_all)[1:2] <- c("traits", "weights")
# Merge the data frames
results_v1 <- merge(results_v1, sum_all, by = c("traits", "weights"), all.x = TRUE)


load("/project/xinhe/xsun/multi_group_ctwas/1.single_tissue/summary/para_pip08.rdata")
results_v2 <- para_sum

results_v1 <- results_v1[results_v1$traits %in% results_v2$trait,]
results_v1 <- results_v1[results_v1$weights %in% results_v2$tissue,]


# Create a combined key column in both data frames
results_v1$key <- paste(results_v1$traits, results_v1$weights, sep = "_")
results_v2$key <- paste(results_v2$trait, results_v2$tissue, sep = "_")

# Sort each data frame by the combined key
results_v1_sorted <- results_v1[order(results_v1$key), ]
results_v2_sorted <- results_v2[order(results_v2$key), ]

# Optionally, drop the key column if it's no longer needed
results_v1_sorted$key <- NULL
results_v2_sorted$key <- NULL

results_v1_sorted$group_pve_gene <- as.numeric(results_v1_sorted$group_pve_gene)
results_v2_sorted$group_pve <- as.numeric(results_v2_sorted$group_pve)
p1 <- ggplot(data = NULL, aes(x = results_v1_sorted$group_pve_gene, y = results_v2_sorted$group_pve)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, color = "red") +
  labs(x = "group_pve (V1)", y = "group_pve (V2)") +
  theme_minimal()

results_v1_sorted$num_gene_susiepip08 <- as.numeric(results_v1_sorted$num_gene_susiepip08)
results_v2_sorted$`#pip>0.8` <- as.numeric(results_v2_sorted$`#pip>0.8`)
# Plot 2: num_gene_susiepip08 vs. #pip>0.8
p2 <- ggplot(data = NULL, aes(x = results_v1_sorted$num_gene_susiepip08, y = results_v2_sorted$`#pip>0.8`)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, color = "red") +
  labs(x = "#pip>0.8 (V1)", y = "#pip>0.8 (V2)") +
  theme_minimal()

results_v1_sorted$num_gene_08_cs <- as.numeric(results_v1_sorted$num_gene_08_cs)
results_v2_sorted$`#pip>0.8&incs` <- as.numeric(results_v2_sorted$`#pip>0.8&incs`)
# Plot 3: num_gene_08_cs vs. #pip>0.8&incs
p3 <- ggplot(data = NULL, aes(x = results_v1_sorted$num_gene_08_cs, y = results_v2_sorted$`#pip>0.8&incs`)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, color = "red") +
  labs(x = "#pip>0.8&incs (V1)", y = "#pip>0.8&incs (V2)") +
  theme_minimal()

grid.arrange(p1, p2, p3, ncol = 3)
```
