---
title: "methylation_analysis"
output: html_document
date: '2024-10-22'
editor_options: 
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, results='hide', warning=FALSE}
library(ggplot2)
library(ggVennDiagram)
library(gridExtra)
library(data.table)
library(ctwas)
library(data.table)
library(tidyverse)
library(RSQLite)
library(dplyr)
library(magrittr)
library(pheatmap)
plot_piechart <- function(ctwas_parameters, colors, by) {
  # Create the initial data frame
  data <- data.frame(
    category = names(ctwas_parameters$prop_heritability),
    percentage = ctwas_parameters$prop_heritability
  )
  
  # Split the category into context and type
  data <- data %>%
    mutate(
      context = sub("\\|.*", "", category),
      type = sub(".*\\|", "", category)
    )
  
  # Aggregate the data based on the 'by' parameter
  if (by == "type") {
    data <- data %>%
      group_by(type) %>%
      summarize(percentage = sum(percentage)) %>%
      mutate(category = type)  # Use type as the new category
  } else if (by == "context") {
    data <- data %>%
      group_by(context) %>%
      summarize(percentage = sum(percentage)) %>%
      mutate(category = context)  # Use context as the new category
  } else {
    stop("Invalid 'by' parameter. Use 'type' or 'context'.")
  }
  
  # Calculate percentage labels for the chart
  data$percentage_label <- paste0(round(data$percentage * 100, 1), "%")
  
  # Create the pie chart
  pie <- ggplot(data, aes(x = "", y = percentage, fill = category)) +
    geom_bar(stat = "identity", width = 1) +
    coord_polar("y", start = 0) +
    theme_void() +  # Remove background and axes
    geom_text(aes(label = percentage_label), 
              position = position_stack(vjust = 0.5), size = 3) +  # Adjust size as needed
    scale_fill_manual(values = colors) +  # Custom colors
    labs(fill = "Category") +  # Legend title
    ggtitle("Percent of Heritability")  # Title
  
  return(pie)
}

draw_gene_piechart_type <- function(data, colors) {
  # Filter data based on combined_pip
  data <- data[data$combined_pip > 0.8, ]
  
  # Count occurrences by eQTL, sQTL, and stQTL
  byeQTL <- nrow(data[data$eQTL_pip / data$combined_pip > 0.8, ])
  bysQTL <- nrow(data[data$sQTL_pip / data$combined_pip > 0.8, ])
  bystQTL <- nrow(data[data$stQTL_pip / data$combined_pip > 0.8, ])
  
  # Count occurrences for combined sQTL and stQTL
  bysQTLstQTL <- nrow(data[((data$stQTL_pip + data$sQTL_pip) / data$combined_pip) > 0.8 &
                             data$stQTL_pip / data$combined_pip < 0.8 &
                             data$sQTL_pip / data$combined_pip < 0.8, ])
  
  # Count unspecified
  unspecified <- nrow(data) - byeQTL - bysQTL - bystQTL - bysQTLstQTL
  
  # Create vectors for plotting
  n <- c(byeQTL, bysQTL, bystQTL, bysQTLstQTL, unspecified)
  prop <- round(n / nrow(data), 3)
  labels = c("by eQTL", "by sQTL", "by stQTL", "by sQTL+stQTL", "unspecified")
  lab.ypos = cumsum(prop) - 0.5 * prop
  
  # Prepare the data frame for plotting
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  
  # Generate the pie chart
  ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0) +
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5)) +
    scale_fill_manual(values = colors) +  # Ensure 'palette' is defined
    theme_void()
}

draw_gene_piechart_tissue <- function(data, colors){
  
  data <- data[data$combined_pip>0.8,]
  tissues <- colnames(data)[3:7]
  tissues <- sub("_pip$", "", tissues)
  colnames(data)[3:7] <- paste0("tissue",c(1:5))
  
  
  bytissue1 <- nrow(data[data$tissue1/data$combined_pip>0.8,])
  bytissue2 <- nrow(data[data$tissue2/data$combined_pip>0.8,])
  bytissue3 <- nrow(data[data$tissue3/data$combined_pip>0.8,])
  bytissue4 <- nrow(data[data$tissue4/data$combined_pip>0.8,])
  bytissue5 <- nrow(data[data$tissue5/data$combined_pip>0.8,])
  unspecified <- nrow(data)-bytissue1-bytissue2-bytissue3-bytissue4-bytissue5
  
  n <- c(bytissue1,bytissue2,bytissue3,bytissue4,bytissue5,unspecified)
  
  prop <- round(n/nrow(data),3)
  labels = c(tissues,"unspecified")
  lab.ypos = cumsum(prop) - 0.5*prop
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  p <- ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5))+
    scale_fill_manual(values = colors) +
    theme_void()
  
  return(p)
}

plot_heatmap <- function(heatmap_data, main) {
  
  rownames(heatmap_data) <- heatmap_data$gene_name
  heatmap_data <- heatmap_data %>% dplyr::select(-gene_name, -combined_pip)
  
  if(nrow(heatmap_data) ==1){
    
    heatmap_data <- rbind(heatmap_data,rep(0,ncol(heatmap_data)))
    rownames(heatmap_data)[2] <- "fake_gene_for_plotting"
    
  }
  
  heatmap_matrix <- as.matrix(heatmap_data)
  
  p <- pheatmap(heatmap_matrix,
                cluster_rows = F,   # Cluster the rows (genes)
                cluster_cols = F,   # Cluster the columns (QTL types)
                color = colorRampPalette(c("white", "red"))(50), # Color gradient
                display_numbers = TRUE, # Display numbers in cells
                main = main,labels_row = rownames(heatmap_data), silent = T)
  
  return(p)
}
```
### Fusion Lasso model of DNA methylation

I built lasso model of DNA methylation with FUSION for Whole Blood and Colon Transverse. Similar to meQTL mapping, for each CpG site, I extracted surrounding 50kb genptypes and train lasso models with cross validation. With heritability cutoff p<0.0001, I have about 16,000 and 48,000 CpG sites in whole blood and colon transverse. Among which, 5,000 and 40,000 CpG sites are also in QTL mapping. Colon have more overlaps than whole blood. The average cross-validation R2 for lasso in whole blood and colon transverse are 0.393 and 0.248 In the single QTL approach (qval < 0.001), we have 7,720 and 91,466 CpG sites.

```{r, echo=FALSE, warning=FALSE}
# Arrange the two plots side by side
library(ggplot2)
library(ggVennDiagram)
library(gridExtra)
library(data.table)
library(ctwas)
fusion_model <- fread("/project2/xinhe/shengqian/fusion_twas-master/methylation_wholeblood_50kb_h00001.profile",sep='\t')
fusion_model <- fusion_model[!is.na(fusion_model$lasso.r2),]
QTL_data <- fread("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mqtl/WholeBlood.regular.perm.fdr.txt")
QTL_data <- QTL_data[QTL_data$qval<0.001,]

venn1 <- ggVennDiagram(list(QTL = QTL_data$cpg_id, FUSION = fusion_model$id),category.names=c("","")) +
  ggtitle("Whole Blood") + theme(legend.position = "none") + scale_fill_gradient(low = NA, high = NA) + annotate("text", x = 0, y = - 2.5, label = "QTL", size = 5) +
  annotate("text", x = 0, y = 6.5, label = "FUSION", size = 5)

fusion_model <- fread("/project2/xinhe/shengqian/fusion_twas-master/methylation_colon_50kb_h00001.profile",sep='\t')
fusion_model <- fusion_model[!is.na(fusion_model$lasso.r2),]
QTL_data <- fread("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mqtl/ColonTransverse.regular.perm.fdr.txt")
QTL_data <- QTL_data[QTL_data$qval<0.001,]

venn2 <- ggVennDiagram(list(QTL = QTL_data$cpg_id, FUSION = fusion_model$id),category.names=c("","")) +
  ggtitle("Colon Transverse") + theme(legend.position = "none") + scale_fill_gradient(low = NA, high = NA) + annotate("text", x = 0, y = - 2.5, label = "QTL", size = 5) +
  annotate("text", x = 0, y = 6.5, label = "FUSION", size = 5)

grid.arrange(
  grobs = list(venn1, venn2),
  ncol = 2
)
```

## IBD results

### cTWAS parameters (50kb, h2 pvalue<0.00001)
```{r, echo=FALSE, fig.width=10, fig.height=8}
param <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.param.RDS")
gwas_n <- 59957
colors <- c(  "#1f77b4", "#ff7f0e", "#2ca02c", "#d62728",  "#9467bd", "#8c564b", "#e377c2", "#7f7f7f",  "#bcbd22",  "#17becf",  "#f7b6d2",  "#c5b0d5",  "#9edae5", "#ffbb78",  "#98df8a",  "#ff9896" )
make_convergence_plots(param, gwas_n,colors = colors)
```

### meQTLs explained 10% heritability
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width=10, fig.height=5}
ctwas_parameters <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.ctwas_parameters.RDS")
pve_pie_by_type_multi <- plot_piechart(ctwas_parameters = ctwas_parameters, colors = colors, by = "type")
pve_pie_by_context_multi <- plot_piechart(ctwas_parameters = ctwas_parameters, colors = colors, by = "context")
aplot::plot_list(pve_pie_by_type_multi,pve_pie_by_context_multi)
#gridExtra::grid.arrange(pve_pie_by_type_multi,pve_pie_by_context_multi, ncol = 2)
```

### cTWAS with eQTLs and meQTLs identifies 28 genes with PIP > 0.8
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
gwas_n <- 59957
finemap_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.finemap_regions_res.RDS")
gene_annot <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
susie_alpha_res <- anno_susie_alpha_res(finemap_res$susie_alpha_res,
                                        mapping_table = gene_annot,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

#combined_pip_by_type <- combine_gene_pips(susie_alpha_res, 
#                                             group_by = "gene_name",
#                                             by = "type",
#                                             method = "combine_cs",
#                                             filter_cs = TRUE,
#                                             include_cs_id = FALSE)

#combined_pip_by_context <- combine_gene_pips(susie_alpha_res, 
#                                             group_by = "gene_name",
#                                             by = "context",
#                                             method = "combine_cs",
#                                             filter_cs = TRUE,
#                                             include_cs_id = FALSE)

combined_pip_by_group <- combine_gene_pips(susie_alpha_res, 
                                             group_by = "gene_name",
                                             by = "group",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = FALSE)

#DT::datatable(combined_pip_by_type[combined_pip_by_type$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by modalities'),options = list(pageLength = 10) )

#DT::datatable(combined_pip_by_context[combined_pip_by_context$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by contexts'),options = list(pageLength = 10) )

DT::datatable(combined_pip_by_group,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by modalities and contexts'),options = list(pageLength = 10) )
```

### Top cTWAS genes from single group models have very little overlap
```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
gene_annot <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
snp_map <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_wholeblood_E/ibd_wholeblood_E.snp_map.RDS")
wholeblood_E_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_wholeblood_E/ibd_wholeblood_E.finemap_regions_res.RDS")
wholeblood_E_res <- anno_finemap_res(wholeblood_E_res$finemap_res,
                                snp_map = snp_map,
                                mapping_table = gene_annot,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")
wholeblood_E_res <- subset(wholeblood_E_res, type != "SNP" & susie_pip > 0.8 & !is.na(cs))
wholeblood_E_res <- wholeblood_E_res[,c("gene_name","susie_pip","context")]
DT::datatable(wholeblood_E_res,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Top eQTL genes in wholeblood'),options = list(pageLength = 10) )

snp_map <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_wholeblood_M/ibd_wholeblood_M.snp_map.RDS")
wholeblood_M_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_wholeblood_M/ibd_wholeblood_M.finemap_regions_res.RDS")
wholeblood_M_res <- anno_finemap_res(wholeblood_M_res$finemap_res,
                                snp_map = snp_map,
                                mapping_table = gene_annot,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")
wholeblood_M_res <- subset(wholeblood_M_res, type != "SNP" & susie_pip > 0.8 & !is.na(cs))
wholeblood_M_res <- wholeblood_M_res[,c("gene_name","susie_pip","context")]
DT::datatable(wholeblood_M_res,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Top meQTL genes in wholeblood'),options = list(pageLength = 10) )


snp_map <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_colon_E/ibd_colon_E.snp_map.RDS")
colon_E_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_colon_E/ibd_colon_E.finemap_regions_res.RDS")
colon_E_res <- anno_finemap_res(colon_E_res$finemap_res,
                                snp_map = snp_map,
                                mapping_table = gene_annot,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")
colon_E_res <- subset(colon_E_res, type != "SNP" & susie_pip > 0.8 & !is.na(cs))
colon_E_res <- colon_E_res[,c("gene_name","susie_pip","context")]
DT::datatable(colon_E_res,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Top eQTL genes in colon'),options = list(pageLength = 10) )

snp_map <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_colon_M/ibd_colon_M.snp_map.RDS")
colon_M_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_colon_M/ibd_colon_M.finemap_regions_res.RDS")
colon_M_res <- anno_finemap_res(colon_M_res$finemap_res,
                                snp_map = snp_map,
                                mapping_table = gene_annot,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")
colon_M_res <- subset(colon_M_res, type != "SNP" & susie_pip > 0.8 & !is.na(cs))
colon_M_res <- colon_M_res[,c("gene_name","susie_pip","context")]
DT::datatable(colon_M_res,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Top meQTL genes in colon'),options = list(pageLength = 10) )

set1 <- wholeblood_E_res$gene_name
set2 <- wholeblood_M_res$gene_name
set3 <- colon_E_res$gene_name
set4 <- colon_M_res$gene_name

library(RColorBrewer)
library(VennDiagram)
myCol <- brewer.pal(4, "Pastel2")
# Chart
aa <- venn.diagram(
  x = list(set1, set2, set3, set4),
  category.names = c("wholeblood E" , "wholeblood M " , "Colon E", "Colon M"),
  filename = NULL,
  lwd = 2,
  lty = 'blank',
  fill = myCol
)

grid.draw(aa)
```

```{r, echo=FALSE, results='hide'}
gwas_n <- 59957
finemap_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_M/ibd_multi_M.finemap_regions_res.RDS")
gene_annot <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
susie_alpha_res <- anno_susie_alpha_res(finemap_res$susie_alpha_res,
                                        mapping_table = gene_annot,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)


genes_M <- combine_gene_pips(susie_alpha_res, 
                               group_by = "gene_name",
                               by = "context",
                               method = "combine_cs",
                               filter_cs = TRUE,
                               include_cs_id = FALSE)

genes_M <- genes_M[genes_M$combined_pip>0.8,]
```

```{r, echo=FALSE, results='hide'}
gwas_n <- 59957
finemap_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E/ibd_multi_E.finemap_regions_res.RDS")
gene_annot <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
susie_alpha_res <- anno_susie_alpha_res(finemap_res$susie_alpha_res,
                                        mapping_table = gene_annot,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)


genes_E <- combine_gene_pips(susie_alpha_res, 
                               group_by = "gene_name",
                               by = "context",
                               method = "combine_cs",
                               filter_cs = TRUE,
                               include_cs_id = FALSE)

genes_E <- genes_E[genes_E$combined_pip>0.8,]
```

```{r, echo=FALSE, results='hide'}
gwas_n <- 59957
finemap_res <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.finemap_regions_res.RDS")
gene_annot <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
susie_alpha_res <- anno_susie_alpha_res(finemap_res$susie_alpha_res,
                                        mapping_table = gene_annot,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)


genes_EM <- combine_gene_pips(susie_alpha_res, 
                               group_by = "gene_name",
                               by = "context",
                               method = "combine_cs",
                               filter_cs = TRUE,
                               include_cs_id = FALSE)

genes_EM <- genes_EM[genes_EM$combined_pip>0.8,]
```

### Adding eQTL to meQTL identifies an additional 16 high PIP genes
1. 12/17 meQTL genes still have combined PIP > 0.8 after adding eQTL
2. 3/17 meQTL genes have decreased combined PIP < 0.8 after adding eQTL
3. 2/17 meQTL genes are lost, due to region selection after adding eQTL
4. Among the 12 overalpped genes:
- One gene (BRD7) is mediation (meQTL pip decreases from 0.96 to 0.09, eQTL pip=0.90).
- Two genes (TNFSF15 and ATG16L1) are competition (eQTL pip = 0.2 and meQTL pip is decreased by 0.2).
- Nine genes are meQTL alone (no eQTL pip).
```{r, echo=FALSE, fig.width=6, fig.height=5}
set1 <- genes_M$gene_name
set2 <- genes_EM$gene_name
aa <- venn.diagram(
  x = list(set1, set2),
  category.names = c("meQTL alone","meQTL + eQTL"),
  filename = NULL,
  lwd = 2,
  lty = 'blank',
  fill = c("#87CEFA","#fb8072")
)

grid.draw(aa)
```

### Adding meQTL to eQTL identifies an additional 11 high PIP genes
```{r, echo=FALSE, fig.width=6, fig.height=5}
set1 <- genes_E$gene_name
set2 <- genes_EM$gene_name
aa <- venn.diagram(
  x = list(set1, set2),
  category.names = c("eQTL alone","meQTL + eQTL"),
  filename = NULL,
  lwd = 2,
  lty = 'blank',
  fill = c("#87CEFA","#fb8072")
)

grid.draw(aa)
```

### Genes that identified by meQTL not eQTL
```{r echo=FALSE, message=FALSE, results='hide', warning=FALSE}
library(ctwas)
library(EnsDb.Hsapiens.v86)
ens_db <- EnsDb.Hsapiens.v86

mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_E_S_M_mapping.RDS")
ctwas_res_multi <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.finemap_regions_res.RDS")
snp_map_multi <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.snp_map.RDS")
weights_multi <- readRDS("/project/xinhe/shengqian/cTWAS_analysis/data/ibd_multi_E_M/ibd_multi_E_M.preprocessed.weights.RDS")

finemap_res_multi <- ctwas_res_multi$finemap_res
finemap_res_multi <- anno_finemap_res(finemap_res_multi,
                                      snp_map = snp_map_multi,
                                      mapping_table = mapping_predictdb,
                                      add_gene_annot = TRUE,
                                      map_by = "molecular_id",
                                      drop_unmapped = TRUE,
                                      add_position = TRUE,
                                      use_gene_pos = "mid")
```

#### ETS1
```{r echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#finemap_res_multi[!is.na(finemap_res_multi$gene_name) & finemap_res_multi$gene_name == "ETS1", ]
#pdf(file = "/project/xinhe/shengqian/TNFSF15.pdf",width = 12,height = 10)#
make_locusplot(finemap_res_multi,
               region_id = "9_114257521_115159681",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```


#### ATG16L1
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/ATG16L1.pdf",width = 12,height = 10)
make_locusplot(finemap_res_multi,
               region_id = "2_232685293_234242343",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```

#### TNFSF15
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/TNFSF15.pdf",width = 12,height = 10)
make_locusplot(finemap_res_multi,
               region_id = "11_128318642_130472680",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```

#### HLA_DRA
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/HLA_DRA.pdf",width = 12,height = 10)
#finemap_res_multi[finemap_res_multi$gene_name=="TNFSF15",]
make_locusplot(finemap_res_multi,
               region_id = "6_31603441_32714887",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```

#### ADCY3
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/ATG16L1.pdf",width = 12,height = 10)
#finemap_res_multi[finemap_res_multi$gene_name=="TNFSF15",]
make_locusplot(finemap_res_multi,
               region_id = "2_24464049_26672117",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```

#### RGS14
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/ATG16L1.pdf",width = 12,height = 10)
make_locusplot(finemap_res_multi,
               region_id = "5_176023592_177824506",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```

#### SEC16A
```{r, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
#pdf(file = "/project/xinhe/shengqian/ATG16L1.pdf",width = 12,height = 10)
make_locusplot(finemap_res_multi,
               region_id = "9_136047132_136605890",
               ens_db = ens_db,
               weights = weights_multi,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = TRUE,
               color_pval_by = "cs",
               color_pip_by = "cs")
```






