---
title: "Multi group analysis: 6 traits, 5 tissues, eQTL + sQTL + apaQTL -- Compute UKBB LD"
author: "XSun"
date: "2024-05-27"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Overview

## Traits

aFib, IBD, LDL, SBP, SCZ, WBC 

[details](https://sq-96.github.io/multigroup_ctwas_analysis/data.html)

## Tissues

The independent tissues are selected by [single tissue analysis](https://sq-96.github.io/multigroup_ctwas_analysis/matching_tissue_v2.html)

## Omics

eQTL, sQTL weights are from GTEx PredictDB 

apaQTL wetights are from https://www.nature.com/articles/s41467-024-46064-7#Sec2. Top 10 SNPs with largest abs(weights) were selected after harmonization

## Settings

1. Weight processing: 

PredictDB:

- drop_strand_ambig = TRUE,
- scale_by_ld_variance = TRUE,
- load_predictdb_LD = F,  

FUSION:

- method_FUSION = "enet",
- fusion_top_n_snps = 10, 
- drop_strand_ambig = TRUE,
- scale_by_ld_variance = F, 
- load_predictdb_LD = F,  

2. Parameter estimation and fine-mapping

- niter_prefit = 5,
- niter = 60,
- L = 3,
- group_prior_var_structure = "shared_type",
- maxSNP = 20000,
- min_nonSNP_PIP = 0.5,

3. Memory requested

- cpus-per-task=2
- mem=200G

100G 2core got killed


```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(ctwas)
library(data.table)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(gridExtra)
#source("/project2/xinhe/shengqian/cTWAS/for_xiaotong/R/ctwas_summarize_finemap_res.R")
load("/project2/xinhe/shared_data/multigroup_ctwas/weights/E_S_A_mapping.RData")
E_S_A_mapping <- E_S_A_mapping[E_S_A_mapping$gene_type=="protein_coding",] #limit to protein coding genes
palette <- c(brewer.pal(12, "Paired"), brewer.pal(12, "Set3"), brewer.pal(6, "Dark2"))

sum_pip_across_contexts <- function(finemap_res, mapping_data){

  finemap_res <- finemap_res[finemap_res$cs_index!=0,]
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

  df_gene <- aggregate(finemap_gene_res$susie_pip, by=list(finemap_gene_res$genename), FUN=sum)
  colnames(df_gene) <- c("genename", "combined_pip")
  for(j in unique(finemap_gene_res$context)){
    tmp_res <- finemap_gene_res[finemap_gene_res$context==j,]
    tmp_res <- tmp_res[,c("genename","susie_pip")]
    tmp_res <- aggregate(tmp_res$susie_pip, by=list(tmp_res$genename), FUN=sum)
    colnames(tmp_res) <- c("genename",j)
    tmp_res[,j] <- round(tmp_res[,j],3)
    df_gene <- df_gene %>% plyr::join(tmp_res, by = "genename") %>% replace(is.na(.), 0)
  }
  df_gene$combined_pip <- round(df_gene$combined_pip,3)
  df_gene <- df_gene[order(-df_gene$combined_pip),]
  return(df_gene)
}

sum_pip_across_types <- function(finemap_res, mapping_data){

  finemap_res <- finemap_res[finemap_res$cs_index!=0,]
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

  df_gene <- aggregate(finemap_gene_res$susie_pip, by=list(finemap_gene_res$genename), FUN=sum)
  colnames(df_gene) <- c("genename", "combined_pip")
  for(j in unique(finemap_gene_res$type)){
    tmp_res <- finemap_gene_res[finemap_gene_res$type==j,]
    tmp_res <- tmp_res[,c("genename","susie_pip")]
    tmp_res <- aggregate(tmp_res$susie_pip, by=list(tmp_res$genename), FUN=sum)
    colnames(tmp_res) <- c("genename",j)
    tmp_res[,j] <- round(tmp_res[,j],3)
    df_gene <- df_gene %>% plyr::join(tmp_res, by = "genename") %>% replace(is.na(.), 0)
  }
  df_gene$combined_pip <- round(df_gene$combined_pip,3)
  df_gene <- df_gene[order(-df_gene$combined_pip),]
  return(df_gene)
}

draw_gene_piechart <- function(data){
  data <- data[data$combined_pip>0.8,]
  byeQTL <- nrow(data[data$eQTL/data$combined_pip>0.8,])
  bysQTL <- nrow(data[data$sQTL/data$combined_pip>0.8,])
  byapaQTL <- nrow(data[data$apaQTL/data$combined_pip>0.8,])
  bysQTLapaQTL <- nrow(data[((data$apaQTL+data$sQTL)/data$combined_pip)>0.8 &
                              data$apaQTL/data$combined_pip < 0.8 &
                              data$sQTL/data$combined_pip < 0.8,])
  unspecified <- nrow(data)-byeQTL-bysQTL-byapaQTL-bysQTLapaQTL
  n <- c(byeQTL,bysQTL,byapaQTL,bysQTLapaQTL,unspecified)
  prop <- round(n/nrow(data),3)
  labels = c("by eQTL","by sQTL","by apaQTL","by sQTL+apaQTL","unspecified")
  lab.ypos = cumsum(prop) - 0.5*prop
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5))+
    scale_fill_manual(values = palette) +
    theme_void()
  #pie(Prop , labels = c("by eQTL","by sQTL","by apaQTL","by sQTL+apaQTL","unspecified"), col=palette)
}



draw_gene_piechart_tissue <- function(data){
  
  data <- data[data$combined_pip>0.8,]
  tissues <- colnames(data)[3:7]
  colnames(data)[3:7] <- paste0("tissue",c(1:5))
  
  
  bytissue1 <- nrow(data[data$tissue1/data$combined_pip>0.8,])
  bytissue2 <- nrow(data[data$tissue2/data$combined_pip>0.8,])
  bytissue3 <- nrow(data[data$tissue3/data$combined_pip>0.8,])
  bytissue4 <- nrow(data[data$tissue4/data$combined_pip>0.8,])
  bytissue5 <- nrow(data[data$tissue5/data$combined_pip>0.8,])
  unspecified <- nrow(data)-bytissue1-bytissue2-bytissue3-bytissue4-bytissue5
  
  n <- c(bytissue1,bytissue2,bytissue3,bytissue4,bytissue5,unspecified)
  
  prop <- round(n/nrow(data),3)
  labels = c(tissues,"unspecified")
  lab.ypos = cumsum(prop) - 0.5*prop
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  p <- ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5))+
    scale_fill_manual(values = palette) +
    theme_void()
  
  print(p)
  print(tissues)

}

sum_pve_across_types <- function(ctwas_parameters) {
  # Round the group_pve values
  pve <- round(ctwas_parameters$group_pve, 4)
  pve <- as.data.frame(pve)

  # Extract SNP PVE for later use
  SNP_pve <- pve["SNP", ]

  # Add type and context columns
  pve$type <- sapply(rownames(pve), function(x) { unlist(strsplit(x, "[|]"))[1] })
  pve$context <- sapply(rownames(pve), function(x) { unlist(strsplit(x, "[|]"))[2] })

  # Remove rows with NA values and sort
  pve <- na.omit(pve)
  pve <- pve[order(rownames(pve)), ]

  # Aggregate PVE by type
  df_pve <- aggregate(pve$pve, by = list(pve$type), FUN = sum)
  colnames(df_pve) <- c("type", "total_pve")
  df_pve$total_pve <- round(df_pve$total_pve, 4)

  # Add context-specific columns
  for (context in unique(pve$context)) {
    context_pve <- aggregate(pve$pve, by = list(pve$type, pve$context), FUN = sum)
    context_pve <- context_pve[context_pve$Group.2 == context, ]
    colnames(context_pve)[3] <- context
    df_pve <- merge(df_pve, context_pve[, c("Group.1", context)], by.x = "type", by.y = "Group.1", all.x = TRUE)
  }

  # Insert SNP PVE
  SNP_row <- c("SNP", SNP_pve, rep(0, ncol(df_pve) - 2))
  df_pve <- rbind(df_pve, SNP_row)

  # Convert to numeric except for the type column
  df_pve[, -1] <- lapply(df_pve[, -1], as.numeric)

  # Sum all rows and add a sum_pve row
  sum_row <- colSums(df_pve[, -1], na.rm = TRUE)
  sum_row <- c("total_pve", sum_row)
  df_pve <- rbind(df_pve, sum_row)

  # Clean up row names and return
  row.names(df_pve) <- NULL
  return(df_pve)
}

load("/project2/xinhe/shared_data/multigroup_ctwas/gwas/samplesize.rdata")

process_fm_res_single <- function(finemap_res, mapping_data) {
  
  finemap_res <- finemap_res
  mapping_data <- mapping_data
  
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res <- finemap_gene_res[finemap_gene_res$cs_index!=0,]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  
  finemap_gene_res <- finemap_gene_res[,c("genename","susie_pip")]
  colnames(finemap_gene_res) <- c("genename","PIP_single_group")
  return(finemap_gene_res)
}


calculate_sum_overlap_and_pve <- function(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters) {
  sum_overlap_for_all <- c()
  sum_pve_for_all <- c()
  
  for (i in seq_along(tissues)) {
    tissue <- tissues[i]
    
    # Load single tissue fine-mapping results
    file_res_single <- paste0("/project/xinhe/xsun/multi_group_ctwas/1.single_tissue/results/", trait, "/", tissue, "/", trait, "_", tissue, ".finemap_res.RDS")
    finemap_res_single <- readRDS(file_res_single)
    finemap_res_single <- process_fm_res_single(finemap_res = finemap_res_single, mapping_data = E_S_A_mapping)
    
    # Load multi-group fine-mapping results
    finemap_res_multi <- combined_pip_by_contexts
    
    # Identify significant genes
    sig_gene_single <- finemap_res_single[finemap_res_single$PIP_single_group > 0.8, ]
    sig_gene_multi <- finemap_res_multi[finemap_res_multi$combined_pip > 0.8, ]
    
    # Merge significant genes
    merged_siggene <- merge(sig_gene_single, sig_gene_multi, by = "genename", all.x = TRUE, all.y = TRUE)
    
    # Calculate overlap statistics
    tmp <- c(
      sum(merged_siggene$PIP_single_group > 0.8, na.rm = TRUE),
      sum(merged_siggene$PIP_single_group > 0.8 & merged_siggene$combined_pip > 0.8, na.rm = TRUE),
      sum(merged_siggene$combined_pip > 0.8, na.rm = TRUE)
    )
    sum_overlap_for_all <- rbind(sum_overlap_for_all, tmp)
    
    # Calculate PVE for multi-group
    group_pve_multi <- ctwas_parameters$group_pve
    group_pve_multi_tissue <- round(group_pve_multi[paste0("eQTL|", tissue)], digits = 4)
    
    # Load single tissue parameter results
    file_para_single <- paste0("/project/xinhe/xsun/multi_group_ctwas/1.single_tissue/results/", trait, "/", tissue, "/", trait, "_", tissue, ".param.RDS")
    para_single <- readRDS(file_para_single)
    group_pve_single_tissue <- round(para_single$group_prior[paste0("eQTL|", tissue)], digits = 4)
    
    # Combine PVE results
    tmp2 <- c(group_pve_single_tissue, group_pve_multi_tissue)
    sum_pve_for_all <- rbind(sum_pve_for_all, tmp2)
  }
  
  # Set row and column names for the result matrices
  rownames(sum_overlap_for_all) <- tissues
  colnames(sum_overlap_for_all) <- c("#of.sig.gene_single_group_cs_filtered", "#of.overlap", "#of.sig.gene_multi_group_cs_filtered")
  
  rownames(sum_pve_for_all) <- tissues
  colnames(sum_pve_for_all) <- c("pve_single_group", "pve_eqtl_multi_group")
  
  return(list(sum_overlap_for_all = sum_overlap_for_all, sum_pve_for_all = sum_pve_for_all))
}

pve_pie_chart <- function(pve_vector, palette=NULL, title) {
  # Create data frame for plotting
  data <- data.frame(
    Group = names(pve_vector),
    Value = pve_vector
  )
  
  # Calculate percentages
  data$Percentage <- round(100 * data$Value / sum(data$Value), 1)
  
  # Set palette if not specified
  if (is.null(palette)) {
    palette <- brewer.pal(min(8, length(data$Group)), "Set3")
  }
  
  # Create pie chart
  ggplot(data, aes(x = "", y = Value, fill = Group)) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +  # This transforms the bar chart into a pie chart
    scale_fill_manual(values = palette, name = "Group") +
    labs(title = title, x = NULL, y = NULL) +
    theme_void() +
    theme(plot.title = element_text(hjust = 0.5), 
          legend.position = "right", 
          legend.title = element_text(face = "bold"), 
          legend.text = element_text(size = 12)) +
    geom_text(aes(label = paste(Percentage, "%", sep="")), position = position_stack(vjust = 0.5))
}

```

# Results

**Results from multi-group analysis**

The results are summarized by 

1. Heritability contribution by contexts: we aggregate the PVE values by omics and tissues, making it easier to understand the distribution of PVE across different genetic contexts.

2. Combined PIP by omics:  we aggregate the Susie PIPs by omics 

3. Combined PIP by contexts: we aggregate the Susie PIPs by tissues, making it easier to understand the distribution of PIP across different genetic contexts.

4. Specific molecular traits of top genes: we creates a pie chart to visualize the proportion of genes classified into different categories based on their PIPs contributed by each genetics contexts. The categories are based on the proportion of each QTL type relative to the combined PIP value: 

- by eQTL: Number of genes where the ratio of eQTL to combined PIP is greater than 0.8.
- by sQTL: Number of genes where the ratio of sQTL to combined PIP is greater than 0.8.
- by apaQTL: Number of genes where the ratio of apaQTL to combined PIP is greater than 0.8.
- by sQTL+apaQTL: Number of genes where the combined ratio of apaQTL and sQTL to combined PIP is greater than 0.8, but neither apaQTL nor sQTL individually exceed 0.8.
- unspecified: Number of genes not classified into any of the above categories.

**Comparing with single group eQTL results**

*Please not that the ealier single group eQTL analyses were performed under L=5 but the current analyses were under L=3*

We compared number of significant genes, overlapping genes and the changes in PVE for eQTLs across five tissues reported by single eQTL analysi


## aFib

TO DO

## IBD

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "IBD-ebi-a-GCST004131"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)


file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)

load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

```

### Comparing with single group eQTL results 

```{r echo=FALSE}

tissues <- c("Cells_Cultured_fibroblasts","Whole_Blood","Adipose_Subcutaneous","Esophagus_Mucosa","Heart_Left_Ventricle")
sprintf("the top tissues from single group analyses are %s",paste0(tissues,collapse = ","))

result <- calculate_sum_overlap_and_pve(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters)

DT::datatable(result$sum_overlap_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','#of overlap for single eqtl analysis and multi group analysis'),options = list(pageLength = 5) )

DT::datatable(result$sum_pve_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','pve changes'),options = list(pageLength = 5) )

```

We then looked into `Whole_Blood`. Please not that the ealier single group eQTL analyses were performed under `L=5` but the current analyses were under `L=3`. 

So we re-ran the single group analysis using `L=3` with for `Whole_Blood`. 


```{r echo=FALSE}
load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_multi_single_IBD-ebi-a-GCST004131.rdata")

unique_single <- df_summary[complete.cases(df_summary$PIP_single_group),]

sprintf("The number of sig genes reported by single group analysis under L=3 is %s",nrow(unique_single[unique_single$PIP_single_group > 0.8 & unique_single$cs_index_single_group !=0,]))
sprintf("The number of overlapped genes is %s",sum(df_summary$overlap))

unique_single <- unique_single[unique_single$PIP_single_group > 0.8 & unique_single$cs_index_single_group !=0 & !unique_single$overlap,]

DT::datatable(unique_single,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Sig genes from single group analysis but not reported in muilti-group analysis'),options = list(pageLength = 5) )


```


locus zoom plot for the regions containing these genes: https://uchicago.box.com/s/ziv0v006pxwnuprhxbagdfr0cnfk5e23



```{r}


DT::datatable(df_summary,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','All genes with pip > 0.8 from single group analysis and all genes with combined pip >0.8 from multi group analysis'),options = list(pageLength = 5) )


ggplot(df_summary, aes(x = combined_pip_multi_group, y = combined_pip_nonzero_cs_multi_group)) +
  geom_point() +  # Add points
  geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +  
  labs(x = "Combined PIP Multi Group (adding up all pips)", 
       y = "Combined PIP Nonzero CS Multi Group (only pips with non-zero CS added)",
       title = "Comparison for combined pip") +
  theme_minimal()
```


## LDL

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()
```

### Comparing with single group eQTL results

```{r echo=FALSE}

tissues <- c("Liver","Spleen","Adipose_Subcutaneous","Adrenal_Gland","Esophagus_Mucosa")
sprintf("the top tissues from single group analyses are %s",paste0(tissues,collapse = ","))

result <- calculate_sum_overlap_and_pve(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters)

DT::datatable(result$sum_overlap_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','#of overlap for single eqtl analysis and multi group analysis'),options = list(pageLength = 5) )
DT::datatable(result$sum_pve_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','pve changes'),options = list(pageLength = 5) )

```


We then looked into `Adipose_Subcutaneous`. Please not that the ealier single group eQTL analyses were performed under `L=5` but the current analyses were under `L=3`.

So we re-ran the single group analysis using `L=3` with for `Adipose_Subcutaneous`.


```{r echo=FALSE}
load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_multi_single_LDL-ukb-d-30780_irnt.rdata")

df_summary$region_ids[df_summary$gene_name=="TPD52"] <- "8:78220626-81044160"

unique_single <- df_summary[complete.cases(df_summary$PIP_single_group),]

sprintf("The number of sig genes reported by single group analysis under L=3 is %s",nrow(unique_single[unique_single$PIP_single_group > 0.8 & unique_single$cs_index_single_group !=0,]))
sprintf("The number of overlapped genes is %s",sum(df_summary$overlap))

unique_single <- unique_single[unique_single$PIP_single_group > 0.8 & unique_single$cs_index_single_group !=0 & !unique_single$overlap,]

DT::datatable(unique_single,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Sig genes from single group analysis but not reported in muilti-group analysis'),options = list(pageLength = 5) )
```


locus zoom plot for the regions containing these genes: https://uchicago.box.com/s/u93gpgil79ybf7akrl1obfkpgkxq5wkn



```{r}
DT::datatable(df_summary,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','All genes with pip > 0.8 from single group analysis and all genes with combined pip >0.8 from multi group analysis'),options = list(pageLength = 5) )

ggplot(df_summary, aes(x = combined_pip_multi_group, y = combined_pip_nonzero_cs_multi_group)) +
  geom_point() +  # Add points
  geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +
  labs(x = "Combined PIP Multi Group (adding up all pips)",
       y = "Combined PIP Nonzero CS Multi Group (only pips with non-zero CS added)",
       title = "Comparison for combined pip") +
  theme_minimal()
```

## SBP

### Results from multi-group analysis

```{r echo=FALSE, message=FALSE}
trait <- "SBP-ukb-a-360"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

finemap_gene_res_highpip$p <- 10^(-finemap_gene_res_highpip$log10p)
DT::datatable(finemap_gene_res_highpip[finemap_gene_res_highpip$gene_name == "NAA60",],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','The one with high PIP but also high TWAS p'),options = list(pageLength = 5) )
load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/naa60.rdata")
DT::datatable(naa60,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','All signals for NAA60'),options = list(pageLength = 5) )

```

plot for this locus https://uchicago.box.com/s/uca02ksxb4hz67lohjpko35nkjhp5pti


We ran regular SuSiE (uniform prior) in a region, and obtain the number of CS from the results. Then we ran fine-mapping step for this region. 

By doing these, we have 

```{r}

estimated_L <- readRDS("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_preL/SBP-ukb-a-360/SBP-ukb-a-360.estimated_L.RDS")

sprintf("the pre-estimated L is %s", estimated_L["16:2714828-3951195"])

load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/NAA60.preL.rdata")
DT::datatable(finemap_res_region_multi,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Updated PIP (run with pre-estimated L)'),options = list(pageLength = 5) )
```


### Comparing with single group eQTL results

```{r echo=FALSE}

tissues <- c("Artery_Tibial","Adipose_Subcutaneous","Brain_Cortex","Heart_Left_Ventricle","Spleen")
sprintf("the top tissues from single group analyses are %s",paste0(tissues,collapse = ","))

result <- calculate_sum_overlap_and_pve(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters)

DT::datatable(result$sum_overlap_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','#of overlap for single eqtl analysis and multi group analysis'),options = list(pageLength = 5) )
DT::datatable(result$sum_pve_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','pve changes'),options = list(pageLength = 5) )

```

## SCZ

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "SCZ-ieu-b-5102"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()
```

### Comparing with single group eQTL results

```{r echo=FALSE}

tissues <- c("Heart_Left_Ventricle","Adrenal_Gland","Artery_Coronary","Brain_Cerebellum","Stomach")
sprintf("the top tissues from single group analyses are %s",paste0(tissues,collapse = ","))

result <- calculate_sum_overlap_and_pve(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters)

DT::datatable(result$sum_overlap_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','#of overlap for single eqtl analysis and multi group analysis'),options = list(pageLength = 5) )
DT::datatable(result$sum_pve_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','pve changes'),options = list(pageLength = 5) )

```

## WBC

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "WBC-ieu-b-30"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

finemap_gene_res_highpip$p <- 10^(-finemap_gene_res_highpip$log10p)

gene_problem <- c("CCZ1","ELMO1","ING5","SLC22A5")

gene_show <- finemap_gene_res_highpip[finemap_gene_res_highpip$gene_name %in% gene_problem,]
gene_show <- gene_show[gene_show$p > 0.1,]
DT::datatable(gene_show,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','The one with high PIP but also high TWAS p'),options = list(pageLength = 5) )

```

Locus zoom plots for these 3 genes

https://uchicago.box.com/s/a1qej83jaabft9z7nqmzewzpsxrg6uui

https://uchicago.box.com/s/fzmv9ypzaa9vt7a4u0al37w7s77qrzdu


https://uchicago.box.com/s/70tk99ifpz7xshknh57u2pnf7jygx5zh


This PIP vs. p-value analysis suggests that some of the high PIP genes can be FPs. When the number of true signals in a region is < L (3 in our case), SuSiE will still try to report L results. This type of FP findings can be removed in the regular SuSiE by the CS filter. But it’s likely less effective in the cTWAS case, because our prior heavily favors genes. We may end up with a single gene in a CS with PIP > 0.95, then that gene will not be filtered by the purity filter.

One simple way to get the true number of signals is: run regular SuSiE (uniform prior) in a region, and obtain the number of CS from the results. This should be a pretty good estimate of the number of true signals. 

By doing these, for these 3 regions, 

For ING5, we have 

```{r}

estimated_L <- readRDS("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_preL/WBC-ieu-b-30/WBC-ieu-b-30.estimated_L.RDS")

sprintf("the pre-estimated L is %s", estimated_L["2:241210506-242147200"])

load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/ING5.preL.rdata")
DT::datatable(finemap_res_region_multi,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Updated PIP (run with pre-estimated L)'),options = list(pageLength = 5) )
```

For CCZ1, region 7:5814895-6534226 didn't pass the screen_regions step 

For ELMO1, region 7:36173929-37515581 didn't pass the screen_regions step 

### Comparing with single group eQTL results

```{r echo=FALSE}

tissues <- c("Whole_Blood","Adipose_Subcutaneous","Artery_Aorta","Skin_Sun_Exposed_Lower_leg","Spleen")
sprintf("the top tissues from single group analyses are %s",paste0(tissues,collapse = ","))

result <- calculate_sum_overlap_and_pve(tissues, trait, combined_pip_by_contexts, E_S_A_mapping, ctwas_parameters)

DT::datatable(result$sum_overlap_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','#of overlap for single eqtl analysis and multi group analysis'),options = list(pageLength = 5) )
DT::datatable(result$sum_pve_for_all,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','pve changes'),options = list(pageLength = 5) )

```