---
title: "6 Traits, 5 tissues, eQTL + sQTL + apaQTL -- compute ukbb LD, weights are from Munro et al"
author: "XSun"
date: "2024-05-27"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Overview

## Traits

aFib, IBD, LDL, SBP, SCZ, WBC 

[details](https://sq-96.github.io/multigroup_ctwas_analysis/data.html)

## Tissues

The independent tissues are selected by [single tissue analysis](https://sq-96.github.io/multigroup_ctwas_analysis/matching_tissue_v2.html)

## Omics

eQTL, sQTL, apaQTL weights are from [Munro et al.](https://sq-96.github.io/multigroup_ctwas_analysis/data_6modality_Munro.html)

## Settings

1. Weight processing: 

PredictDB:

all the PredictDB are converted from FUSION weights

- drop_strand_ambig = TRUE,
- scale_by_ld_variance = F (FUSION converted weights)
- load_predictdb_LD = F,  

2. Parameter estimation and fine-mapping

- niter_prefit = 5,
- niter = 60,
- L = 3,
- group_prior_var_structure = "shared_type",
- maxSNP = 20000,
- min_nonSNP_PIP = 0.5,

3. Memory requested

- cpus-per-task=2
- mem=200G

over 30h running time

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(ctwas)
library(data.table)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(gridExtra)
#source("/project2/xinhe/shengqian/cTWAS/for_xiaotong/R/ctwas_summarize_finemap_res.R")
load("/project2/xinhe/shared_data/multigroup_ctwas/weights/E_S_A_mapping.RData")

E_S_A_mapping_earlier <- get(load("/project2/xinhe/shared_data/multigroup_ctwas/weights/E_S_A_mapping.RData"))
E_S_A_mapping_earlier <- E_S_A_mapping_earlier[E_S_A_mapping_earlier$gene_type=="protein_coding",] #limit to p


E_S_A_mapping <-readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/Munro_mapping.RDS")
E_S_A_mapping <- E_S_A_mapping[E_S_A_mapping$gene_type=="protein_coding",] #limit to protein coding genes


palette <- c(brewer.pal(12, "Paired"), brewer.pal(12, "Set3"), brewer.pal(6, "Dark2"))

sum_pip_across_contexts <- function(finemap_res, mapping_data){

  finemap_res <- finemap_res[finemap_res$cs_index!=0,]
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

  df_gene <- aggregate(finemap_gene_res$susie_pip, by=list(finemap_gene_res$genename), FUN=sum)
  colnames(df_gene) <- c("genename", "combined_pip")
  for(j in unique(finemap_gene_res$context)){
    tmp_res <- finemap_gene_res[finemap_gene_res$context==j,]
    tmp_res <- tmp_res[,c("genename","susie_pip")]
    tmp_res <- aggregate(tmp_res$susie_pip, by=list(tmp_res$genename), FUN=sum)
    colnames(tmp_res) <- c("genename",j)
    tmp_res[,j] <- round(tmp_res[,j],3)
    df_gene <- df_gene %>% plyr::join(tmp_res, by = "genename") %>% replace(is.na(.), 0)
  }
  df_gene$combined_pip <- round(df_gene$combined_pip,3)
  df_gene <- df_gene[order(-df_gene$combined_pip),]
  return(df_gene)
}


sum_pip_across_types <- function(finemap_res, mapping_data){

  finemap_res <- finemap_res[finemap_res$cs_index!=0,]
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

  df_gene <- aggregate(finemap_gene_res$susie_pip, by=list(finemap_gene_res$genename), FUN=sum)
  colnames(df_gene) <- c("genename", "combined_pip")
  for(j in unique(finemap_gene_res$type)){
    tmp_res <- finemap_gene_res[finemap_gene_res$type==j,]
    tmp_res <- tmp_res[,c("genename","susie_pip")]
    tmp_res <- aggregate(tmp_res$susie_pip, by=list(tmp_res$genename), FUN=sum)
    colnames(tmp_res) <- c("genename",j)
    tmp_res[,j] <- round(tmp_res[,j],3)
    df_gene <- df_gene %>% plyr::join(tmp_res, by = "genename") %>% replace(is.na(.), 0)
  }
  df_gene$combined_pip <- round(df_gene$combined_pip,3)
  df_gene <- df_gene[order(-df_gene$combined_pip),]
  return(df_gene)
}

draw_gene_piechart <- function(data){
  data <- data[data$combined_pip>0.8,]
  byeQTL <- nrow(data[data$eQTL/data$combined_pip>0.8,])
  bysQTL <- nrow(data[data$sQTL/data$combined_pip>0.8,])
  byapaQTL <- nrow(data[data$apaQTL/data$combined_pip>0.8,])
  bysQTLapaQTL <- nrow(data[((data$apaQTL+data$sQTL)/data$combined_pip)>0.8 &
                              data$apaQTL/data$combined_pip < 0.8 &
                              data$sQTL/data$combined_pip < 0.8,])
  unspecified <- nrow(data)-byeQTL-bysQTL-byapaQTL-bysQTLapaQTL
  n <- c(byeQTL,bysQTL,byapaQTL,bysQTLapaQTL,unspecified)
  prop <- round(n/nrow(data),3)
  labels = c("by eQTL","by sQTL","by apaQTL","by sQTL+apaQTL","unspecified")
  lab.ypos = cumsum(prop) - 0.5*prop
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5))+
    scale_fill_manual(values = palette) +
    theme_void()
  #pie(Prop , labels = c("by eQTL","by sQTL","by apaQTL","by sQTL+apaQTL","unspecified"), col=palette)
}



draw_gene_piechart_tissue <- function(data){
  
  data <- data[data$combined_pip>0.8,]
  tissues <- colnames(data)[3:7]
  colnames(data)[3:7] <- paste0("tissue",c(1:5))
  
  
  bytissue1 <- nrow(data[data$tissue1/data$combined_pip>0.8,])
  bytissue2 <- nrow(data[data$tissue2/data$combined_pip>0.8,])
  bytissue3 <- nrow(data[data$tissue3/data$combined_pip>0.8,])
  bytissue4 <- nrow(data[data$tissue4/data$combined_pip>0.8,])
  bytissue5 <- nrow(data[data$tissue5/data$combined_pip>0.8,])
  unspecified <- nrow(data)-bytissue1-bytissue2-bytissue3-bytissue4-bytissue5
  
  n <- c(bytissue1,bytissue2,bytissue3,bytissue4,bytissue5,unspecified)
  
  prop <- round(n/nrow(data),3)
  labels = c(tissues,"unspecified")
  lab.ypos = cumsum(prop) - 0.5*prop
  df <- data.frame("n" = n,
                   "class" = labels,
                   "prop" = prop,
                   "lab.ypos" = lab.ypos)
  p <- ggplot(df, aes(x = "", y = prop, fill = class)) +
    geom_bar(width = 1, stat = "identity", color = "white") +
    coord_polar("y", start = 0)+
    geom_text(aes(label = n),
              position = position_stack(vjust = 0.5))+
    scale_fill_manual(values = palette) +
    theme_void()
  
  print(p)
  print(tissues)

}

sum_pve_across_types <- function(ctwas_parameters) {
  # Round the group_pve values
  pve <- round(ctwas_parameters$group_pve, 4)
  pve <- as.data.frame(pve)

  # Extract SNP PVE for later use
  SNP_pve <- pve["SNP", ]

  # Add type and context columns
  pve$type <- sapply(rownames(pve), function(x) { unlist(strsplit(x, "[|]"))[1] })
  pve$context <- sapply(rownames(pve), function(x) { unlist(strsplit(x, "[|]"))[2] })

  # Remove rows with NA values and sort
  pve <- na.omit(pve)
  pve <- pve[order(rownames(pve)), ]

  # Aggregate PVE by type
  df_pve <- aggregate(pve$pve, by = list(pve$type), FUN = sum)
  colnames(df_pve) <- c("type", "total_pve")
  df_pve$total_pve <- round(df_pve$total_pve, 4)

  # Add context-specific columns
  for (context in unique(pve$context)) {
    context_pve <- aggregate(pve$pve, by = list(pve$type, pve$context), FUN = sum)
    context_pve <- context_pve[context_pve$Group.2 == context, ]
    colnames(context_pve)[3] <- context
    df_pve <- merge(df_pve, context_pve[, c("Group.1", context)], by.x = "type", by.y = "Group.1", all.x = TRUE)
  }

  # Insert SNP PVE
  SNP_row <- c("SNP", SNP_pve, rep(0, ncol(df_pve) - 2))
  df_pve <- rbind(df_pve, SNP_row)

  # Convert to numeric except for the type column
  df_pve[, -1] <- lapply(df_pve[, -1], as.numeric)

  # Sum all rows and add a sum_pve row
  sum_row <- colSums(df_pve[, -1], na.rm = TRUE)
  sum_row <- c("total_pve", sum_row)
  df_pve <- rbind(df_pve, sum_row)

  # Clean up row names and return
  row.names(df_pve) <- NULL
  return(df_pve)
}

load("/project2/xinhe/shared_data/multigroup_ctwas/gwas/samplesize.rdata")

pve_pie_chart <- function(pve_vector, palette=NULL, title) {
  # Create data frame for plotting
  data <- data.frame(
    Group = names(pve_vector),
    Value = pve_vector
  )
  
  # Calculate percentages
  data$Percentage <- round(100 * data$Value / sum(data$Value), 1)
  
  # Set palette if not specified
  if (is.null(palette)) {
    palette <- brewer.pal(min(8, length(data$Group)), "Set3")
  }
  
  # Create pie chart
  ggplot(data, aes(x = "", y = Value, fill = Group)) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +  # This transforms the bar chart into a pie chart
    scale_fill_manual(values = palette, name = "Group") +
    labs(title = title, x = NULL, y = NULL) +
    theme_void() +
    theme(plot.title = element_text(hjust = 0.5), 
          legend.position = "right", 
          legend.title = element_text(face = "bold"), 
          legend.text = element_text(size = 12)) +
    geom_text(aes(label = paste(Percentage, "%", sep="")), position = position_stack(vjust = 0.5))
}

sum_pip_across_contexts_nocs <- function(finemap_res, mapping_data){

  #finemap_res <- finemap_res[finemap_res$cs_index!=0,]
  finemap_gene_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_gene_res$gene <- sapply(finemap_gene_res$id, function(x){unlist(strsplit(x,"[|]"))[1]})
  finemap_gene_res <- finemap_gene_res %>% plyr::join(mapping_data, by = "gene") %>% dplyr::select(-gene) %>% na.omit()
  finemap_gene_res <- finemap_gene_res %>% group_by(id) %>% mutate(susie_pip_adjusted = ifelse(n() > 1, susie_pip / n(), susie_pip)) %>% ungroup() %>% dplyr::select(-susie_pip) %>% dplyr::rename(susie_pip = susie_pip_adjusted)

  df_gene <- aggregate(finemap_gene_res$susie_pip, by=list(finemap_gene_res$genename), FUN=sum)
  colnames(df_gene) <- c("genename","combined_pip")
  return(df_gene)
}

```

# Results

**Results from multi-group analysis**

The results are summarized by 

1. Heritability contribution by contexts: we aggregate the PVE values by omics and tissues, making it easier to understand the distribution of PVE across different genetic contexts.

2. Combined PIP by omics:  we aggregate the Susie PIPs by omics 

3. Combined PIP by contexts: we aggregate the Susie PIPs by tissues, making it easier to understand the distribution of PIP across different genetic contexts.

4. Specific molecular traits of top genes: we creates a pie chart to visualize the proportion of genes classified into different categories based on their PIPs contributed by each genetics contexts. The categories are based on the proportion of each QTL type relative to the combined PIP value: 

- by eQTL: Number of genes where the ratio of eQTL to combined PIP is greater than 0.8.
- by sQTL: Number of genes where the ratio of sQTL to combined PIP is greater than 0.8.
- by apaQTL: Number of genes where the ratio of apaQTL to combined PIP is greater than 0.8.
- by sQTL+apaQTL: Number of genes where the combined ratio of apaQTL and sQTL to combined PIP is greater than 0.8, but neither apaQTL nor sQTL individually exceed 0.8.
- unspecified: Number of genes not classified into any of the above categories.

**Comparing with earlier multi-group analysis results**

We compared number of significant genes, overlapping genes. The earlier results are here: https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ukbb.html


## aFib

TO DO

## IBD

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "IBD-ebi-a-GCST004131"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)


file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".ctwas.res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)

load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

```

### Comparing with earlier weights 

#### overlaps

```{r echo=FALSE}

siggene_current <- combined_pip_by_contexts[combined_pip_by_contexts$combined_pip > 0.8,]

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res_earlier <- readRDS(file_res)
combined_pip_by_contexts_earlier <- sum_pip_across_contexts(finemap_res_earlier,E_S_A_mapping_earlier)
siggene_earlier <- combined_pip_by_contexts_earlier[combined_pip_by_contexts_earlier$combined_pip > 0.8,]

sprintf("current # of sig. genes = %s", nrow(siggene_current))
sprintf("earlier # of sig. genes = %s", nrow(siggene_earlier))
sprintf("# of overlap between them = %s", sum(siggene_current$genename %in% siggene_earlier$genename))
```

#### Why earlier genes were not reported in current settings

In the table below, NA means: current weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in current setting, this gene is not included in CS. `combined_pip_current_weights < 0.8` means in current setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_current <- sum_pip_across_contexts_nocs(finemap_res,E_S_A_mapping)

siggene_earlier_total <- siggene_earlier[,c("genename","combined_pip")]
notoverlapped_earlier <- siggene_earlier_total[!siggene_earlier_total$genename%in%siggene_current$genename,]
gene_notsig_current <- merge(notoverlapped_earlier,genepip_current, by="genename", all.x = T)
colnames(gene_notsig_current) <- c("genename","combined_pip_earlier_weights","combined_pip_current_weights")

DT::datatable(gene_notsig_current,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Earlier sig genes but not reported in current setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_current$combined_pip_current_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_current$combined_pip_current_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_current$combined_pip_current_weights < 0.8),na.rm = T))
```


#### Why current genes were not reported in earlier settings

In the table below, NA means: earlier weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in earlier setting, this gene is not included in CS. `combined_pip_earlier_weights < 0.8` means in earlier setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_earlier <- sum_pip_across_contexts_nocs(finemap_res_earlier,E_S_A_mapping_earlier)

siggene_current_total <- siggene_current[,c("genename","combined_pip")]
notoverlapped_current <- siggene_current_total[!siggene_current_total$genename%in%siggene_earlier$genename,]
gene_notsig_earlier <- merge(notoverlapped_current,genepip_earlier, by="genename", all.x = T)
colnames(gene_notsig_earlier) <- c("genename","combined_pip_current_weights","combined_pip_earlier_weights")

DT::datatable(gene_notsig_earlier,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','current sig genes but not reported in earlier setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_earlier$combined_pip_earlier_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights < 0.8),na.rm = T))
```


## LDL

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "LDL-ukb-d-30780_irnt"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".ctwas.res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()
```

### Comparing with earlier weights 

#### overlaps

```{r echo=FALSE}

siggene_current <- combined_pip_by_contexts[combined_pip_by_contexts$combined_pip > 0.8,]

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res_earlier <- readRDS(file_res)
combined_pip_by_contexts_earlier <- sum_pip_across_contexts(finemap_res_earlier,E_S_A_mapping_earlier)
siggene_earlier <- combined_pip_by_contexts_earlier[combined_pip_by_contexts_earlier$combined_pip > 0.8,]

sprintf("current # of sig. genes = %s", nrow(siggene_current))
sprintf("earlier # of sig. genes = %s", nrow(siggene_earlier))
sprintf("# of overlap between them = %s", sum(siggene_current$genename %in% siggene_earlier$genename))
```

#### Why earlier genes were not reported in current settings

In the table below, NA means: current weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in current setting, this gene is not included in CS. `combined_pip_current_weights < 0.8` means in current setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_current <- sum_pip_across_contexts_nocs(finemap_res,E_S_A_mapping)

siggene_earlier_total <- siggene_earlier[,c("genename","combined_pip")]
notoverlapped_earlier <- siggene_earlier_total[!siggene_earlier_total$genename%in%siggene_current$genename,]
gene_notsig_current <- merge(notoverlapped_earlier,genepip_current, by="genename", all.x = T)
colnames(gene_notsig_current) <- c("genename","combined_pip_earlier_weights","combined_pip_current_weights")

DT::datatable(gene_notsig_current,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Earlier sig genes but not reported in current setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_current$combined_pip_current_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_current$combined_pip_current_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_current$combined_pip_current_weights < 0.8),na.rm = T))
```


#### Why current genes were not reported in earlier settings

In the table below, NA means: earlier weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in earlier setting, this gene is not included in CS. `combined_pip_earlier_weights < 0.8` means in earlier setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_earlier <- sum_pip_across_contexts_nocs(finemap_res_earlier,E_S_A_mapping_earlier)

siggene_current_total <- siggene_current[,c("genename","combined_pip")]
notoverlapped_current <- siggene_current_total[!siggene_current_total$genename%in%siggene_earlier$genename,]
gene_notsig_earlier <- merge(notoverlapped_current,genepip_earlier, by="genename", all.x = T)
colnames(gene_notsig_earlier) <- c("genename","combined_pip_current_weights","combined_pip_earlier_weights")

DT::datatable(gene_notsig_earlier,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','current sig genes but not reported in earlier setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_earlier$combined_pip_earlier_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights < 0.8),na.rm = T))
```


## SBP

### Results from multi-group analysis

```{r echo=FALSE, message=FALSE}
trait <- "SBP-ukb-a-360"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".ctwas.res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

finemap_gene_res_highpip$p <- 10^(-finemap_gene_res_highpip$log10p)
DT::datatable(finemap_gene_res_highpip[finemap_gene_res_highpip$gene_name == "NAA60",],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','The one with high PIP but also high TWAS p'),options = list(pageLength = 5) )
load("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/naa60.rdata")
DT::datatable(naa60,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','All signals for NAA60'),options = list(pageLength = 5) )

```

### Comparing with earlier weights 

#### overlaps

```{r echo=FALSE}

siggene_current <- combined_pip_by_contexts[combined_pip_by_contexts$combined_pip > 0.8,]

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res_earlier <- readRDS(file_res)
combined_pip_by_contexts_earlier <- sum_pip_across_contexts(finemap_res_earlier,E_S_A_mapping_earlier)
siggene_earlier <- combined_pip_by_contexts_earlier[combined_pip_by_contexts_earlier$combined_pip > 0.8,]

sprintf("current # of sig. genes = %s", nrow(siggene_current))
sprintf("earlier # of sig. genes = %s", nrow(siggene_earlier))
sprintf("# of overlap between them = %s", sum(siggene_current$genename %in% siggene_earlier$genename))
```

#### Why earlier genes were not reported in current settings

In the table below, NA means: current weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in current setting, this gene is not included in CS. `combined_pip_current_weights < 0.8` means in current setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_current <- sum_pip_across_contexts_nocs(finemap_res,E_S_A_mapping)

siggene_earlier_total <- siggene_earlier[,c("genename","combined_pip")]
notoverlapped_earlier <- siggene_earlier_total[!siggene_earlier_total$genename%in%siggene_current$genename,]
gene_notsig_current <- merge(notoverlapped_earlier,genepip_current, by="genename", all.x = T)
colnames(gene_notsig_current) <- c("genename","combined_pip_earlier_weights","combined_pip_current_weights")

DT::datatable(gene_notsig_current,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Earlier sig genes but not reported in current setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_current$combined_pip_current_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_current$combined_pip_current_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_current$combined_pip_current_weights < 0.8),na.rm = T))
```


#### Why current genes were not reported in earlier settings

In the table below, NA means: earlier weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in earlier setting, this gene is not included in CS. `combined_pip_earlier_weights < 0.8` means in earlier setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_earlier <- sum_pip_across_contexts_nocs(finemap_res_earlier,E_S_A_mapping_earlier)

siggene_current_total <- siggene_current[,c("genename","combined_pip")]
notoverlapped_current <- siggene_current_total[!siggene_current_total$genename%in%siggene_earlier$genename,]
gene_notsig_earlier <- merge(notoverlapped_current,genepip_earlier, by="genename", all.x = T)
colnames(gene_notsig_earlier) <- c("genename","combined_pip_current_weights","combined_pip_earlier_weights")

DT::datatable(gene_notsig_earlier,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','current sig genes but not reported in earlier setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_earlier$combined_pip_earlier_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights < 0.8),na.rm = T))
```


## SCZ

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "SCZ-ieu-b-5102"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".ctwas.res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()
```

### Comparing with earlier weights 

#### overlaps

```{r echo=FALSE}

siggene_current <- combined_pip_by_contexts[combined_pip_by_contexts$combined_pip > 0.8,]

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res_earlier <- readRDS(file_res)
combined_pip_by_contexts_earlier <- sum_pip_across_contexts(finemap_res_earlier,E_S_A_mapping_earlier)
siggene_earlier <- combined_pip_by_contexts_earlier[combined_pip_by_contexts_earlier$combined_pip > 0.8,]

sprintf("current # of sig. genes = %s", nrow(siggene_current))
sprintf("earlier # of sig. genes = %s", nrow(siggene_earlier))
sprintf("# of overlap between them = %s", sum(siggene_current$genename %in% siggene_earlier$genename))
```

#### Why earlier genes were not reported in current settings

In the table below, NA means: current weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in current setting, this gene is not included in CS. `combined_pip_current_weights < 0.8` means in current setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_current <- sum_pip_across_contexts_nocs(finemap_res,E_S_A_mapping)

siggene_earlier_total <- siggene_earlier[,c("genename","combined_pip")]
notoverlapped_earlier <- siggene_earlier_total[!siggene_earlier_total$genename%in%siggene_current$genename,]
gene_notsig_current <- merge(notoverlapped_earlier,genepip_current, by="genename", all.x = T)
colnames(gene_notsig_current) <- c("genename","combined_pip_earlier_weights","combined_pip_current_weights")

DT::datatable(gene_notsig_current,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Earlier sig genes but not reported in current setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_current$combined_pip_current_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_current$combined_pip_current_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_current$combined_pip_current_weights < 0.8),na.rm = T))
```


#### Why current genes were not reported in earlier settings

In the table below, NA means: earlier weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in earlier setting, this gene is not included in CS. `combined_pip_earlier_weights < 0.8` means in earlier setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_earlier <- sum_pip_across_contexts_nocs(finemap_res_earlier,E_S_A_mapping_earlier)

siggene_current_total <- siggene_current[,c("genename","combined_pip")]
notoverlapped_current <- siggene_current_total[!siggene_current_total$genename%in%siggene_earlier$genename,]
gene_notsig_earlier <- merge(notoverlapped_current,genepip_earlier, by="genename", all.x = T)
colnames(gene_notsig_earlier) <- c("genename","combined_pip_current_weights","combined_pip_earlier_weights")

DT::datatable(gene_notsig_earlier,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','current sig genes but not reported in earlier setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_earlier$combined_pip_earlier_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights < 0.8),na.rm = T))
```


## WBC

### Results from multi-group analysis

```{r echo=FALSE}
trait <- "WBC-ieu-b-30"
gwas_n <- samplesize[trait]

file_para <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".param.RDS")
param <- readRDS(file_para)
ctwas_parameters <- summarize_param(param, gwas_n)

group_size <- data.frame(group = names(ctwas_parameters$group_size),
                         group_size = as.vector(ctwas_parameters$group_size))
group_size <- t(group_size)
DT::datatable(group_size,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Group size'),options = list(pageLength = 5) )


para <- sum_pve_across_types(ctwas_parameters)
DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Heritability contribution by contexts'),options = list(pageLength = 5) )

pve_vector <- as.numeric(para$total_pve[1:4])
names(pve_vector) <- para$type[1:4]
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Types", palette)

pve_vector <- as.numeric(para[5,3:7])
pve_vector <- c(as.numeric(para[5,2])-sum(pve_vector),pve_vector)
names(pve_vector) <- c("1SNP",colnames(para)[3:7])
pve_pie_chart(pve_vector, title = "Pie Chart of PVE across Tissues", palette)

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/6.multi_group_munro/results/",trait,"/",trait,".ctwas.res.RDS")
finemap_res <- readRDS(file_res)
combined_pip_by_types <- sum_pip_across_types(finemap_res,E_S_A_mapping)
combined_pip_by_contexts <- sum_pip_across_contexts(finemap_res,E_S_A_mapping)

DT::datatable(combined_pip_by_types[combined_pip_by_types$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by omics'),options = list(pageLength = 5) )

DT::datatable(combined_pip_by_contexts[combined_pip_by_contexts$combined_pip>0.8,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Combined PIP by tissue'),options = list(pageLength = 5) )

draw_gene_piechart(combined_pip_by_types)
draw_gene_piechart_tissue(combined_pip_by_contexts)
load(paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/analy_results/compare_pip_pval_",trait,".rdata"))

#ggplot(finemap_gene_res_highpip, aes(x = log10p, y = susie_pip)) +
ggplot(finemap_gene_res_highpip, aes(x = 10^(-log10p), y = susie_pip)) +
  geom_point() +
  labs(x = "P-values", y = "SuSiE PIP", title = "Scatter plot of imputed TWAS P-value vs SuSiE PIP (CS filtered, PIP>0.5 included)") +
  theme_minimal()

finemap_gene_res_highpip$p <- 10^(-finemap_gene_res_highpip$log10p)

gene_problem <- c("CCZ1","ELMO1","ING5")

DT::datatable(finemap_gene_res_highpip[finemap_gene_res_highpip$gene_name %in% gene_problem,],caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','The one with high PIP but also high TWAS p'),options = list(pageLength = 5) )

```

### Comparing with earlier weights 

#### overlaps

```{r echo=FALSE}

siggene_current <- combined_pip_by_contexts[combined_pip_by_contexts$combined_pip > 0.8,]

file_res <- paste0("/project/xinhe/xsun/multi_group_ctwas/5.multi_group_testing/results_ukbb/",trait,"/",trait,".finemap_res.RDS")
finemap_res_earlier <- readRDS(file_res)
combined_pip_by_contexts_earlier <- sum_pip_across_contexts(finemap_res_earlier,E_S_A_mapping_earlier)
siggene_earlier <- combined_pip_by_contexts_earlier[combined_pip_by_contexts_earlier$combined_pip > 0.8,]

sprintf("current # of sig. genes = %s", nrow(siggene_current))
sprintf("earlier # of sig. genes = %s", nrow(siggene_earlier))
sprintf("# of overlap between them = %s", sum(siggene_current$genename %in% siggene_earlier$genename))
```

#### Why earlier genes were not reported in current settings

In the table below, NA means: current weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in current setting, this gene is not included in CS. `combined_pip_current_weights < 0.8` means in current setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_current <- sum_pip_across_contexts_nocs(finemap_res,E_S_A_mapping)

siggene_earlier_total <- siggene_earlier[,c("genename","combined_pip")]
notoverlapped_earlier <- siggene_earlier_total[!siggene_earlier_total$genename%in%siggene_current$genename,]
gene_notsig_current <- merge(notoverlapped_earlier,genepip_current, by="genename", all.x = T)
colnames(gene_notsig_current) <- c("genename","combined_pip_earlier_weights","combined_pip_current_weights")

DT::datatable(gene_notsig_current,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','Earlier sig genes but not reported in current setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_current$combined_pip_current_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_current$combined_pip_current_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_current$combined_pip_current_weights < 0.8),na.rm = T))
```


#### Why current genes were not reported in earlier settings

In the table below, NA means: earlier weights do not have this gene. If `combined_pip_current_weights >0.8`, it means in earlier setting, this gene is not included in CS. `combined_pip_earlier_weights < 0.8` means in earlier setting, the gene is filtered out by either credible set or combined_pip.

```{r echo=FALSE}
genepip_earlier <- sum_pip_across_contexts_nocs(finemap_res_earlier,E_S_A_mapping_earlier)

siggene_current_total <- siggene_current[,c("genename","combined_pip")]
notoverlapped_current <- siggene_current_total[!siggene_current_total$genename%in%siggene_earlier$genename,]
gene_notsig_earlier <- merge(notoverlapped_current,genepip_earlier, by="genename", all.x = T)
colnames(gene_notsig_earlier) <- c("genename","combined_pip_current_weights","combined_pip_earlier_weights")

DT::datatable(gene_notsig_earlier,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','current sig genes but not reported in earlier setting'),options = list(pageLength = 5) )

sprintf("%s were not included (NAs) because they are not in the weight files", sum(is.na(gene_notsig_earlier$combined_pip_earlier_weights)))

sprintf("%s were not included (combined_pip > 0.8) because they are not in CS", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights >0.8),na.rm = T))

sprintf("%s were not included (combined_pip < 0.8) because of the low combined PIPs", sum(as.numeric(gene_notsig_earlier$combined_pip_earlier_weights < 0.8),na.rm = T))
```

