---
title: "6 Traits, 5 tissues, eQTL + sQTL + stQTL â€“ compute ukbb LD, eqtl, sqtl from predictdb; stQTL from Munro et al; post-processing"
author: "XSun"
date: "2024-12-03"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Introduction

We compare the post-processed results with the original results here: https://sq-96.github.io/multigroup_ctwas_analysis/multi_group_6traits_15weights_ess.html


```{r message=FALSE, warning=FALSE}
library(ctwas)
library(EnsDb.Hsapiens.v86)
library(ggplot2)
library(gridExtra)
library(dplyr)

ens_db <- EnsDb.Hsapiens.v86

mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
mapping_munro <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
mapping_two <- rbind(mapping_predictdb,mapping_munro)


compute_pip_per_cs <- function(combined_data, susie_data) {
  # Initialize an empty list to store results
  details <- list()
  
  # Iterate over each unique gene name in the combined data
  unique_genes <- unique(combined_data$gene_name)
  
  for (genename in unique_genes) {
    # dplyr::filter susie data for the current gene
    susie_alpha_res_multi_per_gene <- susie_data %>%
      dplyr::filter(gene_name == genename)
    
    # Get all unique credible sets for the current gene
    cs_all <- unique(susie_alpha_res_multi_per_gene$susie_set[susie_alpha_res_multi_per_gene$in_cs])
    
    if (length(cs_all) > 1) {
      # dplyr::filter complete cases and those in credible sets
      susie_alpha_res_multi_per_gene <- susie_alpha_res_multi_per_gene %>%
        dplyr::filter(complete.cases(cs), in_cs)
      
      # Summarize the data
      summed_alpha_with_details <- susie_alpha_res_multi_per_gene %>%
        group_by(susie_set) %>%
        summarise(
          total_susie_alpha = round(sum(susie_alpha, na.rm = TRUE), digits = 3),
          num_molecular_traits = n(),
          ids_pip = paste0(id, "(", round(susie_alpha, digits = 3), ")", collapse = ", ")
        )
      
      # Add gene name to the summarized data
      summed_alpha_with_details$gene_name <- genename
      
      # Append the result to the details list
      details[[length(details) + 1]] <- summed_alpha_with_details
    }
  }
  
  # Combine all results into a single data frame
  final_details <- bind_rows(details)
  
  if(nrow(final_details) > 0){
    final_details <- final_details[,c("gene_name","susie_set","total_susie_alpha","num_molecular_traits","ids_pip")]
    colnames(final_details) <- c("gene_name","CS","total_PIP_CS","num_molecular_traits_CS","ids_pip_CS")
  }
  
  
  return(final_details)
}

```

# aFib-ebi-a-GCST006414

```{r message=FALSE, warning=FALSE}
trait <- "aFib-ebi-a-GCST006414"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch 

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed") 

DT::datatable(merge_regionmerge_nold,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','PIP after region merge and fixed PIP for problematic genes'),options = list(pageLength = 5) )

```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```

## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}

```




# LDL-ukb-d-30780_irnt

```{r message=FALSE, warning=FALSE}
trait <- "LDL-ukb-d-30780_irnt"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch 

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed") 

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}
```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```


## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )

```


# IBD-ebi-a-GCST004131

```{r message=FALSE, warning=FALSE}
trait <- "IBD-ebi-a-GCST004131"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch 

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed") 

DT::datatable(merge_regionmerge_nold,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','PIP after region merge and fixed PIP for problematic genes'),options = list(pageLength = 5) )

```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```


## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}

```
# SBP-ukb-a-360

```{r message=FALSE, warning=FALSE}
trait <- "SBP-ukb-a-360"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch 

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed") 

DT::datatable(merge_regionmerge_nold,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','PIP after region merge and fixed PIP for problematic genes'),options = list(pageLength = 5) )

```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```


## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}

```

# SCZ-ieu-b-5102

```{r message=FALSE, warning=FALSE}
trait <- "SCZ-ieu-b-5102"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch 

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed") 

DT::datatable(merge_regionmerge_nold,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','PIP after region merge and fixed PIP for problematic genes'),options = list(pageLength = 5) )

```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) + 
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld, 
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```


## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}

```

# WBC-ieu-b-30

```{r message=FALSE, warning=FALSE}
trait <- "WBC-ieu-b-30"

results_dir_origin <- paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/results/",trait,"/")
ctwas_res_origin <- readRDS(paste0(results_dir_origin,trait,".ctwas.res.RDS"))

finemap_res_origin <- ctwas_res_origin$finemap_res
```

## Region merge

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rm_",trait,".rdata"))

finemap_res_rm <- res_regionmerge$finemap_res
finemap_res_rm_boundary_genes <- finemap_res_rm[finemap_res_rm$id %in%selected_boundary_genes$id,]
finemap_res_rm_boundary_genes_pip <- finemap_res_rm_boundary_genes[,c("id","susie_pip","cs")]


finemap_res_origin_boundary_genes <- finemap_res_origin[finemap_res_origin$id %in%selected_boundary_genes$id,]
finemap_res_origin_boundary_genes_pip <- finemap_res_origin_boundary_genes[,c("id","susie_pip","cs")]

finemap_res_compare_regionmerge <- merge(finemap_res_origin_boundary_genes_pip,finemap_res_rm_boundary_genes_pip, by = "id")
colnames(finemap_res_compare_regionmerge) <- c("id","susie_pip_origin","cs_origin","susie_pip_reginmerge","cs_reginmerge")

DT::datatable(finemap_res_compare_regionmerge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Selected boundary genes (susie_pip > 0.5)'),options = list(pageLength = 10) )
```

## LD-mismatch

```{r message=FALSE, warning=FALSE}

load(paste0("/project/xinhe/xsun/multi_group_ctwas/11.multi_group_1008/post_process_rm_ld/rmld_",trait,".rdata"))

sprintf("The number of problematic regions = %s", length(problematic_region_ids))
sprintf("The number of problematic genes = %s", length(problematic_genes))
sprintf("The number of problematic snps = %s", length(res_ldmismatch$problematic_snps))
sprintf("The number of flipped snps = %s", length(res_ldmismatch$flipped_snps))

finemap_rm_res_problematic_region <- finemap_res_rm[finemap_res_rm$id %in% problematic_genes,]

finemap_noLD_res_problematic_region <- res_rmld$finemap_res
merge_regionmerge_nold <- merge(finemap_rm_res_problematic_region,finemap_noLD_res_problematic_region, by = "id")
merge_regionmerge_nold <- merge_regionmerge_nold[,c("id","susie_pip.x","susie_pip.y")]
colnames(merge_regionmerge_nold) <-  c("id","susie_pip_after_regionmerge","susie_pip_ld-mismatch-fixed")

DT::datatable(merge_regionmerge_nold,caption = htmltools::tags$caption(style = 'caption-side: topleft; text-align = left; color:black;','PIP after region merge and fixed PIP for problematic genes'),options = list(pageLength = 5) )

```



```{r message=FALSE, warning=FALSE, fig.width=10, fig.height=3}

finemap_res_origin <- ctwas_res_origin$finemap_res
finemap_res_origin_gene <- finemap_res_origin[finemap_res_origin$type != "SNP",]

p1 <- ggplot(data = finemap_res_origin_gene, aes(x= abs(z), y= susie_pip)) +
  geom_point() +
  ggtitle("Original ctwas results") +
  theme_minimal()


finemap_res_rm_gene <- finemap_res_rm[finemap_res_rm$type != "SNP",]

p2 <- ggplot(data = finemap_res_rm_gene, aes(x= abs(z), y= susie_pip)) +
  geom_point() +
  ggtitle("After region merge") +
  theme_minimal()


finemap_noLD_res_problematic_region_gene <- finemap_noLD_res_problematic_region[finemap_noLD_res_problematic_region$type !="SNP",]

p3 <- ggplot(data = finemap_noLD_res_problematic_region_gene, aes(x= abs(z), y= susie_pip)) +
  geom_point() +
  ggtitle("After LD mismatch fixed") +
  theme_minimal()


grid.arrange(p1,p2,p3, ncol = 3)
```


## Comparing results without post processing

```{r message=FALSE, warning=FALSE}

susie_alpha_res_origin <- ctwas_res_origin$susie_alpha_res
susie_alpha_res_origin <- anno_susie_alpha_res(susie_alpha_res_origin,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_origin <- combine_gene_pips(susie_alpha_res_origin,
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_origin <- combined_pip_by_type_cs_origin[combined_pip_by_type_cs_origin$combined_pip > 0.8,]

susie_alpha_res_rm <- res_regionmerge$susie_alpha_res
susie_alpha_res_rm <- anno_susie_alpha_res(susie_alpha_res_rm,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rm <- combine_gene_pips(susie_alpha_res_rm,
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rm <- combined_pip_by_type_cs_rm[combined_pip_by_type_cs_rm$combined_pip > 0.8,]



susie_alpha_res_rmld <- res_rmld$susie_alpha_res
susie_alpha_res_rmld <- anno_susie_alpha_res(susie_alpha_res_rmld,
                                        mapping_table = mapping_two,
                                        map_by = "molecular_id",
                                        drop_unmapped = TRUE)

combined_pip_by_type_cs_rmld <- combine_gene_pips(susie_alpha_res_rmld,
                                             group_by = "gene_name",
                                             by = "type",
                                             method = "combine_cs",
                                             filter_cs = TRUE,
                                             include_cs_id = T)
combined_pip_by_type_sig_rmld <- combined_pip_by_type_cs_rmld[combined_pip_by_type_cs_rmld$combined_pip > 0.8,]


origin_unique <- combined_pip_by_type_sig_origin[!combined_pip_by_type_sig_origin$gene_name %in% combined_pip_by_type_sig_rmld$gene_name,]

origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rm, by = "gene_name", all.x = TRUE)

# Then merge the result with combined_pip_by_type_cs_rmld
origin_unique <- merge(origin_unique, combined_pip_by_type_cs_rmld, by = "gene_name", all.x = TRUE)
origin_unique_show <- origin_unique[,c("gene_name","combined_pip.x","combined_pip.y","combined_pip","eQTL_pip.x","eQTL_pip.y","eQTL_pip","sQTL_pip.x","sQTL_pip.y","sQTL_pip","stQTL_pip.x","stQTL_pip.x","stQTL_pip","combined_cs_id.x","combined_cs_id.y","combined_cs_id")]

colnames(origin_unique_show) <- c("gene_name","combined_pip_origin","combined_pip_rm","combined_pip_rmld","eQTL_pip_origin","eQTL_pip_rm","eQTL_pip_rmld","sQTL_pip_origin","sQTL_pip_rm","sQTL_pip_rmld","stQTL_pip_origin","stQTL_pip_origin","stQTL_pip_rmld","combined_cs_id_origin","combined_cs_id_rm","combined_cs_id_rmld")

DT::datatable(origin_unique_show,caption = htmltools::tags$caption(style = 'caption-side: top; text-align: left; color: black;','Genes no longer significant after post processing'),options = list(pageLength = 5))

```

## Exploring allelic heterogeneity

```{r message=FALSE, warning=FALSE}

pip_per_cs <- compute_pip_per_cs(combined_pip_by_type_sig_rmld, susie_alpha_res_rmld)

sprintf("Number of genes having allelic heterogeneity = %s",length(unique(pip_per_cs$gene_name)))

if(nrow(pip_per_cs) > 0) {
  DT::datatable(pip_per_cs,caption = htmltools::tags$caption( style = 'caption-side: topleft; text-align = left; color:black;','PIP per CS'),options = list(pageLength = 5) )
}

```

