---
title: "parameters_apa_li"
author: "XSun"
date: "2025-04-09"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Introduction

We estimated the parameters for the e+s+apa model in this analysis. The apa component follows the approach described in this study https://www.nature.com/articles/s41588-021-00864-5. For each gene, we used the lead QTL to construct a PredictDB model.



```{r warning=F,message=F}
library(ctwas)
library(ggplot2)
library(tidyverse)
library(dplyr)
library(EnsDb.Hsapiens.v86)
ens_db <- EnsDb.Hsapiens.v86

source("/project/xinhe/xsun/multi_group_ctwas/data/samplesize.R")
source("/project/xinhe/xsun/multi_group_ctwas/functions/0.functions.R")

folder_results_susieST <- "/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/snakemake_outputs/"
folder_results_apaonly <- "/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/snakemake_outputs_apaonly/"
folder_results_single <- "/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/ctwas_output/apa/"


folder_results_susieST_susie <- "/project/xinhe/xsun/multi_group_ctwas/15.susie_weights/snakemake_outputs/"
folder_results_apaonly_susie <- "/project/xinhe/xsun/multi_group_ctwas/15.susie_weights/snakemake_outputs_marginaltissue/"
folder_results_single_susie <- "/project/xinhe/xsun/multi_group_ctwas/17.single_eQTL/ctwas_output/stability_weight_unscaled/"

# mapping_predictdb <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
# mapping_munro <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/Munro_mapping.RDS")
# mapping_two <- rbind(mapping_predictdb,mapping_munro)

colors <- c("#ff7f0e", "#2ca02c", "#d62728",  "#9467bd", "#8c564b", "#e377c2", "#7f7f7f",  "#bcbd22",  "#17becf",  "#f7b6d2",  "#c5b0d5",  "#9edae5", "#ffbb78",  "#98df8a",  "#ff9896" )

top_tissues <- c("Liver","Whole_Blood","Brain_Cerebellar_Hemisphere","Adipose_Subcutaneous","Brain_Cerebellum","Heart_Atrial_Appendage","Pituitary")
traits <- c("LDL-ukb-d-30780_irnt","IBD-ebi-a-GCST004131","BMI-panukb","RBC-panukb","SCZ-ieu-b-5102","aFib-ebi-a-GCST006414","T2D-panukb")
names(top_tissues) <- traits


plot_piechart <- function(ctwas_parameters, colors, by, title) {
  # Create the initial data frame
  data <- data.frame(
    category = names(ctwas_parameters$prop_heritability),
    percentage = ctwas_parameters$prop_heritability
  )
  
  # Split the category into context and type
  data <- data %>%
    mutate(
      context = sub("\\|.*", "", category),
      type = sub(".*\\|", "", category)
    )
  
  # Aggregate the data based on the 'by' parameter
  if (by == "type") {
    data <- data %>%
      group_by(type) %>%
      summarize(percentage = sum(percentage)) %>%
      mutate(category = type)  # Use type as the new category
  } else if (by == "context") {
    data <- data %>%
      group_by(context) %>%
      summarize(percentage = sum(percentage)) %>%
      mutate(category = context)  # Use context as the new category
  } else {
    stop("Invalid 'by' parameter. Use 'type' or 'context'.")
  }
  
  # Calculate percentage labels for the chart
  data$percentage_label <- paste0(round(data$percentage * 100, 1), "%")
  
  # Create the pie chart
  pie <- ggplot(data, aes(x = "", y = percentage, fill = category)) +
    geom_bar(stat = "identity", width = 1) +
    coord_polar("y", start = 0) +
    theme_void() +  # Remove background and axes
    geom_text(aes(label = percentage_label), 
              position = position_stack(vjust = 0.5), size = 3) +  # Adjust size as needed
    scale_fill_manual(values = colors) +  # Custom colors
    labs(fill = "Category") +  # Legend title
    ggtitle(title)  # Title
  
  return(pie)
}

plot_multi <- function(p1,p2,p3,title=NULL) {
  
  fix_panel_size <- function(plot, width = 2.1, height = 2) {
    set_panel_size(plot, width = unit(width, "in"), height = unit(height, "in"))
  }
  
  # Apply fixed panel size
  pie1 <- fix_panel_size(p1)
  pie2 <- fix_panel_size(p2)
  pie3 <- fix_panel_size(p3)
  
  # Compute natural widths
  widths <- unit.c(grobWidth(pie1), grobWidth(pie2), grobWidth(pie3))
  
  # Arrange
  p <- grid.arrange(pie1, pie2, pie3,
                    ncol = 3,
                    widths = widths,
                    top = title)
  
  return(p)
  
  
}

```

# apaQTL summary

```{r warning=F,message=F}

cis_files <- list.files(path = "/project2/xinhe/shared_data/multigroup_ctwas/weights/apa_li/",pattern = "cis.3aQTL.txt")

sum <- c()
for (file in cis_files){
  
  tissue <- gsub(pattern = ".cis.3aQTL.txt",replacement = "",x = file)
  
  cisdf <- data.table::fread(paste0("/project2/xinhe/shared_data/multigroup_ctwas/weights/apa_li/",file))
  
  cisdf$fdr <- p.adjust(as.numeric(cisdf$p.value), method = "fdr")
  cisdf_fdr005 <- cisdf[cisdf$fdr < 0.05,]
  count_df <- cisdf_fdr005[, .N, by = transcript]
  avg <- sum(count_df$N)/nrow(count_df)
  
  tmp <- c(tissue,avg,nrow(count_df))
  sum <- rbind(sum,tmp)
}

rownames(sum) <- NULL
colnames(sum) <- c("Tissue","avg_qtl_fdr005","num_gene")

DT::datatable(sum,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Average number of apaQTL per gene'),options = list(pageLength = 10) )
```



# LDL-ukb-d-30780_irnt

```{r warning=F,message=F}
trait <- "LDL-ukb-d-30780_irnt"
st <- "with_susieST"

gwas_n <- samplesize[trait]
```


## %h2g for setting: shared_all, thin = 1

```{r warning=F,message=F}
thin <- 1
var_struc <- "shared_all"
```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```



# IBD-ebi-a-GCST004131

```{r warning=F,message=F}
trait <- "IBD-ebi-a-GCST004131"
st <- "with_susieST"

gwas_n <- samplesize[trait]
```


## %h2g for setting: shared_all, thin = 1

```{r warning=F,message=F}
thin <- 1
var_struc <- "shared_all"
```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```

```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```





## gene level results -- whole blood


```{r warning=F,message=F, fig.height=6, fig.width=12}
results_li <- read.table("/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/data/IBD_genes_li.txt", header = T)

tissues_target <- c("Whole_Blood")
results_li_overlaptissue <- results_li[results_li$Tissue %in% tissues_target,]

ctwas_res <- readRDS("/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/ctwas_output/apa//IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Whole_Blood.thin1.shared_all.L5.finemap_regions_res.RDS")

mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/apa_li.RDS")

susie_alpha_res <- ctwas_res$susie_alpha_res
susie_alpha_res$molecular_id <- sub("\\|[^|]*$", "", susie_alpha_res$id)
susie_alpha_res <- anno_susie_alpha_res(susie_alpha_res,
                                        mapping_table = mapping_table,
                                        map_by = "molecular_id",
                                        drop_unmapped = F)

susie_alpha_res_uniq <- susie_alpha_res[!duplicated(susie_alpha_res$id),]

susie_summary <- susie_alpha_res_uniq %>%
  group_by(gene_name) %>%
  summarise(
    susie_z = mean(z, na.rm = TRUE),             # or median(z) if you prefer
    susie_pip = max(susie_pip, na.rm = TRUE),     # max to reflect strongest evidence
    region_id = region_id,
    id = id,
  )

# Step 2: Merge with results_li
results_li_merged <- results_li_overlaptissue %>%
  left_join(susie_summary, by = c("GeneName" = "gene_name"))

DT::datatable(results_li_merged,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',''),options = list(pageLength = 10) )


results_li_merged <- results_li_merged[complete.cases(results_li_merged$region_id),]


weights <- readRDS("/project/xinhe/xsun/multi_group_ctwas/16.apa_li_weights/ctwas_output/apa//IBD-ebi-a-GCST004131/IBD-ebi-a-GCST004131_Whole_Blood.preprocessed.weights.ST.RDS")
snp_map <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/LD_region_info/snp_map.RDS")


finemap_res <- ctwas_res$finemap_res
finemap_res$molecular_id <- get_molecular_ids(finemap_res)
finemap_res <- anno_finemap_res(finemap_res,
                                snp_map = snp_map,
                                mapping_table = mapping_table,
                                add_gene_annot = TRUE,
                                map_by = "molecular_id",
                                drop_unmapped = TRUE,
                                add_position = TRUE,
                                use_gene_pos = "mid")


for (i in 1:nrow(results_li_merged)) {
  
  region_id <- results_li_merged$region_id[i]
  
  p <- make_locusplot(finemap_res,
               region_id = region_id,
               ens_db = ens_db,
               weights = weights,
               highlight_pip = 0.8,
               filter_protein_coding_genes = TRUE,
               filter_cs = F,
               focal_id = results_li_merged$id[i],
               color_pval_by = "cs",
               color_pip_by = "cs")
  
  print(p)
  
}

```


# T2D-panukb

```{r warning=F,message=F}
trait <- "T2D-panukb"
st <- "with_susieST"

gwas_n <- samplesize[trait]
```


## %h2g for setting: shared_all, thin = 1

```{r warning=F,message=F}
thin <- 1
var_struc <- "shared_all"
```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```

```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```


# BMI-panukb

```{r warning=F,message=F}
trait <- "BMI-panukb"
st <- "with_susieST"

gwas_n <- samplesize[trait]
```


## %h2g for setting: shared_all, thin = 1

```{r warning=F,message=F}
thin <- 1
var_struc <- "shared_all"
```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```


# RBC-panukb

```{r warning=F,message=F}
trait <- "RBC-panukb"
st <- "with_susieST"

gwas_n <- samplesize[trait]
```


## %h2g for setting: shared_all, thin = 1

```{r warning=F,message=F}
thin <- 1
var_struc <- "shared_all"
```


```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```

```{r warning=F,message=F,fig.width=15, fig.height=5}

param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))


ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n)
total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"]
pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group")


param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS"))

ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n)
total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"]
pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st")


top_tissue <- top_tissues[trait]
file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS")

param_single <- readRDS(file_param_single)
ctwas_parameters_single <- summarize_param(param_single, gwas_n)
total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"]

pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st")


plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait)

```


<!-- # aFib-ebi-a-GCST006414 -->

<!-- ```{r warning=F,message=F} -->
<!-- trait <- "aFib-ebi-a-GCST006414" -->
<!-- st <- "with_susieST" -->

<!-- gwas_n <- samplesize[trait] -->
<!-- ``` -->


<!-- ## %h2g for setting: shared_all, thin = 1 -->

<!-- ```{r warning=F,message=F} -->
<!-- thin <- 1 -->
<!-- var_struc <- "shared_all" -->
<!-- ``` -->



<!-- ```{r warning=F,message=F,fig.width=15, fig.height=5} -->

<!-- param_susieST <- readRDS(paste0(folder_results_susieST,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS")) -->


<!-- ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n) -->
<!-- total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"] -->
<!-- pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "top-apa, multi-group") -->


<!-- param_apaonly <- readRDS(paste0(folder_results_apaonly,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS")) -->

<!-- ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n) -->
<!-- total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"] -->
<!-- pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "top-apa, multi tissue apa") -->


<!-- top_tissue <- top_tissues[trait] -->
<!-- file_param_single <- paste0(folder_results_single, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS") -->

<!-- param_single <- readRDS(file_param_single) -->
<!-- ctwas_parameters_single <- summarize_param(param_single, gwas_n) -->
<!-- total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"] -->

<!-- pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "top-apa, single tissue apa") -->


<!-- plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait) -->

<!-- ``` -->


<!-- ```{r warning=F,message=F,fig.width=15, fig.height=5} -->

<!-- param_susieST <- readRDS(paste0(folder_results_susieST_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS")) -->


<!-- ctwas_parameters_susieST <- summarize_param(param_susieST, gwas_n) -->
<!-- total_nonSNPpve_susieST <- 1- ctwas_parameters_susieST$prop_heritability["SNP"] -->
<!-- pve_pie_by_type_susieST <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_susieST, colors = colors, by = "type", title = "multi-group") -->


<!-- param_apaonly <- readRDS(paste0(folder_results_apaonly_susie,"/",trait,"/",trait,".",st,".thin",thin,".",var_struc,".param.RDS")) -->

<!-- ctwas_parameters_apaonly <- summarize_param(param_apaonly,gwas_n) -->
<!-- total_nonSNPpve_apaonly <- 1- ctwas_parameters_apaonly$prop_heritability["SNP"] -->
<!-- pve_pie_by_type_apaonly <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_apaonly, colors = colors, by = "type", title = "multi tissue susie st") -->


<!-- top_tissue <- top_tissues[trait] -->
<!-- file_param_single <- paste0(folder_results_single_susie, trait, "/", trait, "_", top_tissue, ".thin", thin, ".", var_struc, ".param.RDS") -->

<!-- param_single <- readRDS(file_param_single) -->
<!-- ctwas_parameters_single <- summarize_param(param_single, gwas_n) -->
<!-- total_nonSNPpve_single <- 1- ctwas_parameters_single$prop_heritability["SNP"] -->

<!-- pve_pie_by_type_single <- plot_piechart_topn(ctwas_parameters = ctwas_parameters_single, colors = colors, by = "context", title = "single tissue susie st") -->


<!-- plot_multi(pve_pie_by_type_single,pve_pie_by_type_apaonly,pve_pie_by_type_susieST, title=trait) -->

<!-- ``` -->