---
title: "Seven Correlated Tissues Simulation"
author: "shengqian"
date: "2023-12-30"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

A simulation of seven correlated tissues is conducted to evaluate cTWAS performance (parameter estimation, PIP calibration ...). Seven tissues used in this simulation are Artery Aorta, Spleen, Skin (not sun exposed suprapubic), Lung, Adipose Subcutaneous, Pancreas, Heart Artial Appendage. Pairwise correlation of gene expression are with 0.6-0.8. The first three tissues are set to be causal and the other four tissues are non-causal. 

```{r echo=FALSE}
load("/project/xinhe/xsun/ctwas/1.matching_tissue/data/tissue_cor.rdata")
data <- lat[colnames(lat)%in%c("Artery_Aorta","Spleen","Skin_Not_Sun_Exposed_Suprapubic","Lung","Adipose_Subcutaneous","Pancreas","Heart_Atrial_Appendage"),rownames(lat)%in%c("Artery_Aorta","Spleen","Skin_Not_Sun_Exposed_Suprapubic","Lung","Adipose_Subcutaneous","Pancreas","Heart_Atrial_Appendage")]
colnames(data) <- c("Adipose","Lung","Artery","Heart","Skin","Spleen","Pancreas")
rownames(data) <- c("Adipose","Lung","Artery","Heart","Skin","Spleen","Pancreas")
data <- round(data,3)
data
```

It current has two settings:

* 3% PVE, 0.9% $\pi$ for causal tissues, 0.5% PVE, 0.15% $\pi$ for non-causal tissues and 30% PVE, 2.5e-4 $\pi$ for SNP. 
* 3% PVE, 0.9% $\pi$ for causal tissues, 0% PVE, 0% $\pi$ for non-causal tissues and 30% PVE, 2.5e-4 $\pi$ for SNP.

We observed that cTWAS always tend to overestimate PVE of non-causal tissues because parameters won't be shrunk exactly to 0. Therefore, we assign non-zero (but very low) PVE to non-causal tissues (the first setting) to check if it helps simulation results. 

Conclusion: It seems that for tissues with moderate correlation (0.6-0.8), Assigning non-zero (but very low) PVE to non-causal tissues does not outperform zero PVE case (the second simulation). cTWAS estimates parameters more accurately in the second simulation (estimated PVE very close to 0) and has lower false positive rates in the PIP calibration plot.

```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(ctwas)
library(data.table)
require(latex2exp)
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_basic_plots.R")
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_ctwas_plots.R")
source("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/analysis/simulation_help_functions.R")
plot_PIP <- function(configtag, runtag,  simutags, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_config", configtag,"_LDR",".susieIrss.txt")
   f1 <- caliPIP_plot(phenofs, susieIfs, ...) 
   return(f1)
}
plot_PIP_combined <- function(configtag, runtag,  simutags, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_config", configtag,"_LDR",".susieIrss.txt")
   f1 <- caliPIP_plot_combined(phenofs, susieIfs, ...) 
   return(f1)
}
plot_PIP_L1 <- function(configtag, runtag,  simutags, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_config", configtag,"_LDR",".L1.susieIrss.txt")
   f1 <- caliPIP_plot(phenofs, susieIfs, ...) 
   return(f1)
}
plot_PIP_L1_combined <- function(configtag, runtag,  simutags, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_config", configtag,"_LDR",".L1.susieIrss.txt")
   f1 <- caliPIP_plot_combined(phenofs, susieIfs, ...) 
   return(f1)
}
```

## Simulation 1: 3% PVE for Causal Tissues and 0.5% PVE for Non-causal Tissues. 

### Shared Prior Variance Parameters

```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/"
runtag = "ukb-s80.45-3_7corr"
configtag <- 1

simutags <- paste(1, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

results_df <- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
group_size <- results_df[[2]]
results_df <- results_df[[1]]
```

### Results using PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)
```

### Results using Combined PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)
```

### Estimated Prior Inclusion Probability
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$prior_mashr_Artery_Aorta
y2 <- results_df$prior_mashr_Spleen
y3 <- results_df$prior_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_mashr_Lung
y5 <- results_df$prior_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_mashr_Pancreas
y7 <- results_df$prior_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0.0015),c(5,0.0015),c(6,0.0015),c(7,0.0015))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.025),ylab="Prior inclusion")
```

### Estimated PVE
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta
y2 <- results_df$pve_mashr_Spleen
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$pve_mashr_Lung
y5 <- results_df$pve_mashr_Adipose_Subcutaneous
y6 <- results_df$pve_mashr_Pancreas
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0.005),c(5,0.005),c(6,0.005),c(7,0.005))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.06),ylab="PVE")
```

### Estimated Prior Variance
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$prior_var_mashr_Artery_Aorta
y2 <- results_df$prior_var_mashr_Spleen
y3 <- results_df$prior_var_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_var_mashr_Lung
y5 <- results_df$prior_var_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_var_mashr_Pancreas
y7 <- results_df$prior_var_mashr_Heart_Atrial_Appendage

t1 <- 0.03*sample_size/(group_size[2]*0.009)
t2 <- 0.03*sample_size/(group_size[3]*0.009)
t3 <- 0.03*sample_size/(group_size[4]*0.009)
t4 <- 0.005*sample_size/(group_size[5]*0.0015)
t5 <- 0.005*sample_size/(group_size[6]*0.0015)
t6 <- 0.005*sample_size/(group_size[7]*0.0015)
t7 <- 0.005*sample_size/(group_size[8]*0.0015)

truth <- rbind(c(1,t1),c(2,t2),c(3,t3),c(4,t4),c(5,t5),c(6,t6),c(7,t7))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,30),ylab=TeX("$sigma^2$"))
```

### Estimated Enrichment
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta/results_df$prior_SNP
y2 <- results_df$pve_mashr_Spleen/results_df$prior_SNP
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic/results_df$prior_SNP
y4 <- results_df$pve_mashr_Lung/results_df$prior_SNP
y5 <- results_df$pve_mashr_Adipose_Subcutaneous/results_df$prior_SNP
y6 <- results_df$pve_mashr_Pancreas/results_df$prior_SNP
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage/results_df$prior_SNP

truth <- rbind(c(1,120),c(2,120),c(3,120),c(4,20),c(5,20),c(6,20),c(7,20))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,250),ylab= TeX("Enrichment $\\pi_G/\\pi_V$"))
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(configtag, runtag, simutags, main = "Expression Trait PIP Calibration")
f2 <-  plot_PIP_combined(configtag, runtag, simutags, main = "Gene PIP Calibration")
gridExtra::grid.arrange(f1, f2, ncol =2)
```

### Separate effect size parameters

```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/"
runtag = "ukb-s80.45-3_7corr"
configtag <- 2

simutags <- paste(1, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

results_df <- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
group_size <- results_df[[2]]
results_df <- results_df[[1]]
```

### Results using PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)
```

### Results using Combined PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)
```

### Estimated Prior Inclusion Probability
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$prior_mashr_Artery_Aorta
y2 <- results_df$prior_mashr_Spleen
y3 <- results_df$prior_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_mashr_Lung
y5 <- results_df$prior_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_mashr_Pancreas
y7 <- results_df$prior_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0.0015),c(5,0.0015),c(6,0.0015),c(7,0.0015))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.025),ylab="Prior inclusion")
```

### Estimated PVE
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta
y2 <- results_df$pve_mashr_Spleen
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$pve_mashr_Lung
y5 <- results_df$pve_mashr_Adipose_Subcutaneous
y6 <- results_df$pve_mashr_Pancreas
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0.005),c(5,0.005),c(6,0.005),c(7,0.005))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.06),ylab="PVE")
```

### Estimated Prior Variance
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$prior_var_mashr_Artery_Aorta
y2 <- results_df$prior_var_mashr_Spleen
y3 <- results_df$prior_var_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_var_mashr_Lung
y5 <- results_df$prior_var_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_var_mashr_Pancreas
y7 <- results_df$prior_var_mashr_Heart_Atrial_Appendage

t1 <- 0.03*sample_size/(group_size[2]*0.009)
t2 <- 0.03*sample_size/(group_size[3]*0.009)
t3 <- 0.03*sample_size/(group_size[4]*0.009)
t4 <- 0.005*sample_size/(group_size[5]*0.0015)
t5 <- 0.005*sample_size/(group_size[6]*0.0015)
t6 <- 0.005*sample_size/(group_size[7]*0.0015)
t7 <- 0.005*sample_size/(group_size[8]*0.0015)

truth <- rbind(c(1,t1),c(2,t2),c(3,t3),c(4,t4),c(5,t5),c(6,t6),c(7,t7))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,30),ylab=TeX("$sigma^2$"))
```

### Estimated Enrichment
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta/results_df$prior_SNP
y2 <- results_df$pve_mashr_Spleen/results_df$prior_SNP
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic/results_df$prior_SNP
y4 <- results_df$pve_mashr_Lung/results_df$prior_SNP
y5 <- results_df$pve_mashr_Adipose_Subcutaneous/results_df$prior_SNP
y6 <- results_df$pve_mashr_Pancreas/results_df$prior_SNP
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage/results_df$prior_SNP

truth <- rbind(c(1,120),c(2,120),c(3,120),c(4,20),c(5,20),c(6,20),c(7,20))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,250),ylab= TeX("Enrichment $\\pi_G/\\pi_V$"))
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(configtag, runtag, simutags, main = "Expression Trait PIP Calibration")
f2 <-  plot_PIP_combined(configtag, runtag, simutags, main = "Gene PIP Calibration")
gridExtra::grid.arrange(f1, f2, ncol =2)
```

## Simulation 2: 3% PVE for Causal Tissues and 0% PVE for Non-causal Tissues. 

### Shared Prior Variance Parameters
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/"
runtag = "ukb-s80.45-3_7corr"
configtag <- 1

simutags <- paste(2, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

results_df <- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
group_size <- results_df[[2]]
results_df <- results_df[[1]]
```

### Results using PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)
```

### Results using Combined PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)
```

### Estimated Prior Inclusion Probability
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$prior_mashr_Artery_Aorta
y2 <- results_df$prior_mashr_Spleen
y3 <- results_df$prior_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_mashr_Lung
y5 <- results_df$prior_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_mashr_Pancreas
y7 <- results_df$prior_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.025),ylab="Prior inclusion")
```

### Estimated PVE
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta
y2 <- results_df$pve_mashr_Spleen
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$pve_mashr_Lung
y5 <- results_df$pve_mashr_Adipose_Subcutaneous
y6 <- results_df$pve_mashr_Pancreas
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.06),ylab="PVE")
```

### Estimated Prior Variance
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$prior_var_mashr_Artery_Aorta
y2 <- results_df$prior_var_mashr_Spleen
y3 <- results_df$prior_var_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_var_mashr_Lung
y5 <- results_df$prior_var_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_var_mashr_Pancreas
y7 <- results_df$prior_var_mashr_Heart_Atrial_Appendage

t1 <- 0.03*sample_size/(group_size[2]*0.009)
t2 <- 0.03*sample_size/(group_size[3]*0.009)
t3 <- 0.03*sample_size/(group_size[4]*0.009)
t4 <- 0
t5 <- 0
t6 <- 0
t7 <- 0

truth <- rbind(c(1,t1),c(2,t2),c(3,t3),c(4,t4),c(5,t5),c(6,t6),c(7,t7))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,30),ylab=TeX("$sigma^2$"))
```

### Estimated Enrichment
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta/results_df$prior_SNP
y2 <- results_df$pve_mashr_Spleen/results_df$prior_SNP
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic/results_df$prior_SNP
y4 <- results_df$pve_mashr_Lung/results_df$prior_SNP
y5 <- results_df$pve_mashr_Adipose_Subcutaneous/results_df$prior_SNP
y6 <- results_df$pve_mashr_Pancreas/results_df$prior_SNP
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage/results_df$prior_SNP

truth <- rbind(c(1,120),c(2,120),c(3,120),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,250),ylab= TeX("Enrichment $\\pi_G/\\pi_V$"))
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(configtag, runtag, simutags, main = "Expression Trait PIP Calibration")
f2 <-  plot_PIP_combined(configtag, runtag, simutags, main = "Gene PIP Calibration")
gridExtra::grid.arrange(f1, f2, ncol =2)
```

### PIP Calibration Plot (L1=1)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP_L1(configtag, runtag, simutags, main = "Expression Trait PIP Calibration")
f2 <-  plot_PIP_L1_combined(configtag, runtag, simutags, main = "Gene PIP Calibration")
gridExtra::grid.arrange(f1, f2, ncol =2)
```

### Separate effect size parameters
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/"
runtag = "ukb-s80.45-3_7corr"
configtag <- 2

simutags <- paste(2, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

results_df <- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
group_size <- results_df[[2]]
results_df <- results_df[[1]]
```

### Results using PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)
```

### Results using Combined PIP Threshold
```{r echo=FALSE}
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]
#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)
```

### Estimated Prior Inclusion Probability
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$prior_mashr_Artery_Aorta
y2 <- results_df$prior_mashr_Spleen
y3 <- results_df$prior_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_mashr_Lung
y5 <- results_df$prior_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_mashr_Pancreas
y7 <- results_df$prior_mashr_Heart_Atrial_Appendage


truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.025),ylab="Prior inclusion")
```

### Estimated Prior Variance
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$prior_var_mashr_Artery_Aorta
y2 <- results_df$prior_var_mashr_Spleen
y3 <- results_df$prior_var_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$prior_var_mashr_Lung
y5 <- results_df$prior_var_mashr_Adipose_Subcutaneous
y6 <- results_df$prior_var_mashr_Pancreas
y7 <- results_df$prior_var_mashr_Heart_Atrial_Appendage

t1 <- 0.03*sample_size/(group_size[2]*0.009)
t2 <- 0.03*sample_size/(group_size[3]*0.009)
t3 <- 0.03*sample_size/(group_size[4]*0.009)
t4 <- 0
t5 <- 0
t6 <- 0
t7 <- 0

truth <- rbind(c(1,t1),c(2,t2),c(3,t3),c(4,t4),c(5,t5),c(6,t6),c(7,t7))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,60),ylab=TeX("$sigma^2$"))
```

### Estimated Enrichment
```{r, fig.width= 8, fig.height= 5, message=FALSE, warning=FALSE, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta/results_df$prior_SNP
y2 <- results_df$pve_mashr_Spleen/results_df$prior_SNP
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic/results_df$prior_SNP
y4 <- results_df$pve_mashr_Lung/results_df$prior_SNP
y5 <- results_df$pve_mashr_Adipose_Subcutaneous/results_df$prior_SNP
y6 <- results_df$pve_mashr_Pancreas/results_df$prior_SNP
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage/results_df$prior_SNP

truth <- rbind(c(1,120),c(2,120),c(3,120),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,250),ylab= TeX("Enrichment $\\pi_G/\\pi_V$"))
```

### Estimated PVE
```{r, fig.width= 8, fig.height= 5, echo=FALSE}
y1 <- results_df$pve_mashr_Artery_Aorta
y2 <- results_df$pve_mashr_Spleen
y3 <- results_df$pve_mashr_Skin_Not_Sun_Exposed_Suprapubic
y4 <- results_df$pve_mashr_Lung
y5 <- results_df$pve_mashr_Adipose_Subcutaneous
y6 <- results_df$pve_mashr_Pancreas
y7 <- results_df$pve_mashr_Heart_Atrial_Appendage

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.06),ylab="PVE")
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(configtag, runtag, simutags, main = "Expression Trait PIP Calibration")
f2 <-  plot_PIP_combined(configtag, runtag, simutags, main = "Gene PIP Calibration")
gridExtra::grid.arrange(f1, f2, ncol =2)
```
