---
title: "simulation_summary"
output: html_document
date: '2024-10-02'
editor_options: 
  chunk_output_type: console
---

```{r, message=FALSE, warning=FALSE, echo=FALSE}
library(ctwas)
library(data.table)
library(ggbreak) 
source("/project/xinhe/shengqian/cTWAS_simulation/code/simulation_help_functions.R")
source("/project/xinhe/shengqian/cTWAS_simulation/code/summarize_basic_plots.R")
source("/project/xinhe/shengqian/cTWAS_simulation/code/summarize_ctwas_plots.R")
get_top_genes <- function(susieIf){
  finemap_res <- readRDS(susieIf)$finemap_res
  finemap_res <- finemap_res[finemap_res$type!="SNP",]
  finemap_res <- finemap_res[!is.na(finemap_res$cs),]
  finemap_res <- finemap_res[finemap_res$susie_pip>0.8,]
  return(finemap_res$id)
}

get_top_genes_combined <- function(susieIf){
  finemap_res <- readRDS(susieIf)
  res <- combine_gene_pips(finemap_res$susie_alpha_res, 
                             group_by = "molecular_id",
                             by = "context",
                             method = "combine_cs",
                             filter_cs = TRUE,
                             include_cs_id = FALSE)
  res <- res[res$combined_pip>0.8,]
  return(res$molecular_id)
}

get_top_genes_coloc <- function(colocf,group){
  coloc_res <- readRDS(colocf)
  coloc_res <- coloc_res[coloc_res$PP4>0.8,]
  return(paste0(coloc_res$id,"|",group))
}
```
## Simulation 1: Four tissues expression trait
Four expression traits from PredictDB are used in this study, which are Liver, Adipose, Lung and stomach. Liver and Adipose are selected as causal tissues, each with 3% PVE and 0.9% prior. Lung and stomach are non-causal tissues with 0% PVE.

### Number of causal genes detected
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_four_tissues_expression/"
runtag = "four_tissues"
simutags <- paste(1, 1:5, sep = "-")
check_power(results_dir, runtag, simutags, "UKBBLD", 'estimatedL', PIP_threshold = 0.8)
```

### Estimated parameters
```{r, fig.width= 12, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
par(mfrow = c(1, 3))
y1 <- param_UKBB_df$`prior_Liver|eQTL`
y2 <- param_UKBB_df$`prior_Adipose_Subcutaneous|eQTL`
y3 <- param_UKBB_df$`prior_Lung|eQTL`
y4 <- param_UKBB_df$`prior_Stomach|eQTL`
truth <- rbind(c(1,0.009),c(2,0.009),c(3,0),c(4,0))
est <- rbind(cbind(1,y1),cbind(2,y2), cbind(3,y3), cbind(4,y4))
plot_par_4(truth,est,xlabels = c("Liver", "Adipose", "Lung", "Stomach"),ylim=c(0,0.025),ylab="Prior inclusion", main="Prior Inclusion Probability")

y1 <- param_UKBB_df$`pve_Liver|eQTL`
y2 <- param_UKBB_df$`pve_Adipose_Subcutaneous|eQTL`
y3 <- param_UKBB_df$`pve_Lung|eQTL`
y4 <- param_UKBB_df$`prior_Stomach|eQTL`
truth <- rbind(c(1,0.03),c(2,0.03),c(3,0),c(4,0))
est <- rbind(cbind(1,y1),cbind(2,y2), cbind(3,y3), cbind(4,y4))
plot_par_4(truth,est,xlabels = c("Liver", "Adipose", "Lung", "Stomach"),ylim=c(0,0.06),ylab="PVE", main="PVE")

y1 <- param_UKBB_df$`enrichment_Liver|eQTL`
y2 <- param_UKBB_df$`enrichment_Adipose_Subcutaneous|eQTL`
y3 <- param_UKBB_df$`enrichment_Lung|eQTL`
y4 <- param_UKBB_df$`enrichment_Stomach|eQTL`
truth <- rbind(c(1,36),c(2,36),c(3,0),c(4,0))
est <- rbind(cbind(1,y1),cbind(2,y2), cbind(3,y3), cbind(4,y4))
plot_par_4(truth,est,xlabels = c("Liver", "Adipose", "Lung", "Stomach"),ylim=c(0,80),ylab="Enrichment", main="Enrichment")
```

```{r, message=FALSE, warning=FALSE, echo=FALSE}
plot_PIP_estimatedL <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_nonSNPpip <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_estimatedL <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_nonSNPpip <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
f1 <- plot_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Molecular Trait Level")
f2 <- plot_gene_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Gene Level")
gridExtra::grid.arrange(f1, f2, ncol = 2)
```

### Compare to other methods (molecular-tissue pair)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_four_tissues_expression/"
runtag = "four_tissues"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes(susieIfs[i])
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),get_top_genes(susieIfs4[i]))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"mashr_Liver"),get_top_genes_coloc(colocfs2[i],"mashr_Adipose_Subcutaneous"),
                   get_top_genes_coloc(colocfs3[i],"mashr_Lung"),get_top_genes_coloc(colocfs4[i],"mashr_Stomach"))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r, fig.width= 3, fig.height= 3, echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS","coloc","single-cTWAS","multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### Compare to other methods (gene)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_four_tissues_expression/"
runtag = "four_tissues"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  causal_genes <- unique(sapply(causal_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  twas_genes <- unique(sapply(twas_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes_combined(susieIfs[i])
  multi_cTWAS_genes <- unique(sapply(multi_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),get_top_genes(susieIfs4[i]))
  single_cTWAS_genes <- unique(sapply(single_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"mashr_Liver"),get_top_genes_coloc(colocfs2[i],"mashr_Adipose_Subcutaneous"),
                   get_top_genes_coloc(colocfs3[i],"mashr_Lung"),get_top_genes_coloc(colocfs4[i],"mashr_Stomach"))
  
  coloc_genes <- unique(sapply(coloc_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### PIP partition across tissues
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_four_tissues_expression/"
runtag = "four_tissues"
simutags <- paste(1, 1:5, sep = "-")
tissues <- c("Liver", "Adipose_Subcutaneous", "Lung", "Stomach")
fig <- pip_partition_across_tissues(results_dir, runtag, simutags, "UKBBLD", tissues, 0.8)
fig
```

## Simulation 2: Three tissues expression and splicing trait
A simulation of expression and splicing traits from three tissues is conducted to evaluate cTWAS performance (parameter estimation, PIP calibration ...). Three tissues used in this simulation are Liver, Adipose and Lung. Only expression and splicing from liver are causal, each with 3% PVE, 0.9% $\pi$. SNPs have 30% PVE, 2.5e-4 $\pi$.

### Number of causal genes detected
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_three_tissues_expression_splicing/"
runtag = "E_S_three_tissues"
simutags <- paste(1, 1:5, sep = "-")
check_power(results_dir, runtag, simutags, "UKBBLD", 'estimatedL', PIP_threshold = 0.8)
```

### Estimated parameters
```{r, fig.width= 12, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
par(mfrow = c(1, 3))
y1 <- param_UKBB_df$`prior_Liver|expression`
y2 <- param_UKBB_df$`prior_Liver|splicing`
y3 <- param_UKBB_df$`prior_Adipose_Subcutaneous|expression`
y4 <- param_UKBB_df$`prior_Adipose_Subcutaneous|splicing`
y5 <- param_UKBB_df$`prior_Lung|expression`
y6 <- param_UKBB_df$`prior_Lung|splicing`
truth <- rbind(c(1,0.009),c(2,0.009),c(3,0),c(4,0),c(5,0),c(6,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6))
plot_par_6(truth,est,xlabels = c("Liver E", "Liver S", "Adipose E", "Adipose S", "Lung E", "Lung S"),ylim=c(0,0.025),ylab="Prior inclusion", main="Prior Inclusion Probability")

y1 <- param_UKBB_df$`pve_Liver|expression`
y2 <- param_UKBB_df$`pve_Liver|splicing`
y3 <- param_UKBB_df$`pve_Adipose_Subcutaneous|expression`
y4 <- param_UKBB_df$`pve_Adipose_Subcutaneous|splicing`
y5 <- param_UKBB_df$`pve_Lung|expression`
y6 <- param_UKBB_df$`pve_Lung|splicing`
truth <- rbind(c(1,0.03),c(2,0.03),c(3,0),c(4,0),c(5,0),c(6,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6))
plot_par_6(truth,est,xlabels = c("Liver E", "Liver S", "Adipose E", "Adipose S", "Lung E", "Lung S"),ylim=c(0,0.06),ylab="PVE", main="PVE")

y1 <- param_UKBB_df$`enrichment_Liver|expression`
y2 <- param_UKBB_df$`enrichment_Liver|splicing`
y3 <- param_UKBB_df$`enrichment_Adipose_Subcutaneous|expression`
y4 <- param_UKBB_df$`enrichment_Adipose_Subcutaneous|splicing`
y5 <- param_UKBB_df$`enrichment_Lung|expression`
y6 <- param_UKBB_df$`enrichment_Lung|splicing`
truth <- rbind(c(1,36),c(2,36),c(3,0),c(4,0),c(5,0),c(6,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6))
plot_par_6(truth,est,xlabels = c("Liver E", "Liver S", "Adipose E", "Adipose S", "Lung E", "Lung S"),ylim=c(0,80),ylab="Enrichment", main="Enrichment")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
plot_PIP_estimatedL <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_nonSNPpip <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_estimatedL <- function(configtag, runtag,  simutags, LD_type, mapping_table, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   snp_mapfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".snp_map.RDS")
   f1 <- caliPIP_plot_multi_omics(phenofs, susieIfs, snp_mapfs, mapping_table, ...) 
   return(f1)
}

plot_gene_PIP_nonSNPpip <- function(configtag, runtag,  simutags, LD_type, mapping_table, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   snp_mapfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".snp_map.RDS")
   f1 <- caliPIP_plot_multi_omics(phenofs, susieIfs, snp_mapfs, mapping_table, ...) 
   return(f1)
}
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
mapping_table <- readRDS("/project2/xinhe/shared_data/multigroup_ctwas/weights/mapping_files/PredictDB_mapping.RDS")
f1 <- plot_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Molecular Trait Level")
f2 <- plot_gene_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", mapping_table, main = "Gene Level")
gridExtra::grid.arrange(f1, f2, ncol = 2)
```

### Compare to other methods
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_three_tissues_expression_splicing/"
runtag = "E_S_three_tissues"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes(susieIfs[i])
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "single-cTWAS", "multi-cTWAS"))
plot_df$ifcausal <- plot_df$ifcausal + as.numeric(as.factor(plot_df$method))*10
plot_df$ifcausal <- as.factor(plot_df$ifcausal)
library(ggpubr)
```

```{r echo=FALSE}
#pdf(file = "/project/xinhe/shengqian/poster_plot/compare.pdf", width = 4, height = 5)
colset = c("#ebebeb", "#87CEFA", # FOCUS
           "#ebebeb", "#ffff99", # Fusion
           "#ebebeb", "#fb8072", # Fusion-permutation
           "#ebebeb", "#CC79A7", # MR-JTI
           "#ebebeb", "goldenrod", #PMR-Egger
           "#ebebeb", "#87CEFA", # SMR
           "#ebebeb", "#fb8072", # cTWAS
           "#ebebeb", "#bebada") # coloc

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          width = 0.5,
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + scale_y_break(c(50, 600),space=0.5,scales="free")
#dev.off()
```

```{r}
library(ggpubr)

colset = c("#ebebeb", "#7fc97f", "#beaed4", "#fdc086")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "method",     # Change here to fill by 'method' for different colors per bar
          legend = "none", 
          width = 0.5,
          ylab = "Count", 
          xlab = "",
          palette = colset) + 
  grids(linetype = "dashed") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  scale_y_break(c(50, 600), space = 0.5, scales = "free")
```

## Simulation 3: Seven tissues correlated tissues expression trait
A simulation of seven correlated tissues is conducted to evaluate cTWAS performance (parameter estimation, PIP calibration ...). Seven tissues used in this simulation are Artery Aorta, Spleen, Skin (not sun exposed suprapubic), Lung, Adipose Subcutaneous, Pancreas, Heart Artial Appendage. Pairwise correlation of gene expression are with 0.6-0.8. The first three tissues are set to be causal, each with 3% PVE, 0.9% $\pi$ for causal tissues, 0% PVE, 0% $\pi$ for non-causal tissues and 30% PVE, 2.5e-4 $\pi$ for SNP.

### Comparing number of regions screened
```{r, message=FALSE, warning=FALSE, echo=FALSE, eval=FALSE}
num_estimatedL <- c()
for(simutag in simutags){
    regions <- readRDS(paste0(results_dir,runtag,"_simu",simutag,"_","UKBBLD",".estimatedL.screened_region_data.RDS"))
    num_estimatedL <- c(num_estimatedL,length(regions))
}
num_nonSNPpip <- c()
for(simutag in simutags){
    regions <- readRDS(paste0(results_dir,runtag,"_simu",simutag,"_","UKBBLD",".nonSNPpip.screened_region_data.RDS"))
    num_nonSNPpip <- c(num_nonSNPpip,length(regions))
}
data.frame(cbind(simutags,num_estimatedL,num_nonSNPpip))
```

### Number of causal genes detected
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_tissues_expression/"
runtag = "7corr"
simutags <- paste(1, 1:5, sep = "-")
check_power(results_dir, runtag, simutags, "UKBBLD", 'estimatedL', PIP_threshold = 0.8)
```

### Estimated parameters
```{r, fig.width= 12, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
par(mfrow = c(1, 3))
y1 <- param_UKBB_df$`prior_Artery_Aorta|expression`
y2 <- param_UKBB_df$`prior_Spleen|expression`
y3 <- param_UKBB_df$`prior_Skin_Not_Sun_Exposed_Suprapubic|expression`
y4 <- param_UKBB_df$`prior_Lung|expression`
y5 <- param_UKBB_df$`prior_Adipose_Subcutaneous|expression`
y6 <- param_UKBB_df$`prior_Pancreas|expression`
y7 <- param_UKBB_df$`prior_Heart_Atrial_Appendage|expression`
truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.025),ylab="Prior inclusion")

y1 <- param_UKBB_df$`pve_Artery_Aorta|expression`
y2 <- param_UKBB_df$`pve_Spleen|expression`
y3 <- param_UKBB_df$`pve_Skin_Not_Sun_Exposed_Suprapubic|expression`
y4 <- param_UKBB_df$`pve_Lung|expression`
y5 <- param_UKBB_df$`pve_Adipose_Subcutaneous|expression`
y6 <- param_UKBB_df$`pve_Pancreas|expression`
y7 <- param_UKBB_df$`pve_Heart_Atrial_Appendage|expression`

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,0.06),ylab="PVE")

y1 <- param_UKBB_df$`enrichment_Artery_Aorta|expression`
y2 <- param_UKBB_df$`enrichment_Spleen|expression`
y3 <- param_UKBB_df$`enrichment_Skin_Not_Sun_Exposed_Suprapubic|expression`
y4 <- param_UKBB_df$`enrichment_Lung|expression`
y5 <- param_UKBB_df$`enrichment_Adipose_Subcutaneous|expression`
y6 <- param_UKBB_df$`enrichment_Pancreas|expression`
y7 <- param_UKBB_df$`enrichment_Heart_Atrial_Appendage|expression`
  
truth <- rbind(c(1,36),c(2,36),c(3,36),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Artery","Spleen","Skin","Lung","Adipose","Pancreas","Heart"),ylim=c(0,250),ylab= "Enrichment")
```

```{r, message=FALSE, warning=FALSE, echo=FALSE}
plot_PIP_estimatedL <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_nonSNPpip <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_estimatedL <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_nonSNPpip <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}
```

### PIP Calibration Plot
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
f1 <- plot_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Molecular Trait Level")
f2 <- plot_gene_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Gene Level")
gridExtra::grid.arrange(f1, f2, ncol = 2)
```

### Compare to other methods (molecular-tissue pair)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_tissues_expression/"
runtag = "7corr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes(susieIfs[i])
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Artery_Aorta"),get_top_genes_coloc(colocfs2[i],"expression_Spleen"),
                   get_top_genes_coloc(colocfs3[i],"expression_Skin_Not_Sun_Exposed_Suprapubic"),get_top_genes_coloc(colocfs4[i],"expression_Lung"),
                   get_top_genes_coloc(colocfs5[i],"expression_Adipose_Subcutaneous"),get_top_genes_coloc(colocfs6[i],"expression_Pancreas"),
                   get_top_genes_coloc(colocfs7[i],"expression_Heart_Atrial_Appendage"))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### Compare to other methods (gene)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_tissues_expression/"
runtag = "7corr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  causal_genes <- unique(sapply(causal_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  twas_genes <- unique(sapply(twas_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes_combined(susieIfs[i])
  multi_cTWAS_genes <- unique(sapply(multi_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  single_cTWAS_genes <- unique(sapply(single_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Artery_Aorta"),get_top_genes_coloc(colocfs2[i],"expression_Spleen"),
                   get_top_genes_coloc(colocfs3[i],"expression_Skin_Not_Sun_Exposed_Suprapubic"),get_top_genes_coloc(colocfs4[i],"expression_Lung"),
                   get_top_genes_coloc(colocfs5[i],"expression_Adipose_Subcutaneous"),get_top_genes_coloc(colocfs6[i],"expression_Pancreas"),
                   get_top_genes_coloc(colocfs7[i],"expression_Heart_Atrial_Appendage"))
  
  coloc_genes <- unique(sapply(coloc_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### PIP partition across tissues
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_tissues_expression/"
runtag = "7corr"
simutags <- paste(1, 1:5, sep = "-")
tissues <- c("Artery_Aorta","Spleen","Skin_Not_Sun_Exposed_Suprapubic","Lung","Adipose_Subcutaneous","Pancreas","Heart_Atrial_Appendage")
fig <- pip_partition_across_tissues(results_dir, runtag, simutags, "UKBBLD", tissues, 0.8)
fig
```

## Simulation 4: Seven correlated brain tissues expression trait
A simulation of seven correlated brain tissues is conducted to evaluate cTWAS performance (parameter estimation, PIP calibration ...). Three tissues are causal

### Number of causal genes detected
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_brain_tissues_expression/"
runtag = "brain_7corr"
simutags <- paste(1, 1:5, sep = "-")
check_power(results_dir, runtag, simutags, "UKBBLD", 'estimatedL', PIP_threshold = 0.8)
```

### Estimated parameters
```{r, fig.width= 12, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
par(mfrow = c(1, 3))
y1 <- param_UKBB_df$`prior_Brain_Cerebellum|expression`
y2 <- param_UKBB_df$`prior_Brain_Hippocampus|expression`
y3 <- param_UKBB_df$`prior_Brain_Caudate_basal_ganglia|expression`
y4 <- param_UKBB_df$`prior_Brain_Cerebellar_Hemisphere|expression`
y5 <- param_UKBB_df$`prior_Brain_Cortex|expression`
y6 <- param_UKBB_df$`prior_Brain_Hypothalamus|expression`
y7 <- param_UKBB_df$`prior_Brain_Putamen_basal_ganglia|expression`
truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Cerebellum","Hippocampus","Caudate basal ganglia","Cerebellar Hemisphere","Cortex","Hypothalamus","Putamen basal ganglia"),ylim=c(0,0.025),ylab="Prior inclusion")

y1 <- param_UKBB_df$`pve_Brain_Cerebellum|expression`
y2 <- param_UKBB_df$`pve_Brain_Hippocampus|expression`
y3 <- param_UKBB_df$`pve_Brain_Caudate_basal_ganglia|expression`
y4 <- param_UKBB_df$`pve_Brain_Cerebellar_Hemisphere|expression`
y5 <- param_UKBB_df$`pve_Brain_Cortex|expression`
y6 <- param_UKBB_df$`pve_Brain_Hypothalamus|expression`
y7 <- param_UKBB_df$`pve_Brain_Putamen_basal_ganglia|expression`

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Cerebellum","Hippocampus","Caudate basal ganglia","Cerebellar Hemisphere","Cortex","Hypothalamus","Putamen basal ganglia"),ylim=c(0,0.06),ylab="PVE")

y1 <- param_UKBB_df$`enrichment_Brain_Cerebellum|expression`
y2 <- param_UKBB_df$`enrichment_Brain_Hippocampus|expression`
y3 <- param_UKBB_df$`enrichment_Brain_Caudate_basal_ganglia|expression`
y4 <- param_UKBB_df$`enrichment_Brain_Caudate_basal_ganglia|expression`
y5 <- param_UKBB_df$`enrichment_Brain_Cortex|expression`
y6 <- param_UKBB_df$`enrichment_Brain_Hypothalamus|expression`
y7 <- param_UKBB_df$`enrichment_Brain_Putamen_basal_ganglia|expression`
  
truth <- rbind(c(1,36),c(2,36),c(3,36),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Cerebellum","Hippocampus","Caudate basal ganglia","Cerebellar Hemisphere","Cortex","Hypothalamus","Putamen basal ganglia"),ylim=c(0,250),ylab= "Enrichment")
```


```{r, message=FALSE, warning=FALSE, echo=FALSE}
plot_PIP_estimatedL <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_nonSNPpip <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_estimatedL <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_nonSNPpip <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}
```

### PIP Calibration Plot 
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
f1 <- plot_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Molecular Trait Level")
f2 <- plot_gene_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Gene Level")
gridExtra::grid.arrange(f1, f2, ncol = 2)
```

### Compare to other methods (molecular-tissue pair)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_brain_tissues_expression/"
runtag = "brain_7corr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes(susieIfs[i])
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Brain_Cerebellum"),get_top_genes_coloc(colocfs2[i],"expression_Brain_Hippocampus"),
                   get_top_genes_coloc(colocfs3[i],"expression_Brain_Caudate_basal_ganglia"),get_top_genes_coloc(colocfs4[i],"expression_Brain_Cerebellar_Hemisphere"),
                   get_top_genes_coloc(colocfs5[i],"expression_Brain_Cortex"),get_top_genes_coloc(colocfs6[i],"expression_Brain_Hypothalamus"),
                   get_top_genes_coloc(colocfs7[i],"expression_Brain_Putamen_basal_ganglia"))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### Compare to other methods (gene)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_brain_tissues_expression/"
runtag = "brain_7corr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  causal_genes <- unique(sapply(causal_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  twas_genes <- unique(sapply(twas_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes_combined(susieIfs[i])
  multi_cTWAS_genes <- unique(sapply(multi_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  single_cTWAS_genes <- unique(sapply(single_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Brain_Cerebellum"),get_top_genes_coloc(colocfs2[i],"expression_Brain_Hippocampus"),
                   get_top_genes_coloc(colocfs3[i],"expression_Brain_Caudate_basal_ganglia"),get_top_genes_coloc(colocfs4[i],"expression_Brain_Cerebellar_Hemisphere"),
                   get_top_genes_coloc(colocfs5[i],"expression_Brain_Cortex"),get_top_genes_coloc(colocfs6[i],"expression_Brain_Hypothalamus"),
                   get_top_genes_coloc(colocfs7[i],"expression_Brain_Putamen_basal_ganglia"))
  
  coloc_genes <- unique(sapply(coloc_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### PIP partition across tissues
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_correlated_brain_tissues_expression/"
runtag = "brain_7corr"
simutags <- paste(1, 1:5, sep = "-")
tissues <- c("Brain_Cerebellum","Brain_Hippocampus","Brain_Caudate_basal_ganglia","Brain_Cerebellar_Hemisphere","Brain_Cortex","Brain_Hypothalamus","Brain_Putamen_basal_ganglia")
fig <- pip_partition_across_tissues(results_dir, runtag, simutags, "UKBBLD", tissues, 0.8)
fig
```

## Simulation 5: seven tissues uncorrelated tissues expression trait
A simulation of seven uncorrelated tissues is conducted to evaluate cTWAS performance (parameter estimation, PIP calibration ...). Seven tissues used in this simulation are Liver, Lung, Whole Blood, Adipose Subcutaneous, Artery_Tibial, Heart Left Ventricle and Stomach. Pairwise correlation of gene expression are with 0.6-0.8. The first three tissues are set to be causal, each with 3% PVE, 0.9% $\pi$ for causal tissues, 0% PVE, 0% $\pi$ for non-causal tissues and 30% PVE, 2.5e-4 $\pi$ for SNP.

### Number of causal genes detected
```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_uncorrelated_tissues_expression/"
runtag = "7uncorr"
simutags <- paste(1, 1:5, sep = "-")
check_power(results_dir, runtag, simutags, "UKBBLD", 'estimatedL', PIP_threshold = 0.8)
```

### Estimated parameters
```{r, fig.width= 12, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
par(mfrow = c(1, 3))
y1 <- param_UKBB_df$`prior_Liver|expression`
y2 <- param_UKBB_df$`prior_Lung|expression`
y3 <- param_UKBB_df$`prior_Whole_Blood|expression`
y4 <- param_UKBB_df$`prior_Adipose_Subcutaneous|expression`
y5 <- param_UKBB_df$`prior_Artery_Tibial|expression`
y6 <- param_UKBB_df$`prior_Heart_Left_Ventricle|expression`
y7 <- param_UKBB_df$`prior_Stomach|expression`
truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Liver","Lung","Blood","Adipose","Artery","Heart","Stomach"), ylim=c(0,0.025),ylab="Prior inclusion")

y1 <- param_UKBB_df$`pve_Liver|expression`
y2 <- param_UKBB_df$`pve_Lung|expression`
y3 <- param_UKBB_df$`pve_Whole_Blood|expression`
y4 <- param_UKBB_df$`pve_Adipose_Subcutaneous|expression`
y5 <- param_UKBB_df$`pve_Artery_Tibial|expression`
y6 <- param_UKBB_df$`pve_Heart_Left_Ventricle|expression`
y7 <- param_UKBB_df$`size_Stomach|expression`

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Liver","Lung","Blood","Adipose","Artery","Heart","Stomach"), ylim=c(0,0.06),ylab="PVE")

y1 <- param_UKBB_df$`enrichment_Liver|expression`
y2 <- param_UKBB_df$`enrichment_Lung|expression`
y3 <- param_UKBB_df$`enrichment_Whole_Blood|expression`
y4 <- param_UKBB_df$`enrichment_Adipose_Subcutaneous|expression`
y5 <- param_UKBB_df$`enrichment_Artery_Tibial|expression`
y6 <- param_UKBB_df$`enrichment_Heart_Left_Ventricle|expression`
y7 <- param_UKBB_df$`enrichment_Stomach|expression`
  
truth <- rbind(c(1,36),c(2,36),c(3,36),c(4,0),c(5,0),c(6,0),c(7,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c("Liver","Lung","Blood","Adipose","Artery","Heart","Stomach"), ylim=c(0,250),ylab= "Enrichment")
```

```{r, message=FALSE, warning=FALSE, echo=FALSE}
plot_PIP_estimatedL <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_nonSNPpip <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_estimatedL <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".estimatedL.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}

plot_gene_PIP_nonSNPpip <- function(configtag, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".nonSNPpip.finemap_regions_res.RDS")
   f1 <- caliPIP_plot_multi_tissues(phenofs, susieIfs, ...) 
   return(f1)
}
```

### PIP Calibration Plot of molecular traits
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
f1 <- plot_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Molecular Trait Level")
f2 <- plot_gene_PIP_estimatedL(results_dir, runtag, simutags, "UKBBLD", main = "Gene Level")
gridExtra::grid.arrange(f1, f2, ncol = 2)
```

### Compare to other methods
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_uncorrelated_tissues_expression/"
runtag = "7uncorr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes(susieIfs[i])
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Liver"),get_top_genes_coloc(colocfs2[i],"expression_Lung"),
                   get_top_genes_coloc(colocfs3[i],"expression_Whole_Blood"),get_top_genes_coloc(colocfs4[i],"expression_Adipose_Subcutaneous"),
                   get_top_genes_coloc(colocfs5[i],"expression_Artery_Tibial"),get_top_genes_coloc(colocfs6[i],"expression_Heart_Left_Ventricle"),
                   get_top_genes_coloc(colocfs7[i],"expression_Stomach"))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### Compare to other methods (gene)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_uncorrelated_tissues_expression/"
runtag = "7uncorr"
simutags <- paste(1, 1:5, sep = "-")
phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".estimatedL.finemap_regions_res.RDS")
susieIfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight1.estimatedL.finemap_regions_res.RDS")
susieIfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight2.estimatedL.finemap_regions_res.RDS")
susieIfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight3.estimatedL.finemap_regions_res.RDS")
susieIfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight4.estimatedL.finemap_regions_res.RDS")
susieIfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight5.estimatedL.finemap_regions_res.RDS")
susieIfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight6.estimatedL.finemap_regions_res.RDS")
susieIfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".weight7.estimatedL.finemap_regions_res.RDS")
colocfs1 <- paste0(results_dir, runtag, "_simu",simutags, "_weight1.coloc_res.RDS")
colocfs2 <- paste0(results_dir, runtag, "_simu",simutags, "_weight2.coloc_res.RDS")
colocfs3 <- paste0(results_dir, runtag, "_simu",simutags, "_weight3.coloc_res.RDS")
colocfs4 <- paste0(results_dir, runtag, "_simu",simutags, "_weight4.coloc_res.RDS")
colocfs5 <- paste0(results_dir, runtag, "_simu",simutags, "_weight5.coloc_res.RDS")
colocfs6 <- paste0(results_dir, runtag, "_simu",simutags, "_weight6.coloc_res.RDS")
colocfs7 <- paste0(results_dir, runtag, "_simu",simutags, "_weight7.coloc_res.RDS")
zgenefs <- paste0(results_dir, runtag, "_simu",simutags, "_UKBBLD", ".z_gene.RDS")
plot_df <- data.frame()
for(i in 1:5){
  cau <- lapply(phenofs, function(x) {load(x);get_causal_id(phenores)})
  causal_genes <- grep("\\|", cau[[i]], value = TRUE)
  causal_genes <- unique(sapply(causal_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  z_gene <- readRDS(zgenefs[i])
  alpha <- 0.05
  sig_thresh <- qnorm(1-(alpha/nrow(z_gene)/2), lower=T)
  twas_genes <- z_gene$id[abs(z_gene$z)>sig_thresh]
  twas_genes <- unique(sapply(twas_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(intersect(twas_genes,causal_genes)),
                              ifcausal=T))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="TWAS",
                              count=length(twas_genes)-length(intersect(twas_genes,causal_genes)),
                              ifcausal=F))
  
  multi_cTWAS_genes <- get_top_genes_combined(susieIfs[i])
  multi_cTWAS_genes <- unique(sapply(multi_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  print(length(multi_cTWAS_genes))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="multi-cTWAS",
                              count=length(multi_cTWAS_genes)-length(intersect(multi_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  single_cTWAS_genes <- c(get_top_genes(susieIfs1[i]),get_top_genes(susieIfs2[i]),get_top_genes(susieIfs3[i]),
                          get_top_genes(susieIfs4[i]),get_top_genes(susieIfs5[i]),get_top_genes(susieIfs6[i]),get_top_genes(susieIfs7[i]))
  
  single_cTWAS_genes <- unique(sapply(single_cTWAS_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="single-cTWAS",
                              count=length(single_cTWAS_genes)-length(intersect(single_cTWAS_genes,causal_genes)),
                              ifcausal=F))
  
  coloc_genes <- c(get_top_genes_coloc(colocfs1[i],"expression_Liver"),get_top_genes_coloc(colocfs2[i],"expression_Lung"),
                   get_top_genes_coloc(colocfs3[i],"expression_Whole_Blood"),get_top_genes_coloc(colocfs4[i],"expression_Adipose_Subcutaneous"),
                   get_top_genes_coloc(colocfs5[i],"expression_Artery_Tibial"),get_top_genes_coloc(colocfs6[i],"expression_Heart_Left_Ventricle"),
                   get_top_genes_coloc(colocfs7[i],"expression_Stomach"))
  
  coloc_genes <- unique(sapply(coloc_genes, function(x){unlist(strsplit(x, "[|]"))[1]}))
  
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(intersect(coloc_genes,causal_genes)),
                              ifcausal=T))
  plot_df <- rbind(plot_df,
                   data.frame(simutag=i,
                              method="coloc",
                              count=length(coloc_genes)-length(intersect(coloc_genes,causal_genes)),
                              ifcausal=F))
}
```

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "coloc", "single-cTWAS", "multi-cTWAS"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

### PIP partition across tissues
```{r, fig.width= 8, fig.height= 4, echo=FALSE, message=FALSE, results='hide', warning=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_seven_uncorrelated_tissues_expression/"
runtag = "7uncorr"
simutags <- paste(1, 1:5, sep = "-")
tissues <- c("Liver","Lung","Whole_Blood","Adipose_Subcutaneous","Artery_Tibial","Heart_Left_Ventricle","Stomach")
fig <- pip_partition_across_tissues(results_dir, runtag, simutags, "UKBBLD", tissues, 0.8)
fig
```

