---
title: "Single weight simulation"
author: "shengqian"
date: "2024-5-6"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

This single weight simulation study is conducted to evaluate our new cTWAS software performance (parameter estimation, PIP calibration ...). Three expression weights from PredictDB are used in this study, which are Liver, Adipose and Lung. Gene PVE is 3% and SNP PVE is 30%. Gene prior is 0.9% and SNP prior is 0.025%. For each weight, I select causal genes, simulate phenotype/GWAS and perform ctwas analysis. Two types of LD between weight SNPs (calculating gene z score) are used and compared in this study. And their performance are very close because most genes only have one weight SNP.  

```{r, message=FALSE, warning=FALSE, echo=FALSE}
library(ctwas)
source("/project/xinhe/shengqian/cTWAS_simulation/help_functions.R")
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_basic_plots.R")
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_ctwas_plots.R")
source("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/analysis/simulation_help_functions.R")

plot_PIP <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".finemap_regions.res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}

plot_PIP_merged <- function(results_dir, runtag,  simutags, LD_type, ...){
   phenofs <- paste0(results_dir, runtag, "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(results_dir, runtag, "_simu",simutags, "_", LD_type, ".finemap_merged_regions.res.RDS")
   f1 <- caliPIP_plot_csindex(phenofs, susieIfs, ...) 
   return(f1)
}
```


## Simulation 1: Expression trait in liver with 3% PVE and 0.9% Prior

```{r, message=FALSE, warning=FALSE, echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_Liver_expression/"
runtag = "Liver"
simutags <- paste(1, 1:5, sep = "-")
param_GTEX_df <- load_parameters(results_dir, runtag, simutags, "GTEXLD")
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
```

### Number of causal genes detected (GTEX LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "GTEXLD", PIP_threshold = 0.8)
```

### Number of causal genes detected (UKBB LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "UKBBLD", PIP_threshold = 0.8)
```

### Estimated Prior Inclusion Probability and PVE
`````{r, fig.width= 8, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
par(mfrow = c(1, 3))
y1 <- param_GTEX_df$`prior_eQTL|Liver`
y2 <- param_UKBB_df$`prior_eQTL|Liver`
truth <- rbind(c(1,0.009),c(2,0.009))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.025),ylab="Prior inclusion", main="Prior Inclusion Probability")

y1 <- param_GTEX_df$`pve_eQTL|Liver`
y2 <- param_UKBB_df$`pve_eQTL|Liver`
truth <- rbind(c(1,0.05),c(2,0.05))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.1),ylab="PVE", main="PVE")

y1 <- param_GTEX_df$enrichment_
y2 <- param_UKBB_df$enrichment_
truth <- rbind(c(1,36),c(2,36))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,60),ylab="Enrichment", main="Enrichment")
```

### PIP Calibration Plot based on GTEX LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "GTEXLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "GTEXLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```


### PIP Calibration Plot based on UKBB LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "UKBBLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "UKBBLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```


## Simulation 2: Expression trait in adipose with 3% PVE and 0.9% Prior

```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_Adipose_expression/"
runtag = "Adipose"
simutags <- paste(1, 1:5, sep = "-")
param_GTEX_df <- load_parameters(results_dir, runtag, simutags, "GTEXLD")
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
```

### Number of causal genes detected (GTEX LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "GTEXLD", PIP_threshold = 0.8)
```

### Number of causal genes detected (UKBB LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "UKBBLD", PIP_threshold = 0.8)
```

### Estimated Prior Inclusion Probability and PVE
`````{r, fig.width= 8, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
par(mfrow = c(1, 2))
y1 <- param_GTEX_df$`prior_eQTL|Adipose`
y2 <- param_UKBB_df$`prior_eQTL|Adipose`
truth <- rbind(c(1,0.009),c(2,0.009))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.025),ylab="Prior inclusion", main="Prior Inclusion Probability")

y1 <- param_GTEX_df$`pve_eQTL|Adipose`
y2 <- param_UKBB_df$`pve_eQTL|Adipose`
truth <- rbind(c(1,0.05),c(2,0.05))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.1),ylab="PVE", main="PVE")

y1 <- param_GTEX_df$enrichment_
y2 <- param_UKBB_df$enrichment_
truth <- rbind(c(1,36),c(2,36))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,60),ylab="Enrichment", main="Enrichment")
```

### PIP Calibration Plot based on GTEX LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "GTEXLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "GTEXLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```


### PIP Calibration Plot based on UKBB LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "UKBBLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "UKBBLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```

### Simulation 3: Expression trait in lung with 3% PVE and 0.9% Prior

```{r echo=FALSE}
results_dir <- "/project/xinhe/shengqian/cTWAS_simulation/simulation_Lung_expression/"
runtag = "Lung"
simutags <- paste(1, 1:5, sep = "-")
param_GTEX_df <- load_parameters(results_dir, runtag, simutags, "GTEXLD")
param_UKBB_df <- load_parameters(results_dir, runtag, simutags, "UKBBLD")
```

### Number of causal genes detected (GTEX LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "GTEXLD", PIP_threshold = 0.8)
```

### Number of causal genes detected (UKBB LD)
```{r echo=FALSE}
check_n_detected(results_dir, runtag, simutags, "UKBBLD", PIP_threshold = 0.8)
```

### Estimated Prior Inclusion Probability and PVE
`````{r, fig.width= 8, fig.height= 4, message=FALSE, warning=FALSE, echo=FALSE}
par(mfrow = c(1, 2))
y1 <- param_GTEX_df$`prior_eQTL|Lung`
y2 <- param_UKBB_df$`prior_eQTL|Lung`
truth <- rbind(c(1,0.009),c(2,0.009))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.025),ylab="Prior inclusion", main="Prior Inclusion Probability")

y1 <- param_GTEX_df$`pve_eQTL|Lung`
y2 <- param_UKBB_df$`pve_eQTL|Lung`
truth <- rbind(c(1,0.05),c(2,0.05))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,0.1),ylab="PVE", main="PVE")

y1 <- param_GTEX_df$enrichment_
y2 <- param_UKBB_df$enrichment_
truth <- rbind(c(1,36),c(2,36))
est <- rbind(cbind(1,y1),cbind(2,y2))
plot_par_2(truth,est,xlabels = c("GTEX LD", "UKBB LD"),ylim=c(0,60),ylab="Enrichment", main="Enrichment")
```

### PIP Calibration Plot based on GTEX LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "GTEXLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "GTEXLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```


### PIP Calibration Plot based on UKBB LD (filter out cs index 0)
```{r, fig.width= 8, fig.height= 4, echo=FALSE, warning=FALSE}
f1 <- plot_PIP(results_dir, runtag, simutags, "UKBBLD", main = "no merged regions")
f2 <- plot_PIP_merged(results_dir, runtag, simutags, "UKBBLD", main = "merged regions")
gridExtra::grid.arrange(f1, f2, ncol =2)
```
